--- /usr/portage/www-client/firefox/firefox-38.4.0.ebuild	2015-12-23 13:01:02.000000000 -0800
+++ firefox-38.4.0.ebuild	2015-11-12 08:19:29.179703818 -0800
@@ -38,11 +38,11 @@
 DESCRIPTION="Firefox Web Browser"
 HOMEPAGE="http://www.mozilla.com/firefox"
 
-KEYWORDS="~alpha amd64 ~arm hppa ~ia64 ppc ppc64 x86 ~amd64-linux ~x86-linux"
+KEYWORDS="~alpha amd64 ~arm hppa ~ia64 ~ppc ppc64 x86 ~amd64-linux ~x86-linux"
 
 SLOT="0"
 LICENSE="MPL-2.0 GPL-2 LGPL-2.1"
-IUSE="bindist egl hardened +minimal pgo selinux +gmp-autoupdate test"
+IUSE="bindist egl hardened +minimal neon pgo selinux +gmp-autoupdate test 32bit 64bit"
 RESTRICT="!bindist? ( bindist )"
 
 # More URIs appended below...
@@ -55,8 +55,9 @@
 
 # Mesa 7.10 needed for WebGL + bugfixes
 RDEPEND="
-	>=dev-libs/nss-3.20.1
-	>=dev-libs/nspr-4.10.10
+	>=dev-libs/nss-3.19.2
+	>=dev-libs/nspr-4.10.8
+	32bit? ( dev-libs/libevent[abi_x86_32] )
 	selinux? ( sec-policy/selinux-mozilla )"
 
 DEPEND="${RDEPEND}
@@ -87,7 +88,7 @@
 	fi
 fi
 
-QA_PRESTRIPPED="usr/$(get_libdir)/${PN}/firefox"
+#QA_PRESTRIPPED="usr/$(get_libdir)/${PN}/firefox"
 
 BUILD_OBJ_DIR="${S}/ff"
 
@@ -184,6 +185,16 @@
 }
 
 src_configure() {
+	if use 32bit ; then
+		CONF_LIBDIR_OVERRIDE="lib32"
+	elif use 64bit ; then
+		CONF_LIBDIR_OVERRIDE="lib64"
+	else
+		true
+	fi
+
+	QA_PRESTRIPPED="usr/$(get_libdir)/${PN}/firefox"
+
 	MOZILLA_FIVE_HOME="/usr/$(get_libdir)/${PN}"
 	MEXTENSIONS="default"
 	# Google API keys (see http://www.chromium.org/developers/how-tos/api-keys)
@@ -206,6 +217,46 @@
 	# Add full relro support for hardened
 	use hardened && append-ldflags "-Wl,-z,relro,-z,now"
 
+	if use 32bit ; then
+		mozconfig_annotate '' --target=i686-pc-linux-gnu
+		mozconfig_annotate '' --host=i686-pc-linux-gnu
+	#	filter-flags -march=* -mtune=* -mcpu=*
+	#	append-cflags "-march=i686"
+		append-cflags "-m32"
+	#	append-cxxflags "-march=i686"
+		append-cxxflags "-m32"
+		append-ldflags "-m32"
+	#	export ASFLAGS="-march=i686 -m32"
+		export ASFLAGS="-m32"
+	elif use 64bit ; then
+		mozconfig_annotate '' --target=x86_64-pc-linux-gnu
+		mozconfig_annotate '' --host=x86_64-pc-linux-gnu
+	#	filter-flags -march=* -mtune=* -mcpu=*
+	#	append-cflags "-march=x86-64"
+		append-cflags "-m64"
+	#	append-cxxflags "-march=x86-64"
+		append-cxxflags "-m64"
+		append-ldflags "-m64"
+	#	export ASFLAGS="-march=x86-64 -m64"
+		export ASFLAGS="-m64"
+	fi
+
+	if use neon ; then
+		mozconfig_annotate '' --with-fpu=neon
+		mozconfig_annotate '' --with-thumb=yes
+		mozconfig_annotate '' --with-thumb-interwork=no
+	fi
+
+	if [[ ${CHOST} == armv* ]] ; then
+		mozconfig_annotate '' --with-float-abi=hard
+		mozconfig_annotate '' --enable-skia
+
+		if ! use system-libvpx ; then
+			sed -i -e "s|softfp|hard|" \
+				"${S}"/media/libvpx/moz.build
+		fi
+	fi
+
 	use egl && mozconfig_annotate 'Enable EGL as GL provider' --with-gl-provider=EGL
 
 	# Setup api key for location services
@@ -237,6 +288,14 @@
 }
 
 src_compile() {
+	if use 32bit ; then
+		CONF_LIBDIR_OVERRIDE="lib32"
+	elif use 64bit ; then
+		CONF_LIBDIR_OVERRIDE="lib64"
+	else
+		true
+	fi
+
 	if use pgo; then
 		addpredict /root
 		addpredict /etc/gconf
