diff -urp firejail-0.9.66.orig/etc/profile-a-l/eo-common.profile firejail-0.9.66/etc/profile-a-l/eo-common.profile
--- firejail-0.9.66.orig/etc/profile-a-l/eo-common.profile	2021-06-22 08:51:28.000000000 -0700
+++ firejail-0.9.66/etc/profile-a-l/eo-common.profile	2021-07-11 18:27:53.093205531 -0700
@@ -49,5 +49,5 @@ tracelog
 private-cache
 private-dev
 private-etc alternatives,dconf,fonts,gtk-3.0
-private-lib eog,eom,gdk-pixbuf-2.*,gio,girepository-1.*,gvfs,libgconf-2.so.*
+private-lib eog,eom,gcc/*/*/libgcc_s.so.*,gcc/*/*/libstdc++.so.*,gcc/*/*/libgomp.so.*,gdk-pixbuf-2.*,gio,girepository-1.*,gvfs,libgconf-2.so.*
 private-tmp
diff -urp firejail-0.9.66.orig/etc/profile-a-l/less.profile firejail-0.9.66/etc/profile-a-l/less.profile
--- firejail-0.9.66.orig/etc/profile-a-l/less.profile	2021-06-22 08:51:28.000000000 -0700
+++ firejail-0.9.66/etc/profile-a-l/less.profile	2021-07-11 18:35:37.003223776 -0700
@@ -39,7 +39,8 @@ x11 none
 # The user can have a custom coloring script configured in ${HOME}/.lessfilter.
 # Enable private-bin and private-lib if you are not using any filter.
 # private-bin less
-# private-lib
+private-lib libreadline.so.*
+#private-lib libpcre2-*.so.*
 private-cache
 private-dev
 writable-var-log
diff -urp firejail-0.9.66.orig/etc/profile-m-z/pavucontrol.profile firejail-0.9.66/etc/profile-m-z/pavucontrol.profile
--- firejail-0.9.66.orig/etc/profile-m-z/pavucontrol.profile	2021-06-22 08:51:28.000000000 -0700
+++ firejail-0.9.66/etc/profile-m-z/pavucontrol.profile	2021-07-11 18:27:53.093205531 -0700
@@ -47,7 +47,7 @@ private-bin pavucontrol
 private-cache
 private-dev
 private-etc alternatives,asound.conf,avahi,fonts,machine-id,pulse
-private-lib
+private-lib gcc/*/*/libgcc_s.so.*,gcc/*/*/libstdc++.so.*,gcc/*/*/libgomp.so.*,libltdl.so.*,libpulse-mainloop-glib.so.*,libvorbisfile.so.*
 private-tmp
 
 dbus-user none
