diff -urp firejail-0.9.64.orig/Makefile.in firejail-0.9.64/Makefile.in
--- firejail-0.9.64.orig/Makefile.in	2020-10-21 04:54:58.000000000 -0700
+++ firejail-0.9.64/Makefile.in	2020-10-26 22:27:49.919917870 -0700
@@ -144,10 +144,9 @@ ifneq ($(HAVE_MAN),no)
 	install -m 0755 -d $(DESTDIR)$(mandir)/man1 $(DESTDIR)$(mandir)/man5
 	for man in $(MANPAGES); do \
 		rm -f $$man.gz; \
-		gzip -9n $$man; \
 		case "$$man" in \
-			*.1) install -m 0644 $$man.gz $(DESTDIR)$(mandir)/man1/; ;; \
-			*.5) install -m 0644 $$man.gz $(DESTDIR)$(mandir)/man5/; ;; \
+			*.1) install -m 0644 $$man $(DESTDIR)$(mandir)/man1/; ;; \
+			*.5) install -m 0644 $$man $(DESTDIR)$(mandir)/man5/; ;; \
 		esac; \
 	done
 	rm -f $(MANPAGES) $(MANPAGES:%=%.gz)
