--- a/torch/headeronly/util/Half.h.orig	2025-10-08 18:09:57.000000000 -0700
+++ b/torch/headeronly/util/Half.h	2025-10-31 13:31:16.601955458 -0700
@@ -189,7 +189,7 @@ C10_HOST_DEVICE inline float fp16_ieee_t
 #if defined(_MSC_VER) && defined(__clang__)
   __builtin_memcpy(&exp_scale_val, &scale_bits, sizeof(exp_scale_val));
 #else
-  std::memcpy(&exp_scale_val, &scale_bits, sizeof(exp_scale_val));
+  memcpy(&exp_scale_val, &scale_bits, sizeof(exp_scale_val));
 #endif
 
   const float exp_scale = exp_scale_val;
@@ -266,8 +266,8 @@ inline uint16_t fp16_ieee_from_fp32_valu
   constexpr uint32_t scale_to_inf_bits = (uint32_t)239 << 23;
   constexpr uint32_t scale_to_zero_bits = (uint32_t)17 << 23;
   float scale_to_inf_val = 0, scale_to_zero_val = 0;
-  std::memcpy(&scale_to_inf_val, &scale_to_inf_bits, sizeof(scale_to_inf_val));
-  std::memcpy(
+  memcpy(&scale_to_inf_val, &scale_to_inf_bits, sizeof(scale_to_inf_val));
+  memcpy(
       &scale_to_zero_val, &scale_to_zero_bits, sizeof(scale_to_zero_val));
   const float scale_to_inf = scale_to_inf_val;
   const float scale_to_zero = scale_to_zero_val;
