diff -urp mesa-13.0.1.orig/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp mesa-13.0.1/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp
--- mesa-13.0.1.orig/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp	2017-01-24 12:01:20.721510843 -0800
+++ mesa-13.0.1/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp	2017-01-24 12:38:46.799724484 -0800
@@ -95,7 +95,9 @@ clover::llvm::parse_module_library(const
    auto mod = ::llvm::parseBitcodeFile(::llvm::MemoryBufferRef(
                                         as_string(m.secs[0].data), " "), ctx);
    if (!mod)
-      fail(r_log, error(CL_INVALID_PROGRAM), mod.getError().message());
+   {
+      fail(r_log, error(CL_INVALID_PROGRAM), "Malformed bitcode was found.");
+   }
 
    return std::unique_ptr<::llvm::Module>(std::move(*mod));
 }
