diff -urp mesa-13.0.1.orig/src/amd/common/ac_nir_to_llvm.c mesa-13.0.1/src/amd/common/ac_nir_to_llvm.c
--- mesa-13.0.1.orig/src/amd/common/ac_nir_to_llvm.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/amd/common/ac_nir_to_llvm.c	2017-01-24 11:19:09.182865417 -0800
@@ -227,11 +227,11 @@ create_llvm_function(LLVMContextRef ctx,
 		LLVMValueRef P = LLVMGetParam(main_function, i);
 
 		if (i < array_params) {
-			LLVMAddAttribute(P, LLVMByValAttribute);
+			lp_add_function_attr(ctx->main_fn, i + 1, "byval", 5);
 			ac_add_attr_dereferenceable(P, UINT64_MAX);
 		}
 		else
-			LLVMAddAttribute(P, LLVMInRegAttribute);
+			lp_add_function_attr(ctx->main_fn, i + 1, "inreg", 5);
 	}
 
 	if (unsafe_math) {
@@ -2415,7 +2415,8 @@ static LLVMValueRef visit_image_load(str
 				    intrinsic_name, sizeof(intrinsic_name));
 
 		res = emit_llvm_intrinsic(ctx, intrinsic_name, ctx->v4f32,
-					params, 7, LLVMReadOnlyAttribute);
+					params, 7, "readonly"),
+
 	}
 	return to_integer(ctx, res);
 }
diff -urp mesa-13.0.1.orig/src/gallium/auxiliary/draw/draw_llvm.c mesa-13.0.1/src/gallium/auxiliary/draw/draw_llvm.c
--- mesa-13.0.1.orig/src/gallium/auxiliary/draw/draw_llvm.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/auxiliary/draw/draw_llvm.c	2017-01-24 11:19:09.183865456 -0800
@@ -1574,8 +1574,7 @@ draw_llvm_generate(struct draw_llvm *llv
    LLVMSetFunctionCallConv(variant_func, LLVMCCallConv);
    for (i = 0; i < num_arg_types; ++i)
       if (LLVMGetTypeKind(arg_types[i]) == LLVMPointerTypeKind)
-         LLVMAddAttribute(LLVMGetParam(variant_func, i),
-                          LLVMNoAliasAttribute);
+         lp_add_function_attr(variant_func, i + 1, "noalias", 7);
 
    context_ptr               = LLVMGetParam(variant_func, 0);
    io_ptr                    = LLVMGetParam(variant_func, 1);
@@ -2189,8 +2188,7 @@ draw_gs_llvm_generate(struct draw_llvm *
 
    for (i = 0; i < ARRAY_SIZE(arg_types); ++i)
       if (LLVMGetTypeKind(arg_types[i]) == LLVMPointerTypeKind)
-         LLVMAddAttribute(LLVMGetParam(variant_func, i),
-                          LLVMNoAliasAttribute);
+         lp_add_function_attr(variant_func, i + 1, "noalias", 7);
 
    context_ptr               = LLVMGetParam(variant_func, 0);
    input_array               = LLVMGetParam(variant_func, 1);
diff -urp mesa-13.0.1.orig/src/gallium/auxiliary/gallivm/lp_bld_intr.c mesa-13.0.1/src/gallium/auxiliary/gallivm/lp_bld_intr.c
--- mesa-13.0.1.orig/src/gallium/auxiliary/gallivm/lp_bld_intr.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/auxiliary/gallivm/lp_bld_intr.c	2017-01-24 11:19:09.184865494 -0800
@@ -120,13 +120,57 @@ lp_declare_intrinsic(LLVMModuleRef modul
 }
 
 
+#if HAVE_LLVM < 0x0400
+static LLVMAttribute str_to_attr(const char *attr_name, unsigned attr_len)
+{
+   if (!strncmp("alwaysinline", attr_name, attr_len)) {
+      return LLVMAlwaysInlineAttribute;
+   } else if (!strncmp("byval", attr_name, attr_len)) {
+      return LLVMByValAttribute;
+   } else if (!strncmp("inreg", attr_name, attr_len)) {
+      return LLVMInRegAttribute;
+   } else if (!strncmp("noalias", attr_name, attr_len)) {
+      return LLVMNoAlliasAttribute;
+   } else if (!strncmp("readnone", attr_name, attr_len)) {
+      return LLVMReadNoneAttribute;
+   } else if (!strncmp("readonly", attr_name, attr_len)) {
+      return LLVMReadOnlyAttribute;
+   } else {
+      _debug_printf("Unhandled function attribute: %s\n", attr_name);
+      return 0;
+   }
+}
+#endif
+
+void
+lp_add_function_attr(LLVMValueRef function,
+                     int attr_idx,
+                     const char *attr_name,
+                     unsigned attr_len)
+{
+
+#if HAVE_LLVM < 0x0400
+   LLVMAttribute attr = str_to_attr(attr_name, attr_len);
+   if (attr_idx == -1) {
+      LLVMAddFunctionAttr(function, attr);
+   } else {
+      LLVMAddAttribute(LLVMGetParam(function, attr_idx), attr);
+   }
+#else
+   LLVMContextRef context = LLVMGetModuleContext(LLVMGetGlobalParent(function));
+   unsigned kind_id = LLVMGetEnumAttributeKindForName(attr_name, attr_len);
+   LLVMAttributeRef attr = LLVMCreateEnumAttribute(context, kind_id, 0);
+   LLVMAddAttributeAtIndex(function, attr_idx, attr);
+#endif
+}
+
 LLVMValueRef
 lp_build_intrinsic(LLVMBuilderRef builder,
                    const char *name,
                    LLVMTypeRef ret_type,
                    LLVMValueRef *args,
                    unsigned num_args,
-                   LLVMAttribute attr)
+                   const char *attr_str)
 {
    LLVMModuleRef module = LLVMGetGlobalParent(LLVMGetBasicBlockParent(LLVMGetInsertBlock(builder)));
    LLVMValueRef function;
@@ -145,10 +189,14 @@ lp_build_intrinsic(LLVMBuilderRef builde
 
       function = lp_declare_intrinsic(module, name, ret_type, arg_types, num_args);
 
+      if (attr_str) {
+         lp_add_function_attr(function, -1, attr_str, sizeof(attr_str));
+      }
+
       /* NoUnwind indicates that the intrinsic never raises a C++ exception.
        * Set it for all intrinsics.
        */
-      LLVMAddFunctionAttr(function, attr | LLVMNoUnwindAttribute);
+      lp_add_function_attr(function, -1, "nounwind", 8);
 
       if (gallivm_debug & GALLIVM_DEBUG_IR) {
          lp_debug_dump_value(function);
diff -urp mesa-13.0.1.orig/src/gallium/auxiliary/gallivm/lp_bld_intr.h mesa-13.0.1/src/gallium/auxiliary/gallivm/lp_bld_intr.h
--- mesa-13.0.1.orig/src/gallium/auxiliary/gallivm/lp_bld_intr.h	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/auxiliary/gallivm/lp_bld_intr.h	2017-01-24 11:19:09.184865494 -0800
@@ -60,13 +60,24 @@ lp_declare_intrinsic(LLVMModuleRef modul
                      LLVMTypeRef *arg_types,
                      unsigned num_args);
 
+void
+lp_remove_attr(LLVMValueRef value,
+               const char *attr_name,
+               unsigned attr_len);
+
+void
+lp_add_function_attr(LLVMValueRef function,
+                     int attr_idx,
+                     const char *attr_name,
+                     unsigned attr_len);
+
 LLVMValueRef
 lp_build_intrinsic(LLVMBuilderRef builder,
                    const char *name,
                    LLVMTypeRef ret_type,
                    LLVMValueRef *args,
                    unsigned num_args,
-                   LLVMAttribute attr);
+                   const char *attr);
 
 
 LLVMValueRef
diff -urp mesa-13.0.1.orig/src/gallium/auxiliary/gallivm/lp_bld_sample_soa.c mesa-13.0.1/src/gallium/auxiliary/gallivm/lp_bld_sample_soa.c
--- mesa-13.0.1.orig/src/gallium/auxiliary/gallivm/lp_bld_sample_soa.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/auxiliary/gallivm/lp_bld_sample_soa.c	2017-01-24 11:19:09.186865571 -0800
@@ -60,6 +60,7 @@
 #include "lp_bld_struct.h"
 #include "lp_bld_quad.h"
 #include "lp_bld_pack.h"
+#include "lp_bld_intr.h"
 
 
 /**
@@ -3316,7 +3317,8 @@ lp_build_sample_soa_func(struct gallivm_
 
       for (i = 0; i < num_param; ++i) {
          if(LLVMGetTypeKind(arg_types[i]) == LLVMPointerTypeKind) {
-            LLVMAddAttribute(LLVMGetParam(function, i), LLVMNoAliasAttribute);
+
+            lp_add_function_attr(function, i + 1, "noalias", 7);
          }
       }
 
diff -urp mesa-13.0.1.orig/src/gallium/drivers/llvmpipe/lp_state_fs.c mesa-13.0.1/src/gallium/drivers/llvmpipe/lp_state_fs.c
--- mesa-13.0.1.orig/src/gallium/drivers/llvmpipe/lp_state_fs.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/drivers/llvmpipe/lp_state_fs.c	2017-01-24 11:19:22.364371385 -0800
@@ -2296,7 +2296,7 @@ generate_fragment(struct llvmpipe_contex
     */
    for(i = 0; i < ARRAY_SIZE(arg_types); ++i)
       if(LLVMGetTypeKind(arg_types[i]) == LLVMPointerTypeKind)
-         LLVMAddAttribute(LLVMGetParam(function, i), LLVMNoAliasAttribute);
+         lp_add_function_attr(function, i + 1, "noalias", 7);
 
    context_ptr  = LLVMGetParam(function, 0);
    x            = LLVMGetParam(function, 1);
diff -urp mesa-13.0.1.orig/src/gallium/drivers/llvmpipe/lp_state_setup.c mesa-13.0.1/src/gallium/drivers/llvmpipe/lp_state_setup.c
--- mesa-13.0.1.orig/src/gallium/drivers/llvmpipe/lp_state_setup.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/drivers/llvmpipe/lp_state_setup.c	2017-01-24 11:19:22.364371385 -0800
@@ -624,8 +624,7 @@ set_noalias(LLVMBuilderRef builder,
    int i;
    for(i = 0; i < nr_args; ++i)
       if(LLVMGetTypeKind(arg_types[i]) == LLVMPointerTypeKind)
-         LLVMAddAttribute(LLVMGetParam(function, i),
-            LLVMNoAliasAttribute);
+         lp_add_function_attr(function, i + 1, "noalias", 7);
 }
 
 static void
diff -urp mesa-13.0.1.orig/src/gallium/drivers/radeonsi/si_shader.c mesa-13.0.1/src/gallium/drivers/radeonsi/si_shader.c
--- mesa-13.0.1.orig/src/gallium/drivers/radeonsi/si_shader.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/drivers/radeonsi/si_shader.c	2017-01-24 11:19:09.190865724 -0800
@@ -413,7 +413,7 @@ static void declare_input_vs(
 	args[2] = buffer_index;
 	input = lp_build_intrinsic(gallivm->builder,
 		"llvm.SI.vs.load.input", ctx->v4f32, args, 3,
-		LLVMReadNoneAttribute);
+		"readnone");
 
 	/* Break up the vec4 into individual components */
 	for (chan = 0; chan < 4; chan++) {
@@ -808,7 +808,7 @@ static LLVMValueRef build_buffer_load(st
 		         type_names[func]);
 
 		return lp_build_intrinsic(gallivm->builder, name, types[func], args,
-		                          ARRAY_SIZE(args), LLVMReadOnlyAttribute);
+		                          ARRAY_SIZE(args), "readonly");
 	} else {
 		LLVMValueRef args[] = {
 			LLVMBuildBitCast(gallivm->builder, rsrc, ctx->v16i8, ""),
@@ -839,7 +839,7 @@ static LLVMValueRef build_buffer_load(st
 		         type_names[func], arg_type);
 
 		return lp_build_intrinsic(gallivm->builder, name, types[func], args,
-		                          ARRAY_SIZE(args), LLVMReadOnlyAttribute);
+		                          ARRAY_SIZE(args), "readonly");
 	}
 }
 
@@ -1126,14 +1126,14 @@ static LLVMValueRef fetch_input_gs(
 	value = lp_build_intrinsic(gallivm->builder,
 				   "llvm.SI.buffer.load.dword.i32.i32",
 				   ctx->i32, args, 9,
-				   LLVMReadOnlyAttribute);
+				   "readonly");
 	if (tgsi_type_is_64bit(type)) {
 		LLVMValueRef value2;
 		args[2] = lp_build_const_int32(gallivm, (param * 4 + swizzle + 1) * 256);
 		value2 = lp_build_intrinsic(gallivm->builder,
 					    "llvm.SI.buffer.load.dword.i32.i32",
 					    ctx->i32, args, 9,
-					    LLVMReadOnlyAttribute);
+					    "readonly");
 		return si_llvm_emit_fetch_64bit(bld_base, type,
 						value, value2);
 	}
@@ -1285,12 +1285,12 @@ static void interp_fs_input(struct si_sh
 			args[1] = attr_number;
 			front = lp_build_intrinsic(gallivm->builder, intr_name,
 						ctx->f32, args, args[3] ? 4 : 3,
-						LLVMReadNoneAttribute);
+						"readnone");
 
 			args[1] = back_attr_number;
 			back = lp_build_intrinsic(gallivm->builder, intr_name,
 					       ctx->f32, args, args[3] ? 4 : 3,
-					       LLVMReadNoneAttribute);
+					       "readnone");
 
 			result[chan] = LLVMBuildSelect(gallivm->builder,
 						is_face_positive,
@@ -1307,7 +1307,7 @@ static void interp_fs_input(struct si_sh
 		args[3] = interp_param;
 		result[0] = lp_build_intrinsic(gallivm->builder, intr_name,
 					ctx->f32, args, args[3] ? 4 : 3,
-					LLVMReadNoneAttribute);
+					"readnone");
 		result[1] =
 		result[2] = lp_build_const_float(gallivm, 0.0f);
 		result[3] = lp_build_const_float(gallivm, 1.0f);
@@ -1322,7 +1322,7 @@ static void interp_fs_input(struct si_sh
 			args[3] = interp_param;
 			result[chan] = lp_build_intrinsic(gallivm->builder, intr_name,
 						ctx->f32, args, args[3] ? 4 : 3,
-						LLVMReadNoneAttribute);
+						"readnone");
 		}
 	}
 }
@@ -1463,18 +1463,18 @@ static LLVMValueRef get_thread_id(struct
 
 	if (HAVE_LLVM < 0x0308) {
 		tid = lp_build_intrinsic(gallivm->builder, "llvm.SI.tid",
-				ctx->i32,   NULL, 0, LLVMReadNoneAttribute);
+				ctx->i32,   NULL, 0, "readnone");
 	} else {
 		LLVMValueRef tid_args[2];
 		tid_args[0] = lp_build_const_int32(gallivm, 0xffffffff);
 		tid_args[1] = lp_build_const_int32(gallivm, 0);
 		tid_args[1] = lp_build_intrinsic(gallivm->builder,
 					"llvm.amdgcn.mbcnt.lo", ctx->i32,
-					tid_args, 2, LLVMReadNoneAttribute);
+					tid_args, 2, "readnone");
 
 		tid = lp_build_intrinsic(gallivm->builder,
 					"llvm.amdgcn.mbcnt.hi", ctx->i32,
-					tid_args, 2, LLVMReadNoneAttribute);
+					tid_args, 2, "readnone");
 	}
 	set_range_metadata(ctx, tid, 0, 64);
 	return tid;
@@ -1491,7 +1491,7 @@ static LLVMValueRef buffer_load_const(st
 	LLVMValueRef args[2] = {resource, offset};
 
 	return lp_build_intrinsic(builder, "llvm.SI.load.const", ctx->f32, args, 2,
-			       LLVMReadNoneAttribute);
+			       "readnone");
 }
 
 static LLVMValueRef load_sample_position(struct si_shader_context *radeon_bld, LLVMValueRef sample_id)
@@ -1729,7 +1729,7 @@ static void declare_system_value(
 		value = lp_build_intrinsic(gallivm->builder,
 					   "llvm.amdgcn.ps.live",
 					   ctx->i1, NULL, 0,
-					   LLVMReadNoneAttribute);
+					   "readnone");
 		value = LLVMBuildNot(gallivm->builder, value, "");
 		value = LLVMBuildSExt(gallivm->builder, value, ctx->i32, "");
 		break;
@@ -1942,7 +1942,7 @@ static void si_llvm_init_export_args(str
 			packed = lp_build_intrinsic(base->gallivm->builder,
 						    "llvm.SI.packf16",
 						    ctx->i32, pack_args, 2,
-						    LLVMReadNoneAttribute);
+						    "readnone");
 			args[chan + 5] =
 				LLVMBuildBitCast(base->gallivm->builder,
 						 packed, ctx->f32, "");
@@ -2087,7 +2087,7 @@ static LLVMValueRef si_scale_alpha_by_sa
 
 	coverage = lp_build_intrinsic(gallivm->builder, "llvm.ctpop.i32",
 				   ctx->i32,
-				   &coverage, 1, LLVMReadNoneAttribute);
+				   &coverage, 1, "readnone");
 
 	coverage = LLVMBuildUIToFP(gallivm->builder, coverage,
 				   ctx->f32, "");
@@ -3634,7 +3634,7 @@ static void load_emit_buffer(struct si_s
 	emit_data->output[emit_data->chan] = lp_build_intrinsic(
 			builder, intrinsic_name, dst_type,
 			emit_data->args, emit_data->arg_count,
-			LLVMReadOnlyAttribute);
+			"readonly");
 }
 
 static LLVMValueRef get_memory_ptr(struct si_shader_context *ctx,
@@ -3739,7 +3739,7 @@ static void load_emit(
 			lp_build_intrinsic(
 				builder, "llvm.amdgcn.buffer.load.format.v4f32", emit_data->dst_type,
 				emit_data->args, emit_data->arg_count,
-				LLVMReadOnlyAttribute);
+				"readonly");
 	} else {
 		get_image_intr_name("llvm.amdgcn.image.load",
 				emit_data->dst_type,		/* vdata */
@@ -3751,7 +3751,7 @@ static void load_emit(
 			lp_build_intrinsic(
 				builder, intrinsic_name, emit_data->dst_type,
 				emit_data->args, emit_data->arg_count,
-				LLVMReadOnlyAttribute);
+				"readonly");
 	}
 }
 
@@ -4179,7 +4179,7 @@ static void resq_emit(
 		out = lp_build_intrinsic(
 			builder, "llvm.SI.getresinfo.i32", emit_data->dst_type,
 			emit_data->args, emit_data->arg_count,
-			LLVMReadNoneAttribute);
+			"readnone");
 
 		/* Divide the number of layers by 6 to get the number of cubes. */
 		if (inst->Memory.Texture == TGSI_TEXTURE_CUBE_ARRAY) {
@@ -4413,7 +4413,7 @@ static void txq_emit(const struct lp_bui
 	emit_data->output[emit_data->chan] = lp_build_intrinsic(
 		base->gallivm->builder, "llvm.SI.getresinfo.i32",
 		emit_data->dst_type, emit_data->args, emit_data->arg_count,
-		LLVMReadNoneAttribute);
+		"readnone");
 
 	/* Divide the number of layers by 6 to get the number of cubes. */
 	if (target == TGSI_TEXTURE_CUBE_ARRAY ||
@@ -4831,7 +4831,7 @@ static void si_lower_gather4_integer(str
 	emit_data->output[emit_data->chan] =
 		lp_build_intrinsic(builder, intr_name, emit_data->dst_type,
 				   emit_data->args, emit_data->arg_count,
-				   LLVMReadNoneAttribute);
+				   "readnone");
 }
 
 static void build_tex_intrinsic(const struct lp_build_tgsi_action *action,
@@ -4855,7 +4855,7 @@ static void build_tex_intrinsic(const st
 			base->gallivm->builder,
 			"llvm.SI.vs.load.input", emit_data->dst_type,
 			emit_data->args, emit_data->arg_count,
-			LLVMReadNoneAttribute);
+			"readnone");
 		return;
 	}
 
@@ -4932,7 +4932,7 @@ static void build_tex_intrinsic(const st
 	emit_data->output[emit_data->chan] = lp_build_intrinsic(
 		base->gallivm->builder, intr_name, emit_data->dst_type,
 		emit_data->args, emit_data->arg_count,
-		LLVMReadNoneAttribute);
+		"readnone");
 }
 
 static void si_llvm_emit_txqs(
@@ -5030,13 +5030,13 @@ static void si_llvm_emit_ddxy(
 		args[1] = val;
 		tl = lp_build_intrinsic(gallivm->builder,
 					"llvm.amdgcn.ds.bpermute", ctx->i32,
-					args, 2, LLVMReadNoneAttribute);
+					args, 2, "readnone");
 
 		args[0] = LLVMBuildMul(gallivm->builder, trbl_tid,
 				       lp_build_const_int32(gallivm, 4), "");
 		trbl = lp_build_intrinsic(gallivm->builder,
 					  "llvm.amdgcn.ds.bpermute", ctx->i32,
-					  args, 2, LLVMReadNoneAttribute);
+					  args, 2, "readnone");
 	} else {
 		LLVMValueRef store_ptr, load_ptr0, load_ptr1;
 
@@ -5219,7 +5219,7 @@ static void build_interp_intrinsic(const
 		emit_data->output[chan] =
 			lp_build_intrinsic(gallivm->builder, intr_name,
 					   ctx->f32, args, args[3] ? 4 : 3,
-					   LLVMReadNoneAttribute);
+					   "readnone");
 	}
 }
 
@@ -5387,10 +5387,10 @@ static void si_create_function(struct si
 		 * SGPR spilling significantly.
 		 */
 		if (LLVMGetTypeKind(LLVMTypeOf(P)) == LLVMPointerTypeKind) {
-			LLVMAddAttribute(P, LLVMByValAttribute);
+			lp_add_function_attr(ctx->main_fn, i + 1, "byval", 5);
 			lp_add_attr_dereferenceable(P, UINT64_MAX);
 		} else
-			LLVMAddAttribute(P, LLVMInRegAttribute);
+			lp_add_function_attr(ctx->main_fn, i + 1, "inreg", 5);
 	}
 
 	if (ctx->screen->b.debug_flags & DBG_UNSAFE_MATH) {
@@ -6332,7 +6332,7 @@ static int si_generate_gs_copy_shader(st
 						 lp_build_intrinsic(gallivm->builder,
 								 "llvm.SI.buffer.load.dword.i32.i32",
 								 ctx->i32, args, 9,
-								 LLVMReadOnlyAttribute),
+								 "readonly"),
 						 ctx->f32, "");
 		}
 	}
diff -urp mesa-13.0.1.orig/src/gallium/drivers/radeonsi/si_shader_tgsi_alu.c mesa-13.0.1/src/gallium/drivers/radeonsi/si_shader_tgsi_alu.c
--- mesa-13.0.1.orig/src/gallium/drivers/radeonsi/si_shader_tgsi_alu.c	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/drivers/radeonsi/si_shader_tgsi_alu.c	2017-01-24 11:19:09.191865762 -0800
@@ -399,7 +399,7 @@ static void emit_frac(const struct lp_bu
 
 	LLVMValueRef floor = lp_build_intrinsic(builder, intr, emit_data->dst_type,
 						&emit_data->args[0], 1,
-						LLVMReadNoneAttribute);
+						"readnone");
 	emit_data->output[emit_data->chan] = LLVMBuildFSub(builder,
 			emit_data->args[0], floor, "");
 }
@@ -449,7 +449,7 @@ build_tgsi_intrinsic_nomem(const struct
 	emit_data->output[emit_data->chan] =
 		lp_build_intrinsic(base->gallivm->builder, action->intr_name,
 				   emit_data->dst_type, emit_data->args,
-				   emit_data->arg_count, LLVMReadNoneAttribute);
+				   emit_data->arg_count, "readnone");
 }
 
 static void emit_bfi(const struct lp_build_tgsi_action *action,
@@ -507,7 +507,7 @@ static void emit_bfe(const struct lp_bui
 
 	bfe_sm5 = lp_build_intrinsic(builder, action->intr_name,
 				     emit_data->dst_type, emit_data->args,
-				     emit_data->arg_count, LLVMReadNoneAttribute);
+				     emit_data->arg_count, "readnone");
 
 	/* Correct for GLSL semantics. */
 	cond = LLVMBuildICmp(builder, LLVMIntUGE, emit_data->args[2],
@@ -539,7 +539,7 @@ static void emit_lsb(const struct lp_bui
 	LLVMValueRef lsb =
 		lp_build_intrinsic(gallivm->builder, "llvm.cttz.i32",
 				emit_data->dst_type, args, ARRAY_SIZE(args),
-				LLVMReadNoneAttribute);
+				"readnone");
 
 	/* TODO: We need an intrinsic to skip this conditional. */
 	/* Check for zero: */
@@ -566,7 +566,7 @@ static void emit_umsb(const struct lp_bu
 	LLVMValueRef msb =
 		lp_build_intrinsic(builder, "llvm.ctlz.i32",
 				emit_data->dst_type, args, ARRAY_SIZE(args),
-				LLVMReadNoneAttribute);
+				"readnone");
 
 	/* The HW returns the last bit index from MSB, but TGSI wants
 	 * the index from LSB. Invert it by doing "31 - msb". */
@@ -593,7 +593,7 @@ static void emit_imsb(const struct lp_bu
 	LLVMValueRef msb =
 		lp_build_intrinsic(builder, "llvm.AMDGPU.flbit.i32",
 				emit_data->dst_type, &arg, 1,
-				LLVMReadNoneAttribute);
+				"readnone");
 
 	/* The HW returns the last bit index from MSB, but TGSI wants
 	 * the index from LSB. Invert it by doing "31 - msb". */
@@ -917,13 +917,13 @@ static LLVMValueRef build_cube_intrinsic
 		LLVMValueRef out[4];
 
 		out[0] = lp_build_intrinsic(gallivm->builder, "llvm.amdgcn.cubetc",
-					    f32, in, 3, LLVMReadNoneAttribute);
+					    f32, in, 3, "readnone");
 		out[1] = lp_build_intrinsic(gallivm->builder, "llvm.amdgcn.cubesc",
-					    f32, in, 3, LLVMReadNoneAttribute);
+					    f32, in, 3, "readnone");
 		out[2] = lp_build_intrinsic(gallivm->builder, "llvm.amdgcn.cubema",
-					    f32, in, 3, LLVMReadNoneAttribute);
+					    f32, in, 3, "readnone");
 		out[3] = lp_build_intrinsic(gallivm->builder, "llvm.amdgcn.cubeid",
-					    f32, in, 3, LLVMReadNoneAttribute);
+					    f32, in, 3, "readnone");
 
 		return lp_build_gather_values(gallivm, out, 4);
 	} else {
@@ -937,7 +937,7 @@ static LLVMValueRef build_cube_intrinsic
 
 		return lp_build_intrinsic(gallivm->builder, "llvm.AMDGPU.cube",
 					  LLVMTypeOf(vec), &vec, 1,
-					  LLVMReadNoneAttribute);
+					  "readnone");
 	}
 }
 
@@ -959,7 +959,7 @@ static void si_llvm_cube_to_2d_coords(st
 						    lp_build_const_int32(gallivm, i), "");
 
 	coords[2] = lp_build_intrinsic(builder, "llvm.fabs.f32",
-			type, &coords[2], 1, LLVMReadNoneAttribute);
+			type, &coords[2], 1, "readnone");
 	coords[2] = lp_build_emit_llvm_unary(bld_base, TGSI_OPCODE_RCP, coords[2]);
 
 	mad_args[1] = coords[2];
