diff -urp mesa-13.0.1.orig/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp mesa-13.0.1/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp
--- mesa-13.0.1.orig/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp	2016-11-10 14:06:40.000000000 -0800
+++ mesa-13.0.1/src/gallium/state_trackers/clover/llvm/codegen/bitcode.cpp	2017-01-24 11:32:49.789470318 -0800
@@ -37,7 +37,9 @@
 #include "util/algorithm.hpp"
 
 #include <map>
-#include <llvm/Bitcode/ReaderWriter.h>
+#include <llvm/Bitcode/BitcodeWriter.h>
+#include <llvm/Bitcode/BitcodeReader.h>
+#include <llvm/Bitcode/BitstreamReader.h>
 #include <llvm/Support/raw_ostream.h>
 
 using namespace clover;
