--- CMakeLists.txt.orig	2017-02-21 23:27:51.307044183 -0800
+++ CMakeLists.txt	2017-02-21 23:29:38.278139472 -0800
@@ -184,6 +184,10 @@ if(NOT BUILD_CORE AND NOT BUILD_EXTRACTO
   message(FATAL_ERROR "You must select something to build!")
 endif()
 
+if(NOT ${CMAKE_VERSION} VERSION_LESS 3.0)
+    cmake_policy(SET CMP0026 OLD)
+endif()
+
 # Generate revision-extractor
 set(GENREV_SRC
   src/tools/genrevision/genrevision.cpp
