diff -urp work.orig/dep/include/g3dlite/G3D/Array.h work/dep/include/g3dlite/G3D/Array.h
--- work.orig/dep/include/g3dlite/G3D/Array.h	2017-02-23 18:52:49.281476082 -0800
+++ work/dep/include/g3dlite/G3D/Array.h	2017-02-23 19:24:49.524253248 -0800
@@ -18,6 +18,7 @@
 #include "G3D/debug.h"
 #include "G3D/System.h"
 #include "G3D/MemoryManager.h"
+#include "G3D/debugAssert.h"
 #ifdef G3D_DEBUG
 //   For formatting error messages
 #    include "G3D/format.h"
diff -urp work.orig/src/game/movement/spline.cpp work/src/game/movement/spline.cpp
--- work.orig/src/game/movement/spline.cpp	2017-02-23 18:52:49.970503060 -0800
+++ work/src/game/movement/spline.cpp	2017-02-23 19:15:14.060095058 -0800
@@ -19,6 +19,7 @@
 #include "spline.h"
 #include <sstream>
 #include <G3D/Matrix4.h>
+#include <G3D/Vector3.h>
 
 namespace Movement
 {
