--- src/game/Object/Player.h.orig	2015-06-21 14:51:34.711041519 -0700
+++ src/game/Object/Player.h	2015-06-21 14:51:48.294558741 -0700
@@ -1933,6 +1933,7 @@
 
         static Team TeamForRace(uint8 race);
         Team GetTeam() const { return m_team; }
+	TeamId GetTeamId() const { return m_team == ALLIANCE ? TEAM_ALLIANCE : TEAM_HORDE; }
         static uint32 getFactionForRace(uint8 race);
         void setFactionForRace(uint8 race);
 
