--- src/game/WorldHandlers/GridNotifiers.h.orig	2015-06-22 03:24:43.452016150 -0700
+++ src/game/WorldHandlers/GridNotifiers.h	2015-06-22 03:30:44.501903600 -0700
@@ -222,6 +222,25 @@
     };
 
     template<class Check>
+    struct MANGOS_DLL_DECL WorldObjectLastSearcher
+    {
+        uint32 i_phaseMask;
+        WorldObject*& i_object;
+        Check& i_check;
+
+        WorldObjectLastSearcher(WorldObject* & result, Check& check)
+        : i_phaseMask(check.GetFocusObject().GetPhaseMask()), i_object(result), i_check(check) {}
+        
+        void Visit(PlayerMapType& m);
+        void Visit(CreatureMapType& m);
+        void Visit(CorpseMapType& m);
+        void Visit(GameObjectMapType& m);
+        void Visit(DynamicObjectMapType& m);
+
+        template<class NOT_INTERESTED> void Visit(GridRefManager<NOT_INTERESTED>&) {}
+    };
+
+    template<class Check>
     struct MANGOS_DLL_DECL WorldObjectListSearcher
     {
         uint32 i_phaseMask;
