--- src/game/Object/ObjectMgr.cpp.orig	2015-06-21 17:58:02.178699657 -0700
+++ src/game/Object/ObjectMgr.cpp	2015-06-21 17:58:53.603651274 -0700
@@ -9151,7 +9151,7 @@
         uint32 ExtendedCost = fields[4].GetUInt32();
         uint16 conditionId  = fields[5].GetUInt16();
 
-        if (!IsVendorItemValid(isTemplates, tableName, entry, item_id, maxcount, incrtime, ExtendedCost, conditionId, NULL, &skip_vendors))
+        if (!IsVendorItemValid(isTemplates, tableName, entry, item_id, maxcount, incrtime, ExtendedCost, conditionId, type, NULL, &skip_vendors))
             continue;
 
         VendorItemData& vList = vendorList[entry];
