--- AP/client/ap_client_main.cpp.orig	2015-11-30 02:23:05.220297813 -0800
+++ AP/client/ap_client_main.cpp	2015-11-30 02:28:53.421712465 -0800
@@ -639,7 +639,7 @@
 #if !(_GNU_SOURCE || __APPLE__)
 	AquireExecutionSlot::BOINCs_device=selected_device;
 #elif (_GNU_SOURCE || __APPLE__)
-	BOINCs_dev=selected_device;
+	AquireExecutionSlot::BOINCs_dev=selected_device;
 #endif
 #endif
       fprintf(stderr, "Running on device number: %d\n", selected_device);
@@ -968,8 +968,8 @@
 	sprintf(devname_open,"<device%d>",AquireExecutionSlot::BOINCs_device);
 	sprintf(devname_close,"</device%d>",AquireExecutionSlot::BOINCs_device);
 #elif (_GNU_SOURCE || __APPLE__)
-  sprintf(devname_open,"<device%d>",BOINCs_dev);
-  sprintf(devname_close,"</device%d>",BOINCs_dev);
+  sprintf(devname_open,"<device%d>",AquireExecutionSlot::BOINCs_dev);
+  sprintf(devname_close,"</device%d>",AquireExecutionSlot::BOINCs_dev);
 #endif
     while (fgets(buf, sizeof(buf), config_file)){
 		if (xml_match_tag(buf, devname_open)){
