--- setiathome-7.08.orig/AP/client/configure.ac	2015-12-06 18:29:33.541770921 -0800
+++ setiathome-7.08/AP/client/configure.ac	2015-12-06 20:32:29.693652803 -0800
@@ -133,11 +133,11 @@
     AC_HELP_STRING([--disable-graphics],
                    [disable building the client graphics]))
 
-#if test "x$enable_graphics" != xno ; then
-#  enable_graphics=yes
-#else
+if test "x$enable_graphics" != xno ; then
+  enable_graphics=yes
+else
   enable_graphics=no
-#fi
+fi
 
 enable_client=yes
 
