--- webkitgtk-2.39.1/Source/WebCore/CMakeLists.txt.orig	2022-11-11 00:40:23.038185100 -0800
+++ webkitgtk-2.39.1/Source/WebCore/CMakeLists.txt	2022-11-14 12:10:17.826781848 -0800
@@ -2623,6 +2623,14 @@ if (APPLE)
     set_target_properties(WebCore PROPERTIES LINK_FLAGS "-sub_library libobjc -umbrella WebKit -allowable_client WebCoreTestSupport -allowable_client WebKit2 -allowable_client WebKitLegacy")
 endif ()
 
+# Prevent tables from collapsing too much with -Ofast and possible ofast.
+# It also prevents element alignment errors with -Ofast.
+if (COMPILER_IS_CLANG)
+    WEBKIT_ADD_TARGET_CXX_FLAGS(WebCore -fno-finite-math-only -fhonor-infinities -fhonor-nans)
+elseif (CMAKE_COMPILER_IS_GNUCXX)
+    WEBKIT_ADD_TARGET_CXX_FLAGS(WebCore -fno-finite-math-only)
+endif()
+
 # The -ftree-sra optimization (implicit with -O2) causes crashes when
 # allocating large chunks of memory using bmalloc on Intel 32bit.
 # See https://bugs.webkit.org/show_bug.cgi?id=146440
