diff -Nurp squashfuse-0.1.103-libsquashfuse-appimage.orig/configure.ac squashfuse-0.1.103-libsquashfuse-appimage/configure.ac
--- squashfuse-0.1.103-libsquashfuse-appimage.orig/configure.ac	2018-04-03 22:44:40.000000000 -0700
+++ squashfuse-0.1.103-libsquashfuse-appimage/configure.ac	2020-08-11 00:51:17.378875993 -0700
@@ -65,7 +65,7 @@ AS_IF([test "x$enable_high_level$enable_
 	AC_MSG_FAILURE([Nothing left to build]))
 
 
-AC_CONFIG_FILES([Makefile squashfuse.pc])
+AC_CONFIG_FILES([Makefile squashfuse_appimage.pc])
 AC_OUTPUT
 
 AS_ECHO()
diff -Nurp squashfuse-0.1.103-libsquashfuse-appimage.orig/Makefile.am squashfuse-0.1.103-libsquashfuse-appimage/Makefile.am
--- squashfuse-0.1.103-libsquashfuse-appimage.orig/Makefile.am	2020-08-11 00:47:33.420250010 -0700
+++ squashfuse-0.1.103-libsquashfuse-appimage/Makefile.am	2020-08-11 00:51:09.948587953 -0700
@@ -19,7 +19,7 @@ noinst_LTLIBRARIES = libsquashfuse_ll.la
 include_HEADERS = squashfuse.h squashfs_fs.h
 
 pkgconfigdir = @pkgconfigdir@
-pkgconfig_DATA 	= squashfuse.pc
+pkgconfig_DATA 	= squashfuse_appimage.pc
 
 # Main library: libsquashfuse
 libsquashfuse_la_SOURCES = swap.c cache.c table.c dir.c file.c fs.c \
diff -Nurp squashfuse-0.1.103-libsquashfuse-appimage.orig/squashfuse_appimage.pc.in squashfuse-0.1.103-libsquashfuse-appimage/squashfuse_appimage.pc.in
--- squashfuse-0.1.103-libsquashfuse-appimage.orig/squashfuse_appimage.pc.in	1969-12-31 16:00:00.000000000 -0800
+++ squashfuse-0.1.103-libsquashfuse-appimage/squashfuse_appimage.pc.in	2020-08-11 00:49:42.255202422 -0700
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: squashfuse
+Description: squashfuse library to mount squashfs archives for appimage
+Version: @VERSION@
+
+Requires:
+Libs: -L${libdir} -lsquashfuse
+Cflags: -I${includedir}
