diff -urp squashfuse-0.1.105.orig/m4/squashfuse_fuse.m4 squashfuse-0.1.105/m4/squashfuse_fuse.m4
--- squashfuse-0.1.105.orig/m4/squashfuse_fuse.m4	2022-06-14 17:34:42.000000000 -0700
+++ squashfuse-0.1.105/m4/squashfuse_fuse.m4	2023-01-13 13:09:55.459173678 -0800
@@ -142,26 +142,35 @@ AC_DEFUN([SQ_FIND_FUSE],[
 			[$sq_fuse_libs],,
 			[AC_MSG_FAILURE([Can't find FUSE in specified directories])])
 	])
+	AC_ARG_WITH([fuse-slot],
+		[AS_HELP_STRING([--with-fuse-slot=SLOT], [FUSE slot])],
+		[with_fuse_slot=$withval],
+		[with_fuse_slot=3])
+	AC_MSG_NOTICE([with_fuse_slot == $with_fuse_slot])
 	
-	# Use pkgconfig to look for fuse3.
-	AS_IF([test "x$sq_fuse_found" = xyes],,[
-		SQ_SAVE_FLAGS
-		SQ_PKG([fuse3],[fuse3 >= 3.2],
-			[ AC_DEFINE([FUSE_USE_VERSION], [32], [Version of FUSE API to use])
-              SQ_TRY_FUSE(,[sq_fuse_found=yes],
-	        [AC_MSG_FAILURE([Can't find FUSE with pkgconfig])])],
-			[:])
-		SQ_KEEP_FLAGS([FUSE],[$sq_fuse_found])
+	AS_IF([test "x$with_fuse_slot" = x3],[
+		# Use pkgconfig to look for fuse3.
+		AS_IF([test "x$sq_fuse_found" = xyes],,[
+			SQ_SAVE_FLAGS
+			SQ_PKG([fuse3],[fuse3 >= 3.2],
+				[ AC_DEFINE([FUSE_USE_VERSION], [32], [Version of FUSE API to use])
+	              SQ_TRY_FUSE(,[sq_fuse_found=yes],
+		        [AC_MSG_FAILURE([Can't find FUSE with pkgconfig])])],
+				[:])
+			SQ_KEEP_FLAGS([FUSE],[$sq_fuse_found])
+		])
-	])
+	],[])
-	# Use pkgconfig to look for fuse2.
-	AS_IF([test "x$sq_fuse_found" = xyes],,[
-        AC_DEFINE([FUSE_USE_VERSION], [26], [Version of FUSE API to use])
-		SQ_SAVE_FLAGS
-		SQ_PKG([fuse],[fuse >= 2.6],
-            [SQ_TRY_FUSE(,[sq_fuse_found=yes],
-	        [AC_MSG_FAILURE([Can't find FUSE with pkgconfig])])],
-			[:])
-		SQ_KEEP_FLAGS([FUSE],[$sq_fuse_found])
+	AS_IF([test "x$with_fuse_slot" = x2],[
+		# Use pkgconfig to look for fuse2.
+		AS_IF([test "x$sq_fuse_found" = xyes],,[
+	        AC_DEFINE([FUSE_USE_VERSION], [26], [Version of FUSE API to use])
+			SQ_SAVE_FLAGS
+			SQ_PKG([fuse],[fuse >= 2.6],
+	            [SQ_TRY_FUSE(,[sq_fuse_found=yes],
+		        [AC_MSG_FAILURE([Can't find FUSE with pkgconfig])])],
+				[:])
+			SQ_KEEP_FLAGS([FUSE],[$sq_fuse_found])
+		])
-	])
+	],[])
 	
 	# Default search locations
