diff -urp vlc-9999.orig/modules/video_output/caca.c vlc-9999/modules/video_output/caca.c
--- vlc-9999.orig/modules/video_output/caca.c	2016-12-26 15:45:03.390800234 -0800
+++ vlc-9999/modules/video_output/caca.c	2016-12-26 15:48:00.507616813 -0800
@@ -196,6 +196,7 @@ static int Open(vlc_object_t *object)
     vd->manage  = Manage;
 
     /* Fix initial state */
+    vout_display_SendEventFullscreen(vd, false);
     Refresh(vd);
 
     return VLC_SUCCESS;
diff -urp vlc-9999.orig/modules/video_output/decklink.cpp vlc-9999/modules/video_output/decklink.cpp
--- vlc-9999.orig/modules/video_output/decklink.cpp	2016-12-26 15:45:03.393800349 -0800
+++ vlc-9999/modules/video_output/decklink.cpp	2016-12-26 15:49:14.186426764 -0800
@@ -1073,6 +1073,7 @@ static int OpenVideo(vlc_object_t *p_thi
     vd->display = DisplayVideo;
     vd->control = ControlVideo;
     vd->manage  = NULL;
+    vout_display_SendEventFullscreen(vd, false);
 
     return VLC_SUCCESS;
 }
diff -urp vlc-9999.orig/modules/video_output/yuv.c vlc-9999/modules/video_output/yuv.c
--- vlc-9999.orig/modules/video_output/yuv.c	2016-12-26 15:45:03.438802082 -0800
+++ vlc-9999/modules/video_output/yuv.c	2016-12-26 15:49:53.145911196 -0800
@@ -166,6 +166,7 @@ static int Open(vlc_object_t *object)
     vd->control = Control;
     vd->manage  = NULL;
 
+    vout_display_SendEventFullscreen(vd, false);
     vout_display_DeleteWindow(vd, NULL);
     return VLC_SUCCESS;
 }
