diff -urp rocm-core-rocm-5.5.1.orig/CMakeLists.txt rocm-core-rocm-5.5.1/CMakeLists.txt
--- rocm-core-rocm-5.5.1.orig/CMakeLists.txt	2023-08-28 16:19:51.977719740 -0700
+++ rocm-core-rocm-5.5.1/CMakeLists.txt	2023-08-28 16:20:07.161719926 -0700
@@ -39,7 +39,7 @@ include( GNUInstallDirs )
 message( STATUS "ROCM_VERSION = ${ROCM_VERSION}" )
 parse_rocm_version( ${ROCM_VERSION} )
 
-set( CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE PATH "default cpack directory" )
+set( CPACK_PACKAGING_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE PATH "default cpack directory" )
 set( BUILD_SHARED_LIBS ON CACHE BOOL "Build shared library (.so) or not." )
 set( CPACK_GENERATOR "TGZ;DEB;RPM" CACHE STRING "package types to be produced " )
 set( COPYRIGHT_FILE "${CMAKE_CURRENT_SOURCE_DIR}/copyright"  )
diff -urp rocm-core-rocm-5.5.1.orig/rocmmod.in rocm-core-rocm-5.5.1/rocmmod.in
--- rocm-core-rocm-5.5.1.orig/rocmmod.in	2023-02-22 10:47:36.000000000 -0800
+++ rocm-core-rocm-5.5.1/rocmmod.in	2023-08-28 16:20:32.241720235 -0700
@@ -10,7 +10,7 @@ proc ModulesHelp { } {
 }
 
 
-prepend-path PATH "@CPACK_PACKAGING_INSTALL_PREFIX@/@CMAKE_INSTALL_BINDIR@:@CPACK_PACKAGING_INSTALL_PREFIX@/llvm/bin"
+prepend-path PATH "@CPACK_PACKAGING_INSTALL_PREFIX@/@CMAKE_INSTALL_BINDIR@:@EPREFIX_LLVM_PATH@/bin"
 prepend-path CMAKE_PREFIX_PATH "@CPACK_PACKAGING_INSTALL_PREFIX@"
 setenv ROCM_PATH "@CPACK_PACKAGING_INSTALL_PREFIX@"
 setenv HIP_PATH "@CPACK_PACKAGING_INSTALL_PREFIX@"
Only in rocm-core-rocm-5.5.1: rocmmod.in.orig
Only in rocm-core-rocm-5.5.1: rocmmod.in.rej
