diff -urp rocm-core-rocm-5.5.1.orig/CMakeLists.txt rocm-core-rocm-5.5.1/CMakeLists.txt
--- rocm-core-rocm-5.5.1.orig/CMakeLists.txt	2023-08-28 16:01:10.073705941 -0700
+++ rocm-core-rocm-5.5.1/CMakeLists.txt	2023-08-28 16:13:18.777714904 -0700
@@ -69,7 +69,7 @@ set( SRCS rocm_version.cpp )
 add_library( ${CORE_TARGET} ${SRCS} )
 
 set( CXX_FLAGS ${CXX_FLAGS} -g -fPIC -fvisibility=hidden -W -Wall -Wextra -Wno-unused-parameter -Wformat-security -Wundef -Wshadow -Wpointer-arith -Wcast-qual -Wmissing-declarations -Wredundant-decls -Wunreachable-code -std=c++11 )
-set( CMAKE_SHARED_LINKER_FLAGS ${CMAKE_SHARED_LINKER_FLAGS} "-Wl,-z,nodelete -Wl,-no-undefined" )
+set( CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,nodelete -Wl,-no-undefined" )
 target_include_directories( ${CORE_TARGET} PRIVATE ${BUILD_DIR} )
 
 ## Set the VERSION and SOVERSION values
