diff -urp rocm-core-rocm-5.6.0.orig/CMakeLists.txt rocm-core-rocm-5.6.0/CMakeLists.txt
--- rocm-core-rocm-5.6.0.orig/CMakeLists.txt	2023-04-05 00:16:56.000000000 -0700
+++ rocm-core-rocm-5.6.0/CMakeLists.txt	2023-08-28 16:22:10.081721438 -0700
@@ -39,7 +39,7 @@ include( GNUInstallDirs )
 message( STATUS "ROCM_VERSION = ${ROCM_VERSION}" )
 parse_rocm_version( ${ROCM_VERSION} )
 
-set( CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE PATH "default cpack directory" )
+set( CPACK_PACKAGING_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE PATH "default cpack directory" )
 set( BUILD_SHARED_LIBS ON CACHE BOOL "Build shared library (.so) or not." )
 set( CPACK_GENERATOR "TGZ;DEB;RPM" CACHE STRING "package types to be produced " )
 set( COPYRIGHT_FILE "${CMAKE_CURRENT_SOURCE_DIR}/copyright"  )
diff -urp rocm-core-rocm-5.6.0.orig/rocmmod.in rocm-core-rocm-5.6.0/rocmmod.in
--- rocm-core-rocm-5.6.0.orig/rocmmod.in	2023-04-05 00:16:56.000000000 -0700
+++ rocm-core-rocm-5.6.0/rocmmod.in	2023-08-28 16:22:39.437721799 -0700
@@ -10,8 +10,8 @@ proc ModulesHelp { } {
 }
 
 
-prepend-path PATH "@CPACK_PACKAGING_INSTALL_PREFIX@/@CMAKE_INSTALL_BINDIR@:@CPACK_PACKAGING_INSTALL_PREFIX@/llvm/bin"
-prepend-path MANPATH "@CPACK_PACKAGING_INSTALL_PREFIX@/@CMAKE_INSTALL_MANDIR@:@CPACK_PACKAGING_INSTALL_PREFIX@/llvm/share/man1"
+prepend-path PATH "@CPACK_PACKAGING_INSTALL_PREFIX@/@CMAKE_INSTALL_BINDIR@:@EPREFIX_LLVM_PATH@/bin"
+prepend-path MANPATH "@CPACK_PACKAGING_INSTALL_PREFIX@/@CMAKE_INSTALL_MANDIR@:@EPREFIX_LLVM_PATH@/share/man/man1"
 prepend-path CMAKE_PREFIX_PATH "@CPACK_PACKAGING_INSTALL_PREFIX@"
 setenv ROCM_PATH "@CPACK_PACKAGING_INSTALL_PREFIX@"
 setenv HIP_PATH "@CPACK_PACKAGING_INSTALL_PREFIX@"
Only in rocm-core-rocm-5.6.0: rocmmod.in.orig
Only in rocm-core-rocm-5.6.0: rocmmod.in.rej
