--- a/setup.py.orig	2023-03-15 15:32:44.000000000 -0700
+++ b/setup.py	2023-03-19 19:22:35.885491086 -0700
@@ -180,7 +180,7 @@ class BuildExtCommand(setuptools.command
             os.getenv('TENSORSTORE_BAZEL_STARTUP_OPTIONS', ''))
         build_options = shlex.split(
             os.getenv('TENSORSTORE_BAZEL_BUILD_OPTIONS', ''))
-        build_command = [sys.executable, '-u', bazelisk] + startup_options + [
+        build_command = [ 'bazel', # oteodoro: removed bazelisk and call bazel directly
             'build',
             '-c',
             compilation_mode,
