diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/ccan-upstream/ccan/opt/helpers.c bfgminer-5.5.0/ccan-upstream/ccan/opt/helpers.c
--- bfgminer-5.5.0.orig/ccan-upstream/ccan/opt/helpers.c	2017-12-21 15:42:31.000000000 -0800
+++ bfgminer-5.5.0/ccan-upstream/ccan/opt/helpers.c	2019-03-20 22:23:04.191903036 -0700
@@ -11,6 +11,10 @@
 #include "private.h"
 #include <float.h>
 
+#ifdef USE_BOINC
+#include "boinc_wrapper.h"
+#endif
+
 /* Upper bound to sprintf this simple type?  Each 3 bits < 1 digit. */
 #define CHAR_SIZE(type) (((sizeof(type)*CHAR_BIT + 2) / 3) + 1)
 
@@ -183,7 +187,11 @@ char *opt_version_and_exit(const char *v
 	printf("%s\n", version);
 	/* Don't have valgrind complain! */
 	opt_free_table();
+#ifdef USE_BOINC
+	bw_boinc_finish(0);
+#else
 	exit(0);
+#endif
 }
 
 char *opt_usage_and_exit(const char *extra)
@@ -193,7 +201,11 @@ char *opt_usage_and_exit(const char *ext
 	/* Don't have valgrind complain! */
 	opt_alloc.free(usage);
 	opt_free_table();
+#ifdef USE_BOINC
+	bw_boinc_finish(0);
+#else
 	exit(0);
+#endif
 }
 
 void opt_show_bool(char buf[OPT_SHOW_LEN], const bool *b)
diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/ccan-upstream/ccan/opt/opt.c bfgminer-5.5.0/ccan-upstream/ccan/opt/opt.c
--- bfgminer-5.5.0.orig/ccan-upstream/ccan/opt/opt.c	2017-12-21 15:42:31.000000000 -0800
+++ bfgminer-5.5.0/ccan-upstream/ccan/opt/opt.c	2019-03-20 22:18:18.707823660 -0700
@@ -15,6 +15,10 @@ const char *opt_argv0;
 struct opt_alloc opt_alloc = {
 	malloc, realloc, free
 };
+#ifdef USE_BOINC
+#include "boinc_wrapper.h"
+#endif
+
 
 /* Returns string after first '-'. */
 static const char *first_name(const char *names, unsigned *len)
@@ -22,6 +26,7 @@ static const char *first_name(const char
 	*len = strcspn(names + 1, "|= ");
 	return names + 1;
 }
+			boinc_quit_status(status, fmt, __VA_ARGS__); \
 
 static const char *next_name(const char *names, unsigned *len)
 {
@@ -259,7 +264,11 @@ void opt_log_stderr_exit(const char *fmt
 	vfprintf(stderr, fmt, ap);
 	fprintf(stderr, "\n");
 	va_end(ap);
+#ifdef USE_BOINC
+	bw_boinc_finish(1);
+#else
 	exit(1);
+#endif
 }
 
 char *opt_invalid_argument(const char *arg)
diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/configure.ac bfgminer-5.5.0/configure.ac
--- bfgminer-5.5.0.orig/configure.ac	2018-01-03 19:33:24.000000000 -0800
+++ bfgminer-5.5.0/configure.ac	2019-03-20 22:18:18.725824359 -0700
@@ -70,6 +70,7 @@ gl_EARLY
 AC_PROG_GCC_TRADITIONAL
 AM_PROG_CC_C_O
 AC_PROG_RANLIB
+AC_PROG_CXX
 AC_PROG_CPP
 AC_PROG_SED
 
diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/driver-cpu.c bfgminer-5.5.0/driver-cpu.c
--- bfgminer-5.5.0.orig/driver-cpu.c	2018-01-03 19:33:24.000000000 -0800
+++ bfgminer-5.5.0/driver-cpu.c	2019-03-20 22:18:18.736824785 -0700
@@ -233,14 +233,22 @@ double bench_algo_stage3(
 		int flags = fcntl(fd, F_GETFL, 0);
 		if (flags<0) {
 			perror("fcntl(GET) failed");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 		flags = yes ? (flags|O_NONBLOCK) : (flags&~O_NONBLOCK);
 
 		int r = fcntl(fd, F_SETFL, flags);
 		if (r<0) {
 			perror("fcntl(SET) failed");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 	}
 
@@ -264,7 +272,11 @@ static double bench_algo_stage2(
 		int r = pipe(pfd);
 		if (r<0) {
 			perror("pipe - failed to create pipe for --algo auto");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 		// Make pipe non blocking
@@ -276,14 +288,22 @@ static double bench_algo_stage2(
 		sighandler_t sr1 = signal(SIGPIPE, SIG_IGN);
 		if (SIG_ERR==sr0 || SIG_ERR==sr1) {
 			perror("signal - failed to edit signal mask for --algo auto");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 		// Fork a child to do the actual benchmarking
 		pid_t child_pid = fork();
 		if (child_pid<0) {
 			perror("fork - failed to create a child process for --algo auto");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 		// Do the dangerous work in the child, knowing we might crash
@@ -337,7 +357,11 @@ static double bench_algo_stage2(
 
 			} else if (r<0) {
 				perror("bench_algo: waitpid failed. giving up.");
+#ifdef USE_BOINC
+				bw_boinc_finish(1);
+#else
 				exit(1);
+#endif
 			}
 
 			// Give up on child after a ~60s
@@ -356,12 +380,20 @@ static double bench_algo_stage2(
 		r = close(pfd[0]);
 		if (r<0) {
 			perror("close - failed to close read end of pipe for --algo auto");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 		r = close(pfd[1]);
 		if (r<0) {
 			perror("close - failed to close read end of pipe for --algo auto");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 	#elif defined(WIN32)
@@ -370,7 +402,11 @@ static double bench_algo_stage2(
 		HINSTANCE module = GetModuleHandle(0);
 		if (!module) {
 			applog(LOG_ERR, "failed to retrieve module handle");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 		// Create a unique name
@@ -393,7 +429,11 @@ static double bench_algo_stage2(
 		);
 		if (NULL==map_handle) {
 			applog(LOG_ERR, "could not create shared memory");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 		void *shared_mem = MapViewOfFile(
@@ -405,7 +445,11 @@ static double bench_algo_stage2(
 		);
 		if (NULL==shared_mem) {
 			applog(LOG_ERR, "could not map shared memory");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 		SetEnvironmentVariable("BFGMINER_SHARED_MEM", unique_name);
 		CopyMemory(shared_mem, &rate, sizeof(rate));
@@ -416,7 +460,11 @@ static double bench_algo_stage2(
 		DWORD size = GetModuleFileName(module, cmd_line, n);
 		if (0==size) {
 			applog(LOG_ERR, "failed to retrieve module path");
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 		// Construct new command line based on that
@@ -445,7 +493,11 @@ static double bench_algo_stage2(
 		);
 		if (!ok) {
 			applog(LOG_ERR, "CreateProcess failed with error %ld\n", (long)GetLastError() );
+#ifdef USE_BOINC
+			bw_boinc_finish(1);
+#else
 			exit(1);
+#endif
 		}
 
 		// Debug the child (only clean way to catch exceptions)
@@ -564,7 +616,11 @@ void init_max_name_len()
 	name_spaces_pad = (char*) malloc(max_name_len+16);
 	if (0==name_spaces_pad) {
 		perror("malloc failed");
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	}
 }
 
diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/libblkmaker/example.c bfgminer-5.5.0/libblkmaker/example.c
--- bfgminer-5.5.0.orig/libblkmaker/example.c	2017-12-22 20:31:52.000000000 -0800
+++ bfgminer-5.5.0/libblkmaker/example.c	2019-03-20 22:18:18.754825484 -0700
@@ -24,6 +24,10 @@
 #include "private.h"
 #include "testinput.c"
 
+#ifdef USE_BOINC
+#include "boinc_wrapper.h"
+#endif
+
 void testb58() {
 	int rv;
 	const char *iaddr = "11Baf75Ferj6A7AoN565gCQj9kGWbDMHfN9";
@@ -34,9 +38,17 @@ void testb58() {
 	char *buf = &bufx[1];
 	actuallen = 25;
 	if (!b58tobin(buf, &actuallen, addr, addrlen))
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	if (bufx[0] != '\xff')
+#ifdef USE_BOINC
+		bw_boinc_finish(2);
+#else
 		exit(2);
+#endif
 	char cbuf[51];
 	_blkmk_bin2hex(cbuf, buf, 25);
 	printf("Base58 raw data: %s\n", cbuf);
diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/logging.h bfgminer-5.5.0/logging.h
--- bfgminer-5.5.0.orig/logging.h	2018-01-03 19:33:24.000000000 -0800
+++ bfgminer-5.5.0/logging.h	2019-03-20 22:18:18.754825484 -0700
@@ -134,6 +134,7 @@ extern void _bfg_clean_up(bool);
 		fprintf(stderr, "\n" fmt, ##__VA_ARGS__);  \
 	} \
 	fflush(stderr);  \
+	boinc_quit_status(status, fmt, ##__VA_ARGS__); \
 	_quit(status); \
 } while (0)
 
@@ -144,6 +145,7 @@ extern void _bfg_clean_up(bool);
 				##__VA_ARGS__, __FILE__, __func__, __LINE__); \
 		_applog(LOG_ERR, tmp42); \
 	} \
+	boinc_quit_status(status, fmt, ##__VA_ARGS__); \
 	_quit(status); \
 } while (0)
 
@@ -154,6 +156,7 @@ extern void _bfg_clean_up(bool);
 				##__VA_ARGS__, _file, _func, _line); \
 		_applog(LOG_ERR, tmp42); \
 	} \
+	boinc_quit_status(status, fmt, ##__VA_ARGS__); \
 	_quit(status); \
 } while (0)
 
diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/Makefile.am bfgminer-5.5.0/Makefile.am
--- bfgminer-5.5.0.orig/Makefile.am	2018-01-03 19:33:24.000000000 -0800
+++ bfgminer-5.5.0/Makefile.am	2019-03-20 22:21:24.690041090 -0700
@@ -9,6 +9,7 @@
 # Software Foundation; either version 3 of the License, or (at your option)
 # any later version.  See COPYING for more details.
 
+CXX=g++
 ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST	= \
@@ -62,12 +63,14 @@ bfgminer_LDFLAGS += $(PTHREAD_FLAGS)
 bfgminer_LDADD   += $(DLOPEN_FLAGS) @LIBCURL_LIBS@ @JANSSON_LIBS@ @PTHREAD_LIBS@ \
 		  @NCURSES_LIBS@ @PDCURSES_LIBS@ @WS2_LIBS@ \
 		  @UDEV_LIBS@ @LIBUSB_LIBS@ @MM_LIBS@ @RT_LIBS@ \
-                    $(MATH_LIBS)
-bfgminer_CPPFLAGS += @LIBUSB_CFLAGS@ @LIBCURL_CFLAGS@
+                    $(MATH_LIBS) \
+                    -lboinc_graphics2 -lboinc -lboinc_api -lboinc -lboinc_opencl
+bfgminer_CPPFLAGS += @LIBUSB_CFLAGS@ @LIBCURL_CFLAGS@ -pthread
 
 bfgminer_CPPFLAGS += -DCGMINER_PREFIX='"$(pkgdatadir)"'
 
 bfgminer_CPPFLAGS += @JANSSON_CFLAGS@
+bfgminer_CPPFLAGS += -DUSE_BOINC -I/usr/include/boinc -I/usr/share/boinc -I/usr/share/boinc/api -I/usr/share/boinc/lib -I/usr/share/boinc/sched -I/usr/share/boinc/db
 bfgminer_CPPFLAGS += $(PTHREAD_FLAGS)
 bfgminer_CPPFLAGS += $(NCURSES_CPPFLAGS)
 
@@ -80,7 +83,7 @@ bfgminer_SOURCES := miner.c
 bfgminer_SOURCES += miner.h compat.h  \
 	deviceapi.c deviceapi.h \
 		   util.c util.h logging.h		\
-		   sha2.c sha2.h api.c
+		   sha2.c sha2.h api.c boinc_wrapper.cpp
 EXTRA_bfgminer_DEPENDENCIES =
 
 TESTS = test-bfgminer.sh
diff -x '*.orig' -x '*.rej' -Nurp bfgminer-5.5.0.orig/miner.c bfgminer-5.5.0/miner.c
--- bfgminer-5.5.0.orig/miner.c	2018-01-03 19:33:24.000000000 -0800
+++ bfgminer-5.5.0/miner.c	2019-03-20 22:24:14.212620966 -0700
@@ -90,6 +90,12 @@
 #include "lowlevel.h"
 #endif
 
+#ifdef USE_BOINC
+#include "boinc_wrapper.h"
+#endif
+
+#include "filesys.h"
+
 #if defined(unix) || defined(__APPLE__)
 	#include <errno.h>
 	#include <fcntl.h>
@@ -112,6 +118,10 @@ enum bfg_quit_summary {
 	BQS_DETAILED,
 };
 
+#ifdef USE_BOINC
+BW_BOINC_OPTIONS options;
+int boinc_success = 0;
+#endif
 struct strategies strategies[] = {
 	{ "Failover" },
 	{ "Round Robin" },
@@ -416,6 +426,7 @@ struct sigaction termhandler, inthandler
 
 struct thread_q *getq;
 
+static int total_work_finished;
 static int total_work;
 static bool staged_full;
 struct work *staged_work = NULL;
@@ -506,7 +517,11 @@ static void applog_and_exit(const char *
 	vsnprintf(exit_buf, sizeof(exit_buf), fmt, ap);
 	va_end(ap);
 	_applog(LOG_ERR, exit_buf);
+#ifdef USE_BOINC
+	bw_boinc_finish(1);
+#else
 	exit(1);
+#endif
 }
 
 static
@@ -3048,7 +3063,11 @@ static char *opt_verusage_and_exit(const
 	bfg_versioninfo();
 	printf("%s", opt_usage(opt_argv0, extra));
 	fflush(stdout);
+#ifdef USE_BOINC
+	bw_boinc_finish(0);
+#else
 	exit(0);
+#endif
 }
 
 static
@@ -3056,7 +3075,11 @@ const char *my_opt_version_and_exit(void
 {
 	bfg_versioninfo();
 	fflush(stdout);
+#ifdef USE_BOINC
+	bw_boinc_finish(0);
+#else
 	exit(0);
+#endif
 }
 
 /* These options are parsed before anything else */
@@ -4671,6 +4694,9 @@ void update_block_display(const bool wit
 	// We cannot do resizing if called within someone else's console lock
 	if (within_console_lock)
 		return;
+#ifdef USE_BOINC
+			boinc_work_success();
+#endif
 	
 	bfg_console_unlock();
 	if (total_found_goals != active_goals)
@@ -6120,8 +6146,14 @@ tryagain:
 
 	cgtime(&work->tv_getwork);
 
+#ifdef USE_BOINC
+	bw_boinc_begin_critical_section();
+#endif
 	val = json_rpc_call(curl, url, pool->rpc_userpass, rpc_req, false,
 			    false, &work->rolltime, pool, false);
+#ifdef USE_BOINC
+	bw_boinc_end_critical_section();
+#endif
 	pool_stats->getwork_attempts++;
 
 	free(rpc_req);
@@ -6924,6 +6956,9 @@ static void *submit_work_thread(__maybe_
 	int n;
 	CURLMsg *cm;
 	FD_ZERO(&rfds);
+#ifdef USE_BOINC
+	bw_boinc_begin_critical_section();
+#endif
 	while (1) {
 		mutex_lock(&submitting_lock);
 		total_submitting -= tsreduce;
@@ -7114,6 +7149,9 @@ next_write_sws_del:
 			}
 		}
 	}
+#ifdef USE_BOINC
+	bw_boinc_end_critical_section();
+#endif
 	assert(!write_sws);
 	mutex_unlock(&submitting_lock);
 
@@ -9081,6 +9119,9 @@ static void hashmeter(int thr_id, struct
 	
 	timersub(&temp_tv_end, &total_tv_end, &total_diff);
 
+#ifdef USE_BOINC
+	bw_boinc_fraction_done(((double)total_work_finished)/total_work);
+#endif
 	total_mhashes_done += local_mhashes;
 	local_mhashes_done += local_mhashes;
 	/* Only update with opt_log_interval */
@@ -9861,8 +9902,14 @@ tryagain:
 
 	pool->probed = false;
 	cgtime(&tv_getwork);
+#ifdef USE_BOINC
+	bw_boinc_begin_critical_section();
+#endif
 	val = json_rpc_call(curl, pool->rpc_url, pool->rpc_userpass, rpc_req,
 			true, false, &rolltime, pool, false);
+#ifdef USE_BOINC
+	bw_boinc_end_critical_section();
+#endif
 	cgtime(&tv_getwork_reply);
 
 	free(rpc_req);
@@ -10937,6 +10984,7 @@ struct work *clone_queued_work_bymidstat
 
 void __work_completed(struct cgpu_info *cgpu, struct work *work)
 {
+	total_work_finished++;
 	cgpu->queued_count--;
 	HASH_DEL(cgpu->queued_work, work);
 }
@@ -11266,8 +11314,14 @@ retry_pool:
 		curl_easy_setopt(curl, CURLOPT_FORBID_REUSE, 1);
 		curl_easy_setopt(curl, CURLOPT_OPENSOCKETFUNCTION, save_curl_socket);
 		curl_easy_setopt(curl, CURLOPT_OPENSOCKETDATA, pool);
+#ifdef USE_BOINC
+		bw_boinc_begin_critical_section();
+#endif
 		val = json_rpc_call(curl, lp_url, pool->rpc_userpass,
 				    lpreq, false, true, &rolltime, pool, false);
+#ifdef USE_BOINC
+		bw_boinc_end_critical_section();
+#endif
 		pool->lp_socket = CURL_SOCKET_BAD;
 
 		cgtime(&reply);
@@ -11606,6 +11660,10 @@ static void *watchdog_thread(void __mayb
 {
 	const unsigned int interval = WATCHDOG_INTERVAL;
 	struct timeval zero_tv;
+#ifdef USE_BOINC
+	int suspended_prev;
+	int suspended_curr;
+#endif
 
 #ifndef HAVE_PTHREAD_CANCEL
 	pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
@@ -11616,10 +11674,18 @@ static void *watchdog_thread(void __mayb
 	memset(&zero_tv, 0, sizeof(struct timeval));
 	cgtime(&rotate_tv);
 
+
+
 	while (1) {
 		int i;
 		struct timeval now;
 
+#ifdef USE_BOINC
+		applog(LOG_WARNING, "Suspended state: %d", suspended_curr);
+		suspended_curr = bw_boinc_get_status_suspended();
+
+		bw_boinc_begin_critical_section();
+#endif
 		sleep(interval);
 
 		discard_stale();
@@ -11640,6 +11706,42 @@ static void *watchdog_thread(void __mayb
 
 		cgtime(&now);
 
+
+#ifdef USE_BOINC
+
+		if (!sched_paused && (suspended_curr != suspended_prev) && suspended_curr) {
+			sched_paused = true;
+
+			applog(LOG_WARNING, "Suspending...", suspended_curr);
+			rd_lock(&mining_thr_lock);
+			for (i = 0; i < mining_threads; i++)
+				mining_thr[i]->pause = true;
+			rd_unlock(&mining_thr_lock);
+		} else if (sched_paused && (suspended_curr != suspended_prev)  && !suspended_curr) {
+			sched_paused = false;
+
+			applog(LOG_WARNING, "Unsupending...", suspended_curr);
+
+			for (i = 0; i < mining_threads; i++) {
+				struct thr_info *thr;
+
+				thr = get_thread(i);
+				thr->pause = false;
+			}
+
+			for (i = 0; i < total_devices; ++i)
+			{
+				struct cgpu_info *cgpu = get_devices(i);
+
+				/* Don't touch disabled devices */
+				if (cgpu->deven == DEV_DISABLED)
+					continue;
+				proc_enable(cgpu);
+			}
+		}
+
+		suspended_prev = suspended_curr;
+#else
 		if (!sched_paused && !should_run()) {
 			applog(LOG_WARNING, "Pausing execution as per stop time %02d:%02d scheduled",
 			       schedstop.tm.tm_hour, schedstop.tm.tm_min);
@@ -11681,14 +11783,20 @@ static void *watchdog_thread(void __mayb
 				proc_enable(cgpu);
 			}
 		}
+#endif
 
 		for (i = 0; i < total_devices; ++i) {
 			struct cgpu_info *cgpu = get_devices(i);
 			if (!cgpu->disable_watchdog)
 				bfg_watchdog(cgpu, &now);
 		}
+#ifdef USE_BOINC
+		bw_boinc_end_critical_section();
+#endif
 	}
 
+
+
 	return NULL;
 }
 
@@ -11968,6 +12076,54 @@ void _bfg_clean_up(bool restarting)
 #endif
 }
 
+#ifdef USE_BOINC
+void boinc_work_success()
+{
+	boinc_success = 1;
+}
+#endif
+
+void boinc_quit_status(int status, const char *format, ...)
+{
+#ifdef USE_BOINC
+	FILE *fresult;
+
+	char result_virtual_name[256] = "result.txt";
+	char result_physical_name[256];
+	bw_boinc_get_result_physical_name(result_virtual_name, result_physical_name, 256);
+	applog(LOG_WARNING,"creating %s", result_physical_name);
+	if (strlen(result_physical_name) == 0) {
+		applog(LOG_WARNING,"Could not create boinc result file.");
+	        bw_boinc_show_message(1, "Boinc BFGMiner quiting...", 0);
+	        bw_boinc_finish(1);
+	}
+	fresult = bw_boinc_fopen(result_physical_name, "w");
+	if (!fresult) {
+		applog(LOG_WARNING,"Could not create boinc result file.");
+	        bw_boinc_show_message(1, "Boinc BFGMiner quiting...", 0);
+	        bw_boinc_finish(1);
+	}
+
+	char buffer[256];
+	va_list args;
+	va_start (args, format);
+	vsnprintf(buffer, 255, format, args);
+
+	if (status == 1) {
+		fputs("fail", fresult);
+	} else if (status == 0 && boinc_success == 0) {
+		fputs("in progress", fresult);
+	} else if (status == 0 && boinc_success == 1) {
+		fputs("success", fresult);
+	}
+
+	bw_boinc_fclose(fresult);
+	applog(LOG_WARNING,"closed %s", result_physical_name);
+	applog(LOG_WARNING,"quitting...");
+	bw_boinc_finish(status);
+#endif
+}
+
 void _quit(int status)
 {
 	if (status) {
@@ -11986,7 +12142,11 @@ void _quit(int status)
 	}
 #endif
 
+#ifdef USE_BOINC
+	bw_boinc_finish(status);
+#else
 	exit(status);
+#endif
 }
 
 #ifdef HAVE_CURSES
@@ -12238,7 +12398,11 @@ static void fork_monitor()
 
 	if (r < 0) {
 		perror("pipe - failed to create pipe for --monitor");
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	}
 
 	// Make stderr write end of pipe
@@ -12246,12 +12410,20 @@ static void fork_monitor()
 	r = dup2(pfd[1], 2);
 	if (r < 0) {
 		perror("dup2 - failed to alias stderr to write end of pipe for --monitor");
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	}
 	r = close(pfd[1]);
 	if (r < 0) {
 		perror("close - failed to close write end of pipe for --monitor");
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	}
 
 	// Don't allow a dying monitor to kill the main process
@@ -12259,14 +12431,22 @@ static void fork_monitor()
 	sighandler_t sr1 = signal(SIGPIPE, SIG_IGN);
 	if (SIG_ERR == sr0 || SIG_ERR == sr1) {
 		perror("signal - failed to edit signal mask for --monitor");
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	}
 
 	// Fork a child process
 	forkpid = fork();
 	if (forkpid < 0) {
 		perror("fork - failed to fork child process for --monitor");
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	}
 
 	// Child: launch monitor command
@@ -12293,7 +12473,11 @@ static void fork_monitor()
 	r = close(pfd[0]);
 	if (r < 0) {
 		perror("close - failed to close read end of pipe for --monitor");
+#ifdef USE_BOINC
+		bw_boinc_finish(1);
+#else
 		exit(1);
+#endif
 	}
 }
 #endif // defined(unix)
@@ -13251,6 +13435,12 @@ int main(int argc, char *argv[])
 	int rearrange_pools = 0;
 	char *s;
 
+#ifdef USE_BOINC
+	bw_boinc_init();
+
+	bw_boinc_init_options(&options);
+#endif
+
 #ifdef WIN32
 	LoadLibrary("backtrace.dll");
 #endif
@@ -13377,7 +13567,11 @@ int main(int argc, char *argv[])
 				}
 				(void)CloseHandle(map_handle);
 			}
+#ifdef USE_BOINC
+			bw_boinc_finish(0);
+#else
 			exit(0);
+#endif
 		}
 	}
 #endif
@@ -13419,8 +13613,8 @@ int main(int argc, char *argv[])
 			   "Options for command line only");
 
 	opt_parse(&argc, argv, applog_and_exit);
-	if (argc != 1)
-		quit(1, "Unexpected extra commandline arguments");
+	//if (argc != 1)
+	//	quit(1, "Unexpected extra commandline arguments");
 	
 	if (rearrange_pools && rearrange_pools < total_pools)
 	{
@@ -13972,5 +14166,8 @@ retry:
 		push_curl_entry(ce, pool);
 	}
 
+#ifdef USE_BOINC
+	bw_boinc_finish(0);
+#endif
 	return 0;
 }
