diff -urp HIPIFY-rocm-5.2.3.orig/CMakeLists.txt HIPIFY-rocm-5.2.3/CMakeLists.txt
--- HIPIFY-rocm-5.2.3.orig/CMakeLists.txt	2023-09-26 13:43:40.378091878 -0700
+++ HIPIFY-rocm-5.2.3/CMakeLists.txt	2023-09-26 13:44:38.010091215 -0700
@@ -134,13 +134,13 @@ if(UNIX)
     if(FILE_REORG_BACKWARD_COMPATIBILITY)
         include(hipify-backward-compat.cmake)
     endif()
-    set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE PATH "HIP Package Installation Path")
+    set(CPACK_PACKAGING_INSTALL_PREFIX "@ESYSROOT_ROCM_PATH@" CACHE PATH "HIP Package Installation Path")
     #TODO: To be removed
     #In jenkins and docker build , hipify need to be installed in /opt/rocm.
     #Currently build script passes /opt/rocm/hip as install path
     #Workaround for removing same, till prototype changes are merged
     if(CPACK_PACKAGING_INSTALL_PREFIX)
-        string(FIND ${CPACK_PACKAGING_INSTALL_PREFIX} "/opt/rocm" ROCMDIR_FOUND)
+        string(FIND ${CPACK_PACKAGING_INSTALL_PREFIX} "@ESYSROOT_ROCM_PATH@" ROCMDIR_FOUND)
         if(NOT ${ROCMDIR_FOUND} MATCHES "-1")
             get_filename_component(DEST_DIR ${CPACK_PACKAGING_INSTALL_PREFIX} NAME)
             if(DEST_DIR STREQUAL "hip")
Only in HIPIFY-rocm-5.2.3: CMakeLists.txt.orig
Only in HIPIFY-rocm-5.2.3: CMakeLists.txt.rej
diff -urp HIPIFY-rocm-5.2.3.orig/hipify-backward-compat.cmake HIPIFY-rocm-5.2.3/hipify-backward-compat.cmake
--- HIPIFY-rocm-5.2.3.orig/hipify-backward-compat.cmake	2022-04-22 11:35:20.000000000 -0700
+++ HIPIFY-rocm-5.2.3/hipify-backward-compat.cmake	2023-09-26 13:44:02.090091628 -0700
@@ -23,9 +23,9 @@ set(HIPIFY_BUILD_DIR ${CMAKE_CURRENT_BIN
 set(HIPIFY_WRAPPER_DIR ${HIPIFY_BUILD_DIR}/wrapper_dir)
 set(HIPIFY_WRAPPER_BIN_DIR ${HIPIFY_WRAPPER_DIR}/bin)
 
-#With File Reorganization , hipify(and hip)  will be installed in /opt/rocm-ver
-#instead of  /opt/rocm-ver/hip/. For maintaining backward  compatibility
-# the previous location(/opt/rocm-ver/hip/) will have soft link.
+#With File Reorganization , hipify(and hip)  will be installed in @ESYSROOT_ROCM_PATH@
+#instead of  @ESYSROOT_ROCM_PATH@/hip/. For maintaining backward  compatibility
+# the previous location(@ESYSROOT_ROCM_PATH@/hip/) will have soft link.
 #This file is for creating  soft link to binary files and install it in the  previous location
 #Note: soft link added for binary files.
 
