diff -urp HIPIFY-rocm-5.7.0.orig/CMakeLists.txt HIPIFY-rocm-5.7.0/CMakeLists.txt
--- HIPIFY-rocm-5.7.0.orig/CMakeLists.txt	2023-09-27 20:02:59.800836234 -0700
+++ HIPIFY-rocm-5.7.0/CMakeLists.txt	2023-09-27 20:04:10.636835419 -0700
@@ -171,7 +171,8 @@ if (NOT HIPIFY_CLANG_TESTS_ONLY)
     PATTERN "*.sh")
 
   # Headers are already included in HIP SDK, so skip those if including HIPIFY in HIP SDK.
-  if(NOT HIPIFY_INCLUDE_IN_HIP_SDK)
+  option(HIPIFY_INSTALL_HEADERS "Install headers" OFF)
+  if(HIPIFY_INSTALL_HEADERS AND NOT HIPIFY_INCLUDE_IN_HIP_SDK)
     # Install all folders under clang/version/ in CMAKE_INSTALL_PREFIX path.
     install(
       DIRECTORY ${LLVM_DIR}/../../clang/${LIB_CLANG_RES}/
