diff -urp HIPIFY-rocm-5.3.3.orig/CMakeLists.txt HIPIFY-rocm-5.3.3/CMakeLists.txt
--- HIPIFY-rocm-5.3.3.orig/CMakeLists.txt	2023-08-23 14:20:02.345685735 -0700
+++ HIPIFY-rocm-5.3.3/CMakeLists.txt	2023-08-23 14:20:27.229686352 -0700
@@ -90,7 +90,7 @@ if(UNIX)
     if(FILE_REORG_BACKWARD_COMPATIBILITY)
         include(hipify-backward-compat.cmake)
     endif()
-    set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE PATH "HIP Package Installation Path")
+    set(CPACK_PACKAGING_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE PATH "HIP Package Installation Path")
     set(BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}/packages/hipify-clang)
     configure_file(packaging/hipify-clang.txt ${BUILD_DIR}/CMakeLists.txt @ONLY)
     configure_file(${CMAKE_SOURCE_DIR}/LICENSE.txt ${BUILD_DIR}/LICENSE.txt @ONLY)
diff -urp HIPIFY-rocm-5.3.3.orig/hipify-backward-compat.cmake HIPIFY-rocm-5.3.3/hipify-backward-compat.cmake
--- HIPIFY-rocm-5.3.3.orig/hipify-backward-compat.cmake	2022-08-22 08:31:44.000000000 -0700
+++ HIPIFY-rocm-5.3.3/hipify-backward-compat.cmake	2023-08-23 14:20:40.653686685 -0700
@@ -23,9 +23,9 @@ set(HIPIFY_BUILD_DIR ${CMAKE_CURRENT_BIN
 set(HIPIFY_WRAPPER_DIR ${HIPIFY_BUILD_DIR}/wrapper_dir)
 set(HIPIFY_WRAPPER_BIN_DIR ${HIPIFY_WRAPPER_DIR}/bin)
 
-#With File Reorganization , hipify(and hip)  will be installed in /opt/rocm-ver
-#instead of  /opt/rocm-ver/hip/. For maintaining backward  compatibility
-# the previous location(/opt/rocm-ver/hip/) will have soft link.
+#With File Reorganization , hipify(and hip)  will be installed in @ESYSROOT_ROCM_PATH@
+#instead of  @ESYSROOT_ROCM_PATH@/hip/. For maintaining backward  compatibility
+# the previous location(@ESYSROOT_ROCM_PATH@/hip/) will have soft link.
 #This file is for creating  soft link to binary files and install it in the  previous location
 #Note: soft link added for binary files.
 
