diff -urp HIPIFY-rocm-5.6.0.orig/CMakeLists.txt HIPIFY-rocm-5.6.0/CMakeLists.txt
--- HIPIFY-rocm-5.6.0.orig/CMakeLists.txt	2023-08-23 14:38:59.165713940 -0700
+++ HIPIFY-rocm-5.6.0/CMakeLists.txt	2023-08-23 14:39:21.325714490 -0700
@@ -124,12 +124,12 @@ if(UNIX)
     #get rid of any RPATH definations already
     set_target_properties(hipify-clang PROPERTIES INSTALL_RPATH "")
     #set RPATH for the binary
-    set_target_properties(hipify-clang PROPERTIES LINK_FLAGS "-Wl,--disable-new-dtags -Wl,--rpath,$ORIGIN/../lib" )
+    set_target_properties(hipify-clang PROPERTIES LINK_FLAGS "-Wl,--disable-new-dtags -Wl,--rpath,$ORIGIN/../@LIBDIR@" )
 
     if(FILE_REORG_BACKWARD_COMPATIBILITY)
         include(hipify-backward-compat.cmake)
     endif()
-    set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE PATH "HIP Package Installation Path")
+    set(CPACK_PACKAGING_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE PATH "HIP Package Installation Path")
     set(BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}/packages/hipify-clang)
     configure_file(packaging/hipify-clang.txt ${BUILD_DIR}/CMakeLists.txt @ONLY)
     configure_file(${CMAKE_SOURCE_DIR}/LICENSE.txt ${BUILD_DIR}/LICENSE.txt @ONLY)
diff -urp HIPIFY-rocm-5.6.0.orig/hipify-backward-compat.cmake HIPIFY-rocm-5.6.0/hipify-backward-compat.cmake
--- HIPIFY-rocm-5.6.0.orig/hipify-backward-compat.cmake	2023-05-12 18:19:43.000000000 -0700
+++ HIPIFY-rocm-5.6.0/hipify-backward-compat.cmake	2023-08-23 14:39:21.325714490 -0700
@@ -23,7 +23,7 @@ set(HIPIFY_BUILD_DIR ${CMAKE_CURRENT_BIN
 set(HIPIFY_WRAPPER_DIR ${HIPIFY_BUILD_DIR}/wrapper_dir)
 set(HIPIFY_WRAPPER_BIN_DIR ${HIPIFY_WRAPPER_DIR}/bin)
 
-#With File Reorganization , hipify(and hip)  will be installed in /opt/rocm-ver
+#With File Reorganization , hipify(and hip)  will be installed in @ESYSROOT_ROCM_PATH@
-#instead of  /opt/rocm-ver/hip/. For maintaining backward  compatibility
+#instead of  @ESYSROOT_ROCM_PATH@/hip/. For maintaining backward  compatibility
-# the previous location(/opt/rocm-ver/hip/) will have soft link.
+# the previous location(@ESYSROOT_ROCM_PATH@/hip/) will have soft link.
 #This file is for creating  soft link to binary files and install it in the  previous location
