diff -urp work.orig/common/lib/modules/fglrx/build_mod/firegl_public.c work/common/lib/modules/fglrx/build_mod/firegl_public.c
--- work.orig/common/lib/modules/fglrx/build_mod/firegl_public.c	2017-10-04 20:12:45.926028470 -0700
+++ work/common/lib/modules/fglrx/build_mod/firegl_public.c	2017-10-04 20:15:42.533527373 -0700
@@ -269,7 +269,11 @@
 
 static inline void __flush_tlb_all(void)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,7,0)
+        if (boot_cpu_has(X86_FEATURE_PGE))
+#else
         if (cpu_has_pge)
+#endif
         {
                 unsigned long flags, cr4;
                 raw_local_irq_save(flags);
