--- a/backend/src/RenderVideo.py.orig	2024-12-18 13:04:49.193567001 -0800
+++ b/backend/src/RenderVideo.py	2024-12-18 13:25:33.434379721 -0800
@@ -16,7 +16,7 @@ class Render(FFMpegRender):
     inputFile: str, The path to the input file.
     outputFile: str, The path to the output file.
     interpolateTimes: int, this sets the multiplier for the framerate when interpolating, when only upscaling this will be set to 1.
-    encoder: str, The exact name of the encoder ffmpeg will use (default=libx264)
+    encoder: str, The exact name of the encoder ffmpeg will use (default=@DEFAULT_ENCODER@)
     pixelFormat: str, The pixel format ffmpeg will use, (default=yuv420p)
 
     interpolateOptions:
@@ -48,7 +48,7 @@ class Render(FFMpegRender):
         interpolateFactor: int = 1,
         tile_size=None,
         # ffmpeg settings
-        encoder: str = "libx264",
+        encoder: str = "@DEFAULT_ENCODER@",
         pixelFormat: str = "yuv420p",
         benchmark: bool = False,
         overwrite: bool = False,
--- a/src/ui/SettingsTab.py.orig	2024-12-18 13:04:49.197566978 -0800
+++ b/src/ui/SettingsTab.py	2024-12-18 13:25:17.978468926 -0800
@@ -200,7 +200,7 @@ class Settings:
         self.defaultSettings = {
             "precision": "auto",
             "tensorrt_optimization_level": "3",
-            "encoder": "libx264",
+            "encoder": "@DEFAULT_ENCODER@",
             "preview_enabled": "True",
             "scene_change_detection_method": "pyscenedetect",
             "scene_change_detection_enabled": "True",
@@ -215,7 +215,7 @@ class Settings:
         self.allowedSettings = {
             "precision": ("auto", "float32", "float16"),
             "tensorrt_optimization_level": ("0", "1", "2", "3", "4", "5"),
-            "encoder": ("libx264", "libx265", "vp9", "av1"),
+            "encoder": (@ALLOWED_ENCODERS@),
             "preview_enabled": ("True", "False"),
             "scene_change_detection_method": (
                 "mean",
