diff -urp appimaged-8e248f5afe975b8ef65c7e3e5596ab13c6af3a4d.orig/src/main.c appimaged-8e248f5afe975b8ef65c7e3e5596ab13c6af3a4d/src/main.c
--- appimaged-8e248f5afe975b8ef65c7e3e5596ab13c6af3a4d.orig/src/main.c	2020-07-24 23:34:19.000000000 -0700
+++ appimaged-8e248f5afe975b8ef65c7e3e5596ab13c6af3a4d/src/main.c	2020-08-30 01:53:46.729167808 -0700
@@ -567,6 +567,8 @@ int main(int argc, char** argv) {
     add_dir_to_watch(g_build_filename("/opt", NULL));
     add_dir_to_watch(g_build_filename("/usr/local/bin", NULL));
 
+    add_dir_to_watch(g_build_filename("/opt/AppImage", NULL));
+
     // Watch "/Applications" on all mounted partitions, if it exists.
     // TODO: Notice when partitions are mounted and unmounted (patches welcome!)
     struct mntent* ent;
