--- a/src/CMakeLists.txt.orig	2020-08-02 03:03:35.213056609 -0700
+++ b/src/CMakeLists.txt	2020-08-02 03:08:34.594642370 -0700
@@ -9,7 +9,7 @@ pkg_check_modules(LIBAPPIMAGE libappimag
 include_directories(${LIBAPPIMAGE_INCLUDE_DIRS})
 
 add_executable(appimaged main.c notify.c notify.h)
-target_link_libraries(appimaged PRIVATE inotify-tools libappimage_static xdg-basedir dl PkgConfig::GLIB)
+target_link_libraries(appimaged PRIVATE inotify-tools appimage_static xdg-basedir dl PkgConfig::GLIB)
 
 install(
     TARGETS appimaged
