diff -x '*.rej' -x '*.orig' -urpN appimaged-9999.orig/src/CMakeLists.txt appimaged-9999/src/CMakeLists.txt
--- appimaged-9999.orig/src/CMakeLists.txt	2023-01-11 19:59:37.922748011 -0800
+++ appimaged-9999/src/CMakeLists.txt	2023-01-11 20:01:26.111024178 -0800
@@ -3,13 +3,21 @@ cmake_minimum_required(VERSION 3.6)
 
 
 find_package(PkgConfig)
+pkg_check_modules(CAIRO cairo IMPORTED_TARGET)
 pkg_check_modules(GLIB glib-2.0 IMPORTED_TARGET)
+pkg_check_modules(LIBARCHIVE libarchive IMPORTED_TARGET)
+pkg_check_modules(LIBRSVG librsvg-2.0 IMPORTED_TARGET)
+pkg_check_modules(SQUASHFUSE squashfuse_appimage IMPORTED_TARGET)
 
 pkg_check_modules(LIBAPPIMAGE libappimage REQUIRED)
 include_directories(${LIBAPPIMAGE_INCLUDE_DIRS})
 
+find_package(XdgUtils "0.1.1" REQUIRED)
+link_directories(${XDGUTILS_LIBDIR})
+
 add_executable(appimaged main.c notify.c notify.h)
 target_link_libraries(appimaged PRIVATE inotifytools appimage_static xdg-basedir dl PkgConfig::GLIB)
+target_link_libraries(appimaged PRIVATE inotifytools appimage_static appimage_hashlib xdg-basedir boost_filesystem dl stdc++ PkgConfig::CAIRO PkgConfig::GLIB PkgConfig::LIBARCHIVE PkgConfig::LIBRSVG PkgConfig::SQUASHFUSE pthread ${XDGUTILS_LINK_LIBRARIES})
 
 install(
     TARGETS appimaged
