--- a/CMakeLists.txt.orig	2020-08-01 23:12:39.324426470 -0700
+++ b/CMakeLists.txt	2020-08-01 23:15:11.230388954 -0700
@@ -6,13 +6,14 @@ project(appimaged)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 # determine Git commit ID
-execute_process(
-    COMMAND git rev-parse --short HEAD
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    OUTPUT_VARIABLE APPIMAGED_GIT_COMMIT
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-mark_as_advanced(APPIMAGED_GIT_COMMIT)
+#execute_process(
+#    COMMAND git rev-parse --short HEAD
+#    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+#    OUTPUT_VARIABLE APPIMAGED_GIT_COMMIT
+#    OUTPUT_STRIP_TRAILING_WHITESPACE
+#)
+#mark_as_advanced(APPIMAGED_GIT_COMMIT)
+set(APPIMAGED_GIT_COMMIT 8e248f5afe975b8ef65c7e3e5596ab13c6af3a4d)
 
 # set version and build number
 set(APPIMAGED_VERSION 1-alpha)
