diff -ur Urho3D-1.5.orig/Source/Urho3D/Navigation/CrowdManager.cpp Urho3D-1.5/Source/Urho3D/Navigation/CrowdManager.cpp
--- Urho3D-1.5.orig/Source/Urho3D/Navigation/CrowdManager.cpp	2016-02-11 10:25:50.077447834 -0800
+++ Urho3D-1.5/Source/Urho3D/Navigation/CrowdManager.cpp	2016-02-11 10:33:21.013635418 -0800
@@ -65,10 +65,13 @@
     numAreas_.Reserve(DT_CROWD_MAX_QUERY_FILTER_TYPE);
     for (unsigned i = 0; i < DT_CROWD_MAX_QUERY_FILTER_TYPE; ++i)
         numAreas_.Push(0);
+
+    agents_ = (dtCrowdAgent**)dtAlloc(sizeof(dtCrowdAgent*)*DEFAULT_MAX_AGENTS, DT_ALLOC_PERM);
 }
 
 CrowdManager::~CrowdManager()
 {
+    dtFree(agents_);
     dtFreeCrowd(crowd_);
     crowd_ = 0;
 }
@@ -642,6 +645,12 @@
 {
     assert(crowd_ && navigationMesh_);
     URHO3D_PROFILE(UpdateCrowd);
+    int nagents;
+    nagents = crowd_->getActiveAgents(agents_, maxAgents_);
+    for (int i = 0; i < nagents; i++) {
+        dtCrowdAgent* ag = agents_[i];
+        CrowdAgentUpdateCallback(ag, delta);
+    }
     crowd_->update(delta, 0);
 }
 
diff -ur Urho3D-1.5.orig/Source/Urho3D/Navigation/CrowdManager.h Urho3D-1.5/Source/Urho3D/Navigation/CrowdManager.h
--- Urho3D-1.5.orig/Source/Urho3D/Navigation/CrowdManager.h	2016-02-11 10:25:50.077447834 -0800
+++ Urho3D-1.5/Source/Urho3D/Navigation/CrowdManager.h	2016-02-11 10:33:37.269262620 -0800
@@ -193,6 +193,7 @@
     PODVector<unsigned> numAreas_;
     /// Number of obstacle avoidance types configured in the crowd. Limit to DT_CROWD_MAX_OBSTAVOIDANCE_PARAMS.
     unsigned numObstacleAvoidanceTypes_;
+    dtCrowdAgent** agents_;
 };
 
 }
