diff -ur Urho3D-1.5.orig/Source/Urho3D/Resource/JSONFile.cpp Urho3D-1.5/Source/Urho3D/Resource/JSONFile.cpp
--- Urho3D-1.5.orig/Source/Urho3D/Resource/JSONFile.cpp	2016-02-04 17:16:23.177006611 -0800
+++ Urho3D-1.5/Source/Urho3D/Resource/JSONFile.cpp	2016-02-04 17:35:07.292047383 -0800
@@ -198,8 +198,10 @@
             rapidjsonValue.SetObject();
             for (JSONObject::ConstIterator i = jsonObject.Begin(); i != jsonObject.End(); ++i)
             {
-                const char* name = i->first_.CString();
+                const char* _name = i->first_.CString();
                 rapidjson::Value value;
+                rapidjson::Value name;
+		name = _name;
                 rapidjsonValue.AddMember(name, value, allocator);
                 ToRapidjsonValue(rapidjsonValue[name], i->second_, allocator);
             }
