diff -urp a/Source/Urho3D/IO/Compression.cpp b/Source/Urho3D/IO/Compression.cpp
--- a/Source/Urho3D/IO/Compression.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/IO/Compression.cpp	2020-01-09 11:46:02.850924292 -0800
@@ -28,8 +28,8 @@
 #include "../IO/Serializer.h"
 #include "../IO/VectorBuffer.h"
 
-#include <LZ4/lz4.h>
-#include <LZ4/lz4hc.h>
+#include <lz4.h>
+#include <lz4hc.h>
 
 namespace Urho3D
 {
diff -urp a/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp b/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp
--- a/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp	2020-01-09 11:46:02.850924292 -0800
@@ -39,7 +39,7 @@
 #include "../Scene/Scene.h"
 #include "../Scene/SceneEvents.h"
 
-#include <LZ4/lz4.h>
+#include <lz4.h>
 #include <Detour/DetourNavMesh.h>
 #include <Detour/DetourNavMeshBuilder.h>
 #include <DetourTileCache/DetourTileCache.h>
