diff -ur urho3d-9999.orig/Source/Urho3D/Network/Network.h urho3d-9999/Source/Urho3D/Network/Network.h
--- urho3d-9999.orig/Source/Urho3D/Network/Network.h	2016-03-01 07:25:19.892887262 -0800
+++ urho3d-9999/Source/Urho3D/Network/Network.h	2016-03-01 07:29:27.370427207 -0800
@@ -22,6 +22,8 @@
 
 #pragma once
 
+
+#include <kNet/MessageConnection.h>
 #include "../Container/HashSet.h"
 #include "../Core/Object.h"
 #include "../IO/VectorBuffer.h"
