diff -urp -x '*.rej' -x '*.orig' a/Source/Urho3D/Resource/JSONFile.cpp b/Source/Urho3D/Resource/JSONFile.cpp
--- a/Source/Urho3D/Resource/JSONFile.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Resource/JSONFile.cpp	2020-01-09 11:58:18.129085298 -0800
@@ -197,8 +197,9 @@ static void ToRapidjsonValue(rapidjson::
             rapidjsonValue.SetObject();
             for (JSONObject::ConstIterator i = jsonObject.Begin(); i != jsonObject.End(); ++i)
             {
-                const char* name = i->first_.CString();
+                const char* _name = i->first_.CString();
                 rapidjson::Value value;
+                rapidjson::Value name(_name,allocator);
                 ToRapidjsonValue(value, i->second_, allocator);
                 rapidjsonValue.AddMember(StringRef(name), value, allocator);
             }
