diff -urp a/Source/ThirdParty/nanodbc/nanodbc/nanodbc.cpp b/Source/ThirdParty/nanodbc/nanodbc/nanodbc.cpp
--- a/Source/ThirdParty/nanodbc/nanodbc/nanodbc.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/ThirdParty/nanodbc/nanodbc/nanodbc.cpp	2020-01-09 20:43:12.723963494 -0800
@@ -13,7 +13,7 @@
 #pragma warning(disable : 4996) // warning about deprecated declaration
 #endif
 
-#include <nanodbc/nanodbc.h>
+#include <nanodbc.h>
 
 #include <algorithm>
 #include <clocale>
diff -urp a/Source/Urho3D/Database/ODBC/ODBCConnection.cpp b/Source/Urho3D/Database/ODBC/ODBCConnection.cpp
--- a/Source/Urho3D/Database/ODBC/ODBCConnection.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Database/ODBC/ODBCConnection.cpp	2020-01-09 20:43:12.723963494 -0800
@@ -32,6 +32,7 @@
 #endif
 
 #include <sqlext.h>
+#include <nanodbc.h>
 
 namespace Urho3D
 {
diff -urp a/Source/Urho3D/Database/ODBC/ODBCConnection.h b/Source/Urho3D/Database/ODBC/ODBCConnection.h
--- a/Source/Urho3D/Database/ODBC/ODBCConnection.h	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Database/ODBC/ODBCConnection.h	2020-01-09 20:43:12.703962720 -0800
@@ -25,7 +25,7 @@
 #include "../../Core/Object.h"
 #include "../../Database/DbResult.h"
 
-#include <nanodbc/nanodbc.h>
+#include <nanodbc.h>
 
 namespace Urho3D
 {
diff -urp a/Source/Urho3D/Database/ODBC/ODBCResult.h b/Source/Urho3D/Database/ODBC/ODBCResult.h
--- a/Source/Urho3D/Database/ODBC/ODBCResult.h	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Database/ODBC/ODBCResult.h	2020-01-09 20:43:12.703962720 -0800
@@ -24,7 +24,7 @@
 
 #include "../../Core/Variant.h"
 
-#include <nanodbc/nanodbc.h>
+#include <nanodbc.h>
 
 namespace Urho3D
 {
