diff -urp a/Source/Urho3D/Urho2D/SpriterData2D.cpp b/Source/Urho3D/Urho2D/SpriterData2D.cpp
--- a/Source/Urho3D/Urho2D/SpriterData2D.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Urho2D/SpriterData2D.cpp	2020-01-09 11:50:25.410979934 -0800
@@ -25,7 +25,7 @@
 #include "../Math/MathDefs.h"
 #include "../Urho2D/SpriterData2D.h"
 
-#include <PugiXml/pugixml.hpp>
+#include <pugixml.hpp>
 
 #include <cstring>
 
