diff -urp a/Source/Urho3D/Resource/XMLElement.cpp b/Source/Urho3D/Resource/XMLElement.cpp
--- a/Source/Urho3D/Resource/XMLElement.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Resource/XMLElement.cpp	2020-01-09 11:50:25.400979552 -0800
@@ -26,7 +26,7 @@
 #include "../IO/Log.h"
 #include "../Resource/XMLFile.h"
 
-#include <PugiXml/pugixml.hpp>
+#include <pugixml.hpp>
 
 #include "../DebugNew.h"
 
diff -urp a/Source/Urho3D/Resource/XMLFile.cpp b/Source/Urho3D/Resource/XMLFile.cpp
--- a/Source/Urho3D/Resource/XMLFile.cpp	2019-10-05 21:46:51.000000000 -0700
+++ b/Source/Urho3D/Resource/XMLFile.cpp	2020-01-09 11:50:25.390979168 -0800
@@ -32,7 +32,7 @@
 #include "../Resource/ResourceCache.h"
 #include "../Resource/XMLFile.h"
 
-#include <PugiXml/pugixml.hpp>
+#include <pugixml.hpp>
 
 #include "../DebugNew.h"
 
