--- surf.c.orig	2015-05-16 00:04:49.203868291 -0700
+++ surf.c	2015-05-16 00:01:51.030039400 -0700
@@ -560,7 +560,7 @@
                gpointer          user_data) {
 	static guint32 unique_id = 0;
 	webkit_web_context_set_web_extensions_directory(context, SURF_WEB_EXTENSIONS_DIR);
-        webkit_web_context_set_web_extensions_initialization_user_data(context, g_variant_new_uint32(unique_id++));
+        webkit_web_context_set_web_extensions_initialization_user_data(context, g_variant_new_string(scriptfile));
 }
 
 static void
@@ -699,6 +699,7 @@
 	WebKitWebContext *context;
 	WebKitSettings *settings;
 	GdkGeometry hints = { 1, 1 };
+	WebKitUserContentManager *usercontent;
 	char *uri, *ua;
 
 	if(!(c = calloc(1, sizeof(Client))))
@@ -751,7 +752,8 @@
 			G_CALLBACK(initializewebextensions), NULL);
 
 	/* Webview */
-	c->view = WEBKIT_WEB_VIEW(webkit_web_view_new());
+	usercontent = webkit_user_content_manager_new();
+	c->view = WEBKIT_WEB_VIEW(webkit_web_view_new_with_user_content_manager(usercontent));
 	context = webkit_web_view_get_context(c->view);
 	g_signal_connect(G_OBJECT(c->view), //
 			"notify::title",
@@ -856,7 +858,7 @@
 	WebKitUserScript *script;
 	if(g_file_get_contents(scriptfile, &source, NULL, NULL)) {
 		script = webkit_user_script_new(source, WEBKIT_USER_CONTENT_INJECT_ALL_FRAMES, WEBKIT_USER_SCRIPT_INJECT_AT_DOCUMENT_END, NULL, NULL);
-		webkit_user_content_manager_add_script(contentmanager, script);
+		webkit_user_content_manager_add_script(usercontent, script);
 		g_free(source);
 		webkit_user_script_unref(script);
 	}
@@ -865,7 +867,7 @@
 	WebKitUserStyleSheet *style;
 	if(g_file_get_contents(stylefile, &source, NULL, NULL)) {
 		style = webkit_user_style_sheet_new(source, WEBKIT_USER_CONTENT_INJECT_ALL_FRAMES, WEBKIT_USER_STYLE_LEVEL_USER, NULL, NULL);
-		webkit_user_content_manager_add_style_sheet(contentmanager, style);
+		webkit_user_content_manager_add_style_sheet(usercontent, style);
 		g_free(source);
 		webkit_user_style_sheet_unref(style);
 	}
--- surfext.c.orig	2015-05-15 22:18:49.707858083 -0700
+++ surfext.c	2015-05-16 00:03:54.163758901 -0700
@@ -22,7 +22,7 @@
 #include <webkit2/webkit-web-extension.h>
 #include <gmodule.h>
 
-static char *scriptfile     = "~/.surf/script.js";
+static char *scriptfile     = NULL;
 
 static
 void windowobjectcleared (WebKitScriptWorld *world,
@@ -73,7 +73,9 @@
 G_MODULE_EXPORT void
 webkit_web_extension_initialize_with_user_data(WebKitWebExtension *extension,
                                 GVariant *user_data)
-{
+{	
+    scriptfile = g_variant_get_string(user_data, NULL);
+    printf(scriptfile);
     g_signal_connect (extension, "page-created", 
                       G_CALLBACK (web_page_created_callback), 
                       NULL);
