Only in surf-0.6: surf
diff -u surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2014-06-21 18:55:36.674320013 +0000
+++ surf-0.6/surf.c	2014-06-21 19:14:16.756302083 +0000
@@ -799,6 +799,8 @@
 			enablespatialbrowsing, NULL);
 	g_object_set(G_OBJECT(settings), "enable-developer-extras",
 			enableinspector, NULL);
+	g_object_set(G_OBJECT(settings), "enable-smooth-scrolling",
+			TRUE, NULL);
 
 	if(enableinspector) {
 		c->inspector = WEBKIT_WEB_INSPECTOR(
Only in surf-0.6: surf.o
