diff -u surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2014-06-23 13:41:00.784011752 -0700
+++ surf-0.6/surf.c	2014-06-23 14:41:29.444396571 -0700
@@ -1052,8 +1052,13 @@
 
 	/* proxy */
 	if((proxy = getenv("http_proxy")) && strcmp(proxy, "")) {
-		new_proxy = g_strrstr(proxy, "http://") ? g_strdup(proxy) :
-			g_strdup_printf("http://%s", proxy);
+		if (g_strrstr(proxy, "https://")) {
+			new_proxy = g_strdup(proxy);
+		} else if (g_strrstr(proxy, "http://")) {
+			new_proxy = g_strdup(proxy);
+		} else {
+			new_proxy = g_strdup_printf("http://%s", proxy);
+		}
 		puri = soup_uri_new(new_proxy);
 		g_object_set(G_OBJECT(s), "proxy-uri", puri, NULL);
 		soup_uri_free(puri);
