diff -ur surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2016-07-28 15:26:53.529778368 -0700
+++ surf-0.6/surf.c	2016-07-28 15:28:20.312781228 -0700
@@ -178,6 +178,7 @@
 	bscript = FALSE;
 	bajax = FALSE;
 
+	/*
         WebKitURIResponse *response = webkit_download_get_response(download);
 	if (response) {
 		const gchar *uri = webkit_uri_response_get_uri(response);
@@ -199,7 +200,7 @@
 			bajax = TRUE;
 		} else {
 		}
-	}
+	}*/
 }
 
 static void
