diff -ur surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2016-07-29 09:17:06.947894356 -0700
+++ surf-0.6/surf.c	2016-07-29 09:18:19.272896739 -0700
@@ -1039,7 +1039,7 @@
 
 	/* ssl */
 	//g_object_set(G_OBJECT(s), "ssl-ca-file", cafile, NULL);
-	g_object_set(G_OBJECT(s), "ssl-strict", strictssl, NULL);
+	//g_object_set(G_OBJECT(s), "ssl-strict", strictssl, NULL);
 
 	/* proxy */
 	if((proxy = getenv("http_proxy")) && strcmp(proxy, "")) {
