--- surf-0.6.orig/surfext.c	2015-10-27 20:46:15.837205640 -0700
+++ surf-0.6/surfext.c	2015-10-27 20:47:54.123954795 -0700
@@ -29,6 +29,7 @@
 #include <sys/time.h>
 
 #define SCRIPTFILE "/etc/surf/scripts/adblock/blocksrv.py"
+#define DEBUG_TIMER 0
 
 static JsonParser *parserd, *parserdg;
 static JsonNode *rootd, *rootdg;
@@ -340,10 +341,12 @@
 static gboolean
 decide(pat_scan_data *psd)
 {
+#if DEBUG_TIMER
 	struct timeval st, et, rt;
 
 	//printf("processing %s", eleuri);
 	gettimeofday(&st, NULL);
+#endif
 
 	//general scan
 	JsonArray *a = json_node_get_array(rootdg);
@@ -359,11 +362,13 @@
 	json_array_foreach_element (ra, pat_scan, psd);
 	json_node_free(results);
 
+#if DEBUG_TIMER
 	struct timeval tv;
 
 	gettimeofday(&et, NULL);
 	timersub(&et, &st, &rt);
 	printf("--- %ld.%ld seconds ---\n", rt.tv_sec, rt.tv_usec);
+#endif
 
 	return psd->evalblock;
 }
