diff -urp src.orig/atmi-backward-compat.cmake src/atmi-backward-compat.cmake
--- src.orig/atmi-backward-compat.cmake	2022-05-17 08:16:19.000000000 -0700
+++ src/atmi-backward-compat.cmake	2023-08-21 11:51:24.821177182 -0700
@@ -24,9 +24,9 @@ set(ATMI_WRAPPER_DIR ${ATMI_BUILD_DIR}/w
 set(ATMI_WRAPPER_INC_DIR ${ATMI_WRAPPER_DIR}/include)
 set(ATMI_WRAPPER_BIN_DIR ${ATMI_WRAPPER_DIR}/bin)
 if(CMAKE_BUILD_TYPE MATCHES Debug)
-  set(ATMI_LIB_NAME "lib-debug")
+  set(ATMI_LIB_NAME "@LIBDIR@-debug")
 else()
-  set(ATMI_LIB_NAME "lib")
+  set(ATMI_LIB_NAME "@LIBDIR@")
 endif()
 set(ATMI_WRAPPER_LIB_DIR ${ATMI_WRAPPER_DIR}/${ATMI_LIB_NAME})
 
diff -urp src.orig/cmake_modules/FindLibElf.cmake src/cmake_modules/FindLibElf.cmake
--- src.orig/cmake_modules/FindLibElf.cmake	2022-05-17 08:16:19.000000000 -0700
+++ src/cmake_modules/FindLibElf.cmake	2023-08-21 11:51:24.821177182 -0700
@@ -23,8 +23,8 @@ find_library (LIBELF_LIBRARIES
     elf
     PATHS
     /usr/lib/x86_64-linux-gnu
-    /usr/lib
-    /usr/local/lib
+    /usr/@LIBDIR@
+    /usr/local/@LIBDIR@
     ENV LIBRARY_PATH
     ENV LD_LIBRARY_PATH)
 
diff -urp src.orig/cmake_modules/FindROCm.cmake src/cmake_modules/FindROCm.cmake
--- src.orig/cmake_modules/FindROCm.cmake	2022-05-17 08:16:19.000000000 -0700
+++ src/cmake_modules/FindROCm.cmake	2023-08-21 11:51:24.821177182 -0700
@@ -41,15 +41,15 @@ find_library(
   ROCR_LIBRARY
     hsa-runtime64
   HINTS
-    ${ROC_DIR}/lib
+    ${ROC_DIR}/@LIBDIR@
     ${ROC_DIR}
-    ${ROCR_DIR}/lib
+    ${ROCR_DIR}/@LIBDIR@
     ${ROCR_DIR}
-    /opt/rocm/lib
-    /opt/rocm/hsa/lib
-    /usr/local/lib
+    /opt/rocm/@LIBDIR@
+    /opt/rocm/hsa/@LIBDIR@
+    /usr/local/@LIBDIR@
     /usr/lib/x86_64-linux-gnu
-    /usr/lib
+    /usr/@LIBDIR@
     ENV LIBRARY_PATH
     ENV LD_LIBRARY_PATH
 )
@@ -57,15 +57,15 @@ find_library(
   ROCT_LIBRARY
     hsakmt
   HINTS
-    ${ROC_DIR}/lib
+    ${ROC_DIR}/@LIBDIR@
     ${ROC_DIR}
-    ${ROCT_DIR}/lib
+    ${ROCT_DIR}/@LIBDIR@
     ${ROCT_DIR}
-    /opt/rocm/lib
-    /opt/rocm/hsa/lib
-    /usr/local/lib
+    /opt/rocm/@LIBDIR@
+    /opt/rocm/hsa/@LIBDIR@
+    /usr/local/@LIBDIR@
     /usr/lib/x86_64-linux-gnu
-    /usr/lib
+    /usr/@LIBDIR@
     ENV LIBRARY_PATH
     ENV LD_LIBRARY_PATH
 )
diff -urp src.orig/compiler/CMakeLists.txt src/compiler/CMakeLists.txt
--- src.orig/compiler/CMakeLists.txt	2022-05-17 08:16:19.000000000 -0700
+++ src/compiler/CMakeLists.txt	2023-08-21 11:51:24.821177182 -0700
@@ -65,11 +65,11 @@ if(ROCM_FOUND)
             include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
             include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../../include)
 
-            # Install plugin under the lib destination folder.
+            # Install plugin under the @LIBDIR@ destination folder.
             if(CMAKE_BUILD_TYPE MATCHES Debug)
-                install(TARGETS atmi_cplugin LIBRARY DESTINATION "lib-debug" COMPONENT cplugin )
+                install(TARGETS atmi_cplugin LIBRARY DESTINATION "@LIBDIR@-debug" COMPONENT cplugin )
             else()
-                install(TARGETS atmi_cplugin LIBRARY DESTINATION "lib" COMPONENT cplugin )
+                install(TARGETS atmi_cplugin LIBRARY DESTINATION "@LIBDIR@" COMPONENT cplugin )
             endif()
             
             INSTALL(FILES
@@ -91,7 +91,7 @@ if(ROCM_FOUND)
             )
 
             if (NOT CMAKE_INSTALL_RPATH)
-              set(CMAKE_INSTALL_RPATH "$ORIGIN;$ORIGIN/../../hsa/lib;$ORIGIN/../../lib;$ORIGIN/../../lib64;$ORIGIN/../lib64")
+              set(CMAKE_INSTALL_RPATH "$ORIGIN;$ORIGIN/../../hsa/@LIBDIR@;$ORIGIN/../../@LIBDIR@;$ORIGIN/../../@LIBDIR@;$ORIGIN/../@LIBDIR@")
             endif ()
 
        else()
diff -urp src.orig/device_runtime/bc.cmake src/device_runtime/bc.cmake
--- src.orig/device_runtime/bc.cmake	2022-05-17 08:16:19.000000000 -0700
+++ src/device_runtime/bc.cmake	2023-08-21 11:51:24.821177182 -0700
@@ -190,7 +190,7 @@ macro(add_bc_library name dir)
 
   add_custom_command(
     OUTPUT ${name}-${mcpu}.bc
-    COMMAND /bin/cp optout.${mcpu}.bc ${ATMI_RUNTIME_PATH}/lib/${name}-${mcpu}.bc
+    COMMAND /bin/cp optout.${mcpu}.bc ${ATMI_RUNTIME_PATH}/@LIBDIR@/${name}-${mcpu}.bc
     DEPENDS optout.${mcpu}.bc
   )
   add_custom_target(${name}-${mcpu} ALL DEPENDS ${name}-${mcpu}.bc)
diff -urp src.orig/device_runtime/CMakeLists.txt src/device_runtime/CMakeLists.txt
--- src.orig/device_runtime/CMakeLists.txt	2023-08-21 11:51:07.673176757 -0700
+++ src/device_runtime/CMakeLists.txt	2023-08-21 11:51:46.497177720 -0700
@@ -32,7 +32,7 @@ if(LLVM_FOUND)
   if (CLANG_INSTALL_PREFIX)
     set(CLANG_BINDIR ${CLANG_INSTALL_PREFIX}/bin)
     set(CLANG_INCDIR ${CLANG_INSTALL_PREFIX}/include)
-    set(CLANG_LIBDIR ${CLANG_INSTALL_PREFIX}/lib)
+    set(CLANG_LIBDIR ${CLANG_INSTALL_PREFIX}/@LIBDIR@)
     set(AMDLLVM ${CLANG_INSTALL_PREFIX})
   else()
     set(CLANG_BINDIR ${LLVM_BUILD_BINARY_DIR}/bin)
@@ -68,9 +68,10 @@ find_path(ATMI_DE_DEP_LIBHSA_LIBRARIES
     "ockl.bc"
     "ockl.amdgcn.bc"
   PATHS
+    "${AMD_DEVICE_LIBS_PREFIX}/@LIBDIR@/amdgcn/bitcode"
     "${AMD_DEVICE_LIBS_PREFIX}/amdgcn/bitcode"
-    "${AMD_DEVICE_LIBS_PREFIX}/lib/bitcode"
-    "${AMD_DEVICE_LIBS_PREFIX}/lib"
+    "${AMD_DEVICE_LIBS_PREFIX}/@LIBDIR@/bitcode"
+    "${AMD_DEVICE_LIBS_PREFIX}/@LIBDIR@"
   NO_DEFAULT_PATH)
 
 get_filename_component (ATMI_DE_DEP_LIBHSA_LIBRARIES_DIRS ${ATMI_DE_DEP_LIBHSA_LIBRARIES} DIRECTORY)
@@ -116,7 +117,7 @@ add_custom_target(device_header ALL DEPE
 # append to the configure file
 file(APPEND ${ATMI_RUNTIME_PATH}/atmi-config.mak "AMDLLVM=${AMDLLVM}\n")
 file(APPEND ${ATMI_RUNTIME_PATH}/atmi-config.mak "AMDGPU_TARGET_TRIPLE=${AMDGPU_TARGET_TRIPLE}\n")
-# cloc assumes the /path/to/rocm/lib (with the /lib suffix in the path)
+# cloc assumes the /path/to/rocm/@LIBDIR@ (with the /@LIBDIR@ suffix in the path)
 file(APPEND ${ATMI_RUNTIME_PATH}/atmi-config.mak "ROCM_DEVICE_PATH=${ROCM_DEVICE_PATH}/amdgcn/bitcode\n")
 
 target_sources(atmi_runtime PRIVATE
@@ -133,7 +134,7 @@ foreach(mcpu ${mcpus})
   set(optimization_level 2)
 
   # set output dir for .bc files
-  set(OUTPUTDIR ${ATMI_RUNTIME_PATH}/lib)
+  set(OUTPUTDIR ${ATMI_RUNTIME_PATH}/@LIBDIR@)
 
   # create output dir for include files
   execute_process(COMMAND "/bin/mkdir" "-p" "${OUTPUTDIR}")
@@ -153,12 +154,12 @@ foreach(mcpu ${mcpus})
   if(CMAKE_BUILD_TYPE MATCHES Debug)
     install(FILES
       ${OUTPUTDIR}/atmi-${mcpu}.bc
-      DESTINATION lib-debug COMPONENT device_runtime
+      DESTINATION @LIBDIR@-debug COMPONENT device_runtime
       )
   else()
     install(FILES
       ${OUTPUTDIR}/atmi-${mcpu}.bc
-      DESTINATION lib64 COMPONENT device_runtime
+      DESTINATION @LIBDIR@ COMPONENT device_runtime
       )
   endif()
 endforeach()
diff -urp src.orig/runtime/core/CMakeLists.txt src/runtime/core/CMakeLists.txt
--- src.orig/runtime/core/CMakeLists.txt	2023-08-21 11:51:07.677176757 -0700
+++ src/runtime/core/CMakeLists.txt	2023-08-21 11:51:57.809178001 -0700
@@ -60,7 +60,7 @@ add_definitions(-DROCM_VERSION_MINOR=${R
 add_definitions(-fpic)
 
 # set output dir for the .so file
-set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../lib)
+set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../@LIBDIR@)
 
 #set soname version number
 get_version ( "1.0.0" )
@@ -117,7 +117,7 @@ target_link_libraries(
 )
 
 if (NOT CMAKE_INSTALL_RPATH)
-  set(CMAKE_INSTALL_RPATH "$ORIGIN;$ORIGIN/../../hsa/lib;$ORIGIN/../../lib;$ORIGIN/../../lib64;$ORIGIN/../lib64")
+  set(CMAKE_INSTALL_RPATH "$ORIGIN;$ORIGIN/../../hsa/@LIBDIR@;$ORIGIN/../../@LIBDIR@;$ORIGIN/../../@LIBDIR@;$ORIGIN/../@LIBDIR@")
 endif ()
 
 # set output dir for .h files
@@ -140,11 +140,11 @@ add_custom_command(
 
 add_custom_target(core_header ALL DEPENDS ${OUTPUT_INC_DIRECTORY}/atmi.h ${OUTPUT_INC_DIRECTORY}/atmi_runtime.h)
 
-# Install plugin under the lib destination folder.
+# Install plugin under the @LIBDIR@ destination folder.
 if(CMAKE_BUILD_TYPE MATCHES Debug)
-  install(TARGETS atmi_runtime LIBRARY DESTINATION "lib-debug" COMPONENT runtime )
+  install(TARGETS atmi_runtime LIBRARY DESTINATION "@LIBDIR@-debug" COMPONENT runtime )
 else()
-  install(TARGETS atmi_runtime LIBRARY DESTINATION "lib64" COMPONENT runtime )
+  install(TARGETS atmi_runtime LIBRARY DESTINATION "@LIBDIR@" COMPONENT runtime )
 endif()
 
 INSTALL(FILES
