diff -urp nxjson-d2c6fba9d5b0d445722105dd2a64062c1309ac86.orig/CMakeLists.txt nxjson-d2c6fba9d5b0d445722105dd2a64062c1309ac86/CMakeLists.txt
--- nxjson-d2c6fba9d5b0d445722105dd2a64062c1309ac86.orig/CMakeLists.txt	2020-09-27 10:33:16.000000000 -0700
+++ nxjson-d2c6fba9d5b0d445722105dd2a64062c1309ac86/CMakeLists.txt	2020-11-04 21:58:26.409665904 -0800
@@ -12,5 +12,5 @@ if(BUILD_TEST)
     add_subdirectory(test)
 endif()
 
-install(TARGETS nxjson ARCHIVE DESTINATION lib)
+install(TARGETS nxjson ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${CMAKE_CURRENT_LIST_DIR}/nxjson.h DESTINATION include)
