diff -urp "rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda.orig/Rhino.Mocks.Tests/Rhino.Mocks.Tests 2.0.csproj" "rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda/Rhino.Mocks.Tests/Rhino.Mocks.Tests 2.0.csproj"
--- "rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda.orig/Rhino.Mocks.Tests/Rhino.Mocks.Tests 2.0.csproj"	2016-11-09 04:29:37.000000000 -0800
+++ "rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda/Rhino.Mocks.Tests/Rhino.Mocks.Tests 2.0.csproj"	2017-01-29 23:54:19.017872021 -0800
@@ -102,6 +102,9 @@
     <Reference Include="System.Web">
       <Name>System.Web</Name>
     </Reference>
+    <Reference Include="System.Web.ApplicationServices">
+      <Name>System.Web.ApplicationServices</Name>
+    </Reference>
     <Reference Include="System.Windows.Forms" />
     <Reference Include="System.Xml">
       <Name>System.XML</Name>
@@ -402,4 +405,4 @@
     </BootstrapperPackage>
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-</Project>
\ No newline at end of file
+</Project>
diff -urp rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda.orig/Rhino.Mocks.Tests/Rhino.Mocks.Tests.csproj rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda/Rhino.Mocks.Tests/Rhino.Mocks.Tests.csproj
--- rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda.orig/Rhino.Mocks.Tests/Rhino.Mocks.Tests.csproj	2016-11-09 04:29:37.000000000 -0800
+++ rhino-mocks-b1bd7bed0d17a9422e1f993163764637c5023dda/Rhino.Mocks.Tests/Rhino.Mocks.Tests.csproj	2017-01-29 23:54:20.729938426 -0800
@@ -123,6 +123,9 @@
     <Reference Include="System.Web">
       <Name>System.Web</Name>
     </Reference>
+    <Reference Include="System.Web.ApplicationServices">
+      <Name>System.Web.ApplicationServices</Name>
+    </Reference>
     <Reference Include="System.Windows.Forms" />
     <Reference Include="System.Xml">
       <Name>System.XML</Name>
@@ -453,4 +456,4 @@
     </None>
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-</Project>
\ No newline at end of file
+</Project>
