--- a/surf.c.orig	2019-09-25 15:20:42.057920777 -0700
+++ b/surf.c	2019-09-25 15:27:46.601295192 -0700
@@ -35,7 +35,7 @@
 #define LENGTH(x)               (sizeof(x) / sizeof(x[0]))
 #define CLEANMASK(mask)         (mask & (MODKEY|GDK_SHIFT_MASK))
 
-enum { AtomFind, AtomGo, AtomUri, AtomLast };
+enum { AtomFind, AtomGo, AtomUri, AtomScriptFile, AtomLast };
 
 enum {
 	OnDoc   = WEBKIT_HIT_TEST_RESULT_CONTEXT_DOCUMENT,
@@ -328,6 +328,7 @@ setup(void)
 	atoms[AtomFind] = XInternAtom(dpy, "_SURF_FIND", False);
 	atoms[AtomGo] = XInternAtom(dpy, "_SURF_GO", False);
 	atoms[AtomUri] = XInternAtom(dpy, "_SURF_URI", False);
+	atoms[AtomScriptFile] = XInternAtom(dpy, "_SURF_SCRIPT", False);
 
 	gtk_init(NULL, NULL);
 
@@ -567,6 +568,7 @@ loaduri(Client *c, const Arg *a)
 
 	setatom(c, AtomUri, url);
 
+	updatewinid(c);
 	if (strcmp(url, geturi(c)) == 0) {
 		reload(c, a);
 	} else {
@@ -1316,6 +1318,15 @@ processx(GdkXEvent *e, GdkEvent *event,
 				loaduri(c, &a);
 
 				return GDK_FILTER_REMOVE;
+			} else if(ev->atom == atoms[AtomScriptFile]) {
+				a.v = getatom(c, AtomScriptFile);
+				char* source;
+				if(g_file_get_contents(a.v, &source, NULL, NULL)) {
+					webkit_web_view_run_javascript(c->view, source, NULL, NULL, NULL);
+					g_free(source);
+		                }
+				//runscript(a.v, c->view);
+				return GDK_FILTER_REMOVE;
 			}
 		}
 	}
@@ -1404,6 +1415,7 @@ showview(WebKitWebView *v, Client *c)
 
 	setatom(c, AtomFind, "");
 	setatom(c, AtomUri, "about:blank");
+	setatom(c, AtomScriptFile, "");
 }
 
 GtkWidget *
@@ -1516,6 +1528,7 @@ loadchanged(WebKitWebView *v, WebKitLoad
 		seturiparameters(c, uri, loadcommitted);
 		c->https = webkit_web_view_get_tls_info(c->view, &c->cert,
 		                                        &c->tlserr);
+		PAGE_LOAD_COMMITTED(c);
 		break;
 	case WEBKIT_LOAD_FINISHED:
 		seturiparameters(c, uri, loadfinished);
@@ -1525,6 +1538,7 @@ loadchanged(WebKitWebView *v, WebKitLoad
 		    enablescrollbars ? "auto" : "hidden");
 		*/
 		runscript(c);
+		PAGE_LOAD_FINISHED(c);
 		break;
 	}
 	updatetitle(c);
