--- surfext.c.orig	2015-06-18 08:42:20.192850771 -0700
+++ surfext.c	2015-06-18 08:43:40.385860678 -0700
@@ -466,7 +466,7 @@
 webkit_web_extension_initialize_with_user_data(WebKitWebExtension *extension,
                                 GVariant *user_data)
 {
- g_signal_connect (extension, "page-created", 
+ /*g_signal_connect (extension, "page-created", 
                       G_CALLBACK (web_page_created_callback), 
-                      NULL);
+                      NULL);*/
 }
