diff -ur surf-0.6.orig/config.h surf-0.6/config.h
--- surf-0.6.orig/config.h	2016-07-28 16:17:30.792878453 -0700
+++ surf-0.6/config.h	2016-07-28 17:52:09.717065587 -0700
@@ -25,7 +25,7 @@
 static Bool enableinspector = TRUE;
 static Bool enablegeolocation = TRUE;
 static Bool loadimages = TRUE;
-//static Bool hidebackground  = FALSE;
+static Bool hidebackground  = FALSE;
 
 #define RIPSTREAM(p, q) { \
 	.v = (char *[]){ "/bin/sh", "-c", \
