diff -ur surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2016-07-28 15:36:50.013798023 -0700
+++ surf-0.6/surf.c	2016-07-28 15:41:58.108808176 -0700
@@ -766,18 +766,18 @@
 				NULL);
 		g_object_set(G_OBJECT(settings), "enable-webgl", TRUE,
 				NULL);
-		g_object_set(G_OBJECT(settings), "enable-spatial-navigation",
-				enablespatialbrowsing, NULL);
+		//g_object_set(G_OBJECT(settings), "enable-spatial-navigation",
+		//		enablespatialbrowsing, NULL);
 		g_object_set(G_OBJECT(settings), "enable-developer-extras",
 				enableinspector, NULL);
 		g_object_set(G_OBJECT(settings), "enable-smooth-scrolling",
 				TRUE, NULL);
-	        g_object_set(G_OBJECT(settings), "enable-accelerated-2d-canvas",
-	                        TRUE, NULL);
-	        g_object_set(G_OBJECT(settings), "enable-mediasource",
-	                        TRUE, NULL);
-	        g_object_set(G_OBJECT(settings), "enable-media-stream",
-	                        TRUE, NULL);
+	        //g_object_set(G_OBJECT(settings), "enable-accelerated-2d-canvas",
+	        //                TRUE, NULL);
+	        //g_object_set(G_OBJECT(settings), "enable-mediasource",
+	        //                TRUE, NULL);
+	        //g_object_set(G_OBJECT(settings), "enable-media-stream",
+	        //                TRUE, NULL);
 
 		webkit_web_context_set_tls_errors_policy(context, strictssl ? WEBKIT_TLS_ERRORS_POLICY_FAIL : WEBKIT_TLS_ERRORS_POLICY_IGNORE);
 
@@ -1035,9 +1035,9 @@
 	WebKitWebContext *defcontext = webkit_web_context_get_default();
 	//webkit_web_context_set_process_model(defcontext, WEBKIT_PROCESS_MODEL_MULTIPLE_SECONDARY_PROCESSES);
 	webkit_web_context_set_cache_model(defcontext, WEBKIT_CACHE_MODEL_WEB_BROWSER);
-	g_signal_connect(G_OBJECT(defcontext),
-			"initialize-web-extensions",
-			G_CALLBACK(initializewebextensions), NULL);
+	//g_signal_connect(G_OBJECT(defcontext),
+	//		"initialize-web-extensions",
+	//		G_CALLBACK(initializewebextensions), NULL);
 }
 
 static void
