diff -ur surf-0.6.orig/configure.ac surf-0.6/configure.ac
--- surf-0.6.orig/configure.ac	2016-07-28 23:07:42.070689454 -0700
+++ surf-0.6/configure.ac	2016-07-28 23:09:48.414693618 -0700
@@ -28,7 +28,7 @@
 PKG_CHECK_MODULES([GTK3], [gtk+-3.0])
 PKG_CHECK_MODULES([GDK], [gdk-2.0])
 PKG_CHECK_MODULES([GTHREAD], [gthread-2.0])
-PKG_CHECK_MODULES([WEBKIT2GTK], [webkit2gtk-3.0])
+PKG_CHECK_MODULES([WEBKIT2GTK], [webkit-1.0])
 PKG_CHECK_MODULES([JSONGLIB], [json-glib-1.0])
 PKG_CHECK_MODULES([LIBSOUP], [libsoup-2.4])
 
