diff -u surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2014-06-23 20:40:33.418449144 -0700
+++ surf-0.6/surf.c	2014-06-23 20:42:22.072550477 -0700
@@ -962,7 +962,7 @@
 			else if(ev->atom == atoms[AtomScriptFile]) {
 				arg.v = getatom(c, AtomScriptFile);
 				frame = webkit_web_view_get_main_frame(c->view);
-				runscript(arg.v, frame, webkit_web_frame_get_global_context(frame));
+				runscript(arg.v, frame);
 				return GDK_FILTER_REMOVE;
 			}
 		}
