diff -ur surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2016-07-28 21:43:42.255523380 -0700
+++ surf-0.6/surf.c	2016-07-28 21:55:41.311547075 -0700
@@ -76,6 +76,7 @@
 static Atom atoms[AtomLast];
 static WebKitWebView *root_view;
 static Client *clients = NULL;
+//static GdkNativeWindow embed = 0;
 static gboolean showxid = FALSE;
 static char winid[64];
 static gboolean usingproxy = 0;
@@ -643,21 +644,25 @@
 		die("Cannot malloc!\n");
 
 	/* Window */
-	c->win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
-
-	/* TA:  20091214:  Despite what the GNOME docs say, the ICCCM
-	 * is always correct, so we should still call this function.
-	 * But when doing so, we *must* differentiate between a
-	 * WM_CLASS and a resource on the window.  By convention, the
-	 * window class (WM_CLASS) is capped, while the resource is in
-	 * lowercase.   Both these values come as a pair.
-	 */
-	gtk_window_set_wmclass(GTK_WINDOW(c->win), "surf", "Surf");
+	if(0) {
+		;//c->win = gtk_plug_new(embed);
+	} else {
+		c->win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 
-	/* TA:  20091214:  And set the role here as well -- so that
-	 * sessions can pick this up.
-	 */
-	gtk_window_set_role(GTK_WINDOW(c->win), "Surf");
+		/* TA:  20091214:  Despite what the GNOME docs say, the ICCCM
+		 * is always correct, so we should still call this function.
+		 * But when doing so, we *must* differentiate between a
+		 * WM_CLASS and a resource on the window.  By convention, the
+		 * window class (WM_CLASS) is capped, while the resource is in
+		 * lowercase.   Both these values come as a pair.
+		 */
+		gtk_window_set_wmclass(GTK_WINDOW(c->win), "surf", "Surf");
+
+		/* TA:  20091214:  And set the role here as well -- so that
+		 * sessions can pick this up.
+		 */
+		gtk_window_set_role(GTK_WINDOW(c->win), "Surf");
+	}
 	gtk_window_set_default_size(GTK_WINDOW(c->win), 800, 600);
 	g_signal_connect(G_OBJECT(c->win),
 			"destroy",
@@ -1175,6 +1180,9 @@
 	case 'c':
 		cookiefile = EARGF(usage());
 		break;
+	//case 'e':
+	//	embed = strtol(EARGF(usage()), NULL, 0);
+	//	break;
 	case 'i':
 		loadimages = 0;
 		break;
