diff -ur surf-0.6.orig/surf.c surf-0.6/surf.c
--- surf-0.6.orig/surf.c	2016-07-28 15:44:51.811813900 -0700
+++ surf-0.6/surf.c	2016-07-28 15:50:49.998825703 -0700
@@ -680,10 +680,10 @@
 	/* Webview */
 	//usercontent = webkit_user_content_manager_new();
 	if (clients == NULL) {
-		c->view = webkit_web_view_new();
+		c->view = WEBKIT_WEB_VIEW(webkit_web_view_new());
 		//c->view = WEBKIT_WEB_VIEW(webkit_web_view_new_with_user_content_manager(usercontent));
 	} else {
-		c->view = webkit_web_view_new();
+		c->view = WEBKIT_WEB_VIEW(webkit_web_view_new());
 		//c->view = WEBKIT_WEB_VIEW(webkit_web_view_new_with_related_view(root_view));
 	}
 	g_signal_connect(G_OBJECT(c->view),
