diff -urp leveldb-1.22.orig/CMakeLists.txt leveldb-1.22/CMakeLists.txt
--- leveldb-1.22.orig/CMakeLists.txt	2019-05-03 13:24:26.000000000 -0700
+++ leveldb-1.22/CMakeLists.txt	2020-08-18 11:54:10.252473165 -0700
@@ -27,6 +27,8 @@ endif (WIN32)
 option(LEVELDB_BUILD_TESTS "Build LevelDB's unit tests" ON)
 option(LEVELDB_BUILD_BENCHMARKS "Build LevelDB's benchmarks" ON)
 option(LEVELDB_INSTALL "Install LevelDB's header and library" ON)
+option(USE_TCMALLOC "Use tcmalloc" ON)
+option(USE_SNAPPY "Use snappy" ON)
 
 include(TestBigEndian)
 test_big_endian(LEVELDB_IS_BIG_ENDIAN)
@@ -248,12 +250,12 @@ endif(HAVE_CLANG_THREAD_SAFETY)
 if(HAVE_CRC32C)
   target_link_libraries(leveldb crc32c)
 endif(HAVE_CRC32C)
-if(HAVE_SNAPPY)
+if(USE_SNAPPY AND HAVE_SNAPPY)
   target_link_libraries(leveldb snappy)
-endif(HAVE_SNAPPY)
-if(HAVE_TCMALLOC)
+endif(USE_SNAPPY AND HAVE_SNAPPY)
+if(USE_TCMALLOC AND HAVE_TCMALLOC)
   target_link_libraries(leveldb tcmalloc)
-endif(HAVE_TCMALLOC)
+endif(USE_TCMALLOC AND HAVE_TCMALLOC)
 
 # Needed by port_stdcxx.h
 find_package(Threads REQUIRED)
