--- a/caesium/CMakeLists.txt.orig	2019-11-02 20:05:30.642464381 -0700
+++ b/caesium/CMakeLists.txt	2019-11-02 20:09:28.747664807 -0700
@@ -34,11 +34,11 @@ if (APPLE)
 elseif (UNIX OR LINUX)
     install(FILES caesium.h DESTINATION /usr/include)
     install(TARGETS caesium caesium_static
-            LIBRARY DESTINATION /usr/lib
-            ARCHIVE DESTINATION /usr/lib)
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 else ()
     install(FILES caesium.h DESTINATION include)
     install(TARGETS caesium caesium_static
             LIBRARY DESTINATION lib
             ARCHIVE DESTINATION lib)
-endif ()
\ No newline at end of file
+endif ()
