diff -urp boost_1_63_0.orig/boost/intrusive/pointer_traits.hpp boost_1_63_0/boost/intrusive/pointer_traits.hpp
--- boost_1_63_0.orig/boost/intrusive/pointer_traits.hpp	2016-12-22 04:33:16.000000000 -0800
+++ boost_1_63_0/boost/intrusive/pointer_traits.hpp	2017-02-21 20:26:13.271781835 -0800
@@ -129,7 +129,7 @@ struct pointer_traits
       //tries to converts &r to pointer.
       const bool value = boost::intrusive::detail::
          has_member_function_callable_with_pointer_to
-            <Ptr, Ptr (*)(reference)>::value;
+            <pointer, pointer(*)(reference)>::value;
       boost::intrusive::detail::bool_<value> flag;
       return pointer_traits::priv_pointer_to(flag, r);
    }
