--- dotdevelop-9999/main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj.orig	2022-09-16 08:57:36.386680692 -0700
+++ dotdevelop-9999/main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj	2022-09-16 09:16:31.973378087 -0700
@@ -15,7 +15,7 @@
     <ConfigureScript>$(MSBuildProjectDirectory)\..\..\..\..\scripts\configure.cs</ConfigureScript>
     <ConfigureScriptExe>$(MSBuildProjectDirectory)\..\..\..\..\scripts\configure.exe</ConfigureScriptExe>
     <FullBuildInfo>fullbuildinfo</FullBuildInfo>
-    <MonoLauncher Condition=" '$(VisualStudioVersion)' == '' ">mono </MonoLauncher>
+    <MonoLauncher Condition=" '$(VisualStudioVersion)' == '' Or '$(UseMonoLauncher)' == '1' ">mono </MonoLauncher>
     <TargetFrameworkVersion>$(MDFrameworkVersion)</TargetFrameworkVersion>
     <OutputPath>..\..\..\build\bin\</OutputPath>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
