--- dotdevelop-9999/main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj.orig	2022-09-16 10:04:20.741652601 -0700
+++ dotdevelop-9999/main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj	2022-09-16 10:26:49.470095218 -0700
@@ -798,7 +798,7 @@
     <MakeDir Directories="$(FullBuildInfo)" />
     <Csc Sources="$(ConfigureScript)" OutputAssembly="$(ConfigureScriptExe)" ToolExe="$(CscToolExe)" ToolPath="$(CscToolPath)" Condition="!Exists('$(ConfigureScriptExe)')" />
     <Exec Command="$(MonoLauncher)&quot;$(ConfigureScriptExe)&quot; gen-buildinfo $(FullBuildInfo)" WorkingDirectory="$(MSBuildProjectDirectory)" />
-    <Csc Sources="BuildVariables.gen.cs" OutputAssembly="BuildVariables.gen.exe" ToolExe="$(CscToolExe)" ToolPath="$(CscToolPath)" />
+    <Csc Sources="BuildVariables.gen.cs" OutputAssembly="BuildVariables.gen.exe" ToolExe="$(CscToolExe)" ToolPath="$(CscToolPath)" References="mscorlib.dll;System.dll;System.Core.dll;System.Runtime.dll" AdditionalLibPaths="$(ReferencePath)/microsoft.netframework.referenceassemblies.net472/1.0.3/build/.NETFramework/v4.7.2;$(ReferencePath)/microsoft.netframework.referenceassemblies.net472/1.0.3/build/.NETFramework/v4.7.2/Facades" />
     <Exec Command="$(MonoLauncher)&quot;$(MSBuildProjectDirectory)\BuildVariables.gen.exe&quot; ." WorkingDirectory="$(MSBuildProjectDirectory)" />
     <Delete Files="BuildVariables.gen.exe" />
     <MakeDir Directories="$(OutputPath)" />
