--- a/main/external/vs-editor-api/samples/AsyncQuickInfoDemo/AsyncQuickInfoDemo/AsyncQuickInfoDemo.csproj.orig	2022-09-16 19:37:34.757381495 -0700
+++ b/main/external/vs-editor-api/samples/AsyncQuickInfoDemo/AsyncQuickInfoDemo/AsyncQuickInfoDemo.csproj	2022-09-16 19:37:47.077867826 -0700
@@ -5,9 +5,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.VisualStudio.Language" Version="15.6.391-preview-g3c4ef6e5fc" />
-    <PackageReference Include="Microsoft.VisualStudio.Language.StandardClassification" Version="15.6.391-preview-g3c4ef6e5fc" />
-    <PackageReference Include="Microsoft.VisualStudio.Text.UI" Version="15.6.391-preview-g3c4ef6e5fc" />
+    <PackageReference Include="Microsoft.VisualStudio.Language" Version="15.6.27740" />
+    <PackageReference Include="Microsoft.VisualStudio.Language.StandardClassification" Version="15.6.27740" />
+    <PackageReference Include="Microsoft.VisualStudio.Text.UI" Version="15.6.27740" />
   </ItemGroup>
 
   <ItemGroup>
