--- a/cmake/tools/SetupWebKit.cmake.orig	2025-01-25 01:06:24.671954682 -0800
+++ b/cmake/tools/SetupWebKit.cmake	2025-01-25 01:37:37.795390931 -0800
@@ -23,17 +23,17 @@ set(WEBKIT_INCLUDE_PATH ${WEBKIT_PATH}/i
 set(WEBKIT_LIB_PATH ${WEBKIT_PATH}/lib)
 
 if(WEBKIT_LOCAL)
-  if(EXISTS ${WEBKIT_PATH}/cmakeconfig.h)
+  if(EXISTS ${WEBKIT_PATH}/include/cmakeconfig.h)
     # You may need to run:
     # make jsc-compile-debug jsc-copy-headers
     include_directories(
-      ${WEBKIT_PATH}
-      ${WEBKIT_PATH}/JavaScriptCore/Headers/JavaScriptCore
-      ${WEBKIT_PATH}/JavaScriptCore/PrivateHeaders
-      ${WEBKIT_PATH}/bmalloc/Headers
-      ${WEBKIT_PATH}/WTF/Headers
-      ${WEBKIT_PATH}/JavaScriptCore/DerivedSources/inspector
-      ${WEBKIT_PATH}/JavaScriptCore/PrivateHeaders/JavaScriptCore
+      ${WEBKIT_PATH}/include
+      ${WEBKIT_PATH}/include/JavaScriptCore
+      ${WEBKIT_PATH}/include/JavaScriptCore
+      ${WEBKIT_PATH}/include/bmalloc
+      ${WEBKIT_PATH}/include/wtf
+      ${WEBKIT_PATH}/include/JavaScriptCore
+      ${WEBKIT_PATH}/include/JavaScriptCore
     )
   endif()
 
