--- a/drivers/media/usb/gspca/Kconfig.orig	2022-04-08 18:41:16.562860896 -0700
+++ b/drivers/media/usb/gspca/Kconfig	2022-04-15 01:44:18.273390563 -0700
@@ -62,6 +62,16 @@ config USB_GSPCA_DTCS033
 	  To compile this driver as a module, choose M here: the
 	  module will be called gspca_dtcs033.
 
+config USB_GSPCA_EP800
+	tristate "EP800 USB Camera Driver"
+	depends on VIDEO_V4L2 && USB_GSPCA
+	help
+	 Say Y here if you want support for cameras based on the
+	 Endpoints (formerly known as AOX) ep800 chip.
+
+	 To compile this driver as a module, choose M here: the
+	 module will be called gspca_ep800.
+
 config USB_GSPCA_ETOMS
 	tristate "Etoms USB Camera Driver"
 	depends on VIDEO_V4L2 && USB_GSPCA
--- a/drivers/media/usb/gspca/Makefile.orig	2022-04-08 18:41:16.367853483 -0700
+++ b/drivers/media/usb/gspca/Makefile	2022-04-15 01:43:58.512626867 -0700
@@ -4,6 +4,7 @@ obj-$(CONFIG_USB_GSPCA_BENQ)     += gspc
 obj-$(CONFIG_USB_GSPCA_CONEX)    += gspca_conex.o
 obj-$(CONFIG_USB_GSPCA_CPIA1)    += gspca_cpia1.o
 obj-$(CONFIG_USB_GSPCA_DTCS033)  += gspca_dtcs033.o
+obj-$(CONFIG_USB_GSPCA_EP800)    += gspca_ep800.o
 obj-$(CONFIG_USB_GSPCA_ETOMS)    += gspca_etoms.o
 obj-$(CONFIG_USB_GSPCA_FINEPIX)  += gspca_finepix.o
 obj-$(CONFIG_USB_GSPCA_JEILINJ)  += gspca_jeilinj.o
@@ -52,6 +53,7 @@ gspca_benq-objs     := benq.o
 gspca_conex-objs    := conex.o
 gspca_cpia1-objs    := cpia1.o
 gspca_dtcs033-objs  := dtcs033.o
+gspca_ep800-objs    := ep800.o
 gspca_etoms-objs    := etoms.o
 gspca_finepix-objs  := finepix.o
 gspca_jeilinj-objs  := jeilinj.o
