diff -urp sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav/SharpNav.csproj sharpnav-1.0.0_alpha2_p9999/Source/SharpNav/SharpNav.csproj
--- sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav/SharpNav.csproj	2022-09-13 10:49:13.837885232 -0700
+++ sharpnav-1.0.0_alpha2_p9999/Source/SharpNav/SharpNav.csproj	2022-09-13 10:56:59.635684084 -0700
@@ -73,34 +73,34 @@
   </PropertyGroup>
   <!-- Common items -->
   <ItemGroup>
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" />
     <Reference Include="System" />
     <Reference Include="System.Core" />
-    <Reference Include="Newtonsoft.Json">
-      <HintPath>..\packages\Newtonsoft.Json.8.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
-    </Reference>
-    <Reference Include="YamlDotNet">
-      <HintPath>..\packages\YamlDotNet.Signed.3.8.0\lib\net35\YamlDotNet.dll</HintPath>
-    </Reference>
+    <PackageReference Include="Newtonsoft.Json" Version="8.0.3" ExcludeAssets="compile" />
+    <PackageReference Include="YamlDotNet" Version="5.1.0" ExcludeAssets="compile" />
+    <Reference Include="Newtonsoft.Json">
+       <HintPath>$(NuGetPackageRoot2)/packages/newtonsoft.json/8.0.3/lib/net45/Newtonsoft.Json.dll</HintPath>
+    </Reference>
+    <Reference Include="YamlDotNet">
+       <HintPath>$(NuGetPackageRoot2)/packages/yamldotnet/5.1.0/lib/net35/YamlDotNet.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <!-- Configuration specific items -->
   <ItemGroup Condition="'$(Configuration)' == 'OpenTK'">
-    <Reference Include="OpenTK">
-      <HintPath>..\packages\OpenTK.1.1.2225.0\lib\net20\OpenTK.dll</HintPath>
-    </Reference>
+    <PackageReference Include="OpenTK" Version="1.1.2349.61993" ExcludeAssets="compile" />
+    <Reference Include="OpenTK">
+       <HintPath>$(NuGetPackageRoot2)/packages/opentk/1.1.2349.61993/lib/NET40/OpenTK.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <ItemGroup Condition="'$(Configuration)' == 'SharpDX'">
-    <Reference Include="SharpDX">
-      <HintPath>..\packages\SharpDX.3.0.2\lib\net45\SharpDX.dll</HintPath>
-    </Reference>
-    <Reference Include="SharpDX.Mathematics">
-      <HintPath>..\packages\SharpDX.Mathematics.3.0.2\lib\net45\SharpDX.Mathematics.dll</HintPath>
-    </Reference>
+    <PackageReference Include="SharpDX" Version="3.0.2" />
+    <PackageReference Include="SharpDX.Mathematics" Version="3.0.2" />
   </ItemGroup>
   <ItemGroup Condition="'$(Configuration)' == 'MonoGame'">
-    <Reference Include="MonoGame.Framework, Version=3.1.2.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\MonoGame.Framework.Portable.3.2.99.1-Beta\lib\portable-net45+win+wpa81+Xamarin.iOS10+MonoAndroid10+MonoTouch10\MonoGame.Framework.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <PackageReference Include="MonoGame.Framework.DesktopGL" Version="3.7.0.1708" ExcludeAssets="compile" />
+    <Reference Include="MonoGame.Framework">
+       <HintPath>$(NuGetPackageRoot2)/packages/monogame.framework.desktopgl/3.7.0.1708/lib/net45/MonoGame.Framework.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <!-- C# items -->
   <ItemGroup>
@@ -204,4 +192,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
diff -urp sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.CLI/SharpNav.CLI.csproj sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.CLI/SharpNav.CLI.csproj
--- sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.CLI/SharpNav.CLI.csproj	2022-09-13 10:49:13.709880347 -0700
+++ sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.CLI/SharpNav.CLI.csproj	2022-09-13 10:55:44.180797885 -0700
@@ -39,9 +39,11 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Mono.Options">
-      <HintPath>..\packages\Mono.Options.4.4.0.0\lib\net4-client\Mono.Options.dll</HintPath>
-    </Reference>
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" />
+    <PackageReference Include="Mono.Options" Version="4.4.0" ExcludeAssets="compile" />
+    <Reference Include="Mono.Options">
+       <HintPath>$(NuGetPackageRoot2)/packages/mono.options/4.4.0/lib/net4-client/Mono.Options.dll</HintPath>
+    </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="Microsoft.CSharp" />
@@ -72,4 +71,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
diff -urp sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.Examples/SharpNav.Examples.csproj sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.Examples/SharpNav.Examples.csproj
--- sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.Examples/SharpNav.Examples.csproj	2022-09-13 10:49:13.725880957 -0700
+++ sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.Examples/SharpNav.Examples.csproj	2022-09-13 10:55:20.379887713 -0700
@@ -63,16 +63,17 @@
     <StartupObject>SharpNav.Examples.Program</StartupObject>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" />
     <Reference Include="Gwen">
       <HintPath>..\..\Dependencies\Gwen.dll</HintPath>
     </Reference>
     <Reference Include="Gwen.Renderer.OpenTK">
       <HintPath>..\..\Dependencies\Gwen.Renderer.OpenTK.dll</HintPath>
     </Reference>
-    <Reference Include="OpenTK, Version=1.1.0.0, Culture=neutral, PublicKeyToken=bad199fe84eb3df4, processorArchitecture=MSIL">
-      <HintPath>..\packages\OpenTK.1.1.2349.61993\lib\NET40\OpenTK.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <PackageReference Include="OpenTK" Version="1.1.2349.61993" ExcludeAssets="compile" />
+    <Reference Include="OpenTK">
+       <HintPath>$(NuGetPackageRoot2)/packages/opentk/1.1.2349.61993/lib/NET40/OpenTK.dll</HintPath>
+    </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="Microsoft.CSharp" />
@@ -127,4 +125,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
diff -urp sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.GUI/SharpNav.GUI.csproj sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.GUI/SharpNav.GUI.csproj
--- sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.GUI/SharpNav.GUI.csproj	2022-09-13 10:49:13.765882485 -0700
+++ sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.GUI/SharpNav.GUI.csproj	2022-09-13 10:49:46.255122495 -0700
@@ -39,6 +39,7 @@
     <StartupObject />
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" />
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="System.Drawing" />
@@ -79,4 +80,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
diff -urp sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.Tests/SharpNav.Tests.csproj sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.Tests/SharpNav.Tests.csproj
--- sharpnav-1.0.0_alpha2_p9999.orig/Source/SharpNav.Tests/SharpNav.Tests.csproj	2022-09-13 10:49:13.777882943 -0700
+++ sharpnav-1.0.0_alpha2_p9999/Source/SharpNav.Tests/SharpNav.Tests.csproj	2022-09-13 10:55:01.643171281 -0700
@@ -78,41 +78,29 @@
     <AssemblyOriginatorKeyFile>SharpNav.Tests.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" />
     <Reference Include="System" />
     <Reference Include="System.Core" />
-    <Reference Include="Newtonsoft.Json">
-      <HintPath>..\packages\Newtonsoft.Json.8.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
-    </Reference>
-    <Reference Include="nunit.framework">
-      <HintPath>..\packages\NUnit.3.2.1\lib\net45\nunit.framework.dll</HintPath>
-    </Reference>
+    <PackageReference Include="Newtonsoft.Json" Version="8.0.3" ExcludeAssets="compile" />
+    <Reference Include="Newtonsoft.Json">
+       <HintPath>$(NuGetPackageRoot2)/packages/newtonsoft.json/8.0.3/lib/net45/Newtonsoft.Json.dll</HintPath>
+    </Reference>
+    <PackageReference Include="nunit.framework" Version="3.2.1" />
   </ItemGroup>
   <ItemGroup Condition="'$(Configuration)' == 'SharpDX'">
-    <Reference Include="SharpDX">
-      <HintPath>..\packages\SharpDX.3.0.2\lib\net45\SharpDX.dll</HintPath>
-    </Reference>
-    <Reference Include="SharpDX.Mathematics">
-      <HintPath>..\packages\SharpDX.Mathematics.3.0.2\lib\net45\SharpDX.Mathematics.dll</HintPath>
-    </Reference>
+    <PackageReference Include="SharpDX" Version="3.0.2" />
+    <PackageReference Include="SharpDX.Mathematics" Version="3.0.2" />
   </ItemGroup>
   <ItemGroup Condition="'$(Configuration)' == 'OpenTK'">
-	<Reference Include="OpenTK">
-		<HintPath>..\packages\OpenTK.1.1.2225.0\lib\net20\OpenTK.dll</HintPath>
-	</Reference>
+    <PackageReference Include="OpenTK" Version="1.1.2225.0" ExcludeAssets="compile" />
+    <Reference Include="OpenTK">
+       <HintPath>$(NuGetPackageRoot2)/packages/opentk/1.1.2349.61993/lib/NET40/OpenTK.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <ItemGroup Condition="'$(Configuration)' == 'MonoGame'">
-	<Reference Include="MonoGame.Framework, Version=3.5.1.1679, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\MonoGame.Framework.DesktopGL.3.5.1.1679\lib\net40\MonoGame.Framework.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="NVorbis, Version=0.8.4.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\MonoGame.Framework.DesktopGL.3.5.1.1679\lib\net40\NVorbis.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-	<Reference Include="OpenTK, Version=1.1.0.0, Culture=neutral, PublicKeyToken=bad199fe84eb3df4, processorArchitecture=MSIL">
-      <HintPath>..\packages\MonoGame.Framework.DesktopGL.3.5.1.1679\lib\net40\OpenTK.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <PackageReference Include="MonoGame.Framework.DesktopGL" Version="3.7.0.1708" />
+    <PackageReference Include="NVorbis" Version="0.9.1" />
+    <PackageReference Include="OpenTK" Version="1.1.2349.61993" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Collections\Generic\BufferedQueueTests.cs" />
@@ -156,4 +138,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
