--- sharpnav-1.0.0_alpha2_p9999_MonoGame/Source/SharpNav/IO/NavMeshConfigurationFile.cs.orig	2022-09-13 13:50:44.953675508 -0700
+++ sharpnav-1.0.0_alpha2_p9999_MonoGame/Source/SharpNav/IO/NavMeshConfigurationFile.cs	2022-09-13 14:01:36.951259055 -0700
@@ -31,7 +31,7 @@ namespace SharpNav.IO
 
 		public NavMeshConfigurationFile(StreamReader input)
 		{
-			var deserializer = new Deserializer(namingConvention: new HyphenatedNamingConvention());
+			var deserializer = new Deserializer();
 			var data = deserializer.Deserialize<YamlData>(input);
 
 			GenerationSettings = data.Config;
@@ -46,7 +46,7 @@ namespace SharpNav.IO
 			data.Export = ExportPath;
 			data.Meshes = InputMeshes;
 
-			var serializer = new Serializer(SerializationOptions.None, new HyphenatedNamingConvention());
+			var serializer = new Serializer();
 			using (StreamWriter writer = new StreamWriter(path))
 				serializer.Serialize(writer, data);
 		}
