--- a/Source/JavaScriptCore/API/JSHeapFinalizerPrivate.h.orig	2025-12-23 21:40:18.187767363 -0800
+++ b/Source/JavaScriptCore/API/JSHeapFinalizerPrivate.h	2025-12-23 22:09:01.133149441 -0800
@@ -26,7 +26,7 @@
 #ifndef JSHeapFinalizerPrivate_h
 #define JSHeapFinalizerPrivate_h
 
-#include "JSBase.h"
+#include <JavaScriptCore/JSBase.h>
 #include <stdbool.h>
 
 #ifdef __cplusplus
--- a/Source/JavaScriptCore/API/JSHeapFinalizerPrivate.h.orig	2025-12-23 22:30:03.821349952 -0800
+++ b/Source/JavaScriptCore/API/JSHeapFinalizerPrivate.h	2025-12-23 22:48:17.088757599 -0800
@@ -26,7 +26,7 @@
 #ifndef JSHeapFinalizerPrivate_h
 #define JSHeapFinalizerPrivate_h
 
-#include "JSBase.h"
+#include <JavaScriptCore/JSBase.h>
 #include <stdbool.h>
 
 #ifdef __cplusplus
