--- a/Tools/TestWebKitAPI/Tests/WTF/StringImpl.cpp.orig	2025-12-23 20:47:34.087524792 -0800
+++ b/Tools/TestWebKitAPI/Tests/WTF/StringImpl.cpp	2025-12-23 20:48:28.587496137 -0800
@@ -755,12 +755,17 @@ TEST(WTF, ExternalStringImplCreate8bit)
 {
     constexpr Latin1Character buffer[] = "hello";
     constexpr size_t bufferStringLength = sizeof(buffer) - 1;
-    bool freeFunctionCalled = false;
+    static bool freeFunctionCalled = false;
+    freeFunctionCalled = false;
 
     {
-        auto external = ExternalStringImpl::create({ buffer, bufferStringLength }, [&freeFunctionCalled](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) mutable {
-            freeFunctionCalled = true;
-        });
+        auto external = ExternalStringImpl::create(
+            { buffer, bufferStringLength },
+            nullptr,
+            reinterpret_cast<void(*)(void*, void*, unsigned)>(+[](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) {
+                freeFunctionCalled = true;
+            })
+        );
 
         ASSERT_TRUE(external->isExternal());
         ASSERT_TRUE(external->is8Bit());
@@ -777,12 +782,17 @@ TEST(WTF, ExternalStringImplCreate16bit)
 {
     constexpr char16_t buffer[] = { L'h', L'e', L'l', L'l', L'o', L'\0' };
     constexpr size_t bufferStringLength = (sizeof(buffer) - 1) / sizeof(char16_t);
-    bool freeFunctionCalled = false;
+    static bool freeFunctionCalled = false;
+    freeFunctionCalled = false;
 
     {
-        auto external = ExternalStringImpl::create({ buffer, bufferStringLength }, [&freeFunctionCalled](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) mutable {
-            freeFunctionCalled = true;
-        });
+        auto external = ExternalStringImpl::create(
+            { buffer, bufferStringLength },
+            nullptr,
+            reinterpret_cast<void(*)(void*, void*, unsigned)>(+[](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) {
+                freeFunctionCalled = true;
+            })
+        );
 
         ASSERT_TRUE(external->isExternal());
         ASSERT_FALSE(external->is8Bit());
@@ -806,12 +816,17 @@ TEST(WTF, ExternalStringAtom)
 {
     constexpr Latin1Character buffer[] = "hello";
     constexpr size_t bufferStringLength = sizeof(buffer) - 1;
-    bool freeFunctionCalled = false;
+    static bool freeFunctionCalled = false;
+    freeFunctionCalled = false;
 
     {
-        auto external = ExternalStringImpl::create({ buffer, bufferStringLength }, [&freeFunctionCalled](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) mutable {
-            freeFunctionCalled = true;
-        });    
+        auto external = ExternalStringImpl::create(
+            { buffer, bufferStringLength },
+            nullptr,
+            reinterpret_cast<void(*)(void*, void*, unsigned)>(+[](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) {
+                freeFunctionCalled = true;
+            })
+        );
 
         ASSERT_TRUE(external->isExternal());
         ASSERT_FALSE(external->isAtom());
@@ -842,12 +857,17 @@ TEST(WTF, ExternalStringAtom)
 TEST(WTF, ExternalStringToSymbol)
 {
     static constexpr ASCIILiteral buffer = "hello"_s;
-    bool freeFunctionCalled = false;
+    static bool freeFunctionCalled = false;
+    freeFunctionCalled = false;
 
     {
-        auto external = ExternalStringImpl::create(buffer.span8(), [&freeFunctionCalled](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) mutable {
-            freeFunctionCalled = true;
-        });
+        auto external = ExternalStringImpl::create(
+            buffer.span8(),
+            nullptr,
+            reinterpret_cast<void(*)(void*, void*, unsigned)>(+[](ExternalStringImpl* externalStringImpl, void* buffer, unsigned bufferSize) {
+                freeFunctionCalled = true;
+            })
+        );
 
         ASSERT_TRUE(external->isExternal());
         ASSERT_FALSE(external->isSymbol());
