diff -urp eigen-7bf2968fed5f246c0589e1111004cb420fcd7c71.orig/scripts/relicense.py eigen-7bf2968fed5f246c0589e1111004cb420fcd7c71/scripts/relicense.py
--- eigen-7bf2968fed5f246c0589e1111004cb420fcd7c71.orig/scripts/relicense.py	2023-03-07 12:28:05.000000000 -0800
+++ eigen-7bf2968fed5f246c0589e1111004cb420fcd7c71/scripts/relicense.py	2023-07-29 12:47:26.111026091 -0700
@@ -52,7 +52,7 @@ rootdir = sys.argv[1]
 for root, sub_folders, files in os.walk(rootdir):
     for basename in files:
         if basename in exclusions:
-          print 'SKIPPED', filename
+          print('SKIPPED', filename)
           continue
         filename = os.path.join(root, basename)
         fo = file(filename)
@@ -64,6 +64,6 @@ for root, sub_folders, files in os.walk(
           fo = file(filename, "w")
           fo.write(text)
           fo.close()
-          print 'UPDATED', filename
+          print('UPDATED', filename)
         else:
-          print '       ', filename
+          print('       ', filename)
