--- a/src/eglfsxkb/CMakeLists.txt.orig	2022-12-12 21:54:04.388086870 -0800
+++ b/src/eglfsxkb/CMakeLists.txt	2022-12-12 21:59:06.971947020 -0800
@@ -1,4 +1,5 @@
-find_package(XkbCommon REQUIRED)
+find_package(XkbCommon REQUIRED)
+pkg_check_modules(XkbCommon xkbcommon REQUIRED IMPORTED_TARGET)
 
 add_library(eglfsxkb STATIC
     eglfsxkb.cpp
@@ -6,7 +6,7 @@ add_library(eglfsxkb STATIC
 )
 target_link_libraries(eglfsxkb
     Qt5::Core
-    PkgConfig::XkbCommon
+    PkgConfig::XkbCommon
 )
 target_include_directories(eglfsxkb INTERFACE
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
