--- a/SConstruct.orig	2022-03-22 02:47:22.000000000 -0700
+++ b/SConstruct	2022-07-30 10:21:01.919717531 -0700
@@ -73,6 +73,7 @@ env_base.PrependENVPath("PATH", os.geten
 env_base.PrependENVPath("PKG_CONFIG_PATH", os.getenv("PKG_CONFIG_PATH"))
 if "TERM" in os.environ:  # Used for colored output.
     env_base["ENV"]["TERM"] = os.environ["TERM"]
+env_base.PrependENVPath("CCACHE_DIR", os.getenv("CCACHE_DIR"))
 
 env_base.disabled_modules = []
 env_base.use_ptrcall = False
