diff '--color=auto' -urp librsvg-2.58.5.orig/configure librsvg-2.58.5/configure
--- librsvg-2.58.5.orig/configure	2025-04-28 15:40:16.116213853 -0700
+++ librsvg-2.58.5/configure	2025-04-28 16:14:58.144611936 -0700
@@ -15955,12 +15955,7 @@ else
   CROSS_COMPILING_FALSE=
 fi
 
-if test "x$cross_compiling" = "xyes"
-then :
-
-	RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
-
-fi
+RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
 
 
 
diff '--color=auto' -urp librsvg-2.58.5.orig/configure.ac librsvg-2.58.5/configure.ac
--- librsvg-2.58.5.orig/configure.ac	2024-09-27 16:32:32.000000000 -0700
+++ librsvg-2.58.5/configure.ac	2025-04-28 16:15:00.085420666 -0700
@@ -291,9 +291,7 @@ AC_ARG_VAR(RUST_TARGET, [Set rust target
 : ${RUST_TARGET:=$host}
 
 AM_CONDITIONAL([CROSS_COMPILING], [test "x$cross_compiling" = xyes])
-AS_IF([test "x$cross_compiling" = "xyes"], [
-	RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
-])
+RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
 AC_SUBST([RUST_TARGET_SUBDIR])
 
 dnl ===========================================================================
