--- a/src/main/ollama-binaries.ts.orig	2025-09-14 06:52:34.000000000 -0700
+++ b/src/main/ollama-binaries.ts	2026-09-28 07:39:49.987648622 -0700
@@ -27,7 +27,7 @@ export function checkOllama(): Promise<b
       // if check is true, that means there's an error
       const ollamaAlreadyRunning = error?.message.includes(`listen tcp`)
       const ollamaNotInstalled =
-        error?.message.includes(`not found`) ||
+        // error?.message.includes(`not found`) || // Avoid:  Warning: Firejail configuration file /etc/firejail/firejail.config not found, using defaults
         error?.message.includes(`not recognized`) ||
         error?.message.includes(`not internal`) ||
         error?.message.includes(`not an internal`)
@@ -114,7 +114,7 @@ export async function downloadBinaries()
 
 export async function installOllamaLinux(): Promise<string> {
   return new Promise((resolve, reject)=> {
-    exec('gnome-terminal -- bash -c "curl -fsSL https://ollama.com/install.sh | sh; exec bash"', (error) => {
+    exec('true gnome-terminal -- bash -c "curl -fsSL https://ollama.com/install.sh | sh; exec bash"', (error) => { // Not everyone uses or has gnome-terminal installed
       if(error == null){
         resolve('linux-detected') // edge case handled on the frontend
       } else{
--- a/src/main/index.ts.orig	2026-09-28 07:31:25.862600417 -0700
+++ b/src/main/index.ts	2026-09-28 07:39:52.515573646 -0700
@@ -223,15 +223,15 @@ app.whenReady().then(() => {
     return response
   })
 
-  ipcMain.handle('downloadingOllama', async () => {
-    const response = await downloadingOllama()
-    return response
-  })
+  // ipcMain.handle('downloadingOllama', async () => {
+  //   const response = await downloadingOllama()
+  //   return response
+  // })
 
-  ipcMain.handle('installingOllama', async () => {
-    const response = await installingOllama()
-    return response
-  })
+  // ipcMain.handle('installingOllama', async () => {
+  //   const response = await installingOllama()
+  //   return response
+  // })
 
   ipcMain.handle('checkVersion', async () => {
     return new Promise((resolve) => {
