diff -urp emscripten-2.0.12.orig/em++ emscripten-2.0.12/em++
--- emscripten-2.0.12.orig/em++	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/em++	2021-10-14 11:59:26.696014893 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/em++.py" "$@"
diff -urp emscripten-2.0.12.orig/emar emscripten-2.0.12/emar
--- emscripten-2.0.12.orig/emar	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emar	2021-10-14 11:59:27.432014917 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emar.py" "$@"
diff -urp emscripten-2.0.12.orig/embuilder emscripten-2.0.12/embuilder
--- emscripten-2.0.12.orig/embuilder	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/embuilder	2021-10-14 11:59:31.384015047 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/embuilder.py" "$@"
diff -urp emscripten-2.0.12.orig/emcc emscripten-2.0.12/emcc
--- emscripten-2.0.12.orig/emcc	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emcc	2021-10-14 11:59:29.400014982 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emcc.py" "$@"
diff -urp emscripten-2.0.12.orig/emcmake emscripten-2.0.12/emcmake
--- emscripten-2.0.12.orig/emcmake	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emcmake	2021-10-14 11:59:28.760014961 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emcmake.py" "$@"
diff -urp emscripten-2.0.12.orig/em-config emscripten-2.0.12/em-config
--- emscripten-2.0.12.orig/em-config	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/em-config	2021-10-14 11:59:30.712015025 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/em-config.py" "$@"
diff -urp emscripten-2.0.12.orig/emconfigure emscripten-2.0.12/emconfigure
--- emscripten-2.0.12.orig/emconfigure	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emconfigure	2021-10-14 11:59:30.040015003 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emconfigure.py" "$@"
diff -urp emscripten-2.0.12.orig/emmake emscripten-2.0.12/emmake
--- emscripten-2.0.12.orig/emmake	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emmake	2021-10-14 11:59:25.128014841 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emmake.py" "$@"
diff -urp emscripten-2.0.12.orig/emranlib emscripten-2.0.12/emranlib
--- emscripten-2.0.12.orig/emranlib	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emranlib	2021-10-14 11:59:25.944014868 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emranlib.py" "$@"
diff -urp emscripten-2.0.12.orig/emrun emscripten-2.0.12/emrun
--- emscripten-2.0.12.orig/emrun	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emrun	2021-10-14 11:59:34.248015142 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emrun.py" "$@"
diff -urp emscripten-2.0.12.orig/emscons emscripten-2.0.12/emscons
--- emscripten-2.0.12.orig/emscons	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emscons	2021-10-14 11:59:35.032015167 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emscons.py" "$@"
diff -urp emscripten-2.0.12.orig/emsize emscripten-2.0.12/emsize
--- emscripten-2.0.12.orig/emsize	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/emsize	2021-10-14 11:59:28.104014939 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emsize.py" "$@"
diff -urp emscripten-2.0.12.orig/tools/emdump emscripten-2.0.12/tools/emdump
--- emscripten-2.0.12.orig/tools/emdump	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/tools/emdump	2021-10-14 11:59:32.072015070 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
diff -urp emscripten-2.0.12.orig/tools/file_packager emscripten-2.0.12/tools/file_packager
--- emscripten-2.0.12.orig/tools/file_packager	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/tools/file_packager	2021-10-14 11:59:32.760015092 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
diff -urp emscripten-2.0.12.orig/tools/run_python.sh emscripten-2.0.12/tools/run_python.sh
--- emscripten-2.0.12.orig/tools/run_python.sh	2021-01-09 08:21:25.000000000 -0800
+++ emscripten-2.0.12/tools/run_python.sh	2021-10-14 11:59:33.464015116 -0700
@@ -26,4 +26,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
