--- a/embuilder.orig	2020-07-24 13:33:08.199441527 -0700
+++ b/embuilder	2020-07-24 13:12:51.672034337 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/embuilder.py" "$@"
--- a/emcc.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emcc	2020-07-23 13:25:10.639710539 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emcc.py" "$@"
--- a/em++.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/em++	2020-07-23 13:11:57.599096594 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/em++.py" "$@"
--- a/emar.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emar	2020-07-24 13:12:50.842001802 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emar.py" "$@"
--- a/emcmake.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emcmake	2020-07-24 13:12:43.371708986 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emcmake.py" "$@"
--- a/em-config.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/em-config	2020-07-24 13:12:44.121738385 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/em-config.py" "$@"
--- a/emconfigure.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emconfigure	2020-07-24 13:12:44.821765824 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emconfigure.py" "$@"
--- a/emmake.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emmake	2020-07-24 13:12:45.531793655 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emmake.py" "$@"
--- a/emranlib.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emranlib	2020-07-24 13:12:46.221820705 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emranlib.py" "$@"
--- a/emrun.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emrun	2020-07-24 13:12:46.951849320 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emrun.py" "$@"
--- a/emscons.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emscons	2020-07-24 13:12:47.691878325 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emscons.py" "$@"
--- a/emsize.orig	2020-07-20 15:35:39.000000000 -0700
+++ b/emsize	2020-07-24 13:12:15.820628991 -0700
@@ -30,4 +30,6 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+EMSCRIPTEN_DIR=$(dirname $(realpath "${BASH_SOURCE[0]}"))
+
+exec "$PYTHON" "${EMSCRIPTEN_DIR}/emsize.py" "$@"
