diff -urp emscripten-2.0.26.orig/em++ emscripten-2.0.26/em++
--- emscripten-2.0.26.orig/em++	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/em++	2021-10-14 12:03:55.978023766 -0700
@@ -29,8 +29,8 @@ if [ -z "$PYTHON" ]; then
 fi
 
 if [ -z "$_EMCC_CCACHE" ]; then
-  exec "$PYTHON" "$0.py" "$@"
+  exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
 else
   unset _EMCC_CCACHE
-  exec ccache "$0" "$@"
+  exec ccache "$(realpath $(dirname $0))/$0" "$@"
 fi
diff -urp emscripten-2.0.26.orig/emar emscripten-2.0.26/emar
--- emscripten-2.0.26.orig/emar	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emar	2021-10-14 12:03:56.842023795 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emar.py" "$@"
diff -urp emscripten-2.0.26.orig/embuilder emscripten-2.0.26/embuilder
--- emscripten-2.0.26.orig/embuilder	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/embuilder	2021-10-14 12:04:03.226024005 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/embuilder.py" "$@"
diff -urp emscripten-2.0.26.orig/emcc emscripten-2.0.26/emcc
--- emscripten-2.0.26.orig/emcc	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emcc	2021-10-14 12:04:00.890023928 -0700
@@ -29,8 +29,8 @@ if [ -z "$PYTHON" ]; then
 fi
 
 if [ -z "$_EMCC_CCACHE" ]; then
-  exec "$PYTHON" "$0.py" "$@"
+  exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
 else
   unset _EMCC_CCACHE
-  exec ccache "$0" "$@"
+  exec ccache "$(realpath $(dirname $0))/$0" "$@"
 fi
diff -urp emscripten-2.0.26.orig/emcmake emscripten-2.0.26/emcmake
--- emscripten-2.0.26.orig/emcmake	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emcmake	2021-10-14 12:03:59.226023873 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emcmake.py" "$@"
diff -urp emscripten-2.0.26.orig/em-config emscripten-2.0.26/em-config
--- emscripten-2.0.26.orig/em-config	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/em-config	2021-10-14 12:04:02.458023980 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/em-config.py" "$@"
diff -urp emscripten-2.0.26.orig/emconfigure emscripten-2.0.26/emconfigure
--- emscripten-2.0.26.orig/emconfigure	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emconfigure	2021-10-14 12:04:01.689023954 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emconfigure.py" "$@"
diff -urp emscripten-2.0.26.orig/emdump emscripten-2.0.26/emdump
--- emscripten-2.0.26.orig/emdump	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emdump	2021-10-14 12:03:58.426023847 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$(dirname $0)/tools/emdump.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/tools/emdump.py" "$@"
diff -urp emscripten-2.0.26.orig/em-dwp emscripten-2.0.26/em-dwp
--- emscripten-2.0.26.orig/em-dwp	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/em-dwp	2021-10-14 12:04:00.058023901 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
diff -urp emscripten-2.0.26.orig/emdwp emscripten-2.0.26/emdwp
--- emscripten-2.0.26.orig/emdwp	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emdwp	2021-10-14 12:03:53.402023681 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$(dirname $0)/tools/emdwp.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/tools/emdwp.py" "$@"
diff -urp emscripten-2.0.26.orig/emmake emscripten-2.0.26/emmake
--- emscripten-2.0.26.orig/emmake	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emmake	2021-10-14 12:03:50.666023591 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emmake.py" "$@"
diff -urp emscripten-2.0.26.orig/emnm emscripten-2.0.26/emnm
--- emscripten-2.0.26.orig/emnm	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emnm	2021-10-14 12:03:55.018023735 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$(dirname $0)/tools/emnm.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/tools/emnm.py" "$@"
diff -urp emscripten-2.0.26.orig/emprofile emscripten-2.0.26/emprofile
--- emscripten-2.0.26.orig/emprofile	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emprofile	2021-10-14 12:03:52.634023656 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$(dirname $0)/tools/emprofile.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/tools/emprofile.py" "$@"
diff -urp emscripten-2.0.26.orig/emranlib emscripten-2.0.26/emranlib
--- emscripten-2.0.26.orig/emranlib	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emranlib	2021-10-14 12:03:51.578023621 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emranlib.py" "$@"
diff -urp emscripten-2.0.26.orig/emrun emscripten-2.0.26/emrun
--- emscripten-2.0.26.orig/emrun	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emrun	2021-10-14 12:04:09.641024216 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emrun.py" "$@"
diff -urp emscripten-2.0.26.orig/emscons emscripten-2.0.26/emscons
--- emscripten-2.0.26.orig/emscons	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emscons	2021-10-14 12:04:10.857024256 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emscons.py" "$@"
diff -urp emscripten-2.0.26.orig/emsize emscripten-2.0.26/emsize
--- emscripten-2.0.26.orig/emsize	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emsize	2021-10-14 12:03:57.642023821 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/emsize.py" "$@"
diff -urp emscripten-2.0.26.orig/tests/runner emscripten-2.0.26/tests/runner
--- emscripten-2.0.26.orig/tests/runner	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/tests/runner	2021-10-14 12:03:54.170023707 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
diff -urp emscripten-2.0.26.orig/tools/file_packager emscripten-2.0.26/tools/file_packager
--- emscripten-2.0.26.orig/tools/file_packager	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/tools/file_packager	2021-10-14 12:04:06.537024114 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
diff -urp emscripten-2.0.26.orig/tools/run_python_compiler.sh emscripten-2.0.26/tools/run_python_compiler.sh
--- emscripten-2.0.26.orig/tools/run_python_compiler.sh	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/tools/run_python_compiler.sh	2021-10-14 12:04:05.193024070 -0700
@@ -29,8 +29,8 @@ if [ -z "$PYTHON" ]; then
 fi
 
 if [ -z "$_EMCC_CCACHE" ]; then
-  exec "$PYTHON" "$0.py" "$@"
+  exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
 else
   unset _EMCC_CCACHE
-  exec ccache "$0" "$@"
+  exec ccache "$(realpath $(dirname $0))/$0" "$@"
 fi
diff -urp emscripten-2.0.26.orig/tools/run_python.sh emscripten-2.0.26/tools/run_python.sh
--- emscripten-2.0.26.orig/tools/run_python.sh	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/tools/run_python.sh	2021-10-14 12:04:07.657024151 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
diff -urp emscripten-2.0.26.orig/tools/webidl_binder emscripten-2.0.26/tools/webidl_binder
--- emscripten-2.0.26.orig/tools/webidl_binder	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/tools/webidl_binder	2021-10-14 12:04:04.041024032 -0700
@@ -28,4 +28,4 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
-exec "$PYTHON" "$0.py" "$@"
+exec "$PYTHON" "$(realpath $(dirname $0))/$0.py" "$@"
