# Backport to 2.0.26
diff -urp emscripten-2.0.26.orig/emcc.py emscripten-2.0.26/emcc.py
--- emscripten-2.0.26.orig/emcc.py	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/emcc.py	2023-01-02 09:59:25.327693590 -0800
@@ -948,6 +948,21 @@ run_via_emxx = False
 # Main run() function
 #
 def run(args):
+  if run_via_emxx:
+    clang = shared.CLANG_CXX
+  else:
+    clang = shared.CLANG_CC
+
+  # Special case the handling of `-v` because it has a special/different meaning
+  # when used with no other arguments.  In particular, we must handle this early
+  # on, before we inject EMCC_CFLAGS.  This is because tools like cmake and
+  # autoconf will run `emcc -v` to determine the compiler version and we don't
+  # want that to break for users of EMCC_CFLAGS.
+  if len(args) == 2 and args[1] == '-v':
+    # autoconf likes to see 'GNU' in the output to enable shared object support
+    print(version_string(), file=sys.stderr)
+    return shared.check_call([clang, '-v'] + get_clang_flags(), check=False).returncode
+
   # Additional compiler flags that we treat as if they were passed to us on the
   # commandline
   EMCC_CFLAGS = os.environ.get('EMCC_CFLAGS')
@@ -958,6 +973,7 @@ def run(args):
     logger.warning(f'invocation: {cmd} (in {os.getcwd()})')
   if EMCC_CFLAGS:
     args.extend(shlex.split(EMCC_CFLAGS))
+    args += shlex.split(EMCC_CFLAGS)
 
   # Strip args[0] (program name)
   args = args[1:]
@@ -1000,16 +1016,6 @@ There is NO warranty; not even for MERCH
 ''')
     return 0
 
-  if run_via_emxx:
-    clang = shared.CLANG_CXX
-  else:
-    clang = shared.CLANG_CC
-
-  if len(args) == 1 and args[0] == '-v': # -v with no inputs
-    # autoconf likes to see 'GNU' in the output to enable shared object support
-    print(version_string(), file=sys.stderr)
-    return shared.check_call([clang, '-v'] + get_clang_flags(), check=False).returncode
-
   if '-dumpmachine' in args:
     print(get_llvm_target())
     return 0
Only in emscripten-2.0.26: emcc.py.orig
Only in emscripten-2.0.26: emcc.py.rej
diff -urp emscripten-2.0.26.orig/tests/test_other.py emscripten-2.0.26/tests/test_other.py
--- emscripten-2.0.26.orig/tests/test_other.py	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/tests/test_other.py	2023-01-02 09:57:30.263209870 -0800
@@ -154,6 +154,10 @@ class other(RunnerCore):
       os.close(master)
       os.close(slave)
 
+  # Test that running `emcc -v` always works even in the presence of `EMCC_CFLAGS`.
+  # This needs to work because many tools run `emcc -v` internally and it should
+  # always work even if the user has `EMCC_CFLAGS` set.
+  @with_env_modify({'EMCC_CFLAGS': '-should -be -ignored'})
   def test_emcc_v(self):
     for compiler in [EMCC, EMXX]:
       # -v, without input files
Only in emscripten-2.0.26/tests: test_other.py.orig
