--- emscripten-2.0.26/cmake/Modules/Platform/Emscripten.cmake.orig	2021-07-25 10:43:09.000000000 -0700
+++ emscripten-2.0.26/cmake/Modules/Platform/Emscripten.cmake	2023-01-01 21:56:13.584044317 -0800
@@ -124,7 +124,9 @@ if (EMSCRIPTEN_FORCE_COMPILERS)
   # Toolchain script is interpreted multiple times, so don't rerun the check if
   # already done before.
   if (NOT CMAKE_C_COMPILER_VERSION)
-    execute_process(COMMAND "${CMAKE_C_COMPILER}" "-v" RESULT_VARIABLE _cmake_compiler_result ERROR_VARIABLE _cmake_compiler_output OUTPUT_QUIET)
+    execute_process(COMMAND "mktemp" OUTPUT_VARIABLE DUMMY_FILE)
+    file(WRITE "${DUMMY_FILE}" "// dummy")
+    execute_process(COMMAND "${CMAKE_C_COMPILER}" "-v" "${DUMMY_FILE}" RESULT_VARIABLE _cmake_compiler_result ERROR_VARIABLE _cmake_compiler_output OUTPUT_QUIET)
     if (NOT _cmake_compiler_result EQUAL 0)
       message(FATAL_ERROR "Failed to fetch compiler version information with command \"'${CMAKE_C_COMPILER}' -v\"! Process returned with error code ${_cmake_compiler_result}.")
     endif()
