diff -urp emscripten-2.0.14.orig/tools/shared.py emscripten-2.0.14/tools/shared.py
--- emscripten-2.0.14.orig/tools/shared.py	2021-02-14 11:59:37.000000000 -0800
+++ emscripten-2.0.14/tools/shared.py	2021-03-05 11:15:01.856321392 -0800
@@ -333,7 +333,9 @@ def check_sanity(force=False):
 # Some distributions ship with multiple llvm versions so they add
 # the version to the binaries, cope with that
 def build_llvm_tool_path(tool):
-  if config.LLVM_ADD_VERSION:
+  if tool.find('wasm-ld') > -1:
+    return os.path.join('/usr/bin', tool)
+  elif config.LLVM_ADD_VERSION:
     return os.path.join(config.LLVM_ROOT, tool + "-" + config.LLVM_ADD_VERSION)
   else:
     return os.path.join(config.LLVM_ROOT, tool)
Only in emscripten-2.0.14/tools: shared.py.orig
Only in emscripten-2.0.14/tools: shared.py.rej
