diff -urp kpatch-0.9.1.orig/kpatch-build/kpatch-gcc kpatch-0.9.1/kpatch-build/kpatch-gcc
--- kpatch-0.9.1.orig/kpatch-build/kpatch-gcc	2020-03-11 07:02:33.000000000 -0700
+++ kpatch-0.9.1/kpatch-build/kpatch-gcc	2020-09-05 01:16:59.337651319 -0700
@@ -21,6 +21,7 @@ if [[ "$TOOLCHAINCMD" =~ "gcc" ]] ; then
 			# skip copying the temporary .o files created by
 			# recordmcount.pl
 			[[ "$obj" = */.tmp_mc_*.o ]] && break;
+			[[ "$obj" = /tmp/tmp.*.o ]] && break;
 
 			[[ "$obj" = */.tmp_*.o ]] && obj="${obj/.tmp_/}"
 			relobj=${obj//$KPATCH_GCC_SRCDIR\//}
