--- a/deps/npm/node_modules/@npmcli/config/lib/index.js.orig	2023-01-24 08:08:37.000000000 -0800
+++ b/deps/npm/node_modules/@npmcli/config/lib/index.js	2023-01-26 23:52:52.814728491 -0800
@@ -322,7 +322,8 @@ class Config {
     // default the globalconfig file to that location, instead of the default
     // global prefix.  It's weird that `npm get globalconfig --prefix=/foo`
     // returns `/foo/etc/npmrc`, but better to not change it at this point.
-    settableGetter(data, 'globalconfig', () => resolve(this[_get]('prefix'), 'etc/npmrc'))
+    // gentoo deviates wrt global config; store in /etc/npm
+    settableGetter(data, 'globalconfig', () => resolve('/etc', 'npmrc'))
   }
 
   loadHome () {
