--- a/deps/v8/BUILD.gn.orig	2025-02-17 01:42:50.402051961 -0800
+++ b/deps/v8/BUILD.gn	2025-02-17 01:46:29.985103993 -0800
@@ -804,6 +804,7 @@ config("external_startup_data") {
 external_v8_defines = [
   "V8_ENABLE_CHECKS",
   "V8_COMPRESS_POINTERS",
+  "V8_COMPRESS_POINTERS_8GB",
   "V8_COMPRESS_POINTERS_IN_SHARED_CAGE",
   "V8_COMPRESS_POINTERS_IN_ISOLATE_CAGE",
   "V8_31BIT_SMIS_ON_64BIT_ARCH",
@@ -2422,6 +2423,7 @@ action("v8_dump_build_config") {
     "v8_enable_lite_mode=$v8_enable_lite_mode",
     "v8_enable_maglev=$v8_enable_maglev",
     "v8_enable_pointer_compression=$v8_enable_pointer_compression",
+    "v8_enable_pointer_compression_8gb=$v8_enable_pointer_compression_8gb",
     "v8_enable_pointer_compression_shared_cage=" +
         "$v8_enable_pointer_compression_shared_cage",
     "v8_enable_runtime_call_stats=$v8_enable_runtime_call_stats",
--- a/common.gypi.orig	2025-02-17 01:28:56.163591438 -0800
+++ b/common.gypi	2025-02-17 01:46:15.521211092 -0800
@@ -65,6 +65,7 @@
     # options but default values are required here as this file is also used by
     # node-gyp to build addons.
     'v8_enable_pointer_compression%': 0,
+    'v8_enable_pointer_compression_8gb%': 0,
     'v8_enable_31bit_smis_on_64bit_arch%': 0,
 
     # Disable v8 hugepage by default.
@@ -114,6 +115,7 @@
       # V8 pointer compression only supports 64bit architectures.
       ['target_arch in "arm ia32 mips mipsel ppc"', {
         'v8_enable_pointer_compression': 0,
+        'v8_enable_pointer_compression_8gb': 0,
         'v8_enable_31bit_smis_on_64bit_arch': 0,
       }],
       ['target_arch in "ppc64 s390x"', {
@@ -448,6 +450,9 @@
       ['v8_enable_pointer_compression == 1', {
         'defines': ['V8_COMPRESS_POINTERS'],
       }],
+      ['v8_enable_pointer_compression_8gb == 1', {
+        'defines': ['V8_COMPRESS_POINTERS_8GB'],
+      }],
       ['v8_enable_pointer_compression_shared_cage == 1', {
         'defines': ['V8_COMPRESS_POINTERS_IN_SHARED_CAGE'],
       }],
