--- a/configure.py.orig	2026-03-08 11:31:21.073403499 -0700
+++ b/configure.py	2026-03-08 11:32:46.285751372 -0700
@@ -1132,9 +1132,9 @@ parser.add_argument('--v8-with-dchecks',
     default=False,
     help='compile V8 with debug checks and runtime debugging features enabled')
 
-parser.add_argument('--v8-lite-mode',
+parser.add_argument('--v8-enable-lite-mode',
     action='store_true',
-    dest='v8_lite_mode',
+    dest='v8_enable_lite_mode',
     default=False,
     help='compile V8 in lite mode for constrained environments (lowers V8 '+
          'memory footprint, but also implies no just-in-time compilation ' +
@@ -1159,13 +1159,29 @@ parser.add_argument('--v8-enable-hugepag
     help='Enable V8 transparent hugepage support. This feature is only '+
          'available on Linux platform.')
 
-maglev_enabled_by_default_help = f"(Maglev is enabled by default on {','.join(maglev_enabled_architectures)})"
+parser.add_argument('--v8-enable-turbofan',
+    action='store_true',
+    dest='v8_enable_turbofan',
+    default=None,
+    help="Enable V8's turbofan tier.")
+
+parser.add_argument('--v8-enable-sparkplug',
+    action='store_true',
+    dest='v8_enable_sparkplug',
+    default=None,
+    help="Enable V8's sparkplug tier.")
+
+parser.add_argument('--v8-enable-maglev',
+    action='store_true',
+    dest='v8_enable_maglev',
+    default=None,
+    help="Enable V8's Maglev tier.")
 
-parser.add_argument('--v8-disable-maglev',
+parser.add_argument('--v8-enable-drumbrake',
     action='store_true',
-    dest='v8_disable_maglev',
+    dest='v8_enable_drumbrake',
     default=None,
-    help=f"Disable V8's Maglev compiler. {maglev_enabled_by_default_help}")
+    help="Enable V8's drumbrake security hardening for WebAssembly.")
 
 parser.add_argument('--v8-enable-short-builtin-calls',
     action='store_true',
@@ -2005,9 +2021,10 @@ def configure_library(lib, output, pkgna
 def configure_v8(o, configs):
   set_configuration_variable(configs, 'v8_enable_v8_checks', release=0, debug=1)
 
-  o['variables']['v8_enable_webassembly'] = 0 if options.v8_lite_mode else 1
+  o['variables']['v8_enable_webassembly'] = 0 if options.v8_enable_lite_mode else 1
+  o['variables']['v8_enable_drumbrake'] = 1 if options.v8_enable_drumbrake else 0
   o['variables']['v8_enable_javascript_promise_hooks'] = 1
-  o['variables']['v8_enable_lite_mode'] = 1 if options.v8_lite_mode else 0
+  o['variables']['v8_enable_lite_mode'] = 1 if options.v8_enable_lite_mode else 0
   is_gdbjit_supported_arch = (
       'x64' in o['variables']['target_arch'] or
       'ia32' in o['variables']['target_arch'] or
@@ -2024,8 +2041,11 @@ def configure_v8(o, configs):
   o['variables']['v8_random_seed'] = 0  # Use a random seed for hash tables.
   o['variables']['v8_promise_internal_field_count'] = 1 # Add internal field to promises for async hooks.
   o['variables']['v8_use_siphash'] = 0 if options.without_siphash else 1
-  o['variables']['v8_enable_maglev'] = B(not options.v8_disable_maglev and
-                                         o['variables']['target_arch'] in maglev_enabled_architectures)
+  o['variables']['v8_enable_turbofan'] = 1 if options.v8_enable_turbofan else 0
+  o['variables']['v8_enable_sparkplug'] = 1 if options.v8_enable_sparkplug else 0
+  o['variables']['v8_enable_maglev'] = 1 if options.v8_enable_maglev else 0
+  if not o['variables']['target_arch'] in maglev_enabled_architectures:
+    o['variables']['v8_enable_maglev'] = 0
   o['variables']['v8_enable_pointer_compression'] = 1 if options.enable_pointer_compression else 0
   # Using the sandbox requires always allocating array buffer backing stores in the sandbox.
   # We currently have many backing stores tied to pointers from C++ land that are not
@@ -2036,6 +2056,12 @@ def configure_v8(o, configs):
   if options.enable_v8_sandbox and not options.enable_pointer_compression:
     raise Exception(
         '--experimental-enable-v8-sandbox needs --experimental-enable-pointer-compression')
+  if options.v8_enable_lite_mode:
+    o['variables']['v8_enable_webassembly'] = 0
+    o['variables']['v8_enable_turbofan'] = 0
+    o['variables']['v8_enable_sparkplug'] = 0
+    o['variables']['v8_enable_maglev'] = 0
+    o['variables']['v8_enable_gdbjit'] = 0
   o['variables']['v8_enable_sandbox'] = 1 if options.enable_v8_sandbox else 0
   # We set v8_enable_pointer_compression_shared_cage to 0 always, even when
   # pointer compression is enabled so that we don't accidentally enable shared
