--- a/tensorflow_estimator/tools/pip_package/setup.py.orig	2023-03-24 17:42:32.878994951 -0700
+++ b/tensorflow_estimator/tools/pip_package/setup.py	2023-03-24 17:43:02.774995338 -0700
@@ -30,7 +30,7 @@ DOCLINES = __doc__.split('\n')
 # This version string is semver compatible, but incompatible with pip.
 # For pip, we will remove all '-' characters from this string, and use the
 # result for pip.
-_VERSION = '2.12.0-rc0'
+_VERSION = '2.12.0' # oteodoro:  removed suffix
 
 REQUIRED_PACKAGES = [
     # We depend on TensorFlow's declared pip dependencies.
