--- wireshark-1.99.9.orig/epan/dissectors/packet-usb-ptp.c	2015-10-05 18:54:57.066634595 -0700
+++ wireshark-1.99.9/epan/dissectors/packet-usb-ptp.c	2015-10-05 19:02:37.665601255 -0700
@@ -378,7 +378,7 @@
     usb_ptp_device_info = usb_ptp_conv_info->device_info;
     if (!usb_ptp_device_info)
     {
-        usb_ptp_device_info = (usb_ptp_device_info_t *) se_alloc0(sizeof(usb_ptp_device_info_t));
+        usb_ptp_device_info = (usb_ptp_device_info_t *) wmem_alloc0(wmem_epan_scope(), sizeof(usb_ptp_device_info_t));
     }
 
     /* Create Device Info Tree */
@@ -564,7 +564,7 @@
     usb_ptp_conv_info = (usb_ptp_conv_info_t *) usb_conv_info->class_data;
     if(!usb_ptp_conv_info)
     {
-        usb_ptp_conv_info = (usb_ptp_conv_info_t *) se_alloc0(sizeof(usb_ptp_conv_info_t));
+        usb_ptp_conv_info = (usb_ptp_conv_info_t *) wmem_alloc0(wmem_epan_scope(), sizeof(usb_ptp_conv_info_t));
         usb_conv_info->class_data = usb_ptp_conv_info;
         usb_ptp_conv_info->flavor = usb_ptp_flavor(usb_conv_info);
     }
