--- wireshark-1.99.9.orig/epan/dissectors/packet-usb-ptp.c	2015-10-05 20:18:29.519271767 -0700
+++ wireshark-1.99.9/epan/dissectors/packet-usb-ptp.c	2015-10-05 20:50:40.580131987 -0700
@@ -429,12 +429,10 @@
 static void
 dissect_usb_ptp_params(tvbuff_t *tvb, packet_info *pinfo _U_, proto_tree *parent_tree, gint offset,gint hf)
 {
-    guint length_tvb;
     guint32 remaining;
     proto_tree *tree = NULL;
 
-    length_tvb = tvb_length(tvb);
-    remaining = length_tvb-offset;
+    remaining = tvb_captured_length_remaining(tvb,offset);
 
     if (!remaining)
         return;
@@ -606,7 +604,7 @@
 				new_tvb = process_reassembled_data(tvb, offset, pinfo,
 					"Reassembled PTP/MTP data packet", head, &mtp_msg_frag_items,
 					NULL, tree);
-				if (tvb_length(tvb) == ptp_length) {
+				if (tvb_captured_length(tvb) == ptp_length) {
 					pinfo->fragmented = FALSE;
 					complete = TRUE;
 				} else {
@@ -621,7 +619,7 @@
 				complete = TRUE;
 				break;
 			default:
-				head = fragment_add_seq_next(&msg_reassembled_table, tvb, offset, pinfo, pinfo->p2p_dir, NULL, tvb_length(tvb), TRUE);
+				head = fragment_add_seq_next(&msg_reassembled_table, tvb, offset, pinfo, pinfo->p2p_dir, NULL, tvb_captured_length(tvb), TRUE);
 				new_tvb = process_reassembled_data(tvb, offset, pinfo,
 					"Reassembled PTP/MTP data packet", head, &mtp_msg_frag_items,
 					NULL, tree);
@@ -742,7 +740,7 @@
 						 (pinfo->p2p_dir==P2P_DIR_SENT) ? "Sent" : "Rcvd");
 		}
 
-		offset+=tvb_length(next_tvb);
+		offset+=tvb_captured_length(next_tvb);
 
         call_dissector(data_handle, next_tvb, pinfo, tree);
     }
