--- wireshark-1.12.6.orig/epan/dissectors/packet-usb-ptp.c	2015-07-07 18:54:52.950043048 -0700
+++ wireshark-1.12.6/epan/dissectors/packet-usb-ptp.c	2015-07-07 19:22:00.376051438 -0700
@@ -216,11 +216,12 @@
 {
     guint8 length;
     gchar   *str;
+    gint lengthp;
 
     /* First byte is the number of characters in UCS-2, including the terminating NULL */
 	length = tvb_get_guint8(tvb, offset) * 2;
     offset += 1;
-    str = tvb_get_ephemeral_unicode_string(tvb, offset, length, ENC_LITTLE_ENDIAN);
+    str = tvb_get_stringz_enc(wmem_packet_scope(), tvb, offset, &lengthp, ENC_ASCII);
     proto_tree_add_string(tree, hf, tvb, offset, length, str);
     offset += length;
 
