--- wireshark-1.99.9.orig/epan/dissectors/packet-usb-ptp.c	2015-10-05 18:10:18.884828456 -0700
+++ wireshark-1.99.9/epan/dissectors/packet-usb-ptp.c	2015-10-05 18:22:47.260774285 -0700
@@ -962,7 +962,7 @@
     usb_ptp_dissector_handle = find_dissector("usb-ptp");
     //dissector_add_uint("usb.bulk", IF_CLASS_IMAGE, usb_ptp_dissector_handle);
     //dissector_add_uint("usb.bulk", IF_CLASS_VENDOR_SPECIFIC, usb_ptp_dissector_handle); //for zune
-    heur_dissector_add("usb.bulk", dissect_usb_ptp, proto_usb_ptp);
+    heur_dissector_add("usb.bulk", dissect_usb_ptp, "PTP USB bulk endpoint", "ptp_usb_bulk", proto_usb_ptp, HEURISTIC_ENABLE);
 
     data_handle = find_dissector("data");
 }
