diff -ru boinc-client_release-7.4-7.4.42.orig/sched/handle_request.cpp boinc-client_release-7.4-7.4.42/sched/handle_request.cpp
--- boinc-client_release-7.4-7.4.42.orig/sched/handle_request.cpp	2015-12-28 21:02:16.248127012 -0800
+++ boinc-client_release-7.4-7.4.42/sched/handle_request.cpp	2015-12-28 21:03:45.376507004 -0800
@@ -1445,6 +1445,7 @@
     }
 
+    debug_sched("debug_sched");
     sreply.write(fout, sreq);
     log_messages.printf(MSG_NORMAL,
         "Scheduler ran %.3f seconds\n", dtime()-start_time
     );
