diff -ur boinc-client_release-7.4-7.4.42.orig/html/ops/size_census.php boinc-client_release-7.4-7.4.42/html/ops/size_census.php
--- boinc-client_release-7.4-7.4.42.orig/html/ops/size_census.php	2016-01-09 20:58:25.975626706 -0800
+++ boinc-client_release-7.4-7.4.42/html/ops/size_census.php	2016-01-09 21:02:23.686469231 -0800
@@ -64,7 +64,8 @@
     $f = fopen("../../size_census_".$app->name, "w");
     for ($i=1; $i<$app->n_size_classes; $i++) {
         $k = (int)(($i*$n)/$app->n_size_classes);
-        fprintf($f, "%e\n", $a[$k]);
+	if (array_key_exists($k,$a))
+	        fprintf($f, "%e\n", $a[$k]);
     }
     fclose($f);
 }
diff -ur boinc-client_release-7.4-7.4.42.orig/html/ops/update_profile_pages.php boinc-client_release-7.4-7.4.42/html/ops/update_profile_pages.php
--- boinc-client_release-7.4-7.4.42.orig/html/ops/update_profile_pages.php	2016-01-09 20:58:25.976626742 -0800
+++ boinc-client_release-7.4-7.4.42/html/ops/update_profile_pages.php	2016-01-09 20:59:17.193531366 -0800
@@ -304,6 +304,7 @@
     print_debug_msg("$numIds users have profiles AND names.");
 
     $letters = array_keys($members);
+    $letters_used = array();
 
     foreach ($letters as $letter) {
         // NOTE: Array indexing is case sensitive.
