--- boinc-client_release-7.4-7.4.42.orig/sched/sched_send.cpp	2016-01-09 16:59:14.592394858 -0800
+++ boinc-client_release-7.4-7.4.42/sched/sched_send.cpp	2016-01-09 17:04:13.900670541 -0800
@@ -1169,7 +1169,7 @@
     bool usable_gpu = false;
     for (int i=1; i<NPROC_TYPES; i++) {
         COPROC* cp = g_request->coprocs.proc_type_to_coproc(i);
-        if (ssp->have_apps_for_proc_type[i] && cp->count) {
+        if (cp && ssp->have_apps_for_proc_type[i] && cp->count) {
             usable_gpu = true;
             break;
         }
@@ -1485,7 +1485,7 @@
         );
         for (i=1; i<NPROC_TYPES; i++) {
             COPROC* cp = g_request->coprocs.proc_type_to_coproc(i);
-            if (cp->count) {
+            if (cp && cp->count) {
                 log_messages.printf(MSG_NORMAL,
                     "[send] %s: req %.2f sec, %.2f instances; est delay %.2f\n",
                     proc_type_name(i),
