diff -ru boinc-client_release-7.4-7.4.42.orig/sched/Makefile.am boinc-client_release-7.4-7.4.42/sched/Makefile.am
--- boinc-client_release-7.4-7.4.42.orig/sched/Makefile.am	2015-12-26 08:30:34.185251807 -0800
+++ boinc-client_release-7.4-7.4.42/sched/Makefile.am	2015-12-26 08:31:16.939855723 -0800
@@ -27,7 +27,7 @@
 libsched_la_CFLAGS = $(AM_CPPFLAGS)
 libsched_la_CXXFLAGS = $(AM_CPPFLAGS)
 libsched_la_LDFLAGS= -L$(libdir) -rpath $(libdir) -version-number $(LIBBOINC_VERSION)
-libsched_la_LIBADD= $(SSL_LIBS)
+libsched_la_LIBADD= $(SSL_LIBS) $(MYSQL_LIBS)
 
 ## install only headers that are meant for exporting the API !!
 if INSTALL_HEADERS
