diff -urp devhub-0.102.0_p9999.orig/@types/react-native-web/index.d.ts devhub-0.102.0_p9999/@types/react-native-web/index.d.ts
--- devhub-0.102.0_p9999.orig/@types/react-native-web/index.d.ts	2023-01-07 18:21:14.281848521 -0800
+++ devhub-0.102.0_p9999/@types/react-native-web/index.d.ts	2023-01-07 18:24:06.312765410 -0800
@@ -15,6 +15,7 @@ namespace JSX {
       | 'contentinfo' // footer
       | 'form' // form
       | 'heading' // h1, h2, h3, ...
+      | 'image' // icon
       | 'label' // label
       | 'link' // a
       | 'list' // ul
@@ -22,6 +23,7 @@ namespace JSX {
       | 'main' // main
       | 'navigation' // nav
       | 'region' // section
+      | 'togglebutton' // togglebutton
     accessible?: boolean
     className?: string
     tabIndex?: number
diff -urp devhub-0.102.0_p9999.orig/packages/components/src/components/themed/ThemedIcon.tsx devhub-0.102.0_p9999/packages/components/src/components/themed/ThemedIcon.tsx
--- devhub-0.102.0_p9999.orig/packages/components/src/components/themed/ThemedIcon.tsx	2023-01-07 18:21:23.758229152 -0800
+++ devhub-0.102.0_p9999/packages/components/src/components/themed/ThemedIcon.tsx	2023-01-07 19:37:36.547868943 -0800
@@ -26,6 +26,7 @@ export const ThemedIcon = React.memo(
       color: _color,
       family = 'octicon',
       size = normalTextSize,
+      accessibilityRole = 'image',
       themeTransformer,
       ...otherProps
     } = props
