--- a/src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs.orig	2019-04-05 17:15:53.671446980 -0700
+++ b/src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs	2019-04-05 19:48:26.158692324 -0700
@@ -6621,7 +6621,6 @@ namespace Microsoft.AspNetCore.Server.Ke
 
         protected override bool RemoveFast(string key)
         {
-            value = default;
             switch (key.Length)
             {
                 case 13:
