--- a/src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs.orig	2019-07-19 23:54:39.908259948 -0700
+++ b/src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs	2019-07-20 03:33:38.783729442 -0700
@@ -5454,6 +5454,7 @@ namespace Microsoft.AspNetCore.Server.Ke
 
         protected override bool TryGetValueFast(string key, out StringValues value)
         {
+            value = default;
             switch (key.Length)
             {
                 case 13:
@@ -9016,7 +9017,7 @@ namespace Microsoft.AspNetCore.Server.Ke
         {
             get
             {
-                StringValues value;
+                StringValues value = default;
                 if ((_bits & 1L) != 0)
                 {
                     value = _headers._ETag;
