--- a/src/SignalR/perf/Microbenchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks.csproj.orig	2019-04-05 11:50:07.253790087 -0700
+++ b/src/SignalR/perf/Microbenchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks.csproj	2019-04-07 18:21:04.386006621 -0700
@@ -10,7 +10,7 @@
     <Compile Include="$(SignalRSharedSourceRoot)BinaryMessageParser.cs" Link="BinaryMessageParser.cs" />
   </ItemGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition="'$(CompileOnly)' == 'false'">
     <ProjectReference Include="$(SignalRTestUtilsProject)" />
   </ItemGroup>
 
