--- a/eng/targets/Npm.Common.targets	2019-04-05 11:50:00.734550895 -0700
+++ b/eng/targets/Npm.Common.targets	2019-04-07 11:36:23.419117975 -0700
@@ -64,7 +64,7 @@
     <Move SourceFiles="$(_BackupPackageJson)" DestinationFiles="$(PackageJson)" />
   </Target>
 
-  <Target Name="Test" Condition="'$(IsTestProject)' == 'true'">
+  <Target Name="Test" Condition="'$(IsTestProject)' == 'true' and '$(CompileOnly)' == 'false'">
     <Message Importance="High" Text="Running npm tests for $(MSBuildProjectName)" />
     <Exec Command="npm $(NpmTestArgs)" IgnoreStandardErrorWarningFormat="true" />
   </Target>
