--- a/build.sh.orig	2020-06-20 19:43:03.163315998 -0700
+++ b/build.sh	2020-06-20 19:48:24.277445754 -0700
@@ -350,6 +350,7 @@ if [ "$build_repo_tasks" = true ]; then
         -p:Projects="$DIR/eng/tools/RepoTasks/RepoTasks.csproj" \
         -p:Configuration=Release \
         -p:Restore=$run_restore \
+        -p:RestoreDisableParallel=true \
         -p:Build=true \
         -clp:NoSummary \
         ${toolset_build_args[@]+"${toolset_build_args[@]}"}
--- a/eng/common/tools.sh.orig	2020-06-20 19:43:05.723438504 -0700
+++ b/eng/common/tools.sh	2020-06-20 19:48:26.907553132 -0700
@@ -334,7 +334,7 @@ function InitializeToolset {
   fi
   
   echo '<Project Sdk="Microsoft.DotNet.Arcade.Sdk"/>' > "$proj"
-  MSBuild-Core "$proj" $bl /t:__WriteToolsetLocation /clp:ErrorsOnly\;NoSummary /p:__ToolsetLocationOutputFile="$toolset_location_file"
+  MSBuild-Core "$proj" $bl /p:RestoreDisableParallel=true /t:__WriteToolsetLocation /clp:ErrorsOnly\;NoSummary /p:__ToolsetLocationOutputFile="$toolset_location_file"
 
   local toolset_build_proj=`cat "$toolset_location_file"`
 
