--- a/AspNetCore-2.1.9/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj.orig	2019-04-04 15:15:49.488482219 -0700
+++ b/AspNetCore-2.1.9/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj	2019-04-04 15:27:20.602279307 -0700
@@ -5,6 +5,7 @@
     <IsPackable>false</IsPackable>
     <IsTestProject>true</IsTestProject>
     <NpmTestArgs>run test:local -- --no-color --configuration $(Configuration)</NpmTestArgs>
+    <SkipTests>true</SkipTests>
   </PropertyGroup>
 
   <ItemGroup>
