diff -urp FNA-18.01.02.orig/FNA.csproj FNA-18.01.02/FNA.csproj
--- FNA-18.01.02.orig/FNA.csproj	2018-01-02 11:28:52.000000000 -0800
+++ FNA-18.01.02/FNA.csproj	2018-01-09 21:32:47.979386549 -0800
@@ -400,17 +400,6 @@
     <Compile Include="src\Vector2.cs" />
     <Compile Include="src\Vector3.cs" />
     <Compile Include="src\Vector4.cs" />
-    <Compile Include="lib\SDL2-CS\src\SDL2.cs" />
-    <Compile Include="lib\SDL2-CS\src\SDL2_image.cs" />
-    <Compile Include="lib\OpenAL-CS\src\ALC10.cs" />
-    <Compile Include="lib\OpenAL-CS\src\ALC11.cs" />
-    <Compile Include="lib\OpenAL-CS\src\AL10.cs" />
-    <Compile Include="lib\OpenAL-CS\src\AL11.cs" />
-    <Compile Include="lib\OpenAL-CS\src\ALEXT.cs" />
-    <Compile Include="lib\OpenAL-CS\src\EFX.cs" />
-    <Compile Include="lib\MojoShader-CS\MojoShader.cs" />
-    <Compile Include="lib\Vorbisfile-CS\Vorbisfile.cs" />
-    <Compile Include="lib\Theorafile\csharp\Theorafile.cs" />
   </ItemGroup>
   <ItemGroup>
     <None Include="FNA.dll.config">
