diff -urp FNA-17.01.orig/FNA.csproj FNA-17.01/FNA.csproj
--- FNA-17.01.orig/FNA.csproj	2017-01-01 08:24:15.000000000 -0800
+++ FNA-17.01/FNA.csproj	2017-01-08 03:13:00.179291616 -0800
@@ -392,18 +392,6 @@
     <Compile Include="src\Vector2.cs" />
     <Compile Include="src\Vector3.cs" />
     <Compile Include="src\Vector4.cs" />
-    <Compile Include="lib\SDL2-CS\src\SDL2.cs" />
-    <Compile Include="lib\SDL2-CS\src\SDL2_image.cs" />
-    <Compile Include="lib\SDL2-CS\src\LPUtf8StrMarshaler.cs" />
-    <Compile Include="lib\OpenAL-CS\src\ALC10.cs" />
-    <Compile Include="lib\OpenAL-CS\src\ALC11.cs" />
-    <Compile Include="lib\OpenAL-CS\src\AL10.cs" />
-    <Compile Include="lib\OpenAL-CS\src\AL11.cs" />
-    <Compile Include="lib\OpenAL-CS\src\ALEXT.cs" />
-    <Compile Include="lib\OpenAL-CS\src\EFX.cs" />
-    <Compile Include="lib\MojoShader-CS\MojoShader.cs" />
-    <Compile Include="lib\Vorbisfile-CS\Vorbisfile.cs" />
-    <Compile Include="lib\TheoraPlay-CS\TheoraPlay.cs" />
   </ItemGroup>
   <ItemGroup>
     <None Include="FNA.dll.config">
