diff -urp monodevelop-6.1.2.44.orig/main/external/guiunit/src/framework/GuiUnit_NET_4_5.csproj monodevelop-6.1.2.44/main/external/guiunit/src/framework/GuiUnit_NET_4_5.csproj
--- monodevelop-6.1.2.44.orig/main/external/guiunit/src/framework/GuiUnit_NET_4_5.csproj	2017-02-12 00:16:44.015971547 -0800
+++ monodevelop-6.1.2.44/main/external/guiunit/src/framework/GuiUnit_NET_4_5.csproj	2017-02-12 07:59:12.169613446 -0800
@@ -66,6 +66,7 @@
     <Reference Include="System.Xml" />
     <Reference Include="System.Core" />
     <Reference Include="System.Xml.Linq" />
+    <Reference Include="System.Runtime.CompilerServices" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Api\ExpectedExceptionData.cs" />
diff -urp monodevelop-6.1.2.44.orig/main/external/RefactoringEssentials/RefactoringEssentials/RefactoringEssentials.csproj monodevelop-6.1.2.44/main/external/RefactoringEssentials/RefactoringEssentials/RefactoringEssentials.csproj
--- monodevelop-6.1.2.44.orig/main/external/RefactoringEssentials/RefactoringEssentials/RefactoringEssentials.csproj	2017-02-12 00:16:44.051972938 -0800
+++ monodevelop-6.1.2.44/main/external/RefactoringEssentials/RefactoringEssentials/RefactoringEssentials.csproj	2017-02-12 07:59:29.497257004 -0800
@@ -12,7 +12,6 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>RefactoringEssentials</RootNamespace>
     <AssemblyName>RefactoringEssentials</AssemblyName>
-    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>
     <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
