diff -urp monodevelop-6.1.2.44.orig/main/src/addins/GnomePlatform/GnomePlatform.cs monodevelop-6.1.2.44/main/src/addins/GnomePlatform/GnomePlatform.cs
--- monodevelop-6.1.2.44.orig/main/src/addins/GnomePlatform/GnomePlatform.cs	2017-01-03 06:02:42.461350155 -0800
+++ monodevelop-6.1.2.44/main/src/addins/GnomePlatform/GnomePlatform.cs	2017-01-03 06:09:55.185749236 -0800
@@ -88,7 +88,7 @@ namespace MonoDevelop.Platform
 
 		public override void ShowUrl (string url)
 		{
-			Gnome.Url.Show (url);
+			//Gnome.Url.Show (url);
 		}
 		
 		public override string DefaultMonospaceFont {
