--- a/main/external/mono-addins/Mono.Addins/Mono.Addins.csproj.orig	2019-09-05 00:14:22.798596918 -0700
+++ b/main/external/mono-addins/Mono.Addins/Mono.Addins.csproj	2019-09-05 01:42:03.219479914 -0700
@@ -47,6 +47,10 @@
     <Reference Include="System.Xml" />
     <Reference Include="System.Core" />
     <PackageReference Include="NuGet.Build.Packaging" Version="0.2.0" />
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="AssemblyInfo.cs" />
--- a/main/external/xwt/Xwt/Xwt.csproj.orig	2019-09-05 00:14:22.481584744 -0700
+++ b/main/external/xwt/Xwt/Xwt.csproj	2019-09-05 01:43:54.158761179 -0700
@@ -40,6 +40,10 @@
     <Reference Include="System.Core" />
     <Reference Include="System.Xml" />
     <Reference Include="System.Xaml" />
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="AssemblyInfo.cs" />
@@ -375,4 +379,4 @@
   <ItemGroup />
   <ItemGroup />
   <ItemGroup />
-</Project>
\ No newline at end of file
+</Project>
--- a/main/external/nrefactory/ICSharpCode.NRefactory/ICSharpCode.NRefactory.csproj.orig	2019-09-05 00:14:23.930640389 -0700
+++ b/main/external/nrefactory/ICSharpCode.NRefactory/ICSharpCode.NRefactory.csproj	2019-09-05 01:46:39.436141261 -0700
@@ -83,6 +83,10 @@
     <Reference Include="System.Core" />
     <Reference Include="System.Xml" />
     <Reference Include="System.Xml.Linq" />
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Analysis\AbiComparer.cs" />
--- a/main/external/guiunit/src/framework/GuiUnit_NET_4_5.csproj.orig	2019-09-05 00:14:22.732594383 -0700
+++ b/main/external/guiunit/src/framework/GuiUnit_NET_4_5.csproj	2019-09-05 01:48:31.082452278 -0700
@@ -68,6 +68,10 @@
     <Reference Include="System.Xml" />
     <Reference Include="System.Core" />
     <Reference Include="System.Xml.Linq" />
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>    
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Api\ExpectedExceptionData.cs" />
--- a/main/src/addins/MonoDevelop.GtkCore/libstetic/libstetic.csproj.orig	2019-09-05 00:14:24.683669305 -0700
+++ b/main/src/addins/MonoDevelop.GtkCore/libstetic/libstetic.csproj	2019-09-05 01:49:55.800724154 -0700
@@ -50,6 +50,10 @@
     <Reference Include="glade-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f" />
     <Reference Include="Mono.Cairo" />
     <Reference Include="System.Core" />
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>    
   </ItemGroup>
   <ItemGroup>
     <Compile Include="ClassDescriptor.cs" />
--- a/main/external/RefactoringEssentials/RefactoringEssentials/RefactoringEssentials.csproj.orig	2019-09-05 00:16:09.197682960 -0700
+++ b/main/external/RefactoringEssentials/RefactoringEssentials/RefactoringEssentials.csproj	2019-09-05 01:51:27.432263626 -0700
@@ -26,6 +26,10 @@
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="2.9.0-beta4-63001-02" />
     <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic" Version="2.9.0-beta4-63001-02" />
     <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic.Workspaces" Version="2.9.0-beta4-63001-02" />
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>     
   </ItemGroup>
   <ItemGroup>
     <None Update="RefactoringEssentials.Library.nuspec">
--- a/main/external/debugger-libs/Mono.Debugger.Soft/Mono.Debugger.Soft.csproj.orig	2019-09-05 00:14:23.914639774 -0700
+++ b/main/external/debugger-libs/Mono.Debugger.Soft/Mono.Debugger.Soft.csproj	2019-09-05 01:53:55.012965543 -0700
@@ -135,5 +135,9 @@
     <Reference Include="Mono.Cecil.Rocks">
       <HintPath>$(SolutionDir)\packages\Mono.Cecil.0.10.0-beta6\lib\net40\Mono.Cecil.Rocks.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>
   </ItemGroup>
 </Project>
--- a/main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildBuilder.csproj.orig	2019-09-05 00:14:23.973642040 -0700
+++ b/main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildBuilder.csproj	2019-09-05 01:56:03.442928803 -0700
@@ -84,6 +84,10 @@
       <HintPath>..\..\..\packages\System.ValueTuple.4.4.0\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>    
   </ItemGroup>
   <ItemGroup Condition=" '$(Configuration)' == 'Debug' OR '$(Configuration)' == 'Release' ">
     <None Include="app.config">
--- a/main/src/addins/MonoDevelop.UnitTesting.NUnit/NUnitRunner/NUnitRunner.csproj.orig	2019-09-05 00:14:24.258652984 -0700
+++ b/main/src/addins/MonoDevelop.UnitTesting.NUnit/NUnitRunner/NUnitRunner.csproj	2019-09-05 01:52:35.974911641 -0700
@@ -54,6 +54,10 @@
       <HintPath>..\..\..\..\packages\System.ValueTuple.4.4.0\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Microsoft.Build.Utilities.Core">
+      <Private>False</Private>
+      <HintPath>$(MSBuildToolsPath)\Microsoft.Build.Utilities.Core.dll</HintPath>
+    </Reference>       
   </ItemGroup>
   <ItemGroup>
     <Compile Include="NUnitTestRunner.cs" />
--- a/main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildResolver/MonoDevelop.MSBuildResolver.csproj.orig	2019-09-05 00:14:23.972642001 -0700
+++ b/main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildResolver/MonoDevelop.MSBuildResolver.csproj	2019-09-05 02:04:57.178567035 -0700
@@ -37,15 +37,15 @@
   <ItemGroup>
     <Reference Include="System" />
     <Reference Include="Microsoft.Build">
-      <HintPath>$(MSBuild_OSS_BinDir)Microsoft.Build.dll</HintPath>
+      <HintPath>$(MSBuildToolsPath)/Microsoft.Build.dll</HintPath>
       <Private>False</Private>
     </Reference>
     <Reference Include="Microsoft.Build.Framework">
-      <HintPath>$(MSBuild_OSS_BinDir)Microsoft.Build.Framework.dll</HintPath>
+      <HintPath>$(MSBuildToolsPath)/Microsoft.Build.Framework.dll</HintPath>
       <Private>False</Private>
     </Reference>
     <Reference Include="Microsoft.Build.Utilities.Core">
-      <HintPath>$(MSBuild_OSS_BinDir)Microsoft.Build.Utilities.Core.dll</HintPath>
+      <HintPath>$(MSBuildToolsPath)/Microsoft.Build.Utilities.Core.dll</HintPath>
       <Private>False</Private>
     </Reference>
   </ItemGroup>
