--- a/main/configure.ac.orig	2019-09-04 00:40:44.298852151 -0700
+++ b/main/configure.ac	2019-09-04 00:44:06.858479014 -0700
@@ -28,9 +28,9 @@ if test "x$MCS" = "x" ; then
   AC_MSG_ERROR([Can't find "mcs" in your PATH])
 fi
 
-if test "x$FSHARPC" = "x" ; then
-  AC_MSG_ERROR([Can't find "fsharpc" in your PATH])
-fi
+#if test "x$FSHARPC" = "x" ; then
+#  AC_MSG_ERROR([Can't find "fsharpc" in your PATH])
+#fi
 
 dnl Find pkg-config
 AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
