--- a/main/Makefile.am.orig	2019-09-04 16:26:49.488225037 -0700
+++ b/main/Makefile.am	2019-09-04 16:26:43.809005037 -0700
@@ -28,7 +28,6 @@ NUGET_RESTORE = mono external/nuget-bina
 #       see: https://github.com/kzu/NuGet.Restore
 restore-packages:
 	@$(NUGET_RESTORE)
-	msbuild /t:Restore /p:RestoreDisableParallel=true external/RefactoringEssentials/RefactoringEssentials/RefactoringEssentials.csproj
 	mono external/fsharpbinding/.paket/paket.bootstrapper.exe
 	pushd . && cd external/fsharpbinding && mono .paket/paket.exe restore && popd
 
