diff -urp Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e.orig/Source/Eto.Gtk/Forms/Controls/GridViewHandler.cs Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e/Source/Eto.Gtk/Forms/Controls/GridViewHandler.cs
--- Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e.orig/Source/Eto.Gtk/Forms/Controls/GridViewHandler.cs	2017-12-18 16:45:49.000000000 -0800
+++ Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e/Source/Eto.Gtk/Forms/Controls/GridViewHandler.cs	2017-12-31 20:16:19.907918312 -0800
@@ -196,7 +196,9 @@ namespace Eto.GtkSharp.Forms.Controls
 				GridViewDragInfo dragInfo = _dragInfo;
 				if (dragInfo == null && location != null)
 				{
-					if (t.GetDestRowAtPos((int)location.Value.X, (int)location.Value.Y, out var path, out var pos))
+					Gtk.TreePath path;
+					Gtk.TreeViewDropPosition pos;
+					if (t.GetDestRowAtPos((int)location.Value.X, (int)location.Value.Y, out path, out pos))
 					{
 						var item = Handler.model.GetItemAtPath(path);
 						var indecies = path.Indices;
diff -urp Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e.orig/Source/Eto.Gtk/Forms/Controls/TreeGridViewHandler.cs Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e/Source/Eto.Gtk/Forms/Controls/TreeGridViewHandler.cs
--- Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e.orig/Source/Eto.Gtk/Forms/Controls/TreeGridViewHandler.cs	2017-12-18 16:45:49.000000000 -0800
+++ Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e/Source/Eto.Gtk/Forms/Controls/TreeGridViewHandler.cs	2017-12-31 20:15:29.942919112 -0800
@@ -264,7 +264,9 @@ namespace Eto.GtkSharp.Forms.Controls
 				TreeGridViewDragInfo dragInfo = _dragInfo;
 				if (dragInfo == null && location != null)
 				{
-					if (t.GetDestRowAtPos((int)location.Value.X, (int)location.Value.Y, out var path, out var pos))
+					Gtk.TreePath path;
+					Gtk.TreeViewDropPosition pos;
+					if (t.GetDestRowAtPos((int)location.Value.X, (int)location.Value.Y, out path, out pos))
 					{
 						var item = h.model.GetItemAtPath(path);
 						var indices = path.Indices;
