diff -urp Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e.orig/Source/Eto.Gtk/Forms/GtkControl.cs Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e/Source/Eto.Gtk/Forms/GtkControl.cs
--- Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e.orig/Source/Eto.Gtk/Forms/GtkControl.cs	2017-12-31 20:32:42.808902578 -0800
+++ Eto-d7a6d4bbdb5ac1263d3036d9177cadcfb2f0e63e/Source/Eto.Gtk/Forms/GtkControl.cs	2017-12-31 20:34:09.385901192 -0800
@@ -783,7 +783,7 @@ namespace Eto.GtkSharp.Forms
 
 		public void DoDragDrop(DataObject data, DragEffects allowedEffects)
 		{
-			var targets = (data.Handler as DataObjectHandler)?.GetTargets();
+			//var targets = (data.Handler as DataObjectHandler)?.GetTargets();
 
 			DragInfo = new DragInfoObject { Data = data, AllowedEffects = allowedEffects };
 
