diff -urp idb-connector.orig/binding.gyp idb-connector/binding.gyp
--- idb-connector.orig/binding.gyp	1985-10-26 01:15:00.000000000 -0700
+++ idb-connector/binding.gyp	2019-02-23 10:20:12.852407913 -0800
@@ -16,7 +16,6 @@
         '-std=c++0x',
         '-Wno-unknown-pragmas',
         '-Wno-format',
-        '-gxcoff',
         '-O0',
         '-DNAPI_DISABLE_CPP_EXCEPTIONS',
         '-I/QOpenSys/pkgs/include'
@@ -45,4 +44,4 @@
       ]
     }
   ]
-}
\ No newline at end of file
+}
