diff -ur llvm-9999.orig/tools/clang/CMakeLists.txt llvm-9999/tools/clang/CMakeLists.txt
--- llvm-9999.orig/tools/clang/CMakeLists.txt	2016-01-19 21:33:04.221637342 -0800
+++ llvm-9999/tools/clang/CMakeLists.txt	2016-01-19 22:04:46.242848723 -0800
@@ -278,6 +278,8 @@
   ${CMAKE_CURRENT_SOURCE_DIR}/include/clang/Basic/Version.inc.in
   ${CMAKE_CURRENT_BINARY_DIR}/include/clang/Basic/Version.inc)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+
 # Add appropriate flags for GCC
 if (LLVM_COMPILER_IS_GCC_COMPATIBLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-common -Woverloaded-virtual")
diff -ur llvm-9999.orig/tools/clang/lib/Driver/ToolChains.h llvm-9999/tools/clang/lib/Driver/ToolChains.h
--- llvm-9999.orig/tools/clang/lib/Driver/ToolChains.h	2016-01-19 21:33:05.224675441 -0800
+++ llvm-9999/tools/clang/lib/Driver/ToolChains.h	2016-01-19 22:11:59.723304406 -0800
@@ -781,10 +781,10 @@
   Tool *buildLinker() const override;
 };
 
-class LLVM_LIBRARY_VISIBILITY Linux : public Generic_ELF {
+class Linux : public Generic_ELF {
 public:
   Linux(const Driver &D, const llvm::Triple &Triple,
-        const llvm::opt::ArgList &Args);
+        const llvm::opt::ArgList &Args) __attribute__ ((visibility ("default") ));
 
   bool HasNativeLLVMSupport() const override;
 
diff -ur llvm-9999.orig/tools/clang/Makefile llvm-9999/tools/clang/Makefile
--- llvm-9999.orig/tools/clang/Makefile	2016-01-19 21:33:04.221637342 -0800
+++ llvm-9999/tools/clang/Makefile	2016-01-19 21:34:39.344250458 -0800
@@ -51,6 +51,8 @@
   DIRS := $(filter-out docs, $(DIRS))
 endif
 
+CXX.Flags += -fPIC
+
 # Set common Clang build flags.
 CPP.Flags += -I$(PROJ_SRC_DIR)/$(CLANG_LEVEL)/include -I$(PROJ_OBJ_DIR)/$(CLANG_LEVEL)/include
 ifdef CLANG_VENDOR
