diff -ur llvm-3.7.1.src.orig/tools/clang/CMakeLists.txt llvm-3.7.1.src/tools/clang/CMakeLists.txt
--- llvm-3.7.1.src.orig/tools/clang/CMakeLists.txt	2016-01-18 22:47:04.204905318 -0800
+++ llvm-3.7.1.src/tools/clang/CMakeLists.txt	2016-01-18 22:49:53.429548773 -0800
@@ -252,7 +252,7 @@
 
 # Add appropriate flags for GCC
 if (LLVM_COMPILER_IS_GCC_COMPATIBLE)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-common -Woverloaded-virtual -fno-strict-aliasing")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-common -Woverloaded-virtual -fno-strict-aliasing -fPIC")
 
   # Enable -pedantic for Clang even if it's not enabled for LLVM.
   if (NOT LLVM_ENABLE_PEDANTIC)
diff -ur llvm-3.7.1.src.orig/tools/clang/Makefile llvm-3.7.1.src/tools/clang/Makefile
--- llvm-3.7.1.src.orig/tools/clang/Makefile	2016-01-18 22:47:04.204905318 -0800
+++ llvm-3.7.1.src/tools/clang/Makefile	2016-01-18 22:53:18.913599635 -0800
@@ -68,7 +68,7 @@
 #   http://gcc.gnu.org/PR41838
 #
 # We can revisit this when LLVM/Clang support it.
-CXX.Flags += -fno-strict-aliasing
+CXX.Flags += -fno-strict-aliasing -fPIC
 
 # Set up Clang's tblgen.
 ifndef CLANG_TBLGEN
