diff -urp ROCm-Device-Libs-rocm-5.1.3.orig/cmake/OCL.cmake ROCm-Device-Libs-rocm-5.1.3/cmake/OCL.cmake
--- ROCm-Device-Libs-rocm-5.1.3.orig/cmake/OCL.cmake	2022-02-16 13:19:54.000000000 -0800
+++ ROCm-Device-Libs-rocm-5.1.3/cmake/OCL.cmake	2023-08-21 12:09:42.017204405 -0700
@@ -45,7 +45,7 @@ set (BC_EXT .bc)
 set (LIB_SUFFIX ".lib${BC_EXT}")
 set (STRIP_SUFFIX ".strip${BC_EXT}")
 set (FINAL_SUFFIX "${BC_EXT}")
-set (INSTALL_ROOT_SUFFIX "amdgcn/bitcode")
+set (INSTALL_ROOT_SUFFIX "@LIBDIR@/amdgcn/bitcode")
 
 # Set `inc_options` to contain Clang command-line for include directories for
 # current source directory.
diff -urp ROCm-Device-Libs-rocm-5.1.3.orig/cmake/Packages.cmake ROCm-Device-Libs-rocm-5.1.3/cmake/Packages.cmake
--- ROCm-Device-Libs-rocm-5.1.3.orig/cmake/Packages.cmake	2022-02-16 13:19:54.000000000 -0800
+++ ROCm-Device-Libs-rocm-5.1.3/cmake/Packages.cmake	2023-08-21 12:10:06.401205010 -0700
@@ -1,4 +1,4 @@
-set(PACKAGE_PREFIX lib/cmake/AMDDeviceLibs)
+set(PACKAGE_PREFIX @LIBDIR@/cmake/AMDDeviceLibs)
 
 # Generate the build-tree package.
 # We know the absolute path to the build tree, so we leave
@@ -16,7 +16,7 @@ configure_file(AMDDeviceLibsConfig.cmake
   @ONLY)
 
 
-set(install_path_suffix "amdgcn/bitcode")
+set(install_path_suffix "@LIBDIR@/amdgcn/bitcode")
 
 # Generate the install-tree package.
 # We do not know the absolute path to the intall tree until we are installed,
