diff -urp ROCm-Device-Libs-rocm-5.3.3.orig/cmake/OCL.cmake ROCm-Device-Libs-rocm-5.3.3/cmake/OCL.cmake
--- ROCm-Device-Libs-rocm-5.3.3.orig/cmake/OCL.cmake	2022-08-19 07:36:28.000000000 -0700
+++ ROCm-Device-Libs-rocm-5.3.3/cmake/OCL.cmake	2023-08-21 12:13:43.057210385 -0700
@@ -45,7 +45,7 @@ set (BC_EXT .bc)
 set (LIB_SUFFIX ".lib${BC_EXT}")
 set (STRIP_SUFFIX ".strip${BC_EXT}")
 set (FINAL_SUFFIX "${BC_EXT}")
-set (INSTALL_ROOT_SUFFIX "amdgcn/bitcode")
+set (INSTALL_ROOT_SUFFIX "@LIBDIR@/amdgcn/bitcode")
 
 # Set `inc_options` to contain Clang command-line for include directories for
 # current source directory.
diff -urp ROCm-Device-Libs-rocm-5.3.3.orig/cmake/Packages.cmake ROCm-Device-Libs-rocm-5.3.3/cmake/Packages.cmake
--- ROCm-Device-Libs-rocm-5.3.3.orig/cmake/Packages.cmake	2022-08-19 07:36:28.000000000 -0700
+++ ROCm-Device-Libs-rocm-5.3.3/cmake/Packages.cmake	2023-08-21 12:14:42.465211859 -0700
@@ -16,7 +16,7 @@ configure_file(AMDDeviceLibsConfig.cmake
   @ONLY)
 
 
-set(install_path_suffix "amdgcn/bitcode")
+set(install_path_suffix "@LIBDIR@/amdgcn/bitcode")
 
 # Generate the install-tree package.
 # We do not know the absolute path to the intall tree until we are installed,
