--- a/makefile	2011-06-14 13:58:22.000000000 -0400
+++ b/makefile	2013-02-18 23:11:41.029584213 -0500
@@ -1,5 +1,5 @@
 
-OUTPUT_DIR = ./Release
+OUTPUT_DIR = ./build
 
 EXECUTABLE = AMDOverdriveCtrl
 
@@ -8,8 +8,8 @@
 OBJECTS = $(SOURCES:.cpp=.o)
 
 CC = g++
-CFLAGS = -c -I./ADL_SDK/include -O3 -fexpensive-optimizations -W -Wall $(shell wx-config --cxxflags --unicode=yes --debug=no) -DLINUX
-LDFLAGS = -mwindows -s $(shell wx-config --debug=no --libs --unicode=yes) -ldl -o$(OUTPUT_DIR)/$(EXECUTABLE)
+CFLAGS := -c -I/usr/include/ADL $(shell wx-config --cxxflags --unicode=yes --debug=no) -DLINUX $(CXXFLAGS)
+LDFLAGS := $(shell wx-config --debug=no --libs --unicode=yes) -ldl -o$(OUTPUT_DIR)/$(EXECUTABLE) $(LDFLAGS)
 
 .cpp.o:
 	$(CC) $(CFLAGS) $< -o $@
@@ -19,7 +19,7 @@
 $(OUTPUT_DIR)/$(EXECUTABLE): $(OBJECTS) 
 	@rm -rf $(OUTPUT_DIR)
 	@mkdir $(OUTPUT_DIR)
-	$(CC) $(LDFLAGS) $(OBJECTS) -o $@
+	$(CC) $(OBJECTS) $(LDFLAGS) -o $@
 	@echo ...
 	@echo build completed.
 	@echo ...
@@ -32,23 +32,23 @@
 	@echo ...
 
 install: $(OUTPUT_DIR)/$(EXECUTABLE)
-	@rm -f /usr/bin/$(EXECUTABLE)
-	@rm -rf /usr/share/$(EXECUTABLE)
-	@cp $(OUTPUT_DIR)/$(EXECUTABLE) /usr/bin/
-	@mkdir -p /usr/share/$(EXECUTABLE)/images
-	@mkdir -p /usr/share/doc/amdoverdrivectrl/
-	@mkdir -p /usr/share/applications/
-	@mkdir -p /usr/share/menu/
-	@cp -pr ./create_deb/*.png /usr/share/$(EXECUTABLE)/images/
-	@cp -pr ./create_deb/*.xpm /usr/share/$(EXECUTABLE)/images/
-	@cp -pr ./create_deb/AUTHORS /usr/share/$(EXECUTABLE)/
-	@cp -pr ./create_deb/LICENSE /usr/share/$(EXECUTABLE)/
-	@cp -pr ./create_deb/copyright /usr/share/doc/amdoverdrivectrl/copyright
-	@gzip -c -9 ./create_deb/changelog > /usr/share/doc/amdoverdrivectrl/changelog.gz
-	@gzip -c -9 ./create_deb/manpage > /usr/share/man/man1/amdoverdrivectrl.1.gz
-	@rm -rf /usr/share/applications/$(EXECUTABLE).desktop
-	@cp -rf create_deb/$(EXECUTABLE).desktop /usr/share/applications/$(EXECUTABLE).desktop
-	@chmod a+x /usr/share/applications/$(EXECUTABLE).desktop
+	@rm -f ${DESTDIR}/usr/bin/$(EXECUTABLE)
+	@rm -rf ${DESTDIR}/usr/share/$(EXECUTABLE)
+	@mkdir -p ${DESTDIR}/usr/bin
+	@cp $(OUTPUT_DIR)/$(EXECUTABLE) ${DESTDIR}/usr/bin/
+	@mkdir -p ${DESTDIR}/usr/share/$(EXECUTABLE)/images
+	@mkdir -p ${DESTDIR}/usr/share/doc/amdoverdrivectrl/
+	@mkdir -p ${DESTDIR}/usr/share/applications/
+	@mkdir -p ${DESTDIR}/usr/share/menu/
+	@cp -pr ./create_deb/*.png ${DESTDIR}/usr/share/$(EXECUTABLE)/images/
+	@cp -pr ./create_deb/*.xpm ${DESTDIR}/usr/share/$(EXECUTABLE)/images/
+	@mkdir -p ${DESTDIR}/usr/share/man/man1
+	@cp -pr ./create_deb/changelog ${DESTDIR}/usr/share/doc/amdoverdrivectrl/changelog
+	@cp -pr ./create_deb/AUTHORS ${DESTDIR}/usr/share/doc/
+	@cp -pr ./create_deb/manpage ${DESTDIR}/usr/share/man/man1/AMDOverdriveCtrl.1
+	@rm -rf ${DESTDIR}/usr/share/applications/$(EXECUTABLE).desktop
+	@cp -rf create_deb/$(EXECUTABLE).desktop ${DESTDIR}/usr/share/applications/$(EXECUTABLE).desktop
+	@chmod a+x ${DESTDIR}/usr/share/applications/$(EXECUTABLE).desktop
 	@echo ...
 	@echo $(EXECUTABLE) installed.
 	@echo ...

