diff -urp AMDOverdriveCtrl.orig/makefile AMDOverdriveCtrl/makefile
--- AMDOverdriveCtrl.orig/makefile	2013-12-13 15:12:10.000000000 -0800
+++ AMDOverdriveCtrl/makefile	2017-08-06 16:46:04.940060126 -0700
@@ -1,5 +1,5 @@
 
-OUTPUT_DIR = ./Release
+OUTPUT_DIR = ./build
 
 EXECUTABLE = AMDOverdriveCtrl
 
@@ -15,8 +15,8 @@ ifndef OPTFLAGS
 endif
 
 CC = g++
-CFLAGS = -c -I./ADL_SDK/include -fexpensive-optimizations -W $(shell wx-config --cxxflags --unicode=yes --debug=no) -DLINUX $(OPTFLAGS)
-LDFLAGS = -s $(shell wx-config --debug=no --libs --unicode=yes) -ldl
+CFLAGS := -c -I/usr/include/ADL $(shell wx-config --cxxflags --unicode=yes --debug=no) -DLINUX $(CXXFLAGS)
+LDFLAGS := $(shell wx-config --debug=no --libs --unicode=yes) -ldl $(LDFLAGS)
 
 .cpp.o:
 	$(CC) $(CFLAGS) $< -o $@
Only in AMDOverdriveCtrl: makefile.orig
Only in AMDOverdriveCtrl: makefile.rej
