diff -urp MonoGame-3.5.1.orig/Build/Projects/2MGFX.definition MonoGame-3.5.1/Build/Projects/2MGFX.definition
--- MonoGame-3.5.1.orig/Build/Projects/2MGFX.definition	2016-03-29 23:30:30.000000000 -0700
+++ MonoGame-3.5.1/Build/Projects/2MGFX.definition	2017-01-12 22:46:02.254183611 -0800
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<Project Name="2MGFX" Path="Tools/2MGFX" Type="Console" Platforms="Windows">
+<Project Name="2MGFX" Path="Tools/2MGFX" Type="Console" Platforms="Windows,Linux">
 
   <!-- Common assembly references -->
   <References>
@@ -167,18 +167,28 @@
     <!-- The core mgfx generation classes -->
     <Compile Include="ConstantBufferData.cs" />
     <Compile Include="ConstantBufferData.mojo.cs" />
-    <Compile Include="ConstantBufferData.sharpdx.cs" />
+    <Compile Include="ConstantBufferData.sharpdx.cs">
+      <Platforms>Windows</Platforms>
+    </Compile>
     <Compile Include="ConstantBufferData.writer.cs" />
     <Compile Include="EffectObject.cs" />
-    <Compile Include="EffectObject.hlsl.cs" />
-    <Compile Include="EffectObject.pssl.cs" />
+    <Compile Include="EffectObject.hlsl.cs">
+      <Platforms>Windows</Platforms>
+    </Compile>
+    <Compile Include="EffectObject.pssl.cs">
+      <Platforms>Windows</Platforms>
+    </Compile>
     <Compile Include="EffectObject.writer.cs" />    
     <Compile Include="IEffectCompilerOutput.cs" />
     <Compile Include="ShaderCompilerException.cs" />
     <Compile Include="ShaderData.cs" />
     <Compile Include="ShaderData.mojo.cs" />
-    <Compile Include="ShaderData.pssl.cs" />
-    <Compile Include="ShaderData.sharpdx.cs" />
+    <Compile Include="ShaderData.pssl.cs">
+      <Platforms>Windows</Platforms>
+    </Compile>
+    <Compile Include="ShaderData.sharpdx.cs">
+      <Platforms>Windows</Platforms>
+    </Compile>
     <Compile Include="ShaderData.writer.cs" />
     <Compile Include="PassInfo.cs" />
     <Compile Include="SamplerStateInfo.cs" />
diff -urp MonoGame-3.5.1.orig/Build/Projects/2MGFXReferences.definition MonoGame-3.5.1/Build/Projects/2MGFXReferences.definition
--- MonoGame-3.5.1.orig/Build/Projects/2MGFXReferences.definition	2016-03-29 23:30:30.000000000 -0700
+++ MonoGame-3.5.1/Build/Projects/2MGFXReferences.definition	2017-01-12 22:41:35.600933050 -0800
@@ -13,5 +13,10 @@
       Path="ThirdParty/Dependencies/CppNet/CppNet.dll" />
     <NativeBinary Path="ThirdParty\Dependencies\MojoShader\Windows\libmojoshader_64.dll" />
   </Platform>
+  <Platform Type="Linux">
+    <Binary
+      Name="CppNet"
+      Path="ThirdParty/Dependencies/CppNet/CppNet.dll" />
+  </Platform>
 
 </ExternalProject>
diff -urp MonoGame-3.5.1.orig/Build/Projects/MonoGame.Framework.Content.Pipeline.definition MonoGame-3.5.1/Build/Projects/MonoGame.Framework.Content.Pipeline.definition
--- MonoGame-3.5.1.orig/Build/Projects/MonoGame.Framework.Content.Pipeline.definition	2017-01-12 22:41:13.126064356 -0800
+++ MonoGame-3.5.1/Build/Projects/MonoGame.Framework.Content.Pipeline.definition	2017-01-12 22:46:00.878130880 -0800
@@ -145,15 +145,15 @@
     <Compile Include="Processors\VideoProcessor.cs"/>
     <!-- TwoMGFX-->
     <Compile Include="..\Tools\2MGFX\CommandLineParser.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\CommandLineParser.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ConstantBufferData.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ConstantBufferData.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ConstantBufferData.mojo.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ConstantBufferData.mojo.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ConstantBufferData.sharpdx.cs">
@@ -161,11 +161,11 @@
       <Link>Processors\MGFX\ConstantBufferData.sharpdx.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ConstantBufferData.writer.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ConstantBufferData.writer.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\EffectObject.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\EffectObject.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\EffectObject.hlsl.cs">
@@ -177,27 +177,27 @@
       <Link>Processors\MGFX\EffectObject.pssl.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\EffectObject.writer.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\EffectObject.writer.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\IEffectCompilerOutput.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\IEffectCompilerOutput.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\MarshalHelper.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\MarshalHelper.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ShaderCompilerException.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ShaderCompilerException.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ShaderData.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ShaderData.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ShaderData.mojo.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ShaderData.mojo.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ShaderData.pssl.cs">
@@ -209,59 +209,59 @@
       <Link>Processors\MGFX\ShaderData.sharpdx.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ShaderData.writer.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ShaderData.writer.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\MojoShader.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\MojoShader.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\Options.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\Options.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\Parser.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\Parser.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ParseTree.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ParseTree.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ParseTreeTools.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ParseTreeTools.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\PassInfo.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\PassInfo.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\Preprocessor.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\Preprocessor.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\SamplerStateInfo.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\SamplerStateInfo.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\Scanner.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\Scanner.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ShaderInfo.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ShaderInfo.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\ShaderProfile.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\ShaderProfile.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\TechniqueInfo.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\TechniqueInfo.cs</Link>
     </Compile>
     <Compile Include="..\Tools\2MGFX\TextureFilterType.cs">
-      <Platforms>Windows</Platforms>
+      <Platforms>Windows,Linux</Platforms>
       <Link>Processors\MGFX\TextureFilterType.cs</Link>
     </Compile>
     <!-- Microsoft.Xna.Framework.Content.Pipeline.Serialization.Compiler -->
diff -urp MonoGame-3.5.1.orig/MonoGame.Framework.Content.Pipeline/Processors/EffectProcessor.cs MonoGame-3.5.1/MonoGame.Framework.Content.Pipeline/Processors/EffectProcessor.cs
--- MonoGame-3.5.1.orig/MonoGame.Framework.Content.Pipeline/Processors/EffectProcessor.cs	2016-03-29 23:30:30.000000000 -0700
+++ MonoGame-3.5.1/MonoGame.Framework.Content.Pipeline/Processors/EffectProcessor.cs	2017-01-12 22:41:35.601933089 -0800
@@ -7,7 +7,7 @@ using System.IO;
 using System.Text.RegularExpressions;
 using Microsoft.Xna.Framework.Content.Pipeline.Graphics;
 using Microsoft.Xna.Framework.Graphics;
-#if WINDOWS
+#if WINDOWS || (LINUX && DESKTOPGL)
 using TwoMGFX;
 #endif
 
@@ -50,7 +50,7 @@ namespace Microsoft.Xna.Framework.Conten
         /// <remarks>If you get an error during processing, compilation stops immediately. The effect processor displays an error message. Once you fix the current error, it is possible you may get more errors on subsequent compilation attempts.</remarks>
         public override CompiledEffectContent Process(EffectContent input, ContentProcessorContext context)
         {
-#if WINDOWS
+#if WINDOWS || (LINUX && DESKTOPGL)
             var options = new Options();
             options.SourceFile = input.Identity.SourceFilename;
 
@@ -142,7 +142,7 @@ namespace Microsoft.Xna.Framework.Conten
 #endif
         }
 
-#if WINDOWS
+#if WINDOWS || (LINUX && DESKTOPGL)
         private class ContentPipelineEffectCompilerOutput : IEffectCompilerOutput
         {
             private readonly ContentProcessorContext _context;
