diff -urp rocm-cmake-rocm-5.5.1.orig/CHANGELOG.md rocm-cmake-rocm-5.5.1/CHANGELOG.md
--- rocm-cmake-rocm-5.5.1.orig/CHANGELOG.md	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/CHANGELOG.md	2023-08-27 11:52:10.154040213 -0700
@@ -2,7 +2,7 @@
 
 ## [rocm-cmake 0.8.1 for ROCm 5.5.0]
 ### Fixed
-- ROCMInstallTargets: Added compatibility symlinks for included cmake files in `<ROCM>/lib/cmake/<PACKAGE>`.
+- ROCMInstallTargets: Added compatibility symlinks for included cmake files in `<ROCM>/@LIBDIR@/cmake/<PACKAGE>`.
 ### Changed
 - ROCMHeaderWrapper: The wrapper header deprecation message is now a deprecation warning.
 
diff -urp rocm-cmake-rocm-5.5.1.orig/CMakeLists.txt rocm-cmake-rocm-5.5.1/CMakeLists.txt
--- rocm-cmake-rocm-5.5.1.orig/CMakeLists.txt	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/CMakeLists.txt	2023-08-27 11:50:59.550038461 -0700
@@ -4,7 +4,7 @@
 
 cmake_minimum_required (VERSION 3.6)
 
-set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH "")
+#_cmake_modify_IGNORE set(CMAKE_INSTALL_PREFIX "/usr" CACHE PATH "")
 
 project(rocm-cmake LANGUAGES NONE)
 
Only in rocm-cmake-rocm-5.5.1: CMakeLists.txt.orig
diff -urp rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMClangTidy.cmake rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMClangTidy.cmake
--- rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMClangTidy.cmake	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMClangTidy.cmake	2023-08-27 11:50:59.550038461 -0700
@@ -22,7 +22,7 @@ find_program(
           clang-tidy-3.5
     HINTS ${CLANG_TIDY_EXE_HINT}
     PATH_SUFFIXES compiler/bin bin
-    PATHS /opt/rocm/llvm/bin /opt/rocm/hcc /usr/local/opt/llvm/bin)
+    PATHS /usr/lib/llvm/@LLVM_SLOT@/bin /usr/hcc)
 
 execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version OUTPUT_VARIABLE CLANG_TIDY_COMPILER_VERSION_OUTPUT)
 function(rocm_find_clang_tidy_version VAR)
diff -urp rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMCppCheck.cmake rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMCppCheck.cmake
--- rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMCppCheck.cmake	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMCppCheck.cmake	2023-08-27 11:50:59.550038461 -0700
@@ -9,7 +9,7 @@ include(ROCMAnalyzers)
 find_program(
     CPPCHECK_EXE
     NAMES cppcheck
-    PATHS /opt/rocm/bin)
+    PATHS /usr/bin)
 
 function(rocm_find_cppcheck_version VAR)
     execute_process(COMMAND ${CPPCHECK_EXE} --version OUTPUT_VARIABLE VERSION_OUTPUT)
diff -urp rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMCreatePackage.cmake rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMCreatePackage.cmake
--- rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMCreatePackage.cmake	2023-08-27 11:50:45.302038108 -0700
+++ rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMCreatePackage.cmake	2023-08-27 11:50:59.550038461 -0700
@@ -138,7 +138,7 @@ endmacro()
 
 function(rocm_parse_python_syspath DIR_PATH PKG_NAME)
     set(PYTHON_SITE_PACKAGES
-        "/usr/lib/python3/dist-packages;/usr/lib/python2.7/dist-packages"
+        "@PYTHON_SITEDIRS@"
         CACHE STRING "The site packages used for packaging")
     #Group the statements to function
     file(APPEND ${PROJECT_BINARY_DIR}/debian/postinst
diff -urp rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMInstallTargets.cmake rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMInstallTargets.cmake
--- rocm-cmake-rocm-5.5.1.orig/share/rocm/cmake/ROCMInstallTargets.cmake	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/share/rocm/cmake/ROCMInstallTargets.cmake	2023-08-27 11:50:59.550038461 -0700
@@ -6,7 +6,7 @@ cmake_policy(SET CMP0057 NEW)
 
 # todo: consolidate with duplicate in ROCMCreatePackage.cmake
 # Default libdir to "lib", this skips GNUInstallDirs from trying to take a guess if it's unset:
-set(CMAKE_INSTALL_LIBDIR "lib" CACHE STRING "Library install directory")
+set(CMAKE_INSTALL_LIBDIR "@LIBDIR@" CACHE STRING "Library install directory")
 
 include(CMakeParseArguments)
 include(GNUInstallDirs)
diff -urp rocm-cmake-rocm-5.5.1.orig/test/libheaderonly/CMakeLists.txt rocm-cmake-rocm-5.5.1/test/libheaderonly/CMakeLists.txt
--- rocm-cmake-rocm-5.5.1.orig/test/libheaderonly/CMakeLists.txt	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/test/libheaderonly/CMakeLists.txt	2023-08-27 11:50:59.550038461 -0700
@@ -33,8 +33,8 @@ rocm_create_package(
     HEADER_ONLY)
 endif()
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
 
 add_library(headeronly INTERFACE)
diff -urp rocm-cmake-rocm-5.5.1.orig/test/libsimple/CMakeLists.txt rocm-cmake-rocm-5.5.1/test/libsimple/CMakeLists.txt
--- rocm-cmake-rocm-5.5.1.orig/test/libsimple/CMakeLists.txt	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/test/libsimple/CMakeLists.txt	2023-08-27 11:50:59.550038461 -0700
@@ -31,8 +31,8 @@ rocm_create_package(
     LDCONFIG)
 endif()
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
 
 add_library(simple simple.cpp)
diff -urp rocm-cmake-rocm-5.5.1.orig/test/libwrapper/CMakeLists.txt rocm-cmake-rocm-5.5.1/test/libwrapper/CMakeLists.txt
--- rocm-cmake-rocm-5.5.1.orig/test/libwrapper/CMakeLists.txt	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/test/libwrapper/CMakeLists.txt	2023-08-27 11:50:59.550038461 -0700
@@ -32,8 +32,8 @@ rocm_create_package(
     LDCONFIG)
 endif()
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
 
 option(ERR_ON_WARN "Treat all warnings as errors." OFF)
diff -urp rocm-cmake-rocm-5.5.1.orig/test/pass/simple-soversion.cmake rocm-cmake-rocm-5.5.1/test/pass/simple-soversion.cmake
--- rocm-cmake-rocm-5.5.1.orig/test/pass/simple-soversion.cmake	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/test/pass/simple-soversion.cmake	2023-08-27 11:50:59.550038461 -0700
@@ -10,12 +10,12 @@ endfunction()
 
 install_dir(${TEST_DIR}/libsimple CMAKE_ARGS -DBUILD_SHARED_LIBS=On -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=On)
 if(NOT WIN32 AND NOT APPLE)
-    test_expect_file(${PREFIX}/lib/libsimple.so)
-    test_expect_file(${PREFIX}/lib/libsimple.so.1)
-    test_expect_file(${PREFIX}/lib/libsimple.so.1.1.2)
+    test_expect_file(${PREFIX}/@LIBDIR@/libsimple.so)
+    test_expect_file(${PREFIX}/@LIBDIR@/libsimple.so.1)
+    test_expect_file(${PREFIX}/@LIBDIR@/libsimple.so.1.1.2)
 
-    test_expect_realpath(${PREFIX}/lib/libsimple.so ${PREFIX}/lib/libsimple.so.1.1.2)
-    test_expect_realpath(${PREFIX}/lib/libsimple.so.1 ${PREFIX}/lib/libsimple.so.1.1.2)
+    test_expect_realpath(${PREFIX}/@LIBDIR@/libsimple.so ${PREFIX}/@LIBDIR@/libsimple.so.1.1.2)
+    test_expect_realpath(${PREFIX}/@LIBDIR@/libsimple.so.1 ${PREFIX}/@LIBDIR@/libsimple.so.1.1.2)
 
     test_exec(COMMAND ldd ${PREFIX}/bin/simple-main OUTPUT_VARIABLE LIBS)
     test_expect_matches("${LIBS}" "libsimple.so.1 =>")
diff -urp rocm-cmake-rocm-5.5.1.orig/test/pass/simple-soversion-env.cmake rocm-cmake-rocm-5.5.1/test/pass/simple-soversion-env.cmake
--- rocm-cmake-rocm-5.5.1.orig/test/pass/simple-soversion-env.cmake	2023-01-18 13:22:54.000000000 -0800
+++ rocm-cmake-rocm-5.5.1/test/pass/simple-soversion-env.cmake	2023-08-27 11:50:59.550038461 -0700
@@ -12,12 +12,12 @@ set(ENV{ROCM_LIBPATCH_VERSION} "21001")
 install_dir(${TEST_DIR}/libsimple CMAKE_ARGS -DBUILD_SHARED_LIBS=On -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=On)
 
 if(NOT WIN32 AND NOT APPLE)
-    test_expect_file(${PREFIX}/lib/libsimple.so)
-    test_expect_file(${PREFIX}/lib/libsimple.so.1)
-    test_expect_file(${PREFIX}/lib/libsimple.so.1.1.2.$ENV{ROCM_LIBPATCH_VERSION})
+    test_expect_file(${PREFIX}/@LIBDIR@/libsimple.so)
+    test_expect_file(${PREFIX}/@LIBDIR@/libsimple.so.1)
+    test_expect_file(${PREFIX}/@LIBDIR@/libsimple.so.1.1.2.$ENV{ROCM_LIBPATCH_VERSION})
 
-    test_expect_realpath(${PREFIX}/lib/libsimple.so ${PREFIX}/lib/libsimple.so.1.1.2.$ENV{ROCM_LIBPATCH_VERSION})
-    test_expect_realpath(${PREFIX}/lib/libsimple.so.1 ${PREFIX}/lib/libsimple.so.1.1.2.$ENV{ROCM_LIBPATCH_VERSION})
+    test_expect_realpath(${PREFIX}/@LIBDIR@/libsimple.so ${PREFIX}/@LIBDIR@/libsimple.so.1.1.2.$ENV{ROCM_LIBPATCH_VERSION})
+    test_expect_realpath(${PREFIX}/@LIBDIR@/libsimple.so.1 ${PREFIX}/@LIBDIR@/libsimple.so.1.1.2.$ENV{ROCM_LIBPATCH_VERSION})
 
     test_exec(COMMAND ldd ${PREFIX}/bin/simple-main OUTPUT_VARIABLE LIBS)
     test_expect_matches("${LIBS}" "libsimple.so.1 =>")
