diff -urp urho3d-9999.orig/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp urho3d-9999/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp
--- urho3d-9999.orig/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp	2016-11-19 23:43:29.912676727 -0800
+++ urho3d-9999/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp	2016-11-19 23:45:12.839642571 -0800
@@ -442,7 +442,11 @@ bool Graphics::SetMode(int width, int he
             {
 #ifndef __EMSCRIPTEN__
                 if (!window_)
+#ifdef ENABLE_DOLLAR
                     window_ = SDL_CreateWindowFrom(externalWindow_, SDL_WINDOW_OPENGL);
+#else
+                    window_ = SDL_CreateWindowFrom(externalWindow_);
+#endif
                 fullscreen = false;
 #endif
             }
Only in urho3d-9999/Source/Urho3D/Graphics/OpenGL: OGLGraphics.cpp.orig
Only in urho3d-9999/Source/Urho3D/Graphics/OpenGL: OGLGraphics.cpp.rej
diff -urp urho3d-9999.orig/Source/Urho3D/Input/Input.cpp urho3d-9999/Source/Urho3D/Input/Input.cpp
--- urho3d-9999.orig/Source/Urho3D/Input/Input.cpp	2016-11-19 23:43:35.633897163 -0800
+++ urho3d-9999/Source/Urho3D/Input/Input.cpp	2016-11-19 23:44:17.306502816 -0800
@@ -1190,7 +1190,7 @@ bool Input::SaveGestures(Serializer& des
 #ifdef ENABLE_DOLLAR
     return SDL_SaveAllDollarTemplates(wrapper.GetRWOps()) != 0;
 #else
-    return
+    return false;
 #endif
 }
 
Only in urho3d-9999/Source/Urho3D/Input: Input.cpp.orig
