diff -urp Urho3D-1.6.orig/Source/Urho3D/CMakeLists.txt Urho3D-1.6/Source/Urho3D/CMakeLists.txt
--- Urho3D-1.6.orig/Source/Urho3D/CMakeLists.txt	2016-11-30 22:24:17.686304786 -0800
+++ Urho3D-1.6/Source/Urho3D/CMakeLists.txt	2016-11-30 22:45:11.637371380 -0800
@@ -294,6 +294,7 @@ if (NOT ANDROID AND NOT WEB)
 elseif (EMSCRIPTEN)
     # Emscripten does not support generation of shared library in a conventional sense, it just produces "linked" LLVM bitcode
     set (CMAKE_SHARED_LIBRARY_SUFFIX ".bc")
+    set_target_properties(${TARGET_NAME} PROPERTIES OUTPUT_NAME "Urho3D_web")
 endif ()
 if (IOS)
     # Add a custom target to build Mach-O universal binary consisting of iphoneos (universal ARM archs including 'arm64' if 64-bit is enabled) and iphonesimulator (i386 arch and also x86_64 arch if 64-bit is enabled)
@@ -511,6 +512,9 @@ endif ()
 if (ANDROID)
     set (RENAME RENAME Urho3D-${ANDROID_NDK_ABI_NAME}.pc)
 endif ()
+if (WEB)
+    set (RENAME RENAME Urho3D_web.pc)
+endif ()
 install (FILES ${CMAKE_CURRENT_BINARY_DIR}/Urho3D.pc DESTINATION ${DEST_PKGCONFIG_DIR} ${RENAME})
 
 # Define post build steps
