diff -urp urho3d-9999.orig/Source/Urho3D/Input/Input.cpp urho3d-9999/Source/Urho3D/Input/Input.cpp
--- urho3d-9999.orig/Source/Urho3D/Input/Input.cpp	2016-11-19 23:38:55.277095904 -0800
+++ urho3d-9999/Source/Urho3D/Input/Input.cpp	2016-11-19 23:41:09.756276755 -0800
@@ -1187,13 +1187,21 @@ bool Input::RecordGesture()
 bool Input::SaveGestures(Serializer& dest)
 {
     RWOpsWrapper<Serializer> wrapper(dest);
+#ifdef ENABLE_DOLLAR
     return SDL_SaveAllDollarTemplates(wrapper.GetRWOps()) != 0;
+#else
+    return
+#endif
 }
 
 bool Input::SaveGesture(Serializer& dest, unsigned gestureID)
 {
     RWOpsWrapper<Serializer> wrapper(dest);
+#ifdef ENABLE_DOLLAR
     return SDL_SaveDollarTemplate(gestureID, wrapper.GetRWOps()) != 0;
+#else
+    return false;
+#endif
 }
 
 unsigned Input::LoadGestures(Deserializer& source)
@@ -1206,7 +1214,11 @@ unsigned Input::LoadGestures(Deserialize
     }
 
     RWOpsWrapper<Deserializer> wrapper(source);
+#ifdef ENABLE_DOLLAR
     return (unsigned)SDL_LoadDollarTemplates(-1, wrapper.GetRWOps());
+#else
+    return 0;
+#endif
 }
 
 
@@ -1215,15 +1227,21 @@ bool Input::RemoveGesture(unsigned gestu
 #ifdef __EMSCRIPTEN__
     return false;
 #else
+#ifdef ENABLE_DOLLAR
     return SDL_RemoveDollarTemplate(gestureID) != 0;
+#else
+    return false;
+#endif
 #endif
 }
 
 void Input::RemoveAllGestures()
 {
 #ifndef __EMSCRIPTEN__
+#ifdef ENABLE_DOLLAR
     SDL_RemoveAllDollarTemplates();
 #endif
+#endif
 }
 
 SDL_JoystickID Input::OpenJoystick(unsigned index)
Only in urho3d-9999/Source/Urho3D/Input: Input.cpp.orig
Only in urho3d-9999/Source/Urho3D/Input: Input.cpp.rej
