diff -ur Urho3D-1.5.orig/Source/Urho3D/Resource/JSONFile.cpp Urho3D-1.5/Source/Urho3D/Resource/JSONFile.cpp
--- Urho3D-1.5.orig/Source/Urho3D/Resource/JSONFile.cpp	2016-02-04 17:38:52.695683554 -0800
+++ Urho3D-1.5/Source/Urho3D/Resource/JSONFile.cpp	2016-02-04 18:00:26.947153563 -0800
@@ -224,7 +224,7 @@
     ToRapidjsonValue(document, root_, document.GetAllocator());
 
     StringBuffer buffer;
-    PrettyWriter<StringBuffer> writer(buffer, &(document.GetAllocator()));
+    PrettyWriter<StringBuffer> writer(buffer);
     writer.SetIndent(!indendation.Empty() ? indendation.Front() : '\0', indendation.Length());
 
     document.Accept(writer);
