--- a/WORKSPACE.orig	2023-04-02 22:05:17.629245194 -0700
+++ b/WORKSPACE	2023-04-02 22:06:10.141245872 -0700
@@ -12,6 +12,30 @@ http_archive(
     ],
 )
 
+# oteodoro:  Add rules_go for offline install
+http_archive(
+    name = "io_bazel_rules_go",
+    sha256 = "69de5c704a05ff37862f7e0f5534d4f479418afc21806c887db544a316f3cb6b",
+    urls = [
+        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.27.0/rules_go-v0.27.0.tar.gz",
+        "https://github.com/bazelbuild/rules_go/releases/download/v0.27.0/rules_go-v0.27.0.tar.gz",
+    ],
+)
+
+# oteodoro:  Use static version
+load("@io_bazel_rules_go//go:deps.bzl",
+    "go_download_sdk",
+    "go_rules_dependencies",
+    "go_register_toolchains",
+)
+go_rules_dependencies()
+go_download_sdk(
+    name = "go_sdk",
+    version = "1.14.15",
+    sdks = { "linux_amd64": ("go1.14.15.linux-amd64.tar.gz", "c64a57b374a81f7cf1408d2c410a28c6f142414f1ffa9d1062de1d653b0ae0d6"), }
+)
+#go_register_toolchains()
+
 load("@bazel_skylib//lib:versions.bzl", "versions")
 
 versions.check(
