--- a/Documentation/manpages/tool/man-pandoc-filter.py.orig	2020-06-12 11:10:54.603462678 -0700
+++ b/Documentation/manpages/tool/man-pandoc-filter.py	2020-06-17 02:01:26.458443088 -0700
@@ -39,7 +39,7 @@ def remove_references(key, value, format
         pass
         if value[1][0]['t'] == 'Code':
             return value[1][0]
-        return Str(' '.join([e['c'] for e in value[1] if 'c' in e.keys()]))
+        return Str(' '.join([str(e['c']) for e in value[1] if 'c' in e.keys()]))
     return None
 
 def main():
