diff -urp cli-2.1.807.orig/build/AppHostTemplate.proj cli-2.1.807/build/AppHostTemplate.proj
--- cli-2.1.807.orig/build/AppHostTemplate.proj	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/AppHostTemplate.proj	2020-06-19 03:34:43.809439891 -0700
@@ -36,6 +36,7 @@
   <Target Name="EnsureAppHostPackageRestored">
     <PropertyGroup>
       <AppHostTemplateRestoreAdditionalParameters>--runtime $(Rid)</AppHostTemplateRestoreAdditionalParameters>
+      <AppHostTemplateRestoreAdditionalParameters>$(AppHostTemplateRestoreAdditionalParameters) --disable-parallel</AppHostTemplateRestoreAdditionalParameters>
       <AppHostTemplateRestoreAdditionalParameters>$(AppHostTemplateRestoreAdditionalParameters) /p:TargetFramework=$(CliTargetFramework)</AppHostTemplateRestoreAdditionalParameters>
       <AppHostTemplateRestoreAdditionalParameters>$(AppHostTemplateRestoreAdditionalParameters) /p:TemplateFillInPackageName=$(TemplateFillInPackageName)</AppHostTemplateRestoreAdditionalParameters>
       <AppHostTemplateRestoreAdditionalParameters>$(AppHostTemplateRestoreAdditionalParameters) /p:TemplateFillInPackageVersion=$(TemplateFillInPackageVersion)</AppHostTemplateRestoreAdditionalParameters>
diff -urp cli-2.1.807.orig/build/BundledDotnetTools.proj cli-2.1.807/build/BundledDotnetTools.proj
--- cli-2.1.807.orig/build/BundledDotnetTools.proj	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/BundledDotnetTools.proj	2020-06-19 03:32:22.943793257 -0700
@@ -14,6 +14,7 @@
 
     <PropertyGroup>
       <DotnetToolsRestoreAdditionalParameters>--runtime any /p:RestoreProjectStyle=DotnetToolReference</DotnetToolsRestoreAdditionalParameters>
+      <DotnetToolsRestoreAdditionalParameters>$(DotnetToolsRestoreAdditionalParameters) --disable-parallel</DotnetToolsRestoreAdditionalParameters>
       <DotnetToolsRestoreAdditionalParameters>$(DotnetToolsRestoreAdditionalParameters) /p:TargetFramework=$(CliTargetFramework)</DotnetToolsRestoreAdditionalParameters>
       <DotnetToolsRestoreAdditionalParameters>$(DotnetToolsRestoreAdditionalParameters) /p:TemplateFillInPackageName=$(TemplateFillInPackageName)</DotnetToolsRestoreAdditionalParameters>
       <DotnetToolsRestoreAdditionalParameters>$(DotnetToolsRestoreAdditionalParameters) /p:TemplateFillInPackageVersion=$(TemplateFillInPackageVersion)</DotnetToolsRestoreAdditionalParameters>
diff -urp cli-2.1.807.orig/build/BundledTemplates.proj cli-2.1.807/build/BundledTemplates.proj
--- cli-2.1.807.orig/build/BundledTemplates.proj	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/BundledTemplates.proj	2020-06-19 03:33:49.567265830 -0700
@@ -25,7 +25,7 @@
           Condition="!Exists('$(TemplateNuPkgPath)/$(TemplateFillInPackageName.ToLower()).nuspec')">
     <DotNetRestore ToolPath="$(PreviousStageDirectory)"
                    ProjectPath="$(MSBuildThisFileDirectory)/templates/templates.csproj"
-                   AdditionalParameters="/p:TargetFramework=netcoreapp1.0 /p:TemplateFillInPackageName=$(TemplateFillInPackageName) /p:TemplateFillInPackageVersion=$(TemplateFillInPackageVersion)" />
+                   AdditionalParameters="--disable-parallel /p:TargetFramework=netcoreapp1.0 /p:TemplateFillInPackageName=$(TemplateFillInPackageName) /p:TemplateFillInPackageVersion=$(TemplateFillInPackageVersion)" />
   </Target>
 
   <Target Name="PrepareBundledTemplateProps">
diff -urp cli-2.1.807.orig/build/package/Installer.DEB.proj cli-2.1.807/build/package/Installer.DEB.proj
--- cli-2.1.807.orig/build/package/Installer.DEB.proj	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/package/Installer.DEB.proj	2020-06-19 03:34:48.259618243 -0700
@@ -121,7 +121,8 @@
 
       <!-- Run E2E -->
       <DotNetRestore ProjectPath="$(EndToEndTestProject)"
-                     ToolPath="$(DebianInstalledDirectory)" />
+                     ToolPath="$(DebianInstalledDirectory)"
+                     AdditionalParameters="--disable-parallel" />
 
       <DotNetTest ProjectPath="$(EndToEndTestProject)"
                   EnvironmentVariables="@(TestSdkDebTaskEnvironmentVariables)"
@@ -170,7 +171,8 @@
   <Target Name="PrepareDotnetDebTool">
 
     <DotNetRestore ToolPath="$(PreviousStageDirectory)"
-                   WorkingDirectory="$(DotnetDebToolDir)" />
+                   WorkingDirectory="$(DotnetDebToolDir)"
+                   AdditionalParameters="--disable-parallel" />
   </Target>
 
   <Target Name="TestDebuild">
diff -urp cli-2.1.807.orig/build/package/Installer.RPM.targets cli-2.1.807/build/package/Installer.RPM.targets
--- cli-2.1.807.orig/build/package/Installer.RPM.targets	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/package/Installer.RPM.targets	2020-06-19 03:34:47.269578566 -0700
@@ -230,7 +230,8 @@
 
     <!-- Run Tests -->
     <DotNetRestore ProjectPath="$(EndToEndTestProject)"
-                   ToolPath="$(RpmInstalledDirectory)" />
+                   ToolPath="$(RpmInstalledDirectory)"
+                   AdditionalParameters="--disable-parallel" />
 
     <DotNetTest ProjectPath="$(EndToEndTestProject)"
                 EnvironmentVariables="@(TestSdkRpmTaskEnvironmentVariables)"
diff -urp cli-2.1.807.orig/build/package/Symbols.targets cli-2.1.807/build/package/Symbols.targets
--- cli-2.1.807.orig/build/package/Symbols.targets	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/package/Symbols.targets	2020-06-19 03:34:49.429665136 -0700
@@ -23,7 +23,8 @@
   <Target Name="RestoreSymbolsProject"
           DependsOnTargets="Init; SetupProjectsToPack">
     <DotNetRestore ProjectPath="&quot;$(SymbolsProjectFileName)&quot;"
-                   ToolPath="$(PreviousStageDirectory)" />
+                   ToolPath="$(PreviousStageDirectory)"
+                   AdditionalParameters="--disable-parallel" />
   </Target>
 
 </Project>
diff -urp cli-2.1.807.orig/build/Prepare.targets cli-2.1.807/build/Prepare.targets
--- cli-2.1.807.orig/build/Prepare.targets	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/Prepare.targets	2020-06-19 03:34:23.088609423 -0700
@@ -63,7 +63,8 @@
     <CallTarget Targets="CleanSrcLockFiles" />
 
     <DotNetRestore ToolPath="$(PreviousStageDirectory)"
-                   ProjectPath="&quot;%(RestoreSrcPackagesInput.FullPath)&quot;" />
+                   ProjectPath="&quot;%(RestoreSrcPackagesInput.FullPath)&quot;"
+                   AdditionalParameters="--disable-parallel" />
 
   </Target>
 
@@ -89,7 +90,7 @@
 
     <DotNetRestore ToolPath="$(PreviousStageDirectory)"
                    ProjectPath="&quot;%(RestoreToolsPackagesInput.FullPath)&quot;"
-                   AdditionalParameters="/p:UsePortableLinuxSharedFramework=$(UsePortableLinuxSharedFramework)" />
+                   AdditionalParameters="--disable-parallel /p:UsePortableLinuxSharedFramework=$(UsePortableLinuxSharedFramework)" />
 
   </Target>
 
diff -urp cli-2.1.807.orig/build/RestoreDependency.proj cli-2.1.807/build/RestoreDependency.proj
--- cli-2.1.807.orig/build/RestoreDependency.proj	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/RestoreDependency.proj	2020-06-19 03:34:35.399102821 -0700
@@ -35,7 +35,7 @@
           Condition="!Exists('$(DependencyNuPkgPath)/$(DependencyPackageName.ToLower()).nuspec')">
     <DotNetRestore ToolPath="$(PreviousStageDirectory)"
                    ProjectPath="$(MSBuildThisFileDirectory)/sdks/sdks.csproj"
-                   AdditionalParameters="/p:DependencyPackageName=$(DependencyPackageName) /p:DependencyPackageVersion=$(DependencyPackageVersion)" />
+                   AdditionalParameters="--disable-parallel /p:DependencyPackageName=$(DependencyPackageName) /p:DependencyPackageVersion=$(DependencyPackageVersion)" />
   </Target>
 
   <Target Name="PrepareBundledDependencyProps">
diff -urp cli-2.1.807.orig/build/Test.targets cli-2.1.807/build/Test.targets
--- cli-2.1.807.orig/build/Test.targets	2020-05-28 09:13:45.000000000 -0700
+++ cli-2.1.807/build/Test.targets	2020-06-19 03:34:46.229536884 -0700
@@ -58,7 +58,7 @@
                             CreateTestAssetPackageNuPkgs;">
     <DotNetRestore ToolPath="$(OutputDirectory)"
                    ProjectPath="&quot;$(TestDirectory)/Microsoft.DotNet.Cli.Tests.sln&quot;"
-                   AdditionalParameters="/p:PreviousStageProps=$(NextStagePropsPath)"
+                   AdditionalParameters="--disable-parallel /p:PreviousStageProps=$(NextStagePropsPath)"
                    MaxCpuCount="1" />
   </Target>
 
@@ -77,7 +77,7 @@
     <DotNetRestore ToolPath="$(OutputDirectory)"
                    WorkingDirectory="$([System.IO.Directory]::GetParent('%(TestPackageProject.ProjectPath)'))"
                    ProjectPath="%(TestPackageProject.ProjectPath)"
-                   AdditionalParameters="/p:RestoreAdditionalProjectSources=$(TestOutputDir)/packages /p:PreviousStageProps=$(NextStagePropsPath)"
+                   AdditionalParameters="--disable-parallel /p:RestoreAdditionalProjectSources=$(TestOutputDir)/packages /p:PreviousStageProps=$(NextStagePropsPath)"
                    />
  
     <!-- https://github.com/NuGet/Home/issues/4063 -->
