--- a/eng/common/tools.sh.orig	2020-06-20 19:32:53.866098414 -0700
+++ b/eng/common/tools.sh	2020-06-20 19:37:08.538273573 -0700
@@ -327,7 +327,7 @@ function InitializeToolset {
   fi
   
   echo '<Project Sdk="Microsoft.DotNet.Arcade.Sdk"/>' > "$proj"
-  MSBuild-Core "$proj" $bl /t:__WriteToolsetLocation /clp:ErrorsOnly\;NoSummary /p:__ToolsetLocationOutputFile="$toolset_location_file"
+  MSBuild-Core "$proj" $bl /p:RestoreDisableParallel=true /t:__WriteToolsetLocation /clp:ErrorsOnly\;NoSummary /p:__ToolsetLocationOutputFile="$toolset_location_file"
 
   local toolset_build_proj=`cat "$toolset_location_file"`
 
--- a/eng/common/build.sh.orig	2020-06-20 19:32:53.856097935 -0700
+++ b/eng/common/build.sh	2020-06-20 19:34:58.852318657 -0700
@@ -169,6 +169,8 @@ function Build {
   InitializeToolset
   InitializeCustomToolset
 
+  properties="$properties /p:RestoreDisableParallel=true"
+
   if [[ ! -z "$projects" ]]; then
     properties="$properties /p:Projects=$projects"
   fi
