diff -urp civetweb-1.13.orig/CMakeLists.txt civetweb-1.13/CMakeLists.txt
--- civetweb-1.13.orig/CMakeLists.txt	2021-02-20 20:59:51.873084829 -0800
+++ civetweb-1.13/CMakeLists.txt	2021-02-20 21:00:16.130041632 -0800
@@ -145,8 +145,6 @@ endif()
 if (CIVETWEB_ENABLE_LUA)
   if (NOT CIVETWEB_ENABLE_LUA_SHARED)
     # Lua Version
-    set(CIVETWEB_LUA_VERSION 5.2.4 CACHE STRING
-      "The version of Lua to build and include statically")
     set_property(CACHE CIVETWEB_LUA_VERSION PROPERTY VALUE ${CIVETWEB_LUA_VERSION})
     message(STATUS "Lua Version - ${CIVETWEB_LUA_VERSION}")
     mark_as_advanced(CIVETWEB_LUA_VERSION)
diff -urp civetweb-1.13.orig/src/CMakeLists.txt civetweb-1.13/src/CMakeLists.txt
--- civetweb-1.13.orig/src/CMakeLists.txt	2020-09-30 11:24:15.000000000 -0700
+++ civetweb-1.13/src/CMakeLists.txt	2021-02-20 21:02:28.443258698 -0800
@@ -15,7 +15,9 @@ if (BUILD_SHARED_LIBS)
 endif()
 target_include_directories(
   civetweb-c-library PUBLIC
-  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
+  /usr/include/lua${CIVETWEB_LUA_VERSION_MAJOR_MINOR}
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/third_party>)
 install(
   TARGETS civetweb-c-library
   EXPORT ${PROJECT_NAME}-targets
@@ -64,6 +66,22 @@ endif()
 if (CIVETWEB_ENABLE_LUA)
   include(ExternalProject)
 
+  if (CIVETWEB_LUA_VERSION VERSION_GREATER_EQUAL 5.4 AND CIVETWEB_LUA_VERSION VERSION_LESS 5.5)
+    set(LUA_VERSION_NUM 504)
+    set(CIVETWEB_LUA_VERSION_MAJOR_MINOR 5.4)
+  elseif (CIVETWEB_LUA_VERSION VERSION_GREATER_EQUAL 5.3 AND CIVETWEB_LUA_VERSION VERSION_LESS 5.4)
+    set(LUA_VERSION_NUM 503)
+    set(CIVETWEB_LUA_VERSION_MAJOR_MINOR 5.3)
+  elseif (CIVETWEB_LUA_VERSION VERSION_GREATER_EQUAL 5.2 AND CIVETWEB_LUA_VERSION VERSION_LESS 5.3)
+    set(LUA_VERSION_NUM 502)
+    set(CIVETWEB_LUA_VERSION_MAJOR_MINOR 5.2)
+  elseif (CIVETWEB_LUA_VERSION VERSION_GREATER_EQUAL 5.1 AND CIVETWEB_LUA_VERSION VERSION_LESS 5.2)
+    set(LUA_VERSION_NUM 501)
+    set(CIVETWEB_LUA_VERSION_MAJOR_MINOR 5.1)
+  else()
+    message(ERROR "CIVETWEB_LUA_VERSION should be defined")
+  endif()
+  add_definitions(-DLUA_VERSION_NUM=${LUA_VERSION_NUM})
   # Determine if we should print to the output
   if (CIVETWEB_ENABLE_THIRD_PARTY_OUTPUT)
     set(THIRD_PARTY_LOGGING 0)
@@ -112,75 +130,88 @@ if (CIVETWEB_ENABLE_LUA)
     set(LUA_LIBRARIES "${LUA_INSTALL_DIR}/lib/liblua.a")
     add_dependencies(civetweb-c-library lua)
   else()
-    find_package(Lua)
+    find_package(Lua ${CIVETWEB_LUA_VERSION_MAJOR_MINOR} EXACT)
   endif()
 
   # Lua Filesystem Support
-  string(REPLACE "." "_" LUA_FILESYSTEM_VERSION_UNDERSCORE ${CIVETWEB_LUA_FILESYSTEM_VERSION})
-  ExternalProject_Add(luafilesystem
-    URL "https://github.com/keplerproject/luafilesystem/archive/v_${LUA_FILESYSTEM_VERSION_UNDERSCORE}.tar.gz"
-    URL_MD5 ${CIVETWEB_LUA_FILESYSTEM_MD5_HASH}
-    PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
-    PATCH_COMMAND ${CMAKE_COMMAND} -E copy
-      "${CMAKE_CURRENT_SOURCE_DIR}/cmake/luafilesystem/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
-    CMAKE_ARGS
-      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
-      "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
-    LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
-    LOG_UPDATE ${THIRD_PARTY_LOGGING}
-    LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
-    LOG_BUILD ${THIRD_PARTY_LOGGING}
-    LOG_TEST ${THIRD_PARTY_LOGGING}
-    LOG_INSTALL ${THIRD_PARTY_LOGGING})
-  ExternalProject_Get_Property(luafilesystem INSTALL_DIR)
-  set(LUA_FILESYSTEM_INSTALL_DIR ${INSTALL_DIR})
-  unset(INSTALL_DIR)
-  link_directories("${LUA_FILESYSTEM_INSTALL_DIR}/lib")
-  include_directories("${LUA_FILESYSTEM_INSTALL_DIR}/include")
-  set(LUA_LIBRARIES "${LUA_LIBRARIES};${LUA_FILESYSTEM_INSTALL_DIR}/lib/libluafilesystem.a")
-  add_dependencies(civetweb-c-library luafilesystem)
+  if (NOT CIVETWEB_ENABLE_LUA_FILESYSTEM_SHARED)
+    string(REPLACE "." "_" LUA_FILESYSTEM_VERSION_UNDERSCORE ${CIVETWEB_LUA_FILESYSTEM_VERSION})
+    ExternalProject_Add(luafilesystem
+      URL "https://github.com/keplerproject/luafilesystem/archive/v_${LUA_FILESYSTEM_VERSION_UNDERSCORE}.tar.gz"
+      URL_MD5 ${CIVETWEB_LUA_FILESYSTEM_MD5_HASH}
+      PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
+      PATCH_COMMAND ${CMAKE_COMMAND} -E copy
+        "${CMAKE_CURRENT_SOURCE_DIR}/cmake/luafilesystem/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
+      CMAKE_ARGS
+        "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+        "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
+      LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
+      LOG_UPDATE ${THIRD_PARTY_LOGGING}
+      LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
+      LOG_BUILD ${THIRD_PARTY_LOGGING}
+      LOG_TEST ${THIRD_PARTY_LOGGING}
+      LOG_INSTALL ${THIRD_PARTY_LOGGING})
+    ExternalProject_Get_Property(luafilesystem INSTALL_DIR)
+    set(LUA_FILESYSTEM_INSTALL_DIR ${INSTALL_DIR})
+    unset(INSTALL_DIR)
+    link_directories("${LUA_FILESYSTEM_INSTALL_DIR}/lib")
+    include_directories("${LUA_FILESYSTEM_INSTALL_DIR}/include")
+    add_library(lfs_lib STATIC IMPORTED)
+    set_property(TARGET lfs_lib PROPERTY IMPORTED_LOCATION "${LUA_FILESYSTEM_INSTALL_DIR}/lib/libluafilesystem.a")
+    add_dependencies(civetweb-c-library luafilesystem)
+  else()
+    add_library(lfs_lib SHARED IMPORTED)
+    set_property(TARGET lfs_lib PROPERTY IMPORTED_LOCATION "${LUA_CDIR}/lfs.so")
+  endif()
 
   # Lua SQLite Support
-  if (${CIVETWEB_LUA_SQLITE_VERSION} VERSION_EQUAL "0.9.3")
-    set(LUA_SQLITE_FILENAME lsqlite3_fsl09w.zip)
-  elseif (${CIVETWEB_LUA_SQLITE_VERSION} VERSION_EQUAL "0.9.2")
-    set(LUA_SQLITE_FILENAME lsqlite3_fsl09v.zip)
-  elseif (${CIVETWEB_LUA_SQLITE_VERSION} VERSION_EQUAL "0.9.1")
-    set(LUA_SQLITE_FILENAME lsqlite3_fsl09t.zip)
+  if (NOT CIVETWEB_ENABLE_LUA_SQLITE_SHARED)
+    if (${CIVETWEB_LUA_SQLITE_VERSION} VERSION_EQUAL "0.9.3")
+      set(LUA_SQLITE_FILENAME lsqlite3_fsl09w.zip)
+    elseif (${CIVETWEB_LUA_SQLITE_VERSION} VERSION_EQUAL "0.9.2")
+      set(LUA_SQLITE_FILENAME lsqlite3_fsl09v.zip)
+    elseif (${CIVETWEB_LUA_SQLITE_VERSION} VERSION_EQUAL "0.9.1")
+      set(LUA_SQLITE_FILENAME lsqlite3_fsl09t.zip)
+    else()
+      message(FATAL_ERROR "The Lua SQLite archive filename is unknown for version ${CIVETWEB_LUA_SQLITE_VERSION}")
+    endif()
+    ExternalProject_Add(luasqlite
+      URL "http://lua.sqlite.org/index.cgi/zip/${LUA_SQLITE_FILENAME}"
+      URL_MD5 ${CIVETWEB_LUA_SQLITE_MD5_HASH}
+      PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
+      PATCH_COMMAND ${CMAKE_COMMAND} -E copy
+        "${CMAKE_CURRENT_SOURCE_DIR}/cmake/luasqlite/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
+      CMAKE_ARGS
+        "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+        "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
+      LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
+      LOG_UPDATE ${THIRD_PARTY_LOGGING}
+      LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
+      LOG_BUILD ${THIRD_PARTY_LOGGING}
+      LOG_TEST ${THIRD_PARTY_LOGGING}
+      LOG_INSTALL ${THIRD_PARTY_LOGGING})
+    ExternalProject_Get_Property(luasqlite INSTALL_DIR)
+    set(LUA_SQLITE_INSTALL_DIR ${INSTALL_DIR})
+    unset(INSTALL_DIR)
+    link_directories("${LUA_SQLITE_INSTALL_DIR}/lib")
+    add_library(lsqlite3_lib STATIC IMPORTED)
+    set_property(TARGET lsqlite3_lib PROPERTY IMPORTED_LOCATION "${LUA_SQLITE_INSTALL_DIR}/lib/libluasqlite.a")
+    add_dependencies(civetweb-c-library luasqlite)
   else()
-    message(FATAL_ERROR "The Lua SQLite archive filename is unknown for version ${CIVETWEB_LUA_SQLITE_VERSION}")
+    add_library(lsqlite3_lib SHARED IMPORTED)
+    set_property(TARGET lsqlite3_lib PROPERTY IMPORTED_LOCATION "${LUA_CDIR}/lsqlite3.so")
   endif()
-  ExternalProject_Add(luasqlite
-    URL "http://lua.sqlite.org/index.cgi/zip/${LUA_SQLITE_FILENAME}"
-    URL_MD5 ${CIVETWEB_LUA_SQLITE_MD5_HASH}
-    PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
-    PATCH_COMMAND ${CMAKE_COMMAND} -E copy
-      "${CMAKE_CURRENT_SOURCE_DIR}/cmake/luasqlite/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
-    CMAKE_ARGS
-      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
-      "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
-    LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
-    LOG_UPDATE ${THIRD_PARTY_LOGGING}
-    LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
-    LOG_BUILD ${THIRD_PARTY_LOGGING}
-    LOG_TEST ${THIRD_PARTY_LOGGING}
-    LOG_INSTALL ${THIRD_PARTY_LOGGING})
-  ExternalProject_Get_Property(luasqlite INSTALL_DIR)
-  set(LUA_SQLITE_INSTALL_DIR ${INSTALL_DIR})
-  unset(INSTALL_DIR)
-  link_directories("${LUA_SQLITE_INSTALL_DIR}/lib")
-  set(LUA_LIBRARIES "${LUA_LIBRARIES};${LUA_SQLITE_INSTALL_DIR}/lib/libluasqlite.a")
-  add_dependencies(civetweb-c-library luasqlite)
 
   # Lua XML Support
-  if (${CIVETWEB_LUA_XML_VERSION} VERSION_EQUAL "1.8.0")
-    set(LUA_XML_FILENAME LuaXML_130610.zip)
-  elseif (${CIVETWEB_LUA_XML_VERSION} VERSION_EQUAL "1.7.4")
-    set(LUA_XML_FILENAME LuaXML_101012.zip)
-  else()
-    message(FATAL_ERROR "The Lua XML archive filename is unknown for version ${CIVETWEB_LUA_XML_VERSION}")
-  endif()
-  ExternalProject_Add(luaxml
+  if (NOT CIVETWEB_ENABLE_LUA_XML_SHARED)
+    if (${CIVETWEB_LUA_XML_VERSION} VERSION_EQUAL "1.8.0")
+      set(LUA_XML_FILENAME LuaXML_130610.zip)
+    elseif (${CIVETWEB_LUA_XML_VERSION} VERSION_EQUAL "1.7.4")
+      set(LUA_XML_FILENAME LuaXML_101012.zip)
+    else()
+      message(FATAL_ERROR "The Lua XML archive filename is unknown for version ${CIVETWEB_LUA_XML_VERSION}")
+    endif()
+    ExternalProject_Add(luaxml
 # Old:
 #    URL "http://viremo.eludi.net/LuaXML/${LUA_XML_FILENAME}"
 #    URL_MD5 ${CIVETWEB_LUA_XML_MD5_HASH}
@@ -191,59 +222,71 @@ if (CIVETWEB_ENABLE_LUA)
 # An extended version is available at https://github.com/n1tehawk/LuaXML
 # but the last commit there was in 2015
 #
-    URL "https://github.com/n1tehawk/LuaXML/archive/v1.8.0.zip"
-    PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
+      URL "https://github.com/n1tehawk/LuaXML/archive/v1.8.0.zip"
+      PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
 #
 # TODO: fix this patch command (needs someone with deeper CMake know how)
-    PATCH_COMMAND ${CMAKE_COMMAND} -E copy
-      "${CMAKE_CURRENT_SOURCE_DIR}/cmake/luaxml/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
-    CMAKE_ARGS
-      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
-      "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
-    LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
-    LOG_UPDATE ${THIRD_PARTY_LOGGING}
-    LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
-    LOG_BUILD ${THIRD_PARTY_LOGGING}
-    LOG_TEST ${THIRD_PARTY_LOGGING}
-    LOG_INSTALL ${THIRD_PARTY_LOGGING})
-  ExternalProject_Get_Property(luaxml INSTALL_DIR)
-  set(LUA_XML_INSTALL_DIR ${INSTALL_DIR})
-  unset(INSTALL_DIR)
-  link_directories("${LUA_XML_INSTALL_DIR}/lib")
-  set(LUA_LIBRARIES "${LUA_LIBRARIES};${LUA_XML_INSTALL_DIR}/lib/libluaxml.a")
-  add_dependencies(civetweb-c-library luaxml)
+      PATCH_COMMAND ${CMAKE_COMMAND} -E copy
+        "${CMAKE_CURRENT_SOURCE_DIR}/cmake/luaxml/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
+      CMAKE_ARGS
+        "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+        "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
+      LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
+      LOG_UPDATE ${THIRD_PARTY_LOGGING}
+      LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
+      LOG_BUILD ${THIRD_PARTY_LOGGING}
+      LOG_TEST ${THIRD_PARTY_LOGGING}
+      LOG_INSTALL ${THIRD_PARTY_LOGGING})
+    ExternalProject_Get_Property(luaxml INSTALL_DIR)
+    set(LUA_XML_INSTALL_DIR ${INSTALL_DIR})
+    unset(INSTALL_DIR)
+    link_directories("${LUA_XML_INSTALL_DIR}/lib")
+    add_library(luaxml_lib STATIC IMPORTED)
+    set_property(TARGET luaxml_lib PROPERTY IMPORTED_LOCATION "${LUA_XML_INSTALL_DIR}/lib/libluaxml.a")
+    add_dependencies(civetweb-c-library luaxml)
+  else()
+    add_library(luaxml_lib SHARED IMPORTED)
+    set_property(TARGET luaxml_lib PROPERTY IMPORTED_LOCATION "${LUA_CDIR}/LuaXML_lib.so")
+  endif()
 
   # SQLite Support
-  string (REGEX MATCHALL "[0-9]+" SQLITE_VERSION_MATCHES ${CIVETWEB_SQLITE_VERSION})
-  list(GET SQLITE_VERSION_MATCHES 0 SQLITE_VERSION_MAJOR)
-  list(GET SQLITE_VERSION_MATCHES 1 SQLITE_VERSION_MINOR)
-  list(GET SQLITE_VERSION_MATCHES 2 SQLITE_VERSION_PATCH)
-  set(SQLITE_FILE_VERSION ${SQLITE_VERSION_MAJOR}0${SQLITE_VERSION_MINOR}0${SQLITE_VERSION_PATCH}00)
-  ExternalProject_Add(sqlite
-    URL "http://www.sqlite.org/2015/sqlite-amalgamation-${SQLITE_FILE_VERSION}.zip"
-    URL_MD5 ${CIVETWEB_SQLITE_MD5_HASH}
-    PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
-    PATCH_COMMAND ${CMAKE_COMMAND} -E copy
-      "${CMAKE_CURRENT_SOURCE_DIR}/cmake/sqlite/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
-    CMAKE_ARGS
-      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
-      "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
-    LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
-    LOG_UPDATE ${THIRD_PARTY_LOGGING}
-    LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
-    LOG_BUILD ${THIRD_PARTY_LOGGING}
-    LOG_TEST ${THIRD_PARTY_LOGGING}
-    LOG_INSTALL ${THIRD_PARTY_LOGGING})
-  ExternalProject_Get_Property(sqlite INSTALL_DIR)
-  set(SQLITE_INSTALL_DIR ${INSTALL_DIR})
-  unset(INSTALL_DIR)
-  link_directories("${SQLITE_INSTALL_DIR}/lib")
-  include_directories("${SQLITE_INSTALL_DIR}/include")
-  set(LUA_LIBRARIES "${LUA_LIBRARIES};${SQLITE_INSTALL_DIR}/lib/libsqlite.a")
-  add_dependencies(civetweb-c-library sqlite)
+  if (NOT CIVETWEB_ENABLE_SQLITE_SHARED)
+    string (REGEX MATCHALL "[0-9]+" SQLITE_VERSION_MATCHES ${CIVETWEB_SQLITE_VERSION})
+    list(GET SQLITE_VERSION_MATCHES 0 SQLITE_VERSION_MAJOR)
+    list(GET SQLITE_VERSION_MATCHES 1 SQLITE_VERSION_MINOR)
+    list(GET SQLITE_VERSION_MATCHES 2 SQLITE_VERSION_PATCH)
+    set(SQLITE_FILE_VERSION ${SQLITE_VERSION_MAJOR}0${SQLITE_VERSION_MINOR}0${SQLITE_VERSION_PATCH}00)
+    ExternalProject_Add(sqlite
+      URL "http://www.sqlite.org/2015/sqlite-amalgamation-${SQLITE_FILE_VERSION}.zip"
+      URL_MD5 ${CIVETWEB_SQLITE_MD5_HASH}
+      PREFIX "${CIVETWEB_THIRD_PARTY_DIR}"
+      PATCH_COMMAND ${CMAKE_COMMAND} -E copy
+        "${CMAKE_CURRENT_SOURCE_DIR}/cmake/sqlite/CMakeLists.txt" <SOURCE_DIR>/CMakeLists.txt
+      CMAKE_ARGS
+        "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+        "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
+      LOG_DOWNLOAD ${THIRD_PARTY_LOGGING}
+      LOG_UPDATE ${THIRD_PARTY_LOGGING}
+      LOG_CONFIGURE ${THIRD_PARTY_LOGGING}
+      LOG_BUILD ${THIRD_PARTY_LOGGING}
+      LOG_TEST ${THIRD_PARTY_LOGGING}
+      LOG_INSTALL ${THIRD_PARTY_LOGGING})
+    ExternalProject_Get_Property(sqlite INSTALL_DIR)
+    set(SQLITE_INSTALL_DIR ${INSTALL_DIR})
+    unset(INSTALL_DIR)
+    link_directories("${SQLITE_INSTALL_DIR}/lib")
+    include_directories("${SQLITE_INSTALL_DIR}/include")
+    add_library(sqlite_lib STATIC IMPORTED)
+    set_property(TARGET sqlite_lib PROPERTY IMPORTED_LOCATION "${SQLITE_INSTALL_DIR}/lib/libsqlite.a")
+    add_dependencies(civetweb-c-library sqlite)
+  else()
+    add_library(sqlite_lib SHARED IMPORTED)
+    set_property(TARGET sqlite_lib PROPERTY IMPORTED_LOCATION "/usr/${_GET_LIBDIR}/libsqlite3.so")
+  endif()
 
   # Link all the Lua libraries
-  target_link_libraries(civetweb-c-library ${LUA_LIBRARIES})
+  message(STATUS "LUA_LIBRARIES=${LUA_LIBRARIES}")
+  target_link_libraries(civetweb-c-library ${LUA_LIBRARIES} lfs_lib lsqlite3_lib luaxml_lib sqlite_lib)
 endif()
 
 # The web server executable
@@ -264,7 +307,12 @@ if (CIVETWEB_ENABLE_SERVER_EXECUTABLE)
     endif()
     target_include_directories(
         civetweb-c-executable PUBLIC
-        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+        )
+    target_link_directories(
+        civetweb-c-executable PRIVATE
+        ${CMAKE_BINARY_DIR}
+        )
     target_link_libraries(civetweb-c-executable civetweb-c-library)
     if (LIBRT_FOUND)
         target_link_libraries(civetweb-c-executable LIBRT::LIBRT)
@@ -278,9 +326,17 @@ if (CIVETWEB_ENABLE_LUA)
     VERSION ${CIVETWEB_VERSION}
     SOVERSION ${CIVETWEB_VERSION}
   )
+  if (CIVETWEB_ENABLE_LUA_SHARED)
+  target_include_directories(
+    lua-library PUBLIC
+    ${LUA_INC}
+    )
+  else ()
   target_include_directories(
     lua-library PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/third_party/lua-5.2.4>)
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/third_party/lua-${CIVETWEB_LUA_VERSION}/src>
+    )
+  endif ()
   install(
     TARGETS lua-library
     EXPORT ${PROJECT_NAME}-targets
@@ -303,12 +359,17 @@ if (CIVETWEB_ENABLE_CXX)
   if (BUILD_SHARED_LIBS)
     target_compile_definitions(civetweb-cpp PRIVATE CIVETWEB_CXX_DLL_EXPORTS INTERFACE CIVETWEB_CXX_DLL_IMPORTS)
   endif()
+  target_link_directories(
+      civetweb-cpp PRIVATE
+      ${CMAKE_BINARY_DIR}
+      )
   target_link_libraries(
     civetweb-cpp PUBLIC
 	civetweb-c-library)
   target_include_directories(
     civetweb-cpp PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+    )
   install(
     TARGETS civetweb-cpp
     EXPORT ${PROJECT_NAME}-targets
