--- a/datasets/prepare_dataset.py.orig	2024-06-09 10:48:35.913693500 -0700
+++ b/datasets/prepare_dataset.py	2024-06-09 10:49:46.449119819 -0700
@@ -37,7 +37,8 @@ class SceneChangeDetector:
             for j in range(frames_number):
                 print('Calculating histogram differences for ',  os.path.split(video_fn)[-1], '. Frame: ', j, end='\r')
                 ret, frame = video.read()
-                frame_br = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+                a = np.asarray(frame)
+                frame_br = cv2.cvtColor(a, cv2.COLOR_BGR2GRAY)
                 cur_hist = [[np.histogram(frame_br[y * self.hist_block_size:(y + 1) * self.hist_block_size,
                                                    x * self.hist_block_size:(x + 1) * self.hist_block_size],
                                           bins=self.hist_nbins,
@@ -128,9 +129,11 @@ def process_video(video_path, scd, write
                 if args.blur:
                     frame_lr = cv2.GaussianBlur(frame_lr, (args.blur_size, args.blur_size), args.blur_sigma,
                                                 borderType=cv2.BORDER_REFLECT_101)
-                frames_lr.append(cv2.cvtColor(frame_lr, cv2.COLOR_BGR2YUV)[:, :, 0])
+                a = np.asarray(frame_lr)
+                frames_lr.append(cv2.cvtColor(a, cv2.COLOR_BGR2YUV)[:, :, 0])
                 if k == args.temporal_radius:
-                    frame_hr = cv2.cvtColor(frames[k], cv2.COLOR_BGR2YUV)[:, :, 0]
+                    a = np.asarray(frames[k])
+                    frame_hr = cv2.cvtColor(a, cv2.COLOR_BGR2YUV)[:, :, 0]
             if args.type == 'blocks':
                 for y in range(frames_lr[0].shape[0] // args.stride - 1):
                     for x in range(frames_lr[0].shape[1] // args.stride - 1):
--- a/datasets/prepare_div2k_dataset.py.orig	2024-06-09 10:48:35.877693794 -0700
+++ b/datasets/prepare_div2k_dataset.py	2024-06-09 10:50:01.344998664 -0700
@@ -2,6 +2,7 @@ import os
 import argparse
 from tqdm import tqdm
 import cv2
+import numpy as np
 try:
     import tensorflow.compat.v1 as tf
 except ImportError:
@@ -65,8 +66,10 @@ def main():
             image = image[:args.crop_height, :args.crop_width]
             image_lr = image_lr[:args.crop_height // args.scale_factor, :args.crop_width // args.scale_factor]
 
-        image = cv2.cvtColor(image, cv2.COLOR_BGR2YUV)[:, :, 0]
-        image_lr = cv2.cvtColor(image_lr, cv2.COLOR_BGR2YUV)[:, :, 0]
+        a = np.asarray(image, dtype = 'uint8')
+        image = cv2.cvtColor(a, cv2.COLOR_BGR2YUV)[:, :, 0]
+        a = np.asarray(image_lr, dtype = 'uint8')
+        image_lr = cv2.cvtColor(a, cv2.COLOR_BGR2YUV)[:, :, 0]
         if args.type == 'blocks':
             height = image_lr.shape[0]
             width = image_lr.shape[1]
