diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/client_example/CMakeLists.txt composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/client_example/CMakeLists.txt
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/client_example/CMakeLists.txt	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/client_example/CMakeLists.txt	2023-10-02 13:52:21.180446688 -0700
@@ -3,7 +3,7 @@ project(ck_app)
 add_compile_options(-std=c++17)
 
 find_package(composable_kernel 1.0.0 COMPONENTS device_operations)
-find_package(hip REQUIRED PATHS /opt/rocm)
+find_package(hip REQUIRED PATHS @ESYSROOT_ROCM_PATH@)
 message(STATUS "Build with HIP ${hip_VERSION}")
 
 # add all example subdir
Only in composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/client_example: CMakeLists.txt.orig
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/client_example/README.md composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/client_example/README.md
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/client_example/README.md	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/client_example/README.md	2023-10-02 13:52:21.184446688 -0700
@@ -10,8 +10,8 @@ cd client_example/build
 
 ```bash
 cmake                                                                 \
--D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc                             \
--D CMAKE_PREFIX_PATH="/opt/rocm;${PATH_TO_CK_INSTALL_DIRECTORY}"      \
+-D CMAKE_CXX_COMPILER=@EPREFIX_ROCM_PATH@/bin/hipcc                             \
+-D CMAKE_PREFIX_PATH="@EPREFIX_ROCM_PATH@;${PATH_TO_CK_INSTALL_DIRECTORY}"      \
 ..
 ```
 
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/cmake/ClangTidy.cmake composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/cmake/ClangTidy.cmake
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/cmake/ClangTidy.cmake	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/cmake/ClangTidy.cmake	2023-10-02 13:52:21.184446688 -0700
@@ -43,9 +43,8 @@ find_program(CLANG_TIDY_EXE
     PATH_SUFFIXES
         compiler/bin
     PATHS
-        /opt/rocm/llvm/bin
-        /opt/rocm/hcc
-        /usr/local/opt/llvm/bin
+        @ESYSROOT_LLVM_PATH@
+        @ESYSROOT@/usr/hcc
 )
 
 function(find_clang_tidy_version VAR)
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/cmake/CppCheck.cmake composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/cmake/CppCheck.cmake
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/cmake/CppCheck.cmake	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/cmake/CppCheck.cmake	2023-10-02 13:52:21.184446688 -0700
@@ -32,7 +32,7 @@ find_program(CPPCHECK_EXE
     NAMES 
         cppcheck
     PATHS
-        /opt/rocm/bin
+        @ESYSROOT_ROCM_PATH@/bin
 )
 
 ProcessorCount(CPPCHECK_JOBS)
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/CMakeLists.txt composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/CMakeLists.txt
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/CMakeLists.txt	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/CMakeLists.txt	2023-10-02 13:52:21.184446688 -0700
@@ -8,7 +8,7 @@ list(APPEND CMAKE_MODULE_PATH "${PROJECT
 enable_testing()
 
 set(ROCM_SYMLINK_LIBS OFF)
-find_package(ROCM REQUIRED PATHS /opt/rocm)
+find_package(ROCM REQUIRED PATHS @ESYSROOT_ROCM_PATH@)
 
 include(ROCMInstallTargets)
 include(ROCMPackageConfigHelpers)
@@ -19,7 +19,7 @@ include(CheckCXXCompilerFlag)
 
 rocm_setup_version(VERSION 0.2.0)
 include(TargetFlags)
-list(APPEND CMAKE_PREFIX_PATH ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_PREFIX}/llvm ${CMAKE_INSTALL_PREFIX}/hip /opt/rocm /opt/rocm/llvm /opt/rocm/hip)
+list(APPEND CMAKE_PREFIX_PATH ${CMAKE_INSTALL_PREFIX} @ESYSROOT_LLVM_PATH@ ${CMAKE_INSTALL_PREFIX}/hip @ESYSROOT_ROCM_PATH@ @ESYSROOT_ROCM_PATH@/hip)
 
 option(USE_BITINT_EXTENSION_INT4, "Whether to enable clang's BitInt extension to provide int4 data type." OFF)
 
@@ -227,8 +227,8 @@ enable_cppcheck(
         __linux__=1
 )
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
 
 include_directories(BEFORE
Only in composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd: CMakeLists.txt.orig
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/docs/source/dockerhub.rst composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/docs/source/dockerhub.rst
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/docs/source/dockerhub.rst	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/docs/source/dockerhub.rst	2023-10-02 13:53:16.140447808 -0700
@@ -38,8 +38,8 @@ and build the CK::
     mkdir build && cd build
     # Need to specify target ID, example below is for gfx908 and gfx90a
     cmake                                                                                             \
-    -D CMAKE_PREFIX_PATH=/opt/rocm                                                                    \
-    -D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc                                                         \
+    -D CMAKE_PREFIX_PATH=@EPREFIX_ROCM_PATH@                                                                    \
+    -D CMAKE_CXX_COMPILER=@EPREFIX_ROCM_PATH@/bin/hipcc                                                         \
     -D CMAKE_CXX_FLAGS="-O3"                                                                          \
     -D CMAKE_BUILD_TYPE=Release                                                                       \
     -D GPU_TARGETS="gfx908;gfx90a"                                                                    \
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/docs/source/tutorial_hello_world.rst composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/docs/source/tutorial_hello_world.rst
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/docs/source/tutorial_hello_world.rst	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/docs/source/tutorial_hello_world.rst	2023-10-02 13:53:02.332447527 -0700
@@ -80,8 +80,8 @@ Create and go to the "build" directory::
 In the previous section we talked about target GPU architecture. Once you decide which one is right for you, run cmake using the right GPU_TARGETS flag::
 
     cmake  \
-    -D CMAKE_PREFIX_PATH=/opt/rocm  \
-    -D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc  \
+    -D CMAKE_PREFIX_PATH=@EPREFIX_ROCM_PATH@  \
+    -D CMAKE_CXX_COMPILER=@EPREFIX_ROCM_PATH@/bin/hipcc  \
     -D CMAKE_CXX_FLAGS="-O3"  \
     -D CMAKE_BUILD_TYPE=Release  \
     -D BUILD_DEV=OFF  \
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/library/src/tensor_operation_instance/gpu/CMakeLists.txt composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/library/src/tensor_operation_instance/gpu/CMakeLists.txt
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/library/src/tensor_operation_instance/gpu/CMakeLists.txt	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/library/src/tensor_operation_instance/gpu/CMakeLists.txt	2023-10-02 13:52:23.052446726 -0700
@@ -55,7 +55,7 @@ target_compile_options(device_operations
     --offload-arch=gfx90a
 )
 
-# install(TARGETS device_operations LIBRARY DESTINATION lib)
+# install(TARGETS device_operations LIBRARY DESTINATION @LIBDIR@)
 rocm_install(TARGETS device_operations
         EXPORT device_operationsTargets)
 
Only in composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/library/src/tensor_operation_instance/gpu: CMakeLists.txt.orig
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/README.md composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/README.md
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/README.md	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/README.md	2023-10-02 13:52:23.052446726 -0700
@@ -55,8 +55,8 @@ mkdir build && cd build
 
 # Need to specify target ID, example below is for gfx908 and gfx90a
 cmake                                                                                             \
--D CMAKE_PREFIX_PATH=/opt/rocm                                                                    \
--D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc                                                         \
+-D CMAKE_PREFIX_PATH=@EPREFIX_ROCM_PATH@                                                                         \
+-D CMAKE_CXX_COMPILER=@EPREFIX_ROCM_PATH@/bin/hipcc                                                              \
 -D CMAKE_CXX_FLAGS="-O3"                                                                          \
 -D CMAKE_BUILD_TYPE=Release                                                                       \
 -D GPU_TARGETS="gfx908;gfx90a"                                                                    \
Only in composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd: README.md.orig
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/cmake-ck-dev.sh composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/cmake-ck-dev.sh
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/cmake-ck-dev.sh	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/cmake-ck-dev.sh	2023-10-02 13:53:38.588448266 -0700
@@ -6,8 +6,8 @@ rm -rf CMakeFiles
 MY_PROJECT_SOURCE=$1
 
 cmake                                                                                             \
--D CMAKE_PREFIX_PATH=/opt/rocm                                                                    \
--D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc                                                         \
+-D CMAKE_PREFIX_PATH=@EPREFIX_ROCM_PATH@                                                                    \
+-D CMAKE_CXX_COMPILER=@EPREFIX_ROCM_PATH@/bin/hipcc                                                         \
 -D CMAKE_CXX_FLAGS="-O3 -ftemplate-backtrace-limit=0 -gline-tables-only -save-temps=$PWD"         \
 -D CMAKE_BUILD_TYPE=Release                                                                       \
 -D BUILD_DEV=ON                                                                                   \
Only in composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script: cmake-ck-dev.sh.orig
Only in composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script: cmake-ck-dev.sh.rej
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/cmake-ck-release.sh composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/cmake-ck-release.sh
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/cmake-ck-release.sh	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/cmake-ck-release.sh	2023-10-02 13:52:23.052446726 -0700
@@ -6,8 +6,8 @@ rm -rf CMakeFiles
 MY_PROJECT_SOURCE=$1
 
 cmake                                                                                             \
--D CMAKE_PREFIX_PATH=/opt/rocm                                                                    \
--D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc                                                         \
+-D CMAKE_PREFIX_PATH=@ESYSROOT_ROCM_PATH@                                                                         \
+-D CMAKE_CXX_COMPILER=@ESYSROOT_ROCM_PATH@/bin/hipcc                                                              \
 -D CMAKE_CXX_FLAGS="-O3"                                                                          \
 -D CMAKE_BUILD_TYPE=Release                                                                       \
 -D BUILD_DEV=OFF                                                                                  \
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/hipclang_opt.sh composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/hipclang_opt.sh
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/hipclang_opt.sh	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/hipclang_opt.sh	2023-10-02 13:52:23.052446726 -0700
@@ -2,24 +2,24 @@ rm *.ll *.s
 
 BC_FILE=$1
 
-/opt/rocm/llvm/bin/llvm-dis $BC_FILE -o original.ll
-/opt/rocm/llvm/bin/opt -S -inline -inline-threshold=104857 original.ll > inline.ll
-/opt/rocm/llvm/bin/opt -S -sroa inline.ll > sroa.ll
-/opt/rocm/llvm/bin/opt -S -O3 sroa.ll > o3.ll
+@ESYSROOT_LLVM_PATH@/bin/llvm-dis $BC_FILE -o original.ll
+@ESYSROOT_LLVM_PATH@/bin/opt -S -inline -inline-threshold=104857 original.ll > inline.ll
+@ESYSROOT_LLVM_PATH@/bin/opt -S -sroa inline.ll > sroa.ll
+@ESYSROOT_LLVM_PATH@/bin/opt -S -O3 sroa.ll > o3.ll
 
-/opt/rocm/llvm/bin/llc -mcpu=gfx906 original.ll
-/opt/rocm/llvm/bin/llc -mcpu=gfx906 inline.ll
-/opt/rocm/llvm/bin/llc -mcpu=gfx906 sroa.ll
-/opt/rocm/llvm/bin/llc -mcpu=gfx906 o3.ll
+@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx906 original.ll
+@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx906 inline.ll
+@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx906 sroa.ll
+@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx906 o3.ll
 
-#/opt/rocm/llvm/bin/opt -S -O3 -sroa inline.ll > o3.ll
-#/opt/rocm/llvm/bin/opt -S -O3 -sroa o3.ll > o3_2.ll
-#/opt/rocm/llvm/bin/opt -S -O3 -sroa o3_2.ll > o3_3.ll
-#/opt/rocm/llvm/bin/opt -S -O3 -sroa o3_3.ll > o3_4.ll
+#@ESYSROOT_LLVM_PATH@/bin/opt -S -O3 -sroa inline.ll > o3.ll
+#@ESYSROOT_LLVM_PATH@/bin/opt -S -O3 -sroa o3.ll > o3_2.ll
+#@ESYSROOT_LLVM_PATH@/bin/opt -S -O3 -sroa o3_2.ll > o3_3.ll
+#@ESYSROOT_LLVM_PATH@/bin/opt -S -O3 -sroa o3_3.ll > o3_4.ll
 
-#/opt/rocm/llvm/bin/llc -mcpu=gfx908 opt.ll
-#/opt/rocm/llvm/bin/llc -mcpu=gfx908 inline.ll
-#/opt/rocm/llvm/bin/llc -mcpu=gfx908 o3.ll
-#/opt/rocm/llvm/bin/llc -mcpu=gfx908 o3_2.ll
-#/opt/rocm/llvm/bin/llc -mcpu=gfx908 o3_3.ll
-#/opt/rocm/llvm/bin/llc -mcpu=gfx908 o3_4.ll
+#@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx908 opt.ll
+#@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx908 inline.ll
+#@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx908 o3.ll
+#@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx908 o3_2.ll
+#@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx908 o3_3.ll
+#@ESYSROOT_LLVM_PATH@/bin/llc -mcpu=gfx908 o3_4.ll
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/parse_perf_data.py composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/parse_perf_data.py
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/parse_perf_data.py	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/parse_perf_data.py	2023-10-02 13:52:23.052446726 -0700
@@ -69,7 +69,7 @@ def main():
                 compute_units=lst[2]
             if 'InstalledDir' in line:
                 lst=line.split()
-                rocm_vers=lst[1][lst[1].find('/opt/rocm-')+len('/opt/rocm-'):lst[1].rfind('/llvm/bin')]
+                rocm_vers="@ROCM_VERSION@"
     print("Branch name:",branch_name)
     print("Node name:",node_id)
     print("GPU_arch:",gpu_arch)
@@ -287,4 +287,4 @@ def main():
     return regression
 
 if __name__ == '__main__':
-    main()
\ No newline at end of file
+    main()
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/process_perf_data.py composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/process_perf_data.py
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/process_perf_data.py	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/process_perf_data.py	2023-10-02 13:52:23.052446726 -0700
@@ -64,7 +64,7 @@ def get_log_params(logfile):
             environment=lst[2]
         if 'InstalledDir' in line:
             lst=line.split()
-            rocm_vers=lst[1][lst[1].find('/opt/rocm-')+len('/opt/rocm-'):lst[1].rfind('/llvm/bin')]
+            rocm_vers="@ROCM_VERSION@"
     return branch_name, node_id, gpu_arch, compute_units, rocm_vers, hip_vers, environment
 
 def parse_logfile(logfile):
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/run_full_performance_tests.sh composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/run_full_performance_tests.sh
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/run_full_performance_tests.sh	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/run_full_performance_tests.sh	2023-10-02 13:52:23.052446726 -0700
@@ -31,7 +31,7 @@ function print_log_header(){
 	rocminfo | grep "Compute Unit:" >> $1;
 	hipcc --version | grep -e 'HIP version'  >> $1;
 	echo 'Environment type: ' $2 >> $1;
-	/opt/rocm/bin/amdclang++ --version | grep -e 'InstalledDir' >> $1;
+	@EPREFIX_LLVM_PATH@/bin/clang++ --version | grep -e 'InstalledDir' >> $1;
 }
 
 #run gemm tests
diff -urp composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/run_performance_tests.sh composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/run_performance_tests.sh
--- composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd.orig/script/run_performance_tests.sh	2023-04-26 13:48:22.000000000 -0700
+++ composable_kernel-0a8dac4ef1a232abd8f6896a5b016f9e76192ddd/script/run_performance_tests.sh	2023-10-02 13:52:23.052446726 -0700
@@ -28,7 +28,7 @@ function print_log_header(){
 	rocminfo | grep "Compute Unit:" >> $1;
 	hipcc --version | grep -e 'HIP version'  >> $1;
 	echo 'Environment type: ' $2 >> $1;
-	/opt/rocm/bin/amdclang++ --version | grep -e 'InstalledDir' >> $1;
+	@EPREFIX_LLVM_PATH@/bin/clang++ --version | grep -e 'InstalledDir' >> $1;
 }
 
 #run gemm tests
