diff -urp ckolivas-ckpool-5e93cf821949.orig/configure.ac ckolivas-ckpool-5e93cf821949/configure.ac
--- ckolivas-ckpool-5e93cf821949.orig/configure.ac	2017-08-16 21:53:58.000000000 -0700
+++ ckolivas-ckpool-5e93cf821949/configure.ac	2017-10-02 03:19:41.583083436 -0700
@@ -69,10 +69,10 @@ if test x$sse4 = xsse4_1; then
 	AC_DEFINE([USE_SSE4], [1], [Use sse4 assembly instructions for sha256])
 fi
 
-AC_CONFIG_SUBDIRS([src/jansson-2.10])
-JANSSON_LIBS="jansson-2.10/src/.libs/libjansson.a"
+#AC_CONFIG_SUBDIRS([src/jansson-2.10])
+#JANSSON_LIBS="jansson-2.10/src/.libs/libjansson.a"
 
-AC_SUBST(JANSSON_LIBS)
+AC_SUBST(JANSSON_LIBS,[])
 
 AC_ARG_WITH([ckdb],
 	[AC_HELP_STRING([--with-ckdb],[Compile ckpool with ckdb database support (default disabled)])],
Only in ckolivas-ckpool-5e93cf821949: configure.ac.orig
Only in ckolivas-ckpool-5e93cf821949: configure.ac.rej
diff -urp ckolivas-ckpool-5e93cf821949.orig/src/Makefile.am ckolivas-ckpool-5e93cf821949/src/Makefile.am
--- ckolivas-ckpool-5e93cf821949.orig/src/Makefile.am	2017-08-16 21:53:58.000000000 -0700
+++ ckolivas-ckpool-5e93cf821949/src/Makefile.am	2017-10-02 03:20:13.817427145 -0700
@@ -1,7 +1,7 @@
-SUBDIRS	= jansson-2.10
+#SUBDIRS	= jansson-2.10
 
 ACLOCAL_AMFLAGS = -I m4
-AM_CPPFLAGS =  -I$(top_srcdir)/src/jansson-2.10/src
+#AM_CPPFLAGS =  -I$(top_srcdir)/src/jansson-2.10/src
 
 native_objs :=
 
Only in ckolivas-ckpool-5e93cf821949/src: Makefile.am.orig
Only in ckolivas-ckpool-5e93cf821949/src: Makefile.am.rej
