diff -urp ckolivas-ckpool-5e93cf821949.orig/pool/email.php ckolivas-ckpool-5e93cf821949/pool/email.php
--- ckolivas-ckpool-5e93cf821949.orig/pool/email.php	2017-08-16 21:53:58.000000000 -0700
+++ ckolivas-ckpool-5e93cf821949/pool/email.php	2017-10-02 15:08:07.048571351 -0700
@@ -100,7 +100,7 @@ function passReset($to, $code, $whoip, $
  $message = "Someone requested to reset your password.$eol$eol";
  $message .= "You can ignore this message since nothing has changed yet,$eol";
  $message .= "or click on the link below to reset your password.$eol";
- $message .= "$web/index.php?k=reset&code=$code$eol$eol";
+ $message .= 'https://' . $_SERVER['HTTP_HOST'] . ((dirname($_SERVER['REQUEST_URI']) == "/") ? $_SERVER['REQUEST_URI'] : dirname($_SERVER['REQUEST_URI'])) . "/index.php?k=reset&code=$code$eol$eol";
  $message .= $ret;
 
  return sendnoheader($to, "Password Reset", $message, $emailinfo);
