Mitigate against cold boot attack
--- a/data/org.gnome.World.Secrets.gschema.xml.in.orig	2026-02-22 05:59:13.000000000 -0800
+++ b/data/org.gnome.World.Secrets.gschema.xml.in	2026-05-05 20:40:22.929728854 -0700
@@ -23,7 +23,7 @@
             <description>Save every change you made instantly into the database. Please note that you cannot revert changes if Autosave is enabled.</description>
         </key>
         <key type="i" name="database-lock-timeout">
-            <default>5</default>
+            <default>1</default>
             <summary>Lock database after X minutes</summary>
             <description>Automatically lock your database after a given amount of minutes to improve the security.</description>
         </key>
