diff -urp rdc-rocm-5.1.3.orig/CMakeLists.txt rdc-rocm-5.1.3/CMakeLists.txt
--- rdc-rocm-5.1.3.orig/CMakeLists.txt	2022-03-18 09:09:27.000000000 -0700
+++ rdc-rocm-5.1.3/CMakeLists.txt	2023-08-08 21:39:06.698360337 -0700
@@ -46,6 +46,7 @@ if( NOT EXISTS "${CMAKE_SOURCE_DIR}/rasl
     )
 endif()
 
+option(INSTALL_RASLIB "Install the RAS (Reliability, Availability, and Serviceability) plugin." OFF)
 if (BUILD_RASLIB AND NOT DEFINED HSA_DIR)
     message( FATAL_ERROR "Please specify libhsakmt directory which is required by raslib
     cmake -DHSA_DIR=<libhsakmt directory>
@@ -290,15 +291,17 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/
                                          COMPONENT ${CLIENT_COMPONENT})
 
 # Prebuild packages to install
-install(FILES  ${PROJECT_SOURCE_DIR}/ras_prebuild/librdc_ras.so
-    DESTINATION  ${RDC_CLIENT_INSTALL_PREFIX}/${RDC}/lib
-    COMPONENT ${CLIENT_COMPONENT})
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/config
-     DESTINATION  ${RDC_CLIENT_INSTALL_PREFIX}/${RDC}/lib
-     COMPONENT ${CLIENT_COMPONENT})
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/sp3
-     DESTINATION  ${RDC_CLIENT_INSTALL_PREFIX}/${RDC}/lib
-     COMPONENT ${CLIENT_COMPONENT})
+if(INSTALL_RASLIB)
+    install(FILES  ${PROJECT_SOURCE_DIR}/ras_prebuild/librdc_ras.so
+        DESTINATION  ${RDC_CLIENT_INSTALL_PREFIX}/${RDC}/lib
+        COMPONENT ${CLIENT_COMPONENT})
+    install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/config
+        DESTINATION  ${RDC_CLIENT_INSTALL_PREFIX}/${RDC}/lib
+        COMPONENT ${CLIENT_COMPONENT})
+    install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/sp3
+        DESTINATION  ${RDC_CLIENT_INSTALL_PREFIX}/${RDC}/lib
+        COMPONENT ${CLIENT_COMPONENT})
+endif()
 
 
 set(CPACK_PACKAGE_NAME ${RDC_PACKAGE})
Only in rdc-rocm-5.1.3: CMakeLists.txt.orig
Only in rdc-rocm-5.1.3: CMakeLists.txt.rej
