diff -urp rdc-rocm-5.4.3.orig/CMakeLists.txt rdc-rocm-5.4.3/CMakeLists.txt
--- rdc-rocm-5.4.3.orig/CMakeLists.txt	2023-01-25 10:51:10.000000000 -0800
+++ rdc-rocm-5.4.3/CMakeLists.txt	2023-08-08 22:45:15.398463704 -0700
@@ -59,6 +59,12 @@ option(CMAKE_VERBOSE_MAKEFILE "Enable ve
 
 option(CMAKE_EXPORT_COMPILE_COMMANDS "Export compile commands for linters and autocompleters" ON)
 
+option(INSTALL_RASLIB "Install the RAS (Reliability, Availability, and Serviceability) plugin." OFF)
+
+if(NOT INSTALL_RASLIB)
+    add_defintion(-DRASLIB_DISABLED)
+endif()
+
 # this must go after some CMAKE_* variables
 project(${RDC}
     VERSION "${VERSION_STRING}"
@@ -336,15 +342,17 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/
     COMPONENT ${CLIENT_COMPONENT})
 
 # Prebuild packages to install
-install(FILES ${PROJECT_SOURCE_DIR}/ras_prebuild/librdc_ras.so
-    DESTINATION ${RDC_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${RDC}
-    COMPONENT ${CLIENT_COMPONENT})
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/config
-    DESTINATION ${RDC_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/${RDC}
-    COMPONENT ${CLIENT_COMPONENT})
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/sp3
-    DESTINATION ${RDC_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${RDC}
-    COMPONENT ${CLIENT_COMPONENT})
+if(INSTALL_RASLIB)
+    install(FILES ${PROJECT_SOURCE_DIR}/ras_prebuild/librdc_ras.so
+        DESTINATION ${RDC_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${RDC}
+        COMPONENT ${CLIENT_COMPONENT})
+    install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/config
+        DESTINATION ${RDC_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/${RDC}
+        COMPONENT ${CLIENT_COMPONENT})
+    install(DIRECTORY ${PROJECT_SOURCE_DIR}/ras_prebuild/sp3
+        DESTINATION ${RDC_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${RDC}
+        COMPONENT ${CLIENT_COMPONENT})
+endif()
 
 
 #Identify between SLES and Centos for setting symlink for rdc.service
diff -urp rdc-rocm-5.4.3.orig/rdc_libs/rdc/src/RdcRasLib.cc rdc-rocm-5.4.3/rdc_libs/rdc/src/RdcRasLib.cc
--- rdc-rocm-5.4.3.orig/rdc_libs/rdc/src/RdcRasLib.cc	2023-01-25 10:51:10.000000000 -0800
+++ rdc-rocm-5.4.3/rdc_libs/rdc/src/RdcRasLib.cc	2023-08-08 22:45:45.558464490 -0700
@@ -34,6 +34,10 @@ RdcRasLib::RdcRasLib(const char* lib_nam
     , fields_unwatch_(nullptr)
     , rdc_module_init_(nullptr)
     , rdc_module_destroy_(nullptr) {
+#ifdef RASLIB_DISABLED
+    RDC_LOG(RDC_ERROR, "RAS disabled.");
+    return;
+#endif
     rdc_status_t status = lib_loader_.load(lib_name);
     if (status != RDC_ST_OK) {
         RDC_LOG(RDC_ERROR, "RAS related function will not work.");
