diff -urp civetweb-1.13.orig/CMakeLists.txt civetweb-1.13/CMakeLists.txt
--- civetweb-1.13.orig/CMakeLists.txt	2020-11-27 21:49:35.803173542 -0800
+++ civetweb-1.13/CMakeLists.txt	2020-11-27 21:52:34.133998756 -0800
@@ -378,7 +378,7 @@ if (CIVETWEB_ENABLE_CXX)
     add_cxx_compiler_flag(-Werror)
     add_cxx_compiler_flag(/WX)
   endif()
-  add_cxx_compiler_flag(-pedantic-errors)
+  #add_cxx_compiler_flag(-pedantic-errors)
   add_cxx_compiler_flag(-fvisibility=hidden)
   add_cxx_compiler_flag(-fstack-protector-strong RELEASE)
 
@@ -424,7 +424,7 @@ if (NOT ZEPHYR)
   add_c_compiler_flag(-Wno-cast-qual) # const cast
   add_c_compiler_flag(/Wd4820) # padding
 
-  add_c_compiler_flag(-pedantic-errors)
+  #add_c_compiler_flag(-pedantic-errors)
   add_c_compiler_flag(-fvisibility=hidden)
   add_c_compiler_flag(-fstack-protector-strong RELEASE)
   add_c_compiler_flag(-fstack-protector-all DEBUG)
