--- Makefile.orig	2020-03-26 20:28:38.382723044 -0700
+++ Makefile	2020-03-26 20:33:22.023600932 -0700
@@ -110,6 +110,9 @@ ifdef WITH_LUAJIT_SHARED
 endif
 
 ifdef WITH_LUA
+  CFLAGS += -I/usr/include/lua__LUA_VER__
+  LIBS += -llua__LUA_VER__ /usr/lib64/lua/__LUA_VER__/lfs.so /usr/lib64/lua/__LUA_VER__/LuaXML_lib.so -lsqlite3 /usr/lib64/lua/__LUA_VER__/lsqlite3.so
+  CFLAGS += -DLUA_COMPAT_ALL -DUSE_LUA -DLUA_USE_POSIX -DLUA_USE_DLOPEN -DTHREADSAFE=1 -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DUSE_LUA_SQLITE3 -DUSE_LUA_FILE_SYSTEM -DUSE_LUA_LUAXML
   include resources/Makefile.in-lua
 endif
 
