diff -urpN blender-3.0.0.orig/build_files/cmake/platform/platform_unix.cmake blender-3.0.0/build_files/cmake/platform/platform_unix.cmake
--- blender-3.0.0.orig/build_files/cmake/platform/platform_unix.cmake	2022-02-01 10:52:28.740357288 -0800
+++ blender-3.0.0/build_files/cmake/platform/platform_unix.cmake	2022-02-01 11:00:10.425743151 -0800
@@ -304,6 +304,12 @@ if(WITH_USD)
   if(NOT USD_FOUND)
     set(WITH_USD OFF)
   endif()
+
+  find_package_wrapper(TBB2)
+  if(NOT TBB2_FOUND)
+    message(WARNING "TBB2 not found, disabling WITH_USD")
+    set(WITH_USD OFF)
+  endif()
 endif()
 
 if(WITH_BOOST)
diff -urpN blender-3.0.0.orig/source/blender/io/usd/CMakeLists.txt blender-3.0.0/source/blender/io/usd/CMakeLists.txt
--- blender-3.0.0.orig/source/blender/io/usd/CMakeLists.txt	2022-02-01 10:52:28.584351044 -0800
+++ blender-3.0.0/source/blender/io/usd/CMakeLists.txt	2022-02-01 10:53:10.054010774 -0800
@@ -52,7 +52,7 @@ set(INC
 set(INC_SYS
   ${USD_INCLUDE_DIRS}
   ${BOOST_INCLUDE_DIR}
-  ${TBB_INCLUDE_DIR}
+  ${TBB2_INCLUDE_DIR}
   ${PYTHON_INCLUDE_DIRS}
 )
 
@@ -137,12 +137,12 @@ if(WIN32)
 elseif(APPLE)
   target_link_libraries(bf_usd INTERFACE -Wl,-force_load ${USD_LIBRARIES})
 elseif(UNIX)
-  target_link_libraries(bf_usd INTERFACE "-Wl,--whole-archive ${USD_LIBRARIES} -Wl,--no-whole-archive ${TBB_LIBRARIES}")
+  target_link_libraries(bf_usd INTERFACE "-Wl,--whole-archive ${USD_LIBRARIES} -Wl,--no-whole-archive ${TBB2_LIBRARIES}")
 else()
   message(FATAL_ERROR "Unknown how to link USD with your compiler ${CMAKE_CXX_COMPILER_ID}")
 endif()
 
-target_link_libraries(bf_usd INTERFACE ${TBB_LIBRARIES})
+target_link_libraries(bf_usd INTERFACE ${TBB2_LIBRARIES})
 
 if(WITH_GTESTS)
   set(TEST_SRC
