--- setiathome-gpu-8.00.orig/AKv8/client/sah_gfx.cpp	2016-01-02 01:58:49.045953303 -0800
+++ setiathome-gpu-8.00/AKv8/client/sah_gfx.cpp	2016-01-02 02:35:53.656134504 -0800
@@ -294,6 +294,17 @@
     float tripletData[TRIPLET_POT_LEN];
     float gaussFunc[GAUSS_POT_LEN];
 	char buf[512],time_buf[256];
+    const char *TitleText="SETI@home 8";
+
+
+    int s4_id=3;
+
+    if (gdata && gdata->ready) {
+        s4_id=gdata->wu.s4_id;
+        if (s4_id > 2) {
+            TitleText="SETI@home Version 8";
+        }
+    }
 
     draw_pillars();
 
@@ -316,7 +327,7 @@
     get_user_info_string(buf);
 	draw_text(pos, 0.1, 1.0, 0.15, buf);
 	pos[0] = -2.75; pos[1] = 2.15;
-	draw_text_line(pos, 0.2, 2.0, "SETI@Home Enhanced");
+	draw_text_line(pos, 0.2, 2.0, TitleText);
 	pos[0] = -3.5; pos[1] = 1.9;
 
     get_analysis_info_string(buf);
