--- setiathome-gpu-8.00.orig/AKv8/client/Makefile.am	2016-01-02 01:58:49.126956393 -0800
+++ setiathome-gpu-8.00/AKv8/client/Makefile.am	2016-01-02 02:07:57.696904600 -0800
@@ -10,7 +10,7 @@
 if OPENCL
 BOINC_LIBS = -L$(BOINCDIR)/api -L$(BOINCDIR)/api/.libs -lboinc_api -L$(BOINCDIR)/lib -L$(BOINCDIR)/lib/.libs -lboinc -L$(BOINCDIR)/api -L$(BOINCDIR)/api/.libs -lboinc_opencl
 else
-LIBS += $(BOINCDIR)/api/.libs/libboinc_api.a $(BOINCDIR)/lib/.libs/libboinc.a
+LIBS += -lboinc_api -lboinc
 endif
 
 #AM_MAKEFLAGS = -s
@@ -22,7 +22,7 @@
 		$(BOINC_CFLAGS)  $(PTHREAD_CFLAGS)  $(ASMLIB_CFLAGS)
 
 if OPENCL
-CLIENT_CXX_FLAGS += -I$(top_srcdir)/../src/OpenCL_FFT
+CLIENT_CXX_FLAGS += -I$(top_srcdir)/../src/OpenCL_FFT -fpermissive
 endif
 
 CLIENT_LD_FLAGS = $(PTHREAD_CFLAGS) $(LDFLAGS) $(APP_LDFLAGS)
