--- AKv8/client/gdata.h.orig	2015-12-03 13:07:57.146410875 -0800
+++ AKv8/client/gdata.h	2015-12-03 14:02:57.459674859 -0800
@@ -36,6 +36,7 @@
 #define _GDATA_
 
 #include "version.h"
+#include "reduce.h"
 #include "analyze.h"
 #include "seti.h"
 
@@ -114,7 +115,12 @@
 
     SETI_WU_INFO wu;
     double local_progress;
-
+    double progress;
+    double cpu_time;
+    volatile int countdown;
+        // the graphics app sets this to 5 every second,
+        // and the main app decrements it every second.
+        // So if it's zero, it means no graphics app is running
     char status[80];
 
     FFT_INFO fft_info;
@@ -135,4 +141,11 @@
   	memset(status,0,sizeof(status));
 }
 
+// SAH_SHMEM is the shared-memory structure
+
+struct SAH_SHMEM {
+    GDATA gdata;
+    REDUCED_ARRAY_DATA rarray_data;
+};
+
 #endif
