--- AKv8/client/Makefile.am.orig	2015-12-03 11:14:07.771291012 -0800
+++ AKv8/client/Makefile.am	2015-12-03 11:22:48.657970505 -0800
@@ -104,19 +104,20 @@
 endif
 endif #Linux
 
-## if ENABLE_GUI
-##     CLIENT_BIN = $(CLIENT_PROG) $(CLIENT_NATIVE) $(DEBUG_NATIVE) $(DEBUG_PROG)
-##     GUI_BIN = seti_graphics
-##     GUI_C_FLAGS = $(CLIENT_C_FLAGS) $(GRAPHICS_CFLAGS)
-##     GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
-##     CLIENT_LD_FLAGS += $(LDSTATIC)
-##     CLIENT_LD_ADD += -L$(BOINCDIR)/api -L$(BOINCDIR)/lib $(BOINC_LIBS) $(AM_LIBS)
-##     SAH_GRX_SOURCES = sah_gfx_main.cpp
-##else
+if ENABLE_GUI
+    CLIENT_BIN = $(CLIENT_PROG) $(CLIENT_NATIVE) $(DEBUG_NATIVE) $(DEBUG_PROG)
+    GUI_BIN = seti_graphics
+    GUI_C_FLAGS = $(CLIENT_C_FLAGS) $(GRAPHICS_CFLAGS)
+    GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
+    CLIENT_LD_FLAGS += $(LDSTATIC)
+    CLIENT_LD_ADD += -L$(BOINCDIR)/api -L$(BOINCDIR)/lib $(BOINC_LIBS) $(AM_LIBS)
+    SAH_GRX_SOURCES = sah_gfx_main.cpp
+else
 #no graphics
-##     SAH_GRX_SOURCES =
-
-CLIENT_BIN = $(CLIENT_PROG) $(DEBUG_PROG)
+    SAH_GRX_SOURCES =
+    CLIENT_LD_ADD += $(AM_LIBS)
+    CLIENT_BIN = $(CLIENT_PROG) $(DEBUG_PROG)
+endif
 
 all: client-bin
 
@@ -179,7 +180,8 @@
     ../db/xml_util.h \
     ../../src/GPU_lock.cpp \
     ../../src/GPU_lock.h   \
-    version.h
+    version.h \
+    $(SAH_GRX_SOURCES)
 
 if OPENCL
 seti_boinc_opencl_SOURCES = \
@@ -213,31 +215,31 @@
 endif
 
 
-## if ENABLE_GUI
-## noinst_PROGRAMS += seti_graphics
+if ENABLE_GUI
+noinst_PROGRAMS += seti_graphics
 
-## seti_graphics_SOURCES = \
-##    timecvt.cpp \
-##    sah_gfx.cpp \
-##    sah_gfx_base.cpp \
-##    graphics_main.cpp \
-##    sah_version.cpp
+seti_graphics_SOURCES = \
+    timecvt.cpp \
+    sah_gfx.cpp \
+    sah_gfx_base.cpp \
+    graphics_main.cpp \
+    sah_version.cpp
 
-## seti_graphics_CFLAGS = $(GUI_C_FLAGS)
-## seti_graphics_CXXFLAGS = $(GUI_C_FLAGS)
-## seti_graphics_LDFLAGS = $(GUI_LD_FLAGS)
+seti_graphics_CFLAGS = $(GUI_C_FLAGS)
+seti_graphics_CXXFLAGS = $(GUI_C_FLAGS)
+seti_graphics_LDFLAGS = $(GUI_LD_FLAGS)
 
 
 
-## seti_graphics_LDADD = $(GUI_LD_ADD) $(BOINC_LIBS)
+seti_graphics_LDADD = $(GUI_LD_ADD) $(BOINC_LIBS)
 
-## $(DEBUG_NATIVE): $(GUI_BUILD)
-##	@RM@ -f $(DEBUG_NATIVE)
-##	@LN@ sah_graphics $(DEBUG_NATIVE)
+$(DEBUG_NATIVE): $(GUI_BUILD)
+	@RM@ -f $(DEBUG_NATIVE)
+	@LN@ sah_graphics $(DEBUG_NATIVE)
 
-## $(CLIENT_NATIVE): $(DEBUG_NATIVE)
-##	@RM@ -f $(CLIENT_NATIVE)
-##	@CP@ $(DEBUG_NATIVE) $(CLIENT_NATIVE)
+$(CLIENT_NATIVE): $(DEBUG_NATIVE)
+	@RM@ -f $(CLIENT_NATIVE)
+	@CP@ $(DEBUG_NATIVE) $(CLIENT_NATIVE)
 
-## endif
+endif
 
