--- AKv8/client/sah_gfx_main.h.orig	2015-12-03 13:26:27.272623849 -0800
+++ AKv8/client/sah_gfx_main.h	2015-12-03 13:48:22.802445042 -0800
@@ -31,13 +31,14 @@
 
 extern SAH_SHMEM* sah_shmem;
 extern REDUCED_ARRAY_GEN rarray;
+//extern GDATA* sah_graphics;
 extern GDATA* sah_graphics;
 extern bool nographics_flag;
 
-inline bool nographics() {
-    if (nographics_flag) return true;
-    if (!sah_graphics) return true;
-    if (!sah_graphics->countdown) return true;
-    return false;
-}
+inline bool nographics() {
+    if (nographics_flag) return true;
+    if (!sah_graphics) return true;
+    if (!sah_graphics->countdown) return true;
+    return false;
+}
 extern void sah_graphics_init(APP_INIT_DATA&);
