diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-gpu-7.08.orig/AKv8/client/analyzeFuncs.cpp setiathome-gpu-7.08/AKv8/client/analyzeFuncs.cpp
--- setiathome-gpu-7.08.orig/AKv8/client/analyzeFuncs.cpp	2015-12-18 13:18:24.811093019 -0800
+++ setiathome-gpu-7.08/AKv8/client/analyzeFuncs.cpp	2015-12-18 13:30:26.757400808 -0800
@@ -2287,7 +2287,7 @@
   FftOrder=0;
   FftLen=1;
 #ifdef BOINC_APP_GRAPHICS
-  if (!nographics()) strcpy(sah_graphics->status, "Generating FFT Coefficients");
+  if (sah_graphics) strcpy(sah_graphics->status, "Generating FFT Coefficients");
 #endif
 
 
@@ -2526,7 +2526,7 @@
 
 #ifdef DO_SMOOTH
 #ifdef BOINC_APP_GRAPHICS
-  if (!nographics()) strcpy(sah_graphics->status, "Doing Baseline Smoothing");
+  if (sah_graphics) strcpy(sah_graphics->status, "Doing Baseline Smoothing");
 #endif
   retval = v_BaseLineSmooth(
              DataIn, NumDataPoints, swi.analysis_cfg.bsmooth_boxcar_length,
@@ -5301,7 +5301,7 @@
 
     if(dochirp = (chirprateind != last_chirp_ind)) {
 #ifdef BOINC_APP_GRAPHICS
-      if (!nographics()) strcpy(sah_graphics->status, "Chirping data");
+      if (sah_graphics) strcpy(sah_graphics->status, "Chirping data");
 #endif
 
 #ifndef USE_VECTORIZED_CHIRP
@@ -5344,7 +5344,7 @@
     }
 
 #ifdef BOINC_APP_GRAPHICS
-    if (!nographics()) {
+    if (sah_graphics) {
       sah_graphics->fft_info.chirp_rate = chirprate;
       sah_graphics->fft_info.fft_len = fftlen;
       strcpy(sah_graphics->status, "Computing Fast Fourier Transform");
@@ -5366,13 +5366,9 @@
     dotranspose = TripTable[fftlen%37] || ChirpFftPairs[icfft+stripLength].GaussFit || ChirpFftPairs[icfft+stripLength].PulseFind;
 
 #ifdef BOINC_APP_GRAPHICS
-    GRAPH_BUFFER* gbp=0;
-    if (!nographics()) {
-      gbp = sah_graphics->get_generate_buffer(true);
-      if (gbp) {
-        gbp->rarray.init_data(fftlen, NumFfts);
-      }
-    }
+        if (sah_graphics) {
+            rarray.init_data(fftlen, NumFfts);
+        }
 #endif
 
     ifft = 0;
@@ -5542,10 +5538,8 @@
         progress = std::min(progress,1.0);
 #ifdef BOINC_APP_GRAPHICS
         if (!nographics()) {
-        	if (gbp) {
-        	  gbp->rarray.add_source_row(PowerSpectrum+fftlen*ifft);
-        	}
-        	sah_graphics->local_progress = (((float)ifft+1)/NumFfts);
+            rarray.add_source_row(PowerSpectrum+fftlen*ifft);
+            sah_graphics->local_progress = (((float)ifft+1)/NumFfts);
         }
 #endif
         remaining=1.0-(double)(icfft+1)/num_cfft;
@@ -5638,9 +5632,7 @@
   		  progress = std::min(progress,1.0);
 #ifdef BOINC_APP_GRAPHICS
   		  if (!nographics()) {
-  			if (gbp) {
-  			  gbp->rarray.add_source_row(PowerSpectrum+fftlen*ifft);
-  			}
+  			rarray.add_source_row(PowerSpectrum+fftlen*ifft);
   			sah_graphics->local_progress = (((float)ifft+1)/NumFfts);
   		  }
 #endif
@@ -5757,9 +5749,7 @@
 //	  progress = std::min(progress,1.0);
 //  #ifdef BOINC_APP_GRAPHICS
 //        if (!nographics()) {
-//          if (gbp) {
-//            gbp->rarray.add_source_row(PowerSpectrum+fftlen*ifft);
-//          }
+//          rarray.add_source_row(PowerSpectrum+fftlen*ifft);
 //          sah_graphics->local_progress = (((float)ifft+1)/NumFfts);
 //        }
 //  #endif
@@ -5874,7 +5864,7 @@
   }
   retval=0;//R: cause outfile.printf can return non-zero and worker will treat it as error.
 #ifdef BOINC_APP_GRAPHICS
-  if (!nographics()) strcpy(sah_graphics->status, "Work unit done");
+  if (sah_graphics) strcpy(sah_graphics->status, "Work unit done");
 #endif
 
   final_report(); // flop and signal counts to stderr
