--- AKv8/client/sah_gfx.cpp.orig	2015-12-04 19:37:33.306830180 -0800
+++ AKv8/client/sah_gfx.cpp	2015-12-04 21:19:01.447854576 -0800
@@ -134,7 +134,6 @@
 bool mouse_down = false;
 int mouse_x, mouse_y;
 
-extern REDUCED_ARRAY_RENDER rarray;
 SAH_GRAPHICS *sah_graphics_dl;
 
 SAH_GRAPHICS::SAH_GRAPHICS() : track_mem<SAH_GRAPHICS>("SAH_GRAPHICS"), SAH_GRAPHICS_BASE() {}
@@ -725,11 +724,11 @@
 
     pos[0] = -3.50; pos[1] = -2.50; pos[2] = -2.00;
     size[0] = 7.00; size[1] = 2.00; size[2] = 4.00;
-    rarray.init_display(
+    buf1.rarray.init_display(
         graph_style, pos, size, start_hue, hue_change, graph_alpha,
         "Frequency (Hz)","Power","Time (sec)"
     );
-    rarray.init_display(
+    buf2.rarray.init_display(
         graph_style, pos, size, start_hue, hue_change, graph_alpha,
         "Frequency (Hz)","Power","Time (sec)"
     );
