--- setiathome-gpu-8.00.orig/AKv8/client/autocorr.cpp	2016-01-02 01:58:49.082954715 -0800
+++ setiathome-gpu-8.00/AKv8/client/autocorr.cpp	2016-01-02 02:26:39.536864819 -0800
@@ -274,15 +274,16 @@
     if (ai.score > best_autocorr->score || best_autocorr->a.fft_len == 0) {
       *best_autocorr     = ai;
 
+#ifdef BOINC_APP_GRAPHICS
+      if (!nographics()) sah_graphics->ai.copy(&ai);
+#endif
+
       if(verbose>=2  && verbose <6){
         fprintf(stderr,"Best autocorr updated:");
         fprintf(stderr,"score=%.4g, peak_power=%.4g, bin=%d, fft_ind=%d, icfft=%d\n",
                 ai.score,ai.a.peak_power,ai.bin,ai.fft_ind,analysis_state.icfft);
       }
 
-#ifdef BOINC_APP_GRAPHICS
-      if (!nographics()) sah_graphics->ai.copy(&ai);
-#endif
     }
 
     // Report a signal if it excceeds threshold.
@@ -375,6 +376,9 @@
     // if best_autocorr.s.fft_len == 0, there is not yet a first autocorr
     if (was_best) {
       *best_autocorr 			= ai;
+#ifdef BOINC_APP_GRAPHICS
+      if (!nographics()) sah_graphics->ai.copy(&ai);
+#endif      
 	  if(verbose>=2 && verbose <6){
 		fprintf(stderr,"Best autocorr updated:");
 		fprintf(stderr,"score=%.4g, peak_power=%.4g, bin=%d, fft_ind=%d, icfft=%d\n",
@@ -416,6 +420,9 @@
 		time_to_ra_dec(ai.a.time, &ai.a.ra, &ai.a.decl);
 		if(best && best_autocorr->a.peak_power<ai.a.peak_power){//R: update only if really new one
 		    *best_autocorr 			= ai;
+#ifdef BOINC_APP_GRAPHICS
+                    if (!nographics()) sah_graphics->ai.copy(&ai);
+#endif        
 			if(verbose>=2  && verbose <6){
 				fprintf(stderr,"LogAutocorr best autocorr updated:");
 				fprintf(stderr,"score=%.4g, peak_power=%.4g, bin=%d, fft_ind=%d, icfft=%d\n",
@@ -462,6 +469,9 @@
     // if best_autocorr.s.fft_len == 0, there is not yet a first autocorr
     if (was_best) {
       *best_autocorr 			= ai;
+#ifdef BOINC_APP_GRAPHICS
+      if (!nographics()) sah_graphics->ai.copy(&ai);
+#endif      
 	  if(verbose>=2  && verbose <6){
 		fprintf(stderr,"Best autocorr updated:");
 		fprintf(stderr,"score=%.4g, peak_power=%.4g, bin=%d, fft_ind=%d, icfft=%d\n",
