diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-gpu-7.08.orig/AKv8/client/analyzePoT.cpp setiathome-gpu-7.08/AKv8/client/analyzePoT.cpp
--- setiathome-gpu-7.08.orig/AKv8/client/analyzePoT.cpp	2015-12-18 14:06:50.397013484 -0800
+++ setiathome-gpu-7.08/AKv8/client/analyzePoT.cpp	2015-12-18 19:52:34.842682927 -0800
@@ -1151,7 +1151,7 @@
 #endif
 #else
 #ifdef BOINC_APP_GRAPHICS
-    if (!nographics()) strcpy(sah_graphics->status, "Searching for Gaussians");
+    if (sah_graphics) strcpy(sah_graphics->status, "Searching for Gaussians");
 #endif
 
     // If we are back from being interrupted in the middle of gaussian PoT
@@ -1169,7 +1169,7 @@
     // Initial display of local Progress / CPU time. Assumes that
     // we start ThisPoT at 1 each time in!
 #ifdef BOINC_APP_GRAPHICS
-    if (!nographics()) sah_graphics->local_progress = ((float)ThisPoT-1)/(FftLength-1);
+    if (sah_graphics) sah_graphics->local_progress = ((float)ThisPoT-1)/(FftLength-1);
 #endif
     // loop through frequencies
     for(; ThisPoT < FftLength; ThisPoT++) {
@@ -1230,7 +1230,7 @@
   // Look for pulses -------------------------------------------------------
   if(!SkipPulse || !SkipTriplet) {
 #ifdef BOINC_APP_GRAPHICS
-    if (!nographics()) {
+    if (sah_graphics) {
       strcpy(sah_graphics->status, "Searching for Pulses / Triplets");
     // init local progress for pulse search
       sah_graphics->local_progress = 0;
@@ -1249,7 +1249,7 @@
 
 #ifdef BOINC_APP_GRAPHICS
     // Inital display of Local Progress
-    if (!nographics()) {
+    if (sah_graphics) {
        sah_graphics->local_progress = (((ThisPoT-1) * NumPulsePoTs) +
            PulsePoTNum)                   *
           ProgressPerPulsePoT;
@@ -3777,7 +3777,7 @@
 	// At the end of each pulse PoT we update progress.  Progress
 	// will thus get updted several times per frequency bin.
 #ifdef BOINC_APP_GRAPHICS
-        if (!nographics()) {
+        if (sah_graphics) {
 	  sah_graphics->local_progress = (((ThisPoT-1) * NumPulsePoTs) +
                                    PulsePoTNum)                   *
                                   ProgressPerPulsePoT;
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-gpu-7.08.orig/AKv8/client/sah_gfx.cpp setiathome-gpu-7.08/AKv8/client/sah_gfx.cpp
--- setiathome-gpu-7.08.orig/AKv8/client/sah_gfx.cpp	2015-12-18 14:06:50.396013446 -0800
+++ setiathome-gpu-7.08/AKv8/client/sah_gfx.cpp	2015-12-18 14:51:21.763018230 -0800
@@ -202,14 +202,17 @@
 	int s4_id=3;
 
 	if (gdata->ready) {
-          if (gdata->wu.wu) {
-	    s4_id=gdata->wu.wu->group_info->receiver_cfg->s4_id;
-	    if (s4_id==0) {
-	      if (strncmp("ao430",gdata->wu.wu->group_info->receiver_cfg->name,5)) {
-	        locations[0]=gdata->wu.wu->group_info->receiver_cfg->name;
-	      }
-	    }
-	  } 
+          s4_id = gdata->wu.s4_id;
+          // starting after the 3rd receiver_cfg entry, the long form
+          // name appears in receiver_cfg->name
+          if (s4_id>2) {
+            if (strlen(gdata->wu.receiver_name)==0) {
+              locations[0]="SETI@home Multi-Beam";
+            } else {
+              locations[0]=gdata->wu.receiver_name;
+            }
+            s4_id=0;
+          }
 	  ra_string(gdata->wu.start_ra, ra_buf);
 	  dec_string(gdata->wu.start_dec, dec_buf);
 	  sprintf(buf, "From: %s, %s\n"
@@ -323,6 +326,7 @@
     //
 	pos[1] = 0.3;
     boinc_wu_cpu_time(cur_cpu_time);
+        cur_cpu_time=gdata->cpu_time;
 	ndays_to_string(cur_cpu_time/SECONDS_PER_DAY, 0, time_buf );
 	sprintf(buf, "Overall %.3f%% done    CPU time: %s", floor(gdata->progress*100000)/1000, time_buf);
 	draw_text_line(pos, 0.1, 1.0, buf);
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-gpu-7.08.orig/AKv8/client/seti.cpp setiathome-gpu-7.08/AKv8/client/seti.cpp
--- setiathome-gpu-7.08.orig/AKv8/client/seti.cpp	2015-12-18 14:06:50.398013521 -0800
+++ setiathome-gpu-7.08/AKv8/client/seti.cpp	2015-12-18 20:20:14.835990011 -0800
@@ -1206,9 +1206,15 @@
   state.data = data;
 
 #ifdef BOINC_APP_GRAPHICS
-  if (!nographics()) {
-    sah_graphics->wu = swi;
-    sah_graphics->ready = true;
+  if (sah_graphics) {
+      strlcpy(sah_graphics->wu.receiver_name,swi.receiver_cfg.name,255);
+      sah_graphics->wu.s4_id = swi.receiver_cfg.s4_id;
+      sah_graphics->wu.time_recorded = swi.time_recorded;
+      sah_graphics->wu.subband_base = swi.subband_base;
+      sah_graphics->wu.start_ra = swi.start_ra;
+      sah_graphics->wu.start_dec = swi.start_dec;
+      sah_graphics->wu.subband_sample_rate = swi.subband_sample_rate;
+      sah_graphics->ready = true;
   }
 #endif
 
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-gpu-7.08.orig/AKv8/client/worker.cpp setiathome-gpu-7.08/AKv8/client/worker.cpp
--- setiathome-gpu-7.08.orig/AKv8/client/worker.cpp	2015-12-18 14:06:50.399013559 -0800
+++ setiathome-gpu-7.08/AKv8/client/worker.cpp	2015-12-18 20:03:22.716995200 -0800
@@ -121,7 +121,7 @@
     f = boinc_fopen(path.c_str(), "rb");
     if (f) {
 #ifdef BOINC_APP_GRAPHICS
-    if (!nographics())  sprintf(sah_graphics->status, "Scanning data file\n");
+    if (sah_graphics)  sprintf(sah_graphics->status, "Scanning data file\n");
 #endif
         retval = seti_parse_wu(f, analysis_state);
         fclose(f);
@@ -135,7 +135,7 @@
     retval = seti_init_state();
     if (retval) SETIERROR(retval,"from seti_init_state() in read_wu_state()");
 #ifdef BOINC_APP_GRAPHICS
-    if (!nographics()) sprintf(sah_graphics->status, "Scanning state file.\n");
+    if (sah_graphics) sprintf(sah_graphics->status, "Scanning state file.\n");
 #endif
     if (boinc_file_exists(STATE_FILENAME)) {
       retval = parse_state_file(analysis_state);
