diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AKv8/client/analyzeFuncs.cpp setiathome-7.08/AKv8/client/analyzeFuncs.cpp
--- setiathome-7.08.orig/AKv8/client/analyzeFuncs.cpp	2015-12-07 08:59:53.593452917 -0800
+++ setiathome-7.08/AKv8/client/analyzeFuncs.cpp	2015-12-07 09:10:03.885373692 -0800
@@ -5772,9 +5772,7 @@
 
 #ifdef BOINC_APP_GRAPHICS
     if (!nographics()) {
-      if (gbp) {
-        sah_graphics->generate_done(gbp);
-      }
+      memcpy(&sah_shmem->rarray_data, &rarray, sizeof(REDUCED_ARRAY_DATA));
     }
 #endif
     fraction_done(progress,remaining);
