--- AKv8/client/worker.cpp.orig	2015-12-03 20:11:48.944981170 -0800
+++ AKv8/client/worker.cpp	2015-12-03 20:14:23.853969936 -0800
@@ -76,7 +76,6 @@
 #define free_sah_graphics f_free_sah_graphics
 #endif
 #endif
-
 extern int verbose;
 
 // this gets called first on all platforms
@@ -177,7 +176,7 @@
 //fprintf(stderr,"after common_init\n");
 		if (retval) SETIERROR(retval,"from common_init() in worker()");
 
-#ifdef BOINC_APP_GRAPHICS
+#ifdef BOINC_APP_GRAPHICS_OLD
     if (!sah_graphics) {
 #ifdef DYNAMIC_GRAPHICS
       int waited=0;
@@ -207,7 +206,7 @@
     retval = read_wu_state();
 		if (retval) SETIERROR(retval,"from read_wu_state() in worker()");
 
-#ifdef BOINC_APP_GRAPHICS
+#ifdef BOINC_APP_GRAPHICS_OLD
     if (!sah_graphics) {
 #ifdef DYNAMIC_GRAPHICS
       int waited=0;
