diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-gpu-7.08.orig/AKv8/client/sah_gfx_base.cpp setiathome-gpu-7.08/AKv8/client/sah_gfx_base.cpp
--- setiathome-gpu-7.08.orig/AKv8/client/sah_gfx_base.cpp	2015-12-16 12:13:28.397092106 -0800
+++ setiathome-gpu-7.08/AKv8/client/sah_gfx_base.cpp	2015-12-16 12:29:52.235179847 -0800
@@ -37,6 +37,32 @@
 #include <glut/glut.h>
 #endif
 
+#ifdef HAVE_GL_H
+#include "gl.h"
+#elif defined(HAVE_GL_GL_H)
+#include <GL/gl.h>
+#elif defined(HAVE_OPENGL_GL_H)
+#include <OpenGL/gl.h>
+#endif
+
+#ifdef HAVE_GLU_H
+#include "glu.h"
+#elif defined(HAVE_GL_GLU_H)
+#include <GL/glu.h>
+#elif defined(HAVE_OPENGL_GLU_H)
+#include <OpenGL/glu.h>
+#endif
+
+#ifdef HAVE_GLUT_H
+#include "glut.h"
+#elif defined(HAVE_GL_GLUT_H)
+#include <GL/glut.h>
+#elif defined(HAVE_OPENGL_GLUT_H)
+#include <OpenGL/glut.h>
+#elif defined(HAVE_GLUT_GLUT_H)
+#include <GLUT/glut.h>
+#endif
+
 #ifndef _WIN32
 #include <cmath>
 #include <ctime>
Only in setiathome-gpu-7.08/AKv8/client: seti_graphics-sah_gfx_base.o
