--- a/clang/lib/Driver/ToolChains/Linux.cpp.orig	2025-06-05 10:11:31.000000000 -0700
+++ b/clang/lib/Driver/ToolChains/Linux.cpp	2025-11-12 17:28:00.835359873 -0800
@@ -977,6 +977,8 @@ void Linux::AddHIPRuntimeLibArgs(const A
         {"-rpath", Args.MakeArgString(RocmInstallation->getLibPath())});
 
   CmdArgs.push_back("-lamdhip64");
+  if (getenv("HIP_PLATFORM") && strcmp(getenv("HIP_PLATFORM"), "nvidia") != 0)
+    CmdArgs.push_back("-lamd_comgr");
 }
 
 void Linux::AddIAMCUIncludeArgs(const ArgList &DriverArgs,
