--- a/post-remove.sh.orig	2019-07-05 17:58:11.000000000 -0700
+++ b/post-remove.sh	2019-10-16 23:45:01.999210481 -0700
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- a/pre-build.sh.orig	2019-07-05 17:58:11.000000000 -0700
+++ b/pre-build.sh	2019-10-16 23:45:00.079136093 -0700
@@ -41,6 +41,6 @@ find scheduler -name '*.c' -exec grep EX
     | awk -F'[()]' '{print "#define "$2" amd"$2" //"$0}'\
     >> include/rename_symbol.h
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-19.30-838629/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-19.30-838629/firmware/amdgpu $FW_DIR
