--- a/amd/amdgpu/dce_v11_0.c.orig	2019-10-16 23:30:05.182439199 -0700
+++ b/amd/amdgpu/dce_v11_0.c	2019-10-16 23:31:44.589316885 -0700
@@ -1727,10 +1727,12 @@ static void dce_v11_0_afmt_setmode(struc
 	dce_v11_0_audio_write_sad_regs(encoder);
 	dce_v11_0_audio_write_latency_fields(encoder, mode);
 
-#if DRM_VERSION_CODE < DRM_VERSION(4, 14, 0) && \
+#if DRM_VERSION_CODE >= DRM_VERSION(5, 1, 0)
+	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, connector, mode);
+#elif DRM_VERSION_CODE < DRM_VERSION(4, 14, 0) && \
 	!defined(OS_NAME_SUSE_15) && !defined(OS_NAME_SUSE_15_1)
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode);
-#else
+#elif DRM_VERSION_CODE < DRM_VERSION(5, 1, 0)
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode, false);
 #endif
 	if (err < 0) {
