--- a/post-remove.sh.orig	2019-11-23 16:21:44.243026694 -0800
+++ b/post-remove.sh	2019-11-23 16:19:12.061078524 -0800
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- a/pre-build.sh.orig	2019-07-30 14:46:05.000000000 -0700
+++ b/pre-build.sh	2019-11-23 16:21:31.427525797 -0800
@@ -41,6 +41,6 @@ find scheduler -name '*.c' -exec grep EX
     | awk -F'[()]' '{print "#define "$2" amd"$2" //"$0}'\
     >> include/rename_symbol.h
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-19.30-855429/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-19.30-855429/firmware/amdgpu $FW_DIR
