--- a/Makefile.orig	2019-10-16 23:36:37.487672187 -0700
+++ b/Makefile	2019-10-16 23:38:12.566353101 -0700
@@ -189,6 +189,7 @@ export CONFIG_DRM_AMD_DC=y
 export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_DRM_AMD_DC_DCN1_01=y
 export CONFIG_DRM_AMD_DC_DCN2_0=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DBUILD_AS_DKMS
@@ -199,5 +200,6 @@ subdir-ccflags-y += -DCONFIG_DRM_AMD_DC
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_0
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_01
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN2_0
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 obj-m += scheduler/ amd/amdgpu/ ttm/ amd/amdkcl/
--- a/amd/dkms/Makefile.orig	2019-10-16 23:36:37.487672187 -0700
+++ b/amd/dkms/Makefile	2019-10-16 23:38:44.021570998 -0700
@@ -189,6 +189,7 @@ export CONFIG_DRM_AMD_DC=y
 export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_DRM_AMD_DC_DCN1_01=y
 export CONFIG_DRM_AMD_DC_DCN2_0=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DBUILD_AS_DKMS
@@ -199,5 +200,6 @@ subdir-ccflags-y += -DCONFIG_DRM_AMD_DC
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_0
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_01
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN2_0
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 obj-m += scheduler/ amd/amdgpu/ ttm/ amd/amdkcl/
