--- a/post-remove.sh.orig	2020-04-08 20:50:08.000000000 -0700
+++ b/post-remove.sh	2020-04-23 17:05:30.789020697 -0700
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- a/pre-build.sh.orig	2020-04-08 20:50:08.000000000 -0700
+++ b/pre-build.sh	2020-04-23 17:05:28.698940461 -0700
@@ -56,6 +56,6 @@ done
 export KERNELVER
 ./configure
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-5.4.7.53-1048554/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-5.4.7.53-1048554/firmware/amdgpu $FW_DIR
