--- a/amd/amdgpu/amdgpu_drv.c.orig	2019-10-16 23:08:45.751558908 -0700
+++ b/amd/amdgpu/amdgpu_drv.c	2019-10-16 23:10:17.600137822 -0700
@@ -1453,7 +1453,11 @@ static struct drm_driver kms_driver = {
 #if DRM_VERSION_CODE >= DRM_VERSION(4, 20, 0)
 		DRIVER_ATOMIC |
 #endif
+#if DRM_VERSION_CODE < DRM_VERSION(5, 1, 0)
 		DRIVER_HAVE_IRQ | DRIVER_IRQ_SHARED | DRIVER_GEM |
+#else
+		DRIVER_GEM |
+#endif
 #if DRM_VERSION_CODE >= DRM_VERSION(4, 13, 0)
 		DRIVER_SYNCOBJ |
 #endif
