--- a/post-remove.sh.orig	2019-10-03 17:18:36.479015447 -0700
+++ b/post-remove.sh	2019-10-04 14:29:10.790397313 -0700
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware/$1"
-rm -rf $FW_DIR/radeon
-rm -rf $FW_DIR/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware/$1"
+#rm -rf $FW_DIR/radeon
+#rm -rf $FW_DIR/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- a/pre-build.sh.orig	2019-10-03 17:18:36.480015485 -0700
+++ b/pre-build.sh	2019-10-04 14:29:12.454462754 -0700
@@ -44,7 +44,7 @@ find scheduler -name '*.c' -exec grep EX
 export KERNELVER
 ./configure
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-2.8-13/firmware/radeon $FW_DIR
-cp -ar /usr/src/amdgpu-2.8-13/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-2.8-13/firmware/radeon $FW_DIR
+#cp -ar /usr/src/amdgpu-2.8-13/firmware/amdgpu $FW_DIR
