--- a/amd/amdgpu/amdgpu_connectors.c.orig	2019-10-03 02:40:23.350999427 -0700
+++ b/amd/amdgpu/amdgpu_connectors.c	2019-10-03 03:23:10.577003913 -0700
@@ -27,6 +27,9 @@
 #include <drm/drm_edid.h>
 #include <drm/drm_crtc_helper.h>
 #include <drm/drm_fb_helper.h>
+#if DRM_VERSION_CODE >= DRM_VERSION(5, 1, 0)
+#include <drm/drm_probe_helper.h>
+#endif
 #include <drm/amdgpu_drm.h>
 #include "amdgpu.h"
 #include "atom.h"
