--- a/m4/kernel.m4.orig	2019-10-03 05:15:44.162224694 -0700
+++ b/m4/kernel.m4	2019-10-03 13:58:36.146107443 -0700
@@ -340,7 +340,7 @@ AC_DEFUN([AC_KERNEL_COMPILE_IFELSE], [
 	test "x$enable_linux_builtin" = xyes && modpost_flag='modpost=true' # fake modpost stage
 	test "x$enable_linux_builtin" = xyes && kbuild_src_flag='KBUILD_SRC=' # override KBUILD_SRC
 	cp conftest.c conftest.h build
-	AC_TRY_COMMAND(make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-Werror -Wno-error=uninitialized -Wno-error=unused-variable" M=$PWD/build $modpost_flag $kbuild_src_flag >/dev/null)
+	AC_TRY_COMMAND(make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-DCONFIG_MMU_NOTIFIER -Werror -Wno-error=uninitialized -Wno-error=unused-variable" M=$PWD/build $modpost_flag $kbuild_src_flag >/dev/null)
 	AS_IF(
 		[AC_TRY_COMMAND([$3])],
 		[$4],
--- a/amd/dkms/m4/kernel.m4.orig	2019-10-03 05:15:44.162224694 -0700
+++ b/amd/dkms/m4/kernel.m4	2019-10-03 13:58:36.146107443 -0700
@@ -340,7 +340,7 @@ AC_DEFUN([AC_KERNEL_COMPILE_IFELSE], [
 	test "x$enable_linux_builtin" = xyes && modpost_flag='modpost=true' # fake modpost stage
 	test "x$enable_linux_builtin" = xyes && kbuild_src_flag='KBUILD_SRC=' # override KBUILD_SRC
 	cp conftest.c conftest.h build
-	AC_TRY_COMMAND(make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-Werror -Wno-error=uninitialized -Wno-error=unused-variable" M=$PWD/build $modpost_flag $kbuild_src_flag >/dev/null)
+	AC_TRY_COMMAND(make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-DCONFIG_MMU_NOTIFIER -Werror -Wno-error=uninitialized -Wno-error=unused-variable" M=$PWD/build $modpost_flag $kbuild_src_flag >/dev/null)
 	AS_IF(
 		[AC_TRY_COMMAND([$3])],
 		[$4],
