--- a/amd/dkms/Kbuild.orig	2025-07-25 13:06:35.000000000 -0700
+++ b/amd/dkms/Kbuild	2025-10-14 21:24:49.535689232 -0700
@@ -57,6 +57,9 @@ RHEL_MINOR := $(shell $(call get_rhel_ve
 ifneq (,$(RHEL_MAJOR))
 OS_NAME = "rhel"
 OS_VERSION = "$(RHEL_MAJOR).$(RHEL_MINOR)"
+else ifeq (gentoo,$(shell grep "^ID=" /etc/os-release | cut -f 2 -d "="))
+OS_NAME = "gentoo"
+OS_VERSION = $(shell realpath /etc/portage/make.profile | grep -E -o "[0-9]+\.[0-9]+")
 else ifneq (,$(wildcard /etc/os-release))
 OS_NAME = "$(shell sed -n 's/^ID=\(.*\)/\1/p' /etc/os-release | tr -d '\"')"
 # On CentOS/RHEL, users could have installed a kernel not distributed from RHEL
