--- a/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-10-03 16:12:54.841046590 -0700
+++ b/amd/display/amdgpu_dm/amdgpu_dm.c	2019-10-03 16:36:33.941483885 -0700
@@ -2169,6 +2169,8 @@ static int amdgpu_dm_mode_config_init(st
 				     * integrate modeset lock with private objects" upstreamed.
 				     */
 				    adev->ddev,
+#elif DRM_VERSION_CODE >= DRM_VERSION(5, 1, 0)
+				    adev->ddev,
 #endif
 				    &adev->dm.atomic_obj,
 				    &state->base,
