--- a/amd/amdgpu/amdgpu_device.c.orig	2019-10-03 02:40:23.535006662 -0700
+++ b/amd/amdgpu/amdgpu_device.c	2019-10-03 02:57:02.890316067 -0700
@@ -32,6 +32,9 @@
 #include <drm/drmP.h>
 #include <drm/drm_crtc_helper.h>
 #include <drm/drm_atomic_helper.h>
+#if DRM_VERSION_CODE >= DRM_VERSION(5, 1, 0)
+#include <drm/drm_probe_helper.h>
+#endif
 #include <drm/amdgpu_drm.h>
 #include <linux/vgaarb.h>
 #include <linux/vga_switcheroo.h>
