diff -urp -x '*.orig' -x '*.reg' llvm-3.9.1.src.orig/tools/clang/CMakeLists.txt llvm-3.9.1.src/tools/clang/CMakeLists.txt
--- llvm-3.9.1.src.orig/tools/clang/CMakeLists.txt	2017-01-05 20:41:28.891927814 -0800
+++ llvm-3.9.1.src/tools/clang/CMakeLists.txt	2017-01-05 20:43:38.034918400 -0800
@@ -279,6 +279,8 @@ configure_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/include/clang/Basic/Version.inc.in
   ${CMAKE_CURRENT_BINARY_DIR}/include/clang/Basic/Version.inc)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+
 # Add appropriate flags for GCC
 if (LLVM_COMPILER_IS_GCC_COMPATIBLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-common -Woverloaded-virtual")
diff -urp -x '*.orig' -x '*.reg' llvm-3.9.1.src.orig/tools/clang/lib/Driver/ToolChains.h llvm-3.9.1.src/tools/clang/lib/Driver/ToolChains.h
--- llvm-3.9.1.src.orig/tools/clang/lib/Driver/ToolChains.h	2017-01-05 20:41:28.833925573 -0800
+++ llvm-3.9.1.src/tools/clang/lib/Driver/ToolChains.h	2017-01-05 20:43:38.034918400 -0800
@@ -831,10 +831,10 @@ protected:
   Tool *buildLinker() const override;
 };
 
-class LLVM_LIBRARY_VISIBILITY Linux : public Generic_ELF {
+class Linux : public Generic_ELF {
 public:
   Linux(const Driver &D, const llvm::Triple &Triple,
-        const llvm::opt::ArgList &Args);
+        const llvm::opt::ArgList &Args) __attribute__ ((visibility ("default") ));
 
   bool HasNativeLLVMSupport() const override;
 
