diff -urp music-1.0.0_p9999.orig/cmake/Modules/FindTaglib.cmake music-1.0.0_p9999/cmake/Modules/FindTaglib.cmake
--- music-1.0.0_p9999.orig/cmake/Modules/FindTaglib.cmake	2022-11-30 11:35:30.342646566 -0800
+++ music-1.0.0_p9999/cmake/Modules/FindTaglib.cmake	2022-11-30 11:36:18.860579432 -0800
@@ -11,11 +11,11 @@
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
 if(NOT TAGLIB_MIN_VERSION)
-  set(TAGLIB_MIN_VERSION "1.4")
+  set(TAGLIB_MIN_VERSION "2.0")
 endif(NOT TAGLIB_MIN_VERSION)
 
 if(NOT WIN32)
-    find_program(TAGLIBCONFIG_EXECUTABLE NAMES taglib-config PATHS
+    find_program(TAGLIBCONFIG_EXECUTABLE NAMES taglib2-config PATHS
        ${BIN_INSTALL_DIR}
     )
 endif(NOT WIN32)
diff -urp music-1.0.0_p9999.orig/src/Components/Songs/songmodel.cpp music-1.0.0_p9999/src/Components/Songs/songmodel.cpp
--- music-1.0.0_p9999.orig/src/Components/Songs/songmodel.cpp	2022-11-30 11:35:30.618657562 -0800
+++ music-1.0.0_p9999/src/Components/Songs/songmodel.cpp	2022-11-30 11:37:02.442315480 -0800
@@ -4,8 +4,8 @@
 #include <stdio.h>
 #include <iostream>
 #include <QFileInfo>
-#include <taglib/tag.h>
-#include <taglib/fileref.h>
+#include <taglib2/tag.h>
+#include <taglib2/fileref.h>
 
 SongModel::SongModel(QObject *parent):
     QAbstractListModel(parent)
diff -urp music-1.0.0_p9999.orig/src/Components/Utilities/musicscanner.cpp music-1.0.0_p9999/src/Components/Utilities/musicscanner.cpp
--- music-1.0.0_p9999.orig/src/Components/Utilities/musicscanner.cpp	2022-11-30 11:35:30.622657721 -0800
+++ music-1.0.0_p9999/src/Components/Utilities/musicscanner.cpp	2022-11-30 11:36:18.860579432 -0800
@@ -11,21 +11,20 @@
 #include <QThread>
 #include <QBuffer>
 #include <iostream>
-#include <taglib/mpegfile.h>
-#include <taglib/mpegheader.h>
-#include <taglib/tbytevector.h>
-#include <taglib/attachedpictureframe.h>
-#include <taglib/id3v2frame.h>
-#include <taglib/id3v2header.h>
-#include <taglib/mpegfile.h>
-#include <taglib/flacfile.h>
+#include <taglib2/mpegfile.h>
+#include <taglib2/mpegheader.h>
+#include <taglib2/tbytevector.h>
+#include <taglib2/attachedpictureframe.h>
+#include <taglib2/id3v2frame.h>
+#include <taglib2/id3v2header.h>
+#include <taglib2/mpegfile.h>
+#include <taglib2/flacfile.h>
 #include <glob.h>
-#include <taglib/id3v2tag.h>
+#include <taglib2/id3v2tag.h>
 #include <typeinfo>
 //#include "/usr/local/include/taglib/tpicturemap.h"
-#include <taglib/id3v1tag.h>
-#include <taglib/tstring.h>
-#include <taglib/tstring.h>
+#include <taglib2/id3v1tag.h>
+#include <taglib2/tstring.h>
 #include <filesystem>
 #include <iostream>
 #include <vector>
diff -urp music-1.0.0_p9999.orig/src/Components/Utilities/musicscanner.h music-1.0.0_p9999/src/Components/Utilities/musicscanner.h
--- music-1.0.0_p9999.orig/src/Components/Utilities/musicscanner.h	2022-11-30 11:35:30.622657721 -0800
+++ music-1.0.0_p9999/src/Components/Utilities/musicscanner.h	2022-11-30 11:36:18.860579432 -0800
@@ -7,11 +7,11 @@
 
 #include <QFileSystemWatcher>
 
-#include <taglib/taglib.h>
-#include <taglib/tag.h>
-#include <taglib/fileref.h>
-#include <taglib/taglib_config.h>
-#include <taglib/tpropertymap.h>
+#include <taglib2/taglib.h>
+#include <taglib2/tag.h>
+#include <taglib2/fileref.h>
+#include <taglib2/taglib_config.h>
+#include <taglib2/tpropertymap.h>
 
 #include "../Songs/song.h"
 #include "../Albums/album.h"
diff -urp music-1.0.0_p9999.orig/src/main.cpp music-1.0.0_p9999/src/main.cpp
--- music-1.0.0_p9999.orig/src/main.cpp	2022-11-30 11:35:30.698660750 -0800
+++ music-1.0.0_p9999/src/main.cpp	2022-11-30 11:36:59.098182275 -0800
@@ -43,7 +43,7 @@
 #include "Components/Utilities/musicscanner.h"
 #include "Components/Albums/albumartprovider.h"
 #include <QtQuickControls2>
-#include <taglib/tag.h>
+#include <taglib2/tag.h>
 
 #ifdef QT_STATIC
 #  include <QQmlExtensionPlugin>
