diff -urp music-1.0.0_p9999.orig/fluid/CMakeLists.txt music-1.0.0_p9999/fluid/CMakeLists.txt
--- music-1.0.0_p9999.orig/fluid/CMakeLists.txt	2022-11-30 11:11:15.960602186 -0800
+++ music-1.0.0_p9999/fluid/CMakeLists.txt	2022-11-30 11:12:09.102727090 -0800
@@ -28,7 +28,7 @@ option(FLUID_USE_SYSTEM_LCS "Use system-
 add_feature_info("Fluid::LiriCMakeShared" FLUID_USE_SYSTEM_LCS "Use system-wide LiriCMakeShared")
 option(FLUID_WITH_DOCUMENTATION "Build documentation" ON)
 add_feature_info("Fluid::Documentation" FLUID_WITH_DOCUMENTATION "Build Fluid documentation")
-option(FLUID_WITH_DEMO "Build demo application" ON)
+option(FLUID_WITH_DEMO "Build demo application" OFF)
 add_feature_info("Fluid::Demo" FLUID_WITH_DEMO "Build Fluid demo application")
 option(FLUID_WITH_QML_MODULES "Build QML modules" ON)
 add_feature_info("Fluid::QMLModules" FLUID_WITH_QML_MODULES "Build Fluid QML modules")
