diff -ur bitlbee-3.4.1.orig/Makefile bitlbee-3.4.1/Makefile
--- bitlbee-3.4.1.orig/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/Makefile	2015-07-03 16:23:57.456613193 -0700
@@ -16,7 +16,8 @@
 OUTFILE = bitlbee
 
 # Expansion of variables
-subdirobjs = $(foreach dir,$(subdirs),$(dir)/$(dir).o)
+subdirobjs = lib/lib.o
+subdirsos = protocols.so
 
 all: $(OUTFILE) $(OTR_PI) $(SKYPE_PI) doc systemd
 
@@ -115,7 +116,7 @@
 	rm -f $(DESTDIR)$(ETCDIR)/bitlbee.conf
 	-rmdir $(DESTDIR)$(ETCDIR)
 
-install-plugins: install-plugin-otr install-plugin-skype
+install-plugins: install-plugin-otr install-plugin-skype install-plugin-protocols
 
 install-plugin-otr:
 ifdef OTR_PI
@@ -134,6 +135,11 @@
 	$(MAKE) -C protocols/skype install-doc
 endif
 
+install-plugin-protocols:
+	mkdir -p $(DESTDIR)$(PLUGINDIR)/
+	$(INSTALL) -m 0755 protocols.so $(DESTDIR)$(PLUGINDIR)
+	$(INSTALL) -m 0755 protocols/*_mod.so $(DESTDIR)$(PLUGINDIR)
+
 systemd:
 ifdef SYSTEMDSYSTEMUNITDIR
 	mkdir -p init
@@ -175,8 +181,9 @@
 $(objects): Makefile Makefile.settings config.h
 
 $(OUTFILE): $(objects) $(subdirs)
+	@echo $(subdirobjs)
 	@echo '*' Linking $(OUTFILE)
-	@$(CC) $(objects) $(subdirobjs) -o $(OUTFILE) $(LDFLAGS_BITLBEE) $(LDFLAGS) $(EFLAGS)
+	@$(CC) $(objects) $(subdirobjs) -o $(OUTFILE) $(LDFLAGS_BITLBEE) $(LDFLAGS) $(EFLAGS) -Wl,-rpath=$(PLUGINDIR) $(subdirsos)
 ifndef DEBUG
 	@echo '*' Stripping $(OUTFILE)
 	@-$(STRIP) $(OUTFILE)
Only in bitlbee-3.4.1: Makefile.settings
Only in bitlbee-3.4.1: bitlbee.pc
Only in bitlbee-3.4.1: config.h
Only in bitlbee-3.4.1: dbg
Only in bitlbee-3.4.1/doc/user-guide: help.txt
diff -ur bitlbee-3.4.1.orig/protocols/Makefile bitlbee-3.4.1/protocols/Makefile
--- bitlbee-3.4.1.orig/protocols/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/Makefile	2015-07-03 16:23:57.512615336 -0700
@@ -23,11 +23,11 @@
 subdirobjs = $(PROTOOBJS)
 
 # Expansion of variables
-subdirobjs := $(join $(subdirs),$(addprefix /,$(subdirobjs)))
-LFLAGS += -r
+#subdirobjs := $(join $(subdirs),$(addprefix /,$(subdirobjs)))
+LFLAGS +=
 
 # [SH] Phony targets
-all: protocols.o
+all: protocols.so
 check: all
 lcov: check
 gcov:
@@ -36,7 +36,7 @@
 .PHONY: all clean distclean $(subdirs)
 
 clean: $(subdirs)
-	rm -f *.o $(OUTFILE) core
+	rm -f *.o ../protocols.so $(OUTFILE) core
 
 distclean: clean $(subdirs)
 	rm -rf .depend
@@ -46,14 +46,14 @@
 
 ### MAIN PROGRAM
 
-protocols.o: $(objects) $(subdirs)
+protocols.so: $(objects) $(subdirs)
 	@echo '*' Linking protocols.o
-	@$(LD) $(LFLAGS) $(objects) $(subdirobjs) -o protocols.o
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../protocols.so -rpath=$(PLUGINDIR) $(subdirobjs)
 
 $(objects): ../Makefile.settings Makefile
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
 -include .depend/*.d
diff -ur bitlbee-3.4.1.orig/protocols/jabber/Makefile bitlbee-3.4.1/protocols/jabber/Makefile
--- bitlbee-3.4.1.orig/protocols/jabber/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/jabber/Makefile	2015-07-03 16:23:57.518615565 -0700
@@ -14,10 +14,10 @@
 # [SH] Program variables
 objects = conference.o io.o iq.o jabber.o jabber_util.o message.o presence.o s5bytestream.o sasl.o si.o hipchat.o
 
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: jabber_mod.o
+all: jabber_mod.so
 check: all
 lcov: check
 gcov: 
@@ -26,7 +26,7 @@
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../jabber_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -37,10 +37,10 @@
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-jabber_mod.o: $(objects)
-	@echo '*' Linking jabber_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o jabber_mod.o
+jabber_mod.so: $(objects)
+	@echo '*' Linking jabber_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../jabber_mod.so
 
 -include .depend/*.d
diff -ur bitlbee-3.4.1.orig/protocols/msn/Makefile bitlbee-3.4.1/protocols/msn/Makefile
--- bitlbee-3.4.1.orig/protocols/msn/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/msn/Makefile	2015-07-03 16:23:57.514615412 -0700
@@ -14,10 +14,10 @@
 # [SH] Program variables
 objects = msn.o msn_util.o ns.o soap.o tables.o gw.o
 
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: msn_mod.o
+all: msn_mod.so
 check: all
 lcov: check
 gcov: 
@@ -26,7 +26,7 @@
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../msn_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -37,10 +37,10 @@
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-msn_mod.o: $(objects)
-	@echo '*' Linking msn_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o msn_mod.o
+msn_mod.so: $(objects)
+	@echo '*' Linking msn_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../msn_mod.so
 
 -include .depend/*.d
diff -ur bitlbee-3.4.1.orig/protocols/oscar/Makefile bitlbee-3.4.1/protocols/oscar/Makefile
--- bitlbee-3.4.1.orig/protocols/oscar/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/oscar/Makefile	2015-07-03 16:23:57.515615450 -0700
@@ -15,10 +15,10 @@
 # [SH] Program variables
 objects = admin.o auth.o bos.o buddylist.o chat.o chatnav.o conn.o icq.o im.o info.o misc.o msgcookie.o rxhandlers.o rxqueue.o search.o service.o snac.o ssi.o stats.o tlv.o txqueue.o oscar_util.o oscar.o
 
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: oscar_mod.o
+all: oscar_mod.so
 check: all
 lcov: check
 gcov:
@@ -27,7 +27,7 @@
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../oscar_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -38,10 +38,10 @@
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-oscar_mod.o: $(objects)
-	@echo '*' Linking oscar_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o oscar_mod.o
+oscar_mod.so: $(objects)
+	@echo '*' Linking oscar_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../oscar_mod.so
 
 -include .depend/*.d
diff -ur bitlbee-3.4.1.orig/protocols/purple/Makefile bitlbee-3.4.1/protocols/purple/Makefile
--- bitlbee-3.4.1.orig/protocols/purple/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/purple/Makefile	2015-07-03 16:23:57.512615336 -0700
@@ -15,10 +15,10 @@
 objects = ft.o purple.o
 
 CFLAGS += -Wall $(PURPLE_CFLAGS)
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: purple_mod.o
+all: purple_mod.so
 check: all
 lcov: check
 gcov: 
@@ -27,7 +27,7 @@
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../purple_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -38,10 +38,10 @@
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-purple_mod.o: $(objects)
-	@echo '*' Linking purple_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o purple_mod.o
+purple_mod.so: $(objects)
+	@echo '*' Linking purple_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../purple_mod.so `pkg-config x11 --libs` `pkg-config purple --libs`
 
 -include .depend/*.d
diff -ur bitlbee-3.4.1.orig/protocols/purple/purple.c bitlbee-3.4.1/protocols/purple/purple.c
--- bitlbee-3.4.1.orig/protocols/purple/purple.c	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/purple/purple.c	2015-07-03 16:23:57.513615374 -0700
@@ -1455,7 +1455,7 @@
 	funcs.chat_leave = purple_chat_leave;
 	funcs.chat_join = purple_chat_join;
 	funcs.transfer_request = purple_transfer_request;
-
+  
 	help = g_string_new("BitlBee libpurple module supports the following IM protocols:\n");
 
 	/* Add a protocol entry to BitlBee's structures for every protocol
@@ -1464,6 +1464,17 @@
 		PurplePlugin *prot = prots->data;
 		struct prpl *ret;
 
+		fprintf(stderr, "purple.c-protocols: %s\n", prot->info->id);
+
+		if (g_strcasecmp(prot->info->id, "prpl-bigbrownchunx-skypenet") == 0) {
+			//avoid broken plugin
+			continue;
+		}
+
+		if (g_strcasecmp(prot->info->id, "prpl-bigbrownchunx-skype") == 0) {
+			g_set_application_name("bitlbee");
+		}
+
 		/* If we already have this one (as a native module), don't
 		   add a libpurple duplicate. */
 		if (find_protocol(prot->info->id)) {
diff -ur bitlbee-3.4.1.orig/protocols/twitter/Makefile bitlbee-3.4.1/protocols/twitter/Makefile
--- bitlbee-3.4.1.orig/protocols/twitter/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/twitter/Makefile	2015-07-03 16:23:57.518615565 -0700
@@ -14,10 +14,10 @@
 # [SH] Program variables
 objects = twitter.o twitter_http.o twitter_lib.o
 
-LFLAGS += -r
+LFLAGS += 
 
 # [SH] Phony targets
-all: twitter_mod.o
+all: twitter_mod.so
 check: all
 lcov: check
 gcov: 
@@ -26,7 +26,7 @@
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../twitter_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -37,10 +37,10 @@
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-twitter_mod.o: $(objects)
-	@echo '*' Linking twitter_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o twitter_mod.o
+twitter_mod.so: $(objects)
+	@echo '*' Linking twitter_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../twitter_mod.so
 	
 -include .depend/*.d
diff -ur bitlbee-3.4.1.orig/protocols/yahoo/Makefile bitlbee-3.4.1/protocols/yahoo/Makefile
--- bitlbee-3.4.1.orig/protocols/yahoo/Makefile	2015-06-16 15:30:40.000000000 -0700
+++ bitlbee-3.4.1/protocols/yahoo/Makefile	2015-07-03 16:23:57.513615374 -0700
@@ -15,10 +15,10 @@
 objects = yahoo.o libyahoo2.o yahoo_httplib.o yahoo_util.o
 
 CFLAGS += -DSTDC_HEADERS -DHAVE_STRING_H -DHAVE_STRCHR -DHAVE_MEMCPY -DHAVE_GLIB
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: yahoo_mod.o
+all: yahoo_mod.so
 check: all
 lcov: check
 gcov: 
@@ -27,7 +27,7 @@
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../yahoo_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -38,10 +38,10 @@
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -so $@ -fPIC
 
-yahoo_mod.o: $(objects)
-	@echo '*' Linking yahoo_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o yahoo_mod.o
+yahoo_mod.so: $(objects)
+	@echo '*' Linking yahoo_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../yahoo_mod.so
 
 -include .depend/*.d
Only in bitlbee-3.4.1: skype.so
diff -u bitlbee-3.4.1.orig/configure bitlbee-3.4.1/configure
--- bitlbee-3.4.1.orig/configure	2015-07-03 16:36:05.718467807 -0700
+++ bitlbee-3.4.1/configure	2015-07-03 16:36:45.747998479 -0700
@@ -754,7 +754,7 @@
 PURPLE_CFLAGS += $($PKG_CONFIG purple --cflags)
 EOF
 	protocols=$protocols'purple '
-	protoobjs=$protoobjs'purple_mod.o '
+	protoobjs=$protoobjs'purple_mod.so '
 
 	# only disable these if the user didn't enable them explicitly
 	[ "$msn" = "default-on" ] && msn=0
@@ -778,7 +778,7 @@
 else
 	echo '#define WITH_MSN' >> config.h
 	protocols=$protocols'msn '
-	protoobjs=$protoobjs'msn_mod.o '
+	protoobjs=$protoobjs'msn_mod.so '
 fi
 
 if [ "$jabber" = 0 ]; then
@@ -786,7 +786,7 @@
 else
 	echo '#define WITH_JABBER' >> config.h
 	protocols=$protocols'jabber '
-	protoobjs=$protoobjs'jabber_mod.o '
+	protoobjs=$protoobjs'jabber_mod.so '
 fi
 
 if [ "$oscar" = 0 ]; then
@@ -794,7 +794,7 @@
 else
 	echo '#define WITH_OSCAR' >> config.h
 	protocols=$protocols'oscar '
-	protoobjs=$protoobjs'oscar_mod.o '
+	protoobjs=$protoobjs'oscar_mod.so '
 fi
 
 if [ "$yahoo" = 0 ]; then
@@ -802,7 +802,7 @@
 else
 	echo '#define WITH_YAHOO' >> config.h
 	protocols=$protocols'yahoo '
-	protoobjs=$protoobjs'yahoo_mod.o '
+	protoobjs=$protoobjs'yahoo_mod.so '
 fi
 
 if [ "$twitter" = 0 ]; then
@@ -810,7 +810,7 @@
 else
 	echo '#define WITH_TWITTER' >> config.h
 	protocols=$protocols'twitter '
-	protoobjs=$protoobjs'twitter_mod.o '
+	protoobjs=$protoobjs'twitter_mod.so '
 fi
 
 if [ "$protocols" = "PROTOCOLS = " ]; then
Common subdirectories: bitlbee-3.4.1.orig/doc and bitlbee-3.4.1/doc
Common subdirectories: bitlbee-3.4.1.orig/init and bitlbee-3.4.1/init
Common subdirectories: bitlbee-3.4.1.orig/lib and bitlbee-3.4.1/lib
Common subdirectories: bitlbee-3.4.1.orig/protocols and bitlbee-3.4.1/protocols
Common subdirectories: bitlbee-3.4.1.orig/tests and bitlbee-3.4.1/tests
Common subdirectories: bitlbee-3.4.1.orig/utils and bitlbee-3.4.1/utils
