diff -urp bitlbee-3.5.1.orig/configure bitlbee-3.5.1/configure
--- bitlbee-3.5.1.orig/configure	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/configure	2017-02-17 08:36:09.020429881 -0800
@@ -808,7 +808,7 @@ EFLAGS += $($PKG_CONFIG purple --libs)
 PURPLE_CFLAGS += $($PKG_CONFIG purple --cflags)
 EOF
 	protocols=$protocols'purple '
-	protoobjs=$protoobjs'purple_mod.o '
+	protoobjs=$protoobjs'purple_mod.so '
 
 	# only disable these if the user didn't enable them explicitly
 	[ "$msn" = "default-on" ] && msn=0
@@ -839,7 +839,7 @@ if [ "$msn" = 0 ]; then
 else
 	echo '#define WITH_MSN' >> config.h
 	protocols=$protocols'msn '
-	protoobjs=$protoobjs'msn_mod.o '
+	protoobjs=$protoobjs'msn_mod.so '
 fi
 
 if [ "$jabber" = 0 ]; then
@@ -847,7 +847,7 @@ if [ "$jabber" = 0 ]; then
 else
 	echo '#define WITH_JABBER' >> config.h
 	protocols=$protocols'jabber '
-	protoobjs=$protoobjs'jabber_mod.o '
+	protoobjs=$protoobjs'jabber_mod.so '
 fi
 
 if [ "$oscar" = 0 ]; then
@@ -855,7 +855,7 @@ if [ "$oscar" = 0 ]; then
 else
 	echo '#define WITH_OSCAR' >> config.h
 	protocols=$protocols'oscar '
-	protoobjs=$protoobjs'oscar_mod.o '
+	protoobjs=$protoobjs'oscar_mod.so '
 fi
 
 if [ "$twitter" = 0 ]; then
@@ -863,7 +863,7 @@ if [ "$twitter" = 0 ]; then
 else
 	echo '#define WITH_TWITTER' >> config.h
 	protocols=$protocols'twitter '
-	protoobjs=$protoobjs'twitter_mod.o '
+	protoobjs=$protoobjs'twitter_mod.so '
 fi
 
 if [ "$protocols" = "PROTOCOLS = " ]; then
Only in bitlbee-3.5.1: configure.orig
Only in bitlbee-3.5.1: configure.rej
diff -urp bitlbee-3.5.1.orig/Makefile bitlbee-3.5.1/Makefile
--- bitlbee-3.5.1.orig/Makefile	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/Makefile	2017-02-17 08:37:47.000163279 -0800
@@ -16,7 +16,8 @@ subdirs = lib protocols
 OUTFILE = bitlbee
 
 # Expansion of variables
-subdirobjs = $(foreach dir,$(subdirs),$(dir)/$(dir).o)
+subdirobjs = lib/lib.o
+subdirsos = protocols.so
 
 all: $(OUTFILE) $(OTR_PI) $(SKYPE_PI) doc systemd
 
@@ -119,7 +120,7 @@ uninstall-etc:
 	rm -f $(DESTDIR)$(ETCDIR)/bitlbee.conf
 	-rmdir $(DESTDIR)$(ETCDIR)
 
-install-plugins: install-plugin-otr install-plugin-skype
+install-plugins: install-plugin-otr install-plugin-skype install-plugin-protocols
 
 install-plugin-otr:
 ifdef OTR_PI
@@ -138,6 +139,11 @@ ifdef SKYPE_PI
 	$(MAKE) -C protocols/skype install-doc
 endif
 
+install-plugin-protocols:
+	mkdir -p $(DESTDIR)$(PLUGINDIR)/
+	$(INSTALL) -m 0755 protocols.so $(DESTDIR)$(PLUGINDIR)
+	$(INSTALL) -m 0755 protocols/*_mod.so $(DESTDIR)$(PLUGINDIR)
+
 systemd:
 ifdef SYSTEMDSYSTEMUNITDIR
 	mkdir -p init
@@ -179,8 +185,9 @@ $(objects): %.o: $(_SRCDIR_)%.c
 $(objects): Makefile Makefile.settings config.h
 
 $(OUTFILE): $(objects) $(subdirs)
+	@echo $(subdirobjs)
 	@echo '*' Linking $(OUTFILE)
-	@$(CC) $(objects) $(subdirobjs) -o $(OUTFILE) $(LDFLAGS_BITLBEE) $(LDFLAGS) $(EFLAGS)
+	@$(CC) $(objects) $(subdirobjs) -o $(OUTFILE) $(LDFLAGS_BITLBEE) $(LDFLAGS) $(EFLAGS) -Wl,-rpath=$(PLUGINDIR) $(subdirsos)
 ifneq ($(firstword $(STRIP)), \#)
 	@echo '*' Stripping $(OUTFILE)
 	@-$(STRIP) $(OUTFILE)
Only in bitlbee-3.5.1: Makefile.orig
Only in bitlbee-3.5.1: Makefile.rej
diff -urp bitlbee-3.5.1.orig/protocols/jabber/Makefile bitlbee-3.5.1/protocols/jabber/Makefile
--- bitlbee-3.5.1.orig/protocols/jabber/Makefile	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/protocols/jabber/Makefile	2017-02-17 08:36:02.557183610 -0800
@@ -14,10 +14,10 @@ endif
 # [SH] Program variables
 objects = conference.o io.o iq.o jabber.o jabber_util.o message.o presence.o s5bytestream.o sasl.o si.o hipchat.o
 
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: jabber_mod.o
+all: jabber_mod.so
 check: all
 lcov: check
 gcov: 
@@ -26,7 +26,7 @@ gcov:
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../jabber_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -37,10 +37,10 @@ $(objects): ../../Makefile.settings Make
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-jabber_mod.o: $(objects)
-	@echo '*' Linking jabber_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o jabber_mod.o
+jabber_mod.so: $(objects)
+	@echo '*' Linking jabber_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../jabber_mod.so
 
 -include .depend/*.d
diff -urp bitlbee-3.5.1.orig/protocols/Makefile bitlbee-3.5.1/protocols/Makefile
--- bitlbee-3.5.1.orig/protocols/Makefile	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/protocols/Makefile	2017-02-17 08:36:02.557183610 -0800
@@ -23,11 +23,11 @@ subdirs = $(PROTOCOLS)
 subdirobjs = $(PROTOOBJS)
 
 # Expansion of variables
-subdirobjs := $(join $(subdirs),$(addprefix /,$(subdirobjs)))
-LFLAGS += -r
+#subdirobjs := $(join $(subdirs),$(addprefix /,$(subdirobjs)))
+LFLAGS +=
 
 # [SH] Phony targets
-all: protocols.o
+all: protocols.so
 check: all
 lcov: check
 gcov:
@@ -36,7 +36,7 @@ gcov:
 .PHONY: all clean distclean $(subdirs)
 
 clean: $(subdirs)
-	rm -f *.o $(OUTFILE) core
+	rm -f *.o ../protocols.so $(OUTFILE) core
 
 distclean: clean $(subdirs)
 	rm -rf .depend
@@ -46,14 +46,14 @@ $(subdirs):
 
 ### MAIN PROGRAM
 
-protocols.o: $(objects) $(subdirs)
+protocols.so: $(objects) $(subdirs)
 	@echo '*' Linking protocols.o
-	@$(LD) $(LFLAGS) $(objects) $(subdirobjs) -o protocols.o
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../protocols.so -rpath=$(PLUGINDIR) $(subdirobjs)
 
 $(objects): ../Makefile.settings Makefile
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
 -include .depend/*.d
diff -urp bitlbee-3.5.1.orig/protocols/msn/Makefile bitlbee-3.5.1/protocols/msn/Makefile
--- bitlbee-3.5.1.orig/protocols/msn/Makefile	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/protocols/msn/Makefile	2017-02-17 08:36:02.557183610 -0800
@@ -14,10 +14,10 @@ endif
 # [SH] Program variables
 objects = msn.o msn_util.o ns.o soap.o tables.o gw.o
 
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: msn_mod.o
+all: msn_mod.so
 check: all
 lcov: check
 gcov: 
@@ -26,7 +26,7 @@ gcov:
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../msn_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -37,10 +37,10 @@ $(objects): ../../Makefile.settings Make
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-msn_mod.o: $(objects)
-	@echo '*' Linking msn_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o msn_mod.o
+msn_mod.so: $(objects)
+	@echo '*' Linking msn_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../msn_mod.so
 
 -include .depend/*.d
diff -urp bitlbee-3.5.1.orig/protocols/oscar/Makefile bitlbee-3.5.1/protocols/oscar/Makefile
--- bitlbee-3.5.1.orig/protocols/oscar/Makefile	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/protocols/oscar/Makefile	2017-02-17 08:36:02.558183648 -0800
@@ -15,10 +15,10 @@ endif
 # [SH] Program variables
 objects = admin.o auth.o bos.o buddylist.o chat.o chatnav.o conn.o icq.o im.o info.o misc.o msgcookie.o rxhandlers.o rxqueue.o search.o service.o snac.o ssi.o stats.o tlv.o txqueue.o oscar_util.o oscar.o
 
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: oscar_mod.o
+all: oscar_mod.so
 check: all
 lcov: check
 gcov:
@@ -27,7 +27,7 @@ gcov:
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../oscar_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -38,10 +38,10 @@ $(objects): ../../Makefile.settings Make
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-oscar_mod.o: $(objects)
-	@echo '*' Linking oscar_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o oscar_mod.o
+oscar_mod.so: $(objects)
+	@echo '*' Linking oscar_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../oscar_mod.so
 
 -include .depend/*.d
diff -urp bitlbee-3.5.1.orig/protocols/purple/Makefile bitlbee-3.5.1/protocols/purple/Makefile
--- bitlbee-3.5.1.orig/protocols/purple/Makefile	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/protocols/purple/Makefile	2017-02-17 08:36:02.558183648 -0800
@@ -15,10 +15,10 @@ endif
 objects = ft.o purple.o
 
 CFLAGS += -Wall $(PURPLE_CFLAGS)
-LFLAGS += -r
+LFLAGS +=
 
 # [SH] Phony targets
-all: purple_mod.o
+all: purple_mod.so
 check: all
 lcov: check
 gcov: 
@@ -27,7 +27,7 @@ gcov:
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../purple_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -38,10 +38,10 @@ $(objects): ../../Makefile.settings Make
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-purple_mod.o: $(objects)
-	@echo '*' Linking purple_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o purple_mod.o
+purple_mod.so: $(objects)
+	@echo '*' Linking purple_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../purple_mod.so `pkg-config x11 --libs` `pkg-config purple --libs`
 
 -include .depend/*.d
diff -urp bitlbee-3.5.1.orig/protocols/purple/purple.c bitlbee-3.5.1/protocols/purple/purple.c
--- bitlbee-3.5.1.orig/protocols/purple/purple.c	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/protocols/purple/purple.c	2017-02-17 08:36:02.559183686 -0800
@@ -1881,7 +1881,7 @@ void purple_initmodule()
 	funcs.chat_add_settings = purple_chat_add_settings;
 	funcs.chat_free_settings = purple_chat_free_settings;
 	funcs.transfer_request = purple_transfer_request;
-
+  
 	help = g_string_new("BitlBee libpurple module supports the following IM protocols:\n");
 
 	/* Add a protocol entry to BitlBee's structures for every protocol
@@ -1892,6 +1892,17 @@ void purple_initmodule()
 		struct prpl *ret;
 		struct plugin_info *info;
 
+		fprintf(stderr, "purple.c-protocols: %s\n", prot->info->id);
+
+		if (g_strcasecmp(prot->info->id, "prpl-bigbrownchunx-skypenet") == 0) {
+			//avoid broken plugin
+			continue;
+		}
+
+		if (g_strcasecmp(prot->info->id, "prpl-bigbrownchunx-skype") == 0) {
+			g_set_application_name("bitlbee");
+		}
+
 		/* If we already have this one (as a native module), don't
 		   add a libpurple duplicate. */
 		if (find_protocol(prot->info->id)) {
Only in bitlbee-3.5.1/protocols/purple: purple.c.orig
diff -urp bitlbee-3.5.1.orig/protocols/twitter/Makefile bitlbee-3.5.1/protocols/twitter/Makefile
--- bitlbee-3.5.1.orig/protocols/twitter/Makefile	2017-01-30 12:45:59.000000000 -0800
+++ bitlbee-3.5.1/protocols/twitter/Makefile	2017-02-17 08:36:02.560183724 -0800
@@ -14,10 +14,10 @@ endif
 # [SH] Program variables
 objects = twitter.o twitter_http.o twitter_lib.o
 
-LFLAGS += -r
+LFLAGS += 
 
 # [SH] Phony targets
-all: twitter_mod.o
+all: twitter_mod.so
 check: all
 lcov: check
 gcov: 
@@ -26,7 +26,7 @@ gcov:
 .PHONY: all clean distclean
 
 clean:
-	rm -f *.o core
+	rm -f *.o core ../twitter_mod.so
 
 distclean: clean
 	rm -rf .depend
@@ -37,10 +37,10 @@ $(objects): ../../Makefile.settings Make
 
 $(objects): %.o: $(_SRCDIR_)%.c
 	@echo '*' Compiling $<
-	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CFLAGS_BITLBEE) $< -o $@ -fPIC
 
-twitter_mod.o: $(objects)
-	@echo '*' Linking twitter_mod.o
-	@$(LD) $(LFLAGS) $(objects) -o twitter_mod.o
+twitter_mod.so: $(objects)
+	@echo '*' Linking twitter_mod.so
+	@$(LD) -shared $(LFLAGS) $(objects) -o ../twitter_mod.so
 	
 -include .depend/*.d
