--- a/app/src/main/index.js.orig	2018-07-02 03:02:40.000000000 -0700
+++ b/app/src/main/index.js	2019-02-15 18:43:43.593850365 -0800
@@ -5,7 +5,6 @@ const tray = require('./tray')
 const appMenu = require('./menus')
 
 const updater = require('./updater')
-const analytics = require('./analytics')
 const isPlatform = require('./../common/is-platform')
 const window = require('./window.js')
 
@@ -86,7 +85,6 @@ app.on('ready', () => {
 
   // Update and analytics
   updater.init(mainWindow)
-  analytics.init()
 
   // Setup events
   setupWebContentsEvents(mainWindow.webContents)
