diff -urN -x result.txt -x stderr.txt -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/boinc_wrapper.cpp bfgminer-3.10.10/boinc_wrapper.cpp
--- bfgminer-3.10.10.orig/boinc_wrapper.cpp	2016-01-13 14:27:48.059503116 -0800
+++ bfgminer-3.10.10/boinc_wrapper.cpp	2016-01-13 14:32:40.490841629 -0800
@@ -33,7 +33,7 @@
 
 void bw_boinc_init()
 {
-	boinc_init();
+	//boinc_init(); //init simple program
 }
 
 void bw_boinc_finish(int x)
@@ -43,7 +43,13 @@
 
 void bw_boinc_init_options()
 {
+	//more complex init
+	boinc_options_defaults(options);
         options.normal_thread_priority = true;
+	options.check_heartbeat = false;
+	options.handle_process_control = true; //program handles suspend,quit,resume
+	options.direct_process_action = false; //quit and heartbeat exit, suspend=suspend, resume=resume
+	options.send_status_msgs = true; //report cpu time and fraction done progress
 	boinc_init_options(&options);
 }
 
