Subject:  product_version changes for cef 91.1.6+g8a752eb+chromium-91.0.4472.77

Patch by Orson Teodoro <orsonteodoro@hotmail.com>
Date Jun 7, 2021 (unix time 1623051271)

--- obs-browser/obs-browser-plugin.cpp.orig	2021-05-30 10:51:38.000000000 -0700
+++ obs-browser/obs-browser-plugin.cpp	2021-06-07 00:44:16.637582774 -0700
@@ -298,7 +298,11 @@ static void BrowserInit(void)
 	prod_ver << std::to_string(obs_maj) << "." << std::to_string(obs_min)
 		 << "." << std::to_string(obs_pat);
 
+#if CHROME_VERSION_BUILD >= 4430
+	CefString(&settings.user_agent_product) = prod_ver.str();
+#else
 	CefString(&settings.product_version) = prod_ver.str();
+#endif
 
 #ifdef USE_QT_LOOP
 	settings.external_message_pump = true;
