--- a/plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c.orig	2024-08-15 14:53:35.000000000 -0700
+++ b/plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c	2024-10-03 19:18:36.202923773 -0700
@@ -1080,8 +1080,10 @@ static bool vaapi_device_codec_supported
 	switch (codec) {
 	case CODEC_H264:
 		return vaapi_device_h264_supported(path);
+#ifdef ENABLE_HEVC
 	case CODEC_HEVC:
 		return vaapi_device_hevc_supported(path);
+#endif
 	case CODEC_AV1:
 		return vaapi_device_av1_supported(path);
 	default:
