Subject:  Update conditional for web_security for cef 91.1.6+g8a752eb+chromium-91.0.4472.77

Patch by Orson Teodoro <orsonteodoro@hotmail.com>
Date Jun 7, 2021 (unix time 1623051271)

--- obs-browser/browser-scheme.hpp.orig	2021-05-30 10:51:38.000000000 -0700
+++ obs-browser/browser-scheme.hpp	2021-06-07 00:23:13.298285242 -0700
@@ -22,7 +22,7 @@
 #include <string>
 #include <fstream>
 
-#if CHROME_VERSION_BUILD >= 3440
+#if CHROME_VERSION_BUILD >= 3440 && CHROME_VERSION_BUILD <= 4389
 #define ENABLE_LOCAL_FILE_URL_SCHEME 1
 #else
 #define ENABLE_LOCAL_FILE_URL_SCHEME 0
