--- a/clients/CMakeLists.txt.orig	2025-09-26 06:48:25.000000000 -0700
+++ b/clients/CMakeLists.txt	2025-10-13 16:56:25.634572098 -0700
@@ -177,7 +177,7 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CL
       endif()
     else()
       find_package( BLAS REQUIRED )
-      set( BLAS_LIBRARY "${BLAS_LIBRARIES}" )
+      set( BLAS_LIBRARY "${BLAS_LIBRARIES} -lcblas" )
     endif()
   else() # WIN32
     file(TO_CMAKE_PATH "C:/Program\ Files/AMD/AOCL-Windows/amd-blis/lib/ILP64/AOCL-LibBlis-Win-MT.lib" AOCL_BLAS_LIBRARY)
