diff -urp rocBLAS-rocm-5.5.1.orig/CHANGELOG.md rocBLAS-rocm-5.5.1/CHANGELOG.md
--- rocBLAS-rocm-5.5.1.orig/CHANGELOG.md	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/CHANGELOG.md	2023-08-29 19:21:17.929366513 -0700
@@ -265,7 +265,7 @@ Full documentation for rocBLAS is availa
 - function rocblas_Xdgmm algorithm corrected and incx=0 support added
 - dependencies:
   - rocblas-tensile internal component requires msgpack instead of LLVM
-- Moved the following files from /opt/rocm/include to /opt/rocm/include/internal:
+- Moved the following files from /usr/include to /usr/include/internal:
   - rocblas-auxillary.h
   - rocblas-complex-types.h
   - rocblas-functions.h
@@ -273,7 +273,7 @@ Full documentation for rocBLAS is availa
   - rocblas-version.h
   - rocblas_bfloat16.h
 
-  These files should NOT be included directly as this may lead to errors. Instead, /opt/rocm/include/rocblas.h should be included directly. /opt/rocm/include/rocblas_module.f90 can also be direcly used.
+  These files should NOT be included directly as this may lead to errors. Instead, /usr/include/rocblas.h should be included directly. /usr/include/rocblas_module.f90 can also be direcly used.
 
 
 ## rocBLAS 2.26.0 for ROCm 3.7.0
diff -urp rocBLAS-rocm-5.5.1.orig/clients/CMakeLists.txt rocBLAS-rocm-5.5.1/clients/CMakeLists.txt
--- rocBLAS-rocm-5.5.1.orig/clients/CMakeLists.txt	2023-08-29 19:21:02.645366205 -0700
+++ rocBLAS-rocm-5.5.1/clients/CMakeLists.txt	2023-08-29 19:21:18.009366514 -0700
@@ -75,9 +75,9 @@ find_package(OpenMP)
 
 if (TARGET OpenMP::OpenMP_CXX)
   set( COMMON_LINK_LIBS "OpenMP::OpenMP_CXX")
-  list( APPEND COMMON_LINK_LIBS "-L\"${HIP_CLANG_ROOT}/lib\"")
+  list( APPEND COMMON_LINK_LIBS "-L\"${HIP_CLANG_ROOT}/@LIBDIR@\"")
   if (NOT WIN32)
-    list( APPEND COMMON_LINK_LIBS "-Wl,-rpath=${HIP_CLANG_ROOT}/lib")
+    list( APPEND COMMON_LINK_LIBS "-Wl,-rpath=${HIP_CLANG_ROOT}/@LIBDIR@")
   endif()
 endif()
 
@@ -97,11 +97,11 @@ include( client-build-options )
 set( CMAKE_EXPORT_COMPILE_COMMANDS ON )
 
 if( NOT TARGET rocblas )
-  find_package( rocblas REQUIRED CONFIG PATHS ${ROCM_PATH} /opt/rocm ${ROCM_PATH}/rocblas /opt/rocm/rocblas ${ROCBLAS_LIBRARY_DIR})
+  find_package( rocblas REQUIRED CONFIG PATHS ${ROCM_PATH} /usr ${ROCM_PATH}/rocblas /usr/rocblas ${ROCBLAS_LIBRARY_DIR})
 endif( )
 
 # Hip headers required of all clients; clients use hip to allocate device memory
-list( APPEND CMAKE_PREFIX_PATH ${ROCM_PATH} /opt/rocm )
+list( APPEND CMAKE_PREFIX_PATH ${ROCM_PATH} /usr )
 if ( NOT hip_FOUND )
   find_package( hip REQUIRED CONFIG PATHS ${ROCM_PATH} )
 endif( )
@@ -115,14 +115,14 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CL
     if (LINK_BLIS)
       set( BLIS_INCLUDE_DIR ${BUILD_DIR}/deps/blis/include/blis )
       set( BLIS_CPP ../common/blis_interface.cpp )
-      set( BLAS_LIBRARY ${BUILD_DIR}/deps/blis/lib/libblis.a )
+      set( BLAS_LIBRARY ${BUILD_DIR}/deps/blis/@LIBDIR@/libblis.a )
     else()
       set( BLAS_LIBRARY "blas -lcblas" )
     endif()
   else() # WIN32
     set( BLAS_INCLUDE_DIR ${OPENBLAS_DIR}/include CACHE PATH "OpenBLAS library include path" )
     find_library( BLAS_LIBRARY libopenblas
-                  PATHS ${OPENBLAS_DIR}/lib
+                  PATHS ${OPENBLAS_DIR}/@LIBDIR@
                   NO_DEFAULT_PATH
                 )
     if (NOT BLAS_LIBRARY)
diff -urp rocBLAS-rocm-5.5.1.orig/CMakeLists.txt rocBLAS-rocm-5.5.1/CMakeLists.txt
--- rocBLAS-rocm-5.5.1.orig/CMakeLists.txt	2023-08-29 19:21:02.629366205 -0700
+++ rocBLAS-rocm-5.5.1/CMakeLists.txt	2023-08-29 19:21:18.009366514 -0700
@@ -54,7 +54,7 @@ endif()
 # This finds the rocm-cmake project, and installs it if not found
 # rocm-cmake contains common cmake code for rocm projects to help setup and install
 set( PROJECT_EXTERN_DIR ${CMAKE_CURRENT_BINARY_DIR}/extern )
-find_package( ROCM 0.7.3 CONFIG QUIET PATHS ${ROCM_PATH} /opt/rocm )
+find_package( ROCM 0.7.3 CONFIG QUIET PATHS ${ROCM_PATH} /usr )
 if( NOT ROCM_FOUND )
   set( rocm_cmake_tag "master" CACHE STRING "rocm-cmake tag to download" )
   file( DOWNLOAD https://github.com/RadeonOpenCompute/rocm-cmake/archive/${rocm_cmake_tag}.zip
@@ -105,7 +105,7 @@ rocm_setup_version( VERSION ${VERSION_ST
 
 # Users may override HIP path by specifying their own in CMAKE_MODULE_PATH
 # NOTE:  workaround until llvm & hip cmake modules fixes symlink logic in their config files; remove when fixed
-list( APPEND CMAKE_PREFIX_PATH ${ROCM_PATH}/llvm ${ROCM_PATH} ${ROCM_PATH}/hip /opt/rocm/llvm /opt/rocm /opt/rocm/hip )
+list( APPEND CMAKE_PREFIX_PATH ${ROCM_PATH}/llvm ${ROCM_PATH} ${ROCM_PATH}/hip /usr/lib/llvm/@LLVM_SLOT@ /usr /usr/hip )
 
 
 # gpu arch configuration
@@ -148,7 +148,7 @@ endif()
 
 # Find HIP dependencies
 if( CMAKE_CXX_COMPILER_ID MATCHES "Clang" )
-  find_package( hip REQUIRED CONFIG PATHS ${HIP_DIR} ${ROCM_PATH} /opt/rocm )
+  find_package( hip REQUIRED CONFIG PATHS ${HIP_DIR} ${ROCM_PATH} /usr )
 endif( )
 
 message(STATUS "Using AMDGPU_TARGETS: ${AMDGPU_TARGETS}")
@@ -301,7 +301,7 @@ if(BUILD_CODE_COVERAGE)
     COMMAND mkdir -p lcoverage
     COMMAND echo "\\#!/bin/bash" > llvm-gcov.sh
     COMMAND echo "\\# THIS FILE HAS BEEN GENERATED" >> llvm-gcov.sh
-    COMMAND printf "exec /opt/rocm/llvm/bin/llvm-cov gcov $$\\@" >> llvm-gcov.sh
+    COMMAND printf "exec /usr/lib/llvm/@LLVM_SLOT@/bin/llvm-cov gcov $$\\@" >> llvm-gcov.sh
     COMMAND chmod +x llvm-gcov.sh
     )
 
diff -urp rocBLAS-rocm-5.5.1.orig/CONTRIBUTING.rst rocBLAS-rocm-5.5.1/CONTRIBUTING.rst
--- rocBLAS-rocm-5.5.1.orig/CONTRIBUTING.rst	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/CONTRIBUTING.rst	2023-08-29 19:21:18.009366514 -0700
@@ -774,7 +774,7 @@ Format
 ------
 
 C and C++ code is formatted using ``clang-format``. To run clang-format
-use the version in the ``/opt/rocm/llvm/bin`` directory. Please do not use your
+use the version in the ``/usr/lib/llvm/@LLVM_SLOT@/bin`` directory. Please do not use your
 system's built-in ``clang-format``, as this may be an older version that
 will result in different results.
 
@@ -782,14 +782,14 @@ To format a file, use:
 
 ::
 
-    /opt/rocm/llvm/bin/clang-format -style=file -i <path-to-source-file>
+    /usr/lib/llvm/@SLOT@/bin/clang-format -style=file -i <path-to-source-file>
 
 To format all files, run the following script in rocBLAS directory:
 
 ::
 
     #!/bin/bash
-    git ls-files -z *.cc *.cpp *.h *.hpp *.cl *.h.in *.hpp.in *.cpp.in | xargs -0 /opt/rocm/llvm/bin/clang-format -style=file -i
+    git ls-files -z *.cc *.cpp *.h *.hpp *.cl *.h.in *.hpp.in *.cpp.in | xargs -0 /usr/lib/llvm/@LLVM_SLOT@/bin/clang-format -style=file -i
 
 Also, githooks can be installed to format the code per-commit:
 
diff -urp rocBLAS-rocm-5.5.1.orig/deps/external-gtest.cmake rocBLAS-rocm-5.5.1/deps/external-gtest.cmake
--- rocBLAS-rocm-5.5.1.orig/deps/external-gtest.cmake	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/deps/external-gtest.cmake	2023-08-29 19:21:18.009366514 -0700
@@ -90,11 +90,7 @@ ExternalProject_Get_Property( googletest
 set( package_dir "${PREFIX_GTEST}" )
 if( CMAKE_CONFIGURATION_TYPES )
   # Create a package by bundling libraries and header files
-  if( BUILD_64 )
-    set( LIB_DIR lib64 )
-  else( )
-    set( LIB_DIR lib )
-  endif( )
+  set( LIB_DIR @LIBDIR@ )
 
   set( gtest_lib_dir "<BINARY_DIR>/${LIB_DIR}" )
   ExternalProject_Add_Step( googletest createPackage
diff -urp rocBLAS-rocm-5.5.1.orig/deps/external-lapack.cmake rocBLAS-rocm-5.5.1/deps/external-lapack.cmake
--- rocBLAS-rocm-5.5.1.orig/deps/external-lapack.cmake	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/deps/external-lapack.cmake	2023-08-29 19:21:18.009366514 -0700
@@ -38,11 +38,11 @@ enable_language( Fortran )
 include( GNUInstallDirs )
 
 # lapack cmake exports has a bug on debian architectures, they do not take into account the
-# lib/<machine> paths
+# @LIBDIR@/<machine> paths
 # if CMAKE_INSTALL_LIBDIR is of the form above, strip the machine
-# Match against a '/' in CMAKE_INSTALL_LIBDIR, i.e. lib/x86_64-linux-gnu
-if( ${CMAKE_INSTALL_LIBDIR} MATCHES "lib/.*" )
-  list( APPEND lapack_cmake_args "-DCMAKE_INSTALL_LIBDIR=lib" )
+# Match against a '/' in CMAKE_INSTALL_LIBDIR, i.e. @LIBDIR@
+if( ${CMAKE_INSTALL_LIBDIR} MATCHES "@LIBDIR@" )
+  list( APPEND lapack_cmake_args "-DCMAKE_INSTALL_LIBDIR=@LIBDIR@" )
 endif( )
 
 ExternalProject_Add(
diff -urp rocBLAS-rocm-5.5.1.orig/docs/source/Linux_Install_Guide.rst rocBLAS-rocm-5.5.1/docs/source/Linux_Install_Guide.rst
--- rocBLAS-rocm-5.5.1.orig/docs/source/Linux_Install_Guide.rst	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/docs/source/Linux_Install_Guide.rst	2023-08-29 19:21:18.009366514 -0700
@@ -45,9 +45,9 @@ The rocblas.h header file must be includ
 into rocBLAS, and the rocBLAS shared library will become link-time and run-time
 dependent for the user application.
 
-Once installed, find rocblas.h and rocblas_module.f90 in the /opt/rocm/include
+Once installed, find rocblas.h and rocblas_module.f90 in the /usr/include
 directory. Only use these two installed files when needed in user code.
-Find other rocBLAS files in /opt/rocm/include/internal, However, do not include these files directly.
+Find other rocBLAS files in /usr/include/internal, However, do not include these files directly.
 
 
 -------------------------------
@@ -146,7 +146,7 @@ in the table below:
 | ``./install.sh -i``  | Build library, then      |
 |                      | build and install        |
 |                      | rocBLAS package in       |
-|                      | /opt/rocm/rocblas. You   |
+|                      | /usr/rocblas. You   |
 |                      | will be prompted for     |
 |                      | sudo access. This will   |
 |                      | install for all users.   |
@@ -243,7 +243,7 @@ Note that the version of the rocBLAS cli
 | ``./install.sh --clients-only``     | Build rocBLAS clients    |
 |                                     | and use an installed     |
 |                                     | rocBLAS library at       |
-|                                     | ROCM_PATH (/opt/rocm if  |
+|                                     | ROCM_PATH (/usr if  |
 |                                     | not specified).          |
 +-------------------------------------+--------------------------+
 | ``./install.sh --clients-only``     | Build rocBLAS clients    |
diff -urp rocBLAS-rocm-5.5.1.orig/header_compilation_tests.sh rocBLAS-rocm-5.5.1/header_compilation_tests.sh
--- rocBLAS-rocm-5.5.1.orig/header_compilation_tests.sh	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/header_compilation_tests.sh	2023-08-29 19:21:18.009366514 -0700
@@ -39,7 +39,7 @@ out_uptodate()
     done
 }
 
-rocm_path=/opt/rocm
+rocm_path=/usr
 if ! [ -z ${ROCM_PATH+x} ]; then
     rocm_path=${ROCM_PATH}
 fi
@@ -50,7 +50,7 @@ HCC_OPTS="-Werror -DBUILD_WITH_TENSILE=1
 
 GPU_OPTS="-Wno-unused-command-line-argument -fvisibility=hidden -fvisibility-inlines-hidden -hc -fno-gpu-rdc --amdgpu-target=gfx803 --amdgpu-target=gfx900 --amdgpu-target=gfx906 -Werror"
 
-CLANG=${rocm_path}/llvm/bin/clang
+CLANG=${rocm_path}/lib/llvm/@LLVM_SLOT@/bin/clang
 CLANG_OPTS="-xc-header -std=c99"  # auto set in hip_common.h -D__HIP_PLATFORM_HCC__
 
 GCC=/usr/bin/gcc
diff -urp rocBLAS-rocm-5.5.1.orig/install.sh rocBLAS-rocm-5.5.1/install.sh
--- rocBLAS-rocm-5.5.1.orig/install.sh	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/install.sh	2023-08-29 19:21:18.009366514 -0700
@@ -173,7 +173,7 @@ install_packages( )
     library_dependencies_sles+=( "gcc-fortran" "libgomp1" )
 
     # wget is needed for blis
-    if [[ ! -e "${build_dir}/deps/blis/lib/libblis.a" ]]; then
+    if [[ ! -e "${build_dir}/deps/blis/@LIBDIR@/libblis.a" ]]; then
       library_dependencies_ubuntu+=("wget")
       library_dependencies_centos_rhel+=("wget")
       library_dependencies_centos_8+=("wget")
@@ -403,7 +403,7 @@ printf "\033[32mCreating project build d
 install_blis()
 {
     #Download prebuilt AMD multithreaded blis
-    if [[ ! -e "./blis/lib/libblis.a" ]]; then
+    if [[ ! -e "./blis/@LIBDIR@/libblis.a" ]]; then
       case "${ID}" in
           centos|rhel|sles|opensuse-leap)
               wget -nv -O blis.tar.gz https://github.com/amd/blis/releases/download/2.0/aocl-blis-mt-centos-2.0.tar.gz
@@ -421,7 +421,7 @@ install_blis()
       rm -rf blis/amd-blis-mt
       mv amd-blis-mt blis
       rm blis.tar.gz
-      cd blis/lib
+      cd blis/@LIBDIR@
       ln -sf libblis-mt.a libblis.a
     fi
 }
diff -urp rocBLAS-rocm-5.5.1.orig/library/src/CMakeLists.txt rocBLAS-rocm-5.5.1/library/src/CMakeLists.txt
--- rocBLAS-rocm-5.5.1.orig/library/src/CMakeLists.txt	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/library/src/CMakeLists.txt	2023-08-29 19:21:18.029366515 -0700
@@ -537,7 +537,7 @@ endif( )
 
 if( NOT BUILD_SHARED_LIBS )
   # Following Boost conventions of prefixing 'lib' on static built libraries, across all platforms
-  set_target_properties( rocblas PROPERTIES PREFIX "lib" )
+  set_target_properties( rocblas PROPERTIES PREFIX "@LIBDIR@" )
 
   if( BUILD_WITH_TENSILE )
     add_custom_command( TARGET rocblas POST_BUILD
diff -urp rocBLAS-rocm-5.5.1.orig/library/src/tensile_host.cpp rocBLAS-rocm-5.5.1/library/src/tensile_host.cpp
--- rocBLAS-rocm-5.5.1.orig/library/src/tensile_host.cpp	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/library/src/tensile_host.cpp	2023-08-29 19:22:12.201367606 -0700
@@ -66,7 +66,7 @@ extern "C" void rocblas_shutdown();
 #include <libgen.h>
 #include <link.h>
 #include <unistd.h>
-#define ROCBLAS_LIB_PATH "/opt/rocm/lib/rocblas"
+#define ROCBLAS_LIB_PATH "/usr/@LIBDIR@"
 #endif
 
 #ifdef WIN32
@@ -561,47 +561,10 @@ namespace
             }
             else
             {
-                path = ROCBLAS_LIB_PATH;
-
                 // Find the location of librocblas.dll/.so
                 // Fall back on hard-coded path if static library or not found
 
-#ifndef ROCBLAS_STATIC_LIB
-#ifdef WIN32
-                // wchar_t wpath[MAX_PATH + 1] = {0};
-                // if(GetModuleFileNameW(GetModuleHandle("rocblas.dll"), wpath, MAX_PATH + 1))
-                // {
-                //     std::wstring          wspath(wpath);
-                //     std::string           tmp(wspath.begin(), wspath.end());
-
-                std::vector<TCHAR> dll_path(MAX_PATH + 1);
-                if(GetModuleFileNameA(
-                       GetModuleHandleA("rocblas.dll"), dll_path.data(), MAX_PATH + 1))
-                {
-                    std::string           tmp(dll_path.begin(), dll_path.end());
-                    std::filesystem::path exepath = tmp;
-                    if(exepath.has_filename())
-                    {
-                        path = exepath.remove_filename().string();
-                    }
-                }
-#else
-                dl_iterate_phdr(rocblas_dl_iterate_phdr_callback, NULL);
-                if(rocblas_so_path.size())
-                    path = std::string{dirname(&rocblas_so_path[0])};
-#endif
-#endif // ifndef ROCBLAS_STATIC_LIB
-
-                // Find the location of the libraries
-                if(TestPath(path + "/../../Tensile/library"))
-                    path += "/../../Tensile/library";
-                else if(TestPath(path + "library"))
-                    path += "/library";
-                else
-                    path += "/rocblas/library";
-
-                if(TestPath(path + "/" + processor))
-                    path += "/" + processor;
+		path="@EPREFIX@/usr/@LIBDIR@/rocblas/library";
             }
 
 #ifdef TENSILE_YAML
diff -urp rocBLAS-rocm-5.5.1.orig/rmake.py rocBLAS-rocm-5.5.1/rmake.py
--- rocBLAS-rocm-5.5.1.orig/rmake.py	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/rmake.py	2023-08-29 19:21:18.325366521 -0700
@@ -105,7 +105,7 @@ def parse_args():
                         help='Disable on-demand loading of Tensile Library files. (Default is enabled)')
 
     parser.add_argument(     '--library-path', dest='library_dir_installed', type=str, required=False, default="",
-                        help='Specify path to a pre-built rocBLAS library, when building clients only using --clients-only flag. (optional, default: /opt/rocm/rocblas)')
+                        help='Specify path to a pre-built rocBLAS library, when building clients only using --clients-only flag. (optional, default: /usr/rocblas)')
 
     parser.add_argument('-n', '--no_tensile', dest='build_tensile', required=False, default=True, action='store_false',
                         help='Build a subset of rocBLAS library which does not require Tensile.')
@@ -238,7 +238,7 @@ def config_cmd():
         cmake_platform_opts.append(f'-DCMAKE_INSTALL_PREFIX="C:/hipSDK"')
         toolchain = os.path.join(src_path, "toolchain-windows.cmake")
     else:
-        rocm_raw_path = os.getenv('ROCM_PATH', "/opt/rocm")
+        rocm_raw_path = os.getenv('ROCM_PATH', "/usr")
         rocm_path = rocm_raw_path
         cmake_platform_opts.append(f"-DROCM_DIR:PATH={rocm_path} -DCPACK_PACKAGING_INSTALL_PREFIX={rocm_path}")
         cmake_platform_opts.append(f'-DCMAKE_INSTALL_PREFIX="rocblas-install"')
@@ -298,7 +298,7 @@ def config_cmd():
         cmake_options.append(f"-DBUILD_SHARED_LIBS=OFF")
 
     if args.relocatable:
-        rocm_rpath = os.getenv('ROCM_RPATH', "/opt/rocm/lib:/opt/rocm/lib64")
+        rocm_rpath = os.getenv('ROCM_RPATH', "/usr/@LIBDIR@")
         cmake_options.append(f'-DCMAKE_SHARED_LINKER_FLAGS=" -Wl,--enable-new-dtags -Wl,--rpath,{rocm_rpath}"')
 
     if args.skip_ld_conf_entry or args.relocatable:
diff -urp rocBLAS-rocm-5.5.1.orig/scripts/performance/blas/commandrunner.py rocBLAS-rocm-5.5.1/scripts/performance/blas/commandrunner.py
--- rocBLAS-rocm-5.5.1.orig/scripts/performance/blas/commandrunner.py	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/scripts/performance/blas/commandrunner.py	2023-08-29 19:21:18.325366521 -0700
@@ -125,7 +125,7 @@ def import_rocm_smi(install_path):
             import rocm_smi
             smi = rocm_smi
 
-            # The following is needed to call rsmi_init() before other calls as documented in /opt/rocm/rocm_smi/docs/README.md
+            # The following is needed to call rsmi_init() before other calls as documented in /usr/rocm_smi/docs/README.md
             smi.initializeRsmi()
         except ImportError:
             print('WARNING - rocm_smi.py not found!')
@@ -1433,7 +1433,7 @@ def parse_input_arguments(parser):
                         help='Output directory for the summary documentation.')
     parser.add_argument('-d', '--device-num', default=0, type=int,
                         help='Device number to run on.')
-    parser.add_argument('--install-path', default='/opt/rocm', help='Top directory of driver installation.')
+    parser.add_argument('--install-path', default='/usr', help='Top directory of driver installation.')
 
     parser.add_argument('--surface-plot', default=False, action='store_true', help='Adds a surface plot instead of a normal plot.')
 
diff -urp rocBLAS-rocm-5.5.1.orig/scripts/performance/blas/getspecs.py rocBLAS-rocm-5.5.1/scripts/performance/blas/getspecs.py
--- rocBLAS-rocm-5.5.1.orig/scripts/performance/blas/getspecs.py	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/scripts/performance/blas/getspecs.py	2023-08-29 19:21:18.329366521 -0700
@@ -45,13 +45,13 @@ def get_smi_exec(cuda):
     if cuda:
         return "nvidia-smi"
     else:
-        return "/opt/rocm/bin/rocm-smi"
+        return "/usr/bin/rocm-smi"
 
 def getgfx(devicenum, cuda):
     if cuda:
         return "N/A"
     else:
-        cmd = ["/opt/rocm/bin/rocm_agent_enumerator"]
+        cmd = ["/usr/bin/rocm_agent_enumerator"]
         success, cout = _subprocess_helper(cmd)
         if not success:
             return "N/A"
@@ -410,7 +410,7 @@ def validmemtypes(cuda, smi=None):
     if cuda:
         return ["vram"]
     elif smi is not None:
-        # Hardcoded in /opt/rocm/rocm_smi/bindings/rsmiBindings.py
+        # Hardcoded in /usr/rocm_smi/bindings/rsmiBindings.py
         return ["VRAM", "VIS_VRAM", "GTT"]
 
 def getmeminfo(devicenum, mem_type, cuda, smi=None):
@@ -431,12 +431,12 @@ def getmeminfo(devicenum, mem_type, cuda
         return smi.getMemInfo(devicenum, mem_type)
 
 def validversioncomponents(cuda, smi=None):
-    # currently only driver according to /opt/rocm/bin/rocm_smi.py
-    # driver corresponds to 0 in /opt/rocm/bin/rocm_smi.py
+    # currently only driver according to /usr/bin/rocm_smi.py
+    # driver corresponds to 0 in /usr/bin/rocm_smi.py
     if cuda:
         return ['driver']
     else:
-        # currently only driver according to /opt/rocm/bin/rocm_smi.py
+        # currently only driver according to /usr/bin/rocm_smi.py
         return [0]
 
 def getversion(devicenum, component, cuda, smi=None):
diff -urp rocBLAS-rocm-5.5.1.orig/scripts/utilities/check_for_pretuned_sizes_c/Makefile rocBLAS-rocm-5.5.1/scripts/utilities/check_for_pretuned_sizes_c/Makefile
--- rocBLAS-rocm-5.5.1.orig/scripts/utilities/check_for_pretuned_sizes_c/Makefile	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/scripts/utilities/check_for_pretuned_sizes_c/Makefile	2023-08-29 19:21:18.329366521 -0700
@@ -1,9 +1,9 @@
-# Makefile assumes rocBLAS is installed in /opt/rocm/rocblas
+# Makefile assumes rocBLAS is installed in /usr/rocblas
 
-ROCM_INSTALL_DIR=/opt/rocm
+ROCM_INSTALL_DIR=/usr
 ROCBLAS_INSTALL_DIR=$(ROCM_INSTALL_DIR)
 ROCBLAS_INCLUDE=$(ROCBLAS_INSTALL_DIR)/include
-ROCBLAS_LIB_PATH=$(ROCBLAS_INSTALL_DIR)/lib
+ROCBLAS_LIB_PATH=$(ROCBLAS_INSTALL_DIR)/@LIBDIR@
 HIP_INCLUDE=$(ROCM_INSTALL_DIR)/include
 LDFLAGS=-L$(ROCBLAS_LIB_PATH) -lrocblas
 LD=$(ROCM_INSTALL_DIR)/bin/hipcc
diff -urp rocBLAS-rocm-5.5.1.orig/toolchain-linux.cmake rocBLAS-rocm-5.5.1/toolchain-linux.cmake
--- rocBLAS-rocm-5.5.1.orig/toolchain-linux.cmake	2023-03-28 07:57:37.000000000 -0700
+++ rocBLAS-rocm-5.5.1/toolchain-linux.cmake	2023-08-29 19:21:18.441366523 -0700
@@ -6,7 +6,7 @@ endif()
 if (DEFINED ENV{ROCM_PATH})
   set(rocm_bin "$ENV{ROCM_PATH}/bin")
 else()
-  set(rocm_bin "/opt/rocm/bin")
+  set(rocm_bin "/usr/bin")
 endif()
 
 # relying on env and path for backward compatibility with external recipes
