diff -urp rocBLAS-rocm-5.7.0.orig/CMakeLists.txt rocBLAS-rocm-5.7.0/CMakeLists.txt
--- rocBLAS-rocm-5.7.0.orig/CMakeLists.txt	2023-09-08 08:06:24.000000000 -0700
+++ rocBLAS-rocm-5.7.0/CMakeLists.txt	2023-10-02 02:03:46.595579926 -0700
@@ -150,21 +150,6 @@ if(NOT SKIP_LIBRARY)
 
       set( Tensile_TEST_LOCAL_PATH "" CACHE PATH "Use local Tensile directory instead of fetching a GitHub branch" )
 
-      include(virtualenv)
-
-      if (Tensile_TEST_LOCAL_PATH)
-        virtualenv_install(${Tensile_TEST_LOCAL_PATH})
-        message (STATUS "using local Tensile from ${Tensile_TEST_LOCAL_PATH}, copied to ${Tensile_ROOT}")
-      else()
-        # Use the virtual-env setup and download package from specified repot:
-        set( tensile_fork "ROCmSoftwarePlatform" CACHE STRING "Tensile fork to use" )
-        file (STRINGS "tensile_tag.txt" read_tensile_tag)
-        set( tensile_tag ${read_tensile_tag} CACHE STRING "Tensile tag to download" )
-        virtualenv_install("git+https://github.com/${tensile_fork}/Tensile.git@${tensile_tag}")
-        message (STATUS "using GIT Tensile fork=${tensile_fork} from branch=${tensile_tag}")
-      endif()
-      message(STATUS "Adding ${VIRTUALENV_HOME_DIR} to CMAKE_PREFIX_PATH")
-      list(APPEND CMAKE_PREFIX_PATH ${VIRTUALENV_HOME_DIR})
     endif()
     find_package(Tensile ${TENSILE_VERSION} EXACT REQUIRED HIP LLVM)
   endif()
Only in rocBLAS-rocm-5.7.0: CMakeLists.txt.orig
Only in rocBLAS-rocm-5.7.0: CMakeLists.txt.rej
