diff -urp openmp.orig/libomptarget/CMakeLists.txt openmp/libomptarget/CMakeLists.txt
--- a/openmp/libomptarget/CMakeLists.txt	2023-09-30 07:53:18.126358310 -0700
+++ b/openmp/libomptarget/CMakeLists.txt	2023-09-30 07:59:02.318354351 -0700
@@ -35,6 +35,10 @@ if (NOT LIBOMPTARGET_LLVM_INCLUDE_DIRS)
   message(FATAL_ERROR "Missing definition for LIBOMPTARGET_LLVM_INCLUDE_DIRS")
 endif()
 
+set(_OPENMP_RUNTIME_DIR "${PROJECT_SOURCE_DIR}/../llvm_build/projects/openmp/runtime/src")
+set(_OMPT_HEADER_DIR "${PROJECT_SOURCE_DIR}/../openmp/runtime/exports/common.dia.ompt.optional/include")
+include_directories(${LIBOMPTARGET_LLVM_INCLUDE_DIRS} ${_OMPT_HEADER_DIR})
+
 # This is a list of all the targets that are supported/tested right now.
 set (LIBOMPTARGET_ALL_TARGETS "${LIBOMPTARGET_ALL_TARGETS} aarch64-unknown-linux-gnu")
 set (LIBOMPTARGET_ALL_TARGETS "${LIBOMPTARGET_ALL_TARGETS} amdgcn-amd-amdhsa")
Only in openmp/libomptarget: CMakeLists.txt.orig
Only in openmp/libomptarget: CMakeLists.txt.rej
diff -urp openmp.orig/libomptarget/src/CMakeLists.txt openmp/libomptarget/src/CMakeLists.txt
--- a/openmp/libomptarget/src/CMakeLists.txt	2022-03-17 08:38:43.000000000 -0700
+++ b/openmp/libomptarget/src/CMakeLists.txt	2023-09-30 07:57:44.766355243 -0700
@@ -12,6 +12,9 @@
 
 libomptarget_say("Building offloading runtime library libomptarget.")
 
+set(_OPENMP_RUNTIME_DIR "${PROJECT_SOURCE_DIR}/../llvm_build/projects/openmp/runtime/src")
+set(_OMPT_HEADER_DIR "${PROJECT_SOURCE_DIR}/../openmp/runtime/exports/common.dia.ompt.optional/include")
+
 set(LIBOMPTARGET_SRC_FILES
   ${CMAKE_CURRENT_SOURCE_DIR}/api.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/device.cpp
@@ -25,7 +27,7 @@ set(LIBOMPTARGET_SRC_FILES
 
 set(LIBOMPTARGET_SRC_FILES ${LIBOMPTARGET_SRC_FILES} PARENT_SCOPE)
 
-include_directories(${LIBOMPTARGET_LLVM_INCLUDE_DIRS})
+include_directories(${LIBOMPTARGET_LLVM_INCLUDE_DIRS} ${_OPENMP_RUNTIME_DIR} ${_OMPT_HEADER_DIR})
 
 # Build libomptarget library with libdl dependency.
 add_library(omptarget SHARED ${LIBOMPTARGET_SRC_FILES})
Only in openmp/libomptarget/src: CMakeLists.txt.orig
