--- a/openmp/libomptarget/CMakeLists.txt.orig	2023-06-14 10:03:33.000000000 -0700
+++ b/openmp/libomptarget/CMakeLists.txt	2023-08-12 00:04:09.825338028 -0700
@@ -105,16 +105,20 @@ pythonize_bool(LIBOMPTARGET_OMPT_SUPPORT
 
 set(LIBOMPTARGET_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
-find_path (
-  LIBOMP_OMP_TOOLS_INCLUDE_DIR
-  NAMES
-    omp-tools.h
-  PATHS
-    ${CMAKE_BINARY_DIR}/openmp/runtime/src
-    ${CMAKE_BINARY_DIR}/runtime/src
-  )
+if(LIBOMPTARGET_CMAKE_BUILD_TYPE MATCHES debug)
+  find_path (
+    LIBOMP_OMP_TOOLS_INCLUDE_DIR
+    NAMES
+      omp-tools.h
+    PATHS
+      ${CMAKE_BINARY_DIR}/openmp/runtime/src
+      ${CMAKE_BINARY_DIR}/runtime/src
+    )
 
-include_directories(${LIBOMPTARGET_INCLUDE_DIR} ${LIBOMP_OMP_TOOLS_INCLUDE_DIR})
+  include_directories(${LIBOMPTARGET_INCLUDE_DIR} ${LIBOMP_OMP_TOOLS_INCLUDE_DIR})
+else()
+  include_directories(${LIBOMPTARGET_INCLUDE_DIR})
+endif()
 
 # Various LLVM_TOOLS are needed to build libomptarget
 find_package(LLVM QUIET CONFIG PATHS
