--- a/mlir/lib/Dialect/GPU/Transforms/SerializeToHsaco.cpp.orig	2023-06-14 10:03:33.000000000 -0700
+++ b/mlir/lib/Dialect/GPU/Transforms/SerializeToHsaco.cpp	2023-09-16 12:06:17.210880158 -0700
@@ -281,7 +281,7 @@ SerializeToHsacoPass::translateToLLVMIR(
   std::optional<SmallVector<std::unique_ptr<llvm::Module>, 3>> mbModules;
   std::string theRocmPath = getRocmPath();
   llvm::SmallString<32> bitcodePath(theRocmPath);
-  llvm::sys::path::append(bitcodePath, "amdgcn", "bitcode");
+  llvm::sys::path::append(bitcodePath, "@LIBDIR@", "amdgcn", "bitcode");
   mbModules = loadLibraries(bitcodePath, libraries, llvmContext);
 
   if (!mbModules) {
--- a/compiler-rt/test/asan/lit.cfg.py.orig	2023-06-14 10:03:33.000000000 -0700
+++ b/compiler-rt/test/asan/lit.cfg.py	2023-09-16 12:13:18.138866637 -0700
@@ -253,13 +253,13 @@ if config.host_os == 'NetBSD':
 def configure_rocm(config, test_rocm_path):
     if (not os.path.isdir(test_rocm_path)):
         print("no directory found")
-        test_rocm_path = os.path.join('/opt','rocm')
+        test_rocm_path = os.path.join('/usr')
         if (not os.path.isdir(test_rocm_path)):
             test_rocm_path = os.path.abspath(os.path.join(config.llvm_install_dir, os.pardir))
             if (not os.path.isdir(test_rocm_path)):
                 sys.exit("ROCM installation not found, try exporting ASAN_TEST_ROCM variable")
 
-    test_device_libs  = os.path.join(test_rocm_path, 'amdgcn', 'bitcode')
+    test_device_libs  = os.path.join(test_rocm_path, '@LIBDIR@', 'amdgcn', 'bitcode')
     test_hip_path     = os.path.join(test_rocm_path, 'hip')
     hipcc             = os.path.join(test_hip_path, 'bin', 'hipcc')
 
--- a/clang/lib/Driver/ToolChains/AMDGPU.cpp.orig	2023-09-16 13:31:57.246715059 -0700
+++ b/clang/lib/Driver/ToolChains/AMDGPU.cpp	2023-09-16 13:32:28.042714069 -0700
@@ -424,11 +424,13 @@ void RocmInstallationDetector::detectDev
     };
 
     // The possible structures are:
+    // - ${ROCM_ROOT}/@LIBDIR@/amdgcn/bitcode/*
     // - ${ROCM_ROOT}/amdgcn/bitcode/*
     // - ${ROCM_ROOT}/lib/*
     // - ${ROCM_ROOT}/lib/bitcode/*
     // so try to detect these layouts.
     static constexpr std::array<const char *, 2> SubDirsList[] = {
+        {"@LIBDIR@/amdgcn", "bitcode"},
         {"amdgcn", "bitcode"},
         {"lib", ""},
         {"lib", "bitcode"},
