--- a/openmp/libomptarget/tools/prep-libomptarget-bc/CMakeLists.txt.orig	2022-08-11 10:57:36.000000000 -0700
+++ b/openmp/libomptarget/tools/prep-libomptarget-bc/CMakeLists.txt	2023-10-07 16:15:05.543816586 -0700
@@ -9,6 +9,7 @@ set(LLVM_LINK_COMPONENTS
   TransformUtils
   IPO
   )
 add_llvm_tool(prep-libomptarget-bc prep-libomptarget-bc.cpp)
 llvm_update_compile_flags(prep-libomptarget-bc)
+target_link_options(prep-libomptarget-bc PRIVATE "-Wl,-rpath=@ESYSROOT_LLVM_PATH@/@LIBDIR@")
 include_directories( ${LLVM_INCLUDE_DIR} ${LLVM_MAIN_INCLUDE_DIR})
