--- b/player/src/tizplayapp.cpp	2020-03-08 14:18:54.000000000 -0700
+++ a/player/src/tizplayapp.cpp	2020-04-06 04:18:22.026122977 -0700
@@ -67,25 +67,41 @@
 #include "tizgraphtypes.hpp"
 #include "tizomxutil.hpp"
 #include <decoders/tizdecgraphmgr.hpp>
+#ifdef HAVE_ICECAST_CLIENT
 #include <httpclnt/tizhttpclntmgr.hpp>
+#endif
+#ifdef HAVE_ICECAST_SERVER
 #include <httpserv/tizhttpservconfig.hpp>
 #include <httpserv/tizhttpservmgr.hpp>
+#endif
+#ifdef HAVE_CHROMECAST
 #include <services/chromecast/tizchromecastconfig.hpp>
 #include <services/chromecast/tizchromecastmgr.hpp>
+#endif
+#ifdef HAVE_TUNEIN
 #include <services/tunein/tiztuneinconfig.hpp>
 #include <services/tunein/tiztuneinmgr.hpp>
+#endif
+#ifdef HAVE_GMUSIC
 #include <services/googlemusic/tizgmusicconfig.hpp>
 #include <services/googlemusic/tizgmusicmgr.hpp>
+#endif
+#ifdef HAVE_SOUNDCLOUD
 #include <services/soundcloud/tizscloudconfig.hpp>
 #include <services/soundcloud/tizscloudmgr.hpp>
+#endif
 #ifdef HAVE_LIBSPOTIFY
 #include <services/spotify/tizspotifyconfig.hpp>
 #include <services/spotify/tizspotifymgr.hpp>
 #endif
+#ifdef HAVE_PLEX
 #include <services/plex/tizplexconfig.hpp>
 #include <services/plex/tizplexmgr.hpp>
+#endif
+#ifdef HAVE_YOUTUBE
 #include <services/youtube/tizyoutubeconfig.hpp>
 #include <services/youtube/tizyoutubemgr.hpp>
+#endif
 
 #include "tizplayapp.hpp"
 
@@ -505,54 +521,82 @@ void tiz::playapp::set_option_handlers (
   // local audio decoding program options
   popts_.set_option_handler ("decode-local",
                              boost::bind (&tiz::playapp::decode_local, this));
+#ifdef HAVE_ICECAST_SERVER
   // streaming audio server program options
   popts_.set_option_handler ("serve-stream",
                              boost::bind (&tiz::playapp::serve_stream, this));
+#endif
+#ifdef HAVE_ICECAST_CLIENT
   // streaming audio client program options
   popts_.set_option_handler ("decode-stream",
                              boost::bind (&tiz::playapp::decode_stream, this));
+#endif
+#ifdef HAVE_LIBSPOTIFY
   // spotify streaming client program options
   popts_.set_option_handler ("spotify-stream",
                              boost::bind (&tiz::playapp::spotify_stream, this));
+#endif
+#ifdef HAVE_GMUSIC
   // Google music streaming client program options
   popts_.set_option_handler ("gmusic-stream",
                              boost::bind (&tiz::playapp::gmusic_stream, this));
+#endif
+#ifdef HAVE_SOUNDCLOUD
   // SoundCloud music streaming client program options
   popts_.set_option_handler ("scloud-stream",
                              boost::bind (&tiz::playapp::scloud_stream, this));
+#endif
+#ifdef HAVE_TUNEIN
   // Tunein internet radio directory streaming client program options
   popts_.set_option_handler ("tunein-stream",
                              boost::bind (&tiz::playapp::tunein_stream, this));
+#endif
+#ifdef HAVE_YOUTUBE
   // YouTube audio streaming client program options
   popts_.set_option_handler ("youtube-stream",
                              boost::bind (&tiz::playapp::youtube_stream, this));
+#endif
+#ifdef HAVE_PLEX
   // Plex audio streaming client program options
   popts_.set_option_handler ("plex-stream",
                              boost::bind (&tiz::playapp::plex_stream, this));
+#endif
+#ifdef HAVE_CHROMECAST
   // HTTP music streaming on Chromecast device
   popts_.set_option_handler (
       "http-stream-chromecast",
       boost::bind (&tiz::playapp::http_stream_chromecast, this));
+#ifdef HAVE_GMUSIC
   // Google music streaming on Chromecast device
   popts_.set_option_handler (
       "gmusic-stream-chromecast",
       boost::bind (&tiz::playapp::gmusic_stream_chromecast, this));
+#endif
+#ifdef HAVE_SOUNDCLOUD
   // Soudcloud audio streaming on Chromecast device
   popts_.set_option_handler (
       "scloud-stream-chromecast",
       boost::bind (&tiz::playapp::scloud_stream_chromecast, this));
+#endif
+#ifdef HAVE_TUNEIN
   // Tunein audio streaming on Chromecast device
   popts_.set_option_handler (
       "tunein-stream-chromecast",
       boost::bind (&tiz::playapp::tunein_stream_chromecast, this));
+#endif
+#ifdef HAVE_YOUTUBE
   // YouTube audio streaming on Chromecast device
   popts_.set_option_handler (
       "youtube-stream-chromecast",
       boost::bind (&tiz::playapp::youtube_stream_chromecast, this));
+#endif
+#ifdef HAVE_PLEX
   // Plex audio streaming on Chromecast device
   popts_.set_option_handler (
       "plex-stream-chromecast",
       boost::bind (&tiz::playapp::plex_stream_chromecast, this));
+#endif
+#endif
 }
 
 OMX_ERRORTYPE
@@ -782,6 +826,7 @@ tiz::playapp::decode_local ()
   return rc;
 }
 
+#ifdef HAVE_ICECAST_SERVER
 OMX_ERRORTYPE
 tiz::playapp::serve_stream ()
 {
@@ -874,7 +919,9 @@ tiz::playapp::serve_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_ICECAST_CLIENT
 OMX_ERRORTYPE
 tiz::playapp::decode_stream ()
 {
@@ -911,6 +958,7 @@ tiz::playapp::decode_stream ()
 
   return rc;
 }
+#endif
 
 #ifdef HAVE_LIBSPOTIFY
 OMX_ERRORTYPE
@@ -974,14 +1022,9 @@ tiz::playapp::spotify_stream ()
 
   return rc;
 }
-#else
-OMX_ERRORTYPE
-tiz::playapp::spotify_stream ()
-{
-  return OMX_ErrorNone;
-}
 #endif
 
+#ifdef HAVE_GMUSIC
 OMX_ERRORTYPE
 tiz::playapp::gmusic_stream ()
 {
@@ -1038,7 +1081,9 @@ tiz::playapp::gmusic_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_SOUNDCLOUD
 OMX_ERRORTYPE
 tiz::playapp::scloud_stream ()
 {
@@ -1081,7 +1126,9 @@ tiz::playapp::scloud_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_TUNEIN
 OMX_ERRORTYPE
 tiz::playapp::tunein_stream ()
 {
@@ -1126,7 +1173,9 @@ tiz::playapp::tunein_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_YOUTUBE
 OMX_ERRORTYPE
 tiz::playapp::youtube_stream ()
 {
@@ -1170,7 +1219,9 @@ tiz::playapp::youtube_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_PLEX
 OMX_ERRORTYPE
 tiz::playapp::plex_stream ()
 {
@@ -1215,7 +1266,9 @@ tiz::playapp::plex_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_CHROMECAST
 OMX_ERRORTYPE
 tiz::playapp::http_stream_chromecast ()
 {
@@ -1260,7 +1313,10 @@ tiz::playapp::http_stream_chromecast ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_CHROMECAST
+#ifdef HAVE_GMUSIC
 OMX_ERRORTYPE
 tiz::playapp::gmusic_stream_chromecast ()
 {
@@ -1324,7 +1380,11 @@ tiz::playapp::gmusic_stream_chromecast (
 
   return rc;
 }
+#endif
+#endif
 
+#ifdef HAVE_CHROMECAST
+#ifdef HAVE_SOUNDCLOUD
 OMX_ERRORTYPE
 tiz::playapp::scloud_stream_chromecast ()
 {
@@ -1374,7 +1434,11 @@ tiz::playapp::scloud_stream_chromecast (
 
   return rc;
 }
+#endif
+#endif
 
+#ifdef HAVE_CHROMECAST
+#ifdef HAVE_TUNEIN
 OMX_ERRORTYPE
 tiz::playapp::tunein_stream_chromecast ()
 {
@@ -1426,7 +1490,11 @@ tiz::playapp::tunein_stream_chromecast (
 
   return rc;
 }
+#endif
+#endif
 
+#ifdef HAVE_CHROMECAST
+#ifdef HAVE_YOUTUBE
 OMX_ERRORTYPE
 tiz::playapp::youtube_stream_chromecast ()
 {
@@ -1476,7 +1544,11 @@ tiz::playapp::youtube_stream_chromecast
 
   return rc;
 }
+#endif
+#endif
 
+#ifdef HAVE_CHROMECAST
+#ifdef HAVE_PLEX
 OMX_ERRORTYPE
 tiz::playapp::plex_stream_chromecast ()
 {
@@ -1529,6 +1601,8 @@ tiz::playapp::plex_stream_chromecast ()
 
   return rc;
 }
+#endif
+#endif
 
 void tiz::playapp::print_banner () const
 {
Only in tizonia-openmax-il-0.21.0/player/src: tizplayapp.cpp.orig
Only in tizonia-openmax-il-0.21.0/player/src: tizplayapp.cpp.rej
