--- a/plugins/Makefile.am.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/plugins/Makefile.am	2019-08-01 17:30:18.421425861 -0700
@@ -15,47 +15,89 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with Tizonia.  If not, see <http://www.gnu.org/licenses/>.
 
-REQUIRED_SUBDIRS = \
-	chromecast_renderer \
-	file_reader \
-	file_writer \
-	flac_decoder \
-	http_renderer \
-	http_source \
-	mp3_decoder \
-	mp3_encoder \
-	mp3_metadata \
-	mpeg_audio_decoder \
-	ogg_demuxer \
-	ogg_muxer \
-	opus_decoder \
-	opusfile_decoder \
-	pcm_decoder \
-	pcm_renderer_pa \
-	vorbis_decoder \
-	vp8_decoder \
-	webm_demuxer \
-	yuv_renderer
+REQUIRED_SUBDIRS =
 
-LIBSPOTIFY_SUBDIR= \
-	spotify_source
+SUBDIRS = ${REQUIRED_SUBDIRS}
 
-ALSA_SUBDIR= \
-	pcm_renderer_alsa
+if WITH_CHROMECAST
+SUBDIRS += chromecast_renderer
+endif
 
-AAC_SUBDIR= \
-	aac_decoder
+if WITH_FILE_IO
+SUBDIRS += file_reader \
+	file_writer
+endif
 
-SUBDIRS = ${REQUIRED_SUBDIRS}
+if WITH_FLAC
+SUBDIRS += flac_decoder
+endif
+
+if WITH_ICECAST_SERVER
+SUBDIRS += http_renderer
+endif
+
+if WITH_HTTP_SOURCE
+SUBDIRS += http_source
+endif
+
+if WITH_MAD
+SUBDIRS += mp3_decoder
+endif
+
+if WITH_LAME
+SUBDIRS += mp3_encoder
+endif
+
+if WITH_MP3_METADATA_ERASER
+SUBDIRS += mp3_metadata
+endif
+
+if WITH_MP2
+SUBDIRS += mpeg_audio_decoder
+endif
+
+if WITH_OGG
+SUBDIRS += ogg_demuxer \
+	ogg_muxer
+endif
+
+if WITH_OPUS
+SUBDIRS += opus_decoder \
+	opusfile_decoder
+endif
+
+if WITH_LIBSNDFILE
+SUBDIRS += pcm_decoder
+endif
+
+if WITH_PULSEAUDIO
+SUBDIRS += pcm_renderer_pa
+endif
+
+if WITH_VORBIS
+SUBDIRS += vorbis_decoder
+endif
+
+if WITH_VP8
+SUBDIRS += vp8_decoder
+endif
+
+if WITH_WEBM
+SUBDIRS += webm_demuxer
+endif
+
+if WITH_SDL
+SUBDIRS += yuv_renderer
+endif
 
-if WITH_LIBSPOTIFY
-SUBDIRS += ${LIBSPOTIFY_SUBDIR}
+if WITH_SPOTIFY
+SUBDIRS += spotify_source
 endif
 
 if WITH_ALSA
-SUBDIRS += ${ALSA_SUBDIR}
+SUBDIRS += pcm_renderer_alsa
 endif
 
 if WITH_AAC
-SUBDIRS += ${AAC_SUBDIR}
+SUBDIRS += aac_decoder
 endif
