--- a/plugins/configure.ac.orig	2019-08-06 12:39:17.206413212 -0700
+++ b/plugins/configure.ac	2019-08-06 12:42:36.787182672 -0700
@@ -43,6 +43,18 @@ PKG_CHECK_MODULES([CHECK], [check >= 0.9
 
 # Checks for library functions.
 
+AC_ARG_WITH(curl,
+    AS_HELP_STRING([--with-curl],
+        [build with cURL support (default: yes)]),,
+    with_curl=yes)
+AM_CONDITIONAL(WITH_CURL, test "x$with_curl" = xyes)
+
+if test "$with_curl" = yes; then
+    AC_DEFINE(HAVE_CURL, 1, [Support for curl is included])
+else
+    AC_MSG_ERROR([ogg_muxer requires --with-curl])
+fi
+
 AC_ARG_WITH(chromecast,
     AS_HELP_STRING([--with-chromecast],
         [build Chromecast support (default: yes)]),,
