--- a/plugins/http_source/src/Makefile.am.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/plugins/http_source/src/Makefile.am	2019-08-01 17:34:13.042839703 -0700
@@ -19,65 +19,131 @@ libtizhttpsrcdir = $(plugindir)
 
 libtizhttpsrc_LTLIBRARIES = libtizhttpsrc.la
 
-noinst_HEADERS = \
-	httpsrc.h \
+noinst_HEADERS =
+
+if WITH_ICECAST_CLIENT
+noinst_HEADERS += httpsrc.h \
 	httpsrcport.h \
 	httpsrcport_decls.h \
 	httpsrcprc.h \
-	httpsrcprc_decls.h \
-	gmusicprc.h \
-	gmusicprc_decls.h \
-	gmusiccfgport.h \
-	gmusiccfgport_decls.h \
-	scloudprc.h \
+	httpsrcprc_decls.h
+endif
+
+if WITH_SOUNDCLOUD
+noinst_HEADERS += scloudprc.h \
 	scloudprc_decls.h \
 	scloudcfgport.h \
-	scloudcfgport_decls.h \
-	dirbleprc.h \
+	scloudcfgport_decls.h
+endif
+
+if WITH_GMUSIC
+noinst_HEADERS += gmusicprc.h \
+	gmusicprc_decls.h \
+	gmusiccfgport.h \
+	gmusiccfgport_decls.h
+endif
+
+if WITH_DIRBLE
+noinst_HEADERS += dirbleprc.h \
 	dirbleprc_decls.h \
 	dirblecfgport.h \
-	dirblecfgport_decls.h \
-	youtubeprc.h \
+	dirblecfgport_decls.h
+endif
+
+if WITH_YOUTUBE
+noinst_HEADERS += youtubeprc.h \
 	youtubeprc_decls.h \
 	youtubecfgport.h \
-	youtubecfgport_decls.h \
-	plexprc.h \
+	youtubecfgport_decls.h
+endif
+
+if WITH_PLEX
+noinst_HEADERS += plexprc.h \
 	plexprc_decls.h \
 	plexcfgport.h \
 	plexcfgport_decls.h
+endif
 
-libtizhttpsrc_la_SOURCES = \
-	httpsrc.c \
+libtizhttpsrc_la_SOURCES =
+
+if WITH_ICECAST_CLIENT
+libtizhttpsrc_la_SOURCES += httpsrc.c \
 	httpsrcport.c \
-	httpsrcprc.c \
-	gmusicprc.c \
-	gmusiccfgport.c \
-	scloudprc.c \
-	scloudcfgport.c \
-	dirbleprc.c \
-	dirblecfgport.c \
-	youtubeprc.c \
-	youtubecfgport.c \
-	plexprc.c \
+	httpsrcprc.c
+endif
+
+if WITH_GMUSIC
+libtizhttpsrc_la_SOURCES += gmusicprc.c \
+	gmusiccfgport.c
+endif
+
+if WITH_SOUNDCLOUD
+libtizhttpsrc_la_SOURCES += scloudprc.c \
+	scloudcfgport.c
+endif
+
+if WITH_DIRBLE
+libtizhttpsrc_la_SOURCES += dirbleprc.c \
+	dirblecfgport.c
+endif
+
+if WITH_YOUTUBE
+libtizhttpsrc_la_SOURCES += youtubeprc.c \
+	youtubecfgport.c
+endif
+
+if WITH_PLEX
+libtizhttpsrc_la_SOURCES += plexprc.c \
 	plexcfgport.c
+endif
 
 libtizhttpsrc_la_CFLAGS = \
 	@TIZILHEADERS_CFLAGS@ \
 	@TIZPLATFORM_CFLAGS@ \
-	@TIZONIA_CFLAGS@ \
-	@TIZGMUSIC_CFLAGS@ \
-	@TIZSOUNDCLOUD_CFLAGS@ \
-	@TIZDIRBLE_CFLAGS@ \
-	@TIZYOUTUBE_CFLAGS@ \
-	@TIZPLEX_CFLAGS@
+	@TIZONIA_CFLAGS@
+
+if WITH_GMUSIC
+libtizhttpsrc_la_CFLAGS += @TIZGMUSIC_CFLAGS@
+endif
+
+if WITH_SOUNDCLOUD
+libtizhttpsrc_la_CFLAGS += @TIZSOUNDCLOUD_CFLAGS@
+endif
+
+if WITH_DIRBLE
+libtizhttpsrc_la_CFLAGS += @TIZDIRBLE_CFLAGS@
+endif
+
+if WITH_YOUTUBE
+libtizhttpsrc_la_CFLAGS += @TIZYOUTUBE_CFLAGS@
+endif
+
+if WITH_PLEX
+libtizhttpsrc_la_CFLAGS += @TIZPLEX_CFLAGS@
+endif
 
 libtizhttpsrc_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 
 libtizhttpsrc_la_LIBADD = \
 	@TIZPLATFORM_LIBS@ \
-	@TIZONIA_LIBS@ \
-	@TIZGMUSIC_LIBS@ \
-	@TIZSOUNDCLOUD_LIBS@ \
-	@TIZDIRBLE_LIBS@ \
-	@TIZYOUTUBE_LIBS@ \
-	@TIZPLEX_LIBS@
+	@TIZONIA_LIBS@
+
+if WITH_GMUSIC
+libtizhttpsrc_la_LIBADD += @TIZGMUSIC_LIBS@
+endif
+
+if WITH_SOUNDCLOUD
+libtizhttpsrc_la_LIBADD += @TIZSOUNDCLOUD_LIBS@
+endif
+
+if WITH_DIRBLE
+libtizhttpsrc_la_LIBADD += @TIZDIRBLE_LIBS@
+endif
+
+if WITH_YOUTUBE
+libtizhttpsrc_la_LIBADD += @TIZYOUTUBE_LIBS@
+endif
+
+if WITH_PLEX
+libtizhttpsrc_la_LIBADD += @TIZPLEX_LIBS@
+endif
