--- b/player/src/Makefile.am	2020-03-08 14:18:54.000000000 -0700
+++ a/player/src/Makefile.am	2020-04-06 03:28:20.299684497 -0700
@@ -17,24 +17,6 @@
 
 bin_PROGRAMS = tizonia
 
-if WITH_LIBSPOTIFY
-SPOTIFY_SRC= \
-	services/spotify/tizspotifygraph.cpp \
-	services/spotify/tizspotifygraphfsm.cpp \
-	services/spotify/tizspotifygraphops.cpp \
-	services/spotify/tizspotifymgr.cpp
-SPOTIFY_HDR= \
-	services/spotify/tizspotifygraph.hpp \
-	services/spotify/tizspotifygraphfsm.hpp \
-	services/spotify/tizspotifygraphops.hpp \
-	services/spotify/tizspotifyconfig.hpp \
-	services/spotify/tizspotifymgr.hpp
-else
-SPOTIFY_SRC=
-SPOTIFY_HDR=
-endif
-
-
 noinst_HEADERS = \
 	tizomxutil.hpp \
 	tizplayapp.hpp \
@@ -65,59 +47,126 @@ noinst_HEADERS = \
 	tizplaybackevents.hpp \
 	tizprogressdisplay.hpp \
 	decoders/tizdecgraphmgr.hpp \
-	decoders/tizdecgraph.hpp \
-	decoders/tizmp3graph.hpp \
-	decoders/tizaacgraph.hpp \
-	decoders/tizopusgraph.hpp \
-	decoders/tizoggopusgraph.hpp \
-	decoders/tizvorbisgraph.hpp \
-	decoders/tizflacgraph.hpp \
-	decoders/tizoggflacgraph.hpp \
-	decoders/tizpcmgraph.hpp \
-	decoders/tizmpeggraph.hpp \
-	httpserv/tizhttpservconfig.hpp \
+	decoders/tizdecgraph.hpp
+
+if WITH_MP3
+noinst_HEADERS += decoders/tizmp3graph.hpp
+endif
+
+if WITH_AAC
+noinst_HEADERS += decoders/tizaacgraph.hpp
+endif
+
+if WITH_OPUS
+noinst_HEADERS += decoders/tizopusgraph.hpp
+endif
+
+if WITH_OGG
+if WITH_OPUS
+noinst_HEADERS += decoders/tizoggopusgraph.hpp
+endif
+endif
+
+if WITH_VORBIS
+noinst_HEADERS += decoders/tizvorbisgraph.hpp
+endif
+
+if WITH_FLAC
+noinst_HEADERS += decoders/tizflacgraph.hpp
+endif
+
+if WITH_OGG
+if WITH_FLAC
+noinst_HEADERS += decoders/tizoggflacgraph.hpp
+endif
+endif
+
+if WITH_LIBSNDFILE
+noinst_HEADERS += decoders/tizpcmgraph.hpp
+endif
+
+if WITH_MP2
+noinst_HEADERS += decoders/tizmpeggraph.hpp
+endif
+
+if WITH_ICECAST_SERVER
+noinst_HEADERS += httpserv/tizhttpservconfig.hpp \
 	httpserv/tizhttpservgraph.hpp \
 	httpserv/tizhttpservgraphfsm.hpp \
 	httpserv/tizhttpservgraphops.hpp \
-	httpserv/tizhttpservmgr.hpp \
-	httpclnt/tizhttpclntmgr.hpp \
+	httpserv/tizhttpservmgr.hpp
+endif
+
+if WITH_ICECAST_CLIENT
+noinst_HEADERS += httpclnt/tizhttpclntmgr.hpp \
 	httpclnt/tizhttpclntgraph.hpp \
 	httpclnt/tizhttpclntgraphfsm.hpp \
-	httpclnt/tizhttpclntgraphops.hpp \
+	httpclnt/tizhttpclntgraphops.hpp
+endif
+
+noinst_HEADERS += \
 	services/tizservicegraph.hpp \
-	services/tizservicegraphfsm.hpp \
-	${SPOTIFY_HDR} \
-	services/googlemusic/tizgmusicgraph.hpp \
+	services/tizservicegraphfsm.hpp
+
+if WITH_SPOTIFY
+noinst_HEADERS += services/spotify/tizspotifygraph.hpp \
+	services/spotify/tizspotifygraphfsm.hpp \
+	services/spotify/tizspotifygraphops.hpp \
+	services/spotify/tizspotifyconfig.hpp \
+	services/spotify/tizspotifymgr.hpp
+endif
+
+if WITH_GMUSIC
+noinst_HEADERS += services/googlemusic/tizgmusicgraph.hpp \
 	services/googlemusic/tizgmusicgraphops.hpp \
 	services/googlemusic/tizgmusicconfig.hpp \
-	services/googlemusic/tizgmusicmgr.hpp \
-	services/soundcloud/tizscloudgraph.hpp \
+	services/googlemusic/tizgmusicmgr.hpp
+endif
+
+if WITH_SOUNDCLOUD
+noinst_HEADERS += services/soundcloud/tizscloudgraph.hpp \
 	services/soundcloud/tizscloudgraphops.hpp \
 	services/soundcloud/tizscloudconfig.hpp \
-	services/soundcloud/tizscloudmgr.hpp \
-	services/youtube/tizyoutubegraph.hpp \
+	services/soundcloud/tizscloudmgr.hpp
+endif
+
+if WITH_YOUTUBE
+noinst_HEADERS += services/youtube/tizyoutubegraph.hpp \
 	services/youtube/tizyoutubegraphfsm.hpp \
 	services/youtube/tizyoutubegraphops.hpp \
 	services/youtube/tizyoutubeconfig.hpp \
-	services/youtube/tizyoutubemgr.hpp \
-	services/plex/tizplexgraph.hpp \
+	services/youtube/tizyoutubemgr.hpp
+endif
+
+if WITH_PLEX
+noinst_HEADERS += services/plex/tizplexgraph.hpp \
 	services/plex/tizplexgraphops.hpp \
 	services/plex/tizplexconfig.hpp \
-	services/plex/tizplexmgr.hpp \
-	services/tunein/tiztuneingraph.hpp \
+	services/plex/tizplexmgr.hpp
+endif
+
+if WITH_TUNEIN
+noinst_HEADERS += services/tunein/tiztuneingraph.hpp \
 	services/tunein/tiztuneingraphfsm.hpp \
 	services/tunein/tiztuneingraphops.hpp \
 	services/tunein/tiztuneinconfig.hpp \
-	services/tunein/tiztuneinmgr.hpp \
-	mpris/tizmpriscbacks.hpp \
+	services/tunein/tiztuneinmgr.hpp
+endif
+
+if WITH_DBUS
+noinst_HEADERS += mpris/tizmpriscbacks.hpp \
 	mpris/tizmprisprops.hpp \
 	mpris/tizmprisif.hpp \
-	mpris/tizmprismgr.hpp \
-	services/chromecast/tizchromecastgraph.hpp \
+	mpris/tizmprismgr.hpp
+endif
+
+if WITH_CHROMECAST
+noinst_HEADERS += services/chromecast/tizchromecastgraph.hpp \
 	services/chromecast/tizchromecastgraphfsm.hpp \
 	services/chromecast/tizchromecastgraphops.hpp \
 	services/chromecast/tizchromecastconfig.hpp \
 	services/chromecast/tizchromecastmgr.hpp
+endif
 
 
 tizonia_SOURCES = \
@@ -140,53 +189,119 @@ tizonia_SOURCES = \
 	tizgraphcmd.cpp \
 	tizgraph.cpp \
 	decoders/tizdecgraphmgr.cpp \
-	decoders/tizdecgraph.cpp \
-	decoders/tizmp3graph.cpp \
-	decoders/tizaacgraph.cpp \
-	decoders/tizopusgraph.cpp \
-	decoders/tizoggopusgraph.cpp \
-	decoders/tizvorbisgraph.cpp \
-	decoders/tizflacgraph.cpp \
-	decoders/tizoggflacgraph.cpp \
-	decoders/tizpcmgraph.cpp \
-	decoders/tizmpeggraph.cpp \
-	httpserv/tizhttpservmgr.cpp \
+	decoders/tizdecgraph.cpp
+
+if WITH_MP3
+tizonia_SOURCES += decoders/tizmp3graph.cpp
+endif
+
+if WITH_AAC
+tizonia_SOURCES += decoders/tizaacgraph.cpp
+endif
+
+if WITH_OPUS
+tizonia_SOURCES += decoders/tizopusgraph.cpp
+endif
+
+if WITH_OGG
+if WITH_OPUS
+tizonia_SOURCES += decoders/tizoggopusgraph.cpp
+endif
+endif
+
+if WITH_VORBIS
+tizonia_SOURCES += decoders/tizvorbisgraph.cpp
+endif
+
+if WITH_FLAC
+tizonia_SOURCES += decoders/tizflacgraph.cpp
+endif
+
+if WITH_OGG
+if WITH_FLAC
+tizonia_SOURCES += decoders/tizoggflacgraph.cpp
+endif
+endif
+
+if WITH_LIBSNDFILE
+tizonia_SOURCES += decoders/tizpcmgraph.cpp
+endif
+
+if WITH_MP2
+tizonia_SOURCES += decoders/tizmpeggraph.cpp
+endif
+
+if WITH_ICECAST_SERVER
+tizonia_SOURCES += httpserv/tizhttpservmgr.cpp \
 	httpserv/tizhttpservgraph.cpp \
 	httpserv/tizhttpservgraphfsm.cpp \
-	httpserv/tizhttpservgraphops.cpp \
-	httpclnt/tizhttpclntmgr.cpp \
+	httpserv/tizhttpservgraphops.cpp
+endif
+
+if WITH_ICECAST_CLIENT
+tizonia_SOURCES += httpclnt/tizhttpclntmgr.cpp \
 	httpclnt/tizhttpclntgraph.cpp \
 	httpclnt/tizhttpclntgraphfsm.cpp \
-	httpclnt/tizhttpclntgraphops.cpp \
-	services/tizservicegraph.cpp \
-	services/tizservicegraphfsm.cpp \
-	${SPOTIFY_SRC} \
-	services/googlemusic/tizgmusicgraph.cpp \
+	httpclnt/tizhttpclntgraphops.cpp
+endif
+
+tizonia_SOURCES += services/tizservicegraph.cpp \
+	services/tizservicegraphfsm.cpp
+
+if WITH_SPOTIFY
+tizonia_SOURCES += services/spotify/tizspotifygraph.cpp \
+	services/spotify/tizspotifygraphfsm.cpp \
+	services/spotify/tizspotifygraphops.cpp \
+	services/spotify/tizspotifymgr.cpp
+endif
+
+if WITH_GMUSIC
+tizonia_SOURCES += services/googlemusic/tizgmusicgraph.cpp \
 	services/googlemusic/tizgmusicgraphops.cpp \
-	services/googlemusic/tizgmusicmgr.cpp \
-	services/soundcloud/tizscloudgraph.cpp \
+	services/googlemusic/tizgmusicmgr.cpp
+endif
+
+if WITH_SOUNDCLOUD
+tizonia_SOURCES += services/soundcloud/tizscloudgraph.cpp \
 	services/soundcloud/tizscloudgraphops.cpp \
-	services/soundcloud/tizscloudmgr.cpp \
-	services/youtube/tizyoutubegraph.cpp \
+	services/soundcloud/tizscloudmgr.cpp
+endif
+
+if WITH_YOUTUBE
+tizonia_SOURCES += services/youtube/tizyoutubegraph.cpp \
 	services/youtube/tizyoutubegraphfsm.cpp \
 	services/youtube/tizyoutubegraphops.cpp \
-	services/youtube/tizyoutubemgr.cpp \
-	services/plex/tizplexgraph.cpp \
+	services/youtube/tizyoutubemgr.cpp
+endif
+
+if WITH_PLEX
+tizonia_SOURCES += services/plex/tizplexgraph.cpp \
 	services/plex/tizplexgraphops.cpp \
-	services/plex/tizplexmgr.cpp \
-	services/tunein/tiztuneingraph.cpp \
+	services/plex/tizplexmgr.cpp
+endif
+
+if WITH_TUNEIN
+tizonia_SOURCES += services/tunein/tiztuneingraph.cpp \
 	services/tunein/tiztuneingraphfsm.cpp \
 	services/tunein/tiztuneingraphops.cpp \
-	services/tunein/tiztuneinmgr.cpp \
-	services/chromecast/tizchromecastgraph.cpp \
+	services/tunein/tiztuneinmgr.cpp
+endif
+
+if WITH_CHROMECAST
+tizonia_SOURCES += services/chromecast/tizchromecastgraph.cpp \
 	services/chromecast/tizchromecastgraphfsm.cpp \
 	services/chromecast/tizchromecastgraphops.cpp \
-	services/chromecast/tizchromecastmgr.cpp \
-	tizplaybackevents.cpp \
-	tizprogressdisplay.cpp \
-	mpris/tizmprismgr.cpp \
+	services/chromecast/tizchromecastmgr.cpp
+endif
+
+tizonia_SOURCES += tizplaybackevents.cpp \
+	tizprogressdisplay.cpp
+
+if WITH_DBUS
+tizonia_SOURCES += mpris/tizmprismgr.cpp \
 	mpris/tizmprisprops.cpp \
 	mpris/tizmprisif.cpp
+endif
 
 tizonia_CPPFLAGS = \
 	@BOOST_CPPFLAGS@ \
@@ -194,13 +309,29 @@ tizonia_CPPFLAGS = \
 	@TIZPLATFORM_CFLAGS@ \
 	-I$(top_srcdir)/dbus \
 	-I$(top_srcdir)/src \
-	-I$(top_srcdir)/src/services \
-	-I$(top_srcdir)/src/services/googlemusic \
-	-I$(top_srcdir)/src/services/soundcloud \
-	-I$(top_srcdir)/src/services/youtube \
-	-I$(top_srcdir)/src/services/plex \
-	-I$(top_srcdir)/src/services/tunein \
-	@TAGLIB_CFLAGS@ \
+	-I$(top_srcdir)/src/services
+
+if WITH_GMUSIC
+tizonia_CPPFLAGS += -I$(top_srcdir)/src/services/googlemusic
+endif
+
+if WITH_SOUNDCLOUD
+tizonia_CPPFLAGS += -I$(top_srcdir)/src/services/soundcloud
+endif
+
+if WITH_YOUTUBE
+tizonia_CPPFLAGS += -I$(top_srcdir)/src/services/youtube
+endif
+
+if WITH_PLEX
+tizonia_CPPFLAGS += -I$(top_srcdir)/src/services/plex
+endif
+
+if WITH_TUNEIN
+tizonia_CPPFLAGS += -I$(top_srcdir)/src/services/tunein
+endif
+
+tizonia_CPPFLAGS += @TAGLIB_CFLAGS@ \
 	@LIBMEDIAINFO_CFLAGS@ \
 	@TIZDBUSCPLUSPLUS_CFLAGS@ \
 	-D__STDC_CONSTANT_MACROS
@@ -212,7 +343,12 @@ tizonia_LDADD = \
 	@BOOST_THREAD_LIB@ \
 	@BOOST_CHRONO_LIB@ \
 	@TAGLIB_LIBS@ \
-	@LIBMEDIAINFO_LIBS@ \
-	@TIZDBUSCPLUSPLUS_LIBS@ \
+	@LIBMEDIAINFO_LIBS@
+
+if WITH_DBUS
+tizonia_LDADD += @TIZDBUSCPLUSPLUS_LIBS@
+endif
+
+tizonia_LDADD += \
 	@TIZPLATFORM_LIBS@ \
 	@TIZCORE_LIBS@
Only in tizonia-openmax-il-0.21.0/player/src: Makefile.am.orig
Only in tizonia-openmax-il-0.21.0/player/src: Makefile.am.rej
