--- a/plugins/http_source/configure.ac.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/plugins/http_source/configure.ac	2019-08-01 20:10:32.901907110 -0700
@@ -158,6 +158,54 @@ AC_SUBST([plugindir], ['${libdir}/tizoni
 
 # Checks for library functions.
 
+AC_ARG_WITH(curl,
+    AS_HELP_STRING([--with-curl],
+        [build with cURL support (default: yes)]),,
+    with_curl=yes)
+AM_CONDITIONAL(WITH_CURL, test "x$with_curl" = xyes)
+
+if test "$with_curl" = yes; then
+    AC_DEFINE(HAVE_CURL, 1, [Support for curl is included])
+else
+    AC_MSG_ERROR([http_source requires --with-curl.])
+fi
+
+AC_ARG_WITH(icecast-client,
+    AS_HELP_STRING([--with-icecast-client],
+        [build Icecast/SHOUTcast client support (default: yes)]),,
+    with_icecast_client=yes)
+AM_CONDITIONAL(WITH_ICECAST_CLIENT, test "x$with_icecast_client" = xyes)
+
+AC_ARG_WITH(soundcloud,
+    AS_HELP_STRING([--with-soundcloud],
+        [build SoundCloud support (default: yes)]),,
+    with_soundcloud=yes)
+AM_CONDITIONAL(WITH_SOUNDCLOUD, test "x$with_soundcloud" = xyes)
+
+AC_ARG_WITH(gmusic,
+    AS_HELP_STRING([--with-gmusic],
+        [build Google Music support (default: yes)]),,
+    with_gmusic=yes)
+AM_CONDITIONAL(WITH_GMUSIC, test "x$with_gmusic" = xyes)
+
+AC_ARG_WITH(dirble,
+    AS_HELP_STRING([--with-dirble],
+        [build Dirble support (default: yes)]),,
+    with_dirble=yes)
+AM_CONDITIONAL(WITH_DIRBLE, test "x$with_dirble" = xyes)
+
+AC_ARG_WITH(youtube,
+    AS_HELP_STRING([--with-youtube],
+        [build YouTube support (default: yes)]),,
+    with_youtube=yes)
+AM_CONDITIONAL(WITH_YOUTUBE, test "x$with_youtube" = xyes)
+
+AC_ARG_WITH(plex,
+    AS_HELP_STRING([--with-plex],
+        [build Plex support (default: yes)]),,
+    with_plex=yes)
+AM_CONDITIONAL(WITH_PLEX, test "x$with_plex" = xyes)
+
 AC_CONFIG_FILES([Makefile
                  src/Makefile])
 
