--- b/configure.ac	2020-04-06 04:50:14.770610399 -0700
+++ a/configure.ac	2020-04-06 04:50:48.031904731 -0700
@@ -220,7 +220,7 @@ AC_ARG_WITH(webm,
 
 if test -z "$with_http_source" ; then
     if test "$with_icecast_client" = yes || test "$with_gmusic" = yes || test "$with_plex" = yes || test "$with_soundcloud" = yes || test "$with_tunein" = yes || test "$with_youtube" = yes ; then
-        AC_MSG_ERROR([icecast-client, dirble, gmusic, plex, soundcloud, youtube requires --with-http-source.])
+        AC_MSG_ERROR([icecast-client, gmusic, plex, soundcloud, tunein, youtube requires --with-http-source.])
     fi
 fi
 
