diff -urp tizonia-openmax-il-0.21.0.orig/clients/chromecast/configure.ac tizonia-openmax-il-0.21.0/clients/chromecast/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/chromecast/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/chromecast/configure.ac	2020-04-10 21:45:18.416183461 -0700
@@ -35,6 +35,12 @@ LT_INIT
 # Checks for libraries.
 AM_PATH_PYTHON([3.5])
 
+AC_ARG_WITH(chromecast,
+    AS_HELP_STRING([--with-chromecast],
+        [build Chromecast support (default: yes)]),,
+    with_chromecast=yes)
+AM_CONDITIONAL(WITH_CHROMECAST, test "x$with_chromecast" = xyes)
+
 # Checks for header files.
 
 # Checks for typedefs, structures, and compiler characteristics.
diff -urp tizonia-openmax-il-0.21.0.orig/clients/gmusic/configure.ac tizonia-openmax-il-0.21.0/clients/gmusic/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/gmusic/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/gmusic/configure.ac	2020-04-10 21:45:28.056555828 -0700
@@ -35,6 +35,12 @@ LT_INIT
 # Checks for libraries.
 AM_PATH_PYTHON([3.5])
 
+AC_ARG_WITH(gmusic,
+    AS_HELP_STRING([--with-gmusic],
+        [build Google Music support (default: yes)]),,
+    with_gmusic=yes)
+AM_CONDITIONAL(WITH_GMUSIC, test "x$with_gmusic" = xyes)
+
 # Checks for header files.
 
 # Checks for typedefs, structures, and compiler characteristics.
diff -urp tizonia-openmax-il-0.21.0.orig/clients/plex/configure.ac tizonia-openmax-il-0.21.0/clients/plex/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/plex/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/plex/configure.ac	2020-04-10 21:45:37.556922801 -0700
@@ -35,6 +35,12 @@ LT_INIT
 # Checks for libraries.
 AM_PATH_PYTHON([3.5])
 
+AC_ARG_WITH(plex,
+    AS_HELP_STRING([--with-plex],
+        [build Plex support (default: yes)]),,
+    with_plex=yes)
+AM_CONDITIONAL(WITH_PLEX, test "x$with_plex" = xyes)
+
 # Checks for header files.
 
 # Checks for typedefs, structures, and compiler characteristics.
diff -urp tizonia-openmax-il-0.21.0.orig/clients/soundcloud/configure.ac tizonia-openmax-il-0.21.0/clients/soundcloud/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/soundcloud/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/soundcloud/configure.ac	2020-04-10 21:45:49.537385594 -0700
@@ -35,6 +35,12 @@ LT_INIT
 # Checks for libraries.
 AM_PATH_PYTHON([3.5])
 
+AC_ARG_WITH(soundcloud,
+    AS_HELP_STRING([--with-soundcloud],
+        [build SoundCloud support (default: yes)]),,
+    with_soundcloud=yes)
+AM_CONDITIONAL(WITH_SOUNDCLOUD, test "x$with_soundcloud" = xyes)
+
 # Checks for header files.
 
 # Checks for typedefs, structures, and compiler characteristics.
diff -urp tizonia-openmax-il-0.21.0.orig/clients/spotify/configure.ac tizonia-openmax-il-0.21.0/clients/spotify/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/spotify/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/spotify/configure.ac	2020-04-10 21:45:59.877785048 -0700
@@ -35,6 +35,12 @@ LT_INIT
 # Checks for libraries.
 AM_PATH_PYTHON([3.5])
 
+AC_ARG_WITH(spotify,
+    AS_HELP_STRING([--with-spotify],
+        [build Spotify support (default: yes)]),,
+    with_spotify=yes)
+AM_CONDITIONAL(WITH_SPOTIFY, test "x$with_spotify" = xyes)
+
 # Checks for header files.
 
 # Checks for typedefs, structures, and compiler characteristics.
diff -urp tizonia-openmax-il-0.21.0.orig/clients/tunein/configure.ac tizonia-openmax-il-0.21.0/clients/tunein/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/tunein/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/tunein/configure.ac	2020-04-10 21:46:10.018176793 -0700
@@ -35,6 +35,12 @@ LT_INIT
 # Checks for libraries.
 AM_PATH_PYTHON([3.5])
 
+AC_ARG_WITH(tunein,
+    AS_HELP_STRING([--with-tunein],
+        [build TuneIn support (default: yes)]),,
+    with_tunein=yes)
+AM_CONDITIONAL(WITH_TUNEIN, test "x$with_tunein" = xyes)
+
 # Checks for header files.
 
 # Checks for typedefs, structures, and compiler characteristics.
diff -urp tizonia-openmax-il-0.21.0.orig/clients/youtube/configure.ac tizonia-openmax-il-0.21.0/clients/youtube/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/youtube/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/youtube/configure.ac	2020-04-10 21:46:23.178685235 -0700
@@ -35,6 +35,12 @@ LT_INIT
 # Checks for libraries.
 AM_PATH_PYTHON([3.5])
 
+AC_ARG_WITH(youtube,
+    AS_HELP_STRING([--with-youtube],
+        [build YouTube support (default: yes)]),,
+    with_youtube=yes)
+AM_CONDITIONAL(WITH_YOUTUBE, test "x$with_youtube" = xyes)
+
 # Checks for header files.
 
 # Checks for typedefs, structures, and compiler characteristics.
