--- b/clients/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ a/clients/configure.ac	2020-04-09 14:38:59.564976194 -0700
@@ -23,6 +23,47 @@ AM_INIT_AUTOMAKE([foreign color-tests si
 # Checks for programs.
 
 # Checks for libraries.
+AC_ARG_WITH(gmusic,
+    AS_HELP_STRING([--with-gmusic],
+        [build Google Music support (default: yes)]),,
+    with_gmusic=yes)
+AM_CONDITIONAL(WITH_GMUSIC, test "x$with_gmusic" = xyes)
+
+AC_ARG_WITH(soundcloud,
+    AS_HELP_STRING([--with-soundcloud],
+        [build SoundCloud support (default: yes)]),,
+    with_soundcloud=yes)
+AM_CONDITIONAL(WITH_SOUNDCLOUD, test "x$with_soundcloud" = xyes)
+
+AC_ARG_WITH(youtube,
+    AS_HELP_STRING([--with-youtube],
+        [build YouTube support (default: yes)]),,
+    with_youtube=yes)
+AM_CONDITIONAL(WITH_YOUTUBE, test "x$with_youtube" = xyes)
+
+AC_ARG_WITH(plex,
+    AS_HELP_STRING([--with-plex],
+        [build Plex support (default: yes)]),,
+    with_plex=yes)
+AM_CONDITIONAL(WITH_PLEX, test "x$with_plex" = xyes)
+
+AC_ARG_WITH(chromecast,
+    AS_HELP_STRING([--with-chromecast],
+        [build Chromecast support (default: yes)]),,
+    with_chromecast=yes)
+AM_CONDITIONAL(WITH_CHROMECAST, test "x$with_chromecast" = xyes)
+
+AC_ARG_WITH(spotify,
+    AS_HELP_STRING([--with-spotify],
+        [build Spotify support (default: yes)]),,
+    with_spotify=yes)
+AM_CONDITIONAL(WITH_SPOTIFY, test "x$with_spotify" = xyes)
+
+AC_ARG_WITH(tunein,
+    AS_HELP_STRING([--with-tunein],
+        [build TuneIn support (default: yes)]),,
+    with_tunein=yes)
+AM_CONDITIONAL(WITH_TUNEIN, test "x$with_tunein" = xyes)
 
 # Checks for header files.
 
@@ -31,5 +72,42 @@ AM_INIT_AUTOMAKE([foreign color-tests si
 # Checks for library functions.
 
 AC_CONFIG_FILES([Makefile])
-AC_CONFIG_SUBDIRS([gmusic soundcloud youtube plex chromecast spotify tunein])
+if test "$with_gmusic" = yes; then
+	AC_CONFIG_SUBDIRS([gmusic])
+fi
+if test "$with_soundcloud" = yes; then
+	AC_CONFIG_SUBDIRS([soundcloud])
+fi
+if test "$with_youtube" = yes; then
+	AC_CONFIG_SUBDIRS([youtube])
+fi
+if test "$with_plex" = yes; then
+	AC_CONFIG_SUBDIRS([plex])
+fi
+if test "$with_chromecast" = yes; then
+	AC_CONFIG_SUBDIRS([chromecast])
+fi
+if test "$with_spotify" = yes; then
+	AC_CONFIG_SUBDIRS([spotify])
+fi
+if test "$with_tunein" = yes; then
+	AC_CONFIG_SUBDIRS([tunein])
+fi
+
 AC_OUTPUT
+
+AC_MSG_RESULT([
+------------------------------------------------------------------------
+
+  Services
+
+    Google Music: ................ ${with_gmusic}
+    SoundCloud: .................. ${with_soundcloud}
+    YouTube: ..................... ${with_youtube}
+    Plex: ........................ ${with_plex}
+    Chromecast: .................. ${with_chromecast}
+    Spotify: ..................... ${with_spotify}
+    TuneIn: ...................... ${with_tunein}
+
+------------------------------------------------------------------------
+])
--- b/clients/Makefile.am	2020-03-08 14:18:54.000000000 -0700
+++ a/clients/Makefile.am	2020-04-09 14:32:00.708055140 -0700
@@ -15,4 +15,25 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with Tizonia.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = gmusic soundcloud youtube plex chromecast spotify tunein
+SUBDIRS =
+if WITH_GMUSIC
+SUBDIRS += gmusic
+endif
+if WITH_SOUNDCLOUD
+SUBDIRS += soundcloud
+endif
+if WITH_YOUTUBE
+SUBDIRS += youtube
+endif
+if WITH_PLEX
+SUBDIRS += plex
+endif
+if WITH_CHROMECAST
+SUBDIRS += chromecast
+endif
+if WITH_SPOTIFY
+SUBDIRS += spotify
+endif
+if WITH_TUNEIN
+SUBDIRS += tunein
+endif
