--- a/libtizplatform/src/Makefile.am.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/libtizplatform/src/Makefile.am	2019-08-01 17:50:47.939521652 -0700
@@ -55,8 +55,11 @@ libtizplatform_include_HEADERS = \
 	tizhttp.h \
 	tizlimits.h \
 	tizprintf.h \
-	tizshufflelst.h \
-	tizurltransfer.h
+	tizshufflelst.h
+
+if WITH_CURL
+libtizplatform_include_HEADERS += tizurltransfer.h
+endif
 
 libtizplatform_la_SOURCES = \
 	http-parser/http_parser.c \
@@ -80,22 +83,33 @@ libtizplatform_la_SOURCES = \
 	tizhttp.c \
 	tizlimits.c \
 	tizprintf.c \
-	tizshufflelst.c \
-	tizurltransfer.c
+	tizshufflelst.c
+
+if WITH_CURL
+libtizplatform_la_SOURCES += tizurltransfer.c
+endif
 
 libtizplatform_la_CFLAGS = \
 	$(AM_CFLAGS) \
-	@TIZILHEADERS_CFLAGS@ \
-	@LIBCURL_CFLAGS@ \
-	@LOG4C_CFLAGS@
+	@TIZILHEADERS_CFLAGS@
+
+if WITH_CURL
+libtizplatform_la_CFLAGS += @LIBCURL_CFLAGS@
+endif
+
+libtizplatform_la_CFLAGS += @LOG4C_CFLAGS@
 
 libtizplatform_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 
 libtizplatform_la_LIBADD = \
 	-lpthread \
-	@LOG4C_LIBS@ \
-	@LIBCURL_LIBS@ \
-	@UUID_LIBS@
+	@LOG4C_LIBS@
+
+if WITH_CURL
+libtizplatform_la_LIBADD += @LIBCURL_LIBS@
+endif
+
+libtizplatform_la_LIBADD += @UUID_LIBS@
 
 do_subst = sed -e 's,[@]abs_top_builddir[@],$(abs_top_builddir),g' \
 	-e 's,[@]localstatedir[@],$(localstatedir),g' \
