--- a/configure.ac.orig	2019-12-11 13:43:20.654938878 -0800
+++ b/configure.ac	2019-12-11 13:46:10.881487402 -0800
@@ -136,6 +136,7 @@ AC_ARG_WITH(icecast-client,
     AS_HELP_STRING([--with-icecast-client],
         [build Icecast/SHOUTcast client support (default: yes)]),,
     with_icecast_client=yes)
+AM_CONDITIONAL(WITH_ICECAST_CLIENT, test "x$with_icecast_client" = xyes)
 
 AC_ARG_WITH(icecast-server,
     AS_HELP_STRING([--with-icecast-server],
@@ -226,9 +227,13 @@ fi
 AC_CONFIG_FILES([Makefile])
 
 AC_CONFIG_SUBDIRS([3rdparty
-                   include
-                   clients
-                   libtizplatform])
+                   include])
+
+if test "$with_icecast_client" = yes || test "$with_dirble" = yes || test "$with_gmusic" = yes || test "$with_plex" = yes || test "$with_soundcloud" = yes || test "$with_youtube" = yes ; then
+    AC_CONFIG_SUBDIRS([clients])
+fi
+
+AC_CONFIG_SUBDIRS([libtizplatform])
 
 if test "$with_chromecast" = yes; then
 	AC_CONFIG_SUBDIRS([cast])
--- a/Makefile.am.orig	2019-12-11 13:43:20.714941187 -0800
+++ b/Makefile.am	2019-12-11 13:52:29.846043533 -0800
@@ -30,7 +30,33 @@ if ENABLE_PLAYER
 SUBDIRS += config
 endif
 
-SUBDIRS += include clients libtizplatform
+SUBDIRS += include
+
+if WITH_ICECAST_CLIENT
+SUBDIRS += clients
+else
+if WITH_DIRBLE
+SUBDIRS += clients
+else
+if WITH_GMUSIC
+SUBDIRS += clients
+else
+if WITH_PLEX
+SUBDIRS += clients
+else
+if WITH_SOUNDCLOUD
+SUBDIRS += clients
+else
+if WITH_YOUTUBE
+SUBDIRS += clients
+endif # youtube
+endif # soundcloud
+endif # plex
+endif # gmusic
+endif # dirble
+endif # icecast client
+
+SUBDIRS += libtizplatform
 
 if WITH_CHROMECAST
 SUBDIRS += cast
