diff -urp tizonia-openmax-il-0.21.0.orig/clients/chromecast/libtizchromecast/configure.ac tizonia-openmax-il-0.21.0/clients/chromecast/libtizchromecast/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/chromecast/libtizchromecast/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/chromecast/libtizchromecast/configure.ac	2020-04-10 21:30:51.072656590 -0700
@@ -69,6 +69,12 @@ AX_PYTHON_DEVEL([>= '3.5'])
 AX_BOOST_BASE([1.54],, [AC_MSG_ERROR([libtizchromecast needs Boost 1.54])])
 AX_BOOST_PYTHON
 
+AC_ARG_WITH(chromecast,
+    AS_HELP_STRING([--with-chromecast],
+        [build Chromecast support (default: yes)]),,
+    with_chromecast=yes)
+AM_CONDITIONAL(WITH_CHROMECAST, test "x$with_chromecast" = xyes)
+
 # Checks for header files.
 #AC_CHECK_HEADER_STDBOOL
 
diff -urp tizonia-openmax-il-0.21.0.orig/clients/gmusic/libtizgmusic/configure.ac tizonia-openmax-il-0.21.0/clients/gmusic/libtizgmusic/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/gmusic/libtizgmusic/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/gmusic/libtizgmusic/configure.ac	2020-04-10 21:31:34.174335942 -0700
@@ -69,6 +69,12 @@ AX_PYTHON_DEVEL([>= '3.5'])
 AX_BOOST_BASE([1.54],, [AC_MSG_ERROR([libtizgmusic needs Boost 1.54])])
 AX_BOOST_PYTHON
 
+AC_ARG_WITH(gmusic,
+    AS_HELP_STRING([--with-gmusic],
+        [build Google Music support (default: yes)]),,
+    with_gmusic=yes)
+AM_CONDITIONAL(WITH_GMUSIC, test "x$with_gmusic" = xyes)
+
 # Checks for header files.
 #AC_CHECK_HEADER_STDBOOL
 
diff -urp tizonia-openmax-il-0.21.0.orig/clients/plex/libtizplex/configure.ac tizonia-openmax-il-0.21.0/clients/plex/libtizplex/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/plex/libtizplex/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/plex/libtizplex/configure.ac	2020-04-10 21:30:06.330913480 -0700
@@ -67,6 +67,12 @@ AX_PYTHON_DEVEL([>= '3.5'])
 AX_BOOST_BASE([1.54],, [AC_MSG_ERROR([libtizplex needs Boost 1.54])])
 AX_BOOST_PYTHON
 
+AC_ARG_WITH(plex,
+    AS_HELP_STRING([--with-plex],
+        [build Plex support (default: yes)]),,
+    with_plex=yes)
+AM_CONDITIONAL(WITH_PLEX, test "x$with_plex" = xyes)
+
 # Checks for header files.
 #AC_CHECK_HEADER_STDBOOL
 
diff -urp tizonia-openmax-il-0.21.0.orig/clients/soundcloud/libtizsoundcloud/configure.ac tizonia-openmax-il-0.21.0/clients/soundcloud/libtizsoundcloud/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/soundcloud/libtizsoundcloud/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/soundcloud/libtizsoundcloud/configure.ac	2020-04-10 21:28:59.498310007 -0700
@@ -69,6 +69,12 @@ AX_PYTHON_MODULE(soundcloud, [tizsoundcl
 AX_BOOST_BASE([1.54],, [AC_MSG_ERROR([libtizsoundcloud needs Boost 1.54])])
 AX_BOOST_PYTHON
 
+AC_ARG_WITH(soundcloud,
+    AS_HELP_STRING([--with-soundcloud],
+        [build SoundCloud support (default: yes)]),,
+    with_soundcloud=yes)
+AM_CONDITIONAL(WITH_SOUNDCLOUD, test "x$with_soundcloud" = xyes)
+
 # Checks for header files.
 #AC_CHECK_HEADER_STDBOOL
 
diff -urp tizonia-openmax-il-0.21.0.orig/clients/spotify/libtizspotify/configure.ac tizonia-openmax-il-0.21.0/clients/spotify/libtizspotify/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/spotify/libtizspotify/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/spotify/libtizspotify/configure.ac	2020-04-10 21:28:31.537220880 -0700
@@ -67,6 +67,12 @@ AX_PYTHON_DEVEL([>= '3.5'])
 AX_BOOST_BASE([1.54],, [AC_MSG_ERROR([libtizspotify needs Boost 1.54])])
 AX_BOOST_PYTHON
 
+AC_ARG_WITH(spotify,
+    AS_HELP_STRING([--with-spotify],
+        [build Spotify support (default: yes)]),,
+    with_spotify=yes)
+AM_CONDITIONAL(WITH_SPOTIFY, test "x$with_spotify" = xyes)
+
 # Checks for header files.
 #AC_CHECK_HEADER_STDBOOL
 
diff -urp tizonia-openmax-il-0.21.0.orig/clients/tunein/libtiztunein/configure.ac tizonia-openmax-il-0.21.0/clients/tunein/libtiztunein/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/tunein/libtiztunein/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/tunein/libtiztunein/configure.ac	2020-04-10 21:30:21.691511902 -0700
@@ -68,6 +68,12 @@ AX_PYTHON_DEVEL([>= '3.5'])
 AX_BOOST_BASE([1.54],, [AC_MSG_ERROR([libtiztunein needs Boost 1.54])])
 AX_BOOST_PYTHON
 
+AC_ARG_WITH(tunein,
+    AS_HELP_STRING([--with-tunein],
+        [build TuneIn support (default: yes)]),,
+    with_tunein=yes)
+AM_CONDITIONAL(WITH_TUNEIN, test "x$with_tunein" = xyes)
+
 # Checks for header files.
 #AC_CHECK_HEADER_STDBOOL
 
diff -urp tizonia-openmax-il-0.21.0.orig/clients/youtube/libtizyoutube/configure.ac tizonia-openmax-il-0.21.0/clients/youtube/libtizyoutube/configure.ac
--- tizonia-openmax-il-0.21.0.orig/clients/youtube/libtizyoutube/configure.ac	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/youtube/libtizyoutube/configure.ac	2020-04-10 21:31:03.603144794 -0700
@@ -68,6 +68,12 @@ AX_PYTHON_DEVEL([>= '3.5'])
 AX_BOOST_BASE([1.54],, [AC_MSG_ERROR([libtizyoutube needs Boost 1.54])])
 AX_BOOST_PYTHON
 
+AC_ARG_WITH(youtube,
+    AS_HELP_STRING([--with-youtube],
+        [build YouTube support (default: yes)]),,
+    with_youtube=yes)
+AM_CONDITIONAL(WITH_YOUTUBE, test "x$with_youtube" = xyes)
+
 # Checks for header files.
 #AC_CHECK_HEADER_STDBOOL
 
