diff -urp tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel.c tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel.c
--- tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel.c	2020-05-22 15:33:08.000000000 -0700
+++ tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel.c	2020-09-07 12:02:38.796575624 -0700
@@ -134,11 +134,13 @@ init_ports_and_lists (void * ap_obj)
   p_obj->p_cport_ = NULL;
   p_obj->p_proc_ = NULL;
   p_obj->eos_ = false;
+#ifdef HAVE_DBUS
   p_obj->rm_ = 0;
   p_obj->rm_cbacks_.pf_waitend = &wait_complete;
   p_obj->rm_cbacks_.pf_preempt = &preemption_req;
   p_obj->rm_cbacks_.pf_preempt_end = &preemption_complete;
   p_obj->rm_inited_ = false;
+#endif
   p_obj->audio_init_ = null_param;
   p_obj->image_init_ = null_param;
   p_obj->video_init_ = null_param;
diff -urp tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel_decls.h tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel_decls.h
--- tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel_decls.h	2020-05-22 15:33:08.000000000 -0700
+++ tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel_decls.h	2020-09-07 12:02:38.796575624 -0700
@@ -35,7 +35,9 @@ extern "C" {
 
 #include <OMX_Core.h>
 
+#ifdef HAVE_DBUS
 #include <tizrmproxy_c.h>
+#endif
 
 #include "tizservant_decls.h"
 
@@ -116,9 +118,11 @@ struct tiz_krn
   OMX_PTR p_cport_;
   OMX_PTR p_proc_;
   bool eos_;
+#ifdef HAVE_DBUS
   tiz_rm_t rm_;
   tiz_rm_proxy_callbacks_t rm_cbacks_;
   bool rm_inited_;
+#endif
   OMX_PORT_PARAM_TYPE audio_init_;
   OMX_PORT_PARAM_TYPE image_init_;
   OMX_PORT_PARAM_TYPE video_init_;
diff -urp tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel_dispatch.inl tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel_dispatch.inl
--- tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel_dispatch.inl	2020-05-22 15:33:08.000000000 -0700
+++ tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel_dispatch.inl	2020-09-07 12:05:12.922516712 -0700
@@ -909,6 +909,7 @@ static OMX_ERRORTYPE dispatch_idle_to_lo
 
   rc = tiz_srv_deallocate_resources (ap_krn);
 
+#ifdef HAVE_DBUS
   if (OMX_ErrorNone == rc)
     {
       rc = release_rm_resources (ap_krn, handleOf(ap_krn));
@@ -919,6 +920,7 @@ static OMX_ERRORTYPE dispatch_idle_to_lo
       /* Uninitialize the Resource Manager hdl */
       rc = deinit_rm (ap_krn, handleOf(ap_krn));
     }
+#endif
 
   *ap_done = (OMX_ErrorNone == rc && all_depopulated (ap_krn))
     ? true
@@ -936,6 +938,7 @@ static OMX_ERRORTYPE dispatch_loaded_to_
   OMX_ERRORTYPE rc = OMX_ErrorNone;
   assert (ap_done);
 
+#ifdef HAVE_DBUS
   /* Before allocating any resources, we need to initialize the
    * Resource Manager hdl */
   rc = init_rm (ap_krn, handleOf(ap_krn));
@@ -944,6 +947,7 @@ static OMX_ERRORTYPE dispatch_loaded_to_
     {
       rc = acquire_rm_resources (ap_krn, handleOf(ap_krn));
     }
+#endif
 
   if (OMX_ErrorNone == rc)
     {
diff -urp tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel_helpers.inl tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel_helpers.inl
--- tizonia-openmax-il-0.22.0.orig/libtizonia/src/tizkernel_helpers.inl	2020-05-22 15:33:08.000000000 -0700
+++ tizonia-openmax-il-0.22.0/libtizonia/src/tizkernel_helpers.inl	2020-09-07 12:02:38.796575624 -0700
@@ -1032,6 +1032,7 @@ static OMX_ERRORTYPE enqueue_callback_ms
   return tiz_srv_enqueue (ap_obj, p_msg, 1);
 }
 
+#ifdef HAVE_DBUS
 static OMX_ERRORTYPE init_rm (const void *ap_obj, OMX_HANDLETYPE ap_hdl)
 {
   tiz_krn_t *p_obj = (tiz_krn_t *)ap_obj;
@@ -1175,6 +1176,7 @@ static OMX_ERRORTYPE release_rm_resource
 
   return OMX_ErrorNone;
 }
+#endif
 
 static bool all_populated (const void *ap_obj)
 {
