--- a/3rdparty/configure.ac.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/3rdparty/configure.ac	2019-08-01 18:23:13.047428869 -0700
@@ -30,6 +30,14 @@ AM_INIT_AUTOMAKE([foreign color-tests si
 
 # Checks for library functions.
 
+AC_ARG_WITH(dbus,
+    AS_HELP_STRING([--with-dbus],
+        [build with D-Bus support (default: yes)]),,
+    with_dbus=yes)
+AM_CONDITIONAL(WITH_DBUS, test "x$with_dbus" = xyes)
+
 AC_CONFIG_FILES([Makefile])
-AC_CONFIG_SUBDIRS([dbus-cplusplus])
+if test "$with_dbus" = yes; then
+	AC_CONFIG_SUBDIRS([dbus-cplusplus])
+fi
 AC_OUTPUT
