--- b/plugins/http_source/configure.ac	2020-04-06 18:42:54.375629227 -0700
+++ a/plugins/http_source/configure.ac	2020-04-06 19:02:26.520890938 -0700
@@ -188,11 +188,11 @@ AC_ARG_WITH(gmusic,
     with_gmusic=yes)
 AM_CONDITIONAL(WITH_GMUSIC, test "x$with_gmusic" = xyes)
 
-AC_ARG_WITH(dirble,
-    AS_HELP_STRING([--with-dirble],
-        [build Dirble support (default: yes)]),,
-    with_dirble=yes)
-AM_CONDITIONAL(WITH_DIRBLE, test "x$with_dirble" = xyes)
+AC_ARG_WITH(tunein,
+    AS_HELP_STRING([--with-tunein],
+        [build TuneIn support (default: yes)]),,
+    with_tunein=yes)
+AM_CONDITIONAL(WITH_TUNEIN, test "x$with_tunein" = xyes)
 
 AC_ARG_WITH(youtube,
     AS_HELP_STRING([--with-youtube],
