diff -urp tizonia-openmax-il-0.21.0.orig/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h tizonia-openmax-il-0.21.0/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h
--- tizonia-openmax-il-0.21.0.orig/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h	2020-04-09 21:51:02.065408508 -0700
@@ -38,7 +38,9 @@ extern "C" {
 
 #include <tizplatform.h>
 
+#ifdef HAVE_TUNEIN
 #include <tiztunein_c.h>
+#endif
 
 #include "cc_prc_decls.h"
 #include "cc_prc.h"
@@ -50,7 +52,9 @@ struct cc_tunein_prc
   const cc_prc_t _;
   OMX_TIZONIA_AUDIO_PARAM_TUNEINSESSIONTYPE tunein_session_;
   OMX_TIZONIA_AUDIO_PARAM_TUNEINPLAYLISTTYPE tunein_playlist_;
+#ifdef HAVE_TUNEIN
   tiz_tunein_t * p_tunein_;
+#endif
   bool remove_current_url_;
 };
 
diff -urp tizonia-openmax-il-0.21.0.orig/plugins/http_source/configure.ac tizonia-openmax-il-0.21.0/plugins/http_source/configure.ac
--- tizonia-openmax-il-0.21.0.orig/plugins/http_source/configure.ac	2020-04-09 20:37:17.571183323 -0700
+++ tizonia-openmax-il-0.21.0/plugins/http_source/configure.ac	2020-04-09 21:57:11.339622786 -0700
@@ -176,36 +176,60 @@ AC_ARG_WITH(icecast-client,
     with_icecast_client=yes)
 AM_CONDITIONAL(WITH_ICECAST_CLIENT, test "x$with_icecast_client" = xyes)
 
+if test "$with_icecast_client" = yes; then
+    AC_DEFINE(HAVE_ICECAST_CLIENT, 1, [Support for Icecast/SHOUTcast client is included])
+fi
+
 AC_ARG_WITH(soundcloud,
     AS_HELP_STRING([--with-soundcloud],
         [build SoundCloud support (default: yes)]),,
     with_soundcloud=yes)
 AM_CONDITIONAL(WITH_SOUNDCLOUD, test "x$with_soundcloud" = xyes)
 
+if test "$with_soundcloud" = yes; then
+    AC_DEFINE(HAVE_SOUNDCLOUD, 1, [Support for SoundCloud is included])
+fi
+
 AC_ARG_WITH(gmusic,
     AS_HELP_STRING([--with-gmusic],
         [build Google Music support (default: yes)]),,
     with_gmusic=yes)
 AM_CONDITIONAL(WITH_GMUSIC, test "x$with_gmusic" = xyes)
 
+if test "$with_gmusic" = yes; then
+    AC_DEFINE(HAVE_GMUSIC, 1, [Support for Google Music is included])
+fi
+
 AC_ARG_WITH(tunein,
     AS_HELP_STRING([--with-tunein],
         [build TuneIn support (default: yes)]),,
     with_tunein=yes)
 AM_CONDITIONAL(WITH_TUNEIN, test "x$with_tunein" = xyes)
 
+if test "$with_tunein" = yes; then
+    AC_DEFINE(HAVE_TUNEIN, 1, [Support for TuneIn is included])
+fi
+
 AC_ARG_WITH(youtube,
     AS_HELP_STRING([--with-youtube],
         [build YouTube support (default: yes)]),,
     with_youtube=yes)
 AM_CONDITIONAL(WITH_YOUTUBE, test "x$with_youtube" = xyes)
 
+if test "$with_youtube" = yes; then
+    AC_DEFINE(HAVE_YOUTUBE, 1, [Support for YouTube is included])
+fi
+
 AC_ARG_WITH(plex,
     AS_HELP_STRING([--with-plex],
         [build Plex support (default: yes)]),,
     with_plex=yes)
 AM_CONDITIONAL(WITH_PLEX, test "x$with_plex" = xyes)
 
+if test "$with_plex" = yes; then
+    AC_DEFINE(HAVE_PLEX, 1, [Support for Plex is included])
+fi
+
 AC_CONFIG_FILES([Makefile
                  src/Makefile])
 
diff -urp tizonia-openmax-il-0.21.0.orig/plugins/http_source/src/tuneinprc_decls.h tizonia-openmax-il-0.21.0/plugins/http_source/src/tuneinprc_decls.h
--- tizonia-openmax-il-0.21.0.orig/plugins/http_source/src/tuneinprc_decls.h	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/plugins/http_source/src/tuneinprc_decls.h	2020-04-09 21:51:14.815910602 -0700
@@ -39,7 +39,9 @@ extern "C" {
 #include <tizprc_decls.h>
 
 #include <tizplatform.h>
+#if WITH_TUNEIN
 #include <tiztunein_c.h>
+#endif
 
 typedef struct tunein_prc tunein_prc_t;
 struct tunein_prc
@@ -53,7 +55,9 @@ struct tunein_prc
   OMX_TIZONIA_STREAMINGBUFFERTYPE buffer_size_;
   OMX_PARAM_CONTENTURITYPE * p_uri_param_;
   tiz_urltrans_t * p_trans_;
+#ifdef HAVE_TUNEIN
   tiz_tunein_t * p_tunein_;
+#endif
   bool eos_;
   bool port_disabled_;
   bool uri_changed_;
