--- b/plugins/http_source/src/Makefile.am	2020-03-08 14:18:54.000000000 -0700
+++ a/plugins/http_source/src/Makefile.am	2020-04-06 04:02:21.678859657 -0700
@@ -19,65 +19,131 @@ libtizhttpsrcdir = $(plugindir)
 
 libtizhttpsrc_LTLIBRARIES = libtizhttpsrc.la
 
-noinst_HEADERS = \
-	httpsrc.h \
+noinst_HEADERS =
+
+if WITH_ICECAST_CLIENT
+noinst_HEADERS += httpsrc.h \
 	httpsrcport.h \
 	httpsrcport_decls.h \
 	httpsrcprc.h \
-	httpsrcprc_decls.h \
-	gmusicprc.h \
+	httpsrcprc_decls.h
+endif
+
+if WITH_GMUSIC
+noinst_HEADERS += gmusicprc.h \
 	gmusicprc_decls.h \
 	gmusiccfgport.h \
-	gmusiccfgport_decls.h \
-	scloudprc.h \
+	gmusiccfgport_decls.h
+endif
+
+if WITH_SOUNDCLOUD
+noinst_HEADERS += scloudprc.h \
 	scloudprc_decls.h \
 	scloudcfgport.h \
-	scloudcfgport_decls.h \
-	youtubeprc.h \
+	scloudcfgport_decls.h
+endif
+
+if WITH_YOUTUBE
+noinst_HEADERS += youtubeprc.h \
 	youtubeprc_decls.h \
 	youtubecfgport.h \
-	youtubecfgport_decls.h \
-	plexprc.h \
+	youtubecfgport_decls.h
+endif
+
+if WITH_PLEX
+noinst_HEADERS += plexprc.h \
 	plexprc_decls.h \
 	plexcfgport.h \
-	plexcfgport_decls.h \
-	tuneinprc.h \
+	plexcfgport_decls.h
+endif
+
+if WITH_TUNEIN
+noinst_HEADERS += tuneinprc.h \
 	tuneinprc_decls.h \
 	tuneincfgport.h \
 	tuneincfgport_decls.h
+endif
 
-libtizhttpsrc_la_SOURCES = \
-	httpsrc.c \
+libtizhttpsrc_la_SOURCES =
+
+if WITH_ICECAST_CLIENT
+libtizhttpsrc_la_SOURCES += httpsrc.c \
 	httpsrcport.c \
-	httpsrcprc.c \
-	gmusicprc.c \
-	gmusiccfgport.c \
-	scloudprc.c \
-	scloudcfgport.c \
-	youtubeprc.c \
-	youtubecfgport.c \
-	plexprc.c \
-	plexcfgport.c \
-	tuneinprc.c \
+	httpsrcprc.c
+endif
+
+if WITH_GMUSIC
+libtizhttpsrc_la_SOURCES += gmusicprc.c \
+	gmusiccfgport.c
+endif
+
+if WITH_SOUNDCLOUD
+libtizhttpsrc_la_SOURCES += scloudprc.c \
+	scloudcfgport.c
+endif
+
+if WITH_YOUTUBE
+libtizhttpsrc_la_SOURCES += youtubeprc.c \
+	youtubecfgport.c
+endif
+
+if WITH_PLEX
+libtizhttpsrc_la_SOURCES += plexprc.c \
+	plexcfgport.c
+endif
+
+if WITH_TUNEIN
+libtizhttpsrc_la_SOURCES += tuneinprc.c \
 	tuneincfgport.c
+endif
 
 libtizhttpsrc_la_CFLAGS = \
 	@TIZILHEADERS_CFLAGS@ \
 	@TIZPLATFORM_CFLAGS@ \
-	@TIZONIA_CFLAGS@ \
-	@TIZGMUSIC_CFLAGS@ \
-	@TIZSOUNDCLOUD_CFLAGS@ \
-	@TIZYOUTUBE_CFLAGS@ \
-	@TIZPLEX_CFLAGS@ \
+	@TIZONIA_CFLAGS@
+
+if WITH_GMUSIC
+libtizhttpsrc_la_CFLAGS += @TIZGMUSIC_CFLAGS@
+endif
+
+if WITH_SOUNDCLOUD
+libtizhttpsrc_la_CFLAGS += @TIZSOUNDCLOUD_CFLAGS@
+endif
+
+if WITH_YOUTUBE
+libtizhttpsrc_la_CFLAGS += @TIZYOUTUBE_CFLAGS@
+endif
+
+if WITH_PLEX
+libtizhttpsrc_la_CFLAGS += @TIZPLEX_CFLAGS@
+endif
+
+if WITH_TUNEIN
 	@TIZTUNEIN_CFLAGS@
+endif
 
 libtizhttpsrc_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 
 libtizhttpsrc_la_LIBADD = \
 	@TIZPLATFORM_LIBS@ \
-	@TIZONIA_LIBS@ \
-	@TIZGMUSIC_LIBS@ \
-	@TIZSOUNDCLOUD_LIBS@ \
-	@TIZYOUTUBE_LIBS@ \
-	@TIZPLEX_LIBS@ \
-	@TIZTUNEIN_LIBS@
+	@TIZONIA_LIBS@
+
+if WITH_GMUSIC
+libtizhttpsrc_la_LIBADD += @TIZGMUSIC_LIBS@
+endif
+
+if WITH_SOUNDCLOUD
+libtizhttpsrc_la_LIBADD += @TIZSOUNDCLOUD_LIBS@
+endif
+
+if WITH_YOUTUBE
+libtizhttpsrc_la_LIBADD += @TIZYOUTUBE_LIBS@
+endif
+
+if WITH_PLEX
+libtizhttpsrc_la_LIBADD += @TIZPLEX_LIBS@
+endif
+
+if WITH_TUNEIN
+libtizhttpsrc_la_LIBADD += @TIZTUNEIN_LIBS@
+endif
Only in tizonia-openmax-il-0.21.0/plugins/http_source/src: Makefile.am.orig
Only in tizonia-openmax-il-0.21.0/plugins/http_source/src: Makefile.am.rej
