diff -urp tizonia-openmax-il-0.21.0.orig/clients/chromecast/libtizchromecast/libtizchromecast.pc.in tizonia-openmax-il-0.21.0/clients/chromecast/libtizchromecast/libtizchromecast.pc.in
--- tizonia-openmax-il-0.21.0.orig/clients/chromecast/libtizchromecast/libtizchromecast.pc.in	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/chromecast/libtizchromecast/libtizchromecast.pc.in	2020-04-09 19:21:33.142251443 -0700
@@ -7,5 +7,5 @@ Name: libtizchromecast
 Description: Tizonia Chromecast client library
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ltizchromecast
-Libs.private: -lpython3.6 -lboost_python3
-Cflags: -I${includedir} -I${includedir}/python3.6 -I${includedir}/tizonia
+Libs.private: -lpython3.6m -lboost_python36
+Cflags: -I${includedir} -I${includedir}/python3.6m -I${includedir}/tizonia
diff -urp tizonia-openmax-il-0.21.0.orig/clients/plex/libtizplex/libtizplex.pc.in tizonia-openmax-il-0.21.0/clients/plex/libtizplex/libtizplex.pc.in
--- tizonia-openmax-il-0.21.0.orig/clients/plex/libtizplex/libtizplex.pc.in	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/plex/libtizplex/libtizplex.pc.in	2020-04-09 19:21:42.962631790 -0700
@@ -7,5 +7,5 @@ Name: libtizplex
 Description: Tizonia Plex audio client library
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ltizplex
-Libs.private: -lpython3.6 -lboost_python3
-Cflags: -I${includedir} -I${includedir}/python3.6 -I${includedir}/tizonia
+Libs.private: -lpython3.6m -lboost_python36
+Cflags: -I${includedir} -I${includedir}/python3.6m -I${includedir}/tizonia
diff -urp tizonia-openmax-il-0.21.0.orig/clients/soundcloud/libtizsoundcloud/libtizsoundcloud.pc.in tizonia-openmax-il-0.21.0/clients/soundcloud/libtizsoundcloud/libtizsoundcloud.pc.in
--- tizonia-openmax-il-0.21.0.orig/clients/soundcloud/libtizsoundcloud/libtizsoundcloud.pc.in	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/soundcloud/libtizsoundcloud/libtizsoundcloud.pc.in	2020-04-09 19:21:40.622541155 -0700
@@ -7,5 +7,5 @@ Name: libtizsoundcloud
 Description: Tizonia SoundCloud client library
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ltizsoundcloud
-Libs.private: -lpython3.6 -lboost_python3
-Cflags: -I${includedir} -I${includedir}/python3.6 -I${includedir}/tizonia
+Libs.private: -lpython3.6m -lboost_python36
+Cflags: -I${includedir} -I${includedir}/python3.6m -I${includedir}/tizonia
diff -urp tizonia-openmax-il-0.21.0.orig/clients/spotify/libtizspotify/libtizspotify.pc.in tizonia-openmax-il-0.21.0/clients/spotify/libtizspotify/libtizspotify.pc.in
--- tizonia-openmax-il-0.21.0.orig/clients/spotify/libtizspotify/libtizspotify.pc.in	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/spotify/libtizspotify/libtizspotify.pc.in	2020-04-09 19:21:39.542499327 -0700
@@ -7,5 +7,5 @@ Name: libtizspotify
 Description: Tizonia Spotify Web client library
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ltizspotify
-Libs.private: -lpython3.6 -lboost_python3
-Cflags: -I${includedir} -I${includedir}/python3.6 -I${includedir}/tizonia
+Libs.private: -lpython3.6m -lboost_python36
+Cflags: -I${includedir} -I${includedir}/python3.6m -I${includedir}/tizonia
diff -urp tizonia-openmax-il-0.21.0.orig/clients/tunein/libtiztunein/libtiztunein.pc.in tizonia-openmax-il-0.21.0/clients/tunein/libtiztunein/libtiztunein.pc.in
--- tizonia-openmax-il-0.21.0.orig/clients/tunein/libtiztunein/libtiztunein.pc.in	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/tunein/libtiztunein/libtiztunein.pc.in	2020-04-09 19:21:28.702079471 -0700
@@ -7,5 +7,5 @@ Name: libtiztunein
 Description: Tizonia Tunein client library
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ltiztunein
-Libs.private: -lpython3.6 -lboost_python3
-Cflags: -I${includedir} -I${includedir}/python3.6 -I${includedir}/tizonia
+Libs.private: -lpython3.6m -lboost_python36
+Cflags: -I${includedir} -I${includedir}/python3.6m -I${includedir}/tizonia
diff -urp tizonia-openmax-il-0.21.0.orig/clients/youtube/libtizyoutube/libtizyoutube.pc.in tizonia-openmax-il-0.21.0/clients/youtube/libtizyoutube/libtizyoutube.pc.in
--- tizonia-openmax-il-0.21.0.orig/clients/youtube/libtizyoutube/libtizyoutube.pc.in	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/clients/youtube/libtizyoutube/libtizyoutube.pc.in	2020-04-09 19:21:36.402377709 -0700
@@ -7,5 +7,5 @@ Name: libtizyoutube
 Description: Tizonia YouTube audio client library
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ltizyoutube
-Libs.private: -lpython3.6 -lboost_python3
-Cflags: -I${includedir} -I${includedir}/python3.6 -I${includedir}/tizonia
+Libs.private: -lpython3.6m -lboost_python36
+Cflags: -I${includedir} -I${includedir}/python3.6m -I${includedir}/tizonia
