--- a/libtizplatform/configure.ac.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/libtizplatform/configure.ac	2019-08-01 17:47:10.694886842 -0700
@@ -70,9 +70,19 @@ AC_CHECK_LIB([pthread], [pthread_create]
 AC_CHECK_LIB([libuuid], [uuid_generate])
 PKG_CHECK_MODULES([UUID], [uuid >= 2.19.0])
 PKG_CHECK_MODULES([CHECK], [check >= 0.9.4])
+
+AC_ARG_WITH(curl,
+    AS_HELP_STRING([--with-curl],
+        [build with cURL support (default: yes)]),,
+    with_curl=yes)
+AM_CONDITIONAL(WITH_CURL, test "x$with_curl" = xyes)
+
+if test "$with_curl" = yes; then
 # NOTE: Look for libcurl 7.18.0. Before this version, there was no explicit
 # support for pausing transfers.
 PKG_CHECK_MODULES([LIBCURL], [libcurl >= 7.18.0])
+AC_DEFINE(HAVE_CURL, 1, [Support for curl is included])
+fi
 
 AC_CHECK_HEADERS([tizonia/OMX_Core.h tizonia/OMX_Component.h],
 	[tizrmd_found_omx_headers=yes; break;])
