--- b/player/configure.ac	2020-04-06 18:08:19.835541989 -0700
+++ a/player/configure.ac	2020-04-06 18:28:49.572988679 -0700
@@ -156,6 +156,7 @@ AC_ARG_WITH(chromecast,
         [build support for the Chromecast plugin (default: yes)]),,
     with_chromecast=yes)
 AM_CONDITIONAL(WITH_CHROMECAST, test "x$with_chromecast" = xyes)
+
 if test "$with_chromecast" = yes; then
       AC_DEFINE(HAVE_CHROMECAST, 1, [Support for Chromecast is included])
 fi
@@ -166,11 +167,9 @@ AC_ARG_WITH(dbus,
     with_dbus=yes)
 AM_CONDITIONAL(WITH_DBUS, test "x$with_dbus" = xyes)
 
-AC_ARG_WITH(dirble,
-    AS_HELP_STRING([--with-dirble],
-        [build support for the Dirble plugin (default: yes)]),,
-    with_dirble=yes)
-AM_CONDITIONAL(WITH_DIRBLE, test "x$with_dirble" = xyes)
+if test "$with_dbus" = yes; then
+      AC_DEFINE(HAVE_DBUS, 1, [Support for dbus is included])
+fi
 
 AC_ARG_WITH(flac,
     AS_HELP_STRING([--with-flac],
@@ -258,6 +257,10 @@ AC_ARG_WITH(spotify,
     with_spotify=yes)
 AM_CONDITIONAL(WITH_SPOTIFY, test "x$with_spotify" = xyes)
 
+if test "$with_spotify" = yes; then
+      AC_DEFINE(HAVE_LIBSPOTIFY, 1, [Support for libspotify is included])
+fi
+
 AC_ARG_WITH(ogg,
     AS_HELP_STRING([--with-ogg],
         [build support for the OGG plugin (default: yes)]),,
@@ -278,6 +281,16 @@ if test "$with_opus" = yes; then
       AC_DEFINE(HAVE_OPUS, 1, [Support for OPUS is included])
 fi
 
+AC_ARG_WITH(tunein,
+    AS_HELP_STRING([--with-tunein],
+        [build support for the TuneIn plugin (default: yes)]),,
+    with_tunein=yes)
+AM_CONDITIONAL(WITH_TUNEIN, test "x$with_tunein" = xyes)
+
+if test "$with_tunein" = yes; then
+      AC_DEFINE(HAVE_TUNEIN, 1, [Support for TuneIn is included])
+fi
+
 AC_ARG_WITH(vorbis,
     AS_HELP_STRING([--with-vorbis],
         [build support for the Vorbis plugin (default: yes)]),,
@@ -317,13 +330,5 @@ fi
 
 AC_CONFIG_FILES([src/Makefile])
 
-if test "$with_spotify" = yes; then
-      AC_DEFINE(HAVE_LIBSPOTIFY, 1, [Support for libspotify is included])
-fi
-
-if test "$with_dbus" = yes; then
-      AC_DEFINE(HAVE_DBUS, 1, [Support for dbus is included])
-fi
-
 # End the configure script.
 AC_OUTPUT
