diff -urp tizonia-openmax-il-0.22.0.orig/player/src/tizplayapp.cpp tizonia-openmax-il-0.22.0/player/src/tizplayapp.cpp
--- tizonia-openmax-il-0.22.0.orig/player/src/tizplayapp.cpp	2020-05-22 15:33:08.000000000 -0700
+++ tizonia-openmax-il-0.22.0/player/src/tizplayapp.cpp	2020-09-07 10:08:26.592252936 -0700
@@ -69,27 +69,45 @@
 #include "tizgraphtypes.hpp"
 #include "tizomxutil.hpp"
 #include <decoders/tizdecgraphmgr.hpp>
+#ifdef HAVE_ICECAST_CLIENT
 #include <httpclnt/tizhttpclntmgr.hpp>
+#endif
+#ifdef HAVE_ICECAST_SERVER
 #include <httpserv/tizhttpservconfig.hpp>
 #include <httpserv/tizhttpservmgr.hpp>
+#endif
+#ifdef HAVE_CHROMECAST
 #include <services/chromecast/tizchromecastconfig.hpp>
 #include <services/chromecast/tizchromecastmgr.hpp>
+#endif
+#ifdef HAVE_TUNEIN
 #include <services/tunein/tiztuneinconfig.hpp>
 #include <services/tunein/tiztuneinmgr.hpp>
+#endif
+#ifdef HAVE_GMUSIC
 #include <services/googlemusic/tizgmusicconfig.hpp>
 #include <services/googlemusic/tizgmusicmgr.hpp>
+#endif
+#ifdef HAVE_SOUNDCLOUD
 #include <services/soundcloud/tizscloudconfig.hpp>
 #include <services/soundcloud/tizscloudmgr.hpp>
+#endif
 #ifdef HAVE_LIBSPOTIFY
 #include <services/spotify/tizspotifyconfig.hpp>
 #include <services/spotify/tizspotifymgr.hpp>
 #endif
+#ifdef HAVE_PLEX
 #include <services/plex/tizplexconfig.hpp>
 #include <services/plex/tizplexmgr.hpp>
+#endif
+#ifdef HAVE_YOUTUBE
 #include <services/youtube/tizyoutubeconfig.hpp>
 #include <services/youtube/tizyoutubemgr.hpp>
+#endif
+#ifdef HAVE_IHEART
 #include <services/iheart/tiziheartconfig.hpp>
 #include <services/iheart/tiziheartmgr.hpp>
+#endif
 
 #include "tizplayapp.hpp"
 
@@ -651,61 +669,93 @@ void tiz::playapp::set_option_handlers (
   // local audio decoding program options
   popts_.set_option_handler ("decode-local",
                              boost::bind (&tiz::playapp::decode_local, this));
+#ifdef HAVE_ICECAST_SERVER
   // streaming audio server program options
   popts_.set_option_handler ("serve-stream",
                              boost::bind (&tiz::playapp::serve_stream, this));
+#endif
+#ifdef HAVE_ICECAST_CLIENT
   // streaming audio client program options
   popts_.set_option_handler ("decode-stream",
                              boost::bind (&tiz::playapp::decode_stream, this));
+#endif
+#ifdef HAVE_LIBSPOTIFY
   // spotify streaming client program options
   popts_.set_option_handler ("spotify-stream",
                              boost::bind (&tiz::playapp::spotify_stream, this));
+#endif
+#ifdef HAVE_GMUSIC
   // Google music streaming client program options
   popts_.set_option_handler ("gmusic-stream",
                              boost::bind (&tiz::playapp::gmusic_stream, this));
+#endif
+#ifdef HAVE_SOUNDCLOUD
   // SoundCloud music streaming client program options
   popts_.set_option_handler ("scloud-stream",
                              boost::bind (&tiz::playapp::scloud_stream, this));
+#endif
+#ifdef HAVE_TUNEIN
   // Tunein internet radio directory streaming client program options
   popts_.set_option_handler ("tunein-stream",
                              boost::bind (&tiz::playapp::tunein_stream, this));
+#endif
+#ifdef HAVE_YOUTUBE
   // YouTube audio streaming client program options
   popts_.set_option_handler ("youtube-stream",
                              boost::bind (&tiz::playapp::youtube_stream, this));
+#endif
+#ifdef HAVE_PLEX
   // Plex audio streaming client program options
   popts_.set_option_handler ("plex-stream",
                              boost::bind (&tiz::playapp::plex_stream, this));
+#endif
+#ifdef HAVE_IHEART
   // Iheart audio streaming client program options
   popts_.set_option_handler ("iheart-stream",
                              boost::bind (&tiz::playapp::iheart_stream, this));
+#endif
+#ifdef HAVE_CHROMECAST
   // HTTP music streaming on Chromecast device
   popts_.set_option_handler (
       "http-stream-chromecast",
       boost::bind (&tiz::playapp::http_stream_chromecast, this));
+#ifdef HAVE_GMUSIC
   // Google music streaming on Chromecast device
   popts_.set_option_handler (
       "gmusic-stream-chromecast",
       boost::bind (&tiz::playapp::gmusic_stream_chromecast, this));
+#endif
+#ifdef HAVE_SOUNDCLOUD
   // Soudcloud audio streaming on Chromecast device
   popts_.set_option_handler (
       "scloud-stream-chromecast",
       boost::bind (&tiz::playapp::scloud_stream_chromecast, this));
+#endif
+#ifdef HAVE_TUNEIN
   // Tunein audio streaming on Chromecast device
   popts_.set_option_handler (
       "tunein-stream-chromecast",
       boost::bind (&tiz::playapp::tunein_stream_chromecast, this));
+#endif
+#ifdef HAVE_YOUTUBE
   // YouTube audio streaming on Chromecast device
   popts_.set_option_handler (
       "youtube-stream-chromecast",
       boost::bind (&tiz::playapp::youtube_stream_chromecast, this));
+#endif
+#ifdef HAVE_PLEX
   // Plex audio streaming on Chromecast device
   popts_.set_option_handler (
       "plex-stream-chromecast",
       boost::bind (&tiz::playapp::plex_stream_chromecast, this));
+#endif
+#ifdef HAVE_IHEART
   // Iheart audio streaming on Chromecast device
   popts_.set_option_handler (
       "iheart-stream-chromecast",
       boost::bind (&tiz::playapp::iheart_stream_chromecast, this));
+#endif
+#endif
 }
 
 OMX_ERRORTYPE
@@ -984,6 +1034,7 @@ tiz::playapp::decode_local ()
   return rc;
 }
 
+#ifdef HAVE_ICECAST_SERVER
 OMX_ERRORTYPE
 tiz::playapp::serve_stream ()
 {
@@ -1076,7 +1127,9 @@ tiz::playapp::serve_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_ICECAST_CLIENT
 OMX_ERRORTYPE
 tiz::playapp::decode_stream ()
 {
@@ -1113,6 +1166,7 @@ tiz::playapp::decode_stream ()
 
   return rc;
 }
+#endif
 
 #ifdef HAVE_LIBSPOTIFY
 OMX_ERRORTYPE
@@ -1176,14 +1230,9 @@ tiz::playapp::spotify_stream ()
 
   return rc;
 }
-#else
-OMX_ERRORTYPE
-tiz::playapp::spotify_stream ()
-{
-  return OMX_ErrorNone;
-}
 #endif
 
+#ifdef HAVE_GMUSIC
 OMX_ERRORTYPE
 tiz::playapp::gmusic_stream ()
 {
@@ -1240,7 +1289,9 @@ tiz::playapp::gmusic_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_SOUNDCLOUD
 OMX_ERRORTYPE
 tiz::playapp::scloud_stream ()
 {
@@ -1283,7 +1334,9 @@ tiz::playapp::scloud_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_TUNEIN
 OMX_ERRORTYPE
 tiz::playapp::tunein_stream ()
 {
@@ -1328,7 +1381,9 @@ tiz::playapp::tunein_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_YOUTUBE
 OMX_ERRORTYPE
 tiz::playapp::youtube_stream ()
 {
@@ -1372,7 +1427,9 @@ tiz::playapp::youtube_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_PLEX
 OMX_ERRORTYPE
 tiz::playapp::plex_stream ()
 {
@@ -1417,7 +1474,9 @@ tiz::playapp::plex_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_IHEART
 OMX_ERRORTYPE
 tiz::playapp::iheart_stream ()
 {
@@ -1460,7 +1519,9 @@ tiz::playapp::iheart_stream ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_CHROMECAST
 OMX_ERRORTYPE
 tiz::playapp::http_stream_chromecast ()
 {
@@ -1506,6 +1567,7 @@ tiz::playapp::http_stream_chromecast ()
   return rc;
 }
 
+#ifdef HAVE_GMUSIC
 OMX_ERRORTYPE
 tiz::playapp::gmusic_stream_chromecast ()
 {
@@ -1569,7 +1631,9 @@ tiz::playapp::gmusic_stream_chromecast (
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_SOUNDCLOUD
 OMX_ERRORTYPE
 tiz::playapp::scloud_stream_chromecast ()
 {
@@ -1619,7 +1683,9 @@ tiz::playapp::scloud_stream_chromecast (
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_TUNEIN
 OMX_ERRORTYPE
 tiz::playapp::tunein_stream_chromecast ()
 {
@@ -1671,7 +1737,9 @@ tiz::playapp::tunein_stream_chromecast (
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_YOUTUBE
 OMX_ERRORTYPE
 tiz::playapp::youtube_stream_chromecast ()
 {
@@ -1721,7 +1789,9 @@ tiz::playapp::youtube_stream_chromecast
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_PLEX
 OMX_ERRORTYPE
 tiz::playapp::plex_stream_chromecast ()
 {
@@ -1774,7 +1844,9 @@ tiz::playapp::plex_stream_chromecast ()
 
   return rc;
 }
+#endif
 
+#ifdef HAVE_IHEART
 OMX_ERRORTYPE
 tiz::playapp::iheart_stream_chromecast ()
 {
@@ -1824,6 +1896,8 @@ tiz::playapp::iheart_stream_chromecast (
 
   return rc;
 }
+#endif
+#endif
 
 void tiz::playapp::print_banner () const
 {
