--- a/player/src/tizgraphmgr.hpp.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/player/src/tizgraphmgr.hpp	2019-08-03 12:16:50.647671838 -0700
@@ -37,7 +37,9 @@
 
 #include "tizgraphtypes.hpp"
 #include "tizplaybackstatus.hpp"
+#ifdef HAVE_DBUS
 #include "mpris/tizmprismgr.hpp"
+#endif
 #include "tizgraphmgrfsm.hpp"
 
 namespace tiz
@@ -228,8 +230,10 @@ namespace tiz
       OMX_ERRORTYPE graph_error (const OMX_ERRORTYPE error,
                                  const std::string &msg);
 
+#ifdef HAVE_DBUS
       OMX_ERRORTYPE start_mpris (const graphmgr_capabilities &graphmgr_caps);
       OMX_ERRORTYPE stop_mpris ();
+#endif
       OMX_ERRORTYPE do_update_control_ifcs (const control::playback_status_t status,
                                             const std::string &current_song = std::string ());
       OMX_ERRORTYPE do_update_metadata (const track_metadata_map_t &metadata);
@@ -238,8 +242,10 @@ namespace tiz
     protected:
       ops *p_ops_;
       fsm fsm_;
+#ifdef HAVE_DBUS
       control::mprismgr_ptr_t mpris_ptr_;
       control::playback_events_t playback_events_;
+#endif
 
     private:
       OMX_ERRORTYPE init_cmd_queue ();
