--- a/plugins/ogg_muxer/configure.ac.orig	2019-03-13 16:05:10.000000000 -0700
+++ b/plugins/ogg_muxer/configure.ac	2019-08-06 12:36:29.977925586 -0700
@@ -115,6 +115,18 @@ AC_SUBST([plugindir], ['${libdir}/tizoni
 
 # Checks for library functions.
 
+AC_ARG_WITH(curl,
+    AS_HELP_STRING([--with-curl],
+        [build with cURL support (default: yes)]),,
+    with_curl=yes)
+AM_CONDITIONAL(WITH_CURL, test "x$with_curl" = xyes)
+
+if test "$with_curl" = yes; then
+    AC_DEFINE(HAVE_CURL, 1, [Support for curl is included])
+else
+    AC_MSG_ERROR([ogg_muxer requires --with-curl])
+fi
+
 AC_CONFIG_FILES([Makefile
                  src/Makefile])
 
