--- a/Makefile.am.orig	2019-08-01 14:20:38.431013465 -0700
+++ b/Makefile.am	2019-08-01 18:25:52.341850913 -0700
@@ -24,8 +24,24 @@
 
 # SUBDIRS = config include libtizplatform rm libtizcore libtizonia plugins tizonia docs
 
+SUBDIRS = 3rdparty
+
+if ENABLE_PLAYER
+SUBDIRS += config
+endif
+
+SUBDIRS += include clients libtizplatform
+
+if WITH_CHROMECAST
+SUBDIRS += cast
+endif
+
+if WITH_DBUS
+SUBDIRS += rm
+endif
+
+SUBDIRS += libtizcore libtizonia plugins
+
 if ENABLE_PLAYER
-SUBDIRS= 3rdparty config include clients libtizplatform cast rm libtizcore libtizonia plugins player
-else
-SUBDIRS= 3rdparty include clients libtizplatform cast rm libtizcore libtizonia plugins config
+SUBDIRS += player
 endif
