diff -urp tizonia-openmax-il-0.21.0.orig/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h tizonia-openmax-il-0.21.0/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h
--- tizonia-openmax-il-0.21.0.orig/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/plugins/chromecast_renderer/src/cc_tuneinprc_decls.h	2020-04-09 21:51:02.065408508 -0700
@@ -38,7 +38,9 @@ extern "C" {
 
 #include <tizplatform.h>
 
+#ifdef HAVE_TUNEIN
 #include <tiztunein_c.h>
+#endif
 
 #include "cc_prc_decls.h"
 #include "cc_prc.h"
@@ -50,7 +52,9 @@ struct cc_tunein_prc
   const cc_prc_t _;
   OMX_TIZONIA_AUDIO_PARAM_TUNEINSESSIONTYPE tunein_session_;
   OMX_TIZONIA_AUDIO_PARAM_TUNEINPLAYLISTTYPE tunein_playlist_;
+#ifdef HAVE_TUNEIN
   tiz_tunein_t * p_tunein_;
+#endif
   bool remove_current_url_;
 };
 
diff -urp tizonia-openmax-il-0.21.0.orig/plugins/http_source/src/tuneinprc_decls.h tizonia-openmax-il-0.21.0/plugins/http_source/src/tuneinprc_decls.h
--- tizonia-openmax-il-0.21.0.orig/plugins/http_source/src/tuneinprc_decls.h	2020-03-08 14:18:54.000000000 -0700
+++ tizonia-openmax-il-0.21.0/plugins/http_source/src/tuneinprc_decls.h	2020-04-09 21:51:14.815910602 -0700
@@ -39,7 +39,9 @@ extern "C" {
 #include <tizprc_decls.h>
 
 #include <tizplatform.h>
+#ifdef HAVE_TUNEIN
 #include <tiztunein_c.h>
+#endif
 
 typedef struct tunein_prc tunein_prc_t;
 struct tunein_prc
@@ -53,7 +55,9 @@ struct tunein_prc
   OMX_TIZONIA_STREAMINGBUFFERTYPE buffer_size_;
   OMX_PARAM_CONTENTURITYPE * p_uri_param_;
   tiz_urltrans_t * p_trans_;
+#ifdef HAVE_TUNEIN
   tiz_tunein_t * p_tunein_;
+#endif
   bool eos_;
   bool port_disabled_;
   bool uri_changed_;
