diff -urp tizonia-openmax-il-0.22.0.orig/player/src/tizgraphfactory.cpp tizonia-openmax-il-0.22.0/player/src/tizgraphfactory.cpp
--- tizonia-openmax-il-0.22.0.orig/player/src/tizgraphfactory.cpp	2020-05-22 15:33:08.000000000 -0700
+++ tizonia-openmax-il-0.22.0/player/src/tizgraphfactory.cpp	2020-09-06 22:09:14.895813022 -0700
@@ -35,15 +35,38 @@
 #include <boost/make_shared.hpp>
 #include <boost/filesystem.hpp>
 
+#include <tizplatform.h>
+#ifdef HAVE_MP3
 #include "decoders/tizmp3graph.hpp"
+#endif
+#ifdef HAVE_AAC
 #include "decoders/tizaacgraph.hpp"
+#endif
+#ifdef HAVE_OPUS
 #include "decoders/tizopusgraph.hpp"
+#endif
+#ifdef HAVE_OGG
+#ifdef HAVE_OPUS
 #include "decoders/tizoggopusgraph.hpp"
+#endif
+#endif
+#ifdef HAVE_VORBIS
 #include "decoders/tizvorbisgraph.hpp"
+#endif
+#ifdef HAVE_FLAC
 #include "decoders/tizflacgraph.hpp"
+#endif
+#ifdef HAVE_OGG
+#ifdef HAVE_FLAC
 #include "decoders/tizoggflacgraph.hpp"
+#endif
+#endif
+#ifdef HAVE_PCM
 #include "decoders/tizpcmgraph.hpp"
+#endif
+#ifdef HAVE_MP2
 #include "decoders/tizmpeggraph.hpp"
+#endif
 #include "tizprobe.hpp"
 #include "tizgraphfactory.hpp"
 
@@ -64,21 +87,30 @@ tizgraph_ptr_t graph::factory::create_gr
            tiz_domain_to_str (p->get_omx_domain ()));
   TIZ_LOG (TIZ_PRIORITY_DEBUG, "coding type : %s",
            tiz_audio_coding_to_str (p->get_audio_coding_type ()));
-  if (p->get_omx_domain () == OMX_PortDomainAudio
+  if (0) ;
+#ifdef HAVE_MP2
+  else if (p->get_omx_domain () == OMX_PortDomainAudio
       && p->get_audio_coding_type () == OMX_AUDIO_CodingMP2)
   {
     return boost::make_shared< tiz::graph::mpegdecoder >();
   }
+#endif
+#ifdef HAVE_MP3
   else if (p->get_omx_domain () == OMX_PortDomainAudio
       && p->get_audio_coding_type () == OMX_AUDIO_CodingMP3)
   {
     return boost::make_shared< tiz::graph::mp3decoder >();
   }
+#endif
+#ifdef HAVE_FLAC
   else if (p->get_omx_domain () == OMX_PortDomainAudio
            && p->get_audio_coding_type () == OMX_AUDIO_CodingAAC)
   {
     return boost::make_shared< tiz::graph::aacdecoder >();
   }
+#endif
+#ifdef HAVE_OGG
+#ifdef HAVE_OPUS
   else if (p->get_omx_domain () == OMX_PortDomainAudio
            && p->get_audio_coding_type () == OMX_AUDIO_CodingOPUS)
   {
@@ -91,6 +123,10 @@ tizgraph_ptr_t graph::factory::create_gr
         return boost::make_shared< tiz::graph::oggopusdecoder >();
       }
   }
+#endif
+#endif
+#ifdef HAVE_OGG
+#ifdef HAVE_FLAC
   else if (p->get_omx_domain () == OMX_PortDomainAudio
            && p->get_audio_coding_type () == OMX_AUDIO_CodingFLAC)
   {
@@ -105,16 +141,22 @@ tizgraph_ptr_t graph::factory::create_gr
       return boost::make_shared< tiz::graph::flacdecoder >();
     }
   }
+#endif
+#endif
+#ifdef HAVE_VORBIS
   else if (p->get_omx_domain () == OMX_PortDomainAudio
            && p->get_audio_coding_type () == OMX_AUDIO_CodingVORBIS)
   {
     return boost::make_shared< tiz::graph::vorbisdecoder >();
   }
+#endif
+#ifdef HAVE_PCM
   else if (p->get_omx_domain () == OMX_PortDomainAudio
            && p->get_audio_coding_type () == OMX_AUDIO_CodingPCM)
   {
     return boost::make_shared< tiz::graph::pcmdecoder >();
   }
+#endif
   return null_ptr;
 }
 
