--- b/configure.ac	2020-04-06 04:33:00.970319805 -0700
+++ a/configure.ac	2020-04-06 04:38:56.104171802 -0700
@@ -86,12 +86,6 @@ AC_ARG_WITH(chromecast,
     with_chromecast=yes)
 AM_CONDITIONAL(WITH_CHROMECAST, test "x$with_chromecast" = xyes)
 
-AC_ARG_WITH(dirble,
-    AS_HELP_STRING([--with-dirble],
-        [build Dirble support (default: yes)]),,
-    with_dirble=yes)
-AM_CONDITIONAL(WITH_DIRBLE, test "x$with_dirble" = xyes)
-
 AC_ARG_WITH(gmusic,
     AS_HELP_STRING([--with-gmusic],
         [build Google Music support (default: yes)]),,
@@ -116,6 +110,12 @@ AC_ARG_WITH(spotify,
     with_spotify=yes)
 AM_CONDITIONAL(WITH_SPOTIFY, test "x$with_spotify" = xyes)
 
+AC_ARG_WITH(tunein,
+    AS_HELP_STRING([--with-tunein],
+        [build TuneIn support (default: yes)]),,
+    with_tunein=yes)
+AM_CONDITIONAL(WITH_TUNEIN, test "x$with_tunein" = xyes)
+
 AC_ARG_WITH(youtube,
     AS_HELP_STRING([--with-youtube],
         [build YouTube support (default: yes)]),,
@@ -219,7 +219,7 @@ AC_ARG_WITH(webm,
     with_webm=yes)
 
 if test -z "$with_http_source" ; then
-    if test "$with_icecast_client" = yes || test "$with_dirble" = yes || test "$with_gmusic" = yes || test "$with_plex" = yes || test "$with_soundcloud" = yes || test "$with_youtube" = yes ; then
+    if test "$with_icecast_client" = yes || test "$with_gmusic" = yes || test "$with_plex" = yes || test "$with_soundcloud" = yes || test "$with_tunein" = yes || test "$with_youtube" = yes ; then
         AC_MSG_ERROR([icecast-client, dirble, gmusic, plex, soundcloud, youtube requires --with-http-source.])
     fi
 fi
@@ -229,7 +229,7 @@ AC_CONFIG_FILES([Makefile])
 AC_CONFIG_SUBDIRS([3rdparty
                    include])
 
-if test "$with_icecast_client" = yes || test "$with_dirble" = yes || test "$with_gmusic" = yes || test "$with_plex" = yes || test "$with_soundcloud" = yes || test "$with_youtube" = yes ; then
+if test "$with_icecast_client" = yes || test "$with_gmusic" = yes || test "$with_plex" = yes || test "$with_soundcloud" = yes || test "$with_tunein" = yes || test "$with_youtube" = yes ; then
     AC_CONFIG_SUBDIRS([clients])
 fi
 
@@ -275,12 +275,12 @@ AC_MSG_RESULT([
   Services
 
     Chromecast: .................. ${with_chromecast}
-    Dirble: ...................... ${with_dirble}
     Google Music: ................ ${with_gmusic}
     Icecast/SHOUTcast client ..... ${with_icecast_client}
     Plex: ........................ ${with_plex}
     SoundCloud: .................. ${with_soundcloud}
     Spotify: ..................... ${with_spotify}
+    TuneIn: ...................... ${with_tunein}
     YouTube: ..................... ${with_youtube}
 
   Plugins:
--- b/Makefile.am	2020-04-06 04:33:00.970319805 -0700
+++ a/Makefile.am	2020-04-06 04:39:45.196086589 -0700
@@ -35,9 +35,6 @@ SUBDIRS += include
 if WITH_ICECAST_CLIENT
 SUBDIRS += clients
 else
-if WITH_DIRBLE
-SUBDIRS += clients
-else
 if WITH_GMUSIC
 SUBDIRS += clients
 else
@@ -47,13 +44,16 @@ else
 if WITH_SOUNDCLOUD
 SUBDIRS += clients
 else
+if WITH_TUNEIN
+SUBDIRS += clients
+else
 if WITH_YOUTUBE
 SUBDIRS += clients
 endif # youtube
+endif # tunein
 endif # soundcloud
 endif # plex
 endif # gmusic
-endif # dirble
 endif # icecast client
 
 SUBDIRS += libtizplatform
