--- a/plugins/configure.ac.orig	2019-08-01 21:04:22.996710183 -0700
+++ b/plugins/configure.ac	2019-08-01 21:19:31.270307986 -0700
@@ -147,11 +147,11 @@ AC_ARG_WITH(sdl,
 AM_CONDITIONAL(WITH_SDL, test "x$with_sdl" = xyes)
 
 AC_ARG_WITH(spotify,
-    AS_HELP_STRING([--with-libspotify],
-        [build the libspotify-based OpenMAX IL plugin (default: yes)]),,
+    AS_HELP_STRING([--with-spotify],
+        [build the Spotify plugin (default: yes)]),,
     with_spotify=yes)
 
-AM_CONDITIONAL(WITH_LIBSPOTIFY, test "x$with_libspotify" = xyes)
+AM_CONDITIONAL(WITH_SPOTIFY, test "x$with_spotify" = xyes)
 
 AC_ARG_WITH(alsa,
     AS_HELP_STRING([--with-alsa],
