--- b/player/Makefile.am	2020-03-08 14:18:54.000000000 -0700
+++ a/player/Makefile.am	2020-04-06 02:55:25.652697172 -0700
@@ -16,7 +16,13 @@
 # along with Tizonia.  If not, see <http://www.gnu.org/licenses/>.
 
 
-SUBDIRS = tools dbus man src
+SUBDIRS = tools man
+
+if WITH_DBUS
+SUBDIRS += dbus
+endif
+
+SUBDIRS += src
 
 ACLOCAL_AMFLAGS = -I m4
 
Only in tizonia-openmax-il-0.21.0/player: Makefile.am.orig
Only in tizonia-openmax-il-0.21.0/player: Makefile.am.rej
