--- a/next.config.ts.orig	2025-02-15 14:41:35.344055715 -0800
+++ b/next.config.ts	2025-02-15 14:42:28.579710084 -0800
@@ -27,6 +27,26 @@ const nextConfig: NextConfig = {
     ],
     webVitalsAttribution: ['CLS', 'LCP'],
-    webpackMemoryOptimizations: true,
+    webpackMemoryOptimizations: true,
+//
+// # Setting workerThreads=false causes:
+//
+//     munmap_chunk(): invalid pointer
+//     # but shows progress bar
+//
+// # Setting workerThreads=true causes:
+//
+//     > Build error occurred
+//     Error [DataCloneError]: webpack (config, options) {
+//                     const webpack = options.webpack;
+//                     const { dev } = op...<omitted>... } could not be cloned.
+//         at new Promise (<anonymous>) {
+//       code: 25,
+//       [...]
+//       DATA_CLONE_ERR: 25
+//     # without progress bar
+//
+    workerThreads: false,
+    cpus: 1, // For more settings, see also https://github.com/vercel/next.js/blob/v15.1.7/packages/next/src/server/config-shared.ts#L306
   },
   async headers() {
     return [
