diff --git a/index.js b/index.js
index e9fc367a00904fca3385a05b2b435fed23d14d77..b1a9488791465679f29bb18bf0ad0fa8445363e6 100644
--- a/index.js
+++ b/index.js
@@ -2,25 +2,3 @@ const Fs = require('fs');
 const Pdf = require('./lib/pdf-parse.js');
 
 module.exports = Pdf;
-
-let isDebugMode = !module.parent; 
-
-//process.env.AUTO_KENT_DEBUG
-
-
-//for testing purpose
-if (isDebugMode) {
-
-    let PDF_FILE = './test/data/05-versions-space.pdf';
-    let dataBuffer = Fs.readFileSync(PDF_FILE);
-    Pdf(dataBuffer).then(function(data) {
-        Fs.writeFileSync(`${PDF_FILE}.txt`, data.text, {
-            encoding: 'utf8',
-            flag: 'w'
-        });
-        debugger;
-    }).catch(function(err) {
-        debugger;
-    });
-
-}
