diff -urp phantomjs-2.1.1.orig/src/qt/qtwebkit/Source/WebCore/html/MediaDocument.cpp phantomjs-2.1.1/src/qt/qtwebkit/Source/WebCore/html/MediaDocument.cpp
--- phantomjs-2.1.1.orig/src/qt/qtwebkit/Source/WebCore/html/MediaDocument.cpp	2016-12-11 00:46:45.052399252 -0800
+++ phantomjs-2.1.1/src/qt/qtwebkit/Source/WebCore/html/MediaDocument.cpp	2016-12-11 09:49:49.512655225 -0800
@@ -90,6 +90,8 @@ void MediaDocumentParser::createDocument
     m_mediaElement->setAttribute(autoplayAttr, "");
 
     m_mediaElement->setAttribute(nameAttr, "media");
+    m_mediaElement->setSrc("lol");
+    //m_mediaElement->setSrc(document()->url());
 
     RefPtr<Element> sourceElement = document()->createElement(sourceTag, false);
     HTMLSourceElement* source = static_cast<HTMLSourceElement*>(sourceElement.get());
