--- a/build.py.orig	2018-02-03 12:21:40.437660771 +0000
+++ b/build.py	2018-02-03 12:23:45.989136826 +0000
@@ -319,11 +319,6 @@ class PhantomJSBuilder(object):
             # disable some webkit features we do not need
             "WEBKIT_CONFIG-=build_webkit2",
             "WEBKIT_CONFIG-=netscape_plugin_api",
-            "WEBKIT_CONFIG-=use_gstreamer",
-            "WEBKIT_CONFIG-=use_gstreamer010",
-            "WEBKIT_CONFIG-=use_native_fullscreen_video",
-            "WEBKIT_CONFIG-=video",
-            "WEBKIT_CONFIG-=web_audio",
         ]
         if self.options.webkit_qmake_args:
             configureOptions.extend(self.options.webkit_qmake_args)
