--- a/Makefile.orig	2020-06-21 15:12:29.463376312 -0700
+++ b/Makefile	2020-06-21 15:15:11.219637403 -0700
@@ -3,7 +3,7 @@ include Config.mk
 PATH := $(eTCpath)$(PATH)
 SHELL=/bin/bash
 
-.PHONY: ENIGMA all clean Game clean-game emake emake-tests gm2egm libProtocols libEGM required-directories .FORCE
+.PHONY: all clean Game clean-game required-directories .FORCE
 
 $(LIB_PFX)compileEGMf$(LIB_EXT): ENIGMA
 ENIGMA: .FORCE libProtocols$(LIB_EXT) libENIGMAShared$(LIB_EXT)
@@ -18,7 +18,7 @@ clean: .FORCE
 	$(MAKE) -C shared/protos/ clean
 	$(MAKE) -C CommandLine/gm2egm/ clean
 
-all: libENIGMAShared libProtocols libEGM ENIGMA emake emake-tests test-runner .FORCE
+all: .FORCE
 
 Game: .FORCE
 	@$(RM) -f logs/enigma_compile.log
@@ -27,38 +27,8 @@ Game: .FORCE
 clean-game: .FORCE
 	$(MAKE) -C ENIGMAsystem/SHELL clean
 
-libENIGMAShared$(LIB_EXT): libENIGMAShared
-libENIGMAShared: .FORCE
-	$(MAKE) -C shared/
-
-libProtocols$(LIB_EXT): libProtocols
-libProtocols: .FORCE
-	$(MAKE) -C shared/protos/
-
-libEGM$(LIB_EXT): libEGM
-libEGM: .FORCE libProtocols$(LIB_EXT) libENIGMAShared$(LIB_EXT)
-	$(MAKE) -C CommandLine/libEGM/
-
 EMAKE_TARGETS = .FORCE ENIGMA
 
-ifneq ($(CLI_ENABLE_EGM), FALSE)
-	EMAKE_TARGETS += libEGM$(LIB_EXT)
-else
-	EMAKE_TARGETS += libProtocols$(LIB_EXT)
-endif
-
-emake: $(EMAKE_TARGETS) $(LIB_PFX)compileEGMf$(LIB_EXT)
-	$(MAKE) -C CommandLine/emake/
-
-emake-tests: .FORCE libEGM$(LIB_EXT) $(LIB_PFX)compileEGMf$(LIB_EXT)
-	TESTS=TRUE $(MAKE) -C CommandLine/emake/
-
-gm2egm: libEGM$(LIB_EXT) .FORCE
-	$(MAKE) -C CommandLine/gm2egm/
-
-test-runner: emake .FORCE
-	$(MAKE) -C CommandLine/testing/
-
 required-directories: .FORCE
 	@mkdir -p "$(WORKDIR)"
 	@mkdir -p "$(CODEGEN)/Preprocessor_Environment_Editable/"
