--- client-v6.21137.0/build.gradle.orig	2021-05-15 07:47:20.000000000 -0700
+++ client-v6.21137.0/build.gradle	2021-05-30 10:42:58.534125446 -0700
@@ -1,8 +1,15 @@
-plugins {
-	id 'java-library'
-	id 'com.github.johnrengelman.shadow' version '7.0.0'
+buildscript {
+	repositories {
+		maven { url '__MAVEN_PATH__' }
+	}
+	dependencies {
+		classpath 'gradle.plugin.com.github.jengelman.gradle.plugins:shadow:7.0.0'
+	}
 }
 
+apply plugin: 'java-library'
+apply plugin: 'com.github.johnrengelman.shadow'
+
 sourceCompatibility = '1.8'
 targetCompatibility = '1.8'
 
@@ -18,7 +25,7 @@ shadowJar {
 }
 
 repositories {
-	mavenCentral()
+	maven { url '__MAVEN_PATH__' }
 }
 
 dependencies {
