diff -urp rocprofiler-rocm-5.2.3.orig/bin/build_kernel.sh rocprofiler-rocm-5.2.3/bin/build_kernel.sh
--- rocprofiler-rocm-5.2.3.orig/bin/build_kernel.sh	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/bin/build_kernel.sh	2023-09-28 11:27:11.372198383 -0700
@@ -18,7 +18,7 @@ if [ -z "$DST_DIR" ] ; then
 fi
 
 if [ -z "$ROCM_DIR" ] ; then
-  ROCM_DIR=/opt/rocm
+  ROCM_DIR="@EPREFIX_ROCM_PATH@"
 fi
 
 if [ -z "$TGT_LIST" ] ; then
@@ -32,12 +32,12 @@ fi
 
 OCL_VER="2.0"
 
-if [ -e $ROCM_DIR/llvm ] ; then
-  LLVM_DIR=$ROCM_DIR/llvm
-  LIB_DIR=$ROCM_DIR/lib
+if [ -e @EPREFIX_LLVM_PATH@ ] ; then
+  LLVM_DIR=@EPREFIX_LLVM_PATH@
+  LIB_DIR=$ROCM_DIR/@LIBDIR@
 else
   LLVM_DIR=$ROCM_DIR/hcc
-  LIB_DIR=$LLVM_DIR/lib
+  LIB_DIR=$LLVM_DIR/@LIBDIR@
 fi
 
 # Determine whether using new or old device-libs layout
@@ -45,6 +45,8 @@ if [ -e $LIB_DIR/bitcode/opencl.amdgcn.b
   BC_DIR=$LIB_DIR/bitcode
 elif [ -e $LIB_DIR/opencl.amdgcn.bc ]; then
   BC_DIR=$LIB_DIR
+elif [ -e $ROCM_DIR/@LIBDIR@/amdgcn/bitcode/opencl.bc ]; then
+  BC_DIR=$ROCM_DIR/@LIBDIR@/amdgcn/bitcode
 elif [ -e $ROCM_DIR/amdgcn/bitcode/opencl.bc ]; then
   BC_DIR=$ROCM_DIR/amdgcn/bitcode
 else
@@ -52,7 +54,7 @@ else
   exit 1
 fi
 
-CLANG_ROOT=$LLVM_DIR/lib/clang
+CLANG_ROOT=@EPREFIX_CLANG_PATH@
 CLANG_DIR=`ls -d $CLANG_ROOT/* | head -n 1`
 if [ "$CLANG_DIR" = "" ] ; then
   echo "Error: LLVM clang library was not found"
diff -urp rocprofiler-rocm-5.2.3.orig/bin/merge_traces.sh rocprofiler-rocm-5.2.3/bin/merge_traces.sh
--- rocprofiler-rocm-5.2.3.orig/bin/merge_traces.sh	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/bin/merge_traces.sh	2023-09-28 11:27:11.376198383 -0700
@@ -25,7 +25,7 @@
 #A script to merge rocprof traces and then provide a results.json for the aggregate numbers.
 
 ROCPROF=`which rocprof`
-BIN_DIR=$(dirname $(realpath ${ROCPROF:-/opt/rocm/bin/rocprof}))
+BIN_DIR=$(dirname $(realpath ${ROCPROF:-@EPREFIX_ROCM_PATH@/bin/rocprof}))
 
 bin_name=`basename $0`
 
diff -urp rocprofiler-rocm-5.2.3.orig/bin/rpl_run.sh rocprofiler-rocm-5.2.3/bin/rpl_run.sh
--- rocprofiler-rocm-5.2.3.orig/bin/rpl_run.sh	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/bin/rpl_run.sh	2023-09-28 11:30:09.244196337 -0700
@@ -30,11 +30,12 @@ RUN_DIR=`pwd`
 TMP_DIR="/tmp"
 DATA_DIR="rpl_data_${time_stamp}_$$"
 
-RPL_PATH=$ROOT_DIR/lib
-TLIB_PATH=$RPL_PATH/rocprofiler
-TTLIB_PATH=$TT_DIR/tool
-ROCM_LIB_PATH=$ROOT_DIR/lib
-PROF_BIN_DIR=$ROOT_DIR/libexec/rocprofiler
+RPL_PATH=$ROOT_DIR/@LIBDIR@
+TLIB_PATH=$RPL_PATH/rocprofiler
+TTLIB_PATH=$TT_DIR/@LIBDIR@
+ROCM_LIB_PATH=$ROOT_DIR/@LIBDIR@
+PROF_BIN_DIR=$ROOT_DIR/bin
+PROF_LIBEXEC_DIR=$ROOT_DIR/libexec/rocprofiler
 
 if [ -z "$ROCP_PYTHON_VERSION" ] ; then
   ROCP_PYTHON_VERSION=python3
@@ -70,7 +71,7 @@ export ROCP_HSA_INTERCEPT=1
 # Disabling internal ROC Profiler proxy queue (simple version supported for testing purposes)
 unset ROCP_PROXY_QUEUE
 # ROC Profiler metrics definition
-export ROCP_METRICS=$TLIB_PATH/metrics.xml
+export ROCP_METRICS=$TLIB_PATH/metrics.xml
 # Disable AQL-profile read API
 export AQLPROFILE_READ_API=0
 # ROC Profiler package path
@@ -104,7 +105,7 @@ usage() {
   bin_name=`basename $0`
   echo "ROCm Profiling Library (RPL) run script, a part of ROCprofiler library package."
   echo "Full path: $BIN_DIR/$bin_name"
-  echo "Metrics definition: $TLIB_PATH/metrics.xml"
+  echo "Metrics definition: $TLIB_PATH/metrics.xml"
   echo ""
   echo "Usage:"
   echo "  $bin_name [-h] [--list-basic] [--list-derived] [-i <input .txt/.xml file>] [-o <output CSV file>] <app command line>"
@@ -212,7 +213,7 @@ fi
 # setting ROCM_LIB_PATH
 set_rocm_lib_path() {
 
-  for ROCM_LIB_PATH in "$ROOT_DIR/lib" "$ROOT_DIR/lib64" ; do
+  for ROCM_LIB_PATH in $TLIB_PATH ; do
      if [ -d "$ROCM_LIB_PATH" ]; then
         return 0
      fi
@@ -226,7 +227,7 @@ set_rocm_lib_path() {
 OUTPUT_LIST=""
 run() {
   if ! set_rocm_lib_path ; then
-     echo " Fatal could not find ROCm lib directory "
+     echo " Fatal could not find ROCm @LIBDIR@ directory "
      fatal
   fi
   export ROCP_INPUT="$1"
@@ -376,11 +377,11 @@ while [ 1 ] ; do
     export ROCP_METRICS="$2"
   elif [ "$1" = "--list-basic" ] ; then
     export ROCP_INFO=b
-    HSA_TOOLS_LIB="$MY_HSA_TOOLS_LIB" eval "$TLIB_PATH/rocprof-ctrl"
+    HSA_TOOLS_LIB="$MY_HSA_TOOLS_LIB" eval "$TLIB_PATH/rocprof-ctrl"
     exit 1
   elif [ "$1" = "--list-derived" ] ; then
     export ROCP_INFO=d
-    HSA_TOOLS_LIB="$MY_HSA_TOOLS_LIB" eval "$TLIB_PATH/rocprof-ctrl"
+    HSA_TOOLS_LIB="$MY_HSA_TOOLS_LIB" eval "$TLIB_PATH/rocprof-ctrl"
     exit 1
   elif [ "$1" = "--basenames" ] ; then
     if [ "$2" = "on" ] ; then
@@ -541,7 +542,7 @@ elif [ "$input_type" = "txt" -o "$input_
   mkdir -p $RES_DIR
   echo "RPL: output dir '$RES_DIR'"
   if [ "$input_type" = "txt" ] ; then
-    $PROF_BIN_DIR/txt2xml.sh $INPUT_FILE $RES_DIR
+    $PROF_LIBEXEC_DIR/txt2xml.sh $INPUT_FILE $RES_DIR
   else
     echo "<metric></metric>" > $RES_DIR/input.xml
   fi
@@ -573,9 +574,9 @@ if [ -n "$csv_output" ] ; then
   merge_output $OUTPUT_LIST
   if [ "$GEN_STATS" = "1" ] ; then
     db_output=$(echo $csv_output | sed "s/\.csv/.db/")
-    $ROCP_PYTHON_VERSION $PROF_BIN_DIR/tblextr.py $db_output $OUTPUT_LIST
+    $ROCP_PYTHON_VERSION $PROF_LIBEXEC_DIR/tblextr.py $db_output $OUTPUT_LIST
   else
-    $ROCP_PYTHON_VERSION $PROF_BIN_DIR/tblextr.py $csv_output $OUTPUT_LIST
+    $ROCP_PYTHON_VERSION $PROF_LIBEXEC_DIR/tblextr.py $csv_output $OUTPUT_LIST
   fi
   if [ "$?" -ne 0 ] ; then
     echo "Profiling data corrupted: '$OUTPUT_LIST'" | tee "$ROCPROFILER_SESS/error"
Only in rocprofiler-rocm-5.2.3/bin: rpl_run.sh.orig
Only in rocprofiler-rocm-5.2.3/bin: rpl_run.sh.rej
diff -urp rocprofiler-rocm-5.2.3.orig/build.sh rocprofiler-rocm-5.2.3/build.sh
--- rocprofiler-rocm-5.2.3.orig/build.sh	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/build.sh	2023-09-28 11:31:05.772195687 -0700
@@ -2,5 +2,5 @@
 BIN_DIR=`dirname $0`
 BLD_DIR=$BIN_DIR/build
 
-export CMAKE_PREFIX_PATH=/opt/rocm/include/hsa:/opt/rocm
+export CMAKE_PREFIX_PATH=@EPREFIX_ROCM_PATH@/include/hsa:@EPREFIX_ROCM_PATH@
 rm -rf $BLD_DIR && mkdir $BLD_DIR && cd $BLD_DIR && cmake .. && make -j && make mytest
Only in rocprofiler-rocm-5.2.3: build.sh.orig
Only in rocprofiler-rocm-5.2.3: build.sh.rej
diff -urp rocprofiler-rocm-5.2.3.orig/CMakeLists.txt rocprofiler-rocm-5.2.3/CMakeLists.txt
--- rocprofiler-rocm-5.2.3.orig/CMakeLists.txt	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/CMakeLists.txt	2023-09-28 11:35:13.432192838 -0700
@@ -23,10 +23,10 @@
 cmake_minimum_required ( VERSION 2.8.12 )
 
 # Install prefix
-set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH "Install path prefix default")
+set(CMAKE_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE PATH "Install path prefix default")
 
 ## Verbose output.
-set ( CMAKE_VERBOSE_MAKEFILE TRUE CACHE BOOL "Verbose Output" FORCE )
+#_cmake_modify_IGNORE set ( CMAKE_VERBOSE_MAKEFILE TRUE CACHE BOOL "Verbose Output" FORCE )
 
 ## Set module name and project name.
 set ( ROCPROFILER_NAME "rocprofiler" )
@@ -82,7 +82,7 @@ if(USE_PROF_API EQUAL 1)
         HINTS
             ${PROF_API_HEADER_PATH}
         PATHS
-            /opt/rocm/roctracer
+            @EPREFIX_ROCM_PATH@/roctracer
         PATH_SUFFIXES
             include/ext
     )
@@ -116,7 +116,7 @@ if ( DEFINED CMAKE_INSTALL_PREFIX )
   get_filename_component ( prefix_name ${CMAKE_INSTALL_PREFIX} NAME )
   get_filename_component ( prefix_dir ${CMAKE_INSTALL_PREFIX} DIRECTORY )
   if ( prefix_name STREQUAL ${DEST_NAME} )
-    set ( CMAKE_INSTALL_PREFIX ${prefix_dir} )
+#_cmake_modify_IGNORE     set ( CMAKE_INSTALL_PREFIX ${prefix_dir} )
   endif ()
 endif ()
 if ( DEFINED CPACK_PACKAGING_INSTALL_PREFIX )
@@ -160,10 +160,10 @@ install ( FILES
 install ( FILES
           ${CMAKE_CURRENT_SOURCE_DIR}/test/tool/metrics.xml
           ${CMAKE_CURRENT_SOURCE_DIR}/test/tool/gfx_metrics.xml
-          DESTINATION lib/${DEST_NAME} )
+          DESTINATION ${CMAKE_INSTALL_LIBDIR}/${DEST_NAME} )
 # librocprof-tool.so
-install ( FILES ${PROJECT_BINARY_DIR}/test/librocprof-tool.so DESTINATION lib/${DEST_NAME} )
-install ( FILES ${PROJECT_BINARY_DIR}/test/rocprof-ctrl DESTINATION lib/${DEST_NAME}
+install ( FILES ${PROJECT_BINARY_DIR}/test/librocprof-tool.so DESTINATION ${CMAKE_INSTALL_LIBDIR}/${DEST_NAME} )
+install ( FILES ${PROJECT_BINARY_DIR}/test/rocprof-ctrl DESTINATION ${CMAKE_INSTALL_LIBDIR}/${DEST_NAME}
           PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE )
 
 # File reorg Backward compatibility
Only in rocprofiler-rocm-5.2.3: CMakeLists.txt.orig
Only in rocprofiler-rocm-5.2.3: CMakeLists.txt.rej
diff -urp rocprofiler-rocm-5.2.3.orig/cmake_modules/env.cmake rocprofiler-rocm-5.2.3/cmake_modules/env.cmake
--- rocprofiler-rocm-5.2.3.orig/cmake_modules/env.cmake	2023-09-28 11:26:50.456198624 -0700
+++ rocprofiler-rocm-5.2.3/cmake_modules/env.cmake	2023-09-28 11:35:45.352192471 -0700
@@ -105,10 +105,10 @@ elseif ( ${CMAKE_SYSTEM_PROCESSOR} STREQ
 endif ()
 
 ## Find hsa-runtime
-find_package(hsa-runtime64 1.0 REQUIRED HINTS ${CMAKE_INSTALL_PREFIX} PATHS /opt/rocm)
+find_package(hsa-runtime64 1.0 REQUIRED HINTS ${CMAKE_INSTALL_PREFIX} PATHS @ESYSROOT_ROCM_PATH@)
 
 # find KFD thunk
-find_package(hsakmt 1.0 REQUIRED HINTS ${CMAKE_INSTALL_PREFIX} PATHS /opt/rocm)
+find_package(hsakmt 1.0 REQUIRED HINTS ${CMAKE_INSTALL_PREFIX} PATHS @ESYSROOT_ROCM_PATH@)
 
 ## Find ROCm
 find_library ( HSA_KMT_LIB "libhsakmt.so" )
Only in rocprofiler-rocm-5.2.3/cmake_modules: env.cmake.orig
Only in rocprofiler-rocm-5.2.3/cmake_modules: env.cmake.rej
diff -urp rocprofiler-rocm-5.2.3.orig/Readme.txt rocprofiler-rocm-5.2.3/Readme.txt
--- rocprofiler-rocm-5.2.3.orig/Readme.txt	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/Readme.txt	2023-09-28 11:27:11.376198383 -0700
@@ -24,7 +24,7 @@ $ export CMAKE_DEBUG_TRACE=1 # 1 to enab
 To build with the current installed ROCM:
 
 $ cd .../rocprofiler
-$ export CMAKE_PREFIX_PATH=/opt/rocm/include/hsa:/opt/rocm
+$ export CMAKE_PREFIX_PATH=@ESYSROOT_ROCM_PATH@/include/hsa:@ESYSROOT_ROCM_PATH@
 $ mkdir build
 $ cd build
 $ cmake ..
Only in rocprofiler-rocm-5.2.3: Readme.txt.orig
diff -urp rocprofiler-rocm-5.2.3.orig/rocprofiler-backward-compat.cmake rocprofiler-rocm-5.2.3/rocprofiler-backward-compat.cmake
--- rocprofiler-rocm-5.2.3.orig/rocprofiler-backward-compat.cmake	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/rocprofiler-backward-compat.cmake	2023-09-28 11:38:27.032190611 -0700
@@ -23,7 +23,7 @@ set(ROCPROF_BUILD_DIR ${CMAKE_CURRENT_BI
 set(ROCPROF_WRAPPER_DIR ${ROCPROF_BUILD_DIR}/wrapper_dir)
 set(ROCPROF_WRAPPER_INC_DIR ${ROCPROF_WRAPPER_DIR}/include)
 set(ROCPROF_WRAPPER_BIN_DIR ${ROCPROF_WRAPPER_DIR}/bin)
-set(ROCPROF_WRAPPER_LIB_DIR ${ROCPROF_WRAPPER_DIR}/lib)
+set(ROCPROF_WRAPPER_LIB_DIR ${ROCPROF_WRAPPER_DIR}/@LIBDIR@)
 set(ROCPROF_WRAPPER_TOOL_DIR ${ROCPROF_WRAPPER_DIR}/tool)
 
 #Function to generate header template file
@@ -48,7 +48,7 @@ function(create_header_template)
     LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
     OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
    THE SOFTWARE.
-   */\n\n#ifndef @include_guard@\n#define @include_guard@ \n\n#pragma message(\"This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with rocprofiler\")\n@include_statements@ \n\n#endif")
+   */\n\n#ifndef @include_guard@\n#define @include_guard@ \n\n#pragma message(\"This file is deprecated. Use file from include path @EPREFIX_ROCM_PATH@/include/ and prefix with rocprofiler\")\n@include_statements@ \n\n#endif")
 endfunction()
 
 #use header template file and generate wrapper header files
@@ -104,7 +104,7 @@ function(create_library_symlink)
     add_custom_target(link_${file_name} ALL
                   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
                   COMMAND ${CMAKE_COMMAND} -E create_symlink
-                  ../../lib/${file_name} ${ROCPROF_WRAPPER_LIB_DIR}/${file_name})
+                  ../../${CMAKE_INSTALL_LIBDIR}/${file_name} ${ROCPROF_WRAPPER_LIB_DIR}/${file_name})
   endforeach()
   #create symlink to rocprofiler/tool/libtool.so
   # With File reorg,tool renamed to rocprof-tool
@@ -114,7 +114,7 @@ function(create_library_symlink)
   add_custom_target(link_${LIB_TOOL} ALL
                    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
                    COMMAND ${CMAKE_COMMAND} -E create_symlink
-                   ../../lib/${ROCPROFILER_NAME}/${LIB_ROCPROFTOOL} ${ROCPROF_WRAPPER_TOOL_DIR}/${LIB_TOOL})
+                   ../../${CMAKE_INSTALL_LIBDIR}/${ROCPROFILER_NAME}/${LIB_ROCPROFTOOL} ${ROCPROF_WRAPPER_TOOL_DIR}/${LIB_TOOL})
   #create symlink to test binary
   #since its saved in lib folder , the code for the same is added here
   # With File reorg ,binary name changed from ctrl to rocprof-ctrl
@@ -123,7 +123,7 @@ function(create_library_symlink)
   add_custom_target(link_${TEST_CTRL} ALL
                     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
                     COMMAND ${CMAKE_COMMAND} -E create_symlink
-                    ../../lib/${ROCPROFILER_NAME}/${TEST_ROCPROFCTRL} ${ROCPROF_WRAPPER_TOOL_DIR}/${TEST_CTRL})
+                    ../../${CMAKE_INSTALL_LIBDIR}/${ROCPROFILER_NAME}/${TEST_ROCPROFCTRL} ${ROCPROF_WRAPPER_TOOL_DIR}/${TEST_CTRL})
 endfunction()
 
 #Creater a template for header file
Only in rocprofiler-rocm-5.2.3: rocprofiler-backward-compat.cmake.orig
Only in rocprofiler-rocm-5.2.3: rocprofiler-backward-compat.cmake.rej
diff -urp rocprofiler-rocm-5.2.3.orig/test/CMakeLists.txt rocprofiler-rocm-5.2.3/test/CMakeLists.txt
--- rocprofiler-rocm-5.2.3.orig/test/CMakeLists.txt	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/test/CMakeLists.txt	2023-09-28 11:27:11.376198383 -0700
@@ -21,7 +21,7 @@
 ################################################################################
 
 cmake_minimum_required ( VERSION 2.8.12 )
-set ( CMAKE_VERBOSE_MAKEFILE TRUE CACHE BOOL "Verbose Output" FORCE )
+#_cmake_modify_IGNORE set ( CMAKE_VERBOSE_MAKEFILE TRUE CACHE BOOL "Verbose Output" FORCE )
 
 set ( EXE_NAME "rocprof-ctrl" )
 
diff -urp rocprofiler-rocm-5.2.3.orig/test/run.sh rocprofiler-rocm-5.2.3/test/run.sh
--- rocprofiler-rocm-5.2.3.orig/test/run.sh	2022-05-20 16:39:55.000000000 -0700
+++ rocprofiler-rocm-5.2.3/test/run.sh	2023-09-28 11:27:11.376198383 -0700
@@ -57,7 +57,7 @@ eval_test() {
 }
 
 # paths to ROC profiler and oher libraries
-export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD:$PWD/../../lib:/opt/rocm/lib:/opt/rocm/lib/rocprofiler
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD:$PWD/../../@LIBDIR@:@EPREFIX_ROCM_PATH@/@LIBDIR@:@EPREFIX_ROCM_PATH@/@LIBDIR@/rocprofiler:/opt/rocm-${ROCM_VERSION}/lib
 
 # enable tools load failure reporting
 export HSA_TOOLS_REPORT_LOAD_FAILURE=1
