--- a/CMakeLists.txt.orig	2019-12-23 15:11:11.000000000 -0800
+++ b/CMakeLists.txt	2020-08-07 19:37:24.819533078 -0700
@@ -297,7 +297,7 @@ add_executable(wasm-metadce ${wasm-metad
 target_link_libraries(wasm-metadce ${CMAKE_THREAD_LIBS_INIT})
 set_property(TARGET wasm-metadce PROPERTY CXX_STANDARD 14)
 set_property(TARGET wasm-metadce PROPERTY CXX_STANDARD_REQUIRED ON)
-install(TARGETS wasm-metadce DESTINATION bin)
+install(TARGETS wasm-metadce DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 set(asm2wasm_SOURCES
   src/tools/asm2wasm.cpp
@@ -351,7 +351,7 @@ add_executable(wasm-ctor-eval ${wasm-cto
 target_link_libraries(wasm-ctor-eval ${CMAKE_THREAD_LIBS_INIT})
 set_property(TARGET wasm-ctor-eval PROPERTY CXX_STANDARD 14)
 set_property(TARGET wasm-ctor-eval PROPERTY CXX_STANDARD_REQUIRED ON)
-install(TARGETS wasm-ctor-eval DESTINATION bin)
+install(TARGETS wasm-ctor-eval DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 set(wasm-reduce_SOURCES
   src/tools/wasm-reduce.cpp
