diff -urp libappimage-1.0.4-5.orig/cmake/dependencies.cmake libappimage-1.0.4-5/cmake/dependencies.cmake
--- libappimage-1.0.4-5.orig/cmake/dependencies.cmake	2022-05-11 11:28:34.000000000 -0700
+++ libappimage-1.0.4-5/cmake/dependencies.cmake	2023-01-11 18:13:23.534467833 -0800
@@ -112,7 +112,7 @@ if (NOT LIBAPPIMAGE_SHARED_ONLY)
     else()
         message(STATUS "Using system squashfuse")
 
-        import_pkgconfig_target(TARGET_NAME libsquashfuse PKGCONFIG_TARGET squashfuse)
+        import_pkgconfig_target(TARGET_NAME libsquashfuse PKGCONFIG_TARGET squashfuse_appimage)
     endif()
 
 
Only in libappimage-1.0.4-5/cmake: dependencies.cmake.orig
Only in libappimage-1.0.4-5/cmake: dependencies.cmake.rej
