--- libappimage-1.0.4-5/src/libappimage/desktop_integration/Thumbnailer.h.orig	2022-05-11 11:28:34.000000000 -0700
+++ libappimage-1.0.4-5/src/libappimage/desktop_integration/Thumbnailer.h	2023-02-07 17:16:53.723259812 -0800
@@ -61,11 +61,11 @@ namespace appimage {
 
             static constexpr const char* thumbnailFileExtension = ".png";
 
-            static constexpr const char* normalThumbnailsPrefix = "thumbnails/normal";
+            const char* normalThumbnailsPrefix = "thumbnails/normal";
 
             bf::path getNormalThumbnailPath(const std::string& canonicalPathMd5) const;
 
-            static constexpr const char* largeThumbnailPrefix = "thumbnails/large";
+            const char* largeThumbnailPrefix = "thumbnails/large";
 
             bf::path getLargeThumbnailPath(const std::string& canonicalPathMd5) const;
 
