diff -x '*/*.rej' -urpN libappimage-1.0.4-5.orig/cmake/libappimage.pc.in libappimage-1.0.4-5/cmake/libappimage.pc.in
--- libappimage-1.0.4-5.orig/cmake/libappimage.pc.in	2022-05-11 11:28:34.000000000 -0700
+++ libappimage-1.0.4-5/cmake/libappimage.pc.in	2023-01-11 18:15:32.351583431 -0800
@@ -7,6 +7,7 @@ Name: @PROJECT_NAME@
 Description: AppImage management and desktop integration
 Version: @libappimage_VERSION@
 URL: https://github.com/AppImage/libappimage
-
-Libs: -L${libdir} -lappimage
-Cflags: -I${includedir}
+Requires: glib-2.0 gobject-2.0 gio-2.0 zlib cairo librsvg-2.0 libarchive
+# pangocairo gio-2.0 glib-2.0 libappimage_shared libappimage_hashlib libarchive libbsd libcroco-0.6 libcrypto liblzma librsvg-2.0 libxml-2.0 mount squashfuse_appimage xshmfence zlib
+Libs: -L${libdir} -lappimage -lbz2 -lstdc++ -lboost_filesystem -L${libdir}/XdgUtils -lXdgUtilsBaseDir -lXdgUtilsDesktopEntry
+Cflags: -I${includedir}/appimage -I${includedir}/XdgUtils
diff -x '*/*.rej' -urpN libappimage-1.0.4-5.orig/cmake/libappimage_hashlib.pc.in libappimage-1.0.4-5/cmake/libappimage_hashlib.pc.in
--- libappimage-1.0.4-5.orig/cmake/libappimage_hashlib.pc.in	1969-12-31 16:00:00.000000000 -0800
+++ libappimage-1.0.4-5/cmake/libappimage_hashlib.pc.in	2023-01-11 18:15:32.351583431 -0800
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+# Use prefix as base path to make the package relocatable
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name: libappimage_hashlib
+Description: MD5 hash library for libappimage
+Version: @libappimage_VERSION@
+URL: https://github.com/AppImage/libappimage
+Libs: -L${libdir} -l:libappimage_hashlib.a
+Cflags: -I${includedir}/libappimage_hashlib
diff -x '*/*.rej' -urpN libappimage-1.0.4-5.orig/cmake/libappimage_shared.pc.in libappimage-1.0.4-5/cmake/libappimage_shared.pc.in
--- libappimage-1.0.4-5.orig/cmake/libappimage_shared.pc.in	1969-12-31 16:00:00.000000000 -0800
+++ libappimage-1.0.4-5/cmake/libappimage_shared.pc.in	2023-01-11 18:15:32.355583589 -0800
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+# Use prefix as base path to make the package relocatable
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name: libappimage_shared
+Description: Helper functions for analysis of AppImages
+Version: @libappimage_VERSION@
+URL: https://github.com/AppImage/libappimage
+Libs: -L${libdir} -l:libappimage_shared.a
+Cflags: -I${includedir}/appimage
diff -x '*/*.rej' -urpN libappimage-1.0.4-5.orig/cmake/libappimage_static.pc.in libappimage-1.0.4-5/cmake/libappimage_static.pc.in
--- libappimage-1.0.4-5.orig/cmake/libappimage_static.pc.in	1969-12-31 16:00:00.000000000 -0800
+++ libappimage-1.0.4-5/cmake/libappimage_static.pc.in	2023-01-11 18:15:32.355583589 -0800
@@ -0,0 +1,13 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+# Use prefix as base path to make the package relocatable
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name: libappimage_static
+Description: AppImage management and desktop integration
+Version: @libappimage_VERSION@
+URL: https://github.com/AppImage/libappimage
+Requires: glib-2.0 gobject-2.0 gio-2.0 zlib cairo librsvg-2.0 libarchive
+# pangocairo gio-2.0 glib-2.0 libappimage_shared libappimage_hashlib libarchive libbsd libcroco-0.6 libcrypto liblzma librsvg-2.0 libxml-2.0 mount squashfuse_appimage xshmfence zlib
+Libs: -L${libdir} -l:libappimage_static.a -lbz2 -lstdc++ -lboost_filesystem -L${libdir}/XdgUtils -lXdgUtilsBaseDir -lXdgUtilsDesktopEntry
+Cflags: -I${includedir}/appimage -I${includedir}/XdgUtils
diff -x '*/*.rej' -urpN libappimage-1.0.4-5.orig/cmake/xdg-basedir.pc.in libappimage-1.0.4-5/cmake/xdg-basedir.pc.in
--- libappimage-1.0.4-5.orig/cmake/xdg-basedir.pc.in	1969-12-31 16:00:00.000000000 -0800
+++ libappimage-1.0.4-5/cmake/xdg-basedir.pc.in	2023-01-11 18:15:32.355583589 -0800
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+# Use prefix as base path to make the package relocatable
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name: xdg-basedir
+Description: XDG functions for libappimage
+Version: @libappimage_VERSION@
+URL: https://github.com/AppImage/libappimage
+Libs: -L${libdir} -l:libxdg-basedir.a
+Cflags: -I${includedir}/appimage
diff -x '*/*.rej' -urpN libappimage-1.0.4-5.orig/src/CMakeLists.txt libappimage-1.0.4-5/src/CMakeLists.txt
--- libappimage-1.0.4-5.orig/src/CMakeLists.txt	2022-05-11 11:28:34.000000000 -0700
+++ libappimage-1.0.4-5/src/CMakeLists.txt	2023-01-11 18:16:51.374721918 -0800
@@ -48,8 +48,17 @@ install(
 if(NOT LIBAPPIMAGE_SHARED_ONLY)
     # pkg-config
     configure_file("${PROJECT_SOURCE_DIR}/cmake/libappimage.pc.in" "${PROJECT_BINARY_DIR}/libappimage.pc" @ONLY)
+    configure_file("${PROJECT_SOURCE_DIR}/cmake/libappimage_shared.pc.in" "${PROJECT_BINARY_DIR}/libappimage_shared.pc" @ONLY)
+    configure_file("${PROJECT_SOURCE_DIR}/cmake/libappimage_static.pc.in" "${PROJECT_BINARY_DIR}/libappimage_static.pc" @ONLY)
+    configure_file("${PROJECT_SOURCE_DIR}/cmake/libappimage_hashlib.pc.in" "${PROJECT_BINARY_DIR}/libappimage_hashlib.pc" @ONLY)
+    configure_file("${PROJECT_SOURCE_DIR}/cmake/xdg-basedir.pc.in" "${PROJECT_BINARY_DIR}/xdg-basedir.pc" @ONLY)
 
-    install(FILES "${PROJECT_BINARY_DIR}/libappimage.pc"
+    install(FILES
+        "${PROJECT_BINARY_DIR}/libappimage.pc"
+        "${PROJECT_BINARY_DIR}/libappimage_shared.pc"
+        "${PROJECT_BINARY_DIR}/libappimage_static.pc"
+        "${PROJECT_BINARY_DIR}/libappimage_hashlib.pc"
+        "${PROJECT_BINARY_DIR}/xdg-basedir.pc"
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
         COMPONENT libappimage-dev
     )
