diff -urp libappimage-1.0.4-5.orig/tests/libappimage/desktop_integration/TestIntegrationManager.cpp libappimage-1.0.4-5/tests/libappimage/desktop_integration/TestIntegrationManager.cpp
--- libappimage-1.0.4-5.orig/tests/libappimage/desktop_integration/TestIntegrationManager.cpp	2022-05-11 11:28:34.000000000 -0700
+++ libappimage-1.0.4-5/tests/libappimage/desktop_integration/TestIntegrationManager.cpp	2023-01-12 16:16:22.926845542 -0800
@@ -4,6 +4,7 @@
 // library headers
 #include <gtest/gtest.h>
 #include <boost/filesystem.hpp>
+#include <fstream>
 
 // local
 #include "appimage/desktop_integration/exceptions.h"
@@ -31,7 +32,7 @@ protected:
 
     void createStubFile(const bf::path& path, const std::string& content = "") {
         bf::create_directories(path.parent_path());
-        bf::ofstream f(path);
+        std::ofstream f(path.string());
         f << content;
     }
 };
diff -urp libappimage-1.0.4-5.orig/tests/libappimage/desktop_integration/TestThumbnailer.cpp libappimage-1.0.4-5/tests/libappimage/desktop_integration/TestThumbnailer.cpp
--- libappimage-1.0.4-5.orig/tests/libappimage/desktop_integration/TestThumbnailer.cpp	2022-05-11 11:28:34.000000000 -0700
+++ libappimage-1.0.4-5/tests/libappimage/desktop_integration/TestThumbnailer.cpp	2023-01-12 16:16:24.494907622 -0800
@@ -4,6 +4,7 @@
 // library headers
 #include <gtest/gtest.h>
 #include <boost/filesystem.hpp>
+#include <fstream>
 
 // local
 #include "appimage/desktop_integration/exceptions.h"
@@ -30,7 +31,7 @@ protected:
 
     void createStubFile(const bf::path& path, const std::string& content = "") {
         bf::create_directories(path.parent_path());
-        bf::ofstream f(path);
+        std::ofstream f(path.string());
         f << content;
     }
 };
