diff -urp libappimage-1.0.2.orig/cmake/libappimage.pc.in libappimage-1.0.2/cmake/libappimage.pc.in
--- libappimage-1.0.2.orig/cmake/libappimage.pc.in	2019-06-06 19:42:43.000000000 -0700
+++ libappimage-1.0.2/cmake/libappimage.pc.in	2020-08-11 19:32:06.976319363 -0700
@@ -7,6 +7,6 @@ Name: @PROJECT_NAME@
 Description: AppImage management and desktop integration
 Version: @libappimage_VERSION@
 URL: https://github.com/AppImage/libappimage
-
-Libs: -L${libdir} -lappimage
-Cflags: -I${includedir}
+Requires: pangocairo gio-2.0 glib-2.0 libarchive libacl libbsd libcroco-0.6 libcrypto liblzma librsvg-2.0 libxml-2.0 mount squashfuse_appimage xshmfence zlib
+Libs: -L${libdir} -lappimage -lbz2 -lstdc++ -lboost_filesystem -L${libdir}/XdgUtils -lXdgUtilsBaseDir -lXdgUtilsDesktopEntry
+Cflags: -I${includedir} -I${includedir}/XdgUtils
