--- libappimage-1.0.2/src/libappimage/CMakeLists.txt.orig	2019-06-06 19:42:43.000000000 -0700
+++ libappimage-1.0.2/src/libappimage/CMakeLists.txt	2020-08-11 11:10:27.076774539 -0700
@@ -15,7 +15,7 @@ if(LIBAPPIMAGE_DESKTOP_INTEGRATION_ENABL
     list(APPEND libappimage_sources "$<TARGET_OBJECTS:appimage_desktop_integration>")
 endif()
 
-add_library(libappimage_static STATIC ${libappimage_sources})
+add_library(libappimage_static STATIC ${libappimage_sources} libappimage_legacy.cpp)
 add_library(libappimage SHARED ${libappimage_sources} libappimage_legacy.cpp)
 
 configure_file(config.h.in ${PROJECT_BINARY_DIR}/generated-headers/appimage/config.h)
