--- libappimage-1.0.4-5/tests/libappimage/CMakeLists.txt.orig	2023-02-07 16:22:21.895151998 -0800
+++ libappimage-1.0.4-5/tests/libappimage/CMakeLists.txt	2023-02-07 16:37:24.331181735 -0800
@@ -23,7 +23,7 @@ if (NOT LIBAPPIMAGE_SHARED_ONLY)
     )
 
     target_include_directories(test_libappimage++ PRIVATE "${PROJECT_SOURCE_DIR}/src/libappimage")
-    target_link_libraries(test_libappimage++ libappimage Boost::filesystem libarchive libsquashfuse XdgUtils::DesktopEntry XdgUtils::BaseDir gtest gtest_main)
+    target_link_libraries(test_libappimage++ appimage_desktop_integration libappimage Boost::filesystem libarchive libsquashfuse XdgUtils::DesktopEntry XdgUtils::BaseDir gtest gtest_main)
 
     add_test(test_libappimage++ test_libappimage++)
 endif()
