diff -urp flac-1.3.3.orig/src/libFLAC/Makefile.am flac-1.3.3/src/libFLAC/Makefile.am
--- flac-1.3.3.orig/src/libFLAC/Makefile.am	2018-08-20 01:17:21.060232804 -0700
+++ flac-1.3.3/src/libFLAC/Makefile.am	2021-10-10 10:00:04.585390525 -0700
@@ -49,7 +49,7 @@ endif
 endif
 endif
 
-AM_CFLAGS = $(DEBUGCFLAGS) $(CPUCFLAGS) @OGG_CFLAGS@
+AM_CFLAGS = $(DEBUGCFLAGS) $(CPUCFLAGS) @OGG_CFLAGS@ -fPIC
 
 if FLaC__NO_ASM
 else
