diff -urp Tensile-rocm-5.6.0.orig/Tensile/Common.py Tensile-rocm-5.6.0/Tensile/Common.py
--- Tensile-rocm-5.6.0.orig/Tensile/Common.py	2023-08-20 17:56:18.795576738 -0700
+++ Tensile-rocm-5.6.0/Tensile/Common.py	2023-08-20 17:56:41.747577307 -0700
@@ -2054,7 +2054,7 @@ def assignGlobalParameters( config ):
     if os.name == "nt":
       globalParameters["AssemblerPath"] = locateExe(globalParameters["ROCmBinPath"], "clang++.exe")
     else:
-      globalParameters["AssemblerPath"] = locateExe(os.path.join(globalParameters["ROCmPath"], "llvm/bin"), "clang++")
+      globalParameters["AssemblerPath"] = locateExe(os.path.join(globalParameters["ROCmPath"], "lib/llvm/@LLVM_SLOT@/bin"), "clang++")
 
   globalParameters["ROCmSMIPath"] = locateExe(globalParameters["ROCmBinPath"], "rocm-smi")
 
@@ -2066,7 +2066,7 @@ def assignGlobalParameters( config ):
     if os.name == "nt":
       globalParameters["ClangOffloadBundlerPath"] = locateExe(globalParameters["ROCmBinPath"], "clang-offload-bundler.exe")
     else:
-      globalParameters["ClangOffloadBundlerPath"] = locateExe(os.path.join(globalParameters["ROCmPath"], "llvm/bin"), "clang-offload-bundler")
+      globalParameters["ClangOffloadBundlerPath"] = locateExe(os.path.join(globalParameters["ROCmPath"], "lib/llvm/@LLVM_SLOT@/bin"), "clang-offload-bundler")
 
   if "ROCmAgentEnumeratorPath" in config:
     globalParameters["ROCmAgentEnumeratorPath"] = config["ROCmAgentEnumeratorPath"]
diff -urp Tensile-rocm-5.6.0.orig/Tensile/Tests/hipModuleLoad_timing/Makefile Tensile-rocm-5.6.0/Tensile/Tests/hipModuleLoad_timing/Makefile
--- Tensile-rocm-5.6.0.orig/Tensile/Tests/hipModuleLoad_timing/Makefile	2023-06-12 10:51:58.000000000 -0700
+++ Tensile-rocm-5.6.0/Tensile/Tests/hipModuleLoad_timing/Makefile	2023-08-20 17:57:47.463578938 -0700
@@ -22,10 +22,10 @@
 #
 ################################################################################
 
-CXX=/opt/rocm/hip/bin/hipcc
-LIBFLAGS=-L/opt/rocm/hip/lib/
+CXX=@ESYSROOT@/usr/bin/hipcc
+LIBFLAGS=-L@ESYSROOT@/usr/@LIBDIR@/
 LIBS=-lamdhip64
-INCFLAGS=-I/opt/rocm/hip/include/
+INCFLAGS=-I@ESYSROOT@/usr/include/hip
 
 hipModuleLoadTiming.out: hipModuleLoadTiming.o
 	$(CXX) -o $@ $(LIBFLAGS) $^
