diff -uNr work.orig/vboxdrv/include/iprt/string.h work/vboxdrv/include/iprt/string.h
--- work.orig/vboxdrv/include/iprt/string.h	2017-08-16 10:13:25.793010730 +0300
+++ work/vboxdrv/include/iprt/string.h	2017-08-16 10:14:40.830494997 +0300
@@ -32,11 +32,8 @@
 #include <iprt/stdarg.h>
 #include <iprt/err.h> /* for VINF_SUCCESS */
 #if defined(RT_OS_LINUX) && defined(__KERNEL__)
-  RT_C_DECLS_BEGIN
-# define new newhack /* string.h: strreplace */
+/* no C++ hacks ('new' etc) here anymore! */ 
 # include <linux/string.h>
-# undef new
-  RT_C_DECLS_END
 
 #elif defined(IN_XF86_MODULE) && !defined(NO_ANSIC)
   RT_C_DECLS_BEGIN
diff -uNr work.orig/vboxnetadp/include/iprt/string.h work/vboxnetadp/include/iprt/string.h
--- work.orig/vboxnetadp/include/iprt/string.h	2017-08-16 10:13:25.791010717 +0300
+++ work/vboxnetadp/include/iprt/string.h	2017-08-16 10:14:40.830494997 +0300
@@ -32,11 +32,8 @@
 #include <iprt/stdarg.h>
 #include <iprt/err.h> /* for VINF_SUCCESS */
 #if defined(RT_OS_LINUX) && defined(__KERNEL__)
-  RT_C_DECLS_BEGIN
-# define new newhack /* string.h: strreplace */
+/* no C++ hacks ('new' etc) here anymore! */ 
 # include <linux/string.h>
-# undef new
-  RT_C_DECLS_END
 
 #elif defined(IN_XF86_MODULE) && !defined(NO_ANSIC)
   RT_C_DECLS_BEGIN
diff -uNr work.orig/vboxnetflt/include/iprt/string.h work/vboxnetflt/include/iprt/string.h
--- work.orig/vboxnetflt/include/iprt/string.h	2017-08-16 10:13:25.787010691 +0300
+++ work/vboxnetflt/include/iprt/string.h	2017-08-16 10:14:40.830494997 +0300
@@ -32,11 +32,8 @@
 #include <iprt/stdarg.h>
 #include <iprt/err.h> /* for VINF_SUCCESS */
 #if defined(RT_OS_LINUX) && defined(__KERNEL__)
-  RT_C_DECLS_BEGIN
-# define new newhack /* string.h: strreplace */
+/* no C++ hacks ('new' etc) here anymore! */ 
 # include <linux/string.h>
-# undef new
-  RT_C_DECLS_END
 
 #elif defined(IN_XF86_MODULE) && !defined(NO_ANSIC)
   RT_C_DECLS_BEGIN
diff -uNr work.orig/vboxpci/include/iprt/string.h work/vboxpci/include/iprt/string.h
--- work.orig/vboxpci/include/iprt/string.h	2017-08-16 10:13:25.789010704 +0300
+++ work/vboxpci/include/iprt/string.h	2017-08-16 10:14:40.830494997 +0300
@@ -32,11 +32,8 @@
 #include <iprt/stdarg.h>
 #include <iprt/err.h> /* for VINF_SUCCESS */
 #if defined(RT_OS_LINUX) && defined(__KERNEL__)
-  RT_C_DECLS_BEGIN
-# define new newhack /* string.h: strreplace */
+/* no C++ hacks ('new' etc) here anymore! */ 
 # include <linux/string.h>
-# undef new
-  RT_C_DECLS_END
 
 #elif defined(IN_XF86_MODULE) && !defined(NO_ANSIC)
   RT_C_DECLS_BEGIN

