--- Makefile.orig	2025-12-26 23:45:04.057619628 -0800
+++ Makefile	2025-12-26 23:48:39.891870344 -0800
@@ -1456,6 +1456,7 @@ mimalloc-debug:
 			-DMI_BUILD_OBJECT=ON \
 			-DMI_USE_CXX=ON \
 			-DMI_OVERRIDE=OFF \
+			-DMI_SECURE=ON \
 			-DCMAKE_C_FLAGS="$(CFLAGS)" \
 			-DCMAKE_CXX_FLAGS="$(CFLAGS)" \
 			-GNinja \
@@ -1481,6 +1482,7 @@ mimalloc:
 			-DMI_USE_CXX=ON \
 			-DMI_OVERRIDE=OFF \
 			-DMI_OSX_ZONE=OFF \
+			-DMI_SECURE=ON \
 			-DCMAKE_C_FLAGS="$(CFLAGS)" \
 			-GNinja \
 			 . \
