--- a/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs.orig	2019-02-13 17:56:17.000000000 -0800
+++ b/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs	2019-04-17 18:21:39.312650153 -0700
@@ -1229,7 +1229,7 @@ namespace System.Net.Http
                     asyncRead = easy._requestContentStream.ReadAsync(
                        new Memory<byte>(sts.Buffer, 0, Math.Min(sts.Buffer.Length, length)), easy._cancellationToken);
                 }
-                Debug.Assert(asyncRead != null, "Badly implemented stream returned a null task from ReadAsync");
+                //Debug.Assert(asyncRead != null, "Badly implemented stream returned a null task from ReadAsync");
 
                 // Even though it's "Async", it's possible this read could complete synchronously or extremely quickly.  
                 // Check to see if it did, in which case we can also satisfy the libcurl request synchronously in this callback.
