--- a/src/Native/Unix/CMakeLists.txt.orig	2020-04-13 14:23:22.000000000 -0700
+++ b/src/Native/Unix/CMakeLists.txt	2020-06-08 18:20:42.700743093 -0700
@@ -25,7 +25,7 @@ add_compile_options(-fPIC)
 add_compile_options(-I${CMAKE_CURRENT_SOURCE_DIR}/Common)
 add_compile_options(-I${CMAKE_CURRENT_BINARY_DIR}/Common)
 add_compile_options(-g)
-add_compile_options(-Werror)
+#add_compile_options(-Werror)
 if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)
     add_compile_options(-Wno-unreachable-code)
 endif ()
