--- a/eng/common/tools.sh.orig	2020-06-20 18:56:27.476914088 -0700
+++ b/eng/common/tools.sh	2020-06-20 19:05:16.331031879 -0700
@@ -334,7 +334,7 @@ function InitializeToolset {
   fi
   
   echo '<Project Sdk="Microsoft.DotNet.Arcade.Sdk"/>' > "$proj"
-  MSBuild-Core "$proj" $bl /t:__WriteToolsetLocation /clp:ErrorsOnly\;NoSummary /p:__ToolsetLocationOutputFile="$toolset_location_file"
+  MSBuild-Core "$proj" /p:RestoreDisableParallel=true $bl /t:__WriteToolsetLocation /clp:ErrorsOnly\;NoSummary /p:__ToolsetLocationOutputFile="$toolset_location_file"
 
   local toolset_build_proj=`cat "$toolset_location_file"`
 
--- ./eng/common/build.sh.orig	2020-05-20 09:08:02.000000000 -0700
+++ ./eng/common/build.sh	2020-06-20 19:07:16.402890099 -0700
@@ -169,6 +169,8 @@ function Build {
   InitializeToolset
   InitializeCustomToolset
 
+  properties="$properties /p:RestoreDisableParallel=true"
+
   if [[ ! -z "$projects" ]]; then
     properties="$properties /p:Projects=$projects"
   fi
