--- a/init-tools.sh.orig	2020-06-18 17:27:26.513337030 -0700
+++ b/init-tools.sh	2020-06-18 17:28:44.796253155 -0700
@@ -143,8 +143,8 @@ fi
 
 if [ ! -e "$__BUILD_TOOLS_PATH" ]; then
     echo "Restoring BuildTools version $__BUILD_TOOLS_PACKAGE_VERSION..."
-    echo "Running: $__DOTNET_CMD restore \"$__INIT_TOOLS_RESTORE_PROJECT\" --no-cache --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE /p:BuildToolsPackageVersion=$__BUILD_TOOLS_PACKAGE_VERSION /p:ToolsDir=$__TOOLRUNTIME_DIR" | tee -a "$__init_tools_log"
-    "$__DOTNET_CMD" restore "$__INIT_TOOLS_RESTORE_PROJECT" --no-cache --packages "$__PACKAGES_DIR" --source "$__BUILDTOOLS_SOURCE" /p:BuildToolsPackageVersion=$__BUILD_TOOLS_PACKAGE_VERSION /p:ToolsDir="$__TOOLRUNTIME_DIR" | tee -a "$__init_tools_log"
+    echo "Running: $__DOTNET_CMD restore \"$__INIT_TOOLS_RESTORE_PROJECT\" --disable-parallel --no-cache --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE /p:BuildToolsPackageVersion=$__BUILD_TOOLS_PACKAGE_VERSION /p:ToolsDir=$__TOOLRUNTIME_DIR" | tee -a "$__init_tools_log"
+    "$__DOTNET_CMD" restore "$__INIT_TOOLS_RESTORE_PROJECT" --disable-parallel --no-cache --packages "$__PACKAGES_DIR" --source "$__BUILDTOOLS_SOURCE" /p:BuildToolsPackageVersion=$__BUILD_TOOLS_PACKAGE_VERSION /p:ToolsDir="$__TOOLRUNTIME_DIR" | tee -a "$__init_tools_log"
     if [ ! -e "$__BUILD_TOOLS_PATH/init-tools.sh" ]; then
         echo "ERROR: Could not restore build tools correctly." 1>&2
         display_error_message
@@ -168,6 +168,9 @@ fi
 echo "Initializing BuildTools..."
 echo "Running: $__BUILD_TOOLS_PATH/init-tools.sh $__scriptpath $__DOTNET_CMD $__TOOLRUNTIME_DIR $__PACKAGES_DIR" | tee -a "$__init_tools_log"
 
+sed -i -e "s|\$__DOTNET_CMD restore|\$__DOTNET_CMD restore --disable-parallel|g" "$__BUILD_TOOLS_PATH/init-tools.sh"
+sed -i -e "s|\$__dotnet restore|\$__dotnet restore --disable-parallel|g" "$__BUILD_TOOLS_PATH/crossgen.sh"
+
 # Executables restored with .NET Core 2.0 do not have executable permission flags. https://github.com/NuGet/Home/issues/4424
 chmod +x "$__BUILD_TOOLS_PATH/init-tools.sh"
 "$__BUILD_TOOLS_PATH/init-tools.sh" "$__scriptpath" "$__DOTNET_CMD" "$__TOOLRUNTIME_DIR" "$__PACKAGES_DIR" | tee -a "$__init_tools_log"
