--- a/src/Native/build-native.sh.orig	2020-06-08 17:43:29.685450798 -0700
+++ b/src/Native/build-native.sh	2020-06-08 17:58:42.030206266 -0700
@@ -93,6 +93,7 @@ check_native_prereqs()
     fi
 
     # Check for clang
+    hash -p /usr/lib/llvm/$__ClangMajorVersion/bin/clang clang-$__ClangMajorVersion.$__ClangMinorVersion 2>/dev/null || \
     hash clang-$__ClangMajorVersion.$__ClangMinorVersion 2>/dev/null ||  hash clang$__ClangMajorVersion$__ClangMinorVersion 2>/dev/null ||  hash clang 2>/dev/null || { echo >&2 "Please install clang before running this script"; exit 1; }
 }
 
