--- a/src/SharedFrameworkValidation/SharedFrameworkValidation.proj.orig	2020-05-27 22:14:18.000000000 -0700
+++ b/src/SharedFrameworkValidation/SharedFrameworkValidation.proj	2020-06-18 17:52:03.348588059 -0700
@@ -56,7 +56,7 @@
 
   <Target Name="BuildPackagesProject">
     <PropertyGroup>
-      <_RestorePackagesCommand>"$(DotnetToolCommand)" restore $(MSBuildThisFileDirectory)/RestoreSDKProject/RestoreSDKProject.csproj /p:_PackageRID=$(PackageRID) /p:_RuntimeFrameworkVersion=$(MicrosoftNETCoreAppPackageVersion) /p:CoreFxExpectedPrerelease=$(CoreFxExpectedPrerelease)</_RestorePackagesCommand>
+      <_RestorePackagesCommand>"$(DotnetToolCommand)" restore --disable-parallel $(MSBuildThisFileDirectory)/RestoreSDKProject/RestoreSDKProject.csproj /p:_PackageRID=$(PackageRID) /p:_RuntimeFrameworkVersion=$(MicrosoftNETCoreAppPackageVersion) /p:CoreFxExpectedPrerelease=$(CoreFxExpectedPrerelease)</_RestorePackagesCommand>
       <_PublishPackagesProjectCommand>"$(DotnetToolCommand)" publish $(MSBuildThisFileDirectory)/RestoreSDKProject/RestoreSDKProject.csproj /p:_PackageRID=$(PackageRID) /p:_MetaPackageDestinationFolder=$(SharedFrameworkValidationRefPath) -f netcoreapp2.0 -o $(SharedFrameworkValidationRuntimePath) -r $(PackageRID)</_PublishPackagesProjectCommand>
     </PropertyGroup>
     <Exec Command="$(_RestorePackagesCommand)" StandardOutputImportance="Low" />
