--- a/src/System.Transactions.Local/src/System/Transactions/TransactionsEtwProvider.cs.orig	2019-02-13 17:56:17.000000000 -0800
+++ b/src/System.Transactions.Local/src/System/Transactions/TransactionsEtwProvider.cs	2019-04-18 10:44:48.373673825 -0700
@@ -203,7 +203,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                if (transaction != null && transaction.TransactionTraceId != null)
+                if (transaction != null && transaction.TransactionTraceId != TransactionTraceIdentifier.Empty)
                     TransactionCreated(transaction.TransactionTraceId.TransactionIdentifier, type);
                 else
                     TransactionCreated(string.Empty, type);
@@ -229,7 +229,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                if (transaction != null && transaction.TransactionTraceId != null)
+                if (transaction != null && transaction.TransactionTraceId != TransactionTraceIdentifier.Empty)
                     TransactionCloneCreate(transaction.TransactionTraceId.TransactionIdentifier, type);
                 else
                     TransactionCloneCreate(string.Empty, type);
@@ -255,7 +255,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                if (transaction != null && transaction.TransactionTraceId != null)
+                if (transaction != null && transaction.TransactionTraceId != TransactionTraceIdentifier.Empty)
                     TransactionSerialized(transaction.TransactionTraceId.TransactionIdentifier, type);
                 else
                     TransactionSerialized(string.Empty, type);
@@ -352,7 +352,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (transaction != null && transaction.TransactionTraceId != null)
+                if (transaction != null && transaction.TransactionTraceId != TransactionTraceIdentifier.Empty)
                     TransactionRollback(transaction.TransactionTraceId.TransactionIdentifier, type);
                 else
                     TransactionRollback(string.Empty, type);
@@ -378,7 +378,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                if (transaction != null && transaction.TransactionTraceId != null)
+                if (transaction != null && transaction.TransactionTraceId != TransactionTraceIdentifier.Empty)
                     TransactionDependentCloneComplete(transaction.TransactionTraceId.TransactionIdentifier, type);
                 else
                     TransactionDependentCloneComplete(string.Empty, type);
@@ -404,7 +404,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Verbose, ALL_KEYWORDS))
             {
-                if (transaction != null && transaction.TransactionTraceId != null)
+                if (transaction != null && transaction.TransactionTraceId != TransactionTraceIdentifier.Empty)
                     TransactionCommit(transaction.TransactionTraceId.TransactionIdentifier, type);
                 else
                     TransactionCommit(string.Empty, type);
@@ -430,7 +430,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Verbose, ALL_KEYWORDS))
             {
-                if (enlistment != null && enlistment.EnlistmentTraceId != null)
+                if (enlistment != null && enlistment.EnlistmentTraceId != EnlistmentTraceIdentifier.Empty)
                     EnlistmentStatus(enlistment.EnlistmentTraceId.EnlistmentIdentifier, notificationCall.ToString());
                 else
                     EnlistmentStatus(0, notificationCall.ToString());
@@ -455,7 +455,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Verbose, ALL_KEYWORDS))
             {
-                if (enlistment != null && enlistment.EnlistmentTraceId != null)
+                if (enlistment != null && enlistment.EnlistmentTraceId != EnlistmentTraceIdentifier.Empty)
                     EnlistmentDone(enlistment.EnlistmentTraceId.EnlistmentIdentifier);
                 else
                     EnlistmentDone(0);
@@ -480,7 +480,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Verbose, ALL_KEYWORDS))
             {
-                if (enlistment != null && enlistment.EnlistmentTraceId != null)
+                if (enlistment != null && enlistment.EnlistmentTraceId != EnlistmentTraceIdentifier.Empty)
                     EnlistmentPrepared(enlistment.EnlistmentTraceId.EnlistmentIdentifier);
                 else
                     EnlistmentPrepared(0);
@@ -505,7 +505,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (enlistment != null && enlistment.EnlistmentTraceId != null)
+                if (enlistment != null && enlistment.EnlistmentTraceId != EnlistmentTraceIdentifier.Empty)
                     EnlistmentForceRollback(enlistment.EnlistmentTraceId.EnlistmentIdentifier);
                 else
                     EnlistmentForceRollback(0);
@@ -530,7 +530,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (enlistment != null && enlistment.EnlistmentTraceId != null)
+                if (enlistment != null && enlistment.EnlistmentTraceId != EnlistmentTraceIdentifier.Empty)
                     EnlistmentAborted(enlistment.EnlistmentTraceId.EnlistmentIdentifier);
                 else
                     EnlistmentAborted(0);
@@ -556,7 +556,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Verbose, ALL_KEYWORDS))
             {
-                if (enlistment != null && enlistment.EnlistmentTraceId != null)
+                if (enlistment != null && enlistment.EnlistmentTraceId != EnlistmentTraceIdentifier.Empty)
                     EnlistmentCommitted(enlistment.EnlistmentTraceId.EnlistmentIdentifier);
                 else
                     EnlistmentCommitted(0);
@@ -581,7 +581,7 @@ namespace System.Transactions
 
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (enlistment != null && enlistment.EnlistmentTraceId != null)
+                if (enlistment != null && enlistment.EnlistmentTraceId != EnlistmentTraceIdentifier.Empty)
                     EnlistmentInDoubt(enlistment.EnlistmentTraceId.EnlistmentIdentifier);
                 else
                     EnlistmentInDoubt(0);
@@ -845,7 +845,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionScopeCreated(transactionID.TransactionIdentifier.ToString(), transactionScopeResult);
                 else
                     TransactionScopeCreated(String.Empty, transactionScopeResult);
@@ -871,11 +871,11 @@ namespace System.Transactions
             {
                 String currentId = String.Empty;
                 String newId = String.Empty;
-                if (currenttransactionID != null && currenttransactionID.TransactionIdentifier != null)
+                if (currenttransactionID != null && currenttransactionID.TransactionIdentifier != string.Empty)
                 {
                     currentId = currenttransactionID.TransactionIdentifier.ToString();
                 }
-                if (newtransactionID!= null && newtransactionID.TransactionIdentifier != null)
+                if (newtransactionID!= null && newtransactionID.TransactionIdentifier != string.Empty)
                 {
                     newId = newtransactionID.TransactionIdentifier.ToString();
                 }
@@ -899,7 +899,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionScopeNestedIncorrectly(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionScopeNestedIncorrectly(String.Empty);
@@ -922,7 +922,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionScopeDisposed(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionScopeDisposed(String.Empty);
@@ -945,7 +945,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionScopeIncomplete(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionScopeIncomplete(String.Empty);
@@ -988,7 +988,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionScopeTimeout(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionScopeTimeout(String.Empty);
@@ -1011,7 +1011,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionTimeout(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionTimeout(String.Empty);
@@ -1036,7 +1036,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
             {
-                if (enlistmentID != null)
+                if (enlistmentID != EnlistmentTraceIdentifier.Empty)
                     TransactionstateEnlist(enlistmentID.EnlistmentIdentifier.ToString(), enlistmentType.ToString(), enlistmentOption.ToString());
                 else
                     TransactionstateEnlist(String.Empty, enlistmentType.ToString(), enlistmentOption.ToString());
@@ -1059,7 +1059,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Verbose, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionCommitted(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionCommitted(String.Empty);
@@ -1082,7 +1082,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionInDoubt(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionInDoubt(String.Empty);
@@ -1108,9 +1108,9 @@ namespace System.Transactions
             {
                 String id = String.Empty;
                 String distributedId = String.Empty;
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     id= transactionID.TransactionIdentifier.ToString();
-                if (distributedTxID != null && distributedTxID.TransactionIdentifier != null)
+                if (distributedTxID != null && distributedTxID.TransactionIdentifier != string.Empty)
                     distributedId = distributedTxID.TransactionIdentifier.ToString();
                 TransactionPromoted(id, distributedId);
             }
@@ -1132,7 +1132,7 @@ namespace System.Transactions
         {
             if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
             {
-                if (transactionID != null && transactionID.TransactionIdentifier != null)
+                if (transactionID != null && transactionID.TransactionIdentifier != string.Empty)
                     TransactionAborted(transactionID.TransactionIdentifier.ToString());
                 else
                     TransactionAborted(String.Empty);
