--- a/coreclr-2.1.9/src/vm/i386/jithelp.S.orig	2019-02-13 13:18:25.000000000 -0800
+++ b/coreclr-2.1.9/src/vm/i386/jithelp.S	2019-04-01 22:56:09.822656823 -0700
@@ -184,12 +184,12 @@ LOCAL_LABEL(WriteBarrier_ShadowCheckEnd_
     PREPARE_EXTERNAL_VAR g_card_table, eax
     add     edx, [eax]
     pop     eax
-    cmp     BYTE PTR [edx], 0FFh
+    cmp     BYTE PTR [edx], 0xFF
     jne     LOCAL_LABEL(WriteBarrier_UpdateCardTable_\rg)
     ret
 
 LOCAL_LABEL(WriteBarrier_UpdateCardTable_\rg):
-    mov     BYTE PTR [edx], 0FFh
+    mov     BYTE PTR [edx], 0xFF
     ret
 
 LOCAL_LABEL(WriteBarrier_NotInHeap_\rg):
@@ -351,11 +351,11 @@ LOCAL_LABEL(ByRefWriteBarrier_ShadowChec
     PREPARE_EXTERNAL_VAR g_card_table, eax
     add     ecx, [eax]
     pop     eax
-    cmp     BYTE PTR [ecx], 0FFh
+    cmp     BYTE PTR [ecx], 0xFF
     jne     LOCAL_LABEL(ByRefWriteBarrier_UpdateCardTable)
     ret
 LOCAL_LABEL(ByRefWriteBarrier_UpdateCardTable):
-    mov     BYTE PTR [ecx], 0FFh
+    mov     BYTE PTR [ecx], 0xFF
     ret
 
 LOCAL_LABEL(ByRefWriteBarrier_NotInHeap):
@@ -542,7 +542,7 @@ LEAF_ENTRY JIT_Dbl2LngP4x87, _TEXT
     // get some local space
     sub 	esp, 8
 
-    #define arg1 [esp + 0Ch]
+    #define arg1 [esp + 0x0C]
     fld     QWORD PTR arg1          // fetch arg
     fnstcw  WORD PTR arg1           // store FPCW
     movzx   eax, WORD PTR arg1      // zero extend - wide
@@ -580,7 +580,7 @@ LEAF_ENTRY JIT_Dbl2LngSSE3, _TEXT
     // get some local space
     sub     esp, 8
 
-    fld     QWORD PTR [esp + 0Ch]   // fetch arg
+    fld     QWORD PTR [esp + 0x0C]   // fetch arg
     fisttp  QWORD PTR [esp]         // convert
     mov     eax, DWORD PTR [esp]    // reload FP result
     mov     edx, DWORD PTR [esp + 4]
@@ -629,7 +629,7 @@ LEAF_ENTRY JIT_WriteBarrierReg_PreGrow,
 
     shr     edx, 10
     nop     // padding for alignment of constant
-    cmp     BYTE PTR [edx + 0F0F0F0F0h], 0FFh
+    cmp     BYTE PTR [edx + 0x0F0F0F0F0], 0xFF
     jne     LOCAL_LABEL(WriteBarrierPre)
 
 LOCAL_LABEL(NoWriteBarrierPre):
@@ -638,7 +638,7 @@ LOCAL_LABEL(NoWriteBarrierPre):
     nop     // padding for alignment of constant
 
 LOCAL_LABEL(WriteBarrierPre):
-    mov     BYTE PTR [edx+0F0F0F0F0h], 0FFh
+    mov     BYTE PTR [edx+0x0F0F0F0F0], 0xFF
     ret
 LEAF_END JIT_WriteBarrierReg_PreGrow, _TEXT
 
@@ -661,7 +661,7 @@ LEAF_ENTRY JIT_WriteBarrierReg_PostGrow,
 
     shr     edx, 10
     nop     // padding for alignment of constant
-    cmp     BYTE PTR [edx + 0F0F0F0F0h], 0FFh
+    cmp     BYTE PTR [edx + 0x0F0F0F0F0], 0xFF
     jne     LOCAL_LABEL(WriteBarrierPost)
 
 LOCAL_LABEL(NoWriteBarrierPost):
@@ -670,7 +670,7 @@ LOCAL_LABEL(NoWriteBarrierPost):
     nop     // padding for alignment of constant
 
 LOCAL_LABEL(WriteBarrierPost):
-    mov     BYTE PTR [edx + 0F0F0F0F0h], 0FFh
+    mov     BYTE PTR [edx + 0x0F0F0F0F0], 0xFF
     ret
 LEAF_END JIT_WriteBarrierReg_PostGrow,_TEXT
 
