--- a/coreclr-2.1.9/src/vm/amd64/jithelpers_slow.S.orig	2019-02-13 13:18:25.000000000 -0800
+++ b/coreclr-2.1.9/src/vm/amd64/jithelpers_slow.S	2019-04-01 19:42:10.055352622 -0700
@@ -71,15 +71,15 @@ LEAF_ENTRY JIT_WriteBarrier_Debug, _TEXT
 #ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
         // Update the write watch table if necessary
         PREPARE_EXTERNAL_VAR g_sw_ww_enabled_for_gc_heap, r10
-        cmp     byte ptr [r10], 0h
+        cmp     byte ptr [r10], 0x00
         je      CheckCardTable_Debug
         mov     r10, rdi
-        shr     r10, 0Ch // SoftwareWriteWatch::AddressToTableByteIndexShift
+        shr     r10, 0x0C // SoftwareWriteWatch::AddressToTableByteIndexShift
         PREPARE_EXTERNAL_VAR g_sw_ww_table, r11
         add     r10, qword ptr [r11]
-        cmp     byte ptr [r10], 0h
+        cmp     byte ptr [r10], 0x00
         jne     CheckCardTable_Debug
-        mov     byte ptr [r10], 0FFh
+        mov     byte ptr [r10], 0xFF
 #endif
 
     CheckCardTable_Debug:
@@ -99,13 +99,13 @@ LEAF_ENTRY JIT_WriteBarrier_Debug, _TEXT
         mov     r10, [r10]
 
         // Check if this card is dirty
-        cmp     byte ptr [rdi + r10], 0FFh
+        cmp     byte ptr [rdi + r10], 0xFF
 
         jne     UpdateCardTable_Debug
         REPRET
 
     UpdateCardTable_Debug:
-        mov     byte ptr [rdi + r10], 0FFh
+        mov     byte ptr [rdi + r10], 0xFF
 
 #ifdef FEATURE_MANUALLY_MANAGED_CARD_BUNDLES
         // Shift rdi by 0x0A more to get the card bundle byte (we shifted by 0x0B already)
@@ -115,13 +115,13 @@ LEAF_ENTRY JIT_WriteBarrier_Debug, _TEXT
         add     rdi, [r10]
 
         // Check if this bundle byte is dirty
-        cmp     byte ptr [rdi], 0FFh
+        cmp     byte ptr [rdi], 0xFF
 
         jne     UpdateCardBundle_Debug
         REPRET
 
     UpdateCardBundle_Debug:
-        mov     byte ptr [rdi], 0FFh
+        mov     byte ptr [rdi], 0xFF
 #endif
 
         ret
