diff -urp amdgpu-pro-16.40-348864.orig/amd/backport/Makefile amdgpu-pro-16.40-348864/amd/backport/Makefile
--- amdgpu-pro-16.40-348864.orig/amd/backport/Makefile	2016-10-24 18:01:17.000000000 -0700
+++ amdgpu-pro-16.40-348864/amd/backport/Makefile	2017-02-18 15:03:58.924578725 -0800
@@ -13,6 +13,9 @@ endif
 
 ifeq ("rhel",$(OS_NAME))
 ccflags-y += -DOS_NAME_RHEL
+else ifeq ("gentoo",$(OS_NAME))
+ccflags-y += -DOS_NAME_GENTOO
+OS_VERSION = "0.0"
 else ifeq ("ubuntu",$(OS_NAME))
 ccflags-y += -DOS_NAME_UBUNTU
 else ifeq ("steamos",$(OS_NAME))
Only in amdgpu-pro-16.40-348864/amd/dkms: Makefile.orig
Only in amdgpu-pro-16.40-348864/amd/dkms: Makefile.rej
Only in amdgpu-pro-16.40-348864: Makefile.orig
Only in amdgpu-pro-16.40-348864: Makefile.rej
