--- a/emscripten-fastcomp-clang-1.40.1/lib/Basic/Version.cpp.orig	2020-02-14 09:08:24.000000000 -0800
+++ b/emscripten-fastcomp-clang-1.40.1/lib/Basic/Version.cpp	2020-02-25 21:46:27.576550220 -0800
@@ -142,9 +142,9 @@ std::string getClangToolFullVersion(Stri
       " : "
   #include "../../emscripten-version.txt"
 #else
-#include "../../../../emscripten-version.txt"
+#include "../../emscripten-version.txt"
          " : "
-#include "../../../../tools/clang/emscripten-version.txt"
+
 #endif
          ")";
 
