--- AP/client/ap_gfx_base.h.orig	2015-12-02 11:39:31.257402245 -0800
+++ AP/client/ap_gfx_base.h	2015-12-02 11:40:43.655142754 -0800
@@ -78,8 +78,8 @@
     double viewpoint_distance;
     int width, height;
     REDUCED_ARRAY_RENDER rarray;
-    const AP_SHMEM* ap_shmem;
-    const AP_GDATA* ap_gdata;
+    AP_SHMEM* ap_shmem;
+    AP_GDATA* ap_gdata;
 
     AP_GRAPHICS_BASE();
     void data_struct_init();
