diff -ur tolua++-1.0.93.orig/src/bin/lua/compat.lua tolua++-1.0.93/src/bin/lua/compat.lua
--- tolua++-1.0.93.orig/src/bin/lua/compat.lua	2016-02-04 23:27:10.379242829 -0800
+++ tolua++-1.0.93/src/bin/lua/compat.lua	2016-02-04 23:30:20.308536727 -0800
@@ -104,6 +104,7 @@
 strrep = str.rep
 strsub = str.sub
 strupper = str.upper
+gfind = str.gmatch
 
 -------------------------------------------------------------------
 -- os library
