diff -urp tensorstore-0.1.59.orig/setup.py tensorstore-0.1.59/setup.py
--- tensorstore-0.1.59.orig/setup.py	2024-05-07 08:58:35.000000000 -0700
+++ tensorstore-0.1.59/setup.py	2024-05-22 21:28:51.380434595 -0700
@@ -245,9 +245,7 @@ class BuildExtCommand(setuptools.command
         )
 
         build_command = (
-            [sys.executable, '-u', bazelisk]
-            + startup_options
-            + [
+            [   'bazel', # oteodoro: removed bazelisk and call bazel directly
                 'build',
                 '-c',
                 compilation_mode,
Only in tensorstore-0.1.59: setup.py.orig
Only in tensorstore-0.1.59: setup.py.rej
