--- monodevelop-7.6.11_p9999/configure.orig	2022-09-17 18:17:20.837626548 -0700
+++ monodevelop-7.6.11_p9999/configure	2022-09-17 18:48:13.237387395 -0700
@@ -204,7 +204,7 @@ configure_packages ()
 			exit 1
 		fi
 		pushd $path > /dev/null
-		PKG_CONFIG_PATH=$localconf:$PKG_CONFIG_PATH $sct $prefixarg $ops || handle_config_error
+		PKG_CONFIG_PATH=$localconf:$PKG_CONFIG_PATH $sct $prefixarg $ops ${extra_args[@]} || handle_config_error
 		popd > /dev/null
 		create_local_config $path $builddir
 		packages="$packages $path"
@@ -239,6 +239,7 @@ create_local_config ()
 
 echo
 
+extra_args=()
 while test x$1 != x; do
 	case $1 in
 		--enable-tests)
@@ -273,9 +274,7 @@ while test x$1 != x; do
 			exit
 			;;
 		*)
-			echo Unknown argument $1 >&2
-			usage
-			exit 1
+			extra_args+=( $1 )
 			;;
 	esac
 	shift
