--- monodevelop-7.6.11_p9999/main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj.orig	2022-09-16 22:31:03.802849363 -0700
+++ monodevelop-7.6.11_p9999/main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj	2022-09-16 22:31:08.299024868 -0700
@@ -19,7 +19,7 @@
     <ConfigureScript>$(MSBuildProjectDirectory)\..\..\..\..\scripts\configure.cs</ConfigureScript>
     <ConfigureScriptExe>$(MSBuildProjectDirectory)\..\..\..\..\scripts\configure.exe</ConfigureScriptExe>
     <FullBuildInfo>fullbuildinfo</FullBuildInfo>
-    <MonoLauncher Condition=" '$(VisualStudioVersion)' == '' ">mono </MonoLauncher>
+    <MonoLauncher Condition=" '$(VisualStudioVersion)' == '' Or $(UseMonoLauncher) == '1' ">mono </MonoLauncher>
     <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
   </PropertyGroup>
   <Choose>
