--- monodevelop-7.6.11_p9999/main/external/mono-addins/Samples/TextEditor/TextEditor/TextEditor.csproj.orig	2022-09-17 20:44:14.062162800 -0700
+++ monodevelop-7.6.11_p9999/main/external/mono-addins/Samples/TextEditor/TextEditor/TextEditor.csproj	2022-09-17 21:00:28.167671543 -0700
@@ -9,7 +9,7 @@
     <OutputType>Exe</OutputType>
     <RootNamespace>TextEditor</RootNamespace>
     <AssemblyName>TextEditor</AssemblyName>
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|x86' ">
     <DebugSymbols>True</DebugSymbols>
