--- monodevelop-7.6.11_p9999/main/msbuild/MonoDevelop.BeforeCommon.targets.orig	2022-09-17 19:18:03.994034927 -0700
+++ monodevelop-7.6.11_p9999/main/msbuild/MonoDevelop.BeforeCommon.targets	2022-09-17 19:50:01.898901477 -0700
@@ -6,9 +6,8 @@
 
 	<!-- force Roslyn compiler -->
 	<PropertyGroup Condition="'$(OS)'=='Unix' And '$(MSBuildProjectExtension)'=='.csproj'">
-		<_MonoBinDir>$(MSBuildBinPath)\..\..\..\4.5\</_MonoBinDir>
-		<CscToolExe>csc.exe</CscToolExe>
-		<CscToolPath>$(_MonoBinDir)</CscToolPath>
+		<CscToolExe>__CSC_TOOL_EXE__</CscToolExe>
+		<CscToolPath>__CSC_TOOL_PATH__</CscToolPath>
 		<DebugType>portable</DebugType>
 		<CscDebugFileExt>.pdb</CscDebugFileExt>
 		<_DebugFileExt>.pdb</_DebugFileExt>
