--- a/tensorflow/tools/pip_package/setup.py.orig	2024-11-25 09:49:47.870671343 -0800
+++ b/tensorflow/tools/pip_package/setup.py	2024-11-25 09:52:27.949596183 -0800
@@ -78,21 +78,21 @@ REQUIRED_PACKAGES = [
     # NOTE: As numpy has releases that break semver guarantees and several other
     # deps depend on numpy without an upper bound, we must install numpy before
     # everything else.
-    'numpy ~= 1.19.2',
+    'numpy >= 1.19.2',
     # Install other dependencies
-    'absl-py ~= 0.10',
-    'astunparse ~= 1.6.3',
-    'flatbuffers ~= 1.12.0',
-    'google_pasta ~= 0.2',
-    'h5py ~= 3.1.0',
-    'keras_preprocessing ~= 1.1.2',
-    'opt_einsum ~= 3.3.0',
+    'absl-py >= 0.10',
+    'astunparse >= 1.6.3',
+    'flatbuffers >= 1.12.0',
+    'google_pasta >= 0.2',
+    'h5py >= 3.1.0',
+    'keras_preprocessing >= 1.1.2',
+    'opt_einsum >= 3.3.0',
     'protobuf >= 3.9.2',
-    'six ~= 1.15.0',
-    'termcolor ~= 1.1.0',
-    'typing_extensions ~= 3.7.4',
-    'wheel ~= 0.35',
-    'wrapt ~= 1.12.1',
+    'six >= 1.15.0',
+    'termcolor >= 1.1.0',
+    'typing_extensions >= 3.7.4',
+    'wheel >= 0.35',
+    'wrapt >= 1.12.1',
     # These packages need to be pinned exactly as newer versions are
     # incompatible with the rest of the ecosystem
     'gast >= 0.4.0',
@@ -100,11 +100,11 @@ REQUIRED_PACKAGES = [
     # These need to be in sync with the existing TF version
     # They are updated during the release process
     # When updating these, please also update the nightly versions below
-    'tensorboard ~= 2.5',
+    'tensorboard >= 2.5',
     'tensorflow-estimator >= 2.5.0 , < 2.6.0',
     # TODO(scottzhu): OSS keras hasn't been formally released yet.
     # Use keras-nightly at the moment.
-    'keras-nightly ~= 2.5.0.dev',
+    'keras-nightly ~= 2.5.0.dev',
 ]
 
 
@@ -116,28 +116,28 @@ REQUIRED_PACKAGES = [
 if 'tf_nightly' in project_name:
   for i, pkg in enumerate(REQUIRED_PACKAGES):
     if 'tensorboard' in pkg:
-      REQUIRED_PACKAGES[i] = 'tb-nightly ~= 2.5.0.a'
+      REQUIRED_PACKAGES[i] = 'tb-nightly ~= 2.5.0.a'
     elif 'tensorflow_estimator' in pkg:
-      REQUIRED_PACKAGES[i] = 'tf-estimator-nightly ~= 2.5.0.dev'
+      REQUIRED_PACKAGES[i] = 'tf-estimator-nightly ~= 2.5.0.dev'
     elif 'keras' in pkg and 'keras_preprocessing' not in pkg:
-      REQUIRED_PACKAGES[i] = 'keras-nightly ~= 2.5.0.dev'
+      REQUIRED_PACKAGES[i] = 'keras-nightly ~= 2.5.0.dev'
 
 
 # grpcio does not build correctly on big-endian machines due to lack of
 # BoringSSL support.
 # See https://github.com/tensorflow/tensorflow/issues/17882.
 if sys.byteorder == 'little':
-  REQUIRED_PACKAGES.append('grpcio ~= 1.34.0')
+  REQUIRED_PACKAGES.append('grpcio >= 1.34.0')
 
 
 # Packages which are only needed for testing code.
 # Please don't add test-only packages to `REQUIRED_PACKAGES`!
 # Follows the same conventions as `REQUIRED_PACKAGES`
 TEST_PACKAGES = [
-    'portpicker ~= 1.3.1',
-    'scipy ~= 1.5.2',
-    'tblib ~= 1.7.0',
-    'dill ~= 0.3.2',
+    'portpicker >= 1.3.1',
+    'scipy >= 1.5.2',
+    'tblib >= 1.7.0',
+    'dill >= 0.3.2',
 ]
 
 
