--- a/tensorflow/BUILD.orig	2022-01-30 07:39:47.000000000 -0800
+++ b/tensorflow/BUILD	2024-11-25 09:39:31.390799255 -0800
@@ -36,6 +36,7 @@ load(
     "if_mkl_ml",
 )
 load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
+load("@local_config_syslibs//:build_defs.bzl", "if_system_lib")
 
 package(
     default_visibility = [":internal"],
@@ -1021,7 +1022,7 @@ tf_cc_shared_object(
             "/DEBUG:FASTLINK",
         ],
         "//conditions:default": [],
-    }),
+    }) + if_system_lib("nsync", ["-lnsync_cpp"]),
     per_os_targets = True,
     soversion = VERSION,
     visibility = ["//visibility:public"],
