--- a/lib/Driver/ToolChains/AMDGPU.cpp.orig	2023-01-11 23:12:30.000000000 -0800
+++ b/lib/Driver/ToolChains/AMDGPU.cpp	2023-08-27 22:55:03.152948731 -0700
@@ -411,11 +411,13 @@ void RocmInstallationDetector::detectDev
     };
 
     // The possible structures are:
+    // - ${ROCM_ROOT}/@LIBDIR@/amdgcn/bitcode/*
     // - ${ROCM_ROOT}/amdgcn/bitcode/*
     // - ${ROCM_ROOT}/lib/*
     // - ${ROCM_ROOT}/lib/bitcode/*
     // so try to detect these layouts.
     static constexpr std::array<const char *, 2> SubDirsList[] = {
+        {"@LIBDIR@/amdgcn", "bitcode"},
         {"amdgcn", "bitcode"},
         {"lib", ""},
         {"lib", "bitcode"},
