diff -urp cfe-5.0.0.src.orig/lib/Driver/ToolChains/Linux.h cfe-5.0.0.src/lib/Driver/ToolChains/Linux.h
--- cfe-5.0.0.src.orig/lib/Driver/ToolChains/Linux.h	2017-03-07 17:02:16.000000000 -0800
+++ cfe-5.0.0.src/lib/Driver/ToolChains/Linux.h	2017-12-04 20:22:16.343625206 -0800
@@ -20,7 +20,7 @@ namespace toolchains {
 class LLVM_LIBRARY_VISIBILITY Linux : public Generic_ELF {
 public:
   Linux(const Driver &D, const llvm::Triple &Triple,
-        const llvm::opt::ArgList &Args);
+        const llvm::opt::ArgList &Args) __attribute__ ((visibility ("default") ));
 
   bool HasNativeLLVMSupport() const override;
 
