diff -urp OpenShadingLanguage-1.12.9.0.orig/src/cmake/externalpackages.cmake OpenShadingLanguage-1.12.9.0/src/cmake/externalpackages.cmake
--- OpenShadingLanguage-1.12.9.0.orig/src/cmake/externalpackages.cmake	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/cmake/externalpackages.cmake	2023-02-08 19:27:41.642365510 -0800
@@ -117,7 +117,7 @@ checked_find_package (LLVM REQUIRED
                       VERSION_MIN 9.0
                       PRINT LLVM_SYSTEM_LIBRARIES CLANG_LIBRARIES)
 # ensure include directory is added (in case of non-standard locations
-include_directories (BEFORE SYSTEM "${LLVM_INCLUDES}")
+include_directories (BEFORE SYSTEM "${LLVM_INCLUDES};$ENV{CXX_INCLUDES_PATH}")
 link_directories ("${LLVM_LIB_DIR}")
 # Extract and concatenate major & minor, remove wayward patches,
 # dots, and "svn" or other suffixes.
Only in OpenShadingLanguage-1.12.9.0/src/cmake: externalpackages.cmake.orig
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/cmake/llvm_macros.cmake OpenShadingLanguage-1.12.9.0/src/cmake/llvm_macros.cmake
--- OpenShadingLanguage-1.12.9.0.orig/src/cmake/llvm_macros.cmake	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/cmake/llvm_macros.cmake	2023-02-08 19:36:12.646382349 -0800
@@ -93,6 +93,7 @@ function ( EMBED_LLVM_BITCODE_IN_CPP src
         add_custom_command ( OUTPUT ${src_bc}
         COMMAND ${LLVM_BC_GENERATOR}
             ${LLVM_COMPILE_FLAGS}
+            -isystem "$ENV{CXX_INCLUDES_PATH}"
             "-I${CMAKE_CURRENT_SOURCE_DIR}"
             "-I${CMAKE_SOURCE_DIR}/src/include"
             "-I${CMAKE_BINARY_DIR}/include"
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/liboslcomp/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/liboslcomp/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/liboslcomp/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/liboslcomp/CMakeLists.txt	2023-02-08 19:27:41.642365510 -0800
@@ -21,6 +21,7 @@ target_include_directories(${local_lib}
     PUBLIC
         ${CMAKE_INSTALL_FULL_INCLUDEDIR}
         ${IMATH_INCLUDES}
+        $ENV{CXX_INCLUDES_PATH}
     )
 target_link_libraries (${local_lib}
     PUBLIC
Only in OpenShadingLanguage-1.12.9.0/src/liboslcomp: CMakeLists.txt.orig
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/liboslexec/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/liboslexec/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/liboslexec/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/liboslexec/CMakeLists.txt	2023-02-08 19:27:41.646365510 -0800
@@ -504,6 +504,7 @@ target_include_directories (${local_lib}
     PUBLIC
         ${CMAKE_INSTALL_FULL_INCLUDEDIR}
         ${IMATH_INCLUDES}
+        $ENV{CXX_INCLUDES_PATH}
     PRIVATE
         "${CMAKE_SOURCE_DIR}/src/liboslcomp"
     )
Only in OpenShadingLanguage-1.12.9.0/src/liboslexec: CMakeLists.txt.orig
Only in OpenShadingLanguage-1.12.9.0/src/liboslexec: CMakeLists.txt.rej
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/liboslnoise/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/liboslnoise/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/liboslnoise/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/liboslnoise/CMakeLists.txt	2023-02-08 19:27:41.646365510 -0800
@@ -12,6 +12,7 @@ target_include_directories (${local_lib}
     PUBLIC
         ${CMAKE_INSTALL_FULL_INCLUDEDIR}
         ${IMATH_INCLUDES}
+        $ENV{CXX_INCLUDES_PATH}
     PRIVATE
         ../liboslexec)
 target_link_libraries (${local_lib}
Only in OpenShadingLanguage-1.12.9.0/src/liboslnoise: CMakeLists.txt.orig
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/osl.imageio/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/osl.imageio/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/osl.imageio/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/osl.imageio/CMakeLists.txt	2023-02-08 19:28:42.878367528 -0800
@@ -36,7 +36,9 @@ macro (add_oiio_plugin)
 
     target_compile_definitions (${_plugin_name} PRIVATE
                                 ${_plugin_DEFINITIONS})
-    target_include_directories (${_plugin_name} PRIVATE ${_plugin_INCLUDE_DIRS})
+    target_include_directories (${_plugin_name} PUBLIC $ENV{CXX_INCLUDES_PATH}
+                                                PRIVATE ${_plugin_INCLUDE_DIRS}
+                                                ${OPENIMAGEIO_INCLUDES})
     target_link_libraries (${_plugin_name}
                            PUBLIC  OpenImageIO::OpenImageIO
                            PRIVATE ${_plugin_LINK_LIBRARIES})
Only in OpenShadingLanguage-1.12.9.0/src/osl.imageio: CMakeLists.txt.orig
Only in OpenShadingLanguage-1.12.9.0/src/osl.imageio: CMakeLists.txt.rej
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/oslc/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/oslc/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/oslc/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/oslc/CMakeLists.txt	2023-02-08 19:27:41.646365510 -0800
@@ -12,5 +12,6 @@ if (NOT BUILD_SHARED_LIBS)
 endif ()
 
 add_executable ( oslc ${oslc_srcs} )
+target_include_directories ( oslc PUBLIC $ENV{CXX_INCLUDES_PATH} )
 target_link_libraries ( oslc PRIVATE oslcomp ${Boost_LIBRARIES} ${CMAKE_DL_LIBS})
 install_targets (oslc)
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/oslinfo/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/oslinfo/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/oslinfo/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/oslinfo/CMakeLists.txt	2023-02-08 19:27:41.646365510 -0800
@@ -4,6 +4,7 @@
 
 set ( oslinfo_srcs oslinfo.cpp )
 add_executable ( oslinfo ${oslinfo_srcs} )
+target_include_directories (oslinfo PUBLIC $ENV{CXX_INCLUDE_PATH})
 target_link_libraries (oslinfo
                        PRIVATE oslquery
                                ${Boost_LIBRARIES} ${CMAKE_DL_LIBS})
Only in OpenShadingLanguage-1.12.9.0/src/oslinfo: CMakeLists.txt.orig
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/osltoy/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/osltoy/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/osltoy/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/osltoy/CMakeLists.txt	2023-02-08 19:27:41.646365510 -0800
@@ -10,6 +10,7 @@ set (osltoy_srcs
      osltoymain.cpp osltoyapp.cpp codeeditor.cpp osltoyrenderer.cpp)
 add_executable (osltoy ${osltoy_srcs})
 set_target_properties (osltoy PROPERTIES FOLDER "Tools")
+target_include_directories (osltoy PUBLIC $ENV{CXX_INCLUDES_PATH})
 target_link_libraries (osltoy
     PRIVATE
         oslexec
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/testrender/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/testrender/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/testrender/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/testrender/CMakeLists.txt	2023-02-08 19:27:41.646365510 -0800
@@ -58,6 +58,7 @@ endif ()
 
 add_executable (testrender ${testrender_srcs})
 
+target_include_directories (testrender PUBLIC $ENV{CXX_INCLUDES_PATH})
 target_link_libraries (testrender
     PRIVATE
         oslexec oslquery
Only in OpenShadingLanguage-1.12.9.0/src/testrender: CMakeLists.txt.orig
diff -urp OpenShadingLanguage-1.12.9.0.orig/src/testshade/CMakeLists.txt OpenShadingLanguage-1.12.9.0/src/testshade/CMakeLists.txt
--- OpenShadingLanguage-1.12.9.0.orig/src/testshade/CMakeLists.txt	2023-01-30 14:22:07.000000000 -0800
+++ OpenShadingLanguage-1.12.9.0/src/testshade/CMakeLists.txt	2023-02-08 19:27:41.646365510 -0800
@@ -66,6 +66,10 @@ EMBED_LLVM_BITCODE_IN_CPP ( "${rs_srcs}"
 
 add_executable ( testshade ${testshade_srcs} testshademain.cpp )
 
+target_include_directories (testshade
+                            PUBLIC
+                                $ENV{CXX_INCLUDES_PATH})
+
 target_link_libraries (testshade
                        PRIVATE
                            oslexec oslquery)
@@ -83,6 +87,10 @@ set_target_properties (libtestshade
                        OUTPUT_NAME libtestshade${OSL_LIBNAME_SUFFIX}
                        )
 
+target_include_directories (libtestshade
+                            PUBLIC
+                                $ENV{CXX_INCLUDES_PATH})
+
 target_link_libraries (libtestshade
                        PRIVATE
                            oslexec oslquery)
@@ -92,6 +100,9 @@ install_targets ( libtestshade )
 
 # The 'testshade_dso' executable
 add_executable ( testshade_dso testshade_dso.cpp )
+target_include_directories (testshade_dso
+                            PUBLIC
+                                $ENV{CXX_INCLUDES_PATH})
 target_link_libraries (testshade_dso
                        PRIVATE
                            OpenImageIO::OpenImageIO
Only in OpenShadingLanguage-1.12.9.0/src/testshade: CMakeLists.txt.orig
