diff -urp utf8_range.orig/CMakeLists.txt utf8_range/CMakeLists.txt
--- utf8_range.orig/CMakeLists.txt	2024-02-26 19:17:18.652402685 -0800
+++ utf8_range/CMakeLists.txt	2024-02-26 19:18:22.163444465 -0800
@@ -11,13 +11,13 @@ option (utf8_range_ENABLE_INSTALL "Confi
 
 ##
 # Create the lightweight C library
-add_library (utf8_range STATIC
+add_library (utf8_range SHARED
   utf8_range.c
 )
 
 ##
 # A heavier-weight C++ wrapper that supports Abseil.
-add_library (utf8_validity STATIC utf8_validity.cc utf8_range.c)
+add_library (utf8_validity SHARED utf8_validity.cc utf8_range.c)
 
 # Load Abseil dependency.
 if (NOT TARGET absl::strings)
