diff -urpN okcupid-pidgin-b2ff1ecd2f397f0d358fb1a825c73992bc5f4efb.orig/Makefile okcupid-pidgin-b2ff1ecd2f397f0d358fb1a825c73992bc5f4efb/Makefile
--- okcupid-pidgin-b2ff1ecd2f397f0d358fb1a825c73992bc5f4efb.orig/Makefile	2013-03-17 13:18:34.000000000 -0700
+++ okcupid-pidgin-b2ff1ecd2f397f0d358fb1a825c73992bc5f4efb/Makefile	2017-01-04 04:00:17.314226326 -0800
@@ -1,6 +1,6 @@
 #Customisable stuff here
-LINUX32_COMPILER = i686-pc-linux-gnu-gcc
-LINUX64_COMPILER = x86_64-pc-linux-gnu-gcc
+LINUX32_COMPILER = x86_64-pc-linux-gnu-gcc -m32
+LINUX64_COMPILER = x86_64-pc-linux-gnu-gcc -m64
 WIN32_COMPILER = /usr/bin/i586-mingw32-gcc
 WIN32_WINDRES = i586-mingw32-windres
 #LINUX_ARM_COMPILER = arm-pc-linux-gnu-gcc
@@ -33,7 +33,8 @@ FACEBOOK_SOURCES = \
 .PHONY:	all clean install sourcepackage
 
 #all:	libokcupid.so libokcupid.dll libokcupid64.so libokcupidarm.so libokcupidppc.so installers sourcepackage
-all:	libokcupid.so libokcupid.dll libokcupid64.so installers sourcepackage
+#all:	libokcupid.so libokcupid.dll libokcupid64.so installers sourcepackage
+all:	libokcupid.so libokcupid64.so
 
 install:
 	cp libokcupid.so /usr/lib/purple-2/
