--- a/scripts/serverLauncher/startServer.js.orig	2025-01-22 08:17:26.000000000 -0800
+++ b/scripts/serverLauncher/startServer.js	2025-01-22 20:07:04.659221418 -0800
@@ -3,8 +3,8 @@ const fs = require('node:fs').promises;
 const { spawn } = require('node:child_process');
 
 // Set file paths
-const DB_MIGRATION_SCRIPT_PATH = '/app/docker.cjs';
-const SERVER_SCRIPT_PATH = '/app/server.js';
+const DB_MIGRATION_SCRIPT_PATH = '/opt/lobe-chat/docker.cjs';
+const SERVER_SCRIPT_PATH = '/opt/lobe-chat/server.js';
 const PROXYCHAINS_CONF_PATH = '/etc/proxychains4.conf';
 
 // Function to check if a string is a valid IP address
@@ -97,8 +97,8 @@ ${protocol} ${ip} ${port} ${user} ${pass
 // Function to execute a script with child process spawn
 const runScript = (scriptPath, useProxy = false) => {
   const command = useProxy
-    ? ['/bin/proxychains', '-q', '/bin/node', scriptPath]
-    : ['/bin/node', scriptPath];
+    ? ['/usr/bin/proxychains', '-q', '/usr/lib/node/@NODE_SLOT@/bin/node', scriptPath]
+    : ['/usr/lib/node/@NODE_SLOT@/bin/node', scriptPath];
   return new Promise((resolve, reject) => {
     const process = spawn(command.shift(), command, { stdio: 'inherit' });
     process.on('close', (code) =>
