diff -urp oneTBB-2020_U3.orig/cmake/TBBInstallConfig.cmake oneTBB-2020_U3/cmake/TBBInstallConfig.cmake
--- oneTBB-2020_U3.orig/cmake/TBBInstallConfig.cmake	2020-07-10 06:18:01.000000000 -0700
+++ oneTBB-2020_U3/cmake/TBBInstallConfig.cmake	2024-05-03 20:22:28.420760035 -0700
@@ -30,7 +30,7 @@ function(tbb_install_config)
     file(MAKE_DIRECTORY ${config_install_dir})
 
     # --- TBB_LIB_REL_PATH handling ---
-    set(TBB_LIB_REL_PATH "../../../lib")
+    set(TBB_LIB_REL_PATH "../../../$ENV{LIBDIR}")
 
     if (tbb_IC_LIB_REL_PATH)
         file(TO_CMAKE_PATH ${tbb_IC_LIB_REL_PATH} TBB_LIB_REL_PATH)
@@ -58,7 +58,7 @@ function(tbb_install_config)
     # ------
 
     # --- TBB_INC_REL_PATH handling ---
-    set(TBB_INC_REL_PATH "../../../include")
+    set(TBB_INC_REL_PATH "../../../include/tbb/2")
 
     if (tbb_IC_INC_REL_PATH)
         file(TO_CMAKE_PATH ${tbb_IC_INC_REL_PATH} TBB_INC_REL_PATH)
diff -urp oneTBB-2020_U3.orig/cmake/TBBMakeConfig.cmake oneTBB-2020_U3/cmake/TBBMakeConfig.cmake
--- oneTBB-2020_U3.orig/cmake/TBBMakeConfig.cmake	2020-07-10 06:18:01.000000000 -0700
+++ oneTBB-2020_U3/cmake/TBBMakeConfig.cmake	2024-05-03 20:29:05.229291439 -0700
@@ -34,7 +34,7 @@ function(tbb_make_config)
         set(tbb_system_name ${tbb_MK_SYSTEM_NAME})
     endif()
 
-    set(tbb_config_dir ${tbb_MK_TBB_ROOT}/cmake)
+    set(tbb_config_dir ${tbb_MK_TBB_ROOT}/$ENV{LIBDIR}/cmake/TBB:2)
     if (tbb_MK_SAVE_TO)
         set(tbb_config_dir ${tbb_MK_SAVE_TO})
     endif()
@@ -123,7 +123,7 @@ endif()")
         message(FATAL_ERROR "Unsupported OS name: ${tbb_system_name}")
     endif()
 
-    file(READ "${tbb_MK_TBB_ROOT}/include/tbb/tbb_stddef.h" _tbb_stddef)
+    file(READ "${tbb_MK_TBB_ROOT}/include/tbb/2/tbb/tbb_stddef.h" _tbb_stddef)
     string(REGEX REPLACE ".*#define TBB_VERSION_MAJOR ([0-9]+).*" "\\1" _tbb_ver_major "${_tbb_stddef}")
     string(REGEX REPLACE ".*#define TBB_VERSION_MINOR ([0-9]+).*" "\\1" _tbb_ver_minor "${_tbb_stddef}")
     string(REGEX REPLACE ".*#define TBB_INTERFACE_VERSION ([0-9]+).*" "\\1" TBB_INTERFACE_VERSION "${_tbb_stddef}")
@@ -145,7 +145,7 @@ endif()
 
 ${TBB_CHOOSE_COMPILER_SUBDIR}
 
-get_filename_component(_tbb_lib_path \"\${_tbb_root}/${TBB_SHARED_LIB_DIR}/\${_tbb_arch_subdir}/\${_tbb_compiler_subdir}\" ABSOLUTE)
+get_filename_component(_tbb_lib_path \"\${_tbb_root}/$ENV{LIBDIR}/tbb/2\" ABSOLUTE)
 ")
 
     set(TBB_RELEASE_LIB_PATH "\${_tbb_lib_path}")
diff -urp oneTBB-2020_U3.orig/cmake/templates/TBBConfigInternal.cmake.in oneTBB-2020_U3/cmake/templates/TBBConfigInternal.cmake.in
--- oneTBB-2020_U3.orig/cmake/templates/TBBConfigInternal.cmake.in	2020-07-10 06:18:01.000000000 -0700
+++ oneTBB-2020_U3/cmake/templates/TBBConfigInternal.cmake.in	2024-05-03 20:28:43.569479914 -0700
@@ -44,6 +44,8 @@ set(TBB_INTERFACE_VERSION @TBB_INTERFACE
 
 get_filename_component(_tbb_root "${CMAKE_CURRENT_LIST_FILE}" PATH)
 get_filename_component(_tbb_root "${_tbb_root}" PATH)
+get_filename_component(_tbb_root "${_tbb_root}" PATH)
+get_filename_component(_tbb_root "${_tbb_root}" PATH)
 @TBB_CHOOSE_ARCH_AND_COMPILER@
 foreach (_tbb_component ${TBB_FIND_COMPONENTS})
     set(TBB_${_tbb_component}_FOUND 0)
@@ -58,7 +60,7 @@ foreach (_tbb_component ${TBB_FIND_COMPO
         if (NOT TARGET TBB::${_tbb_component})
             add_library(TBB::${_tbb_component} SHARED IMPORTED)
             set_target_properties(TBB::${_tbb_component} PROPERTIES
-                                INTERFACE_INCLUDE_DIRECTORIES "${_tbb_root}/include"@TBB_COMPILE_DEFINITIONS@)
+                                INTERFACE_INCLUDE_DIRECTORIES "${_tbb_root}/include/tbb/2"@TBB_COMPILE_DEFINITIONS@)
 
             if (EXISTS "${_tbb_release_lib}")
                 set_target_properties(TBB::${_tbb_component} PROPERTIES
