diff '--color=auto' -urp oneTBB-2019_U7.orig/Makefile oneTBB-2019_U7/Makefile
--- oneTBB-2019_U7.orig/Makefile	2019-05-23 03:35:29.000000000 -0700
+++ oneTBB-2019_U7/Makefile	2024-08-20 00:04:11.553218919 -0700
@@ -25,32 +25,26 @@ default: tbb tbbmalloc $(if $(use_proxy)
 all: tbb tbbmalloc tbbproxy test examples
 
 tbb: mkdir
-	$(MAKE) -C "$(work_dir)_debug"  -r -f $(tbb_root)/build/Makefile.tbb cfg=debug
-	$(MAKE) -C "$(work_dir)_release"  -r -f $(tbb_root)/build/Makefile.tbb cfg=release
+	$(MAKE) -C "$(work_dir)_$(cfg)"  -r -f $(tbb_root)/build/Makefile.tbb cfg=$(cfg)
 
 tbbmalloc: mkdir
-	$(MAKE) -C "$(work_dir)_debug"  -r -f $(tbb_root)/build/Makefile.tbbmalloc cfg=debug malloc
-	$(MAKE) -C "$(work_dir)_release"  -r -f $(tbb_root)/build/Makefile.tbbmalloc cfg=release malloc
+	$(MAKE) -C "$(work_dir)_$(cfg)"  -r -f $(tbb_root)/build/Makefile.tbbmalloc cfg=$(cfg) malloc
 
 tbbproxy: mkdir
-	$(MAKE) -C "$(work_dir)_debug"  -r -f $(tbb_root)/build/Makefile.tbbproxy cfg=debug tbbproxy
-	$(MAKE) -C "$(work_dir)_release"  -r -f $(tbb_root)/build/Makefile.tbbproxy cfg=release tbbproxy
+	$(MAKE) -C "$(work_dir)_$(cfg)"  -r -f $(tbb_root)/build/Makefile.tbbproxy cfg=$(cfg) tbbproxy
 
 test: tbb tbbmalloc $(if $(use_proxy),tbbproxy)
-	-$(MAKE) -C "$(work_dir)_debug"  -r -f $(tbb_root)/build/Makefile.tbbmalloc cfg=debug malloc_test
-	-$(MAKE) -C "$(work_dir)_debug"  -r -f $(tbb_root)/build/Makefile.test cfg=debug
-	-$(MAKE) -C "$(work_dir)_release"  -r -f $(tbb_root)/build/Makefile.tbbmalloc cfg=release malloc_test
-	-$(MAKE) -C "$(work_dir)_release"  -r -f $(tbb_root)/build/Makefile.test cfg=release
+	-$(MAKE) -C "$(work_dir)_$(cfg)"  -r -f $(tbb_root)/build/Makefile.tbbmalloc cfg=$(cfg) malloc_test
+	-$(MAKE) -C "$(work_dir)_$(cfg)"  -r -f $(tbb_root)/build/Makefile.test cfg=$(cfg)
 
 rml: mkdir
-	$(MAKE) -C "$(work_dir)_debug"  -r -f $(tbb_root)/build/Makefile.rml cfg=debug
-	$(MAKE) -C "$(work_dir)_release"  -r -f $(tbb_root)/build/Makefile.rml cfg=release
+	$(MAKE) -C "$(work_dir)_$(cfg)"  -r -f $(tbb_root)/build/Makefile.rml cfg=$(cfg)
 
 examples: tbb tbbmalloc
-	$(MAKE) -C examples -r -f Makefile tbb_root=.. release test
+	$(MAKE) -C examples -r -f Makefile tbb_root=.. $(cfg) test
 
 python: tbb
-	$(MAKE) -C "$(work_dir)_release" -rf $(tbb_root)/python/Makefile install
+	$(MAKE) -C "$(work_dir)_$(cfg)" -rf $(tbb_root)/python/Makefile install
 
 doxygen:
 	doxygen Doxyfile
@@ -58,19 +52,16 @@ doxygen:
 .PHONY: clean clean_examples mkdir info
 
 clean: clean_examples
-	$(shell $(RM) $(work_dir)_release$(SLASH)*.* >$(NUL) 2>$(NUL))
-	$(shell $(RD) $(work_dir)_release >$(NUL) 2>$(NUL))
-	$(shell $(RM) $(work_dir)_debug$(SLASH)*.* >$(NUL) 2>$(NUL))
-	$(shell $(RD) $(work_dir)_debug >$(NUL) 2>$(NUL))
+	$(shell $(RM) $(work_dir)_$(cfg)$(SLASH)*.* >$(NUL) 2>$(NUL))
+	$(shell $(RD) $(work_dir)_$(cfg) >$(NUL) 2>$(NUL))
 	@echo clean done
 
 clean_examples:
 	$(shell $(MAKE) -s -i -r -C examples -f Makefile tbb_root=.. clean >$(NUL) 2>$(NUL))
 
 mkdir:
-	$(shell $(MD) "$(work_dir)_release" >$(NUL) 2>$(NUL))
-	$(shell $(MD) "$(work_dir)_debug" >$(NUL) 2>$(NUL))
-	@echo Created $(work_dir)_release and ..._debug directories
+	$(shell $(MD) "$(work_dir)_$(cfg)" >$(NUL) 2>$(NUL))
+	@echo Created $(work_dir)_$(cfg) and ..._$(cfg) directories
 
 info:
 	@echo OS: $(tbb_os)
Only in oneTBB-2019_U7: Makefile.orig
Only in oneTBB-2019_U7: Makefile.rej
