diff -urp mojoshader-76293ed6d5c4bb33875abb92979309e2797cc6ed.orig/CMakeLists.txt mojoshader-76293ed6d5c4bb33875abb92979309e2797cc6ed/CMakeLists.txt
--- mojoshader-76293ed6d5c4bb33875abb92979309e2797cc6ed.orig/CMakeLists.txt	2021-11-29 23:22:38.357408016 -0800
+++ mojoshader-76293ed6d5c4bb33875abb92979309e2797cc6ed/CMakeLists.txt	2021-11-29 23:24:10.274024302 -0800
@@ -22,46 +22,8 @@ OPTION(XNA4_VERTEXTEXTURE "Build MojoSha
 
 INCLUDE_DIRECTORIES(.)
 
-# If Mercurial is installed and we are in a mercurial repository, include the rev# and changeset as version information.
-FIND_PROGRAM(HG hg DOC "Path to hg command line app: http://www.selenic.com/mercurial/")
-IF(NOT HG)
-    MESSAGE(STATUS "Mercurial (hg) not found. You can go on, but version info will be wrong.")
-    SET(MOJOSHADER_VERSION -1)
-    SET(MOJOSHADER_CHANGESET "???")
-ELSE(NOT HG)
-    MARK_AS_ADVANCED(HG)
-
-    # See if we are in an hg repository.
-    EXECUTE_PROCESS(
-        COMMAND hg root
-        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-        RESULT_VARIABLE HGVERSION_RC
-        ERROR_QUIET
-    )
-    IF(NOT HGVERSION_RC EQUAL 0)
-        MESSAGE(STATUS "Mercurial (hg) repository not found. You can go on, but version info will be wrong.")
-        SET(MOJOSHADER_VERSION -1)
-        SET(MOJOSHADER_CHANGESET "???")
-    ELSE(NOT HGVERSION_RC EQUAL 0)
-        # Query the rev and changeset.
-        EXECUTE_PROCESS(
-            COMMAND hg tip --template {rev}
-            WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-            RESULT_VARIABLE HGVERSION_RC
-            OUTPUT_VARIABLE MOJOSHADER_VERSION
-            ERROR_QUIET
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-        )
-        EXECUTE_PROCESS(
-            COMMAND hg tip --template hg-{rev}:{node|short}
-            WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-            RESULT_VARIABLE HGVERSION_RC
-            OUTPUT_VARIABLE MOJOSHADER_CHANGESET
-            ERROR_QUIET
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-        )
-    ENDIF(NOT HGVERSION_RC EQUAL 0)
-ENDIF(NOT HG)
+SET(MOJOSHADER_VERSION -1)
+SET(MOJOSHADER_CHANGESET "???")
 
 WRITE_FILE(
     "${CMAKE_CURRENT_SOURCE_DIR}/mojoshader_version.h"
