--- OpenSubdiv-3_6_0.orig/CMakeLists.txt	2024-05-03 15:54:03.472830917 -0700
+++ OpenSubdiv-3_6_0/CMakeLists.txt	2024-05-03 18:20:55.106975721 -0700
@@ -354,7 +354,15 @@ if(NOT NO_OMP)
     find_package(OpenMP)
 endif()
 if(NOT NO_TBB)
-    find_package(TBB 2018 COMPONENTS tbb)
+    if(USE_ONETBB)                                    # oteodoro: change section start
+        find_package(TBB 2021
+            COMPONENTS tbb
+            HINTS /usr/$ENV{LIBDIR}/cmake/TBB)
+    else()
+        find_package(TBB 2020
+            #COMPONENTS tbb
+            HINTS /usr/$ENV{LIBDIR}/cmake/TBB:2)
+    endif()                                           # oteodoro: change section end
 endif()
 if (NOT NO_OPENGL)
     find_package(OpenGL)
