diff -ur Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/Scene/SplinePath.pkg Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/Scene/SplinePath.pkg
--- Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/Scene/SplinePath.pkg	2016-02-09 17:00:27.303902850 -0800
+++ Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/Scene/SplinePath.pkg	2016-02-09 17:08:00.420128488 -0800
@@ -24,4 +24,20 @@
     tolua_property__get_set float speed;
     tolua_readonly tolua_property__get_set float length;
     tolua_property__get_set Node* controlledNode;
-};
\ No newline at end of file
+};
+
+${
+
+/* get function: interpolationMode of class  SplinePath */
+#define TOLUA_DISABLE_tolua_get_SplinePath_interpolationMode
+static int tolua_get_SplinePath_interpolationMode(lua_State* tolua_S)
+{
+  SplinePath* self = (SplinePath*)  tolua_tousertype(tolua_S,1,0);
+#ifndef TOLUA_RELEASE
+  if (!self) tolua_error(tolua_S,"invalid 'self' in accessing variable 'interpolationMode'",NULL);
+#endif
+   tolua_pushnumber(tolua_S, (lua_Number)self->GetInterpolationMode());
+ return 1;
+}
+
+$}
diff -ur Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/Scene/ValueAnimation.pkg Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/Scene/ValueAnimation.pkg
--- Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/Scene/ValueAnimation.pkg	2016-02-09 17:00:27.303902850 -0800
+++ Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/Scene/ValueAnimation.pkg	2016-02-09 17:07:56.548981165 -0800
@@ -38,4 +38,17 @@
 {
     return ToluaNewObjectGC<ValueAnimation>(tolua_S);
 }
+
+/* get function: valueType of class  ValueAnimation */
+#define TOLUA_DISABLE_tolua_get_ValueAnimation_valueType
+static int tolua_get_ValueAnimation_valueType(lua_State* tolua_S)
+{
+  ValueAnimation* self = (ValueAnimation*)  tolua_tousertype(tolua_S,1,0);
+#ifndef TOLUA_RELEASE
+  if (!self) tolua_error(tolua_S,"invalid 'self' in accessing variable 'valueType'",NULL);
+#endif
+   tolua_pushnumber(tolua_S, (lua_Number)self->GetValueType());
+ return 1;
+}
+
 $}
