diff -ur Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/UI/UIElement.pkg Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/UI/UIElement.pkg
--- Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/UI/UIElement.pkg	2016-02-09 20:32:40.425342720 -0800
+++ Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/UI/UIElement.pkg	2016-02-09 20:33:52.968093936 -0800
@@ -319,4 +319,22 @@
 
 #define GetStyle GetAppliedStyle
 #define SetColorAttr SetColor
+
+/* set function: colorAttr of class  UIElement */
+#define TOLUA_DISABLE_tolua_set_UIElement_color_ref
+static int tolua_set_UIElement_color_ref(lua_State* tolua_S)
+{
+  UIElement* self = (UIElement*)  tolua_tousertype(tolua_S,1,0);
+#ifndef TOLUA_RELEASE
+  tolua_Error tolua_err;
+  if (!self) tolua_error(tolua_S,"invalid 'self' in accessing variable 'colorAttr'",NULL);
+  if ((tolua_isvaluenil(tolua_S,2,&tolua_err) || !tolua_isusertype(tolua_S,2,"Color",0,&tolua_err)))
+   tolua_error(tolua_S,"#vinvalid type in variable assignment.",&tolua_err);
+#endif
+  self->SetColorAttr(*((Urho3D::Color*)(tolua_tousertype(tolua_S,2,0)))
+)
+;
+ return 0;
+}
+
 $}
