diff -ur urho3d-9999.orig/Source/Urho3D/AngelScript/GraphicsAPI.cpp urho3d-9999/Source/Urho3D/AngelScript/GraphicsAPI.cpp
--- urho3d-9999.orig/Source/Urho3D/AngelScript/GraphicsAPI.cpp	2016-02-24 09:04:24.982574427 -0800
+++ urho3d-9999/Source/Urho3D/AngelScript/GraphicsAPI.cpp	2016-02-24 09:10:46.639494628 -0800
@@ -506,6 +506,7 @@
     engine->RegisterObjectMethod("Texture2D", "bool SetData(Image@+, bool useAlpha = false)", asFUNCTION(Texture2DSetData), asCALL_CDECL_OBJLAST);
     engine->RegisterObjectMethod("Texture2D", "RenderSurface@+ get_renderSurface() const", asMETHOD(Texture2D, GetRenderSurface), asCALL_THISCALL);
     engine->RegisterObjectMethod("Texture2D", "Image@+ GetImage() const", asFUNCTION(Texture2DGetImage), asCALL_CDECL_OBJLAST);
+    engine->RegisterObjectMethod("Texture2D", "Texture2D& opAssign(const Texture2D&in)", asMETHODPR(Texture2D, operator =, (const Texture2D&), Texture2D&), asCALL_THISCALL);
 
     RegisterTexture<Texture3D>(engine, "Texture3D");
     engine->RegisterObjectMethod("Texture3D", "bool SetSize(int, int, int, uint, TextureUsage usage = TEXTURE_STATIC)", asMETHOD(Texture3D, SetSize), asCALL_THISCALL);
diff -ur urho3d-9999.orig/Source/Urho3D/AngelScript/ResourceAPI.cpp urho3d-9999/Source/Urho3D/AngelScript/ResourceAPI.cpp
--- urho3d-9999.orig/Source/Urho3D/AngelScript/ResourceAPI.cpp	2016-02-24 09:04:24.981574388 -0800
+++ urho3d-9999/Source/Urho3D/AngelScript/ResourceAPI.cpp	2016-02-24 09:08:13.611513659 -0800
@@ -232,6 +232,7 @@
     engine->RegisterObjectMethod("Image", "bool get_cubemap() const", asMETHOD(Image, IsCubemap), asCALL_THISCALL);
     engine->RegisterObjectMethod("Image", "bool get_array() const", asMETHOD(Image, IsArray), asCALL_THISCALL);
     engine->RegisterObjectMethod("Image", "bool get_sRGB() const", asMETHOD(Image, IsSRGB), asCALL_THISCALL);
+    engine->RegisterObjectMethod("Image", "Image& opAssign(const Image&in)", asMETHODPR(Image, operator =, (const Image&), Image&), asCALL_THISCALL);
 }
 
 static void ConstructJSONValue(JSONValue* ptr)
diff -ur urho3d-9999.orig/Source/Urho3D/AngelScript/UIAPI.cpp urho3d-9999/Source/Urho3D/AngelScript/UIAPI.cpp
--- urho3d-9999.orig/Source/Urho3D/AngelScript/UIAPI.cpp	2016-02-24 09:04:24.983574466 -0800
+++ urho3d-9999/Source/Urho3D/AngelScript/UIAPI.cpp	2016-02-24 09:06:20.050074016 -0800
@@ -147,6 +147,7 @@
     engine->RegisterObjectMethod("Sprite", "const IntRect& get_imageRect() const", asMETHOD(Sprite, GetImageRect), asCALL_THISCALL);
     engine->RegisterObjectMethod("Sprite", "void set_blendMode(BlendMode)", asMETHOD(Sprite, SetBlendMode), asCALL_THISCALL);
     engine->RegisterObjectMethod("Sprite", "BlendMode get_blendMode() const", asMETHOD(Sprite, GetBlendMode), asCALL_THISCALL);
+    engine->RegisterObjectMethod("Sprite", "Sprite& opAssign(const Sprite&in)", asMETHODPR(Sprite, operator =, (const Sprite&), Sprite&), asCALL_THISCALL);
 }
 
 static void RegisterCursor(asIScriptEngine* engine)
