diff -urp Urho3D-1.6.orig/Source/Urho3D/CMakeLists.txt Urho3D-1.6/Source/Urho3D/CMakeLists.txt
--- Urho3D-1.6.orig/Source/Urho3D/CMakeLists.txt	2016-11-25 16:42:15.201775656 -0800
+++ Urho3D-1.6/Source/Urho3D/CMakeLists.txt	2016-11-25 16:55:29.871997123 -0800
@@ -268,6 +268,10 @@ if (URHO3D_LUA)
     # This is more practical than patching its header files in many places to make them work with relative path
     list (APPEND INCLUDE_DIRS ${CMAKE_BINARY_DIR}/${DEST_INCLUDE_DIR}/ThirdParty/Lua${JIT})
 endif ()
+if (EMSCRIPTEN)
+    list (APPEND INCLUDE_DIRS ${EMSCRIPTEN_ROOT_PATH}/system/include)
+endif ()
+
 
 # Setup library output path
 if (ANDROID)
