diff -ur urho3d-9999.orig/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg urho3d-9999/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg
--- urho3d-9999.orig/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg	2016-02-28 16:06:50.642224000 -0800
+++ urho3d-9999/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg	2016-03-01 03:22:36.175226223 -0800
@@ -103,6 +103,7 @@
     tolua_readonly tolua_property__get_set bool sRGBSupport;
     tolua_readonly tolua_property__get_set bool sRGBWriteSupport;
     tolua_readonly tolua_property__get_set IntVector2 desktopResolution;
+    tolua_property__get_set Image* windowIcon;
 };
 
 Graphics* GetGraphics();
