diff -ur urho3d-9999.orig/Source/Urho3D/AngelScript/Script.cpp urho3d-9999/Source/Urho3D/AngelScript/Script.cpp
--- urho3d-9999.orig/Source/Urho3D/AngelScript/Script.cpp	2016-02-28 11:31:28.159271327 -0800
+++ urho3d-9999/Source/Urho3D/AngelScript/Script.cpp	2016-02-28 11:33:04.714951847 -0800
@@ -86,6 +86,7 @@
     scriptEngine_->SetEngineProperty(asEP_ALLOW_UNSAFE_REFERENCES, (asPWORD)true);
     scriptEngine_->SetEngineProperty(asEP_ALLOW_IMPLICIT_HANDLE_TYPES, (asPWORD)true);
     scriptEngine_->SetEngineProperty(asEP_BUILD_WITHOUT_LINE_CUES, (asPWORD)true);
+    //scriptEngine_->SetEngineProperty(asEP_PROPERTY_ACCESSOR_MODE, (asPWORD)2);
     scriptEngine_->SetMessageCallback(asMETHOD(Script, MessageCallback), this, asCALL_THISCALL);
 
     // Create the context for immediate execution
