diff -ur urho3d-9999.orig/Source/Urho3D/Urho2D/Sprite2D.h urho3d-9999/Source/Urho3D/Urho2D/Sprite2D.h
--- urho3d-9999.orig/Source/Urho3D/Urho2D/Sprite2D.h	2016-02-28 07:08:19.101719897 -0800
+++ urho3d-9999/Source/Urho3D/Urho2D/Sprite2D.h	2016-02-28 08:08:23.844987541 -0800
@@ -23,6 +23,7 @@
 #pragma once
 
 #include "../Resource/Resource.h"
+#include "../Graphics/Texture2D.h"
 
 namespace Urho3D
 {
