diff -ur urho3d-9999.orig/Source/Urho3D/AngelScript/ResourceAPI.cpp urho3d-9999/Source/Urho3D/AngelScript/ResourceAPI.cpp
--- urho3d-9999.orig/Source/Urho3D/AngelScript/ResourceAPI.cpp	2016-02-23 18:57:52.145824167 -0800
+++ urho3d-9999/Source/Urho3D/AngelScript/ResourceAPI.cpp	2016-02-23 22:08:03.392800119 -0800
@@ -564,6 +564,7 @@
     engine->RegisterObjectMethod("XMLFile", "XMLElement get_root()", asFUNCTION(XMLFileGetRootDefault), asCALL_CDECL_OBJLAST);
     engine->RegisterObjectMethod("XMLFile", "void Patch(XMLFile@+)", asMETHODPR(XMLFile, Patch, (XMLFile*), void), asCALL_THISCALL);
     engine->RegisterObjectMethod("XMLFile", "void Patch(XMLElement)", asMETHODPR(XMLFile, Patch, (XMLElement), void), asCALL_THISCALL);
+    engine->RegisterObjectMethod("XMLFile", "XMLFile& opAssign(const XMLFile&in)", asMETHODPR(XMLFile, operator =, (const XMLFile&), XMLFile&), asCALL_THISCALL);
 }
 
 void RegisterResourceAPI(asIScriptEngine* engine)
diff -ur urho3d-9999.orig/Source/Urho3D/LuaScript/pkgs/Resource/XMLFile.pkg urho3d-9999/Source/Urho3D/LuaScript/pkgs/Resource/XMLFile.pkg
--- urho3d-9999.orig/Source/Urho3D/LuaScript/pkgs/Resource/XMLFile.pkg	2016-02-23 18:57:52.164824893 -0800
+++ urho3d-9999/Source/Urho3D/LuaScript/pkgs/Resource/XMLFile.pkg	2016-02-23 22:09:21.486821960 -0800
@@ -3,6 +3,7 @@
 class XMLFile : public Resource
 {
     XMLFile();
+    //XMLFile(const XMLFile *xmlFile);
     ~XMLFile();
 
     bool FromString(const String source);
