diff -ur urho3d-9999.orig/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg urho3d-9999/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg
--- urho3d-9999.orig/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg	2016-03-01 03:42:00.695395039 -0800
+++ urho3d-9999/Source/Urho3D/LuaScript/pkgs/Graphics/Graphics.pkg	2016-03-01 04:07:28.673922456 -0800
@@ -128,4 +128,17 @@
 
 #define TOLUA_DISABLE_tolua_get_graphics_ptr
 #define tolua_get_graphics_ptr tolua_GraphicsLuaAPI_GetGraphics00
+
+/* get function: windowIcon of class  Graphics */
+#define TOLUA_DISABLE_tolua_get_Graphics_windowIcon_ptr
+static int tolua_get_Graphics_windowIcon_ptr(lua_State* tolua_S)
+{
+  Graphics* self = (Graphics*)  tolua_tousertype(tolua_S,1,0);
+#ifndef TOLUA_RELEASE
+  if (!self) tolua_error(tolua_S,"invalid 'self' in accessing variable 'windowIcon'",NULL);
+#endif
+   tolua_pushusertype(tolua_S,(void*)0,"Image");
+ return 1;
+}
+
 $}
