--- Urho3D-1.5.orig/Source/Urho3D/LuaScript/ToluaUtils.cpp	2016-02-09 13:16:12.537307407 -0800
+++ Urho3D-1.5/Source/Urho3D/LuaScript/ToluaUtils.cpp	2016-02-09 13:22:07.392788917 -0800
@@ -285,6 +285,11 @@
 
 void ToluaPushVariant(lua_State* L, const Variant* variant, const char* type)
 {
     String typeName(type);
+	if (typeName == "VariantType" && typeName == "InterpolationMode")
+	{
+		tolua_pushnumber(L, (lua_Number)variant->GetType());
+		return;
+	}
     switch (variant->GetType())
     {
