diff -urp Urho3D-1.6.orig/Source/ThirdParty/SDL/CMakeLists.txt Urho3D-1.6/Source/ThirdParty/SDL/CMakeLists.txt
--- Urho3D-1.6.orig/Source/ThirdParty/SDL/CMakeLists.txt	2016-11-25 20:20:45.020492261 -0800
+++ Urho3D-1.6/Source/ThirdParty/SDL/CMakeLists.txt	2016-11-25 20:18:33.934649209 -0800
@@ -214,6 +214,7 @@ set (INCLUDE_DIRS ${SDL2_BINARY_DIR}/inc
 #  you only need to have a platform override them if they are disabling.
 # Urho3D - bug fix - commented out double initialization of the OPT_DEF_ASM variable, it has been initialized properly in "Compiler info" section above
 if(EMSCRIPTEN)
+  list(APPEND INCLUDE_DIRS ${SDL2_SOURCE_DIR}/include ${EMSCRIPTEN_ROOT_PATH}/system/include)
   # Set up default values for the currently supported set of subsystems:
   # Emscripten/Javascript does not have assembly support, a dynamic library
   # loading architecture, low-level CPU inspection or multithreading.
