diff -urp Urho3D-1.6.orig/Source/ThirdParty/SDL/include/SDL_shape.h Urho3D-1.6/Source/ThirdParty/SDL/include/SDL_shape.h
--- Urho3D-1.6.orig/Source/ThirdParty/SDL/include/SDL_shape.h	2016-08-08 04:44:35.000000000 -0700
+++ Urho3D-1.6/Source/ThirdParty/SDL/include/SDL_shape.h	2016-11-25 22:00:37.206735080 -0800
@@ -26,7 +26,6 @@
 #include "SDL_pixels.h"
 #include "SDL_rect.h"
 #include "SDL_surface.h"
-#include "SDL_video.h"
 
 #include "begin_code.h"
 /* Set up for C function definitions, even when using C++ */
diff -urp Urho3D-1.6.orig/Source/ThirdParty/SDL/src/video/SDL_sysvideo.h Urho3D-1.6/Source/ThirdParty/SDL/src/video/SDL_sysvideo.h
--- Urho3D-1.6.orig/Source/ThirdParty/SDL/src/video/SDL_sysvideo.h	2016-08-08 04:44:35.000000000 -0700
+++ Urho3D-1.6/Source/ThirdParty/SDL/src/video/SDL_sysvideo.h	2016-11-25 22:00:54.130363588 -0800
@@ -25,6 +25,7 @@
 
 #include "SDL_messagebox.h"
 #include "SDL_shape.h"
+#include "SDL_video.h"
 #include "SDL_thread.h"
 
 /* The SDL video driver */
