diff -ur Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/UI/Text3D.pkg Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/UI/Text3D.pkg
--- Urho3D-1.5.orig/Source/Urho3D/LuaScript/pkgs/UI/Text3D.pkg	2016-02-09 23:57:50.469310312 -0800
+++ Urho3D-1.5/Source/Urho3D/LuaScript/pkgs/UI/Text3D.pkg	2016-02-09 23:59:14.893569919 -0800
@@ -57,6 +57,8 @@
     IntVector2 GetCharPosition(unsigned index);
     IntVector2 GetCharSize(unsigned index);
     const Color& GetColor(Corner corner) const;
+    Corner GetCorner() const;
+    void SetCorner(Corner corner);
     float GetOpacity() const;
     FaceCameraMode GetFaceCameraMode() const;
     
@@ -73,8 +75,8 @@
     tolua_property__get_set Color effectColor;
     tolua_property__get_set float effectDepthBias;
     tolua_property__get_set int width;
-    tolua_property__get_set Color color; // Write only property.
-    tolua_property__get_set Corner corner;
+    tolua_property__get_set Color color; // Write only property.
+    tolua_property__get_set Corner corner;
 
     tolua_readonly tolua_property__get_set int rowHeight;
     tolua_readonly tolua_property__get_set unsigned numRows;
@@ -98,4 +100,4 @@
 
 #define TOLUA_DISABLE_tolua_get_Text3D_color_ref
 #define tolua_get_Text3D_color_ref NULL
-$}
\ No newline at end of file
+$}
