diff -urp astropulse-cpu-7.01.orig/AKv8/client/analyzeReport.cpp astropulse-cpu-7.01/AKv8/client/analyzeReport.cpp
--- astropulse-cpu-7.01.orig/AKv8/client/analyzeReport.cpp	2017-01-19 16:59:52.569759793 -0800
+++ astropulse-cpu-7.01/AKv8/client/analyzeReport.cpp	2017-01-19 17:06:15.342219299 -0800
@@ -175,7 +175,7 @@ int result_spike(SPIKE_INFO &si) {
 
   //R: & JWS: sanity check for found result
   if(si.s.peak_power > si.s.fft_len){
-    boinc_temporary_exit(5*60,"Impossible Spike, retrying from checkpoint.",true);
+    boinc_temporary_exit(5*60,"Impossible Spike, retrying from checkpoint.");
   }
 
 
@@ -231,7 +231,7 @@ int result_gaussian(GAUSS_INFO &gi) {
 
   //R: & JWS: sanity check for found result
   if(gi.g.peak_power > swi.analysis_cfg.gauss_pot_length){
-    boinc_temporary_exit(5*60,"Improbable Gaussian, retrying from checkpoint.",true);}
+    boinc_temporary_exit(5*60,"Improbable Gaussian, retrying from checkpoint.");}
 
 
   int retval=0;
@@ -282,7 +282,7 @@ BOINCASSERT(_CrtCheckMemory());
 
   //JWS: sanity check for found result.
   if(Power > (float)pot_len){
-    boinc_temporary_exit(5*60,"Impossible Triplet, retrying from checkpoint.",true);
+    boinc_temporary_exit(5*60,"Impossible Triplet, retrying from checkpoint.");
   }
 
   if (!inv) inv = (int*)calloc_a(swi.analysis_cfg.triplet_pot_length, sizeof(int), MEM_ALIGN);
@@ -419,7 +419,7 @@ BOINCASSERT(_CrtCheckMemory());
   //JWS: sanity check for found result. The limit would be 4/3 * period,
   // but discrete math and short PoTs can exceed that slightly.
   if(PulsePower > 1.4 * period){
-    boinc_temporary_exit(5*60,"Impossible Pulse, retrying from checkpoint.",true);
+    boinc_temporary_exit(5*60,"Impossible Pulse, retrying from checkpoint.");
   }
 
   // pulse info
diff -urp astropulse-cpu-7.01.orig/AKv8/client/autocorr.cpp astropulse-cpu-7.01/AKv8/client/autocorr.cpp
--- astropulse-cpu-7.01.orig/AKv8/client/autocorr.cpp	2017-01-19 16:59:52.569759793 -0800
+++ astropulse-cpu-7.01/AKv8/client/autocorr.cpp	2017-01-19 17:06:13.967167358 -0800
@@ -118,7 +118,7 @@ int FindAutoCorrelation(
 
     //JWS: Sanity check, delayed peaks cannot be higher than the zero delay power.
     if (temp > AutoCorrelation[0]) {
-      boinc_temporary_exit(5*60,"Impossible Autocorr power, retrying from checkpoint.",true);
+      boinc_temporary_exit(5*60,"Impossible Autocorr power, retrying from checkpoint.");
     }
 
     //  autocorr info
diff -urp astropulse-cpu-7.01.orig/client/analyzeReport.cpp astropulse-cpu-7.01/client/analyzeReport.cpp
--- astropulse-cpu-7.01.orig/client/analyzeReport.cpp	2017-01-19 16:59:52.233747100 -0800
+++ astropulse-cpu-7.01/client/analyzeReport.cpp	2017-01-19 17:06:46.829408657 -0800
@@ -175,7 +175,7 @@ int result_spike(SPIKE_INFO &si) {
 
   //R: & JWS: sanity check for found result
   if(si.s.peak_power > si.s.fft_len){
-    boinc_temporary_exit(5*60,"Impossible Spike, retrying from checkpoint.",true);
+    boinc_temporary_exit(5*60,"Impossible Spike, retrying from checkpoint.");
   }
 
 
@@ -231,7 +231,7 @@ int result_gaussian(GAUSS_INFO &gi) {
 
   //R: & JWS: sanity check for found result
   if(gi.g.peak_power > swi.analysis_cfg.gauss_pot_length){
-    boinc_temporary_exit(5*60,"Improbable Gaussian, retrying from checkpoint.",true);}
+    boinc_temporary_exit(5*60,"Improbable Gaussian, retrying from checkpoint.");}
 
 
   int retval=0;
@@ -282,7 +282,7 @@ BOINCASSERT(_CrtCheckMemory());
 
   //JWS: sanity check for found result.
   if(Power > (float)pot_len){
-    boinc_temporary_exit(5*60,"Impossible Triplet, retrying from checkpoint.",true);
+    boinc_temporary_exit(5*60,"Impossible Triplet, retrying from checkpoint.");
   }
 
   if (!inv) inv = (int*)calloc_a(swi.analysis_cfg.triplet_pot_length, sizeof(int), MEM_ALIGN);
@@ -419,7 +419,7 @@ BOINCASSERT(_CrtCheckMemory());
   //JWS: sanity check for found result. The limit would be 4/3 * period,
   // but discrete math and short PoTs can exceed that slightly.
   if(PulsePower > 1.4 * period){
-    boinc_temporary_exit(5*60,"Impossible Pulse, retrying from checkpoint.",true);
+    boinc_temporary_exit(5*60,"Impossible Pulse, retrying from checkpoint.");
   }
 
   // pulse info
diff -urp astropulse-cpu-7.01.orig/client/autocorr.cpp astropulse-cpu-7.01/client/autocorr.cpp
--- astropulse-cpu-7.01.orig/client/autocorr.cpp	2017-01-19 16:59:52.234747137 -0800
+++ astropulse-cpu-7.01/client/autocorr.cpp	2017-01-19 17:06:12.414108694 -0800
@@ -118,7 +118,7 @@ int FindAutoCorrelation(
 
     //JWS: Sanity check, delayed peaks cannot be higher than the zero delay power.
     if (temp > AutoCorrelation[0]) {
-      boinc_temporary_exit(5*60,"Impossible Autocorr power, retrying from checkpoint.",true);
+      boinc_temporary_exit(5*60,"Impossible Autocorr power, retrying from checkpoint.");
     }
 
     //  autocorr info
