diff -urp astropulse-cpu-7.01.orig/AP/server/m4/optimizations.m4 astropulse-cpu-7.01/AP/server/m4/optimizations.m4
--- astropulse-cpu-7.01.orig/AP/server/m4/optimizations.m4	2017-01-19 15:18:18.885425130 -0800
+++ astropulse-cpu-7.01/AP/server/m4/optimizations.m4	2017-01-19 15:21:40.422142509 -0800
@@ -13,7 +13,11 @@ if test x_$enable_sse3 = x_yes ; then
     [Define to 1 if you want to use SSE3 optimizations])
 # put compiler specific flags here
   if test x_$ac_cv_c_compiler_gnu = x_yes ; then
-    CFLAGS="-march=prescott -msse3  -mfpmath=sse ${CFLAGS}"
+    if test -n "`uname -m | egrep 'x86_64|amd64'`" ; then
+      CFLAGS="-march=opteron-sse3 -msse3  -mfpmath=sse ${CFLAGS}"
+    else
+      CFLAGS="-march=prescott -msse3  -mfpmath=sse ${CFLAGS}"
+    fi
   fi
 fi
 
@@ -27,7 +31,17 @@ if test x_$enable_sse2 = x_yes ; then
     [Define to 1 if you want to use SSE2 optimizations])
 # put compiler specific flags here
   if test x_$ac_cv_c_compiler_gnu = x_yes ; then
-    CFLAGS="-march=pentium4 -msse2 -mfpmath=sse ${CFLAGS}"
+    if test -n "`uname -m | egrep 'x86_64|amd64'`" ; then
+      CFLAGS="-msse2 -mfpmath=sse ${CFLAGS}"
+      if test -z "echo $CFLAGS | grep march=" ; then
+        CFLAGS="-march=opteron ${CFLAGS}"
+      fi
+    else
+      CFLAGS="-msse2 -mfpmath=sse ${CFLAGS}"
+      if test -z "echo $CFLAGS | grep march=" ; then
+        CFLAGS="-march=pentium4 ${CFLAGS}"
+      fi
+    fi
   fi
 fi
 
