--- src/game/Object/Object.cpp.orig	2015-06-22 02:49:05.931927675 -0700
+++ src/game/Object/Object.cpp	2015-06-22 02:50:39.946569145 -0700
@@ -829,6 +829,14 @@
     }
 }
 
+void Object::UpdateUInt32Value(uint16 index, uint32 value)
+{
+    MANGOS_ASSERT(index < m_valuesCount || PrintIndexError(index, true));
+
+    m_uint32Values[index] = value;
+    m_changedValues[index] = true;
+}
+
 void Object::SetUInt64Value(uint16 index, const uint64& value)
 {
     MANGOS_ASSERT(index + 1 < m_valuesCount || PrintIndexError(index, true));
