--- contrib/vmap_assembler/CMakeLists.txt.orig	2015-06-20 17:26:39.657381213 -0700
+++ contrib/vmap_assembler/CMakeLists.txt	2015-06-20 17:27:57.287355387 -0700
@@ -82,6 +82,24 @@
 
 target_link_libraries(vmap g3dlite z)
 
+set(EXECUTABLE_LINK_FLAGS "")
+
+if(UNIX)
+  if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+    set(EXECUTABLE_LINK_FLAGS "-Wl, --no-as-needed -pthread -lrt ${EXECUTABLE_LINK_FLAGS}")
+  elseif(CMAKE_SYSTEM_NAME MATCHES "Linux")
+    set(EXECUTABLE_LINK_FLAGS "-Wl,--no-as-needed -ldl -pthread -lrt ${EXECUTABLE_LINK_FLAGS}")
+  endif()
+endif()
+
+if(APPLE)
+  set(EXECUTABLE_LINK_FLAGS "-framework Carbon ${EXECUTABLE_LINK_FLAGS}")
+endif()
+
+set_target_properties(${EXECUTABLE_NAME} PROPERTIES LINK_FLAGS
+  "${EXECUTABLE_LINK_FLAGS}"
+)
+
 add_executable(vmap_assembler vmap_assembler.cpp)
 target_link_libraries(vmap_assembler vmap)
 
