--- dep/StormLib/CMakeLists.txt.orig	2015-06-21 22:05:43.837336457 -0700
+++ dep/StormLib/CMakeLists.txt	2015-06-21 22:06:14.492513633 -0700
@@ -283,7 +283,7 @@
     endif()
 endif()
 
-add_library(StormLib STATIC ${SRC_FILES} ${SRC_ADDITIONAL_FILES})
+add_library(StormLib SHARED ${SRC_FILES} ${SRC_ADDITIONAL_FILES})
 target_link_libraries(StormLib ${LINK_LIBS})
 
 if(APPLE)
@@ -300,4 +300,4 @@
 # On Win32, build StormLib.dll since we don't want to clash with Storm.dll
 if(WIN32)
     set_target_properties(StormLib PROPERTIES OUTPUT_NAME StormLib)
-endif()
\ No newline at end of file
+endif()
