--- src/game/Object/Unit.h.orig	2015-06-22 02:04:23.174664961 -0700
+++ src/game/Object/Unit.h	2015-06-22 02:16:00.189140543 -0700
@@ -1715,6 +1715,13 @@
          */
         int32 ModifyHealth(int32 val);
 
+        // Eluna-related health functions
+        bool HealthAbovePctHealed(int32 pct, uint32 heal) const { return uint64(GetHealth()) + uint64(heal) > CountPctFromMaxHealth(pct); }
+        bool IsFullHealth() const { return GetHealth() == GetMaxHealth(); }
+        bool HealthBelowPct(int32 pct) const { return GetHealth() < CountPctFromMaxHealth(pct); }
+        bool HealthBelowPctDamaged(int32 pct, uint32 damage) const { return int64(GetHealth()) - int64(damage) < int64(CountPctFromMaxHealth(pct)); }
+        bool HealthAbovePct(int32 pct) const { return GetHealth() > CountPctFromMaxHealth(pct); }
+
         /** 
          * Gets the power type for this Unit
          * @return The type of power this Unit uses
