--- src/game/Object/ObjectMgr.cpp.orig	2015-06-21 15:06:47.672480472 -0700
+++ src/game/Object/ObjectMgr.cpp	2015-06-21 15:08:09.240557578 -0700
@@ -239,6 +239,17 @@
     return NULL;
 }
 
+void ObjectMgr::AddLocaleString(std::string const& s, LocaleConstant locale, StringVector& data)
+{
+    if (!s.empty())
+    {
+        if (data.size() <= size_t(locale))
+            data.resize(locale + 1);
+
+        data[locale] = s;
+    }
+}
+
 void ObjectMgr::LoadCreatureLocales()
 {
     mCreatureLocaleMap.clear();                             // need for reload case
