diff -urp rocFFT-rocm-5.6.0.orig/clients/tests/cmake/FindFFTW.cmake rocFFT-rocm-5.6.0/clients/tests/cmake/FindFFTW.cmake
--- rocFFT-rocm-5.6.0.orig/clients/tests/cmake/FindFFTW.cmake	2023-06-01 18:18:31.000000000 -0700
+++ rocFFT-rocm-5.6.0/clients/tests/cmake/FindFFTW.cmake	2023-07-30 00:28:56.676122364 -0700
@@ -47,11 +47,11 @@ if( FFTW_FIND_REQUIRED_FLOAT OR FFTW_FIN
   find_library( FFTW_LIBRARIES_SINGLE
       NAMES fftw3f fftw3f-3 fftw3 fftw3-3
       HINTS
-          ${FFTW_ROOT}/lib
-          $ENV{FFTW_ROOT}/lib
+          ${FFTW_ROOT}/lib64
+          $ENV{FFTW_ROOT}/lib64
       PATHS
-          /usr/lib
-          /usr/local/lib
+          /usr/lib64
+          /usr/local/lib64
       PATH_SUFFIXES
           x86_64-linux-gnu
       DOC "FFTW dynamic library single"
@@ -79,11 +79,11 @@ if( FFTW_FIND_REQUIRED_DOUBLE )
   find_library( FFTW_LIBRARIES_DOUBLE
       NAMES fftw3
       HINTS
-          ${FFTW_ROOT}/lib
-          $ENV{FFTW_ROOT}/lib
+          ${FFTW_ROOT}/lib64
+          $ENV{FFTW_ROOT}/lib64
       PATHS
-          /usr/lib
-          /usr/local/lib
+          /usr/lib64
+          /usr/local/lib64
       PATH_SUFFIXES
           x86_64-linux-gnu
       DOC "FFTW dynamic library double"
diff -urp rocFFT-rocm-5.6.0.orig/clients/tests/CMakeLists.txt rocFFT-rocm-5.6.0/clients/tests/CMakeLists.txt
--- rocFFT-rocm-5.6.0.orig/clients/tests/CMakeLists.txt	2023-07-30 00:23:33.944113958 -0700
+++ rocFFT-rocm-5.6.0/clients/tests/CMakeLists.txt	2023-07-30 00:28:54.996122320 -0700
@@ -233,7 +233,7 @@ if( BUILD_CLIENTS_TESTS_OPENMP )
   target_compile_options( rocfft-test PRIVATE -DBUILD_CLIENTS_TESTS_OPENMP )
   if( CMAKE_CXX_COMPILER MATCHES ".*/hipcc$" )  
     target_compile_options( rocfft-test PRIVATE -fopenmp )
-    target_link_libraries( rocfft-test PRIVATE -fopenmp -L${HIP_CLANG_ROOT}/lib -Wl,-rpath=${HIP_CLANG_ROOT}/lib )
+    target_link_libraries( rocfft-test PRIVATE -fopenmp -L${HIP_CLANG_ROOT}/lib64 -Wl,-rpath=${HIP_CLANG_ROOT}/lib64 )
     target_include_directories( rocfft-test PRIVATE ${HIP_CLANG_ROOT}/include )
   else()
     if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
diff -urp rocFFT-rocm-5.6.0.orig/library/src/CMakeLists.txt rocFFT-rocm-5.6.0/library/src/CMakeLists.txt
--- rocFFT-rocm-5.6.0.orig/library/src/CMakeLists.txt	2023-07-30 00:23:33.948113958 -0700
+++ rocFFT-rocm-5.6.0/library/src/CMakeLists.txt	2023-07-30 00:28:53.172122273 -0700
@@ -46,7 +46,7 @@ else()
   if( WIN32 )
     set( ROCFFT_RTC_LINK_LIBS "${HIP_PATH}/lib/hiprtc.lib" )
   else()
-    set( ROCFFT_RTC_LINK_LIBS -L${ROCM_PATH}/lib -lhiprtc -ldl )
+    set( ROCFFT_RTC_LINK_LIBS -L${ROCM_PATH}/lib64 -lhiprtc -ldl )
   endif()
 endif()
 
