diff -urp oiio-Release-2.1.15.0.orig/src/raw.imageio/rawinput.cpp oiio-Release-2.1.15.0/src/raw.imageio/rawinput.cpp
--- oiio-Release-2.1.15.0.orig/src/raw.imageio/rawinput.cpp	2020-05-10 20:43:52.000000000 -0700
+++ oiio-Release-2.1.15.0/src/raw.imageio/rawinput.cpp	2020-10-05 19:45:17.693403362 -0700
@@ -653,8 +653,13 @@ RawInput::open_raw(bool unpack, const st
 #if LIBRAW_VERSION >= LIBRAW_MAKE_VERSION(0, 17, 0)
     if (other.parsed_gps.gpsparsed) {
         add("GPS", "Latitude", other.parsed_gps.latitude, false, 0.0f);
+#if LIBRAW_VERSION >= LIBRAW_MAKE_VERSION(0, 20, 0)
+        add("GPS", "Longitude", other.parsed_gps.longitude, false,
+            0.0f);
+#else
         add("GPS", "Longitude", other.parsed_gps.longtitude, false,
             0.0f);  // N.B. wrong spelling!
+#endif
         add("GPS", "TimeStamp", other.parsed_gps.gpstimestamp, false, 0.0f);
         add("GPS", "Altitude", other.parsed_gps.altitude, false, 0.0f);
         add("GPS", "LatitudeRef", string_view(&other.parsed_gps.latref, 1),
