diff -urp BulletSharpPInvoke-0.9.orig/BulletSharpGen/BulletSharpGen.csproj BulletSharpPInvoke-0.9/BulletSharpGen/BulletSharpGen.csproj
--- BulletSharpPInvoke-0.9.orig/BulletSharpGen/BulletSharpGen.csproj	2016-10-05 13:52:09.000000000 -0700
+++ BulletSharpPInvoke-0.9/BulletSharpGen/BulletSharpGen.csproj	2017-01-11 04:54:04.765389147 -0800
@@ -43,10 +43,7 @@
     <None Include="packages.config" />
   </ItemGroup>
   <ItemGroup>
-    <Reference Include="ClangSharp, Version=0.0.0.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>packages\clang.sharp.3.6.2\lib\net40\ClangSharp.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <Reference Include="ClangSharp" />
     <Reference Include="Microsoft.CodeAnalysis, Version=1.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>packages\Microsoft.CodeAnalysis.Common.1.2.1\lib\net45\Microsoft.CodeAnalysis.dll</HintPath>
       <Private>True</Private>
@@ -139,13 +136,6 @@
     <Analyzer Include="packages\Microsoft.CodeAnalysis.Analyzers.1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Import Project="packages\clang.sharp.3.6.2\build\net40\clang.sharp.targets" Condition="Exists('packages\clang.sharp.3.6.2\build\net40\clang.sharp.targets')" />
-  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
-    <PropertyGroup>
-      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
-    </PropertyGroup>
-    <Error Condition="!Exists('packages\clang.sharp.3.6.2\build\net40\clang.sharp.targets')" Text="$([System.String]::Format('$(ErrorText)', 'packages\clang.sharp.3.6.2\build\net40\clang.sharp.targets'))" />
-  </Target>
   <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
        Other similar extension points exist, see Microsoft.Common.targets.
   <Target Name="BeforeBuild">
@@ -153,4 +143,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
