diff -urp mujoco-2.3.7.orig/python/setup.py mujoco-2.3.7/python/setup.py
--- mujoco-2.3.7.orig/python/setup.py	2023-07-20 02:13:55.000000000 -0700
+++ mujoco-2.3.7/python/setup.py	2023-07-29 10:05:04.634772369 -0700
@@ -191,6 +191,8 @@ class BuildCMakeExtension(build_ext.buil
 
   def _copy_external_libraries(self):
     dst = os.path.dirname(self.get_ext_fullpath(self.extensions[0].name))
+#    os.makedirs(os.path.join("..", dst), exist_ok=True) # oteodoro: fix bug and avoid arch lookup
+    os.makedirs(dst, exist_ok=True) # oteodoro: fix bug and avoid arch lookup
     for directory, _, filenames in os.walk(os.environ[MUJOCO_PATH]):
       for pattern in get_external_lib_patterns():
         for filename in fnmatch.filter(filenames, pattern):
Only in mujoco-2.3.7/python: setup.py.orig
Only in mujoco-2.3.7/python: setup.py.rej
