unchanged:
--- hipBLASLt-rocm-5.6.0.orig/CMakeLists.txt	2023-08-19 18:56:10.309522195 -0700
+++ hipBLASLt-rocm-5.6.0/CMakeLists.txt	2023-08-19 19:00:06.441528054 -0700
@@ -78,7 +78,7 @@ include( ROCMHeaderWrapper )
 
 # Append our library helper cmake path and the cmake path for hip (for convenience)
 # Users may override HIP path by specifying their own in CMAKE_MODULE_PATH
-list( APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake  ${ROCM_PATH}/lib/cmake/hip /opt/rocm/lib/cmake/hip ${HIP_PATH}/cmake )
+list( APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake  ${ROCM_PATH}/lib64/cmake/hip /opt/rocm/lib64/cmake/hip ${HIP_PATH}/cmake )
 
 # Set a default build type if none was specified
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
@@ -96,7 +96,7 @@ endif()
 project(hipblaslt LANGUAGES CXX)
 
 # Force library install path to lib (CentOS 7 defaults to lib64)
-set(CMAKE_INSTALL_LIBDIR "lib" CACHE INTERNAL "Installation directory for libraries" FORCE)
+set(CMAKE_INSTALL_LIBDIR "lib64" CACHE INTERNAL "Installation directory for libraries" FORCE)
 
 
 # Build options
@@ -305,7 +305,7 @@ if( NOT CPACK_PACKAGING_INSTALL_PREFIX )
     set( CPACK_PACKAGING_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" )
 endif()
 
-set( CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "\${CPACK_PACKAGING_INSTALL_PREFIX}" "\${CPACK_PACKAGING_INSTALL_PREFIX}/include" "\${CPACK_PACKAGING_INSTALL_PREFIX}/lib" )
+set( CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "\${CPACK_PACKAGING_INSTALL_PREFIX}" "\${CPACK_PACKAGING_INSTALL_PREFIX}/include" "\${CPACK_PACKAGING_INSTALL_PREFIX}/lib64" )
 
 # Give hipblaslt compiled for CUDA backend a different name
 if( NOT USE_CUDA )
unchanged:
--- hipBLASLt-rocm-5.6.0.orig/deps/external-lapack.cmake	2023-05-28 21:19:48.000000000 -0700
+++ hipBLASLt-rocm-5.6.0/deps/external-lapack.cmake	2023-08-19 19:02:06.297531027 -0700
@@ -40,8 +40,8 @@ include( GNUInstallDirs )
 # lib/<machine> paths
 # if CMAKE_INSTALL_LIBDIR is of the form above, strip the machine
 # Match against a '/' in CMAKE_INSTALL_LIBDIR, i.e. lib/x86_64-linux-gnu
-if( ${CMAKE_INSTALL_LIBDIR} MATCHES "lib/.*" )
-  list( APPEND lapack_cmake_args "-DCMAKE_INSTALL_LIBDIR=lib" )
+if( ${CMAKE_INSTALL_LIBDIR} MATCHES "lib64/.*" )
+  list( APPEND lapack_cmake_args "-DCMAKE_INSTALL_LIBDIR=lib64" )
 endif( )
 
 ExternalProject_Add(
only in patch2:
unchanged:
--- hipBLASLt-rocm-5.6.0.orig/tensilelite/Tensile/cmake/TensileConfig.cmake	2023-05-28 21:19:48.000000000 -0700
+++ hipBLASLt-rocm-5.6.0/tensilelite/Tensile/cmake/TensileConfig.cmake	2023-08-18 22:10:40.675668095 -0700
@@ -35,8 +35,8 @@ if(NOT DEFINED Tensile_ROOT)
         execute_process(COMMAND "${Tensile_PREFIX}/bin/TensileGetPath" OUTPUT_VARIABLE Tensile_ROOT)
     endif()
 endif()
-list(APPEND CMAKE_MODULE_PATH "${Tensile_ROOT}/Source/cmake/")
-list(APPEND CMAKE_MODULE_PATH "${Tensile_ROOT}/Source/")
+list(APPEND CMAKE_MODULE_PATH "${Tensile_ROOT}/Tensile/Source/cmake/")
+list(APPEND CMAKE_MODULE_PATH "${Tensile_ROOT}/Tensile/Source/")
 
 if("HIP" IN_LIST Tensile_FIND_COMPONENTS)
     set(TENSILE_USE_HIP ON CACHE BOOL "Use HIP")
@@ -69,8 +69,8 @@ else()
     set(TENSILE_STATIC_ONLY OFF CACHE BOOL "Disable exporting symbols from shared library.")
 endif()
 
-add_subdirectory("${Tensile_ROOT}/Source" "Tensile")
-include("${Tensile_ROOT}/Source/TensileCreateLibrary.cmake")
+add_subdirectory("${Tensile_ROOT}/Tensile/Source" "Tensile")
+include("${Tensile_ROOT}/Tensile/Source/TensileCreateLibrary.cmake")
 
 # Output target: ${Tensile_VAR_PREFIX}_LIBRARY_TARGET. Ensures that the libs get built in Tensile_OUTPUT_PATH/library.
 function(TensileCreateLibraryFiles
@@ -120,7 +120,7 @@ function(TensileCreateLibraryFiles
   if(Tensile_TENSILE_ROOT)
     set(Script "${Tensile_TENSILE_ROOT}/bin/TensileCreateLibrary")
   else()
-    set(Script "${Tensile_ROOT}/bin/TensileCreateLibrary")
+    set(Script "${Tensile_ROOT}/Tensile/bin/TensileCreateLibrary")
   endif()
 
   message(STATUS "Tensile script: ${Script}")
