--- a/cmake/virtualenv.cmake.orig	2025-07-25 11:47:20.000000000 -0700
+++ b/cmake/virtualenv.cmake	2025-12-05 17:39:15.342120735 -0800
@@ -13,11 +13,6 @@ get_filename_component(VIRTUALENV_PYTHON
 set(VIRTUALENV_HOME_DIR ${CMAKE_BINARY_DIR}/virtualenv CACHE PATH "Path to virtual environment")
 
 function(virtualenv_create)
-    execute_process(
-      COMMAND ${VIRTUALENV_PYTHON_EXE} -m venv ${VIRTUALENV_HOME_DIR} --system-site-packages --clear
-      COMMAND_ECHO STDOUT
-    )
-
     if(WIN32)
         set(VIRTUALENV_BIN_DIR ${VIRTUALENV_HOME_DIR}/Scripts CACHE PATH "Path to virtualenv bin directory")
     else()
@@ -27,21 +22,26 @@ endfunction()
 
 function(virtualenv_install)
     virtualenv_create()
-    execute_process(
-      COMMAND ${VIRTUALENV_BIN_DIR}/${VIRTUALENV_PYTHON_EXENAME} -m pip install --upgrade pip
-      COMMAND_ECHO STDOUT
-    )
-    execute_process(
-      COMMAND ${VIRTUALENV_BIN_DIR}/${VIRTUALENV_PYTHON_EXENAME} -m pip install --upgrade setuptools
-      COMMAND_ECHO STDOUT
-    )
-    execute_process(
-      COMMAND ${VIRTUALENV_BIN_DIR}/${VIRTUALENV_PYTHON_EXENAME} -m pip install ${ARGN}
-      COMMAND_ECHO STDOUT
-      RESULT_VARIABLE return_code
-      ERROR_VARIABLE error_message
-      OUTPUT_VARIABLE output_message      
-    )
+
+    find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
+    # Create virtual environment for TensileLite if it does not exist
+    if(NOT EXISTS "${CMAKE_BINARY_DIR}/virtualenv")
+      message(STATUS "Creating TensileLite Python virtual environment")
+      execute_process(COMMAND ${Python3_EXECUTABLE} -m venv virtualenv
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+      execute_process(COMMAND ${CMAKE_BINARY_DIR}/virtualenv/bin/pip install --quiet --upgrade pip
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+      execute_process(COMMAND ${CMAKE_BINARY_DIR}/virtualenv/bin/pip install --quiet PyYAML msgpack
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+      execute_process(COMMAND ${CMAKE_BINARY_DIR}/virtualenv/bin/python -m pip install --quiet -e ${ARGN}
+                      --use-pep517 --config-settings editable_mode=compat --no-build-isolation
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+                      COMMAND_ECHO STDOUT
+                      RESULT_VARIABLE return_code
+                      ERROR_VARIABLE error_message
+                      OUTPUT_VARIABLE output_message)
+    endif()
 
     if(return_code)
         message("Error Code: ${rc}")
