--- a/CMakeLists.txt.orig	2025-12-05 15:55:08.530447882 -0800
+++ b/CMakeLists.txt	2025-12-05 15:58:54.644609937 -0800
@@ -223,17 +223,27 @@ else()
       option(TENSILE_USE_MSGPACK   "Use msgpack for parsing config files." ON)
     endif()
 
+    find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
+    # Create virtual environment for TensileLite if it does not exist
+    if(NOT EXISTS "${CMAKE_BINARY_DIR}/virtualenv")
+      message(STATUS "Creating TensileLite Python virtual environment")
+      execute_process(COMMAND ${Python3_EXECUTABLE} -m venv virtualenv
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+      execute_process(COMMAND ${CMAKE_BINARY_DIR}/virtualenv/bin/pip install --quiet --upgrade pip
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+      execute_process(COMMAND ${CMAKE_BINARY_DIR}/virtualenv/bin/pip install --quiet PyYAML msgpack
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+      execute_process(COMMAND ${CMAKE_BINARY_DIR}/virtualenv/bin/python -m pip install --quiet -e ${CMAKE_SOURCE_DIR}/tensilelite
+                      --use-pep517 --config-settings editable_mode=compat --no-build-isolation
+                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+    endif()
+    set(VIRTUALENV_HOME_DIR "${CMAKE_BINARY_DIR}/virtualenv" CACHE PATH "path to virtual env")
+
     if (WIN32)
       set( Tensile_ROOT "${CMAKE_BINARY_DIR}/virtualenv/Lib/site-packages/Tensile" )
     endif()
 
-    include(virtualenv)
-    if (Tensile_TEST_LOCAL_PATH)
-      virtualenv_install(${Tensile_TEST_LOCAL_PATH})
-      message (STATUS "using local Tensile from ${Tensile_TEST_LOCAL_PATH}, copied to ${Tensile_ROOT}")
-    else()
-      virtualenv_install(${CMAKE_SOURCE_DIR}/tensilelite)
-    endif()
     message(STATUS "Adding ${VIRTUALENV_HOME_DIR} to CMAKE_PREFIX_PATH")
     list(APPEND CMAKE_PREFIX_PATH ${VIRTUALENV_HOME_DIR})
     cmake_policy(SET CMP0074 NEW)
