--- a/library/src/amd_detail/rocblaslt/src/CMakeLists.txt.orig	2025-11-19 20:11:39.489113420 -0800
+++ b/library/src/amd_detail/rocblaslt/src/CMakeLists.txt	2025-11-19 20:12:29.520743585 -0800
@@ -98,7 +98,9 @@ if( BUILD_WITH_TENSILE )
     )
   endif()
 
-  TensileCreateExtOpLibraries("${PROJECT_BINARY_DIR}/Tensile/library" "${Tensile_ARCHITECTURE}")
+  if(USE_TENSILELITE)
+    TensileCreateExtOpLibraries("${PROJECT_BINARY_DIR}/Tensile/library" "${Tensile_ARCHITECTURE}")
+  endif()
 
   # Create a unique name for TensileHost compiled for rocBLAS
   set_target_properties( TensileHost PROPERTIES OUTPUT_NAME rocblaslt-tensile CXX_EXTENSIONS NO )
