--- AP/client/Makefile.am.orig	2015-12-05 23:11:33.664345277 -0800
+++ AP/client/Makefile.am	2015-12-05 23:14:32.604278246 -0800
@@ -122,8 +122,9 @@
 	ap_graphics.h					\
 	sbtf.h						\
         ap_version.h                                    \
-	ap_shmem.h                                      \
+	ap_gfx_main.h                                   \
 	$(AP_GRX_SOURCES)
+#
 
 if OPENCL
 ap_client_opencl_SOURCES = \
