diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/Makefile.am setiathome-7.08/AP/client/Makefile.am
--- setiathome-7.08.orig/AP/client/Makefile.am	2015-12-06 10:44:40.909246784 -0800
+++ setiathome-7.08/AP/client/Makefile.am	2015-12-06 17:43:21.731834427 -0800
@@ -92,14 +92,11 @@
 CLEANFILES = $(CLIENT_BIN)
 
 ap_client_cpu_SOURCES = \
-        ../server/db/ap_schema.h			\
-	$(SETI_BOINC_DIR)/db/db_table.h			\
         $(SETI_BOINC_DIR)/db/sqlblob.cpp		\
 	$(SETI_BOINC_DIR)/db/sqlrow.cpp			\
 	$(SETI_BOINC_DIR)/db/xml_util.cpp		\
 	$(SETI_BOINC_DIR)/client/lcgamm.cpp		\
         ../server/db/ap_schema.cpp			\
-        ap_config.h					\
 	ap_client_main.cpp				\
 	ap_science.cpp					\
 	ap_fileio.cpp					\
@@ -116,15 +113,17 @@
 	../../src/GPU_lock.h				\
 	../../src/gcc_includes.h			\
 	../../src/intrinsics.h				\
+	ap_gfx_main.h                                   \
 	astropulse.h					\
 	ap_debug.h					\
 	mtrand.h					\
 	ap_graphics.h					\
 	sbtf.h						\
         ap_version.h                                    \
-	ap_gfx_main.h                                   \
+        ap_config.h					\
+	$(SETI_BOINC_DIR)/db/db_table.h			\
+        ../db/ap_schema.h			        \
 	$(AP_GRX_SOURCES)
-#
 
 if OPENCL
 ap_client_opencl_SOURCES = \
Only in setiathome-7.08/AP/client: ap_7.01r1864_sse3_clGPU_x86_64-pc-linux-gnu
Only in setiathome-7.08/AP/client: ap_client
Only in setiathome-7.08/AP/client: ap_client-CLInfo.o
Only in setiathome-7.08/AP/client: ap_client-GPU_lock.o
Only in setiathome-7.08/AP/client: ap_client-ap_client_main.o
Only in setiathome-7.08/AP/client: ap_client-ap_debug.o
Only in setiathome-7.08/AP/client: ap_client-ap_fileio.o
Only in setiathome-7.08/AP/client: ap_client-ap_fold.o
Only in setiathome-7.08/AP/client: ap_client-ap_remove_radar.o
Only in setiathome-7.08/AP/client: ap_client-ap_schema.o
Only in setiathome-7.08/AP/client: ap_client-ap_science.o
Only in setiathome-7.08/AP/client: ap_client-ap_timer.o
Only in setiathome-7.08/AP/client: ap_client-ap_version.o
Only in setiathome-7.08/AP/client: ap_client-fft_execute.o
Only in setiathome-7.08/AP/client: ap_client-fft_kernelstring.o
Only in setiathome-7.08/AP/client: ap_client-fft_setup.o
Only in setiathome-7.08/AP/client: ap_client-lcgamm.o
Only in setiathome-7.08/AP/client: ap_client-malloc_a.o
Only in setiathome-7.08/AP/client: ap_client-mtrand.o
Only in setiathome-7.08/AP/client: ap_client-sbtf.o
Only in setiathome-7.08/AP/client: ap_client-sqlblob.o
Only in setiathome-7.08/AP/client: ap_client-sqlrow.o
Only in setiathome-7.08/AP/client: ap_client-xml_util.o
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/ap_client_main.cpp setiathome-7.08/AP/client/ap_client_main.cpp
--- setiathome-7.08.orig/AP/client/ap_client_main.cpp	2015-12-06 10:44:41.125255258 -0800
+++ setiathome-7.08/AP/client/ap_client_main.cpp	2015-12-06 17:42:26.155726382 -0800
@@ -74,19 +74,16 @@
 	void CLInfo(void);
 #endif
 #ifdef BOINC_APP_GRAPHICS
-extern REDUCED_ARRAY_GEN rarray;
 // #include "ap_graphics.h"   Not part of the main program any more
-#include "reduce.h"
-#include "ap_gfx_main.h"
+//#include "ap_gfx_main.h"
+#include "ap_gfx_main.cpp"
 #include "graphics2.h"
 #endif
 #if _GNU_SOURCE || __APPLE__ || __MINGW32__ || __FreeBSD__ //uje: gcc
 #include "gcc_includes.h"
 //#include "ap_shmem.h"
-#include "ap_gfx_main.h"
 #include "ap_version.h"
 #else
-#include "ap_gfx_main.h"
 #endif
 
 #ifdef __MINGW32__
@@ -95,6 +92,7 @@
 
 #include <time.h>
 
+
 long starttime = time(0);
 int lastcputicks = clock();
 float cputime = 0;
@@ -563,14 +561,30 @@
     strcpy(app_init_data.project_dir, "./");
 #endif
 }
-#if _GNU_SOURCE || __APPLE__ || __MINGW32__ || __FreeBSD__
-  ap_graphics_init(app_init_data); //uje: Renamed ap_graphics_init(app_init_data) to avoid confusions with stock code
-#else // !_GNU_SOURCE
-  // We've moved the state variable into the graphics shmem segment,
-  // so we always need to initialize graphics.
+#ifdef BOINC_APP_GRAPHICS
   ap_graphics_init(app_init_data);
-#endif
+#else
+   // Checking for blank statefile
+   AP_SHMEM* &ap_shmem=Astropulse::client.ap_shmem;
+    // ap_shmem is defined extern in ap_gfx_main.h XXX
+    ap_shmem = (AP_SHMEM *)(boinc_graphics_make_shmem("astropulse", sizeof(AP_SHMEM)));//XXX
+    if (!ap_shmem) {
+        fprintf(stderr, "boinc_graphics_make_shmem failed: %d\n", errno);
+        exit(ERR_SHMEM_NAME);
+    }
 
+    // initialize ap_shmem with placement new.
+        // VC++ doesn't support placement new in debug mode!?!?!  What the hell?
+#if !defined(_DEBUG) || !defined(_MSC_VER)
+    ap_shmem = new (ap_shmem) AP_SHMEM();
+#else
+    // VC++ doesn't support placement new in debug mode!?!?!  What the hell?
+        // Fortunately AP_SHMEM doesn't contain any self-referential pointers.
+    AP_SHMEM *tmp=new AP_SHMEM();
+        memcpy(ap_shmem,tmp,sizeof(AP_SHMEM));
+        delete tmp;
+#endif
+#endif
 
   /* Possible arguments:
    * -standalone
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/ap_gfx_base.h setiathome-7.08/AP/client/ap_gfx_base.h
--- setiathome-7.08.orig/AP/client/ap_gfx_base.h	2015-12-06 10:44:41.125255258 -0800
+++ setiathome-7.08/AP/client/ap_gfx_base.h	2015-12-06 14:32:06.201635735 -0800
@@ -77,9 +77,9 @@
     APP_INIT_DATA app_init_data;
     double viewpoint_distance;
     int width, height;
-    REDUCED_ARRAY_GEN rarray;
-    const AP_SHMEM* ap_shmem;
-    const AP_GDATA* ap_gdata;
+    REDUCED_ARRAY_RENDER rarray;
+    AP_SHMEM* ap_shmem;
+    AP_GDATA* ap_gdata;
 
     AP_GRAPHICS_BASE();
     void data_struct_init();
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/ap_gfx_main.cpp setiathome-7.08/AP/client/ap_gfx_main.cpp
--- setiathome-7.08.orig/AP/client/ap_gfx_main.cpp	2015-12-06 10:44:41.003250471 -0800
+++ setiathome-7.08/AP/client/ap_gfx_main.cpp	2015-12-06 15:52:23.727229137 -0800
@@ -30,17 +30,20 @@
 #include "astropulse.h"
 #include "ap_gfx_main.h"
 #include "boinc_api.h"
-#if BOINC_APP_GRAPHICS
 #include "graphics2.h"
 
 REDUCED_ARRAY_GEN rarray;
+AP_GDATA* ap_gdata;
+#ifdef BOINC_APP_GRAPHICS
 bool nographics_flag = false;
 #else
 bool nographics_flag = true;
 #endif
-AP_GDATA* ap_gdata;
 
 void update_shmem() {
+    if (ap_gdata->countdown > 0) {
+        ap_gdata->countdown--;
+    }
 }
 
 void ap_graphics_init(APP_INIT_DATA& app_init_data) {
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/ap_gfx_main.h setiathome-7.08/AP/client/ap_gfx_main.h
--- setiathome-7.08.orig/AP/client/ap_gfx_main.h	2015-12-06 10:44:41.003250471 -0800
+++ setiathome-7.08/AP/client/ap_gfx_main.h	2015-12-06 15:11:45.314961464 -0800
@@ -30,7 +30,9 @@
 #include <string>
 #include "db_table.h"
 
+#ifndef _MSC_VER //R:There is no such file in source tree and MS VC++ can't take it from nowhere.
 #include "ap_config.h"
+#endif
 #include "ap_schema.h"
 
 #include "ap_const.h"
@@ -51,11 +53,8 @@
 #include "gdata.h"
 #include "boinc_api.h"
 
-#if BOINC_APP_GRAPHICS
 extern REDUCED_ARRAY_GEN rarray;
-#endif
-extern AP_GDATA* ap_graphics;
+extern AP_GDATA* ap_gdata;
 extern bool nographics_flag;
-
-inline bool nographics() {return nographics_flag;}
+extern bool nographics() { return nographics_flag; }
 extern void ap_graphics_init(APP_INIT_DATA&);
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/ap_graphics.cpp setiathome-7.08/AP/client/ap_graphics.cpp
--- setiathome-7.08.orig/AP/client/ap_graphics.cpp	2015-12-06 10:44:41.004250510 -0800
+++ setiathome-7.08/AP/client/ap_graphics.cpp	2015-12-06 14:34:07.672287643 -0800
@@ -526,6 +526,7 @@
     ap_shmem = (AP_SHMEM*)boinc_graphics_get_shmem("astropulse");
     ap_gdata = &(ap_shmem->ap_gdata);
   }
+  ap_gdata->countdown = 5;
 
   if (last_time != 0) {
     dt = time_of_day - last_time;
Only in setiathome-7.08/AP/client: astropulse-7.01_x86_64-pc-linux-gnu.debug
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/gdata.h setiathome-7.08/AP/client/gdata.h
--- setiathome-7.08.orig/AP/client/gdata.h	2015-12-06 10:44:41.044252081 -0800
+++ setiathome-7.08/AP/client/gdata.h	2015-12-06 14:20:53.357987008 -0800
@@ -35,6 +35,7 @@
 #ifndef _AP_GDATA_
 #define _AP_GDATA_
 
+#include "ap_state.h"
 #include "ap_version.h"
 #include "reduce.h"
 
@@ -63,6 +64,10 @@
 #if BOINC_APP_GRAPHICS
   char status[80];
 #endif
+  volatile int countdown;
+      // the graphics app sets this to 5 every second,
+      // and the main app decrements it every second.
+      // So if it's zero, it means no graphics app is running
 
 };
 
@@ -70,6 +75,7 @@
 inline AP_GDATA::AP_GDATA() :
     version_major(gmajor_version),
     version_minor(gminor_version),
+    countdown(0),
 local_progress(0) {
 #if BOINC_APP_GRAPHICS
 	memset(status,0,sizeof(status));
