--- AP/client/ap_client_main.cpp.orig	2015-12-05 22:39:24.510480049 -0800
+++ AP/client/ap_client_main.cpp	2015-12-05 23:01:27.721873353 -0800
@@ -74,13 +74,16 @@
 	void CLInfo(void);
 #endif
 #ifdef BOINC_APP_GRAPHICS
+extern REDUCED_ARRAY_GEN rarray;
 // #include "ap_graphics.h"   Not part of the main program any more
-// #include "ap_gfx_main.h"
+#include "reduce.h"
+#include "ap_gfx_main.h"
 #include "graphics2.h"
 #endif
 #if _GNU_SOURCE || __APPLE__ || __MINGW32__ || __FreeBSD__ //uje: gcc
 #include "gcc_includes.h"
-#include "ap_shmem.h"
+//#include "ap_shmem.h"
+#include "ap_gfx_main.h"
 #include "ap_version.h"
 #else
 #include "ap_gfx_main.h"
@@ -153,7 +156,7 @@
 	#error "This define bugged still"
 #endif
 #ifdef BOINC_APP_GRAPHICS
-	#error "Graphics build unsupported, look at power[] allocation and over optimizations"
+	//#error "Graphics build unsupported, look at power[] allocation and over optimizations"
 #endif
 
 #if defined(_PGO_INSTRUMENT)
@@ -2468,6 +2471,7 @@
         * samples converted is equal to the length of one FFT.
         * If one of the "random indices" is within 100,000 bytes of our data_chunk_now, we randomize all of the data.
         */
+    double max = 0.0;
 #if !USE_OPENCL
         convert_bits_to_float();
 #if 0
@@ -2518,7 +2522,6 @@
 #endif
         /* Graphics stuff */
   #ifdef BOINC_APP_GRAPHICS
-        double max = 0.0;
 	if(!nographics()) {
           rarray.init_data(state.fft_len/2, state.dm_chunk_small); // const
         }
