diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/Makefile.am setiathome-7.08/AP/client/Makefile.am
--- setiathome-7.08.orig/AP/client/Makefile.am	2015-12-07 14:56:19.962213611 -0800
+++ setiathome-7.08/AP/client/Makefile.am	2015-12-07 15:01:31.773991639 -0800
@@ -64,7 +64,7 @@
      CLIENT_BIN = $(CLIENT_PROG) $(CLIENT_NATIVE) $(DEBUG_NATIVE) $(DEBUG_PROG)
      GUI_BIN = ap_graphics
      GUI_C_FLAGS = $(CLIENT_C_FLAGS) $(GRAPHICS_CFLAGS)
-     GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
+     GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
      CLIENT_LD_FLAGS += $(LDSTATIC)
      CLIENT_LD_ADD += -L$(BOINCDIR)/api -L$(BOINCDIR)/lib $(BOINC_LIBS) $(AM_LIBS)
      AP_GRX_SOURCES = ap_gfx_main.cpp
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/Makefile.incl setiathome-7.08/AP/client/Makefile.incl
--- setiathome-7.08.orig/AP/client/Makefile.incl	2015-12-07 14:56:19.962213611 -0800
+++ setiathome-7.08/AP/client/Makefile.incl	2015-12-07 15:00:30.159663268 -0800
@@ -33,9 +33,9 @@
 APP_LDFLAGS = @ASMLIB_LDFLAGS@ @APP_LDFLAGS@
 APP_CFLAGS = @ASMLIB_CFLAGS@ @APP_CFLAGS@
 
-##GRAPHICS_CFLAGS = @GRAPHICS_CFLAGS@
+GRAPHICS_CFLAGS = @GRAPHICS_CFLAGS@
 DEFS = @DEFS@
-##GRAPHICS_LIBS = @GRAPHICS_LIBS@
+GRAPHICS_LIBS = @GRAPHICS_LIBS@
 SSE_LEVEL=@SSE_LEVEL@
 SVN_REV_NUM = @SVN_REV_NUM@
 OCL_Y = @OCL_Y@
