--- AKv8/client/seti.h.orig	2015-12-03 19:52:59.795196778 -0800
+++ AKv8/client/seti.h	2015-12-03 19:54:27.649626475 -0800
@@ -192,19 +192,19 @@
 extern int_func_t f_glut_is_initialized;
 #endif
 
-inline bool nographics() {
-#ifdef BOINC_APP_GRAPHICS
-#ifdef DYNAMIC_GRAPHICS
-  return (nographics_flag || !sah_graphics || !graphics_lib_handle ||
-      !(f_glut_is_initialized && f_glut_is_initialized()));
-#else
-  return (nographics_flag || !sah_graphics );
-      // || !xwin_glut_is_initialized());    // should add this back somehow
-#endif
-#else
-  return true;
-#endif
-}
+//inline bool nographics() {
+//#ifdef BOINC_APP_GRAPHICS
+//#ifdef DYNAMIC_GRAPHICS
+//  return (nographics_flag || !sah_graphics || !graphics_lib_handle ||
+//      !(f_glut_is_initialized && f_glut_is_initialized()));
+//#else
+//  return (nographics_flag || !sah_graphics );
+//      // || !xwin_glut_is_initialized());    // should add this back somehow
+//#endif
+//#else
+//  return true;
+//#endif
+//}
 
 #ifndef AP_CLIENT
 struct SPIKE_INFO : public track_mem<SPIKE_INFO> {
