diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AP/client/ap_client_main.cpp setiathome-7.08/AP/client/ap_client_main.cpp
--- setiathome-7.08.orig/AP/client/ap_client_main.cpp	2015-12-08 23:28:58.583639745 -0800
+++ setiathome-7.08/AP/client/ap_client_main.cpp	2015-12-08 23:50:41.864689677 -0800
@@ -49,9 +49,13 @@
 #ifdef HAVE_STDINT_H
 #include <stdint.h>
 #endif
+#ifdef HAVE_SYS_STAT_H
+#include <sys/stat.h>
+#endif
 #include "boinc_api.h"
 #include "util.h"
 #include "str_util.h"
+#include "shmem.h"
 
 #include "astropulse.h"
 #include "diagnostics.h"
@@ -78,6 +82,8 @@
 #include "ap_gfx_main.h"
 //#include "ap_gfx_main.cpp"
 #include "graphics2.h"
+#else
+bool nographics_flag;
 #endif
 #if _GNU_SOURCE || __APPLE__ || __MINGW32__ || __FreeBSD__ //uje: gcc
 #include "gcc_includes.h"
@@ -420,6 +426,56 @@
 
 APP_INIT_DATA app_init_data;
 
+#ifndef BOINC_APP_GRAPHICS
+/* copy some code from boinc-client */
+
+#ifdef __EMX__
+static key_t get_shmem_name(const char* prog_name) {
+    char cwd[256], path[256];
+    boinc_getcwd(cwd);
+    sprintf(path, "%s/init_data.xml", cwd);
+    return ftok(path, 2);
+}
+#else
+// Unix/Linux/Mac applications always use mmap() for gfx communication
+//
+static void get_shmem_name(const char* prog_name, char* shmem_name) {
+    APP_INIT_DATA aid;
+    int retval = boinc_get_init_data(aid);
+    if (retval) aid.slot = 0;
+    sprintf(shmem_name, "boinc_%s_%d", prog_name, aid.slot);
+}
+#endif
+
+void* boinc_graphics_make_shmem(const char* prog_name, int size) {
+#ifdef _WIN32
+    HANDLE shmem_handle;
+    char shmem_name[256];
+    void* p;
+    get_shmem_name(prog_name, shmem_name);
+    shmem_handle = create_shmem(shmem_name, size, &p);
+    if (shmem_handle == NULL) return 0;
+    return p;
+#else
+    void* p;
+#ifdef __EMX__
+    key_t key = get_shmem_name(prog_name);
+    int retval = create_shmem(key, size, 0, &p);
+#else
+    // V6 Unix/Linux/Mac applications always use mmap() shared memory for graphics communication
+    char shmem_name[256];
+    get_shmem_name(prog_name, shmem_name);
+    int retval = create_shmem_mmap(shmem_name, size, &p);
+    // Graphics app may be run by a different user & group than worker app
+    // Although create_shmem passed 0666 to open(), it was modified by umask
+    if (retval == 0) chmod(shmem_name, 0666);
+#endif
+    if (retval) return 0;
+    return p;
+#endif
+}
+#endif /* !BOINC_APP_GRAPHICS */
+
 #ifndef min
 #define min(a,b)            (((a) < (b)) ? (a) : (b))
 #endif
Only in setiathome-7.08/AP/client: astropulse-7.01_x86_64-pc-linux-gnu.debug
