--- a/CMakeLists.txt.orig2	2020-04-03 23:00:33.287088025 -0700
+++ b/CMakeLists.txt	2020-04-03 23:10:17.439579475 -0700
@@ -2,6 +2,7 @@ CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 PROJECT(MojoShader)
 
 OPTION(BUILD_SHARED_LIBS "Build MojoShader as a shared library" OFF)
+OPTION(BUILD_STATIC_LIBS "Build MojoShader as a static library" OFF)
 OPTION(PROFILE_D3D "Build MojoShader with support for the D3D profile" ON)
 OPTION(PROFILE_BYTECODE "Build MojoShader with support for the BYTECODE profile" ON)
 OPTION(PROFILE_GLSL120 "Build MojoShader with support for the GLSL120 profile" ON)
@@ -154,7 +155,7 @@ IF(XNA4_VERTEXTEXTURE)
     ADD_DEFINITIONS(-DMOJOSHADER_XNA4_VERTEX_TEXTURES)
 ENDIF(XNA4_VERTEXTEXTURE)
 
-ADD_LIBRARY(mojoshader
+SET(mojoshader-libsrc
     mojoshader.c
     mojoshader_common.c
     mojoshader_opengl.c
@@ -167,23 +168,53 @@ ADD_LIBRARY(mojoshader
     profiles/mojoshader_profile_spirv.c
     profiles/mojoshader_profile_common.c
 )
-IF(EFFECT_SUPPORT)
-    TARGET_SOURCES(mojoshader PRIVATE
-        mojoshader_effects.c
-    )
-ENDIF(EFFECT_SUPPORT)
-IF(COMPILER_SUPPORT)
-    TARGET_SOURCES(mojoshader PRIVATE
-        mojoshader_compiler.c
-        mojoshader_preprocessor.c
-        mojoshader_lexer.c
-        mojoshader_assembler.c
-    )
-ENDIF(COMPILER_SUPPORT)
+
+ADD_LIBRARY(mojoshader-libobj OBJECT ${mojoshader-libsrc})
+SET_PROPERTY(TARGET mojoshader-libobj PROPERTY POSITION_INDEPENDENT_CODE 1)
+
+IF(BUILD_SHARED_LIBS)
+    ADD_LIBRARY(mojoshader SHARED $<TARGET_OBJECTS:mojoshader-libobj>)
+    IF(EFFECT_SUPPORT)
+        TARGET_SOURCES(mojoshader PRIVATE
+            mojoshader_effects.c
+        )
+    ENDIF(EFFECT_SUPPORT)
+    IF(COMPILER_SUPPORT)
+        TARGET_SOURCES(mojoshader PRIVATE
+            mojoshader_compiler.c
+            mojoshader_preprocessor.c
+            mojoshader_lexer.c
+            mojoshader_assembler.c
+        )
+    ENDIF(COMPILER_SUPPORT)
+ENDIF(BUILD_SHARED_LIBS)
+
+IF(BUILD_STATIC_LIBS)
+    ADD_LIBRARY(mojoshader-static STATIC $<TARGET_OBJECTS:mojoshader-libobj>)
+    IF(EFFECT_SUPPORT)
+        TARGET_SOURCES(mojoshader-static PRIVATE
+            mojoshader_effects.c
+        )
+    ENDIF(EFFECT_SUPPORT)
+    IF(COMPILER_SUPPORT)
+        TARGET_SOURCES(mojoshader-static PRIVATE
+            mojoshader_compiler.c
+            mojoshader_preprocessor.c
+            mojoshader_lexer.c
+            mojoshader_assembler.c
+        )
+    ENDIF(COMPILER_SUPPORT)
+ENDIF(BUILD_STATIC_LIBS)
+
 IF(BUILD_SHARED_LIBS)
     TARGET_LINK_LIBRARIES(mojoshader ${LIBM} ${LOBJC} ${CARBON_FRAMEWORK})
     INSTALL(TARGETS mojoshader LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 ENDIF(BUILD_SHARED_LIBS)
+IF(BUILD_STATIC_LIBS)
+    TARGET_LINK_LIBRARIES(mojoshader-static ${LIBM} ${LOBJC} ${CARBON_FRAMEWORK})
+    INSTALL(TARGETS mojoshader-static LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    SET_TARGET_PROPERTIES(mojoshader-static PROPERTIES OUTPUT_NAME "mojoshader")
+ENDIF(BUILD_STATIC_LIBS)
 
 SET_SOURCE_FILES_PROPERTIES(
     mojoshader_compiler.c
