--- a/build/configure.ac.pthread.orig	2022-08-25 07:30:29.150520798 -0700
+++ b/build/configure.ac.pthread	2022-08-25 07:37:19.434686856 -0700
@@ -138,8 +138,8 @@ dnl   <program> with the given flags and
 dnl   success and <false-action> on failure.
 AC_DEFUN([CAIRO_CHECK_PTHREAD],[dnl
 	CAIRO_CC_TRY_LINK_WITH_ENV_SILENT(
-		[CFLAGS="$CFLAGS $2";
-		 LIBS="$LIBS $3"],
+		[CFLAGS="$2";
+		 LIBS="$3"],
 		[$4],
 		[$1_CFLAGS="$2";
 		 $1_LIBS="$3";
