--- a/build/configure.ac.pthread.orig	2018-08-16 18:10:53.000000000 -0700
+++ b/build/configure.ac.pthread	2022-08-25 06:52:33.748995629 -0700
@@ -82,7 +82,10 @@ dnl ------------------------------------
 dnl A program to test all the features we want to be able to run the test
 dnl suite or other thready cairo applications that want real threads.
 m4_define([testsuite_pthread_program],[dnl
-libcairo_pthread_program
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE /* for PTHREAD_MUTEX_INITIALIZER under linux */
+#endif
+#include <pthread.h>
 
 pthread_once_t once_control = PTHREAD_ONCE_INIT;
 void test_once_init (void) {}
