diff -urp cairo-1.16.0.orig/src/cairo-xlib-xrender-private.h cairo-1.16.0/src/cairo-xlib-xrender-private.h
--- cairo-1.16.0.orig/src/cairo-xlib-xrender-private.h	2018-08-16 18:10:53.000000000 -0700
+++ cairo-1.16.0/src/cairo-xlib-xrender-private.h	2021-12-23 12:32:34.689026613 -0800
@@ -103,33 +103,41 @@ __attribute__((__unused__)) static void
 #if !HAVE_XRENDERCREATELINEARGRADIENT
 #define XRenderCreateLinearGradient			_int_consume
 
+#  ifndef _XRENDER_H_
 typedef struct _XLinearGradient {
     XPointFixed p1;
     XPointFixed p2;
 } XLinearGradient;
+#  endif
 #endif
 
 #if !HAVE_XRENDERCREATERADIALGRADIENT
 #define XRenderCreateRadialGradient			_int_consume
 
+#  ifndef _XRENDER_H_
 typedef struct _XCircle {
     XFixed x;
     XFixed y;
     XFixed radius;
 } XCircle;
+#  endif
+#  ifndef _XRENDER_H_
 typedef struct _XRadialGradient {
     XCircle inner;
     XCircle outer;
 } XRadialGradient;
+#  endif
 #endif
 
 #if !HAVE_XRENDERCREATECONICALGRADIENT
 #define XRenderCreateConicalGradient			_int_consume
 
+#  ifndef _XRENDER_H_
 typedef struct _XConicalGradient {
     XPointFixed center;
     XFixed angle; /* in degrees */
 } XConicalGradient;
+#  endif
 #endif
 
 
