diff -urp fluid-1.2.0_p9999.orig/qbs/shared/modules/WaylandScanner/WaylandScanner.qbs fluid-1.2.0_p9999/qbs/shared/modules/WaylandScanner/WaylandScanner.qbs
--- fluid-1.2.0_p9999.orig/qbs/shared/modules/WaylandScanner/WaylandScanner.qbs	2022-11-30 10:02:51.750801723 -0800
+++ fluid-1.2.0_p9999/qbs/shared/modules/WaylandScanner/WaylandScanner.qbs	2022-11-30 10:03:23.840060231 -0800
@@ -11,11 +11,11 @@ Module {
         inputs: ["wayland.client.protocol"]
 
         Artifact {
-            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-client-protocol.h"
+            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-client-protocol.h"
             fileTags: ["hpp", "hpp_private"]
         }
         Artifact {
-            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-protocol.c"
+            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-protocol.c"
             fileTags: ["c"]
         }
 
@@ -28,11 +28,11 @@ Module {
         inputs: ["wayland.client.protocol"]
 
         Artifact {
-            filePath: "qwayland-" + FileInfo.baseName(input.fileName) + ".h"
+            filePath: "qwayland-" + FileInfo.baseName(input.fileName) + ".h"
             fileTags: ["hpp", "hpp_private"]
         }
         Artifact {
-            filePath: "qwayland-" + FileInfo.baseName(input.fileName) + ".cpp"
+            filePath: "qwayland-" + FileInfo.baseName(input.fileName) + ".cpp"
             fileTags: ["cpp"]
         }
 
@@ -45,11 +45,11 @@ Module {
         inputs: ["wayland.server.protocol"]
 
         Artifact {
-            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-server-protocol.h"
+            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-server-protocol.h"
             fileTags: ["hpp", "hpp_private"]
         }
         Artifact {
-            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-protocol.c"
+            filePath: "wayland-" + FileInfo.baseName(input.fileName) + "-protocol.c"
             fileTags: ["c"]
         }
 
@@ -62,11 +62,11 @@ Module {
         inputs: ["wayland.server.protocol"]
 
         Artifact {
-            filePath: "qwayland-server-" + FileInfo.baseName(input.fileName) + ".h"
+            filePath: "qwayland-server-" + FileInfo.baseName(input.fileName) + ".h"
             fileTags: ["hpp", "hpp_private"]
         }
         Artifact {
-            filePath: "qwayland-server-" + FileInfo.baseName(input.fileName) + ".cpp"
+            filePath: "qwayland-server-" + FileInfo.baseName(input.fileName) + ".cpp"
             fileTags: ["cpp"]
         }
 
diff -urp fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.cpp fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.cpp
--- fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.cpp	2022-11-30 10:02:50.830765640 -0800
+++ fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.cpp	2022-11-30 10:03:23.840060231 -0800
@@ -61,7 +61,7 @@ FluidDecoration *FluidDecorationManager:
 
 FluidDecoration::FluidDecoration(FluidDecorationManager *parent, QWindow *window, struct ::zfluid_decoration_v1 *object)
     : QWaylandClientExtensionTemplate<FluidDecoration>(1)
-    , QtWayland::zfluid_decoration_v1(object)
+    , Aurora::Client::PrivateClient::zfluid_decoration_v1(object)
     , m_manager(parent)
     , m_window(window)
 {
diff -urp fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.h fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.h
--- fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.h	2022-11-30 10:02:50.830765640 -0800
+++ fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.h	2022-11-30 10:04:43.239173957 -0800
@@ -11,12 +11,12 @@
 
 #include <wayland-client.h>
 
-#include "qwayland-fluid-decoration-unstable-v1.h"
+#include "qwayland-fluid-decoration-unstable-v1.h"
 
 class FluidDecoration;
 
 class FluidDecorationManager : public QWaylandClientExtensionTemplate<FluidDecorationManager>
-        , public QtWayland::zfluid_decoration_manager_v1
+        , public Aurora::Client::PrivateClient::zfluid_decoration_manager_v1
 {
     Q_OBJECT
 public:
@@ -37,7 +37,7 @@ private:
 };
 
 class FluidDecoration : public QWaylandClientExtensionTemplate<FluidDecoration>
-        , public QtWayland::zfluid_decoration_v1
+        , public Aurora::Client::PrivateClient::zfluid_decoration_v1
 {
     Q_OBJECT
 public:
