diff -urp fluid-1.2.0_p9999.orig/src/imports/controls-private/CMakeLists.txt fluid-1.2.0_p9999/src/imports/controls-private/CMakeLists.txt
--- fluid-1.2.0_p9999.orig/src/imports/controls-private/CMakeLists.txt	2022-11-30 16:03:24.818826038 -0800
+++ fluid-1.2.0_p9999/src/imports/controls-private/CMakeLists.txt	2022-11-30 16:04:19.916968817 -0800
@@ -4,12 +4,11 @@ if(LINUX AND NOT ANDROID)
         extensions/fluiddecorationv1.h
     )
 
-    find_package(QtWaylandScanner REQUIRED)
+    if(NOT TARGET Liri::AuroraClient)
+        find_package(Liri1AuroraClient REQUIRED)
+    endif()
     find_package(Wayland 1.15 COMPONENTS Client REQUIRED)
     find_package(Qt5WaylandClient REQUIRED)
-    ecm_add_qtwayland_client_protocol(SOURCES
-        PROTOCOL "${CMAKE_CURRENT_SOURCE_DIR}/extensions/fluid-decoration-unstable-v1.xml"
-        BASENAME "fluid-decoration-unstable-v1")
     set(DEFINES FLUID_ENABLE_WAYLAND=1)
     set(LIBRARIES Qt5::GuiPrivate Qt5::WaylandClient Wayland::Client)
 endif()
@@ -37,6 +36,14 @@ liri_add_qml_plugin(fluidcontrolsprivate
         Qt5::QuickControls2
     LIBRARIES
         ${LIBRARIES}
+        Liri::AuroraClient
 )
 
+if(LINUX AND NOT ANDROID)
+    liri_generate_wayland_protocol_client_sources(fluidcontrolsprivate
+        FILES
+            "${CMAKE_CURRENT_SOURCE_DIR}/extensions/fluid-decoration-unstable-v1.xml"
+    )
+endif()
+
 liri_finalize_qml_plugin(fluidcontrolsprivate)
diff -urp fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.cpp fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.cpp
--- fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.cpp	2022-11-30 16:03:24.818826038 -0800
+++ fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.cpp	2022-11-30 16:03:47.471706969 -0800
@@ -61,7 +61,7 @@ FluidDecoration *FluidDecorationManager:
 
 FluidDecoration::FluidDecoration(FluidDecorationManager *parent, QWindow *window, struct ::zfluid_decoration_v1 *object)
     : QWaylandClientExtensionTemplate<FluidDecoration>(1)
-    , QtWayland::zfluid_decoration_v1(object)
+    , Aurora::Client::PrivateClient::zfluid_decoration_v1(object)
     , m_manager(parent)
     , m_window(window)
 {
diff -urp fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.h fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.h
--- fluid-1.2.0_p9999.orig/src/imports/controls-private/extensions/fluiddecorationv1.h	2022-11-30 16:03:24.818826038 -0800
+++ fluid-1.2.0_p9999/src/imports/controls-private/extensions/fluiddecorationv1.h	2022-11-30 16:03:47.471706969 -0800
@@ -11,12 +11,12 @@
 
 #include <wayland-client.h>
 
-#include "qwayland-fluid-decoration-unstable-v1.h"
+#include "aurora-client-fluid-decoration-unstable-v1.h"
 
 class FluidDecoration;
 
 class FluidDecorationManager : public QWaylandClientExtensionTemplate<FluidDecorationManager>
-        , public QtWayland::zfluid_decoration_manager_v1
+        , public Aurora::Client::PrivateClient::zfluid_decoration_manager_v1
 {
     Q_OBJECT
 public:
@@ -37,7 +37,7 @@ private:
 };
 
 class FluidDecoration : public QWaylandClientExtensionTemplate<FluidDecoration>
-        , public QtWayland::zfluid_decoration_v1
+        , public Aurora::Client::PrivateClient::zfluid_decoration_v1
 {
     Q_OBJECT
 public:
