--- a/scripts/fetch_icons.sh	2022-12-06 20:12:55.694586770 -0800
+++ b/scripts/fetch_icons.sh	2022-12-06 20:19:26.066259600 -0800
@@ -12,7 +12,11 @@ TXT_FILE=${rootdir}/src/demo/qml/icons.t
 function copy_icon()
 {
     for NAME in $ICONS; do
-        source_path=$GIT_DIR/src/$CATEGORY/$NAME/materialicons/24px.svg
+        if [[ -e $GIT_DIR/src/$CATEGORY/$NAME/materialicons/24px.svg ]] ; then
+            source_path=$GIT_DIR/src/$CATEGORY/$NAME/materialicons/24px.svg
+        else
+	     source_path=$GIT_DIR/src/$CATEGORY/$NAME/materialiconstwotone/24px.svg
+        fi
         [ ! -f $source_path ] && continue
         dest_path=$TARGET_DIR/$CATEGORY/$NAME.svg
         if [ ! -f $dest_path ]; then
@@ -30,7 +34,7 @@ else
     :; #git clone --depth 1 $GIT_URL
 fi
 
-CATEGORIES=$(ls -1 -d  $GIT_DIR/src/*/*/materialicons/*.svg | awk -F/ '{ print $3 }' | uniq)
+CATEGORIES=$(ls -1 -d  $GIT_DIR/src/*/*/{materialicons,materialiconstwotone}/*.svg | awk -F/ '{ print $3 }' | uniq)
 
 rm -rf $TARGET_DIR
 mkdir -p $TARGET_DIR
