--- a/dom/indexedDB/moz.build.orig	2022-10-25 09:20:10.000000000 -0700
+++ b/dom/indexedDB/moz.build	2022-11-06 19:36:02.783717171 -0800
@@ -123,3 +123,15 @@ XPIDL_SOURCES += [
 ]
 
 XPIDL_MODULE = "dom_indexeddb"
+
+if CONFIG["CC_TYPE"] in ("clang"):
+    CXXFLAGS += [
+        # Placeholder for -fno-tree-loop-vectorize below
+        "-O3"
+    ]
+
+if CONFIG["CC_TYPE"] in ("gcc"):
+    CXXFLAGS += [
+        # Testing
+        "-fno-tree-loop-vectorize"
+    ]
