--- a/build/moz.configure/init.configure.orig	2020-01-17 11:25:13.000000000 -0800
+++ b/build/moz.configure/init.configure	2020-02-05 22:41:21.646480074 -0800
@@ -875,7 +875,7 @@ target = help_host_target | real_target
 @depends(host, target)
 @checking('whether cross compiling')
 def cross_compiling(host, target):
-    return host != target
+    return True
 
 
 set_config('CROSS_COMPILE', cross_compiling)
