--- a/LibGit2Sharp/LibGit2Sharp.csproj.orig	2019-08-12 21:18:36.163948439 -0700
+++ b/LibGit2Sharp/LibGit2Sharp.csproj	2019-08-12 21:39:16.278732236 -0700
@@ -1,5 +1,9 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
+  <ItemGroup>
+    <EmbeddedResource Include="libgit2_filename.txt" />
+  </ItemGroup>
+
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;net46</TargetFrameworks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
