diff -urp flang-rocm-5.3.3.orig/CMakeLists.txt flang-rocm-5.3.3/CMakeLists.txt
--- flang-rocm-5.3.3.orig/CMakeLists.txt	2022-08-02 10:12:15.000000000 -0700
+++ flang-rocm-5.3.3/CMakeLists.txt	2023-11-11 20:58:49.060649013 -0800
@@ -67,8 +67,8 @@ set(CMAKE_HOST_SYSTEM_PROCESSOR ${TARGET
 # standalone project, using LLVM as an external library:
 if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
   project(Flang)
-  # Set default libdir to be "lib" for ROCm, distros will override this anyway:
-  set(CMAKE_INSTALL_LIBDIR "lib" CACHE STRING "Library install directory")
+  # Set default libdir to be "@LIBDIR@" for ROCm, distros will override this anyway:
+  set(CMAKE_INSTALL_LIBDIR "@LIBDIR@" CACHE STRING "Library install directory")
   include(GNUInstallDirs)
   # Rely on llvm-config.
   set(CONFIG_OUTPUT)
@@ -119,12 +119,18 @@ if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURR
   endif()
 
   set(LLVM_TOOLS_BINARY_DIR ${TOOLS_BINARY_DIR} CACHE PATH "Path to llvm/bin")
-  set(LLVM_LIBRARY_DIR ${LIBRARY_DIR} CACHE PATH "Path to llvm/lib")
+  set(LLVM_LIBRARY_DIR ${LIBRARY_DIR} CACHE PATH "Path to llvm/@LIBDIR@")
   set(LLVM_MAIN_INCLUDE_DIR ${INCLUDE_DIR} CACHE PATH "Path to llvm/include")
   set(LLVM_BINARY_DIR ${LLVM_OBJ_ROOT} CACHE PATH "Path to LLVM build tree")
   set(LLVM_MAIN_SRC_DIR ${MAIN_SRC_DIR} CACHE PATH "Path to LLVM source tree")
 
-  set(LLVM_CMAKE_PATH "${LLVM_BINARY_DIR}/lib/cmake/llvm"
+  if(USE_AAOC)
+    set(_LLVM_CMAKE_PATH "${LLVM_BINARY_DIR}/lib/cmake/llvm")
+  else()
+    set(_LLVM_CMAKE_PATH "${LLVM_BINARY_DIR}/@LIBDIR@/cmake/llvm")
+  endif()
+
+  set(LLVM_CMAKE_PATH "${_LLVM_CMAKE_PATH}"
      CACHE PATH "Path to LLVM cmake modules")
   set(LLVMCONFIG_FILE "${LLVM_CMAKE_PATH}/LLVMConfig.cmake")
   if(EXISTS ${LLVMCONFIG_FILE})
@@ -144,7 +150,7 @@ if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURR
 
   # They are used as destination of target generators.
   set(LLVM_RUNTIME_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/bin)
-  set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/lib)
+  set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/@LIBDIR@)
 
   option(LLVM_INSTALL_TOOLCHAIN_ONLY
     "Only include toolchain files in the 'install' target." OFF)
@@ -168,8 +174,8 @@ if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURR
   link_directories("${LLVM_LIBRARY_DIR}")
 
   set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
-  set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
-  set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+  set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/@LIBDIR@ )
+  set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/@LIBDIR@ )
   if(LLVM_INCLUDE_TESTS)
     set(Python_ADDITIONAL_VERSIONS 2.7)
     include(FindPythonInterp)
@@ -407,8 +413,8 @@ option(FLANG_INCLUDE_TESTS
 get_property(FLANG_TABLEGEN_TARGETS GLOBAL PROPERTY FLANG_TABLEGEN_TARGETS)
 list(APPEND LLVM_COMMON_DEPENDS ${FLANG_TABLEGEN_TARGETS})
 
-set(FLANG_LIB_DIR ${CMAKE_CURRENT_BINARY_DIR}/lib)
-set(FLANG_RTE_LIB_DIR ${CMAKE_CURRENT_BINARY_DIR}/lib)
+set(FLANG_LIB_DIR ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
+set(FLANG_RTE_LIB_DIR ${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@)
 add_definitions( -DPGFLANG )
 
 if (LLVM_INSTALL_RUNTIME)
Only in flang-rocm-5.3.3: CMakeLists.txt.orig
Only in flang-rocm-5.3.3: CMakeLists.txt.rej
diff -urp flang-rocm-5.3.3.orig/runtime/flang/CMakeLists.txt flang-rocm-5.3.3/runtime/flang/CMakeLists.txt
--- flang-rocm-5.3.3.orig/runtime/flang/CMakeLists.txt	2023-11-11 20:56:12.832650810 -0800
+++ flang-rocm-5.3.3/runtime/flang/CMakeLists.txt	2023-11-11 20:57:52.308649666 -0800
@@ -558,7 +558,7 @@ set_property(TARGET flang_shared PROPERT
 #
 add_dependencies(flang_shared flang_static)
 
-target_link_libraries(flang_shared ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/lib/libflangrti.so)
+target_link_libraries(flang_shared ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/@LIBDIR@/libflangrti.so)
 # Resolve symbols against libm and librt
 target_link_libraries(flang_shared m rt)
 
Only in flang-rocm-5.3.3/runtime/flang: CMakeLists.txt.orig
diff -urp flang-rocm-5.3.3.orig/runtime/flangrti/CMakeLists.txt flang-rocm-5.3.3/runtime/flangrti/CMakeLists.txt
--- flang-rocm-5.3.3.orig/runtime/flangrti/CMakeLists.txt	2022-08-02 10:12:15.000000000 -0700
+++ flang-rocm-5.3.3/runtime/flangrti/CMakeLists.txt	2023-11-11 20:57:52.308649666 -0800
@@ -95,7 +95,7 @@ if (NOT DEFINED LIBOMP_EXPORT_DIR)
     FLANG_LIBOMP
     libomp.so
     HINTS
-    ${CMAKE_BINARY_DIR}/lib
+    ${CMAKE_BINARY_DIR}/@LIBDIR@
     ${OPENMP_BUILD_DIR}
     NO_DEFAULT_PATH)
     message (STATUS "FLANGRTI ${FLANG_LIBOMP}")
@@ -105,7 +105,7 @@ endif()
 find_library( 
   LIBPGMATH
   libpgmath.so
-  HINTS ${CMAKE_BINARY_DIR}/lib)
+  HINTS ${CMAKE_BINARY_DIR}/@LIBDIR@)
 target_link_libraries(flangrti_shared ${LIBPGMATH} -Wl,-rpath,\$ORIGIN)
 target_link_libraries(flangrti_shared ${LIBQUADMATH_LOC})
 
Only in flang-rocm-5.3.3/runtime/flangrti: CMakeLists.txt.orig
diff -urp flang-rocm-5.3.3.orig/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake flang-rocm-5.3.3/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake
--- flang-rocm-5.3.3.orig/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake	2022-08-02 10:12:15.000000000 -0700
+++ flang-rocm-5.3.3/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake	2023-11-11 20:57:52.308649666 -0800
@@ -49,7 +49,7 @@ macro(find_llvm_parts)
       string(STRIP "${CONFIG_OUTPUT}" LLVM_CMAKE_PATH)
     else()
       set(LLVM_CMAKE_PATH
-          "${LLVM_BINARY_DIR}/lib${LLVM_LIBDIR_SUFFIX}/cmake/llvm")
+          "${LLVM_BINARY_DIR}/@LIBDIR@/cmake/llvm")
     endif()
   else()
     set(LLVM_FOUND OFF)
diff -urp flang-rocm-5.3.3.orig/runtime/libpgmath/CMakeLists.txt flang-rocm-5.3.3/runtime/libpgmath/CMakeLists.txt
--- flang-rocm-5.3.3.orig/runtime/libpgmath/CMakeLists.txt	2022-08-02 10:12:15.000000000 -0700
+++ flang-rocm-5.3.3/runtime/libpgmath/CMakeLists.txt	2023-11-11 20:59:00.836648878 -0800
@@ -20,8 +20,8 @@ if(POLICY CMP0022)
   cmake_policy(SET CMP0022 NEW) # Required when interacting with LLVM and Clang
 endif()
 
-# Set default libdir to be "lib" for ROCm, distros will override this anyway:
-set(CMAKE_INSTALL_LIBDIR "lib" CACHE STRING "Library install directory")
+# Set default libdir to be "@LIBDIR@" for ROCm, distros will override this anyway:
+set(CMAKE_INSTALL_LIBDIR "@LIBDIR@" CACHE STRING "Library install directory")
 include(GNUInstallDirs)
 
 # Add path for custom modules
@@ -111,7 +111,7 @@ set(LIBPGMATH_BASE_DIR ${CMAKE_CURRENT_S
 set(LIBPGMATH_SRC_DIR ${LIBPGMATH_BASE_DIR}/lib)
 set(LIBPGMATH_TOOLS_DIR ${LIBPGMATH_BASE_DIR}/tools)
 set(LIBPGMATH_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
-set(LIBPGMATH_RUNTIME_PATH ${CMAKE_BINARY_DIR}/lib)
+set(LIBPGMATH_RUNTIME_PATH ${CMAKE_BINARY_DIR}/@LIBDIR@)
 set(LIBPGMATH_LIBRARY_NAME pgmath)
 set(LIBPGMATH_RTL ${CMAKE_SHARED_LIBRARY_PREFIX}${LIBPGMATH_LIBRARY_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
Only in flang-rocm-5.3.3/runtime/libpgmath: CMakeLists.txt.orig
Only in flang-rocm-5.3.3/runtime/libpgmath/lib: CMakeLists.txt.orig
Only in flang-rocm-5.3.3/runtime/libpgmath/lib: CMakeLists.txt.rej
