diff -urp zeromq-4.3.4.orig/CMakeLists.txt zeromq-4.3.4/CMakeLists.txt
--- zeromq-4.3.4.orig/CMakeLists.txt	2021-01-17 05:18:16.000000000 -0800
+++ zeromq-4.3.4/CMakeLists.txt	2021-02-15 23:27:50.651340085 -0800
@@ -1586,6 +1586,19 @@ if(BUILD_SHARED)
   endif()
 endif()
 
+if(ENABLE_CURVE)
+  option(WITH_CURVE_KEYGEN "Build the curve_keygen tool" OFF)
+  if(WITH_CURVE_KEYGEN)
+    add_executable(curve_keygen tools/curve_keygen.cpp)
+    if(BUILD_SHARED)
+      target_link_libraries(curve_keygen libzmq)
+    else()
+      target_link_libraries(curve_keygen libzmq-static)
+    endif()
+    install(TARGETS curve_keygen RUNTIME DESTINATION bin COMPONENT CurveTools)
+  endif()
+endif()
+
 # -----------------------------------------------------------------------------
 # tests
 
@@ -1752,6 +1765,7 @@ if(MSVC AND ENABLE_CPACK)
 
   cpack_add_component_group(Development DISPLAY_NAME "ZeroMQ software development kit" EXPANDED)
   cpack_add_component(PerfTools DISPLAY_NAME "ZeroMQ performance tools" INSTALL_TYPES FullInstall DevInstall)
+  cpack_add_component(CurveTools DISPLAY_NAME "ZeroMQ CURVE tools" INSTALL_TYPES FullInstall DevInstall)
   cpack_add_component(SourceCode DISPLAY_NAME "ZeroMQ source code" DISABLED INSTALL_TYPES FullInstall)
   cpack_add_component(
     SDK
