--- a/script/icon-gen.mts.orig	2024-11-01 04:00:47.000000000 -0700
+++ b/script/icon-gen.mts	2025-02-28 17:08:53.247057609 -0800
@@ -1,8 +1,6 @@
 // eslint-disable-line unicorn/filename-case
 import sharp from 'sharp';
-import icongenRaw from 'icon-gen';
-
-const icongen = icongenRaw as unknown as typeof icongenRaw['default'];
+import icongen from 'icon-gen';
 
 const svg2png = (from: string, to: string, width: number, height: number) => sharp(from)
   .png()
