diff -urp naudio-1.8.orig/NAudio/NAudio.csproj naudio-1.8/NAudio/NAudio.csproj
--- naudio-1.8.orig/NAudio/NAudio.csproj	2017-01-03 18:43:59.945371712 -0800
+++ naudio-1.8/NAudio/NAudio.csproj	2017-01-03 18:46:09.863297490 -0800
@@ -189,7 +189,6 @@
     <Compile Include="Wave\SampleProviders\SampleToWaveProvider24.cs" />
     <Compile Include="Wave\SampleProviders\SignalGenerator.cs" />
     <Compile Include="Dsp\SmbPitchShifter.cs" />
-    <Compile Include="Wave\SampleProviders\SmbPitchShiftingSampleProvider.cs" />
     <Compile Include="Wave\SampleProviders\StereoToMonoSampleProvider.cs" />
     <Compile Include="Wave\SampleProviders\WaveToSampleProvider64.cs" />
     <Compile Include="Wave\SampleProviders\WdlResamplingSampleProvider.cs" />
@@ -265,9 +264,6 @@
     <Compile Include="Gui\Pot.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="Gui\Pot.designer.cs">
-      <DependentUpon>Pot.cs</DependentUpon>
-    </Compile>
     <Compile Include="Gui\VolumeMeter.cs">
       <SubType>Component</SubType>
     </Compile>
@@ -338,9 +334,6 @@
     <Compile Include="Utils\ProgressLog.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="Utils\ProgressLog.designer.cs">
-      <DependentUpon>ProgressLog.cs</DependentUpon>
-    </Compile>
     <Compile Include="CoreAudioApi\AudioEndpointVolume.cs" />
     <Compile Include="CoreAudioApi\AudioMeterInformation.cs" />
     <Compile Include="CoreAudioApi\DeviceState.cs" />
@@ -366,10 +359,6 @@
     <Compile Include="CoreAudioApi\MMDeviceEnumerator.cs" />
     <Compile Include="CoreAudioApi\PropertyStore.cs" />
     <Compile Include="CoreAudioApi\PropertyStoreProperty.cs" />
-    <Compile Include="Wave\Asio\AsioDriver.cs" />
-    <Compile Include="Wave\Asio\AsioDriverExt.cs" />
-    <Compile Include="Wave\Asio\AsioSampleConvertor.cs" />
-    <Compile Include="Wave\Asio\AsioStructures.cs" />
     <Compile Include="Wave\Compression\AcmDriver.cs" />
     <Compile Include="Wave\Compression\AcmDriverDetails.cs" />
     <Compile Include="Wave\Compression\AcmDriverDetailsSupportFlags.cs" />
@@ -466,7 +455,6 @@
     <Compile Include="Wave\WaveProviders\MonoToStereoProvider16.cs" />
     <Compile Include="Wave\WaveProviders\VolumeWaveProvider16.cs" />
     <Compile Include="Wave\WaveProviders\WaveFloatTo16Provider.cs" />
-    <Compile Include="Wave\WaveProviders\Wave16ToFloatProvider.cs" />
     <Compile Include="Wave\WaveProviders\WaveInProvider.cs" />
     <Compile Include="Wave\WaveProviders\WaveProvider16.cs" />
     <Compile Include="Wave\WaveProviders\WaveProvider32.cs" />
@@ -562,11 +550,11 @@
   </ItemGroup>
   <ItemGroup />
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+       Other similar extension points exist, see Microsoft.Common.targets.
+  <Target Name="BeforeBuild">
+  </Target>
+  <Target Name="AfterBuild">
+  </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
