--- a/service/xpra.orig	2019-03-07 23:46:38.000000000 -0800
+++ b/service/xpra	2019-03-25 14:11:10.241474256 -0700
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/sbin/openrc-run
 ### BEGIN INIT INFO
 # Provides:          xpra
 # Required-Start:    $local_fs $network $named $time $syslog
@@ -22,7 +22,26 @@ fi
 [ -e /etc/sysconfig/xpra ] && . /etc/sysconfig/xpra
 [ -e /etc/default/xpra ] && . /etc/default/xpra
 
+depend()
+{
+  need localmount
+}
+
+create_tmpfs() {
+  local user="$1"
+  local path="$2"
+  mkdir -p $path
+  chmod 755 $path
+  chown -R $user:xpra $path
+}
+
 start() {
+  for user in $(cat /etc/group | grep xpra | cut -f4 -d':' | tr ',' ' ') ; do
+    uid=$(cat /etc/passwd | grep $u | cut -f 3 -d':')
+    create_tmpfs $user /run/user/$uid/xpra/
+  done
+  create_tmpfs root /run/xpra/
+
   echo 'Starting service…' >&2
   xpra proxy $VIRTUAL_DISPLAY \
 	--bind=/run/xpra/system --auth=$AUTH \
@@ -47,18 +62,3 @@ stop() {
   	echo 'Service stopped' >&2
   fi
 }
-
-case "$1" in
-  start)
-    start
-    ;;
-  stop)
-    stop
-    ;;
-  restart)
-    stop
-    start
-    ;;
-  *)
-    echo "Usage: $0 {start|stop|restart}"
-esac
