--- a/xpra/os_util.py.orig	2021-11-14 23:11:22.765209487 -0800
+++ b/xpra/os_util.py	2021-11-14 23:12:29.684205140 -0800
@@ -785,14 +785,7 @@ def find_lib_ldconfig(libname):
         if os.path.exists(t):
             ldconfig = t
             break
-    import subprocess
-    p = subprocess.Popen("%s -p" % ldconfig, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
-    data = bytestostr(p.communicate()[0])
-
-    libpath = re.search(pattern, data, re.MULTILINE)        #@UndefinedVariable
-    if libpath:
-        libpath = libpath.group(1)
-    return libpath
+    return ""
 
 def find_lib(libname):
     #it would be better to rely on dlopen to find the paths
