diff -urp xpra-3.0.9.orig/xpra/scripts/config.py xpra-3.0.9/xpra/scripts/config.py
--- xpra-3.0.9.orig/xpra/scripts/config.py	2020-03-21 06:16:55.000000000 -0700
+++ xpra-3.0.9/xpra/scripts/config.py	2020-05-13 16:00:03.957585623 -0700
@@ -1215,12 +1215,31 @@ def validate_config(d={}, discard=NO_FIL
         nd[k] = v
     return nd
 
+def get_speaker_override(d={}):
+    from xpra.util import envbool
+    speaker = "on"
+    try:
+        speaker = os.environ.get("XPRA_SPEAKER")
+        if speaker is None:
+            speaker = d["speaker"]
+        else:
+            if speaker == "1" or speaker == "on":
+                speaker = "on"
+            elif speaker == "disabled":
+                speaker = "disabled"
+            else:
+                speaker = "off"
+    except ValueError:
+        speaker = d["speaker"]
+    d.update({"speaker" : speaker})
+    return d
 
 def make_defaults_struct(extras_defaults={}, extras_types={}, extras_validation={}, username="", uid=getuid(), gid=getgid()):
     #populate config with default values:
     if not username and uid:
         username = get_username_for_uid(uid)
     defaults = read_xpra_defaults(username, uid, gid)
+    defaults = get_speaker_override(defaults)
     return dict_to_validated_config(defaults, extras_defaults, extras_types, extras_validation)
 
 def dict_to_validated_config(d={}, extras_defaults={}, extras_types={}, extras_validation={}):
