--- a/setup.py.orig	2022-12-25 22:59:08.748690292 -0800
+++ b/setup.py	2022-12-25 23:01:48.583222868 -0800
@@ -631,7 +631,11 @@ def remove_from_keywords(kw, key, value)
     values = kw.get(key)
     while values and value in values:
         values.remove(value)
-
+def has_value_from_keywords(kw, key, value):
+    values = kw.get(key)
+    while values and value in values:
+        return True
+    return False
 
 def checkdirs(*dirs):
     for d in dirs:
@@ -808,6 +812,36 @@ def exec_pkgconfig(*pkgs_options, **ekw)
         print("adding LDFLAGS=%s" % (LDFLAGS, ))
     add_tokens(CFLAGS, "extra_compile_args")
     add_tokens(LDFLAGS, "extra_link_args")
+    has_retpoline=False
+    # Translate flags
+    if os.environ.get("CC", "").find("clang")>=0:
+        for k in ['extra_compile_args', 'extra_link_args']:
+            if has_value_from_keywords(kw, k, '-mindirect-branch=thunk'):
+                remove_from_keywords(kw, k, "-mindirect-branch=thunk")
+                remove_from_keywords(kw, k, "-mindirect-branch-register")
+                add_to_keywords(kw, k, '-mretpoline')
+            if has_value_from_keywords(kw, k, '-flto=thin'):
+                remove_from_keywords(kw, k, "-flto=thin")
+                add_to_keywords(kw, k, '-flto')
+            if has_value_from_keywords(kw, k, '-flto=auto'):
+                remove_from_keywords(kw, k, "-flto=auto")
+                add_to_keywords(kw, k, '-flto')
+            for x in range(1,256):
+                if has_value_from_keywords(kw, k, "-flto=%s" % x):
+                    remove_from_keywords(kw, k, "-flto=%s" % x)
+                    add_to_keywords(kw, k, '-flto')
+    else:
+        for k in ['extra_compile_args', 'extra_link_args']:
+            if has_value_from_keywords(kw, k, '-mindirect-branch=thunk'):
+                remove_from_keywords(kw, k, "-mretpoline")
+                add_to_keywords(kw, k, '-mindirect-branch=thunk')
+                add_to_keywords(kw, k, '-mindirect-branch-register')
+            if has_value_from_keywords(kw, k, '-flto=thin'):
+                remove_from_keywords(kw, k, "-flto=thin")
+                add_to_keywords(kw, k, '-flto')
+            if has_value_from_keywords(kw, k, '-fuse-ld=thin'):
+                remove_from_keywords(kw, k, "-fuse-ld=bfd")
+                add_to_keywords(kw, k, '-fuse-ld=bfd')
     #add_to_keywords(kw, 'include_dirs', '.')
     if debug_ENABLED and WIN32 and MINGW_PREFIX:
         extra_compile_args.append("-DDEBUG")
