diff -urp xpra-4.0.3.orig/service/xpra-nosocketactivation.service xpra-4.0.3/service/xpra-nosocketactivation.service
--- xpra-4.0.3.orig/service/xpra-nosocketactivation.service	2020-05-10 10:00:50.000000000 -0700
+++ xpra-4.0.3/service/xpra-nosocketactivation.service	2020-08-26 16:24:57.570815465 -0700
@@ -6,7 +6,7 @@ After=network.target
 
 [Service]
 Type=simple
-EnvironmentFile=-/etc/default/xpra
+EnvironmentFile=-/etc/conf.d/xpra
 ExecStart=/usr/bin/xpra proxy :14500 --daemon=no \
     --bind-tcp=0.0.0.0:14500 --tcp-auth=${TCP_AUTH} \
     --ssl-cert=/etc/xpra/ssl-cert.pem --ssl=on \
diff -urp xpra-4.0.3.orig/service/xpra.service xpra-4.0.3/service/xpra.service
--- xpra-4.0.3.orig/service/xpra.service	2020-05-10 10:00:50.000000000 -0700
+++ xpra-4.0.3/service/xpra.service	2020-08-26 16:25:03.221042552 -0700
@@ -7,7 +7,7 @@ Requires=xpra.socket
 
 [Service]
 Type=simple
-EnvironmentFile=-/etc/sysconfig/xpra
+EnvironmentFile=-/etc/conf.d/xpra
 ExecStart=/usr/bin/xpra proxy :14500 --daemon=no \
     --tcp-auth=${TCP_AUTH} \
     --ssl-cert=/etc/xpra/ssl-cert.pem --ssl=on \
diff -urp xpra-4.0.3.orig/setup.py xpra-4.0.3/setup.py
--- xpra-4.0.3.orig/setup.py	2020-08-26 13:20:53.291303030 -0700
+++ xpra-4.0.3/setup.py	2020-08-26 16:27:07.236025320 -0700
@@ -1567,7 +1567,9 @@ else:
                         copytodir("service/xpra-nosocketactivation.service", systemd_dir, dst_name="xpra.service")
                 else:
                     copytodir("service/xpra", "/etc/init.d")
-                if os.path.exists("/etc/sysconfig"):
+                if os.path.exists("/etc/conf.d"):
+                    copytodir("etc/sysconfig/xpra", "/etc/conf.d")
+                elif os.path.exists("/etc/sysconfig"):
                     copytodir("etc/sysconfig/xpra", "/etc/sysconfig")
                 elif os.path.exists("/etc/default"):
                     copytodir("etc/sysconfig/xpra", "/etc/default")
