--- a/cmake/Dependencies.cmake.orig	2023-07-28 08:50:07.548404902 -0700
+++ b/cmake/Dependencies.cmake	2023-07-28 08:52:58.372409351 -0700
@@ -27,7 +27,7 @@
 
 # Workaround until hcc & hip cmake modules fixes symlink logic in their config files.
 # (Thanks to rocBLAS devs for finding workaround for this problem!)
-list(APPEND CMAKE_PREFIX_PATH /opt/rocm /opt/rocm/hip)
+list(APPEND CMAKE_PREFIX_PATH /usr/share)
 
 # ROCm cmake package
 find_package(ROCM 0.7.3 QUIET CONFIG PATHS ${CMAKE_PREFIX_PATH})
