--- a/clients/tests/CMakeLists.txt.orig	2023-07-28 08:34:00.796379722 -0700
+++ b/clients/tests/CMakeLists.txt	2023-07-28 08:35:12.208381582 -0700
@@ -29,39 +29,6 @@ if(BUILD_MEMSTAT)
   add_compile_options(-DROCSPARSE_WITH_MEMSTAT)
 endif()
 
-#
-# Client matrices.
-#
-if(NOT EXISTS "${CMAKE_MATRICES_DIR}")
-  #
-  # Download.
-  #
-  set(CMAKE_MATRICES_DIR ${PROJECT_BINARY_DIR}/matrices CACHE STRING "Matrices directory.")
-
-  if(NOT TARGET rocsparse)
-    set(CONVERT_SOURCE ${CMAKE_SOURCE_DIR}/../deps/convert.cpp CACHE STRING "Convert tool mtx2csr.")
-    include(${CMAKE_SOURCE_DIR}/../cmake/ClientMatrices.cmake)
-  else()
-   set(CONVERT_SOURCE ${CMAKE_SOURCE_DIR}/deps/convert.cpp CACHE STRING "Convert tool mtx2csr.")
-    include(${CMAKE_SOURCE_DIR}/cmake/ClientMatrices.cmake)
-  endif()
-
-else()
-  #
-  # Copy.
-  #
-  if(NOT CMAKE_MATRICES_DIR STREQUAL "${PROJECT_BINARY_DIR}/matrices")
-    message("Copy matrix files from ${CMAKE_MATRICES_DIR} to ${PROJECT_BINARY_DIR}/matrices")
-
-    execute_process(COMMAND cp -r ${CMAKE_MATRICES_DIR} ${PROJECT_BINARY_DIR}/matrices RESULT_VARIABLE STATUS WORKING_DIRECTORY ${CMAKE_MATRICES_DIR})
-
-    if(STATUS AND NOT STATUS EQUAL 0)
-      message(FATAL_ERROR "Failed to copy matrix .csr files, aborting.")
-    endif()
-  endif()
-
-endif()
-
 set(ROCSPARSE_TEST_SOURCES
   test_axpby.cpp
   test_axpyi.cpp
