diff -urp boinc-client_release-7.2-7.2.44.orig/api/Makefile.am boinc-client_release-7.2-7.2.44/api/Makefile.am
--- boinc-client_release-7.2-7.2.44.orig/api/Makefile.am	2014-03-20 11:03:19.000000000 -0700
+++ boinc-client_release-7.2-7.2.44/api/Makefile.am	2017-01-19 01:47:38.409829556 -0800
@@ -42,7 +42,7 @@ if BUILD_GRAPHICS_API
 lib_LTLIBRARIES += libboinc_graphics2.la
 libboinc_graphics2_la_SOURCES = $(graphics2_files)
 # libboinc_graphics2_la_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/lib
-libboinc_graphics2_la_LDFLAGS = -L$(libdir) -rpath $(libdir) -version-number $(LIBBOINC_VERSION) -ljpeg
+libboinc_graphics2_la_LDFLAGS = -L$(libdir) -rpath $(libdir) -version-number $(LIBBOINC_VERSION) -ljpeg -lglut
 endif #BUILD_GRAPHICS_API
 
 lib_LTLIBRARIES += libboinc_opencl.la
Only in boinc-client_release-7.2-7.2.44: Makefile.am.orig
Only in boinc-client_release-7.2-7.2.44: Makefile.am.rej
