diff -urp aurora-scanner-0.0.0_p9999.orig/src/waylandscanner/waylandscanner.cpp aurora-scanner-0.0.0_p9999/src/waylandscanner/waylandscanner.cpp
--- aurora-scanner-0.0.0_p9999.orig/src/waylandscanner/waylandscanner.cpp	2022-11-30 10:18:30.731607478 -0800
+++ aurora-scanner-0.0.0_p9999/src/waylandscanner/waylandscanner.cpp	2022-11-30 10:21:25.818466239 -0800
@@ -640,9 +640,9 @@ bool Scanner::process()
 
     if (m_option == ServerCode) {
         if (m_headerPath.isEmpty())
-            printf("#include \"aurora-server-%s.h\"\n", QByteArray(m_protocolName).replace('_', '-').constData());
+            printf("#include \"qwayland-server-%s.h\"\n", QByteArray(m_protocolName).replace('_', '-').constData());
         else
-            printf("#include <%s/aurora-server-%s.h>\n", m_headerPath.constData(), QByteArray(m_protocolName).replace('_', '-').constData());
+            printf("#include <%s/qwayland-server-%s.h>\n", m_headerPath.constData(), QByteArray(m_protocolName).replace('_', '-').constData());
         printf("\n");
         printf("namespace Aurora {\n");
         printf("\n");
@@ -1094,9 +1094,9 @@ bool Scanner::process()
 
     if (m_option == ClientCode) {
         if (m_headerPath.isEmpty())
-            printf("#include \"aurora-client-%s.h\"\n", QByteArray(m_protocolName).replace('_', '-').constData());
+            printf("#include \"qwayland-%s.h\"\n", QByteArray(m_protocolName).replace('_', '-').constData());
         else
-            printf("#include <%s/aurora-client-%s.h>\n", m_headerPath.constData(), QByteArray(m_protocolName).replace('_', '-').constData());
+            printf("#include <%s/qwayland-%s.h>\n", m_headerPath.constData(), QByteArray(m_protocolName).replace('_', '-').constData());
         printf("\n");
         printf("namespace Aurora {\n");
         printf("\n");
