--- a/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/browser.js.orig	2019-06-11 00:19:36.032447115 -0700
+++ b/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/browser.js	2019-06-11 00:24:30.123761694 -0700
@@ -130,7 +130,11 @@ function getBrowserConfig(wco) {
                 sri: buildOptions.subresourceIntegrity,
             }),
         ]),
-        node: false, target: "electron-renderer"
+        node: {
+              fs: 'empty',
+              crypto: 'empty'
+        },
+        target: "electron-renderer"
     };
 }
 exports.getBrowserConfig = getBrowserConfig;
