--- a/src/browser/vcs/vcs-view/vcs-item.ts.orig	2018-12-31 06:50:25.000000000 -0800
+++ b/src/browser/vcs/vcs-view/vcs-item.ts	2019-02-28 12:43:22.144500954 -0800
@@ -125,6 +125,6 @@ export interface VcsItemCreateResult<T e
 export abstract class VcsItemFactory<T extends VcsItem> {
     abstract create(
         fileChanges: VcsFileChange[],
-        ...extras: any[],
+        ...extras: any[]
     ): VcsItemCreateResult<T> | Promise<VcsItemCreateResult<T>>;
 }
