--- a/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/browser.js.orig	1985-10-26 01:15:00.000000000 -0700
+++ b/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/browser.js	2019-03-08 15:42:45.726607648 -0800
@@ -130,7 +130,11 @@ function getBrowserConfig(wco) {
                 sri: buildOptions.subresourceIntegrity,
             }),
         ]),
-        node: false,
+        node: {
+              fs: 'empty',
+              crypto: 'empty'
+        },
+        target: "electron-renderer"
     };
 }
 exports.getBrowserConfig = getBrowserConfig;
