--- a/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/browser.js.orig	2019-03-09 23:04:38.298959164 -0800
+++ b/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/browser.js	2019-03-09 23:12:07.054928919 -0800
@@ -130,7 +130,11 @@ function getBrowserConfig(wco) {
                 sri: buildOptions.subresourceIntegrity,
             }),
         ]),
-        node: false, target: "electron-renderer" 
+        node: {
+              fs: 'empty',
+              crypto: 'empty'
+        },
+        target: "electron-renderer"
     };
 }
 exports.getBrowserConfig = getBrowserConfig;
