diff -urp chromium-55.0.2883.75.orig/tools/gn/bootstrap/bootstrap.py chromium-55.0.2883.75/tools/gn/bootstrap/bootstrap.py
--- chromium-55.0.2883.75.orig/tools/gn/bootstrap/bootstrap.py	2016-12-27 19:53:16.246594097 -0800
+++ chromium-55.0.2883.75/tools/gn/bootstrap/bootstrap.py	2016-12-27 19:58:37.514955077 -0800
@@ -565,7 +565,7 @@ def write_gn_ninja(path, root_gen_dir, o
 
   if is_linux:
     libs.extend(['-lrt', '-latomic'])
-    ldflags.extend(['-pthread'])
+    ldflags.extend(['-m32 -pthread'])
 
     static_libraries['xdg_user_dirs'] = {
         'sources': [
