--- AKv8/configure.ac.orig	2015-11-30 23:12:05.686569727 -0800
+++ AKv8/configure.ac	2015-11-30 23:13:05.139843730 -0800
@@ -517,7 +517,9 @@
 #endif
 
 #if !defined(CUSTOM_STRING) && defined(COMPILER_STRING)
-#define CUSTOM_STRING PACKAGE_STRING" "SVN_REV" "COMPILER_STRING
+#define QUOTE(s) #s
+#define xstr(s) QUOTE(s)
+#define CUSTOM_STRING xstr(PACKAGE_STRING)" "SVN_REV" "COMPILER_STRING
 #endif
 
 #include "std_fixes.h"
