diff -urp ROCclr-rocm-5.6.0.orig/cmake/ROCclrHSA.cmake ROCclr-rocm-5.6.0/cmake/ROCclrHSA.cmake
--- ROCclr-rocm-5.6.0.orig/cmake/ROCclrHSA.cmake	2023-03-15 02:14:00.000000000 -0700
+++ ROCclr-rocm-5.6.0/cmake/ROCclrHSA.cmake	2023-09-03 18:40:16.716528287 -0700
@@ -20,11 +20,10 @@
 
 find_package(hsa-runtime64 1.6 REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    @ESYSROOT_ROCM_PATH@/
   PATH_SUFFIXES
     cmake/hsa-runtime64
-    lib/cmake/hsa-runtime64
-    lib64/cmake/hsa-runtime64)
+    @LIBDIR@/cmake/hsa-runtime64)
 target_link_libraries(rocclr PUBLIC hsa-runtime64::hsa-runtime64)
 
 find_package(NUMA)
diff -urp ROCclr-rocm-5.6.0.orig/cmake/ROCclrLC.cmake ROCclr-rocm-5.6.0/cmake/ROCclrLC.cmake
--- ROCclr-rocm-5.6.0.orig/cmake/ROCclrLC.cmake	2023-03-15 02:14:00.000000000 -0700
+++ ROCclr-rocm-5.6.0/cmake/ROCclrLC.cmake	2023-09-03 18:40:15.508528303 -0700
@@ -20,10 +20,10 @@
 
 find_package(amd_comgr REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    @ESYSROOT_ROCM_PATH@/
   PATH_SUFFIXES
     cmake/amd_comgr
-    lib/cmake/amd_comgr)
+    @LIBDIR@/cmake/amd_comgr)
 
 target_compile_definitions(rocclr PUBLIC WITH_LIGHTNING_COMPILER USE_COMGR_LIBRARY)
 if(BUILD_SHARED_LIBS)
diff -urp ROCclr-rocm-5.6.0.orig/elf/test/CMakeLists.txt ROCclr-rocm-5.6.0/elf/test/CMakeLists.txt
--- ROCclr-rocm-5.6.0.orig/elf/test/CMakeLists.txt	2023-03-15 02:14:00.000000000 -0700
+++ ROCclr-rocm-5.6.0/elf/test/CMakeLists.txt	2023-09-03 18:39:19.716529048 -0700
@@ -26,14 +26,14 @@ cmake_minimum_required(VERSION 3.5.1)
 
 find_package(amd_comgr REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    @EPREFIX_ROCM_PATH@/
   PATH_SUFFIXES
     cmake/amd_comgr
-    lib/cmake/amd_comgr)
+    @LIBDIR@/cmake/amd_comgr)
 
 find_package(hsa-runtime64 REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    @EPREFIX_ROCM_PATH@/
   PATH_SUFFIXES
-    cmake/hsa-runtime64)
+    @LIBDIR@/cmake/hsa-runtime64)
 
@@ -42,8 +42,8 @@ find_package(Threads REQUIRED)
 # Look for ROCclr which contains elfio
 find_package(ROCclr REQUIRED CONFIG
   PATHS
-    /opt/rocm
-    /opt/rocm/rocclr)
+    @EPREFIX_ROCM_PATH@
+    @EPREFIX_ROCM_PATH@/rocclr)
 
 add_executable(elf_test main.cpp)
 set_target_properties(
@@ -60,4 +60,4 @@ add_definitions(-DUSE_COMGR_LIBRARY -DCO
 
 target_link_libraries(elf_test PRIVATE amdrocclr_static)
 
-#-------------------------------------elf_test--------------------------------------#
\ No newline at end of file
+#-------------------------------------elf_test--------------------------------------#
diff -urp ROCclr-rocm-5.6.0.orig/elf/test/main.cpp ROCclr-rocm-5.6.0/elf/test/main.cpp
--- ROCclr-rocm-5.6.0.orig/elf/test/main.cpp	2023-03-15 02:14:00.000000000 -0700
+++ ROCclr-rocm-5.6.0/elf/test/main.cpp	2023-09-03 18:40:43.636527927 -0700
@@ -45,8 +45,8 @@ static constexpr size_t rodataSymbolInfo
 
 // Elf::COMMENT,        ".comment",        1, SHT_PROGBITS, 0,
 static const amd::Elf::SymbolInfo commentSymbolInfos_[] = {
-    { ".comment", nullptr, 0, "compile", "-g -I/opt/include",  strlen("-g -I/opt/include") + 1 },
-    { ".comment", nullptr, 0, "link",    "-g -l/opt/rocm/lib", strlen("-g -l/opt/rocm/lib") + 1 },
+    { ".comment", nullptr, 0, "compile", "-g -I@EPREFIX_ROCM_PATH@/include",  strlen("-g -I@EPREFIX_ROCM_PATH@/include") + 1 },
+    { ".comment", nullptr, 0, "link",    "-g -l@EPREFIX_ROCM_PATH@/@LIBDIR@", strlen("-g -l@EPREFIX_ROCM_PATH@/@LIBDIR@") + 1 },
   };
 static constexpr size_t commentSymbolInfosSize_ = sizeof(commentSymbolInfos_)
     / sizeof(commentSymbolInfos_[0]);
diff -urp ROCclr-rocm-5.6.0.orig/os/os_posix.cpp ROCclr-rocm-5.6.0/os/os_posix.cpp
--- ROCclr-rocm-5.6.0.orig/os/os_posix.cpp	2023-03-15 02:14:00.000000000 -0700
+++ ROCclr-rocm-5.6.0/os/os_posix.cpp	2023-09-03 18:44:03.396525260 -0700
@@ -449,7 +449,7 @@ uint64_t Os::timerResolutionNanos() {
 
 const char* Os::libraryExtension() { return MACOS_SWITCH(".dylib", ".so"); }
 
-const char* Os::libraryPrefix() { return "lib"; }
+const char* Os::libraryPrefix() { return "@LIBDIR@"; }
 
 const char* Os::objectExtension() { return ".o"; }
 
diff -urp ROCclr-rocm-5.6.0.orig/README.md ROCclr-rocm-5.6.0/README.md
--- ROCclr-rocm-5.6.0.orig/README.md	2023-03-15 02:14:00.000000000 -0700
+++ ROCclr-rocm-5.6.0/README.md	2023-09-03 18:38:46.372529493 -0700
@@ -42,7 +42,7 @@ Here is command to build ROCclr:
 ```bash
 cd "$ROCclr_DIR"
 mkdir -p build; cd build
-cmake -DOPENCL_DIR="$OPENCL_DIR" -DCMAKE_INSTALL_PREFIX=/opt/rocm/rocclr ..
+cmake -DOPENCL_DIR="$OPENCL_DIR" -DCMAKE_INSTALL_PREFIX=@EPREFIX_ROCM_PATH@/rocclr ..
 make -j$(nproc)
 sudo make install
 ```
@@ -55,7 +55,7 @@ git clone -b main https://github.com/ROC
 export HIP_DIR="$(readlink -f HIP)"
 cd "$HIP_DIR"
 mkdir -p build; cd build
-cmake -DCMAKE_PREFIX_PATH="$ROCclr_DIR/build;/opt/rocm/" ..
+cmake -DCMAKE_PREFIX_PATH="$ROCclr_DIR/build;@EPREFIX_ROCM_PATH@/" ..
 make -j$(nproc)
 ```
 
