diff -urp opencl.orig/amdocl/CMakeLists.txt opencl/amdocl/CMakeLists.txt
--- opencl.orig/amdocl/CMakeLists.txt	2023-09-27 12:02:57.373167544 -0700
+++ opencl/amdocl/CMakeLists.txt	2023-09-27 12:03:12.837167366 -0700
@@ -100,6 +100,20 @@ target_include_directories(amdocl PRIVAT
   ${ROCCLR_PATH}/include
   )
 
+target_compile_definitions(amdocl
+  PRIVATE
+   -DCL_TARGET_OPENCL_VERSION=220
+   -DHAVE_CL2_HPP
+   -DOPENCL_MAJOR=2
+   -DOPENCL_MINOR=1
+   -DOPENCL_C_MAJOR=2
+   -DOPENCL_C_MINOR=0
+   -DCL_TARGET_OPENCL_VERSION=220
+   -DCL_USE_DEPRECATED_OPENCL_1_0_APIS
+   -DCL_USE_DEPRECATED_OPENCL_1_1_APIS
+   -DCL_USE_DEPRECATED_OPENCL_1_2_APIS
+   -DCL_USE_DEPRECATED_OPENCL_2_0_APIS)
+
 if(WIN32)
   target_sources(amdocl PRIVATE
   cl_runtime.cpp)
Only in opencl/amdocl: CMakeLists.txt.orig
diff -urp opencl.orig/tools/cltrace/CMakeLists.txt opencl/tools/cltrace/CMakeLists.txt
--- opencl.orig/tools/cltrace/CMakeLists.txt	2023-09-27 12:02:57.373167544 -0700
+++ opencl/tools/cltrace/CMakeLists.txt	2023-09-27 12:04:11.913166687 -0700
@@ -14,6 +14,20 @@ target_include_directories(cltrace PRIVA
   ${ROCCLR_PATH}/include
   )
 
+target_compile_definitions(amdocl
+  PRIVATE
+   -DCL_TARGET_OPENCL_VERSION=220
+   -DHAVE_CL2_HPP
+   -DOPENCL_MAJOR=2
+   -DOPENCL_MINOR=1
+   -DOPENCL_C_MAJOR=2
+   -DOPENCL_C_MINOR=0
+   -DCL_TARGET_OPENCL_VERSION=220
+   -DCL_USE_DEPRECATED_OPENCL_1_0_APIS
+   -DCL_USE_DEPRECATED_OPENCL_1_1_APIS
+   -DCL_USE_DEPRECATED_OPENCL_1_2_APIS
+   -DCL_USE_DEPRECATED_OPENCL_2_0_APIS)
+
 INSTALL(TARGETS cltrace
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
