diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.acml magma-2.8.0/make.inc-examples/make.inc.acml
--- magma-2.8.0.orig/make.inc-examples/make.inc.acml	2024-05-06 14:17:35.940124871 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.acml	2024-05-06 14:18:13.451804028 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.acml.orig
Only in magma-2.8.0/make.inc-examples: make.inc.acml.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.atlas magma-2.8.0/make.inc-examples/make.inc.atlas
--- magma-2.8.0.orig/make.inc-examples/make.inc.atlas	2024-05-06 14:17:35.940124871 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.atlas	2024-05-06 14:18:27.719681993 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.atlas.orig
Only in magma-2.8.0/make.inc-examples: make.inc.atlas.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.hip-gcc-mkl magma-2.8.0/make.inc-examples/make.inc.hip-gcc-mkl
--- magma-2.8.0.orig/make.inc-examples/make.inc.hip-gcc-mkl	2024-05-06 14:17:35.940124871 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.hip-gcc-mkl	2024-05-06 14:17:53.147977690 -0700
@@ -81,6 +81,7 @@ else ifeq ($(BACKEND),hip)
     # not specific enough. TODO: find better link?
     GPU_TARGET ?= gfx900 gfx906 gfx908
 endif
+@GPU_TARGET_OVERRIDE@
 
 # Use -fPIC to make shared (.so) and static (.a) library;
 # can be commented out if making only static library.
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.hip-gcc-openblas magma-2.8.0/make.inc-examples/make.inc.hip-gcc-openblas
--- magma-2.8.0.orig/make.inc-examples/make.inc.hip-gcc-openblas	2024-05-06 14:17:35.940124871 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.hip-gcc-openblas	2024-05-06 14:17:53.147977690 -0700
@@ -76,7 +76,7 @@ RANLIB      ?= ranlib
 ifeq ($(BACKEND),cuda)
     # See a full table: https://nouveau.freedesktop.org/wiki/CodeNames/
     # note: provided by freedesktop, which is reliable, but not from the vendor
-    GPU_TARGET = Volta Turing Ampere
+    GPU_TARGET ?= Volta Turing Ampere
 else ifeq ($(BACKEND),hip)
     # See a full table:
     #   https://llvm.org/docs/AMDGPUUsage.html#amdgpu-processor-table
@@ -85,6 +85,7 @@ else ifeq ($(BACKEND),hip)
     # not specific enough. TODO: find better link?
     GPU_TARGET ?= gfx900 gfx906 gfx908
 endif
+@GPU_TARGET_OVERRIDE@
 
 # Use -fPIC to make shared (.so) and static (.a) library;
 # can be commented out if making only static library.
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.macos magma-2.8.0/make.inc-examples/make.inc.macos
--- magma-2.8.0.orig/make.inc-examples/make.inc.macos	2024-05-06 14:17:35.940124871 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.macos	2024-05-06 14:22:35.209565135 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.macos-debug magma-2.8.0/make.inc-examples/make.inc.macos-debug
--- magma-2.8.0.orig/make.inc-examples/make.inc.macos-debug	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.macos-debug	2024-05-06 14:18:48.071507921 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.macos-debug.orig
Only in magma-2.8.0/make.inc-examples: make.inc.macos-debug.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.macos-mkl magma-2.8.0/make.inc-examples/make.inc.macos-mkl
--- magma-2.8.0.orig/make.inc-examples/make.inc.macos-mkl	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.macos-mkl	2024-05-06 14:22:47.241462221 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.macos-mkl.orig
Only in magma-2.8.0/make.inc-examples: make.inc.macos-mkl.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.macos-openmp magma-2.8.0/make.inc-examples/make.inc.macos-openmp
--- magma-2.8.0.orig/make.inc-examples/make.inc.macos-openmp	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.macos-openmp	2024-05-06 14:22:54.617399131 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.macos-openmp.orig
Only in magma-2.8.0/make.inc-examples: make.inc.macos-openmp.rej
Only in magma-2.8.0/make.inc-examples: make.inc.macos.orig
Only in magma-2.8.0/make.inc-examples: make.inc.macos.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc magma-2.8.0/make.inc-examples/make.inc.mkl-gcc
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-gcc	2024-05-06 14:23:17.529203155 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc-debug magma-2.8.0/make.inc-examples/make.inc.mkl-gcc-debug
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc-debug	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-gcc-debug	2024-05-06 14:23:12.169249001 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc-debug.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc-debug.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc-ilp64 magma-2.8.0/make.inc-examples/make.inc.mkl-gcc-ilp64
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc-ilp64	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-gcc-ilp64	2024-05-06 14:23:08.073284036 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc-ilp64-debug magma-2.8.0/make.inc-examples/make.inc.mkl-gcc-ilp64-debug
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-gcc-ilp64-debug	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-gcc-ilp64-debug	2024-05-06 14:23:01.657338915 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc-ilp64-debug.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc-ilp64-debug.rej
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc-ilp64.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc-ilp64.rej
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-gcc.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc magma-2.8.0/make.inc-examples/make.inc.mkl-icc
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc	2024-05-06 14:17:35.944124837 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-icc	2024-05-06 14:22:36.249556240 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc-debug magma-2.8.0/make.inc-examples/make.inc.mkl-icc-debug
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc-debug	2024-05-06 14:17:35.948124802 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-icc-debug	2024-05-06 14:22:41.337512720 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc-debug.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc-debug.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc-ilp64 magma-2.8.0/make.inc-examples/make.inc.mkl-icc-ilp64
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc-ilp64	2024-05-06 14:17:35.948124802 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-icc-ilp64	2024-05-06 14:22:55.641390372 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc-ilp64-debug magma-2.8.0/make.inc-examples/make.inc.mkl-icc-ilp64-debug
--- magma-2.8.0.orig/make.inc-examples/make.inc.mkl-icc-ilp64-debug	2024-05-06 14:17:35.948124802 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.mkl-icc-ilp64-debug	2024-05-06 14:23:19.593185501 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc-ilp64-debug.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc-ilp64-debug.rej
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc-ilp64.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc-ilp64.rej
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc.orig
Only in magma-2.8.0/make.inc-examples: make.inc.mkl-icc.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.openblas magma-2.8.0/make.inc-examples/make.inc.openblas
--- magma-2.8.0.orig/make.inc-examples/make.inc.openblas	2024-05-06 14:17:35.948124802 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.openblas	2024-05-06 14:17:53.151977656 -0700
@@ -54,10 +54,11 @@ RANLIB    = ranlib
 
 # set our GPU targets
 ifeq ($(BACKEND),cuda)
-    GPU_TARGET = Volta Turing Ampere
+    GPU_TARGET ?= Volta Turing Ampere
 else ifeq ($(BACKEND),hip)
-    GPU_TARGET = gfx900 gfx906 gfx908
+    GPU_TARGET ?= gfx900 gfx906 gfx908
 endif
+@GPU_TARGET_OVERRIDE@
 
 # Use -fPIC to make shared (.so) and static (.a) library;
 # can be commented out if making only static library.
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.openblas-debug magma-2.8.0/make.inc-examples/make.inc.openblas-debug
--- magma-2.8.0.orig/make.inc-examples/make.inc.openblas-debug	2024-05-06 14:17:35.948124802 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.openblas-debug	2024-05-06 14:22:42.665501361 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.openblas-debug.orig
Only in magma-2.8.0/make.inc-examples: make.inc.openblas-debug.rej
diff -urp magma-2.8.0.orig/make.inc-examples/make.inc.power9-essl magma-2.8.0/make.inc-examples/make.inc.power9-essl
--- magma-2.8.0.orig/make.inc-examples/make.inc.power9-essl	2024-05-06 14:17:35.948124802 -0700
+++ magma-2.8.0/make.inc-examples/make.inc.power9-essl	2024-05-06 14:22:52.873414048 -0700
@@ -20,6 +20,7 @@
 # See http://developer.nvidia.com/cuda-gpus
 #
 GPU_TARGET ?= Volta Turing Ampere
+@GPU_TARGET_OVERRIDE@
 
 # --------------------
 # programs
Only in magma-2.8.0/make.inc-examples: make.inc.power9-essl.orig
Only in magma-2.8.0/make.inc-examples: make.inc.power9-essl.rej
