--- src/hsa-backward-compat.cmake.orig	2023-06-28 05:56:26.000000000 -0700
+++ src/hsa-backward-compat.cmake	2023-08-20 02:16:01.678176985 -0700
@@ -19,7 +19,7 @@
 
 set(HSA_WRAPPER_DIR ${CMAKE_CURRENT_BINARY_DIR}/wrapper_dir)
 set(HSA_WRAPPER_INC_DIR ${HSA_WRAPPER_DIR}/include/hsa)
-set(HSA_WRAPPER_LIB_DIR ${HSA_WRAPPER_DIR}/lib)
+set(HSA_WRAPPER_LIB_DIR ${HSA_WRAPPER_DIR}/lib64)
 
 #Function to generate header template file
 function(create_header_template)
