diff -urp ROCR-Runtime-rocm-5.7.1.orig/src/CMakeLists.txt ROCR-Runtime-rocm-5.7.1/src/CMakeLists.txt
--- src/CMakeLists.txt	2024-04-01 14:36:31.530240952 -0700
+++ src/CMakeLists.txt	2024-04-01 14:40:53.971851180 -0700
@@ -275,6 +275,9 @@ endif()
 ## Link dependencies.
 target_link_libraries ( ${CORE_RUNTIME_TARGET} PRIVATE hsakmt::hsakmt PkgConfig::drm)
 target_link_libraries ( ${CORE_RUNTIME_TARGET} PRIVATE elf::elf dl pthread rt )
+target_link_options ( ${CORE_RUNTIME_TARGET} PUBLIC -Wl,-rpath=@ROCM_PATH@/@LIBDIR@ )
+target_link_directories ( ${CORE_RUNTIME_TARGET} PUBLIC @ROCM_PATH@/@LIBDIR@ )
+target_link_libraries ( ${CORE_RUNTIME_TARGET} PUBLIC amd_comgr )
 
 ## Set the VERSION and SOVERSION values
 set_property ( TARGET ${CORE_RUNTIME_TARGET} PROPERTY VERSION "${SO_VERSION_STRING}" )
