diff -urp src.orig/CMakeLists.txt src/CMakeLists.txt
--- src.orig/CMakeLists.txt	2024-04-01 15:52:37.781545344 -0700
+++ src/CMakeLists.txt	2024-04-01 15:53:42.260978812 -0700
@@ -272,6 +272,9 @@ if(${IMAGE_SUPPORT})
 endif()
 
 ## Link dependencies.
+target_link_options ( ${CORE_RUNTIME_TARGET} PUBLIC -Wl,-rpath=@ROCM_PATH@/@LIBDIR@ )
+target_link_directories ( ${CORE_RUNTIME_TARGET} PUBLIC @ROCM_PATH@/@LIBDIR@ )
+target_link_libraries ( ${CORE_RUNTIME_TARGET} PUBLIC hsakmt )
 target_link_libraries ( ${CORE_RUNTIME_TARGET} PRIVATE hsakmt::hsakmt PkgConfig::drm)
 target_link_libraries ( ${CORE_RUNTIME_TARGET} PRIVATE elf::elf dl pthread rt )
 
Only in src: CMakeLists.txt.orig
Only in src: CMakeLists.txt.rej
