diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/CrowdAgent.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/CrowdAgent.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/CrowdAgent.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/CrowdAgent.cpp	2021-03-04 22:55:48.637098600 -0800
@@ -33,8 +33,13 @@
 #include "../Scene/Node.h"
 #include "../Scene/Scene.h"
 
+#ifdef URHO3D_SYSTEM_RECASTNAVIGATION
+#include <recastnavigation/DetourCommon.h>
+#include <recastnavigation/DetourCrowd.h>
+#else
 #include <Detour/DetourCommon.h>
 #include <DetourCrowd/DetourCrowd.h>
+#endif
 
 #include "../DebugNew.h"
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/CrowdManager.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/CrowdManager.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/CrowdManager.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/CrowdManager.cpp	2021-03-04 23:00:21.043642706 -0800
@@ -34,7 +34,11 @@
 #include "../Scene/Scene.h"
 #include "../Scene/SceneEvents.h"
 
+#ifdef URHO3D_SYSTEM_RECASTNAVIGATION
+#include <recastnavigation/DetourCrowd.h>
+#else
 #include <DetourCrowd/DetourCrowd.h>
+#endif
 
 #include "../DebugNew.h"
 
@@ -69,11 +73,6 @@ static const StringVector obstacleAvoida
     "   Adaptive Depth"
 };
 
-void CrowdAgentUpdateCallback(dtCrowdAgent* ag, float dt)
-{
-    static_cast<CrowdAgent*>(ag->params.userData)->OnCrowdUpdate(ag, dt);
-}
-
 CrowdManager::CrowdManager(Context* context) :
     Component(context),
     maxAgents_(DEFAULT_MAX_AGENTS),
@@ -83,10 +82,13 @@ CrowdManager::CrowdManager(Context* cont
     numAreas_.Reserve(DT_CROWD_MAX_QUERY_FILTER_TYPE);
     for (unsigned i = 0; i < DT_CROWD_MAX_QUERY_FILTER_TYPE; ++i)
         numAreas_.Push(0);
+
+    agents_ = (dtCrowdAgent**)dtAlloc(sizeof(dtCrowdAgent*)*DEFAULT_MAX_AGENTS, DT_ALLOC_PERM);
 }
 
 CrowdManager::~CrowdManager()
 {
+    dtFree(agents_);
     dtFreeCrowd(crowd_);
     crowd_ = nullptr;
 }
@@ -127,7 +129,11 @@ void CrowdManager::ApplyAttributes()
     navigationMeshId_ = navigationMesh_ ? navigationMesh_->GetID() : 0;
 
     // If the Detour crowd initialization parameters have changed then recreate it
-    if (crowd_ && (navMeshChange || crowd_->getAgentCount() != maxAgents_ || crowd_->getMaxAgentRadius() != maxAgentRadius_))
+    if (crowd_ && (navMeshChange || crowd_->getAgentCount() != maxAgents_
+#ifndef URHO3D_SYSTEM_RECASTNAVIGATION
+        || crowd_->getMaxAgentRadius() != maxAgentRadius_
+#endif
+    ))
         CreateCrowd();
 }
 
@@ -584,7 +590,7 @@ bool CrowdManager::CreateCrowd()
     // Initialize the crowd
     if (maxAgentRadius_ == 0.f)
         maxAgentRadius_ = navigationMesh_->GetAgentRadius();
-    if (!crowd_->init(maxAgents_, maxAgentRadius_, navigationMesh_->navMesh_, CrowdAgentUpdateCallback))
+    if (!crowd_->init(maxAgents_, maxAgentRadius_, navigationMesh_->navMesh_))
     {
         URHO3D_LOGERROR("Could not initialize DetourCrowd");
         return false;
@@ -679,6 +685,13 @@ void CrowdManager::Update(float delta)
 {
     assert(crowd_ && navigationMesh_);
     URHO3D_PROFILE(UpdateCrowd);
+
+    int nagents = crowd_->getActiveAgents(agents_, maxAgents_);
+    for (int i = 0; i < nagents; i++) {
+        dtCrowdAgent* ag = agents_[i];
+        static_cast<CrowdAgent*>(ag->params.userData)->OnCrowdUpdate(ag, delta);
+    }
+
     crowd_->update(delta, nullptr);
 }
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/CrowdManager.h Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/CrowdManager.h
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/CrowdManager.h	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/CrowdManager.h	2021-03-04 23:27:14.673865210 -0800
@@ -195,6 +195,7 @@ private:
     PODVector<unsigned> numAreas_;
     /// Number of obstacle avoidance types configured in the crowd. Limit to DT_CROWD_MAX_OBSTAVOIDANCE_PARAMS.
     unsigned numObstacleAvoidanceTypes_{};
+    dtCrowdAgent** agents_;
 };
 
 }
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/DynamicNavigationMesh.cpp	2021-03-04 22:55:48.637098600 -0800
@@ -40,11 +40,19 @@
 #include "../Scene/SceneEvents.h"
 
 #include <LZ4/lz4.h>
+#ifdef URHO3D_SYSTEM_RECASTNAVIGATION
+#include <recastnavigation/DetourNavMesh.h>
+#include <recastnavigation/DetourNavMeshBuilder.h>
+#include <recastnavigation/DetourTileCache.h>
+#include <recastnavigation/DetourTileCacheBuilder.h>
+#include <recastnavigation/Recast.h>
+#else
 #include <Detour/DetourNavMesh.h>
 #include <Detour/DetourNavMeshBuilder.h>
 #include <DetourTileCache/DetourTileCache.h>
 #include <DetourTileCache/DetourTileCacheBuilder.h>
 #include <Recast/Recast.h>
+#endif
 
 // DebugNew is deliberately not used because the macro 'free' conflicts with DetourTileCache's LinearAllocator interface
 //#include "../DebugNew.h"
@@ -1086,7 +1094,11 @@ void DynamicNavigationMesh::AddObstacle(
 
         // Because dtTileCache doesn't process obstacle requests while updating tiles
         // it's necessary update until sufficient request space is available
+#ifdef URHO3D_SYSTEM_RECASTNAVIGATION
+        for(int i=0;i<128;i++)
+#else
         while (tileCache_->isObstacleQueueFull())
+#endif
             tileCache_->update(1, navMesh_);
 
         if (dtStatusFailed(tileCache_->addObstacle(pos, obstacle->GetRadius(), obstacle->GetHeight(), &refHolder)))
@@ -1126,7 +1138,11 @@ void DynamicNavigationMesh::RemoveObstac
     {
         // Because dtTileCache doesn't process obstacle requests while updating tiles
         // it's necessary update until sufficient request space is available
+#ifdef URHO3D_SYSTEM_RECASTNAVIGATION
+        for(int i=0;i<128;i++)
+#else
         while (tileCache_->isObstacleQueueFull())
+#endif
             tileCache_->update(1, navMesh_);
 
         if (dtStatusFailed(tileCache_->removeObstacle(obstacle->obstacleId_)))
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/NavBuildData.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/NavBuildData.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/NavBuildData.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/NavBuildData.cpp	2021-03-04 22:55:48.637098600 -0800
@@ -24,8 +24,13 @@
 
 #include "../Navigation/NavBuildData.h"
 
+#ifdef URHO3D_SYSTEM_RECASTNAVIGATION
+#include <recastnavigation/DetourTileCacheBuilder.h>
+#include <recastnavigation/Recast.h>
+#else
 #include <DetourTileCache/DetourTileCacheBuilder.h>
 #include <Recast/Recast.h>
+#endif
 
 namespace Urho3D
 {
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/NavigationMesh.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/NavigationMesh.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs.orig/Source/Urho3D/Navigation/NavigationMesh.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45-native-abi_x86_64.amd64-shared-libs/Source/Urho3D/Navigation/NavigationMesh.cpp	2021-03-04 22:55:48.637098600 -0800
@@ -48,10 +48,17 @@
 #include "../Scene/Scene.h"
 
 #include <cfloat>
+#ifdef URHO3D_SYSTEM_RECASTNAVIGATION
+#include <recastnavigation/DetourNavMesh.h>
+#include <recastnavigation/DetourNavMeshBuilder.h>
+#include <recastnavigation/DetourNavMeshQuery.h>
+#include <recastnavigation/Recast.h>
+#else
 #include <Detour/DetourNavMesh.h>
 #include <Detour/DetourNavMeshBuilder.h>
 #include <Detour/DetourNavMeshQuery.h>
 #include <Recast/Recast.h>
+#endif
 
 #include "../DebugNew.h"
 
