diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Audio/Audio.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Audio/Audio.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Audio/Audio.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Audio/Audio.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -32,7 +32,11 @@
 #include "../Core/Profiler.h"
 #include "../IO/Log.h"
 
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
+#endif
 
 #include "../DebugNew.h"
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Core/Context.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Core/Context.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Core/Context.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Core/Context.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -27,7 +27,11 @@
 #include "../IO/Log.h"
 
 #ifndef MINI_URHO
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
+#endif
 #ifdef URHO3D_IK
 #include <ik/log.h>
 #include <ik/memory.h>
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Core/ProcessUtils.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Core/ProcessUtils.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Core/ProcessUtils.cpp	2021-02-25 18:53:46.508240525 -0800
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Core/ProcessUtils.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -98,8 +98,12 @@ inline void SetFPUState(unsigned control
 #endif
 
 #ifndef MINI_URHO
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
 #endif
+#endif
 
 #include "../DebugNew.h"
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Engine/Application.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Engine/Application.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Engine/Application.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Engine/Application.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -28,8 +28,12 @@
 
 #if defined(IOS) || defined(TVOS)
 #include "../Graphics/Graphics.h"
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
 #endif
+#endif
 
 #include "../DebugNew.h"
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Graphics/Direct3D9/D3D9Graphics.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Graphics/Direct3D9/D3D9Graphics.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Graphics/Direct3D9/D3D9Graphics.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Graphics/Direct3D9/D3D9Graphics.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -40,8 +40,14 @@
 #include "../../IO/Log.h"
 #include "../../Resource/ResourceCache.h"
 
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_syswm.h>
+#else
 #include <SDL/SDL.h>
 #include <SDL/SDL_syswm.h>
+#endif
+
 
 #include "../../DebugNew.h"
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -43,7 +43,11 @@
 #include "../../IO/Log.h"
 #include "../../Resource/ResourceCache.h"
 
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
+#endif
 
 #include "../../DebugNew.h"
 
@@ -421,7 +425,11 @@ bool Graphics::SetMode(int width, int he
             {
 #ifndef __EMSCRIPTEN__
                 if (!window_)
+#ifdef ENABLE_DOLLAR
                     window_ = SDL_CreateWindowFrom(externalWindow_, SDL_WINDOW_OPENGL);
+#else
+                    window_ = SDL_CreateWindowFrom(externalWindow_);
+#endif
                 fullscreen = false;
 #endif
             }
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Input/Input.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Input/Input.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Input/Input.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Input/Input.cpp	2021-02-25 18:58:03.210248848 -0800
@@ -43,7 +43,11 @@
 #endif
 
 
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
+#endif
 
 #ifdef __EMSCRIPTEN__
 #include <emscripten/html5.h>
@@ -1190,6 +1194,7 @@ void Input::SetScreenKeyboardVisible(boo
 
 void Input::SetTouchEmulation(bool enable)
 {
+#ifndef URHO3D_SYSTEM_SDL
 #if !defined(__ANDROID__) && !defined(IOS)
     if (enable != touchEmulation_)
     {
@@ -1209,6 +1214,7 @@ void Input::SetTouchEmulation(bool enabl
         touchEmulation_ = enable;
     }
 #endif
+#endif
 }
 
 bool Input::RecordGesture()
@@ -1226,13 +1232,21 @@ bool Input::RecordGesture()
 bool Input::SaveGestures(Serializer& dest)
 {
     RWOpsWrapper<Serializer> wrapper(dest);
+#ifdef ENABLE_DOLLAR
     return SDL_SaveAllDollarTemplates(wrapper.GetRWOps()) != 0;
+#else
+    return false;
+#endif
 }
 
 bool Input::SaveGesture(Serializer& dest, unsigned gestureID)
 {
     RWOpsWrapper<Serializer> wrapper(dest);
+#ifdef ENABLE_DOLLAR
     return SDL_SaveDollarTemplate(gestureID, wrapper.GetRWOps()) != 0;
+#else
+    return false;
+#endif
 }
 
 unsigned Input::LoadGestures(Deserializer& source)
@@ -1245,7 +1259,11 @@ unsigned Input::LoadGestures(Deserialize
     }
 
     RWOpsWrapper<Deserializer> wrapper(source);
+#ifdef ENABLE_DOLLAR
     return (unsigned)SDL_LoadDollarTemplates(-1, wrapper.GetRWOps());
+#else
+    return 0;
+#endif
 }
 
 
@@ -1254,15 +1272,19 @@ bool Input::RemoveGesture(unsigned gestu
 #ifdef __EMSCRIPTEN__
     return false;
 #else
+#ifdef ENABLE_DOLLAR
     return SDL_RemoveDollarTemplate(gestureID) != 0;
 #endif
+#endif
 }
 
 void Input::RemoveAllGestures()
 {
 #ifndef __EMSCRIPTEN__
+#ifdef ENABLE_DOLLAR
     SDL_RemoveAllDollarTemplates();
 #endif
+#endif
 }
 
 SDL_JoystickID Input::OpenJoystick(unsigned index)
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/IO/File.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/IO/File.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/IO/File.cpp	2021-02-25 18:53:46.524241149 -0800
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/IO/File.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -29,9 +29,14 @@
 #include "../IO/MemoryBuffer.h"
 #include "../IO/PackageFile.h"
 
+
 #ifdef __ANDROID__
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL_rwops.h>
+#else
 #include <SDL/SDL_rwops.h>
 #endif
+#endif
 
 #include <cstdio>
 #ifdef URHO3D_SYSTEM_LZ4
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/IO/FileSystem.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/IO/FileSystem.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/IO/FileSystem.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/IO/FileSystem.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -32,13 +32,22 @@
 #include "../IO/IOEvents.h"
 #include "../IO/Log.h"
 
+
 #ifdef __ANDROID__
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL_rwops.h>
+#else
 #include <SDL/SDL_rwops.h>
 #endif
+#endif
 
 #ifndef MINI_URHO
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL_filesystem.h>
+#else
 #include <SDL/SDL_filesystem.h>
 #endif
+#endif
 
 #include <sys/stat.h>
 #include <cstdio>
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/UI/Cursor.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/UI/Cursor.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/UI/Cursor.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/UI/Cursor.cpp	2021-02-25 18:56:15.070032981 -0800
@@ -29,7 +29,11 @@
 #include "../Resource/ResourceCache.h"
 #include "../UI/UI.h"
 
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL_mouse.h>
+#else
 #include <SDL/SDL_mouse.h>
+#endif
 
 #include "../DebugNew.h"
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/UI/LineEdit.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/UI/LineEdit.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/UI/LineEdit.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/UI/LineEdit.cpp	2021-02-25 18:56:15.074033137 -0800
@@ -31,7 +31,11 @@
 
 #include "../DebugNew.h"
 
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
+#endif
 
 namespace Urho3D
 {
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/UI/UI.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/UI/UI.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/UI/UI.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/UI/UI.cpp	2021-02-25 18:56:15.074033137 -0800
@@ -64,7 +64,11 @@
 #include "../UI/UIComponent.h"
 
 #include <cassert>
+#ifdef URHO3D_SYSTEM_SDL
+#include <SDL2/SDL.h>
+#else
 #include <SDL/SDL.h>
+#endif
 
 #include "../DebugNew.h"
 
