diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Resource/XMLElement.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Resource/XMLElement.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Resource/XMLElement.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Resource/XMLElement.cpp	2021-02-22 05:37:01.821376133 -0800
@@ -26,7 +26,11 @@
 #include "../IO/Log.h"
 #include "../Resource/XMLFile.h"
 
+#ifdef URHO3D_SYSTEM_PUGIXML
+#include <pugixml.hpp>
+#else
 #include <PugiXml/pugixml.hpp>
+#endif
 
 #include "../DebugNew.h"
 
diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Resource/XMLFile.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Resource/XMLFile.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig/Source/Urho3D/Resource/XMLFile.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Resource/XMLFile.cpp	2021-02-22 05:37:17.229986460 -0800
@@ -32,7 +32,11 @@
 #include "../Resource/ResourceCache.h"
 #include "../Resource/XMLFile.h"
 
+#ifdef URHO3D_SYSTEM_PUGIXML
+#include <pugixml.hpp>
+#else
 #include <PugiXml/pugixml.hpp>
+#endif
 
 #include "../DebugNew.h"
 
