diff -urp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig2/Source/Urho3D/Physics/RaycastVehicle.cpp Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Physics/RaycastVehicle.cpp
--- Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45.orig2/Source/Urho3D/Physics/RaycastVehicle.cpp	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-d34dda158ecd7694fcfd55684caade7e131b8a45/Source/Urho3D/Physics/RaycastVehicle.cpp	2021-02-22 04:51:47.682892941 -0800
@@ -28,8 +28,13 @@
 #include "../IO/Log.h"
 #include "../Physics/RaycastVehicle.h"
 
+#ifdef URHO3D_SYSTEM_BULLET
+#include <bullet/BulletDynamics/Vehicle/btRaycastVehicle.h>
+#include <bullet/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.h>
+#else
 #include <Bullet/BulletDynamics/Vehicle/btRaycastVehicle.h>
 #include <Bullet/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.h>
+#endif
 
 namespace Urho3D
 {
