--- a/examples/CMakeLists.txt.orig	2019-03-17 06:20:08.000000000 -0700
+++ b/examples/CMakeLists.txt	2020-07-18 13:47:11.235684201 -0700
@@ -41,3 +41,10 @@ target_link_libraries(strip-id3v1 tag)
 
 add_executable(inspect inspect.cpp)
 target_link_libraries(inspect tag2)
+
+INSTALL(TARGETS
+  tagreader tagreader_c tagwriter framelist strip-id3v1
+  LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+  RUNTIME DESTINATION ${BIN_INSTALL_DIR}
+  ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
+)
