diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/bindings/c/CMakeLists.txt taglib-65a6a4e225fff1288148de92721418e1c634713b/bindings/c/CMakeLists.txt
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/bindings/c/CMakeLists.txt	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/bindings/c/CMakeLists.txt	2020-07-18 15:49:59.062984557 -0700
@@ -16,21 +16,21 @@ include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}/../../taglib/trueaudio
 )
 
-set(tag_c_HDRS tag_c.h)
+set(tag2_c_HDRS tag_c.h)
 
-add_library(tag_c tag_c.cpp ${tag_c_HDRS})
+add_library(tag2_c tag_c.cpp ${tag2_c_HDRS})
 
-target_link_libraries(tag_c tag)
-set_target_properties(tag_c PROPERTIES
-  PUBLIC_HEADER "${tag_c_HDRS}"
+target_link_libraries(tag2_c tag)
+set_target_properties(tag2_c PROPERTIES
+  PUBLIC_HEADER "${tag2_c_HDRS}"
   DEFINE_SYMBOL MAKE_TAGLIB_LIB
 )
 if(VISIBILITY_HIDDEN)
-	set_target_properties(tag_c PROPERTIES C_VISIBILITY_PRESET hidden
+	set_target_properties(tag2_c PROPERTIES C_VISIBILITY_PRESET hidden
 	)
 endif()
 if(BUILD_FRAMEWORK)
-  set_target_properties(tag_c PROPERTIES FRAMEWORK TRUE)
+  set_target_properties(tag2_c PROPERTIES FRAMEWORK TRUE)
 endif()
 
 # On Solaris we need to explicitly add the C++ standard and runtime
@@ -51,25 +51,25 @@ if(HAVE_CRUN_LIB)
   if(NOT ROGUEWAVE_STDCXX_LIBRARY)
     message(FATAL_ERROR "Did not find supported STL library (tried stdcxx4 and stdcxx)")
   endif()
-  target_link_libraries(tag_c ${ROGUEWAVE_STDCXX_LIBRARY} Crun)
+  target_link_libraries(tag2_c ${ROGUEWAVE_STDCXX_LIBRARY} Crun)
 endif()
 
-set_target_properties(tag_c PROPERTIES
+set_target_properties(tag2_c PROPERTIES
   VERSION 0.0.0
   SOVERSION 0
   DEFINE_SYMBOL MAKE_TAGLIB_C_LIB
   INSTALL_NAME_DIR ${LIB_INSTALL_DIR}
 )
-install(TARGETS tag_c
+install(TARGETS tag2_c
 	FRAMEWORK DESTINATION ${FRAMEWORK_INSTALL_DIR}
 	LIBRARY DESTINATION ${LIB_INSTALL_DIR}
 	RUNTIME DESTINATION ${BIN_INSTALL_DIR}
 	ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
-	PUBLIC_HEADER DESTINATION ${INCLUDE_INSTALL_DIR}/taglib
+	PUBLIC_HEADER DESTINATION ${INCLUDE_INSTALL_DIR}/taglib2
 )
 
 if(NOT BUILD_FRAMEWORK)
-	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/taglib_c.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/taglib_c.pc)
-	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/taglib_c.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
+	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/taglib_c.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/taglib2_c.pc)
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/taglib2_c.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 endif()
 
diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/bindings/c/taglib_c.pc.cmake taglib-65a6a4e225fff1288148de92721418e1c634713b/bindings/c/taglib_c.pc.cmake
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/bindings/c/taglib_c.pc.cmake	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/bindings/c/taglib_c.pc.cmake	2020-07-18 15:49:59.062984557 -0700
@@ -8,5 +8,5 @@ Name: TagLib C Bindings
 Description: Audio meta-data library (C bindings)
 Requires: taglib
 Version: ${TAGLIB_LIB_VERSION_STRING}
-Libs: -L${LIB_INSTALL_DIR} -ltag_c
+Libs: -L${LIB_INSTALL_DIR} -ltag2_c
 Cflags: -I${INCLUDE_INSTALL_DIR}/taglib 
diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/CMakeLists.txt taglib-65a6a4e225fff1288148de92721418e1c634713b/CMakeLists.txt
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/CMakeLists.txt	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/CMakeLists.txt	2020-07-18 15:49:59.062984557 -0700
@@ -94,7 +94,7 @@ endif()
 # 2. If any interfaces have been added, removed, or changed since the last update, increment current, and set revision to 0.
 # 3. If any interfaces have been added since the last public release, then increment age.
 # 4. If any interfaces have been removed since the last public release, then set age to 0.
-set(TAGLIB_SOVERSION_CURRENT  18)
+set(TAGLIB_SOVERSION_CURRENT  2018)
 set(TAGLIB_SOVERSION_REVISION 0)
 set(TAGLIB_SOVERSION_AGE      17)
 
@@ -105,18 +105,18 @@ math(EXPR TAGLIB_SOVERSION_PATCH "${TAGL
 include(ConfigureChecks.cmake)
 
 if(NOT WIN32)
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/taglib-config.cmake" "${CMAKE_CURRENT_BINARY_DIR}/taglib-config")
-  install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/taglib-config" DESTINATION "${BIN_INSTALL_DIR}")
+  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/taglib-config.cmake" "${CMAKE_CURRENT_BINARY_DIR}/taglib2-config")
+  install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/taglib2-config" DESTINATION "${BIN_INSTALL_DIR}")
 endif()
 
 if(WIN32)
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/taglib-config.cmd.cmake" "${CMAKE_CURRENT_BINARY_DIR}/taglib-config.cmd")
-  install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/taglib-config.cmd" DESTINATION "${BIN_INSTALL_DIR}")
+  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/taglib-config.cmd.cmake" "${CMAKE_CURRENT_BINARY_DIR}/taglib2-config.cmd")
+  install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/taglib2-config.cmd" DESTINATION "${BIN_INSTALL_DIR}")
 endif()
 
 if(NOT BUILD_FRAMEWORK)
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/taglib.pc.cmake" "${CMAKE_CURRENT_BINARY_DIR}/taglib.pc" @ONLY)
-  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/taglib.pc" DESTINATION "${LIB_INSTALL_DIR}/pkgconfig")
+  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/taglib.pc.cmake" "${CMAKE_CURRENT_BINARY_DIR}/taglib2.pc" @ONLY)
+  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/taglib2.pc" DESTINATION "${LIB_INSTALL_DIR}/pkgconfig")
 endif()
 
 if(NOT HAVE_ZLIB AND ZLIB_SOURCE)
diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/examples/CMakeLists.txt taglib-65a6a4e225fff1288148de92721418e1c634713b/examples/CMakeLists.txt
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/examples/CMakeLists.txt	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/examples/CMakeLists.txt	2020-07-18 15:49:59.062984557 -0700
@@ -15,12 +15,12 @@ endif()
 ########### next target ###############
 
 add_executable(tagreader tagreader.cpp)
-target_link_libraries(tagreader tag)
+target_link_libraries(tagreader tag2)
 
 ########### next target ###############
 
 add_executable(tagreader_c tagreader_c.c)
-target_link_libraries(tagreader_c tag_c)
+target_link_libraries(tagreader_c tag2_c)
 
 ########### next target ###############
 
@@ -30,14 +30,14 @@ target_link_libraries(tagwriter tag)
 ########### next target ###############
 
 add_executable(framelist framelist.cpp)
-target_link_libraries(framelist tag)
+target_link_libraries(framelist tag2)
 
 ########### next target ###############
 
 add_executable(strip-id3v1 strip-id3v1.cpp)
-target_link_libraries(strip-id3v1 tag)
+target_link_libraries(strip-id3v1 tag2)
 
 ########### next target ###############
 
 add_executable(inspect inspect.cpp)
-target_link_libraries(inspect tag)
+target_link_libraries(inspect tag2)
diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/taglib/CMakeLists.txt taglib-65a6a4e225fff1288148de92721418e1c634713b/taglib/CMakeLists.txt
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/taglib/CMakeLists.txt	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/taglib/CMakeLists.txt	2020-07-18 15:49:59.062984557 -0700
@@ -37,7 +37,7 @@ elseif(HAVE_ZLIB_SOURCE)
   include_directories(${ZLIB_SOURCE})
 endif()
 
-set(tag_HDRS
+set(tag2_HDRS
   tag.h
   fileref.h
   audioproperties.h
@@ -362,7 +362,7 @@ if(HAVE_ZLIB_SOURCE)
   )
 endif()
 
-set(tag_LIB_SRCS
+set(tag2_LIB_SRCS
   ${mpeg_SRCS} ${id3v1_SRCS} ${id3v2_SRCS} ${frames_SRCS} ${ogg_SRCS}
   ${vorbis_SRCS} ${oggflacs_SRCS} ${mpc_SRCS} ${ape_SRCS} ${toolkit_SRCS} ${flacs_SRCS}
   ${wavpack_SRCS} ${speex_SRCS} ${trueaudio_SRCS} ${riff_SRCS} ${aiff_SRCS} ${wav_SRCS}
@@ -376,28 +376,28 @@ set(tag_LIB_SRCS
   tagutils.cpp
 )
 
-add_library(tag ${tag_LIB_SRCS} ${tag_HDRS})
+add_library(tag2 ${tag2_LIB_SRCS} ${tag2_HDRS})
 
 if(HAVE_ZLIB AND NOT HAVE_ZLIB_SOURCE)
-  target_link_libraries(tag ${ZLIB_LIBRARIES})
+  target_link_libraries(tag2 ${ZLIB_LIBRARIES})
 endif()
 
-set_target_properties(tag PROPERTIES
+set_target_properties(tag2 PROPERTIES
   VERSION ${TAGLIB_SOVERSION_MAJOR}.${TAGLIB_SOVERSION_MINOR}.${TAGLIB_SOVERSION_PATCH}
   SOVERSION ${TAGLIB_SOVERSION_MAJOR}
   INSTALL_NAME_DIR ${LIB_INSTALL_DIR}
   DEFINE_SYMBOL MAKE_TAGLIB_LIB
   LINK_INTERFACE_LIBRARIES ""
-  PUBLIC_HEADER "${tag_HDRS}"
+  PUBLIC_HEADER "${tag2_HDRS}"
 )
 if(VISIBILITY_HIDDEN)
-	set_target_properties(tag PROPERTIES C_VISIBILITY_PRESET hidden
+	set_target_properties(tag2 PROPERTIES C_VISIBILITY_PRESET hidden
 	)
 endif()
 
 if(BUILD_FRAMEWORK)
   unset(INSTALL_NAME_DIR)
-  set_target_properties(tag PROPERTIES 
+  set_target_properties(tag2 PROPERTIES 
       FRAMEWORK TRUE
       MACOSX_RPATH 1
       VERSION "A"
@@ -405,10 +405,10 @@ if(BUILD_FRAMEWORK)
       )
 endif()
 
-install(TARGETS tag
+install(TARGETS tag2
   FRAMEWORK DESTINATION ${FRAMEWORK_INSTALL_DIR}
   LIBRARY DESTINATION ${LIB_INSTALL_DIR}
-  RUNTIME DESTINATION ${BIN_INSTALL_DIR}
+  RUNTIME DESTINATION ${BIN_INSTALL_DIR}/taglib2
   ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
-  PUBLIC_HEADER DESTINATION ${INCLUDE_INSTALL_DIR}/taglib
+  PUBLIC_HEADER DESTINATION ${INCLUDE_INSTALL_DIR}/taglib2
 )
diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/taglib/toolkit/taglib.h taglib-65a6a4e225fff1288148de92721418e1c634713b/taglib/toolkit/taglib.h
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/taglib/toolkit/taglib.h	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/taglib/toolkit/taglib.h	2020-07-18 15:50:53.945094788 -0700
@@ -26,9 +26,9 @@
 #ifndef TAGLIB_H
 #define TAGLIB_H
 
-#define TAGLIB_MAJOR_VERSION 1
-#define TAGLIB_MINOR_VERSION 11
-#define TAGLIB_PATCH_VERSION 1
+#define TAGLIB_MAJOR_VERSION 2
+#define TAGLIB_MINOR_VERSION 0
+#define TAGLIB_PATCH_VERSION 20190317
 
 #if (defined(_MSC_VER) && _MSC_VER >= 1600)
 #define TAGLIB_CONSTRUCT_BITSET(x) static_cast<unsigned long long>(x)
Only in taglib-65a6a4e225fff1288148de92721418e1c634713b/taglib/toolkit: taglib.h.rej
diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/taglib.pc.cmake taglib-65a6a4e225fff1288148de92721418e1c634713b/taglib.pc.cmake
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/taglib.pc.cmake	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/taglib.pc.cmake	2020-07-18 15:50:01.313071065 -0700
@@ -7,5 +7,5 @@ Name: TagLib
 Description: Audio meta-data library
 Requires: 
 Version: @TAGLIB_LIB_VERSION_STRING@
-Libs: -L${libdir} -ltag
+Libs: -L${libdir} -ltag2
 Cflags: -I${includedir}/taglib
diff -urp taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/tests/CMakeLists.txt taglib-65a6a4e225fff1288148de92721418e1c634713b/tests/CMakeLists.txt
--- taglib-65a6a4e225fff1288148de92721418e1c634713b.orig/tests/CMakeLists.txt	2019-03-17 06:20:08.000000000 -0700
+++ taglib-65a6a4e225fff1288148de92721418e1c634713b/tests/CMakeLists.txt	2020-07-18 15:50:01.313071065 -0700
@@ -79,7 +79,7 @@ SET(test_runner_SRCS
 INCLUDE_DIRECTORIES(${CPPUNIT_INCLUDE_DIR})
 
 ADD_EXECUTABLE(test_runner ${test_runner_SRCS})
-TARGET_LINK_LIBRARIES(test_runner tag ${CPPUNIT_LIBRARIES})
+TARGET_LINK_LIBRARIES(test_runner tag2 ${CPPUNIT_LIBRARIES})
 
 ADD_TEST(test_runner test_runner)
 ADD_CUSTOM_TARGET(check COMMAND ${CMAKE_CTEST_COMMAND} -V
