--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp.orig	2024-10-01 16:22:16.763884566 -0700
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp	2024-10-01 16:24:47.306916072 -0700
@@ -1411,7 +1411,13 @@ MediaResult FFmpegVideoDecoder<LIBAV_VER
   b.mPlanes[0].mHeight = mFrame->height;
   if (mCodecContext->pix_fmt == AV_PIX_FMT_YUV444P ||
       mCodecContext->pix_fmt == AV_PIX_FMT_YUV444P10LE ||
+#if defined(FFMPEG_AV1_DECODE) \
+    && ( ( LIBAVCODEC_VERSION_MAJOR == 58 && LIBAVCODEC_VERSION_MINOR >= 91 ) \
+    ||   ( LIBAVCODEC_VERSION_MAJOR >= 59 ) )
       mCodecContext->pix_fmt == AV_PIX_FMT_GBRP ||
+#else
+      0 ||
+#endif
       mCodecContext->pix_fmt == AV_PIX_FMT_GBRP10LE
 #if LIBAVCODEC_VERSION_MAJOR >= 57
       || mCodecContext->pix_fmt == AV_PIX_FMT_YUV444P12LE
