--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp.orig	2023-02-13 22:08:53.000000000 -0800
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp	2023-02-28 15:51:49.628164799 -0800
@@ -872,7 +872,9 @@ MediaResult FFmpegVideoDecoder<LIBAV_VER
         flag |= MediaInfoFlag::VIDEO_VP9;
         break;
 #endif
-#ifdef FFMPEG_AV1_DECODE
+#if defined(FFMPEG_AV1_DECODE) \
+  || ( LIBAVCODEC_VERSION_MAJOR == 58 && LIBAVCODEC_VERSION_MINOR >= 54 ) \
+  || ( LIBAVCODEC_VERSION_MAJOR >= 59 )
       case AV_CODEC_ID_AV1:
         flag |= MediaInfoFlag::VIDEO_AV1;
         break;
@@ -1374,7 +1376,9 @@ AVCodecID FFmpegVideoDecoder<LIBAV_VER>:
   }
 #endif
 
-#if defined(FFMPEG_AV1_DECODE)
+#if defined(FFMPEG_AV1_DECODE) \
+  || ( LIBAVCODEC_VERSION_MAJOR == 58 && LIBAVCODEC_VERSION_MINOR >= 54 ) \
+  || ( LIBAVCODEC_VERSION_MAJOR >= 59 )
   if (AOMDecoder::IsAV1(aMimeType)) {
     return AV_CODEC_ID_AV1;
   }
