--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp.orig	2023-02-13 17:31:50.000000000 -0800
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp	2023-02-28 18:33:40.532290263 -0800
@@ -1054,7 +1054,9 @@ MediaResult FFmpegVideoDecoder<LIBAV_VER
 #if LIBAVCODEC_VERSION_MAJOR >= 57
       || mCodecContext->pix_fmt == AV_PIX_FMT_YUV444P12LE
 #endif
-#if defined(FFMPEG_AV1_DECODE)
+#if defined(FFMPEG_AV1_DECODE) \
+    || ( LIBAVCODEC_VERSION_MAJOR == 58 && LIBAVCODEC_VERSION_MINOR >= 91 ) \
+    || ( LIBAVCODEC_VERSION_MAJOR >= 59 )
       || mCodecContext->pix_fmt == AV_PIX_FMT_GBRP
 #endif
   ) {
@@ -1225,7 +1227,9 @@ AVCodecID FFmpegVideoDecoder<LIBAV_VER>:
   }
 #endif
 
-#if defined(FFMPEG_AV1_DECODE)
+#if defined(FFMPEG_AV1_DECODE) \
+  || ( LIBAVCODEC_VERSION_MAJOR == 58 && LIBAVCODEC_VERSION_MINOR >= 91 ) \
+  || ( LIBAVCODEC_VERSION_MAJOR >= 59 )
   if (AOMDecoder::IsAV1(aMimeType)) {
     return AV_CODEC_ID_AV1;
   }
