--- a/toolkit/moz.configure.orig	2024-04-18 16:45:56.292050101 -0700
+++ b/toolkit/moz.configure	2024-04-18 16:51:56.485062185 -0700
@@ -2438,17 +2438,29 @@ with only_when(compile_environment):
 # FFmpeg's ffvpx configuration
 # ==============================================================
 
+option("--with-ffvpx",
+    help="{Enable,Disable} the vendored ffmpeg 5.x fork called ffvpx.  Enables decode only for dav1d, flac, mp3, opus, vpx{8,9} on 64-bit; decode only for flac on 32-bit.",
+    nargs=1,
+    choices=("yes", "no", "default"),
+    default="default",
+)
 
-@depends(target)
+@depends("--with-ffvpx", target)
 def ffvpx(target):
-    use_nasm = True
+    if any("default" in value for value in values) or any("yes" in value for value in values):
+        use_nasm = True
+    else:
+        use_nasm = False
     audio_only = False
     flags = []
 
     # This enables audio and video codecs paths on Windows  x86 and x86_64,
     # macOS (all arch), and Linux x86_64. On other arch / OS combinations,
     # only audio codecs are enabled.
-    if target.kernel == "WINNT":
+    if any("no" in value for value in values):
+        # Do not set MOZ_FFVPX_AUDIOONLY preprocessor codepaths
+        pass
+    elif target.kernel == "WINNT":
         if target.cpu == "x86":
             # 32-bit windows need to prefix symbols with an underscore.
             flags = ["-DPIC", "-DWIN32", "-DPREFIX", "-Pconfig_win32.asm"]
