--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp.orig	2024-06-18 05:04:46.000000000 -0700
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp	2024-06-20 11:43:04.661097919 -0700
@@ -1411,7 +1411,13 @@ MediaResult FFmpegVideoDecoder<LIBAV_VER
   b.mPlanes[0].mHeight = mFrame->height;
   if (mCodecContext->pix_fmt == AV_PIX_FMT_YUV444P ||
       mCodecContext->pix_fmt == AV_PIX_FMT_YUV444P10LE ||
+#if defined(FFMPEG_AV1_DECODE) \
+    && ( ( LIBAVCODEC_VERSION_MAJOR == 58 && LIBAVCODEC_VERSION_MINOR >= 91 ) \
+    ||   ( LIBAVCODEC_VERSION_MAJOR >= 59 ) )
       mCodecContext->pix_fmt == AV_PIX_FMT_GBRP ||
+#else
+      0
+#endif
       mCodecContext->pix_fmt == AV_PIX_FMT_GBRP10LE
 #if LIBAVCODEC_VERSION_MAJOR >= 57
       || mCodecContext->pix_fmt == AV_PIX_FMT_YUV444P12LE
@@ -1648,7 +1654,9 @@ AVCodecID FFmpegVideoDecoder<LIBAV_VER>:
   }
 #endif
 
-#if defined(FFMPEG_AV1_DECODE)
+#if defined(FFMPEG_AV1_DECODE) \
+  && ( ( LIBAVCODEC_VERSION_MAJOR == 58 && LIBAVCODEC_VERSION_MINOR >= 91 ) \
+  ||   ( LIBAVCODEC_VERSION_MAJOR >= 59 ) )
   if (AOMDecoder::IsAV1(aMimeType)) {
     return AV_CODEC_ID_AV1;
   }
