diff -urp Box2D-2.3.1.orig/Box2D/Box2D/CMakeLists.txt Box2D-2.3.1/Box2D/Box2D/CMakeLists.txt
--- Box2D-2.3.1.orig/Box2D/Box2D/CMakeLists.txt	2014-04-05 17:43:12.000000000 -0700
+++ Box2D-2.3.1/Box2D/Box2D/CMakeLists.txt	2021-02-23 15:42:40.369872380 -0800
@@ -145,6 +145,7 @@ if(BOX2D_BUILD_SHARED)
 		OUTPUT_NAME "Box2D"
 		CLEAN_DIRECT_OUTPUT 1
 		VERSION ${BOX2D_VERSION}
+		SOVERSION ${BOX2D_VERSION_MAJOR}
 	)
 endif()
 
@@ -184,14 +185,14 @@ source_group(Rope FILES ${BOX2D_Rope_SRC
 
 if(BOX2D_INSTALL)
 	# install headers
-	install(FILES ${BOX2D_General_HDRS} DESTINATION include/Box2D)
-	install(FILES ${BOX2D_Collision_HDRS} DESTINATION include/Box2D/Collision)
-	install(FILES ${BOX2D_Shapes_HDRS} DESTINATION include/Box2D/Collision/Shapes)
-	install(FILES ${BOX2D_Common_HDRS} DESTINATION include/Box2D/Common)
-	install(FILES ${BOX2D_Dynamics_HDRS} DESTINATION include/Box2D/Dynamics)
-	install(FILES ${BOX2D_Contacts_HDRS} DESTINATION include/Box2D/Dynamics/Contacts)
-	install(FILES ${BOX2D_Joints_HDRS} DESTINATION include/Box2D/Dynamics/Joints)
-	install(FILES ${BOX2D_Rope_HDRS} DESTINATION include/Box2D/Rope)
+	install(FILES ${BOX2D_General_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR})
+	install(FILES ${BOX2D_Collision_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/Collision)
+	install(FILES ${BOX2D_Shapes_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/Collision/Shapes)
+	install(FILES ${BOX2D_Common_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/Common)
+	install(FILES ${BOX2D_Dynamics_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/Dynamics)
+	install(FILES ${BOX2D_Contacts_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/Dynamics/Contacts)
+	install(FILES ${BOX2D_Joints_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/Dynamics/Joints)
+	install(FILES ${BOX2D_Rope_HDRS} DESTINATION include/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/Rope)
 
 	# install libraries
 	if(BOX2D_BUILD_SHARED)
@@ -201,18 +202,19 @@ if(BOX2D_INSTALL)
                   RUNTIME DESTINATION bin)
 	endif()
 	if(BOX2D_BUILD_STATIC)
+	    set_target_properties(Box2D PROPERTIES SUFFIX "-${BOX2D_VERSION_MAJOR_MINOR}.a")
 		install(TARGETS Box2D EXPORT Box2D-targets DESTINATION ${LIB_INSTALL_DIR})
 	endif()
 
 	# install build system hooks for third-party apps
-	install(EXPORT Box2D-targets DESTINATION ${LIB_INSTALL_DIR}/Box2D)
+	install(EXPORT Box2D-targets DESTINATION ${LIB_INSTALL_DIR}/Box2D-${BOX2D_VERSION_MAJOR_MINOR})
 
         set (BOX2D_INCLUDE_DIR ${CMAKE_INSTALL_PREFIX}/include)
         set (BOX2D_INCLUDE_DIRS ${BOX2D_INCLUDE_DIR} )
         set (BOX2D_LIBRARY_DIRS ${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR})
         set (BOX2D_LIBRARY Box2D)
         set (BOX2D_LIBRARIES ${BOX2D_LIBRARY})
-        set (BOX2D_USE_FILE ${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}/cmake/Box2D/UseBox2D.cmake)
+        set (BOX2D_USE_FILE ${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}/cmake/Box2D-${BOX2D_VERSION_MAJOR_MINOR}/UseBox2D.cmake)
         configure_file(Box2DConfig.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/Box2DConfig.cmake @ONLY ESCAPE_QUOTES)
-        install(FILES ${CMAKE_CURRENT_BINARY_DIR}/Box2DConfig.cmake UseBox2D.cmake DESTINATION ${LIB_INSTALL_DIR}/cmake/Box2D)
+        install(FILES ${CMAKE_CURRENT_BINARY_DIR}/Box2DConfig.cmake UseBox2D.cmake DESTINATION ${LIB_INSTALL_DIR}/cmake/Box2D-${BOX2D_VERSION_MAJOR_MINOR})
 endif(BOX2D_INSTALL)
diff -urp Box2D-2.3.1.orig/Box2D/CMakeLists.txt Box2D-2.3.1/Box2D/CMakeLists.txt
--- Box2D-2.3.1.orig/Box2D/CMakeLists.txt	2014-04-05 17:43:12.000000000 -0700
+++ Box2D-2.3.1/Box2D/CMakeLists.txt	2021-02-23 15:36:28.835336477 -0800
@@ -14,7 +14,9 @@ option(BOX2D_BUILD_SHARED "Build Box2D s
 option(BOX2D_BUILD_STATIC "Build Box2D static libraries" ON)
 option(BOX2D_BUILD_EXAMPLES "Build Box2D examples" ON)
 
-set(BOX2D_VERSION 2.3.0)
+set(BOX2D_VERSION 2.3.1)
+set(BOX2D_VERSION_MAJOR_MINOR 2.3)
+set(BOX2D_VERSION_MAJOR 2)
 set(LIB_INSTALL_DIR lib${LIB_SUFFIX})
 
 # The Box2D library.
@@ -26,11 +28,9 @@ if(BOX2D_BUILD_EXAMPLES)
 
   # Testbed and dependencies.
   find_package(OpenGL REQUIRED)
-  add_subdirectory(freeglut)
-  add_subdirectory(glui)
   add_subdirectory(Testbed)
 endif(BOX2D_BUILD_EXAMPLES)
 
 if(BOX2D_INSTALL_DOC)
-  install(DIRECTORY Documentation DESTINATION share/doc/Box2D PATTERN ".svn" EXCLUDE)
+  install(DIRECTORY Documentation DESTINATION share/doc/${DOC_DEST_DIR} PATTERN ".svn" EXCLUDE)
 endif(BOX2D_INSTALL_DOC)
diff -urp Box2D-2.3.1.orig/Box2D/Testbed/CMakeLists.txt Box2D-2.3.1/Box2D/Testbed/CMakeLists.txt
--- Box2D-2.3.1.orig/Box2D/Testbed/CMakeLists.txt	2014-04-05 17:43:12.000000000 -0700
+++ Box2D-2.3.1/Box2D/Testbed/CMakeLists.txt	2021-02-23 15:36:04.602388452 -0800
@@ -3,9 +3,12 @@ add_definitions( -DFREEGLUT_EXPORTS -DFR
 
 # Define the framework files.
 set(Testbed_Framework_SRCS
+	Framework/imgui.cpp
+	Framework/DebugDraw.h
+	Framework/DebugDraw.cpp
 	Framework/Main.cpp
-	Framework/Render.cpp
-	Framework/Render.h
+	Framework/RenderGL3.cpp
+	Framework/RenderGL3.h
 	Framework/Test.cpp
 	Framework/Test.h
 )
@@ -64,11 +67,16 @@ source_group(Framework FILES ${Testbed_F
 source_group(Tests FILES ${Testbed_Tests_SRCS})
 
 include_directories (
+	Framework
 	${OPENGL_INCLUDE_DIR}
 	${Box2D_SOURCE_DIR}
 )
 
-if(APPLE)
+if(UNIX)
+	include_directories( /usr/include/X11 )
+	link_directories( /usr/lib${LIB_SUFFIX} )
+	set (OPENGL_LIBRARIES GL GLEW GLU glut X11)
+elseif(APPLE)
 	# We are not using the Apple's framework version, but X11's
 	include_directories( /usr/X11/include )
 	link_directories( /usr/X11/lib )
@@ -85,7 +93,8 @@ add_executable(Testbed
 target_link_libraries (
 	Testbed
 	Box2D
-	freeglut_static
+	freeglut
+	glfw
 	glui
 	${ADDITIONAL_LIBRARIES}
 	${OPENGL_LIBRARIES}
