diff -urp box2d-2.4.0.orig/testbed/CMakeLists.txt box2d-2.4.0/testbed/CMakeLists.txt
--- box2d-2.4.0.orig/testbed/CMakeLists.txt	2020-07-26 23:03:45.000000000 -0700
+++ box2d-2.4.0/testbed/CMakeLists.txt	2021-02-21 23:12:57.083231419 -0800
@@ -71,6 +71,7 @@ set (TESTBED_SOURCE_FILES
 
 add_executable(testbed ${TESTBED_SOURCE_FILES})
 target_include_directories(testbed PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+set_target_properties(testbed PROPERTIES LINK_FLAGS "-Wl,-rpath,${EXAMPLES_INSTALL_RPATH}")
 target_link_libraries(testbed PUBLIC box2d glfw imgui sajson glad)
 set_target_properties(testbed PROPERTIES
 	CXX_STANDARD 11
