--- a/tensorflow/tools/pip_package/setup.py.orig	2023-03-20 16:12:24.000000000 -0700
+++ b/tensorflow/tools/pip_package/setup.py	2023-07-26 14:07:04.700399539 -0700
@@ -86,7 +86,7 @@ REQUIRED_PACKAGES = [
     'astunparse >= 1.6.0',
     'flatbuffers >= 2.0',
     # TODO(b/213222745) gast versions above 0.4.0 break TF's tests
-    'gast >= 0.2.1, <= 0.4.0',
+    'gast >= 0.5.3', # oteoodoro:  allow to test python 3.10
     'google_pasta >= 0.1.1',
     'h5py >= 2.9.0',
     # TODO(b/239052279): replace with external dependency on JAX repo once JAX
@@ -94,7 +94,7 @@ REQUIRED_PACKAGES = [
     'jax >= 0.3.15',
     'libclang >= 13.0.0',
     # TODO(b/263178356): numpy 1.24 breaks TF's tests
-    'numpy >= 1.22, <1.24',
+    'numpy >= 1.22, <1.24', # otoeodoro:  distro does not have <1.24
     'opt_einsum >= 2.3.2',
     'packaging',
     # TODO(b/182876485): Protobuf 3.20 results in linker errors on Windows
