diff -urp tensorflow-2.13.0.orig/tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc tensorflow-2.13.0/tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc
--- tensorflow-2.13.0.orig/tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc	2023-06-28 11:38:35.000000000 -0700
+++ tensorflow-2.13.0/tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc	2023-08-15 20:19:49.911226701 -0700
@@ -785,7 +785,7 @@ StatusOr<std::vector<uint8_t>> EmitModul
   // Locate lld.
   // TODO(whchung@gmail.com): change to tensorflow::ROCmRoot() after
   // ROCm-Device-Libs PR.
-  std::string lld_path = tsl::io::JoinPath("/opt/rocm", "llvm/bin");
+  std::string lld_path = tsl::io::JoinPath("@EPREFIX@/usr", "llvm/bin");
   auto lld_program = llvm::sys::findProgramByName("ld.lld", {lld_path});
   if (!lld_program) {
     return xla::InternalError("unable to find ld.lld in PATH: %s",
diff -urp tensorflow-2.13.0.orig/tensorflow/compiler/xla/stream_executor/gpu/asm_compiler.cc tensorflow-2.13.0/tensorflow/compiler/xla/stream_executor/gpu/asm_compiler.cc
--- tensorflow-2.13.0.orig/tensorflow/compiler/xla/stream_executor/gpu/asm_compiler.cc	2023-06-28 11:38:35.000000000 -0700
+++ tensorflow-2.13.0/tensorflow/compiler/xla/stream_executor/gpu/asm_compiler.cc	2023-08-15 20:25:28.775215817 -0700
@@ -436,7 +436,7 @@ static std::string findRocmExecutable(co
 tsl::StatusOr<std::vector<uint8_t>> BundleGpuAsm(
     std::vector<HsacoImage> images, const std::string rocm_root_dir) {
   std::string clang_offload_bundler_path =
-      findRocmExecutable("llvm/bin/clang-offload-bundler", rocm_root_dir);
+      findRocmExecutable("lib/llvm/bin/@LLVM_SLOT@/clang-offload-bundler", rocm_root_dir);
 
   // Initialise the "--inputs" / "--targets" arguments for the
   // clang-offload-bundler with a dummy file / host target triple...
diff -urp tensorflow-2.13.0.orig/tensorflow/tsl/platform/default/rocm_rocdl_path.cc tensorflow-2.13.0/tensorflow/tsl/platform/default/rocm_rocdl_path.cc
--- tensorflow-2.13.0.orig/tensorflow/tsl/platform/default/rocm_rocdl_path.cc	2023-06-28 11:38:35.000000000 -0700
+++ tensorflow-2.13.0/tensorflow/tsl/platform/default/rocm_rocdl_path.cc	2023-08-15 20:22:32.887221466 -0700
@@ -35,6 +35,6 @@ string RocmRoot() {
 #endif
 }
 
-string RocdlRoot() { return io::JoinPath(RocmRoot(), "amdgcn/bitcode"); }
+string RocdlRoot() { return io::JoinPath(RocmRoot(), "lib/amdgcn/bitcode"); }
 
 }  // namespace tsl
