--- AppImageKit-08800854de05f4f6f7c1f3901dc165b8518822e1/CMakeLists.txt.orig	2020-08-10 01:58:48.220509810 -0700
+++ AppImageKit-08800854de05f4f6f7c1f3901dc165b8518822e1/CMakeLists.txt	2020-08-10 02:00:41.464950877 -0700
@@ -27,12 +27,13 @@ endif()
 #####################
 
 # determine Git commit ID
-execute_process(
-    COMMAND git rev-parse --short HEAD
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    OUTPUT_VARIABLE GIT_COMMIT
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
+#execute_process(
+#    COMMAND git rev-parse --short HEAD
+#    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+#    OUTPUT_VARIABLE GIT_COMMIT
+#    OUTPUT_STRIP_TRAILING_WHITESPACE
+#)
+set(GIT_COMMIT "08800854de05f4f6f7c1f3901dc165b8518822e1")
 
 # set version and build number
 set(VERSION 1-alpha)
