--- CsQuery-5a22e28a39c139cbb8170a0eeeee59e73f9e02f9.orig/source/CsQuery/CsQuery.csproj	2017-01-30 02:11:41.296605040 -0800
+++ CsQuery-5a22e28a39c139cbb8170a0eeeee59e73f9e02f9/source/CsQuery/CsQuery.csproj	2017-01-30 03:13:52.897019492 -0800
@@ -527,7 +527,7 @@
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <PropertyGroup>
-    <PostBuildEvent>if not $(ConfigurationName) ==Release GOTO finished
+    <PostBuildEvent Condition=" '$(OS)' != 'Unix' ">if not $(ConfigurationName) == Release GOTO finished
 ;
 
 if not $(SolutionName) == CsQuery GOTO finished;
@@ -543,6 +543,18 @@ xcopy $(TargetDir)CsQuery.xml $(Solution
 ;
 :finished
 </PostBuildEvent>
+  <PostBuildEvent Condition=" '$(OS)' == 'Unix' ">
+if '$(ConfigurationName)' != 'Release' ; then
+     true;
+  if '$(SolutionName)' != 'CsQuery' ; then
+    true ;
+  else
+    mkdir -p $(SolutionDir)../Distribution/ ;
+    cp $(TargetDir)CsQuery.dll $(SolutionDir)../Distribution/ ;
+    cp $(TargetDir)CsQuery.pdb $(SolutionDir)../Distribution/ ;
+    cp $(TargetDir)CsQuery.xml $(SolutionDir)../Distribution/ ;
+  fi;
+fi</PostBuildEvent>
   </PropertyGroup>
   <ProjectExtensions>
     <VisualStudio>
@@ -557,4 +564,4 @@ xcopy $(TargetDir)CsQuery.xml $(Solution
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
