diff -urp CsQuery-5a22e28a39c139cbb8170a0eeeee59e73f9e02f9.orig/source/CsQuery.sln CsQuery-5a22e28a39c139cbb8170a0eeeee59e73f9e02f9/source/CsQuery.sln
--- CsQuery-5a22e28a39c139cbb8170a0eeeee59e73f9e02f9.orig/source/CsQuery.sln	2015-12-15 05:06:00.000000000 -0800
+++ CsQuery-5a22e28a39c139cbb8170a0eeeee59e73f9e02f9/source/CsQuery.sln	2017-01-30 05:17:12.263527636 -0800
@@ -8,8 +8,6 @@ Project("{2150E333-8FDC-42A3-9474-1A3956
 		CsQuery.vsmdi = CsQuery.vsmdi
 	EndProjectSection
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Csquery.Tests", "CsQuery.Tests\Csquery.Tests.csproj", "{F8920741-0BB4-4939-A774-0F9059B2B7F0}"
-EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".nuget", ".nuget", "{2F9B2385-B8D2-4682-A76A-F308C3429A5E}"
 	ProjectSection(SolutionItems) = preProject
 		.nuget\packages.config = .nuget\packages.config
@@ -34,12 +32,6 @@ Global
 		{C38DFFE9-321B-4941-B031-29D718AB5C31}.Release|Any CPU.Build.0 = Release|Any CPU
 		{C38DFFE9-321B-4941-B031-29D718AB5C31}.Release|x64.ActiveCfg = Release|x64
 		{C38DFFE9-321B-4941-B031-29D718AB5C31}.Release|x64.Build.0 = Release|x64
-		{F8920741-0BB4-4939-A774-0F9059B2B7F0}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{F8920741-0BB4-4939-A774-0F9059B2B7F0}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{F8920741-0BB4-4939-A774-0F9059B2B7F0}.Debug|x64.ActiveCfg = Debug|Any CPU
-		{F8920741-0BB4-4939-A774-0F9059B2B7F0}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{F8920741-0BB4-4939-A774-0F9059B2B7F0}.Release|Any CPU.Build.0 = Release|Any CPU
-		{F8920741-0BB4-4939-A774-0F9059B2B7F0}.Release|x64.ActiveCfg = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
