diff -urp go-appimage-6249fabe78b8725307ba2fbcdc548b56dc70937d.orig/src/appimaged/desktop.go go-appimage-6249fabe78b8725307ba2fbcdc548b56dc70937d/src/appimaged/desktop.go
--- go-appimage-6249fabe78b8725307ba2fbcdc548b56dc70937d.orig/src/appimaged/desktop.go	2021-03-11 13:19:42.000000000 -0800
+++ go-appimage-6249fabe78b8725307ba2fbcdc548b56dc70937d/src/appimaged/desktop.go	2021-05-19 02:08:08.245094646 -0700
@@ -45,7 +45,7 @@ func writeDesktopFile(ai AppImage) {
 	var cfg *ini.File
 	ini.PrettyFormat = false
 	startingPoint := false //An easy way to tell if extracting the desktop file worked.
-	arg0abs, err := filepath.Abs(os.Args[0])
+	arg0abs, err := filepath.Abs("/usr/bin/appimaged")
 
 	// FIXME: KDE seems to have a problem when the AppImage is on a partition of which the disklabel contains "_"?
 	// Then the desktop file won't run the application
Only in go-appimage-6249fabe78b8725307ba2fbcdc548b56dc70937d/src/appimaged: desktop.go.orig
Only in go-appimage-6249fabe78b8725307ba2fbcdc548b56dc70937d/src/appimaged: desktop.go.rej
