diff -urp go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087.orig/scripts/build.sh go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087/scripts/build.sh
--- go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087.orig/scripts/build.sh	2020-08-30 22:24:56.274200899 -0700
+++ go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087/scripts/build.sh	2020-08-30 22:34:17.516338425 -0700
@@ -91,10 +91,10 @@ fi
 # Make appimagetool AppImage
 rm -rf appimagetool.AppDir || true
 mkdir -p appimagetool.AppDir/usr/bin
-( cd appimagetool.AppDir/usr/bin/ ; wget -c https://github.com/probonopd/static-tools/releases/download/continuous/desktop-file-validate-$ARCHITECTURE -O desktop-file-validate )
-( cd appimagetool.AppDir/usr/bin/ ; wget -c https://github.com/probonopd/static-tools/releases/download/continuous/mksquashfs-$ARCHITECTURE -O mksquashfs )
-( cd appimagetool.AppDir/usr/bin/ ; wget -c https://github.com/probonopd/static-tools/releases/download/continuous/patchelf-$ARCHITECTURE -O patchelf )
-( cd appimagetool.AppDir/usr/bin/ ; wget -c https://github.com/AppImage/AppImageKit/releases/download/continuous/runtime-$ARCHITECTURE -O runtime-${ARCHITECTURE2} )
+( cd appimagetool.AppDir/usr/bin/ ; cp -a "${EROOT}/usr/bin/desktop-file-validate" ./ )
+( cd appimagetool.AppDir/usr/bin/ ; cp -a "${EROOT}/usr/bin/mksquashfs" ./ )
+( cd appimagetool.AppDir/usr/bin/ ; cp -a "${EROOT}/usr/bin/patchelf" ./ )
+( cd appimagetool.AppDir/usr/bin/ ; cp -a "${EROOT}/usr/${LIBDIR}/AppImageKit/runtime-${ARCHITECTURE2}" ./ )
 ( cd appimagetool.AppDir/usr/bin/ ; wget -c https://github.com/probonopd/uploadtool/raw/master/upload.sh -O uploadtool )
 chmod +x appimagetool.AppDir/usr/bin/*
 cp appimagetool-${_GOHOSTARCH} appimagetool.AppDir/usr/bin/appimagetool
@@ -116,8 +116,8 @@ PATH=./appimagetool.AppDir/usr/bin/:$PAT
 # Make appimaged AppImage
 rm -rf appimaged.AppDir || true
 mkdir -p appimaged.AppDir/usr/bin
-( cd appimaged.AppDir/usr/bin/ ; wget -c https://github.com/probonopd/static-tools/releases/download/continuous/bsdtar-$ARCHITECTURE -O bsdtar )
-( cd appimaged.AppDir/usr/bin/ ; wget -c https://github.com/probonopd/static-tools/releases/download/continuous/unsquashfs-$ARCHITECTURE -O unsquashfs )
+( cd appimaged.AppDir/usr/bin/ ; cp -a "${EROOT}/usr/bin/bsdtar" ./ )
+( cd appimaged.AppDir/usr/bin/ ; cp -a "${EROOT}/usr/bin/unsquashfs" ./ )
 chmod +x appimaged.AppDir/usr/bin/*
 cp appimaged-${_GOHOSTARCH} appimaged.AppDir/usr/bin/appimaged
 ( cd appimaged.AppDir/ ; ln -s usr/bin/appimaged AppRun)
