diff -urp go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087.orig/src/appimaged/appimaged.go go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087/src/appimaged/appimaged.go
--- go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087.orig/src/appimaged/appimaged.go	2020-08-16 12:45:05.000000000 -0700
+++ go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087/src/appimaged/appimaged.go	2020-08-28 21:05:38.518185293 -0700
@@ -12,6 +12,7 @@ import (
 	"strconv"
 	"strings"
 	"sync"
+	"syscall"
 	"time"
 
 	"github.com/adrg/xdg"
@@ -405,7 +406,18 @@ func watchDirectories() {
 	mounts, _ := procfs.GetMounts()
 	// FIXME: This breaks when the partition label has "-", see https://github.com/prometheus/procfs/issues/227
 
+	os_type := runtime.GOOS
 	for _, mount := range mounts {
+		if os_type == "linux" {
+			info, errormp := os.Stat(mount.MountPoint)
+			if errormp != nil {
+				continue
+			}
+			sys := info.Sys().(*syscall.Stat_t)
+			if int(sys.Uid) != os.Getuid() && int(sys.Gid) != os.Getgid() {
+				continue
+			}
+		}
 		if *verbosePtr == true {
 			log.Println("main: MountPoint", mount.MountPoint)
 		}
diff -urp go_build.orig/src/github.com/probonopd/go-appimage/src/appimaged/appimaged.go go_build/src/github.com/probonopd/go-appimage/src/appimaged/appimaged.go
--- go_build.orig/src/github.com/probonopd/go-appimage/src/appimaged/appimaged.go	2020-08-17 04:30:51.512523615 -0700
+++ go_build/src/github.com/probonopd/go-appimage/src/appimaged/appimaged.go	2020-08-17 04:30:16.991172642 -0700
@@ -12,6 +12,7 @@ import (
 	"strconv"
 	"strings"
 	"sync"
+	"syscall"
 	"time"
 
 	"github.com/adrg/xdg"
@@ -405,7 +406,18 @@ func watchDirectories() {
 	mounts, _ := procfs.GetMounts()
 	// FIXME: This breaks when the partition label has "-", see https://github.com/prometheus/procfs/issues/227
 
+	os_type := runtime.GOOS
 	for _, mount := range mounts {
+		if os_type == "linux" {
+			info, errormp := os.Stat(mount.MountPoint)
+			if errormp != nil {
+				continue
+			}
+			sys := info.Sys().(*syscall.Stat_t)
+			if int(sys.Uid) != os.Getuid() && int(sys.Gid) != os.Getgid() {
+				continue
+			}
+		}
 		if *verbosePtr == true {
 			log.Println("main: MountPoint", mount.MountPoint)
 		}
