diff -urp go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087.orig/src/appimaged/prerequisites.go go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087/src/appimaged/prerequisites.go
--- go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087.orig/src/appimaged/prerequisites.go	2020-08-16 12:45:05.000000000 -0700
+++ go-appimage-0c02e035e6a93d6e9bb6e3c5623dcc4333540087/src/appimaged/prerequisites.go	2020-08-17 18:58:20.116592599 -0700
@@ -133,7 +133,11 @@ func checkPrerequisites() {
 }
 
 func checkIfSystemdServiceRunning(servicenames []string) bool {
-
+	_, err0 := os.Stat("/lib/systemd/systemd")
+	_, err1 := os.Stat("/usr/lib/systemd/systemd")
+	if (os.IsNotExist(err0) && os.IsNotExist(err1)) {
+		return false
+	}
 	conn, err := systemddbus.NewUserConnection()
 	defer conn.Close()
 	helpers.PrintError("pre: checkIfSystemdServiceRunning", err)
