diff -urp go-appimage.orig/src/appimaged/prerequisites.go go-appimage/src/appimaged/prerequisites.go
--- go-appimage.orig/src/appimaged/prerequisites.go	2023-01-15 01:29:09.668392313 -0800
+++ go-appimage/src/appimaged/prerequisites.go	2023-01-15 01:32:34.336512919 -0800
@@ -136,6 +136,12 @@ func checkPrerequisites() {
 
 func checkIfSystemdServiceRunning(servicenames []string) bool {
 
+	_, err0 := os.Stat("/lib/systemd/systemd")
+	_, err1 := os.Stat("/usr/lib/systemd/systemd")
+	if (os.IsNotExist(err0) && os.IsNotExist(err1)) {
+		return false
+	}
+
 	cont := context.Background()
 
 	conn, err := systemddbus.NewUserConnectionContext(cont)
Only in go-appimage/src/appimaged: prerequisites.go.orig
Only in go-appimage/src/appimaged: prerequisites.go.rej
