--- AKv8/client/main.cpp.orig	2015-12-03 18:18:04.831077485 -0800
+++ AKv8/client/main.cpp	2015-12-03 18:50:50.486142695 -0800
@@ -391,7 +391,7 @@
     }
 
 
-#ifdef BOINC_APP_GRAPHICS
+#ifdef BOINC_APP_GRAPHICS_OLD
     if (!nographics_flag) {
       run_stage=GRXINIT;
 #ifdef DYNAMIC_GRAPHICS
@@ -406,6 +406,9 @@
       run_stage=POSTINIT;
 //fprintf(stderr,"before boinc_init\n");
 {
+#ifdef BOINC_APP_GRAPHICS
+    sah_graphics_init(app_init_data);
+#endif
 #if USE_OPENCL || USE_CUDA
 	BOINC_OPTIONS options;
 	boinc_options_defaults(options);
