--- AKv8/client/sah_gfx.h.orig	2015-12-04 10:58:24.784318018 -0800
+++ AKv8/client/sah_gfx.h	2015-12-04 10:58:43.295035506 -0800
@@ -67,6 +67,7 @@
     static SAH_GRAPHICS_BASE *get_sah_graphics();
 };
 
+#if 0
 #if !defined(DYNAMIC_GRAPHICS) || defined(GUISO)
 extern "C" {
   SAH_GRAPHICS_BASE *get_sah_graphics();
@@ -81,6 +82,7 @@
 extern vvfunc_t free_sah_graphics;
 #endif
 #endif
+#endif
 
 #endif
 
