--- AKv8/client/sah_gfx.cpp.orig	2015-12-05 22:02:05.552174412 -0800
+++ AKv8/client/sah_gfx.cpp	2015-12-05 22:04:17.213480065 -0800
@@ -259,7 +259,7 @@
 		sprintf(buf, "%s\n"
 			"Doppler drift rate %6.4f Hz/sec  Resolution %5.3f Hz\n",
 			status,
-            fft_info.chirp_rate, cnvt_fftlen_hz(fft_info.fft_len)
+            fft_info.chirp_rate, wu.subband_sample_rate/fft_info.fft_len
         );
 	} else {
 		strcpy( buf, "" );
