--- src/tools/vmap-assembler/CMakeLists.txt.orig	2015-06-20 15:32:18.394138560 -0700
+++ src/tools/vmap-assembler/CMakeLists.txt	2015-06-20 15:35:57.893353667 -0700
@@ -50,6 +50,24 @@
 
 target_link_libraries(${EXECUTABLE_NAME} vmap g3dlite zlib)
 
+set(EXECUTABLE_LINK_FLAGS "")
+
+if(UNIX)
+  if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+    set(EXECUTABLE_LINK_FLAGS "-Wl, --no-as-needed -pthread -lrt ${EXECUTABLE_LINK_FLAGS}")
+  elseif(CMAKE_SYSTEM_NAME MATCHES "Linux")
+    set(EXECUTABLE_LINK_FLAGS "-Wl,--no-as-needed -ldl -pthread -lrt ${EXECUTABLE_LINK_FLAGS}")
+  endif()
+endif()
+
+if(APPLE)
+  set(EXECUTABLE_LINK_FLAGS "-framework Carbon ${EXECUTABLE_LINK_FLAGS}")
+endif()
+
+set_target_properties(${EXECUTABLE_NAME} PROPERTIES LINK_FLAGS
+  "${EXECUTABLE_LINK_FLAGS}"
+)
+
 #Output the compiled exes to build/bin/$(Configuration)/tools directory on windows by default
 if(WIN32)
   if ( MSVC )
