--- ./src/shared/Log.h.orig	2015-06-21 13:21:28.204657528 -0700
+++ ./src/shared/Log.h	2015-06-21 13:34:35.138640263 -0700
@@ -146,6 +146,10 @@
                 { fclose(dberLogfile); }
             dberLogfile = NULL;
 
+            if (elunaErrLogfile != NULL)
+                fclose(elunaErrLogfile);
+            elunaErrLogfile = NULL;
+
             if (eventAiErLogfile != NULL)
                 { fclose(eventAiErLogfile); }
             eventAiErLogfile = NULL;
@@ -234,7 +238,18 @@
          * @param str...
          */
         void outChar(const char* str, ...)        ATTR_PRINTF(2, 3);
-
+        /**
+         * @brief any log level
+         *
+         * @param str...
+         */
+        void outErrorEluna();
+        /**
+         * @brief any log level
+         *
+         * @param str...
+         */
+        void outErrorEluna(const char* str, ...)        ATTR_PRINTF(2, 3);
         /**
          * @brief any log level
          *
@@ -403,6 +418,7 @@
         FILE* gmLogfile; /**< TODO */
         FILE* charLogfile; /**< TODO */
         FILE* dberLogfile; /**< TODO */
+	FILE* elunaErrLogfile; /**< TODO */
         FILE* eventAiErLogfile; /**< TODO */
         FILE* scriptErrLogFile; /**< TODO */
         FILE* worldLogfile; /**< TODO */
