diff -urp work.orig/src/game/WorldHandlers/World.cpp work/src/game/WorldHandlers/World.cpp
--- work.orig/src/game/WorldHandlers/World.cpp	2017-02-24 10:14:26.236759911 -0800
+++ work/src/game/WorldHandlers/World.cpp	2017-02-24 13:31:43.992734653 -0800
@@ -1889,7 +1889,7 @@ void World::ServerMaintenanceStart()
         if (itr->second->GetPlayer() && itr->second->GetPlayer()->IsInWorld())
             { itr->second->GetPlayer()->SaveToDB(); }
 
-    CharacterDatabase.PExecute("UPDATE saved_variables SET NextMaintenanceDate = '"UI64FMTD"'", uint64(m_NextMaintenanceDate));
+    CharacterDatabase.PExecute("UPDATE saved_variables SET NextMaintenanceDate = '" UI64FMTD "'", uint64(m_NextMaintenanceDate));
 }
 
 void World::InitServerMaintenanceCheck()
@@ -1900,7 +1900,7 @@ void World::InitServerMaintenanceCheck()
         DEBUG_LOG("Maintenance date not found in SavedVariables, reseting it now.");
         uint32 mDate = GetDateLastMaintenanceDay();
         m_NextMaintenanceDate = mDate == GetDateToday() ?  mDate : mDate + 7;
-        CharacterDatabase.PExecute("INSERT INTO saved_variables (NextMaintenanceDate) VALUES ('"UI64FMTD"')", uint64(m_NextMaintenanceDate));
+        CharacterDatabase.PExecute("INSERT INTO saved_variables (NextMaintenanceDate) VALUES ('" UI64FMTD "')", uint64(m_NextMaintenanceDate));
     }
     else
     {
