diff -urp work.orig/src/game/MotionGenerators/PathFinder.cpp work/src/game/MotionGenerators/PathFinder.cpp
--- work.orig/src/game/MotionGenerators/PathFinder.cpp	2017-02-23 20:45:20.553658048 -0800
+++ work/src/game/MotionGenerators/PathFinder.cpp	2017-02-23 20:53:11.727823979 -0800
@@ -29,6 +29,7 @@
 #include "Creature.h"
 #include "PathFinder.h"
 #include "Log.h"
+#include "Map.h"
 
 ////////////////// PathFinder //////////////////
 PathFinder::PathFinder(const Unit* owner) :
