diff -urp work.orig/src/game/Object/PlayerLogger.h work/src/game/Object/PlayerLogger.h
--- work.orig/src/game/Object/PlayerLogger.h	2017-02-23 21:15:28.916844776 -0800
+++ work/src/game/Object/PlayerLogger.h	2017-02-23 21:27:18.672691526 -0800
@@ -25,6 +25,8 @@
 #ifndef MANGOS_H_PLAYERLOGGER
 #define MANGOS_H_PLAYERLOGGER
 
+#include "Log.h"
+
 enum PlayerLogEntity
 {
     PLAYER_LOG_DAMAGE_GET   = 0,
@@ -196,4 +198,4 @@ private:
     uint8 logActiveMask;
 };
 
-#endif
\ No newline at end of file
+#endif
