--- src/tools/Movemap-Generator/CMakeLists.txt.orig	2015-06-20 15:52:33.780620072 -0700
+++ src/tools/Movemap-Generator/CMakeLists.txt	2015-06-20 15:53:41.160139071 -0700
@@ -69,6 +69,24 @@
 
 target_link_libraries(${EXECUTABLE_NAME} g3dlite vmap detour recast zlib shared)
 
+set(EXECUTABLE_LINK_FLAGS "")
+
+if(UNIX)
+  if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+    set(EXECUTABLE_LINK_FLAGS "-Wl, --no-as-needed -pthread -lrt ${EXECUTABLE_LINK_FLAGS}")
+  elseif(CMAKE_SYSTEM_NAME MATCHES "Linux")
+    set(EXECUTABLE_LINK_FLAGS "-Wl,--no-as-needed -ldl -pthread -lrt ${EXECUTABLE_LINK_FLAGS}")
+  endif()
+endif()
+
+if(APPLE)
+  set(EXECUTABLE_LINK_FLAGS "-framework Carbon ${EXECUTABLE_LINK_FLAGS}")
+endif()
+
+set_target_properties(${EXECUTABLE_NAME} PROPERTIES LINK_FLAGS
+  "${EXECUTABLE_LINK_FLAGS}"
+)
+
 #Output the compiled exes to build/bin/$(Configuration)/tools directory on windows by default
 if(WIN32)
   if ( MSVC )
