--- src/game/Server/SharedDefines.h.orig	2015-06-21 14:48:12.298334178 -0700
+++ src/game/Server/SharedDefines.h	2015-06-21 14:51:18.388420001 -0700
@@ -730,6 +730,16 @@
 
 #define LANGUAGES_COUNT   21
 
+enum TeamId
+{
+    TEAM_ALLIANCE = 0,
+    TEAM_HORDE,
+    TEAM_NEUTRAL
+};
+
+// Temporarily define TEAM_INDEX_NEUTRAL till BG update is done
+#define TEAM_INDEX_NEUTRAL TEAM_NEUTRAL
+
 // In fact !=0 values is alliance/horde root faction ids
 enum Team
 {
