diff -urp howdy-3.0_beta_pre9999.orig/howdy-gtk/src/init.py howdy-3.0_beta_pre9999/howdy-gtk/src/init.py
--- howdy-3.0_beta_pre9999.orig/howdy-gtk/src/init.py	2023-11-23 16:34:11.016708851 -0800
+++ howdy-3.0_beta_pre9999/howdy-gtk/src/init.py	2023-11-23 16:34:49.028709863 -0800
@@ -2,7 +2,4 @@
 # Opens auth ui if requested, otherwise starts normal ui
 import sys
 
-if "--start-auth-ui" in sys.argv:
-	import authsticky
-else:
-	import window
+import window
