diff -urp genkernel-4.2.6.orig/defaults/linuxrc genkernel-4.2.6/defaults/linuxrc
--- genkernel-4.2.6.orig/defaults/linuxrc	2022-02-05 19:03:05.334025862 -0800
+++ genkernel-4.2.6/defaults/linuxrc	2022-02-05 19:04:24.879229307 -0800
@@ -645,7 +645,10 @@ fi
 # Initialize LUKS root device except for livecd's
 if [ "${CDROOT}" != '1' ]
 then
-	start_entry
+	if [ -z "${CRYPT_ROOT_PLAIN}" ]
+	then
+		start_entry
+	fi
 	start_LUKS
 	if [ "${NORESUME}" != '1' ] && [ -n "${REAL_RESUME}" ]
 	then
@@ -1037,7 +1040,10 @@ then
 
 	cache_cd_contents
 
-	start_entry
+	if [ -z "${CRYPT_ROOT_PLAIN}" ]
+	then
+		start_entry
+	fi
 
 	# If encrypted, find key and mount, otherwise mount as usual
 	if [ -n "${CRYPT_ROOT}" ]
