--- a/gen_initramfs.sh.orig	2022-09-25 00:14:33.314392452 -0700
+++ b/gen_initramfs.sh	2022-09-25 01:10:13.413758773 -0700
@@ -654,6 +654,24 @@ append_eudev() {
 	echo "" > usr/lib/udev/rules.d/80-net-name-slot.rules \
 		|| gen_die "Failed to disable predictable network interface naming rule"
 
+	local udevdir=$(get_udevdir)
+	local udevdir_initramfs="/usr/lib/udev"
+	local udev_files=(
+		$(find /etc/genkernel/udev/rules.d -name "*.rules")
+		$(find "${udevdir}" -name "60-persistent-storage.rules")
+	)
+
+	mkdir -p "${TDIR}"/usr/lib/udev/rules.d || gen_die "Failed to create '${TDIR}/usr/lib/udev/rules.d'!"
+
+	local udev_files
+	for udev_file in "${udev_files[@]}"
+	do
+		print_info 5 "Adding ${udev_file}"
+		local dest_file="${TDIR%/}${udev_file/${udevdir}/${udevdir_initramfs}}"
+		cp -aL "${udev_file}" "${dest_file}" \
+			|| gen_die "Failed to copy '${udev_file}' to '${dest_file}'"
+	done
+
 	log_future_cpio_content
 	find . -print0 | "${CPIO_COMMAND}" ${CPIO_ARGS} --append -F "${CPIO_ARCHIVE}" \
 		|| gen_die "Failed to append ${PN} to cpio!"
