diff -urp rccl-rocm-5.3.3.orig/test/CMakeLists.txt rccl-rocm-5.3.3/test/CMakeLists.txt
--- rccl-rocm-5.3.3.orig/test/CMakeLists.txt	2022-10-05 10:08:56.000000000 -0700
+++ rccl-rocm-5.3.3/test/CMakeLists.txt	2023-07-26 08:16:17.075851341 -0700
@@ -5,10 +5,10 @@ if(BUILD_TESTS)
 
   message("Building unit tests (Installed in /test/UnitTests)")
 
-  find_program(CHRPATH chrpath)
-  if(NOT CHRPATH)
-      message(FATAL_ERROR "chrpath is required for UnitTests. Please install (e.g. sudo apt-get install chrpath)")
-  endif()
+  #find_program(CHRPATH chrpath)
+  #if(NOT CHRPATH)
+  #    message(FATAL_ERROR "chrpath is required for UnitTests. Please install (e.g. sudo apt-get install chrpath)")
+  #endif()
 
   find_package(hsa-runtime64 PATHS /opt/rocm )
   if(${hsa-runtime64_FOUND})
