diff -urp rccl-rocm-5.6.0.orig/cmake/Dependencies.cmake rccl-rocm-5.6.0/cmake/Dependencies.cmake
--- rccl-rocm-5.6.0.orig/cmake/Dependencies.cmake	2023-06-14 09:24:41.000000000 -0700
+++ rccl-rocm-5.6.0/cmake/Dependencies.cmake	2023-08-21 11:27:09.857141084 -0700
@@ -60,10 +60,8 @@ if(NOT GTest_FOUND AND BUILD_TESTS OR IN
                      UPDATE_DISCONNECTED TRUE
     )
     set(GTEST_INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/gtest/include CACHE PATH "")
-    if(EXISTS ${CMAKE_CURRENT_BINARY_DIR}/gtest/lib)
-        set(GTEST_BOTH_LIBRARIES ${CMAKE_CURRENT_BINARY_DIR}/gtest/lib/libgtest.a;${CMAKE_CURRENT_BINARY_DIR}/gtest/lib/libgtest_main.a CACHE PATH "")
-    elseif(EXISTS ${CMAKE_CURRENT_BINARY_DIR}/gtest/lib64)
-        set(GTEST_BOTH_LIBRARIES ${CMAKE_CURRENT_BINARY_DIR}/gtest/lib64/libgtest.a;${CMAKE_CURRENT_BINARY_DIR}/gtest/lib64/libgtest_main.a CACHE PATH "")
+    if(EXISTS ${CMAKE_CURRENT_BINARY_DIR}/gtest/@LIBDIR@)
+        set(GTEST_BOTH_LIBRARIES ${CMAKE_CURRENT_BINARY_DIR}/gtest/@LIBDIR@/libgtest.a;${CMAKE_CURRENT_BINARY_DIR}/gtest/@LIBDIR@/libgtest_main.a CACHE PATH "")
     else()
         message(FATAL_ERROR "Cannot find gtest library installation path.")
     find_package(GTest REQUIRED CONFIG PATHS ${GTEST_ROOT})
diff -urp rccl-rocm-5.6.0.orig/CMakeLists.txt rccl-rocm-5.6.0/CMakeLists.txt
--- rccl-rocm-5.6.0.orig/CMakeLists.txt	2023-08-21 11:23:57.765136318 -0700
+++ rccl-rocm-5.6.0/CMakeLists.txt	2023-08-21 11:24:45.393137499 -0700
@@ -511,7 +511,7 @@ target_link_libraries(rccl INTERFACE hip
 
 if(HAVE_BFD)
   target_link_libraries(rccl PRIVATE bfd dl z)
-  find_library(HAVE_IBERTY iberty PATHS /usr/lib64 /usr/lib/
+  find_library(HAVE_IBERTY iberty PATHS /usr/@LIBDIR@
     PATH_SUFFIXES x86_64-linux-gnu)
   if(HAVE_IBERTY)
     message("iberty found @  ${HAVE_IBERTY} ")
diff -urp rccl-rocm-5.6.0.orig/test/CMakeLists.txt rccl-rocm-5.6.0/test/CMakeLists.txt
--- rccl-rocm-5.6.0.orig/test/CMakeLists.txt	2023-08-21 11:23:57.741136317 -0700
+++ rccl-rocm-5.6.0/test/CMakeLists.txt	2023-08-21 11:27:20.481141347 -0700
@@ -25,7 +25,7 @@ if(BUILD_TESTS)
     INCLUDE_DIRECTORIES(${ROCR_HDR})
 
     # Search for ROCr library file in user defined locations
-    find_library(ROCR_LIB ${CORE_RUNTIME_TARGET} PATHS ${ROCR_LIB_DIR} "/opt/rocm" PATH_SUFFIXES lib lib64 REQUIRED)
+    find_library(ROCR_LIB ${CORE_RUNTIME_TARGET} PATHS ${ROCR_LIB_DIR} "/opt/rocm" PATH_SUFFIXES @LIBDIR@ REQUIRED)
   endif()
 
   include_directories(${GTEST_INCLUDE_DIRS} ./common)
