diff -urp -x '*.orig' -x '*.rej' slntools-705869e96a2f0e401be03f8e8478df3e1f2b9373.orig/Main/SLNTools.exe/CommandLine/CommandLineParser.cs slntools-705869e96a2f0e401be03f8e8478df3e1f2b9373/Main/SLNTools.exe/CommandLine/CommandLineParser.cs
--- slntools-705869e96a2f0e401be03f8e8478df3e1f2b9373.orig/Main/SLNTools.exe/CommandLine/CommandLineParser.cs	2015-08-16 14:08:39.000000000 -0700
+++ slntools-705869e96a2f0e401be03f8e8478df3e1f2b9373/Main/SLNTools.exe/CommandLine/CommandLineParser.cs	2017-01-02 05:43:41.351019015 -0800
@@ -544,6 +544,8 @@ namespace CWDev.SLNTools.CommandLine
         /// <returns>Returns the number of columns in the current console window</returns>
         public static int GetConsoleWindowWidth()
         {
+            return Console.WindowWidth;
+/*
             int screenWidth;
             CONSOLE_SCREEN_BUFFER_INFO csbi = new CONSOLE_SCREEN_BUFFER_INFO();
 
@@ -551,6 +553,7 @@ namespace CWDev.SLNTools.CommandLine
             rc = GetConsoleScreenBufferInfo(GetStdHandle(STD_OUTPUT_HANDLE), ref csbi);
             screenWidth = csbi.dwSize.x;
             return screenWidth;
+*/
         }
 
         /// <summary>
@@ -1456,4 +1459,4 @@ namespace CWDev.SLNTools.CommandLine
         private Argument defaultArgument;
         private ErrorReporter reporter;
     }
-}
\ No newline at end of file
+}
