--- ./external/llvm-project/mlir/cmake/modules/AddMLIR.cmake.orig	2022-01-20 11:34:11.000000000 -0800
+++ ./external/llvm-project/mlir/cmake/modules/AddMLIR.cmake	2023-08-07 11:10:17.859127330 -0700
@@ -150,9 +150,9 @@ function(add_mlir_library_install name)
   install(TARGETS ${name}
     COMPONENT ${name}
     ${export_to_mlirtargets}
-    LIBRARY DESTINATION lib${LLVM_LIBDIR_SUFFIX}
-    ARCHIVE DESTINATION lib${LLVM_LIBDIR_SUFFIX}
-    RUNTIME DESTINATION bin)
+    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 
   if (NOT LLVM_ENABLE_IDE)
     add_llvm_install_targets(install-${name}
