diff -urp rocMLIR-rocm-5.2.3.orig/cmake/llvm-project.cmake rocMLIR-rocm-5.2.3/cmake/llvm-project.cmake
--- rocMLIR-rocm-5.2.3.orig/cmake/llvm-project.cmake	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/cmake/llvm-project.cmake	2023-10-04 11:37:12.707021538 -0700
@@ -3,7 +3,7 @@ message(STATUS "Adding LLVM git-submodul
 # Passed to lit.site.cfg.py.so that the out of tree Standalone dialect test
 # can find MLIR's CMake configuration
 set(MLIR_CMAKE_CONFIG_DIR
-   "${CMAKE_BINARY_DIR}/lib${LLVM_LIBDIR_SUFFIX}/cmake/mlir")
+   "${CMAKE_BINARY_DIR}/lib${LLVM_LIBDIR_SUFFIX}/cmake/mlir") # oteodoro:  review
 
 # MLIR settings
 set(MLIR_TABLEGEN_EXE mlir-tblgen)
@@ -19,7 +19,7 @@ set(LLVM_PROJ_SRC "${CMAKE_SOURCE_DIR}/e
 # Configure ROCm support.
 if (NOT DEFINED ROCM_PATH)
   if (NOT DEFINED ENV{ROCM_PATH})
-    set(ROCM_PATH "/opt/rocm" CACHE PATH "Path to which ROCm has been installed")
+    set(ROCM_PATH "@ESYSROOT_ROCM_PATH@" CACHE PATH "Path to which ROCm has been installed") # oteodoro:  edited
   else()
     set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Path to which ROCm has been installed")
   endif()
@@ -28,13 +28,16 @@ message(STATUS "ROCM_PATH: ${ROCM_PATH}"
 
 # Cmake module paths
 list(APPEND CMAKE_MODULE_PATH
-  "${ROCM_PATH}/hip/cmake"
+  "${ROCM_PATH}/@LIBDIR@/cmake/hip" # oteodoro:  edited
 )
 list(APPEND CMAKE_MODULE_PATH
-  "${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/mlir"
+  "${CMAKE_CURRENT_SOURCE_DIR}/external/llvm-project/mlir/cmake/modules" # oteodoro:  added
+  #"${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@/cmake/mlir" # oteodoro:  edited.  Using @LIBDIR@ breaks AddMLIR cmake detect.
 )
 list(APPEND CMAKE_MODULE_PATH
-  "${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/lib/cmake/llvm/"
+  "${CMAKE_CURRENT_SOURCE_DIR}/external/llvm-project/mlir/cmake/modules" # oteodoro:  added
+  "${CMAKE_CURRENT_SOURCE_DIR}/external/llvm-project/llvm/cmake/modules" # oteodoro:  added
+  #"${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/@LIBDIR@/cmake/llvm/" # oteodoro:  edited.  Using @LIBDIR@ breaks TableGen, AddLLVM, LLVMDistributionSupport, HandleLLVMOptions cmake detection
 )
 
 # Include dirs for MLIR and LLVM
@@ -48,8 +51,6 @@ list(APPEND LLVM_INCLUDE_DIRS
 )
 
 # Linker flags
-list(APPEND CMAKE_EXE_LINKER_FLAGS
-  " -Wl,-rpath -Wl,${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/lib"
-)
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-rpath -Wl,${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/@LIBDIR@") # oteodoro - backport from 5.4.3
 
 add_subdirectory("${LLVM_PROJ_SRC}/llvm" "external/llvm-project/llvm" EXCLUDE_FROM_ALL)
diff -urp rocMLIR-rocm-5.2.3.orig/CMakeLists.txt rocMLIR-rocm-5.2.3/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/CMakeLists.txt	2023-10-04 11:37:12.719021538 -0700
@@ -43,9 +43,9 @@ set(LLVM_TARGETS_TO_BUILD "X86;AMDGPU" C
 # Pointers to: 1) external LLVM bins/libs, and 2) MIOpen Dialect bins/libs
 set(LLVM_MAIN_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/external/llvm-project/llvm" CACHE PATH "Path to LLVM sources")
 set(LLVM_EXTERNAL_BIN_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/bin" CACHE PATH "")
-set(LLVM_EXTERNAL_LIB_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/lib" CACHE PATH "")
+set(LLVM_EXTERNAL_LIB_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/@LIBDIR@" CACHE PATH "") # oteodoro:  edited
 set(MLIR_MIOPEN_BIN_DIR "${CMAKE_CURRENT_BINARY_DIR}/bin" CACHE PATH "")
-set(MLIR_MIOPEN_LIB_DIR "${CMAKE_CURRENT_BINARY_DIR}/lib" CACHE PATH "")
+set(MLIR_MIOPEN_LIB_DIR "${CMAKE_CURRENT_BINARY_DIR}/@LIBDIR@" CACHE PATH "") # oteodoro:  edited
 message(STATUS "LLVM_EXTERNAL_BIN_DIR: ${LLVM_EXTERNAL_BIN_DIR}")
 message(STATUS "MLIR_MIOPEN_BIN_DIR: ${MLIR_MIOPEN_BIN_DIR}")
 
@@ -68,7 +68,7 @@ if( BUILD_FAT_LIBMLIRMIOPEN )
   # rocm-runner is not supported with static libraries
   set(MLIR_ENABLE_ROCM_RUNNER 0 CACHE BOOL "")
   set(MLIR_INCLUDE_INTEGRATION_TESTS OFF CACHE BOOL "")
-  set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE STRING "")
+  set(CMAKE_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE STRING "") # oteodoro:  edited
 else()
   set(BUILD_SHARED_LIBS ON CACHE BOOL "")
   set(LLVM_BUILD_LLVM_DYLIB ON CACHE BOOL "")
diff -urp rocMLIR-rocm-5.2.3.orig/external/llvm-project/clang/lib/Driver/ToolChains/AMDGPU.cpp rocMLIR-rocm-5.2.3/external/llvm-project/clang/lib/Driver/ToolChains/AMDGPU.cpp
--- rocMLIR-rocm-5.2.3.orig/external/llvm-project/clang/lib/Driver/ToolChains/AMDGPU.cpp	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/external/llvm-project/clang/lib/Driver/ToolChains/AMDGPU.cpp	2023-10-04 11:37:12.743021538 -0700
@@ -238,8 +238,8 @@ RocmInstallationDetector::getInstallatio
       }
     }
 
-    // Some versions of the rocm llvm package install to /opt/rocm/llvm/bin
-    // Some versions of the aomp package install to /opt/rocm/aomp/bin
+    // Some versions of the rocm llvm package install to @EPREFIX_LLVM_PATH@/bin  // oteodoro: edited
+    // Some versions of the aomp package install to /opt/aomp/@LLVM_SLOT@/bin           // oteodoro: edited
     if (ParentName == "llvm" || ParentName.startswith("aomp"))
       ParentDir = llvm::sys::path::parent_path(ParentDir);
 
@@ -267,9 +267,11 @@ RocmInstallationDetector::getInstallatio
   ROCmSearchDirs.emplace_back(D.ResourceDir,
                               /*StrictChecking=*/true);
 
-  ROCmSearchDirs.emplace_back(D.SysRoot + "/opt/rocm",
+  ROCmSearchDirs.emplace_back(D.SysRoot + "@ROCM_PATH@", // oteodoro:  edited
                               /*StrictChecking=*/true);
 
+// oteodoro:  disabled
+#if 0
   // Find the latest /opt/rocm-{release} directory.
   std::error_code EC;
   std::string LatestROCm;
@@ -305,6 +307,7 @@ RocmInstallationDetector::getInstallatio
   if (!LatestROCm.empty())
     ROCmSearchDirs.emplace_back(D.SysRoot + "/opt/" + LatestROCm,
                                 /*StrictChecking=*/true);
+#endif
 
   DoPrintROCmSearchDirs();
   return ROCmSearchDirs;
@@ -411,11 +414,13 @@ void RocmInstallationDetector::detectDev
     };
 
     // The possible structures are:
+    // - ${ROCM_ROOT}/@LIBDIR@/amdgcn/bitcode/* // oteodoro:  added
     // - ${ROCM_ROOT}/amdgcn/bitcode/*
     // - ${ROCM_ROOT}/lib/*
     // - ${ROCM_ROOT}/lib/bitcode/*
     // so try to detect these layouts.
     static constexpr std::array<const char *, 2> SubDirsList[] = {
+        {"@LIBDIR@/amdgcn", "bitcode"}, // oteodoro:  edited
         {"amdgcn", "bitcode"},
         {"lib", ""},
         {"lib", "bitcode"},
diff -urp rocMLIR-rocm-5.2.3.orig/external/llvm-project/clang/tools/amdgpu-arch/CMakeLists.txt rocMLIR-rocm-5.2.3/external/llvm-project/clang/tools/amdgpu-arch/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/external/llvm-project/clang/tools/amdgpu-arch/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/external/llvm-project/clang/tools/amdgpu-arch/CMakeLists.txt	2023-10-04 11:37:12.751021538 -0700
@@ -6,7 +6,7 @@
 # //
 # //===----------------------------------------------------------------------===//
 
-find_package(hsa-runtime64 QUIET 1.2.0 HINTS ${CMAKE_INSTALL_PREFIX} PATHS /opt/rocm)
+find_package(hsa-runtime64 QUIET 1.2.0 HINTS ${CMAKE_INSTALL_PREFIX} PATHS @ESYSROOT_ROCM_PATH@) # oteodoro:  edited
 if (NOT ${hsa-runtime64_FOUND})
   message(STATUS "Not building amdgpu-arch: hsa-runtime64 not found")
   return()
diff -urp rocMLIR-rocm-5.2.3.orig/external/llvm-project/mlir/lib/Dialect/GPU/CMakeLists.txt rocMLIR-rocm-5.2.3/external/llvm-project/mlir/lib/Dialect/GPU/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/external/llvm-project/mlir/lib/Dialect/GPU/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/external/llvm-project/mlir/lib/Dialect/GPU/CMakeLists.txt	2023-10-04 11:37:12.803021538 -0700
@@ -127,7 +127,7 @@ if(MLIR_ENABLE_ROCM_CONVERSIONS)
       "Building mlir with ROCm support requires the AMDGPU backend")
   endif()
 
-  set(DEFAULT_ROCM_PATH "/opt/rocm" CACHE PATH "Fallback path to search for ROCm installs")
+  set(DEFAULT_ROCM_PATH "@EPREFIX_ROCM_PATH@" CACHE PATH "Fallback path to search for ROCm installs") # oteodoro:  edited
   target_compile_definitions(obj.MLIRGPUTransforms
     PRIVATE
     __DEFAULT_ROCM_PATH__="${DEFAULT_ROCM_PATH}"
Only in rocMLIR-rocm-5.2.3/external/llvm-project/mlir/lib/Dialect/GPU: CMakeLists.txt.orig
diff -urp rocMLIR-rocm-5.2.3.orig/external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt rocMLIR-rocm-5.2.3/external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt	2023-10-04 11:37:12.827021538 -0700
@@ -153,23 +153,27 @@ if(MLIR_ENABLE_ROCM_RUNNER)
   # Configure ROCm support.
   if (NOT DEFINED ROCM_PATH)
     if (NOT DEFINED ENV{ROCM_PATH})
-      set(ROCM_PATH "/opt/rocm" CACHE PATH "Path to which ROCm has been installed")
+      set(ROCM_PATH "@ESYSROOT_ROCM_PATH@" CACHE PATH "Path to which ROCm has been installed") # oteodoro:  edited
     else()
       set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Path to which ROCm has been installed")
     endif()
-    set(HIP_PATH "${ROCM_PATH}/hip" CACHE PATH "Path to which HIP has been installed")
+    set(HIP_PATH "${ROCM_PATH}" CACHE PATH "Path to which HIP has been installed") # oteodoro:  edited
   endif()
-  set(CMAKE_MODULE_PATH "${HIP_PATH}/cmake" ${CMAKE_MODULE_PATH})
-  find_package(HIP)
-  if (NOT HIP_FOUND)
+  set(CMAKE_MODULE_PATH "${ROCM_PATH}/@LIBDIR@/cmake/hip" ${CMAKE_MODULE_PATH}) # oteodoro:  edited
+  find_package(hip) # oteodoro:  edited
+  if (NOT hip_FOUND) # oteodoro:  edited
     message(SEND_ERROR "Building mlir with ROCm support requires a working ROCm and HIP install")
   else()
     message(STATUS "ROCm HIP version: ${HIP_VERSION}")
   endif()
 
+  if (NOT DEFINED HIP_ROOT_DIR)
+    set(HIP_ROOT_DIR "/usr") # oteodoro:  added
+  endif()
+
   # Locate HIP runtime library.
   find_library(ROCM_RUNTIME_LIBRARY amdhip64
-               PATHS "${HIP_ROOT_DIR}/lib")
+               PATHS "${HIP_ROOT_DIR}/@LIBDIR@") # oteodoro:  edited
   if (NOT ROCM_RUNTIME_LIBRARY)
     message(SEND_ERROR "Could not locate ROCm HIP runtime library")
   else()
diff -urp rocMLIR-rocm-5.2.3.orig/external/llvm-project/openmp/libomptarget/plugins/amdgpu/CMakeLists.txt rocMLIR-rocm-5.2.3/external/llvm-project/openmp/libomptarget/plugins/amdgpu/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/external/llvm-project/openmp/libomptarget/plugins/amdgpu/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/external/llvm-project/openmp/libomptarget/plugins/amdgpu/CMakeLists.txt	2023-10-04 11:37:12.859021538 -0700
@@ -20,7 +20,7 @@ if (NOT LIBOMPTARGET_BUILD_AMDGPU_PLUGIN
 endif()
 
 # as of rocm-3.7, hsa is installed with cmake packages and kmt is found via hsa
-find_package(hsa-runtime64 QUIET 1.2.0 HINTS ${CMAKE_INSTALL_PREFIX} PATHS /opt/rocm)
+find_package(hsa-runtime64 QUIET 1.2.0 HINTS ${CMAKE_INSTALL_PREFIX} PATHS @ESYSROOT_ROCM_PATH@) # oteodoro:  edited
 
 if(NOT LIBOMPTARGET_DEP_LIBELF_FOUND)
   libomptarget_say("Not building AMDGPU plugin: LIBELF not found")
diff -urp rocMLIR-rocm-5.2.3.orig/mlir/CMakeLists.txt rocMLIR-rocm-5.2.3/mlir/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/mlir/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/mlir/CMakeLists.txt	2023-10-04 11:37:12.871021538 -0700
@@ -4,7 +4,7 @@ message(STATUS "LLVM_BUILD_LIBRARY_DIR:
 message(STATUS "LLVM_DEFINITIONS: ${LLVM_DEFINITIONS}")
 
 set(LLVM_RUNTIME_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/bin)
-set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/lib)
+set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/@LIBDIR@) # oteodoro:  edited
 set(MLIR_BINARY_DIR ${CMAKE_BINARY_DIR})
 set(MLIR_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
diff -urp rocMLIR-rocm-5.2.3.orig/mlir/lib/Dialect/MIOpen/CMakeLists.txt rocMLIR-rocm-5.2.3/mlir/lib/Dialect/MIOpen/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/mlir/lib/Dialect/MIOpen/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/mlir/lib/Dialect/MIOpen/CMakeLists.txt	2023-10-04 11:37:12.871021538 -0700
@@ -1,7 +1,7 @@
 # Configure ROCm support.
 if (NOT DEFINED ROCM_PATH)
   if (NOT DEFINED ENV{ROCM_PATH})
-    set(ROCM_PATH "/opt/rocm" CACHE PATH "Path to which ROCm has been installed")
+    set(ROCM_PATH "@ESYSROOT_ROCM_PATH@" CACHE PATH "Path to which ROCm has been installed") # oteodoro:  edited
   else()
     set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Path to which ROCm has been installed")
   endif()
diff -urp rocMLIR-rocm-5.2.3.orig/mlir/lib/Dialect/MIOpen/Tuning/CMakeLists.txt rocMLIR-rocm-5.2.3/mlir/lib/Dialect/MIOpen/Tuning/CMakeLists.txt
--- rocMLIR-rocm-5.2.3.orig/mlir/lib/Dialect/MIOpen/Tuning/CMakeLists.txt	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/mlir/lib/Dialect/MIOpen/Tuning/CMakeLists.txt	2023-10-04 11:37:12.871021538 -0700
@@ -18,7 +18,7 @@ if (MLIR_MIOPEN_SQLITE_ENABLED)
   if (SQLITE3_FOUND)
     set(MIOPEN_PATH ${ROCM_PATH}/miopen)
     if (NOT MIOPEN_SYSTEM_DB_PATH)
-      set(MIOPEN_SYSTEM_DB_PATH "${MIOPEN_PATH}/share/miopen/db/miopen.db")
+      set(MIOPEN_SYSTEM_DB_PATH "@EPREFIX_ROCM_PATH@/share/miopen/db/miopen.db") # oteodoro:  edited
     endif()
     if (EXISTS "${MIOPEN_SYSTEM_DB_PATH}")
       message(STATUS "miopen system perfdb path: ${MIOPEN_SYSTEM_DB_PATH}")
Only in rocMLIR-rocm-5.2.3/mlir/lib/Dialect/MIOpen/Tuning: CMakeLists.txt.orig
diff -urp rocMLIR-rocm-5.2.3.orig/mlir/lib/Dialect/MIOpen/Tuning/SqliteDb.cpp rocMLIR-rocm-5.2.3/mlir/lib/Dialect/MIOpen/Tuning/SqliteDb.cpp
--- rocMLIR-rocm-5.2.3.orig/mlir/lib/Dialect/MIOpen/Tuning/SqliteDb.cpp	2022-05-13 11:07:57.000000000 -0700
+++ rocMLIR-rocm-5.2.3/mlir/lib/Dialect/MIOpen/Tuning/SqliteDb.cpp	2023-10-04 11:43:01.003023642 -0700
@@ -152,7 +152,7 @@ SQLitePerfDb::SQLitePerfDb(const std::st
 }
 
 SQLitePerfDb mlir::getDb(const llvm::SmallString<8> &arch, std::size_t num_cu) {
-  // DB path: "/opt/rocm/miopen/share/miopen/db/miopen.db"
+  // DB path: "@EPREFIX_ROCM_PATH@/miopen/share/miopen/db/miopen.db"
   return {MIOPEN_SYSTEM_DB_PATH, true, std::string(arch), num_cu};
 }
 
Only in rocMLIR-rocm-5.2.3/mlir/tools/mlir-rocm-runner: CMakeLists.txt.orig
Only in rocMLIR-rocm-5.2.3/mlir/tools/mlir-rocm-runner: CMakeLists.txt.rej
