diff -urp rocMLIR-rocm-5.5.0.orig/mlir/include/mlir/Dialect/Rock/IR/RockOps.td rocMLIR-rocm-5.5.0/mlir/include/mlir/Dialect/Rock/IR/RockOps.td
--- rocMLIR-rocm-5.5.0.orig/mlir/include/mlir/Dialect/Rock/IR/RockOps.td	2023-01-04 06:20:51.000000000 -0800
+++ rocMLIR-rocm-5.5.0/mlir/include/mlir/Dialect/Rock/IR/RockOps.td	2023-08-06 17:53:16.729506745 -0700
@@ -231,7 +231,7 @@ def Rock_ConvertingCopyKernelOp :
 }
 
 def Rock_TransformOp :
-    Rock_Op<"transform", [NoSideEffect, ViewLikeOpInterface]>,
+    Rock_Op<"transform", [NoMemoryEffect, ViewLikeOpInterface]>,
     Arguments<(ins AnyShaped:$input, Rock_TransformMapAttr:$transform)>,
     Results<(outs AnyShaped:$output)> {
   let summary = "Tensor transformation";
@@ -271,7 +271,7 @@ def Rock_TransformOp :
 
 def Rock_TensorUntransformCastOp :
     Rock_Op<"tensor_untransform_cast",
-      [NoSideEffect, AllTypesMatch<["transformedResult", "transformedArg"]>]>,
+      [NoMemoryEffect, AllTypesMatch<["transformedResult", "transformedArg"]>]>,
     Arguments<(ins AnyRankedTensor:$transformedResult,
                       AnyRankedTensor:$transformedArg)>,
     Results<(outs AnyRankedTensor:$untransformed)> {
@@ -393,7 +393,7 @@ def Rock_LDSBarrierOp:
 }
 
 def Rock_WorkgroupIdOp:
-    Rock_Op<"workgroup_id", [NoSideEffect,
+    Rock_Op<"workgroup_id", [NoMemoryEffect,
       DeclareOpInterfaceMethods<InferIntRangeInterface>]>,
     Results<(outs Index:$id)> {
   let summary = "Get current workgroup ID";
@@ -404,7 +404,7 @@ def Rock_WorkgroupIdOp:
 }
 
 def Rock_WorkitemIdOp:
-    Rock_Op<"workitem_id", [NoSideEffect,
+    Rock_Op<"workitem_id", [NoMemoryEffect,
       DeclareOpInterfaceMethods<InferIntRangeInterface>]>,
     Results<(outs Index:$id)> {
   let summary = "Get current workitem ID";
@@ -416,7 +416,7 @@ def Rock_WorkitemIdOp:
 
 // extract_slice
 def Rock_ExtractSliceOp :
-    Rock_Op<"extract_slice", [NoSideEffect,
+    Rock_Op<"extract_slice", [NoMemoryEffect,
       AllElementTypesMatch<["vector", "result"]>]>,
     Arguments<(ins
       VectorOfRank<[1]>:$vector,
@@ -439,7 +439,7 @@ def Rock_ExtractSliceOp :
 
 // insert_slice
 def Rock_InsertSliceOp :
-    Rock_Op<"insert_slice", [NoSideEffect,
+    Rock_Op<"insert_slice", [NoMemoryEffect,
       AllElementTypesMatch<["source", "dest"]>,
       AllTypesMatch<["dest", "result"]>]>,
     Arguments<(ins
@@ -464,7 +464,7 @@ def Rock_InsertSliceOp :
 
 // yield
 def Rock_YieldOp :
-    Rock_Op<"yield", [NoSideEffect, Terminator, ReturnLike]> {
+    Rock_Op<"yield", [NoMemoryEffect, Terminator, ReturnLike]> {
   let summary = "yield for Rock loops";
 
   let description = [{
@@ -650,7 +650,7 @@ def Rock_TransformingForOp :
 
 // index_diff_update
 def Rock_IndexDiffUpdateOp :
-    Rock_Op<"index_diff_update", [NoSideEffect, AttrSizedOperandSegments,
+    Rock_Op<"index_diff_update", [NoMemoryEffect, AttrSizedOperandSegments,
       SameVariadicResultSize, AllTypesMatch<["lowerIndices", "lowerDiff"]>]>,
     Arguments<(ins Rock_TransformMapAttr:$map,
       Variadic<Index>:$upperDiffs,
