--- a/external/llvm-project/mlir/cmake/modules/AddMLIR.cmake.orig	2023-01-04 06:20:51.000000000 -0800
+++ b/external/llvm-project/mlir/cmake/modules/AddMLIR.cmake	2023-08-07 11:05:31.799119879 -0700
@@ -533,13 +533,13 @@ function(add_mlir_library_install name)
   install(TARGETS ${name}
     COMPONENT ${name}
     ${export_to_mlirtargets}
-    LIBRARY DESTINATION lib${LLVM_LIBDIR_SUFFIX}
-    ARCHIVE DESTINATION lib${LLVM_LIBDIR_SUFFIX}
+    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     # Note that CMake will create a directory like:
     #   objects-${CMAKE_BUILD_TYPE}/obj.LibName
     # and put object files there.
-    OBJECTS DESTINATION lib${LLVM_LIBDIR_SUFFIX}
+    OBJECTS DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   )
 
   if (NOT LLVM_ENABLE_IDE)
