diff -urp rocMLIR-rocm-5.1.3.orig/mlir/tools/miopen-opt/CMakeLists.txt rocMLIR-rocm-5.1.3/mlir/tools/miopen-opt/CMakeLists.txt
--- rocMLIR-rocm-5.1.3.orig/mlir/tools/miopen-opt/CMakeLists.txt	2022-01-20 11:34:11.000000000 -0800
+++ rocMLIR-rocm-5.1.3/mlir/tools/miopen-opt/CMakeLists.txt	2023-08-06 03:51:16.060190905 -0700
@@ -4,11 +4,21 @@ set(LLVM_OPTIONAL_SOURCES
 
 get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
 get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
-set(LLVM_LINK_COMPONENTS
-  Core
-  Support
-  AsmParser
-  )
+#set(LLVM_LINK_COMPONENTS
+#  Core
+#  Support
+#  AsmParser
+#  )
+
+find_library(LLVM_SLOT_LIB
+  LLVM-${LLVM_VERSION_MAJOR}
+  PATHS /usr/lib/llvm/${LLVM_VERSION_MAJOR}/${CMAKE_INSTALL_LIBDIR}
+)
+
+find_library(LLVM_SUPPORT_LIB
+  libLLVMSupport.a LLVMSupport
+  PATHS /usr/lib/llvm/${LLVM_VERSION_MAJOR}/${CMAKE_INSTALL_LIBDIR}
+)
 
 if(MLIR_INCLUDE_TESTS)
   set(test_libs
@@ -66,7 +76,7 @@ add_llvm_tool(miopen-opt
   DEPENDS
   ${LIBS}
   )
-target_link_libraries(miopen-opt PRIVATE ${LIBS})
+target_link_libraries(miopen-opt PRIVATE ${LIBS} ${LLVM_SLOT_LIB} ${LLVM_SUPPORT_LIB})
 llvm_update_compile_flags(miopen-opt)
 
 mlir_check_all_link_libraries(miopen-opt)
Only in rocMLIR-rocm-5.1.3/mlir/tools/miopen-opt: CMakeLists.txt.orig
diff -urp rocMLIR-rocm-5.1.3.orig/mlir/tools/miopen-translate/CMakeLists.txt rocMLIR-rocm-5.1.3/mlir/tools/miopen-translate/CMakeLists.txt
--- rocMLIR-rocm-5.1.3.orig/mlir/tools/miopen-translate/CMakeLists.txt	2022-01-20 11:34:11.000000000 -0800
+++ rocMLIR-rocm-5.1.3/mlir/tools/miopen-translate/CMakeLists.txt	2023-08-06 03:51:42.796191601 -0700
@@ -1,6 +1,11 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
+#set(LLVM_LINK_COMPONENTS
+#  Support
+#  )
+
+find_library(LLVM_SUPPORT_LIB
+  libLLVMSupport.a LLVMSupport
+  PATHS /usr/lib/llvm/${LLVM_VERSION_MAJOR}/${CMAKE_INSTALL_LIBDIR}
+)
 
 get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
 get_property(translation_libs GLOBAL PROPERTY MLIR_TRANSLATION_LIBS)
@@ -27,6 +32,7 @@ target_link_libraries(miopen-translate
   MLIRTranslation
   MLIRSupport
   MLIRROCDLToLLVMIRTranslation
+  ${LLVM_SUPPORT_LIB}
   )
 
 mlir_check_link_libraries(miopen-translate)
Only in rocMLIR-rocm-5.1.3/mlir/tools/miopen-translate: CMakeLists.txt.orig
Only in rocMLIR-rocm-5.1.3/mlir/tools/miopen-translate: CMakeLists.txt.rej
diff -urp rocMLIR-rocm-5.1.3.orig/mlir/tools/mlir-miopen-driver/CMakeLists.txt rocMLIR-rocm-5.1.3/mlir/tools/mlir-miopen-driver/CMakeLists.txt
--- rocMLIR-rocm-5.1.3.orig/mlir/tools/mlir-miopen-driver/CMakeLists.txt	2022-01-20 11:34:11.000000000 -0800
+++ rocMLIR-rocm-5.1.3/mlir/tools/mlir-miopen-driver/CMakeLists.txt	2023-08-06 03:51:16.436190915 -0700
@@ -1,6 +1,11 @@
-set(LLVM_LINK_COMPONENTS
-  Support
-  )
+#set(LLVM_LINK_COMPONENTS
+#  Support
+#  )
+
+find_library(LLVM_SUPPORT_LIB
+  libLLVMSupport.a LLVMSupport
+  PATHS /usr/lib/llvm/${LLVM_VERSION_MAJOR}/${CMAKE_INSTALL_LIBDIR}
+)
 
 get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
 get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
@@ -37,7 +42,7 @@ add_llvm_executable(mlir-miopen-driver
 
 
 llvm_update_compile_flags(mlir-miopen-driver)
-target_link_libraries(mlir-miopen-driver PRIVATE ${LIBS})
+target_link_libraries(mlir-miopen-driver PRIVATE ${LIBS} ${LLVM_SUPPORT_LIB})
 mlir_check_link_libraries(mlir-miopen-driver)
 
 # Performance test scripts
Only in rocMLIR-rocm-5.1.3/mlir/tools/mlir-miopen-driver: CMakeLists.txt.orig
