--- rocMLIR-rocm-5.5.0/external/llvm-project/llvm/cmake/modules/AddLLVM.cmake.orig	2023-01-04 06:20:51.000000000 -0800
+++ rocMLIR-rocm-5.5.0/external/llvm-project/llvm/cmake/modules/AddLLVM.cmake	2023-08-07 11:49:16.219188234 -0700
@@ -2286,7 +2286,7 @@ function(llvm_setup_rpath name)
     # FIXME: update this when there is better solution.
     set(_install_rpath "${LLVM_LIBRARY_OUTPUT_INTDIR}" "${CMAKE_INSTALL_PREFIX}/lib${LLVM_LIBDIR_SUFFIX}" ${extra_libdir})
   elseif(UNIX)
-    set(_install_rpath "\$ORIGIN/../lib${LLVM_LIBDIR_SUFFIX}" ${extra_libdir})
+    set(_install_rpath "/usr/${CMAKE_INSTALL_LIBDIR}/rocMLIR/${CMAKE_INSTALL_LIBDIR}" ${extra_libdir})
     if(${CMAKE_SYSTEM_NAME} MATCHES "(FreeBSD|DragonFly)")
       set_property(TARGET ${name} APPEND_STRING PROPERTY
                    LINK_FLAGS " -Wl,-z,origin ")
