--- a/ycmd/completers/general/general_completer_store.py.orig	2020-02-01 18:27:01.881133059 -0800
+++ b/ycmd/completers/general/general_completer_store.py	2020-02-01 18:30:07.118504597 -0800
@@ -26,7 +26,6 @@ from builtins import *  # noqa
 from ycmd.completers.completer import Completer
 from ycmd.completers.all.identifier_completer import IdentifierCompleter
 from ycmd.completers.general.filename_completer import FilenameCompleter
-from ycmd.completers.general.ultisnips_completer import UltiSnipsCompleter
 
 
 class GeneralCompleterStore( Completer ):
@@ -41,14 +40,10 @@ class GeneralCompleterStore( Completer )
     super( GeneralCompleterStore, self ).__init__( user_options )
     self._identifier_completer = IdentifierCompleter( user_options )
     self._filename_completer = FilenameCompleter( user_options )
-    self._ultisnips_completer = UltiSnipsCompleter( user_options )
     self._non_filename_completers = [ self._identifier_completer ]
-    if user_options.get( 'use_ultisnips_completer', True ):
-      self._non_filename_completers.append( self._ultisnips_completer )
 
     self._all_completers = [ self._identifier_completer,
-                             self._filename_completer,
-                             self._ultisnips_completer ]
+                             self._filename_completer ]
     self._current_query_completers = []
 
 
