--- a/ycmd/__main__.py.orig	2021-04-08 16:39:00.000000000 -0700
+++ b/ycmd/__main__.py	2021-04-14 13:25:42.000372351 -0700
@@ -24,22 +24,13 @@ if sys.version_info[ 0 ] < 3:
 ROOT_DIR = os.path.abspath( os.path.join( os.path.dirname( __file__ ), '..' ) )
 DIR_OF_THIRD_PARTY = os.path.join( ROOT_DIR, 'third_party' )
 DIR_OF_WATCHDOG_DEPS = os.path.join( DIR_OF_THIRD_PARTY, 'watchdog_deps' )
-DIR_OF_REQUESTS_DEPS = os.path.join( DIR_OF_THIRD_PARTY, 'requests_deps' )
 sys.path[ 0:0 ] = [
     os.path.join( ROOT_DIR ),
     os.path.join( DIR_OF_THIRD_PARTY, 'bottle' ),
     os.path.join( DIR_OF_THIRD_PARTY, 'regex-build' ),
-    os.path.join( DIR_OF_THIRD_PARTY, 'frozendict' ),
     os.path.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'jedi' ),
     os.path.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'parso' ),
-    os.path.join( DIR_OF_REQUESTS_DEPS, 'requests' ),
-    os.path.join( DIR_OF_REQUESTS_DEPS, 'chardet' ),
-    os.path.join( DIR_OF_REQUESTS_DEPS, 'certifi' ),
-    os.path.join( DIR_OF_REQUESTS_DEPS, 'idna' ),
-    os.path.join( DIR_OF_REQUESTS_DEPS, 'urllib3', 'src' ),
-    os.path.join( DIR_OF_WATCHDOG_DEPS, 'watchdog', 'build', 'lib3' ),
-    os.path.join( DIR_OF_WATCHDOG_DEPS, 'pathtools' ),
-    os.path.join( DIR_OF_THIRD_PARTY, 'waitress' ) ]
+    os.path.join( DIR_OF_WATCHDOG_DEPS, 'watchdog', 'build', 'lib3' )]
 sys.path.append( os.path.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'numpydoc' ) )
 
 import atexit
--- a/run_tests.py.orig	2021-04-08 16:39:00.000000000 -0700
+++ b/run_tests.py	2021-04-14 13:25:51.510735755 -0700
@@ -19,12 +19,9 @@ LIBCLANG_DIR = p.join( DIR_OF_THIRD_PART
 python_path = [
   p.join( DIR_OF_THIRD_PARTY, 'bottle' ),
   p.join( DIR_OF_THIRD_PARTY, 'regex-build' ),
-  p.join( DIR_OF_THIRD_PARTY, 'frozendict' ),
   p.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'jedi' ),
   p.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'parso' ),
   p.join( DIR_OF_WATCHDOG_DEPS, 'watchdog', 'build', 'lib3' ),
-  p.join( DIR_OF_WATCHDOG_DEPS, 'pathtools' ),
-  p.join( DIR_OF_THIRD_PARTY, 'waitress' ),
 ]
 if os.environ.get( 'PYTHONPATH' ) is not None:
   python_path.append( os.environ[ 'PYTHONPATH' ] )
