--- a/ycmd/completers/go/go_completer.py.orig	2020-02-01 19:55:50.970699623 -0800
+++ b/ycmd/completers/go/go_completer.py	2020-02-01 20:04:23.050719466 -0800
@@ -51,12 +51,8 @@ GOCODE_PANIC_MESSAGE = ( 'Gocode panicke
 DIR_OF_THIRD_PARTY = os.path.abspath(
   os.path.join( os.path.dirname( __file__ ), '..', '..', '..', 'third_party' ) )
 GO_BINARIES = dict( {
-  'gocode': os.path.join( DIR_OF_THIRD_PARTY,
-                          'gocode',
-                          ExecutableName( 'gocode' ) ),
-  'godef': os.path.join( DIR_OF_THIRD_PARTY,
-                         'godef',
-                         ExecutableName( 'godef' ) )
+  'gocode': ___GOCODE_BIN_PATH___,
+  'godef': ___GODEF_BIN_PATH___
 } )
 
 LOGFILE_FORMAT = 'gocode_{port}_{std}_'
