diff -urp ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c.orig/ycmd/completers/cs/cs_completer.py ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c/ycmd/completers/cs/cs_completer.py
--- ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c.orig/ycmd/completers/cs/cs_completer.py	2014-12-14 19:29:30.000000000 -0800
+++ ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c/ycmd/completers/cs/cs_completer.py	2017-01-17 01:14:28.846864990 -0800
@@ -40,8 +40,7 @@ FILE_TOO_SHORT_MESSAGE = (
     MIN_LINES_IN_FILE_TO_PARSE ) )
 NO_DIAGNOSTIC_MESSAGE = 'No diagnostic for current line!'
 PATH_TO_OMNISHARP_BINARY = os.path.join(
-  os.path.abspath( os.path.dirname( __file__ ) ),
-  '../../../third_party/OmniSharpServer/OmniSharp/bin/Debug/OmniSharp.exe' )
+  os.path.abspath( 'GENTOO_OMNISHARP' )
 
 
 #TODO: Handle this better than dummy classes
