diff -urp a/ycmd/completers/javascript/tern_completer.py b/ycmd/completers/javascript/tern_completer.py
--- a/ycmd/completers/javascript/tern_completer.py	2020-01-08 05:10:45.000000000 -0800
+++ b/ycmd/completers/javascript/tern_completer.py	2020-01-11 22:41:30.074442979 -0800
@@ -34,18 +34,7 @@ from ycmd.completers.completer import Co
 from ycmd.completers.completer_utils import GetFileLines
 from ycmd.utils import LOGGER
 
-PATH_TO_TERN_BINARY = os.path.abspath(
-  os.path.join(
-    os.path.dirname( __file__ ),
-    '..',
-    '..',
-    '..',
-    'third_party',
-    'tern_runtime',
-    'node_modules',
-    'tern',
-    'bin',
-    'tern' ) )
+PATH_TO_TERN_BINARY = "___TERN_BIN_PATH___"
 
 PATH_TO_NODE = utils.FindExecutable( 'node' )
 
