--- a/ycmd/utils.py	2020-05-16 16:27:33.132403153 -0700
+++ b/ycmd/utils.py	2020-05-16 16:35:12.110935918 -0700
@@ -29,7 +29,7 @@ import threading
 LOGGER = logging.getLogger( 'ycmd' )
 ROOT_DIR = os.path.normpath( os.path.join( os.path.dirname( __file__ ), '..' ) )
 DIR_OF_THIRD_PARTY = os.path.join( ROOT_DIR, 'third_party' )
-LIBCLANG_DIR = os.path.join( DIR_OF_THIRD_PARTY, 'clang', 'lib' )
+LIBCLANG_DIR = "__LIBCLANG_LIB_DIR__"
 if hasattr( os, 'add_dll_directory' ):
   os.add_dll_directory( LIBCLANG_DIR )
 
Only in ycmd-43_p20200516/ycmd: utils.py.orig
Only in ycmd-43_p20200516/ycmd: utils.py.rej
