--- a/ycmd/completers/go/go_completer.py	2020-05-16 16:49:58.536662892 -0700
+++ b/ycmd/completers/go/go_completer.py	2020-05-16 16:51:36.960632541 -0700
@@ -24,14 +24,7 @@ from ycmd import utils
 from ycmd.completers.language_server import language_server_completer
 
 
-PATH_TO_GOPLS = os.path.abspath( os.path.join( os.path.dirname( __file__ ),
-  '..',
-  '..',
-  '..',
-  'third_party',
-  'go',
-  'bin',
-  utils.ExecutableName( 'gopls' ) ) )
+PATH_TO_GOPLS = "___GOPLS_BIN_PATH___"
 
 
 def ShouldEnableGoCompleter( user_options ):
Only in ycmd-43_p20200516/ycmd/completers/go: go_completer.py.orig
Only in ycmd-43_p20200516/ycmd/completers/go: go_completer.py.rej
