diff -urp ycmd-f1e6afab72bad0a64d5b16f64d0c58bd0d9ddc07.orig/ycmd/completers/cs/cs_completer.py ycmd-f1e6afab72bad0a64d5b16f64d0c58bd0d9ddc07/ycmd/completers/cs/cs_completer.py
--- ycmd-f1e6afab72bad0a64d5b16f64d0c58bd0d9ddc07.orig/ycmd/completers/cs/cs_completer.py	2017-01-17 20:11:11.000000000 -0800
+++ ycmd-f1e6afab72bad0a64d5b16f64d0c58bd0d9ddc07/ycmd/completers/cs/cs_completer.py	2017-01-28 14:42:49.036776432 -0800
@@ -398,7 +398,7 @@ class CsharpSolutionCompleter( object ):
       if self._ServerIsRunning():
         self._logger.info( 'Stopping OmniSharp server with PID {0}'.format(
                                self._omnisharp_phandle.pid ) )
-        self._GetResponse( '/stopserver' )
+        self._GetResponse( '/stopserver', returns_json = False )
         try:
           utils.WaitUntilProcessIsTerminated( self._omnisharp_phandle,
                                               timeout = 5 )
@@ -431,7 +431,8 @@ class CsharpSolutionCompleter( object ):
   def _ReloadSolution( self ):
     """ Reloads the solutions in the OmniSharp server """
     self._logger.info( 'Reloading Solution in OmniSharp server' )
-    return self._GetResponse( '/reloadsolution' )
+    self._GetResponse( '/reloadsolution', returns_json = False )
+    return True
 
 
   def CompletionType( self, request_data ):
@@ -534,13 +535,13 @@ class CsharpSolutionCompleter( object ):
   def _DefaultParameters( self, request_data ):
     """ Some very common request parameters """
     parameters = {}
-    parameters[ 'line' ] = request_data[ 'line_num' ]
-    parameters[ 'column' ] = request_data[ 'column_codepoint' ]
+    parameters[ 'Line' ] = request_data[ 'line_num' ]
+    parameters[ 'Column' ] = request_data[ 'column_codepoint' ]
 
     filepath = request_data[ 'filepath' ]
-    parameters[ 'buffer' ] = (
+    parameters[ 'Buffer' ] = (
       request_data[ 'file_data' ][ filepath ][ 'contents' ] )
-    parameters[ 'filename' ] = filepath
+    parameters[ 'FileName' ] = filepath
     return parameters
 
 
@@ -582,11 +583,17 @@ class CsharpSolutionCompleter( object ):
     return 'http://localhost:' + str( self._omnisharp_port )
 
 
-  def _GetResponse( self, handler, parameters = {}, timeout = None ):
+  def _GetResponse( self, handler, parameters = {}, timeout = None,
+                    returns_json = True ):
     """ Handle communication with server """
     target = urllib.parse.urljoin( self._ServerLocation(), handler )
-    response = requests.post( target, data = parameters, timeout = timeout )
-    return response.json()
+    self._logger.info( "Request: " + str(parameters) )
+    response = requests.post( target, json = parameters, timeout = timeout )
+    self._logger.info( "Response: " + response.text )
+    if returns_json:
+      return response.json()
+    else:
+      return None
 
 
   def _ChooseOmnisharpPort( self ):
