--- a/ycmd/server_utils.py.orig	2020-02-02 05:20:59.844142320 -0800
+++ b/ycmd/server_utils.py	2020-02-02 14:23:44.281385517 -0800
@@ -35,7 +35,6 @@ def SetUpPythonPath():
   sys.path[ 0:0 ] = [ p.join( ROOT_DIR ),
                       p.join( DIR_OF_THIRD_PARTY, 'bottle' ),
                       p.join( DIR_OF_THIRD_PARTY, 'cregex', 'regex_3' ),
-                      p.join( DIR_OF_THIRD_PARTY, 'frozendict' ),
                       p.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'jedi' ),
                       p.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'numpydoc' ),
                       p.join( DIR_OF_THIRD_PARTY, 'jedi_deps', 'parso' ),
@@ -43,8 +42,5 @@ def SetUpPythonPath():
                       p.join( DIR_OF_THIRD_PARTY, 'requests_deps', 'chardet' ),
                       p.join( DIR_OF_THIRD_PARTY, 'requests_deps', 'certifi' ),
                       p.join( DIR_OF_THIRD_PARTY, 'requests_deps', 'idna' ),
-                      p.join( DIR_OF_THIRD_PARTY,
-                              'requests_deps',
-                              'urllib3',
-                              'src' ),
+                      p.join( DIR_OF_THIRD_PARTY, 'requests_deps', 'urllib3', 'src' ),
                       p.join( DIR_OF_THIRD_PARTY, 'waitress' ) ]
