diff -urp ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c.orig/build.sh ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c/build.sh
--- ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c.orig/build.sh	2017-01-17 01:18:43.900878611 -0800
+++ ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c/build.sh	2017-01-17 01:26:23.907860189 -0800
@@ -28,29 +28,29 @@ function homebrew_cmake_install {
 }
 
 function python_finder {
-  python_library="-DPYTHON_LIBRARY="
-  python_include="-DPYTHON_INCLUDE_DIR="
+  python_library="-DPYTHON_LIBRARY=GENTOO_PYTHON_LIBRARY"
+  python_include="-DPYTHON_INCLUDE_DIR=GENTOO_PYTHON_INCLUDE_DIR"
 
   # The CMake 'FindPythonLibs' Module does not work properly.
   # So we are forced to do its job for it.
-  python_prefix=$(python-config --prefix | sed 's/^[ \t]*//')
-  if [ -f "${python_prefix}/Python" ]; then
-    python_library+="${python_prefix}/Python"
-    python_include+="${python_prefix}/Headers"
-  else
-    which_python=$(python -c 'import sys;print(sys.version)' | sed 's/^[ \t]*//')
-    which_python="python${which_python:0:3}"
-    lib_python="${python_prefix}/lib/lib${which_python}"
-    if [ -f "${lib_python}.a" ]; then
-      python_library+="${lib_python}.a"
-    # This check is for for CYGWIN
-    elif [ -f "${lib_python}.dll.a" ]; then
-      python_library+="${lib_python}.dll.a"
-    else
-      python_library+="${lib_python}.dylib"
-    fi
-    python_include+="${python_prefix}/include/${which_python}"
-  fi
+  #python_prefix=$(python-config --prefix | sed 's/^[ \t]*//')
+  #if [ -f "${python_prefix}/Python" ]; then
+  #  python_library+="${python_prefix}/Python"
+  #  python_include+="${python_prefix}/Headers"
+  #else
+  #  which_python=$(python -c 'import sys;print(sys.version)' | sed 's/^[ \t]*//')
+  #  which_python="python${which_python:0:3}"
+  #  lib_python="${python_prefix}/lib/lib${which_python}"
+  #  if [ -f "${lib_python}.a" ]; then
+  #    python_library+="${lib_python}.a"
+  #  # This check is for for CYGWIN
+  #  elif [ -f "${lib_python}.dll.a" ]; then
+  #    python_library+="${lib_python}.dll.a"
+  #  else
+  #    python_library+="${lib_python}.dylib"
+  #  fi
+  #  python_include+="${python_prefix}/include/${which_python}"
+  #fi
 
   echo "${python_library} ${python_include}"
 }
