diff -urp ycmd-e5d696e9b58729d89df60fadebc15fc38e18c466.orig/ycmd/completers/python/jedi_completer.py ycmd-e5d696e9b58729d89df60fadebc15fc38e18c466/ycmd/completers/python/jedi_completer.py
--- ycmd-e5d696e9b58729d89df60fadebc15fc38e18c466.orig/ycmd/completers/python/jedi_completer.py	2018-01-01 20:11:05.108540258 -0800
+++ ycmd-e5d696e9b58729d89df60fadebc15fc38e18c466/ycmd/completers/python/jedi_completer.py	2018-01-01 20:12:42.794538695 -0800
@@ -166,9 +166,7 @@ class JediCompleter( Completer ):
         json.dump( { 'hmac_secret': ToUnicode(
                         b64encode( self._hmac_secret ) ) },
                    hmac_file )
-        command = [
-          self._python_binary_path,
-          PATH_TO_JEDIHTTP,
+        command = [ PATH_TO_JEDIHTTP,
           '--port', str( self._jedihttp_port ),
           '--log', self._GetLoggingLevel(),
           '--hmac-file-secret', hmac_file.name,
Only in ycmd-e5d696e9b58729d89df60fadebc15fc38e18c466/ycmd/completers/python: jedi_completer.py.orig
