--- a/ycmd/completers/cs/cs_completer.py	2020-05-16 17:25:07.249239547 -0700
+++ b/ycmd/completers/cs/cs_completer.py	2020-05-16 17:29:05.638375644 -0700
@@ -38,16 +38,8 @@ SERVER_NOT_FOUND_MSG = ( 'OmniSharp serv
                          '"./install.py --cs-completer".' )
 INVALID_FILE_MESSAGE = 'File is invalid.'
 NO_DIAGNOSTIC_MESSAGE = 'No diagnostic for current line!'
-PATH_TO_ROSLYN_OMNISHARP = os.path.join(
-  os.path.abspath( os.path.dirname( __file__ ) ),
-  '..', '..', '..', 'third_party', 'omnisharp-roslyn'
-)
-PATH_TO_OMNISHARP_ROSLYN_BINARY = os.path.join(
-  PATH_TO_ROSLYN_OMNISHARP, 'Omnisharp.exe' )
-if ( not os.path.isfile( PATH_TO_OMNISHARP_ROSLYN_BINARY )
-     and os.path.isfile( os.path.join( PATH_TO_ROSLYN_OMNISHARP, 'run' ) ) ):
-  PATH_TO_OMNISHARP_ROSLYN_BINARY = (
-    os.path.join( PATH_TO_ROSLYN_OMNISHARP, 'run' ) )
+PATH_TO_ROSLYN_OMNISHARP = "___OMNISHARP_DIR_PATH___"
+PATH_TO_ROSLYN_OMNISHARP_BINARY = "___OMNISHARP_BIN_ABSPATH___"
 LOGFILE_FORMAT = 'omnisharp_{port}_{sln}_{std}_'
 
 
@@ -406,10 +398,6 @@ class CsharpSolutionCompleter( object ):
                 '-s',
                 str( self._solution_path ) ]
 
-    if ( not utils.OnWindows()
-         and self._roslyn_path.endswith( '.exe' ) ):
-      command.insert( 0, 'mono' )
-
     LOGGER.info( 'Starting OmniSharp server with: %s', command )
 
     solutionfile = os.path.basename( self._solution_path )
Only in ycmd-43_p20200516/ycmd/completers/cs: cs_completer.py.orig
Only in ycmd-43_p20200516/ycmd/completers/cs: cs_completer.py.rej
