diff -urp ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c.orig/ycmd/completers/cs/cs_completer.py ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c/ycmd/completers/cs/cs_completer.py
--- ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c.orig/ycmd/completers/cs/cs_completer.py	2017-01-29 18:50:30.491678780 -0800
+++ ycmd-0af509bcda444853836ba740b6b4fbcdea3f8f6c/ycmd/completers/cs/cs_completer.py	2017-01-29 18:53:03.579577307 -0800
@@ -248,9 +248,6 @@ class CsharpCompleter( Completer ):
                          '-s',
                          '"{0}"'.format( path_to_solutionfile ) ] )
 
-    if not utils.OnWindows() and not utils.OnCygwin():
-      command = 'mono ' + command
-
     if utils.OnCygwin():
       command = command + ' --client-path-mode Cygwin'
 
