--- a/ycmd/utils.py.orig	2020-02-02 05:20:59.834141932 -0800
+++ b/ycmd/utils.py	2020-02-02 12:02:39.261898783 -0800
@@ -28,8 +28,7 @@ import threading
 
 LOGGER = logging.getLogger( 'ycmd' )
 ROOT_DIR = os.path.normpath( os.path.join( os.path.dirname( __file__ ), '..' ) )
-DIR_OF_THIRD_PARTY = os.path.join( ROOT_DIR, 'third_party' )
-LIBCLANG_DIR = os.path.join( DIR_OF_THIRD_PARTY, 'clang', 'lib' )
+LIBCLANG_DIR = "__LIBCLANG_LIB_DIR__"
 
 
 from collections.abc import Mapping
