diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/cs/cs_completer.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/cs/cs_completer.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/cs/cs_completer.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/cs/cs_completer.py	2017-01-14 18:41:54.376488818 -0800
@@ -45,9 +45,6 @@ SERVER_NOT_FOUND_MSG = ( 'OmniSharp serv
 INVALID_FILE_MESSAGE = 'File is invalid.'
 NO_DIAGNOSTIC_MESSAGE = 'No diagnostic for current line!'
-PATH_TO_OMNISHARP_BINARY = os.path.abspath(
-  os.path.join( os.path.dirname( __file__ ), '..', '..', '..',
-                'third_party', 'OmniSharpServer', 'OmniSharp',
-                'bin', 'Release', 'OmniSharp.exe' ) )
+PATH_TO_OMNISHARP_BINARY = "GENTOO_OMNISHARP"
 LOGFILE_FORMAT = 'omnisharp_{port}_{sln}_{std}_'
 
 
diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/go/go_completer.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/go/go_completer.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/go/go_completer.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/go/go_completer.py	2017-01-15 13:49:24.633595226 -0800
@@ -48,14 +48,8 @@ GOCODE_NO_COMPLETIONS_MESSAGE = 'No comp
 GOCODE_PANIC_MESSAGE = ( 'Gocode panicked trying to find completions, '
                          'you likely have a syntax error.' )
 
-DIR_OF_THIRD_PARTY = os.path.abspath(
-  os.path.join( os.path.dirname( __file__ ), '..', '..', '..', 'third_party' ) )
 GO_BINARIES = dict( {
-  'gocode': os.path.join( DIR_OF_THIRD_PARTY,
-                          'gocode',
-                          ExecutableName( 'gocode' ) ),
-  'godef': os.path.join( DIR_OF_THIRD_PARTY,
-                         'godef',
-                         ExecutableName( 'godef' ) )
+  'gocode': "/usr/bin/gocode",
+  'godef': "/usr/bin/godef"
 } )
 
diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/javascript/tern_completer.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/javascript/tern_completer.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/javascript/tern_completer.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/javascript/tern_completer.py	2017-01-15 15:07:59.579496831 -0800
@@ -38,14 +38,3 @@ _logger = logging.getLogger( __name__ )
 
-PATH_TO_TERN_BINARY = os.path.abspath(
-  os.path.join(
-    os.path.dirname( __file__ ),
-    '..',
-    '..',
-    '..',
-    'third_party',
-    'tern_runtime',
-    'node_modules',
-    'tern',
-    'bin',
-    'tern' ) )
+PATH_TO_TERN_BINARY = "/usr/lib64/node/tern/0.20.0/bin/tern"
 
diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/python/jedi_completer.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/python/jedi_completer.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/python/jedi_completer.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/python/jedi_completer.py	2017-01-15 13:48:30.490488432 -0800
@@ -47,8 +47,6 @@ BINARY_NOT_FOUND_MESSAGE = ( 'The specif
                              'was not found. Did you specify it correctly?' )
 LOGFILE_FORMAT = 'jedihttp_{port}_{std}_'
-PATH_TO_JEDIHTTP = os.path.abspath(
-  os.path.join( os.path.dirname( __file__ ), '..', '..', '..',
-                'third_party', 'JediHTTP', 'jedihttp.py' ) )
+PATH_TO_JEDIHTTP = "/usr/bin/jedihttp.py"
 
 
 class JediCompleter( Completer ):
diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/rust/rust_completer.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/rust/rust_completer.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/completers/rust/rust_completer.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/completers/rust/rust_completer.py	2017-01-15 13:50:18.551693184 -0800
@@ -42,14 +42,9 @@ from os import path as p
 
 _logger = logging.getLogger( __name__ )
 
-DIR_OF_THIRD_PARTY = p.abspath(
-  p.join( p.dirname( __file__ ), '..', '..', '..', 'third_party' ) )
-
 RACERD_BINARY_NAME = 'racerd' + ( '.exe' if utils.OnWindows() else '' )
-RACERD_BINARY_RELEASE = p.join( DIR_OF_THIRD_PARTY, 'racerd', 'target',
-                        'release', RACERD_BINARY_NAME )
-RACERD_BINARY_DEBUG = p.join( DIR_OF_THIRD_PARTY, 'racerd', 'target',
-                        'debug', RACERD_BINARY_NAME )
+RACERD_BINARY_RELEASE = "/usr/bin/"+RACERD_BINARY_NAME
+RACERD_BINARY_DEBUG = "/usr/bin"+RACERD_BINARY_NAME
 
 RACERD_HMAC_HEADER = 'x-racerd-hmac'
 HMAC_SECRET_LENGTH = 16
