--- a/build.py.orig	2020-01-08 05:10:45.000000000 -0800
+++ b/build.py	2020-01-11 22:29:22.956444063 -0800
@@ -36,14 +36,14 @@ DIR_OF_THIS_SCRIPT = p.dirname( p.abspat
 DIR_OF_THIRD_PARTY = p.join( DIR_OF_THIS_SCRIPT, 'third_party' )
 LIBCLANG_DIR = p.join( DIR_OF_THIRD_PARTY, 'clang', 'lib' )
 
-for folder in os.listdir( DIR_OF_THIRD_PARTY ):
-  abs_folder_path = p.join( DIR_OF_THIRD_PARTY, folder )
-  if p.isdir( abs_folder_path ) and not os.listdir( abs_folder_path ):
-    sys.exit(
-      'ERROR: folder {} in {} is empty; you probably forgot to run:\n'
-      '\tgit submodule update --init --recursive\n'.format( folder,
-                                                            DIR_OF_THIRD_PARTY )
-    )
+#for folder in os.listdir( DIR_OF_THIRD_PARTY ):
+#  abs_folder_path = p.join( DIR_OF_THIRD_PARTY, folder )
+#  if p.isdir( abs_folder_path ) and not os.listdir( abs_folder_path ):
+#    sys.exit(
+#      'ERROR: folder {} in {} is empty; you probably forgot to run:\n'
+#      '\tgit submodule update --init --recursive\n'.format( folder,
+#                                                            DIR_OF_THIRD_PARTY )
+#    )
 
 sys.path[ 0:0 ] = [ p.join( DIR_OF_THIRD_PARTY, 'requests_deps', 'requests' ),
                     p.join( DIR_OF_THIRD_PARTY,
