--- a/ycmd/completers/cs/cs_completer.py.orig	2020-01-08 05:10:45.000000000 -0800
+++ b/ycmd/completers/cs/cs_completer.py	2020-01-11 23:14:20.599779147 -0800
@@ -399,10 +399,6 @@ class CsharpSolutionCompleter( object ):
                   '-s',
                   str( self._solution_path ) ]
 
-      if ( not utils.OnWindows()
-           and PATH_TO_ROSLYN_OMNISHARP_BINARY.endswith( '.exe' ) ):
-        command.insert( 0, 'mono' )
-
       LOGGER.info( 'Starting OmniSharp server with: %s', command )
 
       solutionfile = os.path.basename( self._solution_path )
