--- a/ycmd/completers/cs/cs_completer.py.orig	2020-02-01 20:12:41.380209842 -0800
+++ b/ycmd/completers/cs/cs_completer.py	2020-02-01 20:17:05.960560885 -0800
@@ -41,10 +41,7 @@ SERVER_NOT_FOUND_MSG = ( 'OmniSharp serv
                          '"./install.py --cs-completer".' )
 INVALID_FILE_MESSAGE = 'File is invalid.'
 NO_DIAGNOSTIC_MESSAGE = 'No diagnostic for current line!'
-PATH_TO_OMNISHARP_BINARY = os.path.abspath(
-  os.path.join( os.path.dirname( __file__ ), '..', '..', '..',
-                'third_party', 'OmniSharpServer', 'OmniSharp',
-                'bin', 'Release', 'OmniSharp.exe' ) )
+PATH_TO_OMNISHARP_BINARY = ___OMNISHARP_BIN_ABSPATH___
 LOGFILE_FORMAT = 'omnisharp_{port}_{sln}_{std}_'
 
 
@@ -357,9 +354,6 @@ class CsharpSolutionCompleter( object ):
                   '-s',
                   u'{0}'.format( path_to_solutionfile ) ]
 
-      if not utils.OnWindows() and not utils.OnCygwin():
-        command.insert( 0, 'mono' )
-
       if utils.OnCygwin():
         command.extend( [ '--client-path-mode', 'Cygwin' ] )
 
