--- a/ycmd/completers/rust/rust_completer.py.orig	2020-02-01 21:12:24.789885057 -0800
+++ b/ycmd/completers/rust/rust_completer.py	2020-02-01 21:14:54.675705754 -0800
@@ -42,14 +42,8 @@ from os import path as p
 
 _logger = logging.getLogger( __name__ )
 
-DIR_OF_THIRD_PARTY = p.abspath(
-  p.join( p.dirname( __file__ ), '..', '..', '..', 'third_party' ) )
-
-RACERD_BINARY_NAME = 'racerd' + ( '.exe' if utils.OnWindows() else '' )
-RACERD_BINARY_RELEASE = p.join( DIR_OF_THIRD_PARTY, 'racerd', 'target',
-                        'release', RACERD_BINARY_NAME )
-RACERD_BINARY_DEBUG = p.join( DIR_OF_THIRD_PARTY, 'racerd', 'target',
-                        'debug', RACERD_BINARY_NAME )
+RACERD_BINARY_RELEASE = ___RACERD_BIN_ABSPATH___
+RACERD_BINARY_DEBUG = ___RACERD_BIN_ABSPATH___
 
 RACERD_HMAC_HEADER = 'x-racerd-hmac'
 HMAC_SECRET_LENGTH = 16
