diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/server_utils.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/server_utils.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/server_utils.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/server_utils.py	2017-01-15 22:02:36.303225877 -0800
@@ -76,7 +76,7 @@ _logger = logging.getLogger( __name__ )
 
 
 def ExpectedCoreVersion():
-  filepath = os.path.join( DIR_OF_CURRENT_SCRIPT, '..', VERSION_FILENAME )
+  filepath = os.path.join( DIR_OF_CURRENT_SCRIPT, VERSION_FILENAME )
   with io.open( filepath, encoding = 'utf8' ) as f:
     return int( f.read() )
 
