diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/build.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/build.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/build.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/build.py	2017-01-14 00:39:29.727986997 -0800
@@ -30,13 +30,13 @@ if not ( ( PY_MAJOR == 2 and PY_MINOR >=
 DIR_OF_THIS_SCRIPT = p.dirname( p.abspath( __file__ ) )
 DIR_OF_THIRD_PARTY = p.join( DIR_OF_THIS_SCRIPT, 'third_party' )
 
-for folder in os.listdir( DIR_OF_THIRD_PARTY ):
-  abs_folder_path = p.join( DIR_OF_THIRD_PARTY, folder )
-  if p.isdir( abs_folder_path ) and not os.listdir( abs_folder_path ):
-    sys.exit(
-      'ERROR: some folders in {0} are empty; you probably forgot to run:\n'
-      '\tgit submodule update --init --recursive\n'.format( DIR_OF_THIRD_PARTY )
-    )
+#for folder in os.listdir( DIR_OF_THIRD_PARTY ):
+#  abs_folder_path = p.join( DIR_OF_THIRD_PARTY, folder )
+#  if p.isdir( abs_folder_path ) and not os.listdir( abs_folder_path ):
+#    sys.exit(
+#      'ERROR: some folders in {0} are empty; you probably forgot to run:\n'
+#      '\tgit submodule update --init --recursive\n'.format( DIR_OF_THIRD_PARTY )
+#    )
 
 sys.path.insert( 1, p.abspath( p.join( DIR_OF_THIRD_PARTY, 'argparse' ) ) )
 
