diff -urp a/ycmd/completers/go/go_completer.py b/ycmd/completers/go/go_completer.py
--- a/ycmd/completers/go/go_completer.py	2020-01-08 05:10:45.000000000 -0800
+++ b/ycmd/completers/go/go_completer.py	2020-01-11 22:46:25.375780466 -0800
@@ -32,19 +32,7 @@ from ycmd.completers.language_server imp
 from ycmd.completers.language_server import language_server_completer
 
 
-PATH_TO_GOPLS = os.path.abspath( os.path.join( os.path.dirname( __file__ ),
-  '..',
-  '..',
-  '..',
-  'third_party',
-  'go',
-  'src',
-  'golang.org',
-  'x',
-  'tools',
-  'cmd',
-  'gopls',
-  utils.ExecutableName( 'gopls' ) ) )
+PATH_TO_GOPLS = "___GOPLS_BIN_PATH___"
 
 
 def ShouldEnableGoCompleter( user_options ):
