--- a/ycmd/server_utils.py.orig	2020-05-17 22:00:59.756126257 -0700
+++ b/ycmd/server_utils.py	2020-05-17 22:06:51.749884213 -0700
@@ -127,11 +127,7 @@ def SetUpPythonPath():
   # We don't add this path in AddNearestThirdPartyFoldersToSysPath because
   # loading the regex module in YCM may cause a segmentation fault if the module
   # is compiled for a different version of Python than the one running YCM.
-  regex_folder = os.path.join( DIR_OF_CURRENT_SCRIPT,
-                               '..',
-                               'third_party',
-                               'cregex',
-                               'regex_{}'.format( sys.version_info[ 0 ] ) )
+  regex_folder = os.path.join( DIR_OF_CURRENT_SCRIPT, '..', 'third_party', 'cregex', 'regex_{}'.format( sys.version_info[ 0 ] ) )
   sys.path.insert( 0, regex_folder )
 
   AddNearestThirdPartyFoldersToSysPath( __file__ )
