--- a/ycmd/completers/cs/cs_completer.py.orig	2020-02-01 20:20:40.238945364 -0800
+++ b/ycmd/completers/cs/cs_completer.py	2020-02-01 20:29:20.509307695 -0800
@@ -37,16 +37,8 @@ SERVER_NOT_FOUND_MSG = ( 'OmniSharp serv
                          '"./install.py --cs-completer".' )
 INVALID_FILE_MESSAGE = 'File is invalid.'
 NO_DIAGNOSTIC_MESSAGE = 'No diagnostic for current line!'
-PATH_TO_ROSLYN_OMNISHARP = os.path.join(
-  os.path.abspath( os.path.dirname( __file__ ) ),
-  '..', '..', '..', 'third_party', 'omnisharp-roslyn'
-)
-PATH_TO_ROSLYN_OMNISHARP_BINARY = os.path.join(
-  PATH_TO_ROSLYN_OMNISHARP, 'Omnisharp.exe' )
-if ( not os.path.isfile( PATH_TO_ROSLYN_OMNISHARP_BINARY )
-     and os.path.isfile( os.path.join( PATH_TO_ROSLYN_OMNISHARP, 'run' ) ) ):
-  PATH_TO_ROSLYN_OMNISHARP_BINARY = (
-    os.path.join( PATH_TO_ROSLYN_OMNISHARP, 'run' ) )
+PATH_TO_ROSLYN_OMNISHARP = ___OMNISHARP_DIR_PATH___
+PATH_TO_ROSLYN_OMNISHARP_BINARY = ___OMNISHARP_BIN_ABSPATH___
 LOGFILE_FORMAT = 'omnisharp_{port}_{sln}_{std}_'
 
 
@@ -387,10 +379,6 @@ class CsharpSolutionCompleter:
                   '-s',
                   str( self._solution_path ) ]
 
-      if ( not utils.OnWindows()
-           and PATH_TO_ROSLYN_OMNISHARP_BINARY.endswith( '.exe' ) ):
-        command.insert( 0, 'mono' )
-
       LOGGER.info( 'Starting OmniSharp server with: %s', command )
 
       solutionfile = os.path.basename( self._solution_path )
