diff -urp libspng-0.6.2.orig/meson.build libspng-0.6.2/meson.build
--- libspng-0.6.2.orig/meson.build	2021-01-28 12:46:13.000000000 -0800
+++ libspng-0.6.2/meson.build	2021-02-11 22:35:02.503061602 -0800
@@ -18,11 +18,6 @@ endif
 
 cc = meson.get_compiler('c')
 
-# Check for GNU target_clones attribute
-if cc.compiles(files('tests/target_clones.c'), args : '-Werror', name : 'have target_clones')
-    add_project_arguments('-DSPNG_ENABLE_TARGET_CLONES', language : 'c')
-endif
-
 if get_option('use_miniz') == true
     add_project_arguments('-DSPNG_USE_MINIZ', language : 'c')
     zlib_dep = dependency('miniz', fallback : [ 'miniz', 'miniz_dep'])
