--- corefx-1.1.0.orig/init-tools.sh	2017-01-27 17:53:13.645747651 -0800
+++ corefx-1.1.0/init-tools.sh	2017-01-27 18:12:20.316334532 -0800
@@ -115,6 +115,7 @@ if [ ! -e $__INIT_TOOLS_DONE_MARKER ]; t
     if [ ! -e $__BUILD_TOOLS_PATH ]; then
         echo "Restoring BuildTools version $__BUILD_TOOLS_PACKAGE_VERSION..."
         echo "Running: $__DOTNET_CMD restore \"$__PROJECT_JSON_FILE\" --no-cache --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE"
+        LD_LIBRARY_PATH="$__DOTNET_PATH/shared/Microsoft.NETCore.App/1.0.0" \
         $__DOTNET_CMD restore "$__PROJECT_JSON_FILE" --no-cache --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE
         if [ ! -e "$__BUILD_TOOLS_PATH/init-tools.sh" ]; then echo "ERROR: Could not restore build tools correctly. See '$__init_tools_log' for more details."1>&2; fi
     fi
