diff -urp opt.orig/google/earth/pro/google-earth-pro.desktop opt/google/earth/pro/google-earth-pro.desktop
--- opt.orig/google/earth/pro/google-earth-pro.desktop	2017-08-18 10:06:26.000000000 -0700
+++ opt/google/earth/pro/google-earth-pro.desktop	2017-08-31 12:43:11.129543796 -0700
@@ -3,10 +3,10 @@ Version=1.0
 Name=Google Earth
 GenericName=3D planet viewer
 Comment=Explore, search and discover the planet
-Exec=/opt/google/earth/pro/google-earth-pro %f
+Exec=/opt/googleearthpro/google-earth-pro %f
 Terminal=false
-MultipleArgs=false
+X-MultipleArgs=false
 Icon=google-earth-pro
 Type=Application
-Categories=Application;Network
-MimeType=application/vnd.google-earth.kml+xml;application/vnd.google-earth.kmz;application/earthviewer;application/keyhole
+Categories=Network;
+MimeType=application/vnd.google-earth.kml+xml;application/vnd.google-earth.kmz;application/earthviewer;application/keyhole;
