--- GDevelop-5.0.0-beta97/ExtLibs/CMakeLists.txt.orig	2020-06-21 06:45:08.000000000 -0700
+++ GDevelop-5.0.0-beta97/ExtLibs/CMakeLists.txt	2020-07-26 03:57:21.630438407 -0700
@@ -2,7 +2,8 @@
 # do it if not already cloned to avoid triggering a whole
 # recompilation every time CMake is run.
 find_package(Git)
-if(GIT_FOUND AND NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/SFML/readme.txt")
+if(GIT_FOUND)
+	if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/SFML/readme.txt")
 	message( "Cloning SFML in ExtLibs/SFML with Git..." )
 	execute_process(
 		COMMAND ${GIT_EXECUTABLE} clone "https://www.github.com/SFML/SFML.git" SFML
@@ -14,6 +15,7 @@ if(GIT_FOUND AND NOT EXISTS "${CMAKE_CUR
 		COMMAND ${GIT_EXECUTABLE} reset --hard 2.4.1
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/SFML
 		OUTPUT_QUIET)
+	endif()
 
 	message( "Applying the patches..." )
 	file(GLOB SFML_PATCHES
