--- a/GDevelop.js/Gruntfile.js.orig	2025-01-02 06:16:30.000000000 -0800
+++ b/GDevelop.js/Gruntfile.js	2025-01-19 09:07:59.802609724 -0800
@@ -4,6 +4,8 @@ module.exports = function (grunt) {
   const fs = require('fs');
   const path = require('path');
   const isWin = /^win/.test(process.platform);
+  const isLinux = /^linux/.test(process.platform);
+  const isNinja = grunt.option('ninja') || false;
   const useMinGW = grunt.option('use-MinGW') || false;
 
   const possibleVariants = [
@@ -71,6 +73,19 @@ module.exports = function (grunt) {
         "⚠️ Can't find CMake in its usual Program Files folder. Make sure you have cmake in your PATH instead."
       );
     }
+  } else if (isLinux) {
+      cmakeBinary = 'emcmake cmake';
+      if (isNinja) {
+        cmakeGeneratorArgs = [
+          '-G "Ninja"',
+          `-DCMAKE_MAKE_PROGRAM="/usr/bin/ninja"`,
+        ];
+      } else {
+        cmakeGeneratorArgs = [
+          '-G "Unix Makefiles"',
+          `-DCMAKE_MAKE_PROGRAM="/usr/bin/make"`,
+        ];
+      }
   }
 
   grunt.initConfig({
