This file contains Unix (LF) line endings.
--- a/Extensions/CMakeUtils.txt.orig	2023-11-29 21:02:56.974994320 -0800
+++ b/Extensions/CMakeUtils.txt	2023-11-29 21:04:20.270996489 -0800
@@ -28,6 +28,7 @@ function(gd_add_extension_definitions ta
 		add_definitions(-DMACOS)
 	else()
 		add_definitions(-DLINUX)
+		add_definitions( -D__linux__ )
 	endif()
 
 	if(WIN32) # Windows specific defines
--- a/GDevelop.js/CMakeLists.txt.orig	2023-11-29 21:02:56.978994320 -0800
+++ b/GDevelop.js/CMakeLists.txt	2023-11-29 21:05:22.014998097 -0800
@@ -44,6 +44,7 @@ endif()
 
 add_definitions(-Dlinux)
 add_definitions(-DLINUX)
+add_definitions(-D__linux__)
 add_definitions(-DGD_API=)
 add_definitions(-DGD_CORE_API=)
 add_definitions(-DGD_EXTENSION_API=)
--- a/GDJS/CMakeLists.txt.orig	2023-11-29 21:02:56.978994320 -0800
+++ b/GDJS/CMakeLists.txt	2023-11-29 21:05:45.438998707 -0800
@@ -27,6 +27,7 @@ else()
 		add_definitions(-DMACOS)
 	else()
 		add_definitions(-DLINUX)
+		add_definitions(-D__linux__)
 	endif()
 	add_definitions(-DGD_API=)
 	add_definitions(-DGD_CORE_API=)
