--- GDevelop-5.0.0-beta97/GDevelop.js/update-bindings.js.orig	2020-06-21 06:45:08.000000000 -0700
+++ GDevelop-5.0.0-beta97/GDevelop.js/update-bindings.js	2020-07-26 04:13:40.998618324 -0700
@@ -9,13 +9,7 @@ var debug = false; //If true, add additi
 var fs = require('fs');
 var exec = require('child_process').exec;
 
-if (!process.env.EMSDK) {
-  console.error('EMSDK env. variable is not set');
-  console.log(
-    'Please set Emscripten environment by launching `emsdk_env` script'
-  );
-}
-var webIdlBinderPath = process.env.EMSDK + '/upstream/emscripten/tools/webidl_binder.py';
+var webIdlBinderPath = process.env.EMSCRIPTEN + '/tools/webidl_binder.py';
 
 generateGlueFromBinding(function(err) {
   if (err) return fatalError(err);
