--- a/newIDE/app/scripts/service-worker-template/service-worker-template.js.orig	2025-01-02 06:16:30.000000000 -0800
+++ b/newIDE/app/scripts/service-worker-template/service-worker-template.js	2025-01-19 15:09:10.668259335 -0800
@@ -1,3 +1,4 @@
+import { precacheAndRoute } from 'workbox-precaching/precacheAndRoute';
 if (typeof importScripts === 'function') {
   importScripts(
     'https://storage.googleapis.com/workbox-cdn/releases/3.5.0/workbox-sw.js'
@@ -22,7 +23,7 @@ if (typeof importScripts === 'function')
     ]);
 
     /* injection point for manifest files.  */
-    workbox.precaching.precacheAndRoute([]);
+    workbox.precaching.precacheAndRoute(self.__WB_MANIFEST);
 
     /* custom cache rules*/
     workbox.routing.registerNavigationRoute('/index.html', {
