--- a/GDJS/Runtime/runtimegame.ts.orig	2025-01-02 06:16:30.000000000 -0800
+++ b/GDJS/Runtime/runtimegame.ts	2025-01-19 08:40:53.464285182 -0800
@@ -185,7 +185,7 @@ namespace gdjs {
      */
     _debuggerClient: gdjs.AbstractDebuggerClient | null;
     _sessionMetricsInitialized: boolean = false;
-    _disableMetrics: boolean = false;
+    _disableMetrics: boolean = true;
     _isPreview: boolean;
 
     /**
@@ -961,7 +961,7 @@ namespace gdjs {
           }
         });
         setTimeout(() => {
-          this._setupSessionMetrics();
+          ;
         }, 4000);
         if (this._captureManager) {
           this._captureManager.setupCaptureOptions(this._isPreview);
@@ -991,10 +991,7 @@ namespace gdjs {
      * Set if the session should be registered.
      */
     enableMetrics(enable: boolean): void {
-      this._disableMetrics = !enable;
-      if (enable) {
-        this._setupSessionMetrics();
-      }
+      ;
     }
 
     /**
@@ -1039,6 +1036,7 @@ namespace gdjs {
      * time.
      */
     _setupSessionMetrics() {
+      return;
       if (this._sessionMetricsInitialized) {
         return;
       }
@@ -1054,7 +1052,7 @@ namespace gdjs {
       if (!this._data.properties.projectUuid) {
         return;
       }
-      const baseUrl = 'https://api.gdevelop-app.com/analytics';
+      const baseUrl = '';
       this._playerId = this._makePlayerUuid();
       /**
        * The duration that is already sent to the service
--- a/newIDE/app/src/Utils/GDevelopServices/ApiConfigs.js.orig	2025-01-02 06:16:30.000000000 -0800
+++ b/newIDE/app/src/Utils/GDevelopServices/ApiConfigs.js	2025-01-19 08:39:23.376809168 -0800
@@ -67,8 +67,8 @@ export const GDevelopAssetApi = {
 
 export const GDevelopAnalyticsApi = {
   baseUrl: isDev
-    ? 'https://api-dev.gdevelop.io/analytics'
-    : 'https://api.gdevelop.io/analytics',
+    ? ''
+    : '',
 };
 
 export const GDevelopGameApi = {
--- a/Extensions/Firebase/tests/FirebaseExtension.spec.js.orig	2025-01-02 06:16:30.000000000 -0800
+++ b/Extensions/Firebase/tests/FirebaseExtension.spec.js	2025-01-19 08:39:23.376809168 -0800
@@ -36,7 +36,6 @@ const firebaseConfig = {
   storageBucket: 'gdtest-e11a5.appspot.com',
   messagingSenderId: '254035412678',
   appId: '1:254035412678:web:2ddd6b83019b7f259b79c7',
-  measurementId: 'G-4REML26L59',
 };
 
 // The tests require an internet connection, as a real Firebase instance is used.
