--- a/CMakeLists.txt.orig	2022-02-01 04:15:41.000000000 -0800
+++ b/CMakeLists.txt	2022-02-08 19:20:20.448352287 -0800
@@ -59,12 +59,26 @@ ENDIF()
 
 #Activate C++11
 include(CheckCXXCompilerFlag)
-CHECK_CXX_COMPILER_FLAG("-std=gnu++11" COMPILER_SUPPORTS_CXX11)
-CHECK_CXX_COMPILER_FLAG("-std=gnu++0x" COMPILER_SUPPORTS_CXX0X)
+string(FIND "${CMAKE_CXX_COMPILER}" "em++" FOUND_EMXX)
+if(FOUND_EMXX GREATER -1)
+	# Broken test
+	execute_process(COMMAND "${CMAKE_CXX_COMPILER} --cflags -std=gnu++11" RESULT_VARIABLE COMPILER_SUPPORTS_CXX11)
+	execute_process(COMMAND "${CMAKE_CXX_COMPILER} --cflags -std=gnu++0x" RESULT_VARIABLE COMPILER_SUPPORTS_CXX0X)
+	if(COMPILER_SUPPORTS_CXX11)
+		message("COMPILER_SUPPORTS_CXX11 - Success")
+	endif()
+	if(COMPILER_SUPPORTS_CXX0X)
+		message("COMPILER_SUPPORTS_CXX11 - Success")
+	endif()
+
+else()
+	CHECK_CXX_COMPILER_FLAG("-std=gnu++11" COMPILER_SUPPORTS_CXX11)
+	CHECK_CXX_COMPILER_FLAG("-std=gnu++0x" COMPILER_SUPPORTS_CXX0X)
+endif()
 if(COMPILER_SUPPORTS_CXX11)
        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
 elseif(COMPILER_SUPPORTS_CXX0X)
        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++0x")
 else()
     message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support (with GNU extensions). Please use a different C++ compiler.")
 endif()
