--- LateralGM-1.8.6.844.orig/org/lateralgm/messages/Messages.java	2016-01-31 19:32:37.756893335 -0800
+++ LateralGM-1.8.6.844/org/lateralgm/messages/Messages.java	2016-01-31 20:01:45.571584567 -0800
@@ -50,6 +50,11 @@
 	public static void updateLangPack()
 		{
 		String langbundle = LANGUAGE_BUNDLE_NAME;
+		if (Prefs.languageName == null)
+			{
+			LANGUAGE_BUNDLE = ResourceBundle.getBundle(langbundle, new Locale("",""));
+			return;
+			}
 		if (Prefs.languageName.contains("English"))
 			{
 			langbundle = LANGUAGE_BUNDLE_NAME; //$NON-NLS-1$
