--- a/drivers/gpu/drm/scheduler/sched_main.c.orig	2019-02-12 20:59:18.753040052 -0800
+++ b/drivers/gpu/drm/scheduler/sched_main.c	2019-02-12 21:37:48.021944535 -0800
@@ -380,8 +380,6 @@ void drm_sched_stop(struct drm_gpu_sched
 	unsigned long flags;
 	struct dma_fence *last_fence =  NULL;
 
-			if (s_fence->finished.error < 0)
-				drm_sched_expel_job_unlocked(s_job);
 	kthread_park(sched->thread);
 
 	/*
@@ -571,8 +569,6 @@ void drm_sched_wakeup(struct drm_gpu_sch
  */
 static struct drm_sched_entity *
 drm_sched_select_entity(struct drm_gpu_scheduler *sched)
-			if (s_fence->finished.error < 0)
-				drm_sched_expel_job_unlocked(sched_job);
 {
 	struct drm_sched_entity *entity;
 	int i;
