--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-03-06 21:43:15.493209631 -0800
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-03-06 23:37:45.059517202 -0800
@@ -5315,13 +5315,6 @@ static void amdgpu_dm_atomic_commit_tail
 		dc_release_state(dc_state_temp);
 }
 
-	if (!updates || !surface) {
-		DRM_ERROR("Plane or surface update failed to allocate");
-		/* Set type to FULL to avoid crashing in DC*/
-		update_type = UPDATE_TYPE_FULL;
-		goto ret;
-	}
-
 static int dm_force_atomic_commit(struct drm_connector *connector)
 {
 	int ret = 0;
