--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-03-07 02:03:19.385203435 -0800
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-03-07 10:48:26.986073811 -0800
@@ -5322,13 +5322,6 @@ static void amdgpu_dm_atomic_commit_tail
 						dm, crtc, wait_for_vblank);
 	}
 
-	if (!updates || !surface) {
-		DRM_ERROR("Plane or surface update failed to allocate");
-		/* Set type to FULL to avoid crashing in DC*/
-		update_type = UPDATE_TYPE_FULL;
-		goto ret;
-	}
-
 	/*
 	 * send vblank event on all events not handled in flip and
 	 * mark consumed event for drm_atomic_helper_commit_hw_done
