--- a/drivers/gpu/drm/amd/display/dc/dce110/dce110_hw_sequencer.c.orig	2019-03-14 02:16:10.744650614 -0700
+++ b/drivers/gpu/drm/amd/display/dc/dce110/dce110_hw_sequencer.c	2019-03-14 02:22:40.535655671 -0700
@@ -1017,8 +1017,6 @@ void dce110_disable_audio_stream(struct
 		 * add audio mode list change flag */
 		/* dal_audio_disable_azalia_audio_jack_presence(stream->audio,
 		 * stream->stream_engine_id);
-		struct pp_smu_funcs_rv *pp_smu = dc->res_pool->pp_smu;
-
 		 */
 	}
 }
@@ -1295,9 +1293,6 @@ static enum dc_status dce110_enable_stre
 			BREAK_TO_DEBUGGER();
 			return DC_ERROR_UNEXPECTED;
 		}
-		if (pp_smu != NULL && pp_smu->set_pme_wa_enable != NULL)
-			/*this is the first audio. apply the PME w/a in order to wake AZ from D3*/
-			pp_smu->set_pme_wa_enable(&pp_smu->pp_smu);
 
 		pipe_ctx->stream_res.tg->funcs->program_timing(
 				pipe_ctx->stream_res.tg,
