--- a/net/core/skbuff.c.orig	2021-02-07 21:45:52.665548430 -0800
+++ b/net/core/skbuff.c	2021-02-07 21:46:47.871691098 -0800
@@ -473,6 +473,7 @@ struct sk_buff *__napi_alloc_skb(struct
 	struct napi_alloc_cache *nc;
 	struct sk_buff *skb;
 	void *data;
+	bool pfmemalloc;
 
 	len += NET_SKB_PAD + NET_IP_ALIGN;
 
@@ -488,7 +489,6 @@ struct sk_buff *__napi_alloc_skb(struct
 		goto skb_success;
 	}
 
-	nc = this_cpu_ptr(&napi_alloc_cache);
 	len += SKB_DATA_ALIGN(sizeof(struct skb_shared_info));
 	len = SKB_DATA_ALIGN(len);
 
