--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c.orig	2019-05-13 17:25:39.105657215 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c	2019-05-13 17:26:38.967986142 -0700
@@ -2863,14 +2863,9 @@ int amdgpu_vm_make_compute(struct amdgpu
 		return r;
 
 	/* Sanity checks */
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 14, 0)
-	if (!RB_EMPTY_ROOT(&vm->va) || vm->root.entries) {
-#else
-	if (!RB_EMPTY_ROOT(&vm->va.rb_root) || vm->root.entries) {
-#endif
-		r = -EINVAL;
+	r = amdgpu_vm_check_clean_reserved(adev, vm);
+	if (r)
 		goto unreserve_bo;
-	}
 
 	if (pasid) {
 		unsigned long flags;
