--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_cs.c.orig	2019-05-12 22:20:55.985300545 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_cs.c	2019-05-12 22:24:36.573804407 -0700
@@ -1167,10 +1167,11 @@ static int amdgpu_cs_process_fence_dep(s
 
 #if DRM_VERSION_CODE >= DRM_VERSION(4, 13, 0)
 static int amdgpu_syncobj_lookup_and_add_to_sync(struct amdgpu_cs_parser *p,
-						 uint32_t handle)
+						 uint32_t handle, u64 point,
+						 u64 flags)
 {
-	int r;
 	struct dma_fence *fence;
+	int r;
 #if defined(BUILD_AS_DKMS)
 #if DRM_VERSION_CODE < DRM_VERSION(4, 14, 0)
 	r = drm_syncobj_fence_get(p->filp, handle, &fence);
@@ -1184,8 +1185,12 @@ static int amdgpu_syncobj_lookup_and_add
 #else
 	r = drm_syncobj_find_fence(p->filp, handle, 0, 0, &fence);
 #endif
-	if (r)
+	r = drm_syncobj_find_fence(p->filp, handle, point, flags, &fence);
+	if (r) {
+		DRM_ERROR("syncobj %u failed to find fence @ %llu (%d)!\n",
+			  handle, point, r);
 		return r;
+	}
 
 	r = amdgpu_sync_fence(p->adev, &p->job->sync, fence, true);
 	dma_fence_put(fence);
@@ -1322,20 +1327,34 @@ static int amdgpu_cs_dependencies(struct
 
 		chunk = &p->chunks[i];
 
-		if (chunk->chunk_id == AMDGPU_CHUNK_ID_DEPENDENCIES ||
-		    chunk->chunk_id == AMDGPU_CHUNK_ID_SCHEDULED_DEPENDENCIES) {
+		switch (chunk->chunk_id) {
+		case AMDGPU_CHUNK_ID_DEPENDENCIES:
+		case AMDGPU_CHUNK_ID_SCHEDULED_DEPENDENCIES:
 			r = amdgpu_cs_process_fence_dep(p, chunk);
 			if (r)
 				return r;
+			break;
 #if DRM_VERSION_CODE >= DRM_VERSION(4, 13, 0)
-		} else if (chunk->chunk_id == AMDGPU_CHUNK_ID_SYNCOBJ_IN) {
+		case AMDGPU_CHUNK_ID_SYNCOBJ_IN:
 			r = amdgpu_cs_process_syncobj_in_dep(p, chunk);
 			if (r)
 				return r;
-		} else if (chunk->chunk_id == AMDGPU_CHUNK_ID_SYNCOBJ_OUT) {
+			break;
+		case AMDGPU_CHUNK_ID_SYNCOBJ_OUT:
 			r = amdgpu_cs_process_syncobj_out_dep(p, chunk);
 			if (r)
 				return r;
+			break;
+		case AMDGPU_CHUNK_ID_SYNCOBJ_TIMELINE_WAIT:
+			r = amdgpu_cs_process_syncobj_timeline_in_dep(p, chunk);
+			if (r)
+				return r;
+			break;
+		case AMDGPU_CHUNK_ID_SYNCOBJ_TIMELINE_SIGNAL:
+			r = amdgpu_cs_process_syncobj_timeline_out_dep(p, chunk);
+			if (r)
+				return r;
+			break;
 #endif
 		}
 	}
