--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-13 13:36:42.858708476 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-13 13:39:48.785959960 -0700
@@ -6526,15 +6526,15 @@ static void amdgpu_dm_atomic_commit_tail
 #if DRM_VERSION_CODE >= DRM_VERSION(4, 10, 0)
 #ifdef CONFIG_DEBUG_FS
 		/* The stream has changed so CRC capture needs to re-enabled. */
-		if (dm_new_crtc_state->crc_enabled)
+		if (dm_new_crtc_state->crc_enabled) {
+			dm_new_crtc_state->crc_enabled = false;
 #if DRM_VERSION_CODE >= DRM_VERSION(4, 20, 0) || !defined(BUILD_AS_DKMS)
 			amdgpu_dm_crtc_set_crc_source(crtc, "auto");
 #else
-		{
 			size_t values_cnt;
 			amdgpu_dm_crtc_set_crc_source(crtc, "auto", &values_cnt);
-		}
 #endif
+		}
 #endif
 #endif
 	}
