--- a/drivers/gpu/drm/amd/display/dc/core/dc.c.orig	2019-07-10 23:49:37.653161582 -0700
+++ b/drivers/gpu/drm/amd/display/dc/core/dc.c	2019-07-11 00:18:58.211829372 -0700
@@ -2062,14 +2062,6 @@ void dc_commit_updates_for_stream(struct
 		update_surface_trace(dc, srf_updates, surface_count);
 
 
-		for (i = 0; i < dc->res_pool->pipe_count; i++) {
-			struct pipe_ctx *new_pipe = &context->res_ctx.pipe_ctx[i];
-			struct pipe_ctx *old_pipe = &dc->current_state->res_ctx.pipe_ctx[i];
-
-			if (new_pipe->plane_state && new_pipe->plane_state != old_pipe->plane_state)
-				new_pipe->plane_state->force_full_update = true;
-		}
-
 	if (update_type >= UPDATE_TYPE_FULL) {
 
 		/* initialize scratch memory for building context */
