--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_dma_buf.c.orig	2019-10-06 17:14:30.484159895 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_dma_buf.c	2019-10-06 17:16:18.747315618 -0700
@@ -408,11 +408,11 @@ amdgpu_gem_prime_import_sg_table(struct
 	if (attach->dmabuf->ops != &amdgpu_dmabuf_ops)
 		bo->prime_shared_count = 1;
 
-	reservation_object_unlock(resv);
+	dma_resv_unlock(resv);
 	return &bo->gem_base;
 
 error:
-	reservation_object_unlock(resv);
+	dma_resv_unlock(resv);
 	return ERR_PTR(ret);
 }
 
