--- a/drivers/gpu/drm/v3d/v3d_gem.c.orig	2019-02-12 20:32:17.700764075 -0800
+++ b/drivers/gpu/drm/v3d/v3d_gem.c	2019-02-12 20:35:45.298748914 -0800
@@ -584,8 +584,7 @@ v3d_submit_cl_ioctl(struct drm_device *d
 	/* Update the return sync object for the */
 	sync_out = drm_syncobj_find(file_priv, args->out_sync);
 	if (sync_out) {
-		drm_syncobj_replace_fence(sync_out, 0,
-					  &exec->render.base.s_fence->finished);
+		drm_syncobj_replace_fence(sync_out, exec->render_done_fence);
 		drm_syncobj_put(sync_out);
 	}
 
