--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-13 09:29:41.732204010 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-13 09:32:34.018909477 -0700
@@ -3172,6 +3172,10 @@ static int fill_plane_attributes(struct
 	if (ret)
 		return ret;
 
+	ret = fill_plane_color_attributes(plane_state, dc_plane_state,
+					  &dc_plane_state->color_space);
+	if (ret)
+		return ret;
 
 	/* In case of gamma set, update gamma value */
 #if DRM_VERSION_CODE >= DRM_VERSION(4, 6, 0)
@@ -3700,11 +3704,6 @@ create_stream_for_sink(struct amdgpu_dm_
 	if (!dm_state)
 		drm_mode_set_crtcinfo(&mode, 0);
 
-	ret = fill_plane_color_attributes(plane_state, dc_plane_state,
-					  &dc_plane_state->color_space);
-	if (ret)
-		return ret;
-
 	/*
 	* If scaling is enabled and refresh rate didn't change
 	* we copy the vic and polarities of the old timings
