--- a/drivers/gpu/drm/amd/powerplay/hwmgr/vega20_hwmgr.c.orig	2019-02-12 08:59:43.238471050 -0800
+++ b/drivers/gpu/drm/amd/powerplay/hwmgr/vega20_hwmgr.c	2019-02-12 09:39:27.849607891 -0800
@@ -921,16 +921,6 @@ static int vega20_send_clock_ratio(struc
 			data->registry_data.fclk_gfxclk_ratio);
 }
 
-static int vega20_send_clock_ratio(struct pp_hwmgr *hwmgr)
-{
-	struct vega20_hwmgr *data =
-			(struct vega20_hwmgr *)(hwmgr->backend);
-
-	return smum_send_msg_to_smc_with_parameter(hwmgr,
-			PPSMC_MSG_SetFclkGfxClkRatio,
-			data->registry_data.fclk_gfxclk_ratio);
-}
-
 static int vega20_disable_all_smu_features(struct pp_hwmgr *hwmgr)
 {
 	struct vega20_hwmgr *data =
