--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-04-18 14:26:20.960139654 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-04-18 14:27:43.031291909 -0700
@@ -5273,12 +5273,13 @@ static void amdgpu_dm_commit_planes(stru
 
 			amdgpu_bo_unreserve(abo);
 
-			fill_plane_buffer_attributes(dm->adev, afb, dc_plane,
-				&bundle->plane_infos[planes_count].tiling_info,
-				&bundle->plane_infos[planes_count].plane_size,
-				&bundle->plane_infos[planes_count].dcc,
-				&bundle->flip_addrs[planes_count].address,
-				tiling_flags);
+			fill_dc_plane_info_and_addr(
+				dm->adev, new_plane_state, tiling_flags,
+				&bundle->plane_infos[planes_count],
+				&bundle->flip_addrs[planes_count].address);
+
+			bundle->surface_updates[planes_count].plane_info =
+				&bundle->plane_infos[planes_count];
 
 			bundle->flip_addrs[planes_count].flip_immediate =
 					(crtc->state->pageflip_flags & DRM_MODE_PAGE_FLIP_ASYNC) != 0;
