--- a/drivers/gpu/drm/v3d/v3d_sched.c.orig	2019-05-04 18:40:16.717172629 -0700
+++ b/drivers/gpu/drm/v3d/v3d_sched.c	2019-05-04 18:48:36.646547507 -0700
@@ -234,7 +234,7 @@ v3d_gpu_reset_for_timeout(struct v3d_dev
 	for (q = 0; q < V3D_MAX_QUEUES; q++) {
 		struct drm_gpu_scheduler *sched = &v3d->queue[q].sched;
 
-		drm_sched_stop(sched);
+		drm_sched_stop(sched, sched_job);
 
 		if(sched_job)
 			drm_sched_increase_karma(sched_job);
