--- a/drivers/gpu/drm/amd/amdkfd/kfd_events.c.orig	2019-05-10 12:06:27.997741026 -0700
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_events.c	2019-05-10 12:07:39.820329793 -0700
@@ -1027,15 +1027,25 @@ void kfd_signal_vm_fault_event(struct kf
 void kfd_signal_reset_event(struct kfd_dev *dev)
 {
 	struct kfd_hsa_hw_exception_data hw_exception_data;
+	struct kfd_hsa_memory_exception_data memory_exception_data;
 	struct kfd_process *p;
 	struct kfd_event *ev;
 	unsigned int temp;
 	uint32_t id, idx;
+	int reset_cause = atomic_read(&dev->sram_ecc_flag) ?
+			KFD_HW_EXCEPTION_ECC :
+			KFD_HW_EXCEPTION_GPU_HANG;
 
 	/* Whole gpu reset caused by GPU hang , and  memory is lost */
 	memset(&hw_exception_data, 0, sizeof(hw_exception_data));
 	hw_exception_data.gpu_id = dev->id;
 	hw_exception_data.memory_lost = 1;
+	hw_exception_data.reset_cause = reset_cause;
+
+	memset(&memory_exception_data, 0, sizeof(memory_exception_data));
+	memory_exception_data.ErrorType = KFD_MEM_ERR_SRAM_ECC;
+	memory_exception_data.gpu_id = dev->id;
+	memory_exception_data.failure.imprecise = true;
 
 	idx  = srcu_read_lock(&kfd_processes_srcu);
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 9, 0)
