--- a/drivers/gpu/drm/amd/powerplay/hwmgr/vega12_hwmgr.c.orig	2019-03-23 11:52:37.512021580 -0700
+++ b/drivers/gpu/drm/amd/powerplay/hwmgr/vega12_hwmgr.c	2019-03-23 11:54:09.607490224 -0700
@@ -770,22 +770,6 @@ static int vega12_run_acg_btc(struct pp_
 	return 0;
 }
 
-static int vega12_run_acg_btc(struct pp_hwmgr *hwmgr)
-{
-	uint32_t result;
-
-	PP_ASSERT_WITH_CODE(
-		smum_send_msg_to_smc(hwmgr, PPSMC_MSG_RunAcgBtc) == 0,
-		"[Run_ACG_BTC] Attempt to run ACG BTC failed!",
-		return -EINVAL);
-
-	result = smum_get_argument(hwmgr);
-	PP_ASSERT_WITH_CODE(result == 1,
-			"Failed to run ACG BTC!", return -EINVAL);
-
-	return 0;
-}
-
 static int vega12_set_allowed_featuresmask(struct pp_hwmgr *hwmgr)
 {
 	struct vega12_hwmgr *data =
@@ -965,11 +949,6 @@ static int vega12_enable_dpm_tasks(struc
 			result = tmp_result);
 
 	tmp_result = vega12_run_acg_btc(hwmgr);
-	PP_ASSERT_WITH_CODE(!tmp_result,
-			"Failed to run ACG BTC!",
-			result = tmp_result);
-
-	tmp_result = vega12_run_acg_btc(hwmgr);
 	PP_ASSERT_WITH_CODE(!tmp_result,
 			"Failed to run ACG BTC!",
 			result = tmp_result);
