--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_cs.c.orig	2019-02-12 01:28:54.413294734 -0800
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_cs.c	2019-02-12 01:30:53.067895436 -0800
@@ -964,10 +964,6 @@ static int amdgpu_cs_vm_handling(struct
 	if (r)
 		return r;
 
-	r = reservation_object_reserve_shared(vm->root.base.bo->tbo.resv);
-	if (r)
-		return r;
-
 	p->job->vm_pd_addr = amdgpu_gmc_pd_addr(vm->root.base.bo);
 
 	if (amdgpu_vm_debug) {
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c.orig	2019-02-12 01:29:47.525349708 -0800
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c	2019-02-12 01:32:16.361130186 -0800
@@ -774,10 +774,6 @@ static int amdgpu_vm_clear_bo(struct amd
 
 	ring = container_of(vm->entity.rq->sched, struct amdgpu_ring, sched);
 
-	r = reservation_object_reserve_shared(bo->tbo.resv);
-	if (r)
-		return r;
-
 	r = ttm_bo_validate(&bo->tbo, &bo->placement, &ctx);
 	if (r)
 		goto error;
@@ -1845,10 +1841,6 @@ static int amdgpu_vm_bo_update_mapping(s
 	if (r)
 		goto error_free;
 
-	r = reservation_object_reserve_shared(vm->root.base.bo->tbo.resv);
-	if (r)
-		goto error_free;
-
 	r = amdgpu_vm_update_ptes(&params, start, last + 1, addr, flags);
 	if (r)
 		goto error_free;
