--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-02-11 21:44:17.667001441 -0800
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-02-11 21:49:02.028044965 -0800
@@ -1582,8 +1582,18 @@ static int amdgpu_dm_backlight_update_st
 	 * bd->props.brightness is 8 bit format and needs to be converted by
 	 * scaling via copy lower byte to upper byte of 16 bit value.
 	 */
+	uint32_t brightness = bd->props.brightness * 0x101;
+
+ 	/*
+ 	 * PWM interperts 0 as 100% rather than 0% because of HW
+	 * limitation for level 0.  So limiting minimum brightness level
+ 	 * to 1.
+ 	 */
+ 	if (bd->props.brightness < 1)
+		brightness = 0x101;
+
 	if (dc_link_set_backlight_level(dm->backlight_link,
-			(bd->props.brightness * 0x101), 0, 0))
+			brightness, 0, 0))
 		return 0;
 	else
 		return 1;
