--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-15 17:27:38.889893026 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-15 17:30:55.517553691 -0700
@@ -5303,11 +5303,16 @@ static void amdgpu_dm_commit_planes(stru
 
 			abo = gem_to_amdgpu_bo(fb->obj[0]);
 
-			/* Wait for all fences on this FB */
+			/*
+			 * Wait for all fences on this FB. Do limited wait to avoid
+			 * deadlock during GPU reset when this fence will not signal
+			 * but we hold reservation lock for the BO.
+			 */
 			r = reservation_object_wait_timeout_rcu(abo->tbo.resv, true,
 								false,
-								MAX_SCHEDULE_TIMEOUT);
-			WARN_ON(r < 0);
+							msecs_to_jiffies(5000));
+			if (unlikely(r <= 0))
+				DRM_ERROR("Waiting for fences timed out or interrupted!");
 
 			/*
 			 * TODO This might fail and hence better not used, wait
