--- a/drivers/gpu/drm/ttm/ttm_execbuf_util.c.orig	2019-02-12 01:18:40.759609149 -0800
+++ b/drivers/gpu/drm/ttm/ttm_execbuf_util.c	2019-02-12 01:23:41.731228182 -0800
@@ -126,10 +126,11 @@ int ttm_eu_reserve_buffers(struct ww_acq
 		}
 
 		if (!ret) {
-			if (!entry->shared)
+			if (!entry->num_shared)
 				continue;
 
-			ret = reservation_object_reserve_shared(bo->resv);
+			ret = reservation_object_reserve_shared(bo->resv,
+								entry->num_shared);
 			if (!ret)
 				continue;
 		}
@@ -150,8 +151,9 @@ int ttm_eu_reserve_buffers(struct ww_acq
 			}
 		}
 
-		if (!ret && entry->shared)
-			ret = reservation_object_reserve_shared(bo->resv);
+		if (!ret && entry->num_shared)
+			ret = reservation_object_reserve_shared(bo->resv,
+								entry->num_shared);
 
 		if (unlikely(ret != 0)) {
 			if (ret == -EINTR)
