--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c.orig	2019-10-26 21:57:07.517835596 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c	2019-10-26 21:58:46.711679472 -0700
@@ -718,11 +718,6 @@ struct amdgpu_amdkfd_fence *to_amdgpu_am
 int amdgpu_amdkfd_evict_userptr(struct kgd_mem *mem, struct mm_struct *mm)
 {
 	return 0;
-struct kfd2kgd_calls *amdgpu_amdkfd_arcturus_get_functions(void)
-{
-	return NULL;
-}
-
 }
 
 struct kfd2kgd_calls *amdgpu_amdkfd_gfx_7_get_functions(void)
@@ -739,6 +734,11 @@ struct kfd2kgd_calls *amdgpu_amdkfd_gfx_
 {
 	return NULL;
 }
+
+struct kfd2kgd_calls *amdgpu_amdkfd_arcturus_get_functions(void)
+{
+	return NULL;
+}
 
 struct kfd_dev *kgd2kfd_probe(struct kgd_dev *kgd, struct pci_dev *pdev,
 			      const struct kfd2kgd_calls *f2g)
