--- a/drivers/gpu/drm/amd/powerplay/smu_v11_0.c.orig	2019-10-10 01:56:39.834709983 -0700
+++ b/drivers/gpu/drm/amd/powerplay/smu_v11_0.c	2019-10-10 01:58:41.845381090 -0700
@@ -1425,17 +1425,17 @@ smu_v11_0_get_fan_control_mode(struct sm
 }
 
 static int
-smu_v11_0_smc_fan_control(struct smu_context *smu, bool start)
+smu_v11_0_auto_fan_control(struct smu_context *smu, bool auto_fan_control)
 {
 	int ret = 0;
 
 	if (!smu_feature_is_supported(smu, SMU_FEATURE_FAN_CONTROL_BIT))
 		return 0;
 
-	ret = smu_feature_set_enabled(smu, SMU_FEATURE_FAN_CONTROL_BIT, start);
+	ret = smu_feature_set_enabled(smu, SMU_FEATURE_FAN_CONTROL_BIT, auto_fan_control);
 	if (ret)
 		pr_err("[%s]%s smc FAN CONTROL feature failed!",
-		       __func__, (start ? "Start" : "Stop"));
+		       __func__, (auto_fan_control ? "Start" : "Stop"));
 
 	return ret;
 }
