--- a/drivers/gpu/drm/amd/amdgpu/Makefile.orig	2019-10-26 04:36:00.086905313 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/Makefile	2019-10-26 04:39:57.855154689 -0700
@@ -53,7 +53,7 @@ amdgpu-y += amdgpu_device.o amdgpu_kms.o
 	amdgpu_ucode.o amdgpu_bo_list.o amdgpu_ctx.o amdgpu_sync.o \
 	amdgpu_gtt_mgr.o amdgpu_vram_mgr.o amdgpu_virt.o amdgpu_atomfirmware.o \
 	amdgpu_vf_error.o amdgpu_sched.o amdgpu_debugfs.o amdgpu_ids.o \
-	amdgpu_gmc.o amdgpu_xgmi.o amdgpu_csa.o amdgpu_sem.o amdgpu_ras.o \
+	amdgpu_gmc.o amdgpu_mmhub.o amdgpu_xgmi.o amdgpu_csa.o amdgpu_sem.o amdgpu_ras.o \
 	amdgpu_vm_cpu.o amdgpu_vm_sdma.o amdgpu_pmu.o amdgpu_discovery.o \
 	smu_v11_0_i2c.o
 
