--- a/include/linux/tcp.h.orig	2024-03-17 02:18:30.500390274 -0700
+++ b/include/linux/tcp.h	2024-03-17 02:19:18.300069169 -0700
@@ -370,7 +370,8 @@ struct tcp_sock {
 	u8	dup_ack_counter:2,
 		tlp_retrans:1,	/* TLP is a retransmission */
 		fast_ack_mode:2, /* which fast ack mode ? */
-		unused:3;
+		tlp_orig_data_app_limited:1, /* app-limited before TLP rtx? */
+		unused:2;
 	u8	thin_lto    : 1,/* Use linear timeouts for thin streams */
 		recvmsg_inq : 1,/* Indicate # of bytes in queue upon recvmsg */
 		fastopen_connect:1, /* FASTOPEN_CONNECT sockopt */
