--- a/kernel/locking/rtmutex.c.orig	2020-12-27 17:16:23.578038271 -0800
+++ b/kernel/locking/rtmutex.c	2020-12-27 17:18:51.483939743 -0800
@@ -242,7 +242,7 @@ static inline bool unlock_rt_mutex_safe(
  * Only use with rt_mutex_waiter_{less,equal}()
  */
 #define task_to_waiter(p)	\
-	&(struct rt_mutex_waiter){ .prio = (p)->prio, .deadline = (p)->dl.deadline, .task = (p) }
+	&(struct rt_mutex_waiter){ .prio = (p)->prio, .deadline = __tsk_deadline(p), .task = (p) }
 
 static inline int
 rt_mutex_waiter_less(struct rt_mutex_waiter *left,
