--- a/drivers/misc/lkdtm/perms.c.orig	2022-05-22 12:52:31.000000000 -0700
+++ b/drivers/misc/lkdtm/perms.c	2022-06-14 23:49:26.390010915 -0700
@@ -193,7 +193,7 @@ void lkdtm_EXEC_VMALLOC(void)
 
 void lkdtm_EXEC_RODATA(void)
 {
-	execute_location(dereference_function_descriptor(lkdtm_rodata_do_nothing),
+	execute_location(dereference_function_descriptor((void *)lkdtm_rodata_do_nothing),
 			 CODE_AS_IS);
 }
 
