--- a/Documentation/dev-tools/index.rst.orig	2024-01-07 12:18:38.000000000 -0800
+++ b/Documentation/dev-tools/index.rst	2024-03-16 17:29:12.330062405 -0700
@@ -34,6 +34,7 @@ Documentation/dev-tools/testing-overview
    kselftest
    kunit/index
    ktap
+   pgo
 
 
 .. only::  subproject and html
--- a/Makefile.orig	2024-03-16 17:27:07.363135550 -0700
+++ b/Makefile	2024-03-16 17:29:26.173943489 -0700
@@ -742,6 +742,9 @@ endif # KBUILD_EXTMOD
 # Defaults to vmlinux, but the arch makefile usually adds further targets
 all: vmlinux
 
+CFLAGS_PGO_CLANG := -fprofile-generate
+export CFLAGS_PGO_CLANG
+
 CFLAGS_GCOV	:=
 CFLAGS_GCOV_DEFAULT	:= 1
 
--- a/arch/x86/Kconfig.orig	2024-03-16 17:26:31.355444667 -0700
+++ b/arch/x86/Kconfig	2024-03-16 17:29:46.525768661 -0700
@@ -116,6 +116,7 @@ config X86
 	select ARCH_USES_CFI_TRAPS		if X86_64 && CFI_CLANG
 	select ARCH_SUPPORTS_LTO_CLANG
 	select ARCH_SUPPORTS_LTO_CLANG_THIN
+	select ARCH_SUPPORTS_PGO_CLANG		if X86_64
 	select ARCH_USE_BUILTIN_BSWAP
 	select ARCH_USE_CMPXCHG_LOCKREF		if X86_CMPXCHG64
 	select ARCH_USE_MEMTEST
--- a/arch/x86/um/vdso/Makefile.orig	2024-01-07 12:18:38.000000000 -0800
+++ b/arch/x86/um/vdso/Makefile	2024-03-16 17:30:01.401640865 -0700
@@ -67,3 +67,4 @@ quiet_cmd_vdso = VDSO    $@
 
 VDSO_LDFLAGS = -fPIC -shared -Wl,--hash-style=sysv -z noexecstack
 GCOV_PROFILE := n
+PGO_PROFILE := n
--- a/kernel/Makefile.orig	2024-01-07 12:18:38.000000000 -0800
+++ b/kernel/Makefile	2024-03-16 17:30:17.465502854 -0700
@@ -114,6 +114,7 @@ obj-$(CONFIG_SHADOW_CALL_STACK) += scs.o
 obj-$(CONFIG_HAVE_STATIC_CALL) += static_call.o
 obj-$(CONFIG_HAVE_STATIC_CALL_INLINE) += static_call_inline.o
 obj-$(CONFIG_CFI_CLANG) += cfi.o
+obj-$(CONFIG_PGO_CLANG) += pgo/
 
 obj-$(CONFIG_PERF_EVENTS) += events/
 
