--- a/Makefile.orig	2024-02-13 17:13:51.206625228 -0800
+++ b/Makefile	2024-02-13 17:14:55.961981070 -0800
@@ -686,6 +686,9 @@ endif # KBUILD_EXTMOD
 # Defaults to vmlinux, but the arch makefile usually adds further targets
 all: vmlinux
 
+CFLAGS_PGO_CLANG := -fprofile-generate
+export CFLAGS_PGO_CLANG
+
 CFLAGS_GCOV	:=
 CFLAGS_GCOV_DEFAULT	:= 1
 
--- a/arch/x86/Kconfig.orig	2024-02-13 17:13:15.646978907 -0800
+++ b/arch/x86/Kconfig	2024-02-13 17:15:22.905713008 -0800
@@ -108,6 +108,7 @@ config X86
 	select ARCH_SUPPORTS_KMAP_LOCAL_FORCE_MAP	if NR_CPUS <= 4096
 	select ARCH_SUPPORTS_LTO_CLANG
 	select ARCH_SUPPORTS_LTO_CLANG_THIN
+	select ARCH_SUPPORTS_PGO_CLANG		if X86_64
 	select ARCH_USE_BUILTIN_BSWAP
 	select ARCH_USE_MEMTEST
 	select ARCH_USE_QUEUED_RWLOCKS
--- a/kernel/Makefile.orig	2024-02-13 17:12:41.187321604 -0800
+++ b/kernel/Makefile	2024-02-13 17:15:45.741485798 -0800
@@ -116,6 +116,7 @@ obj-$(CONFIG_SHADOW_CALL_STACK) += scs.o
 obj-$(CONFIG_HAVE_STATIC_CALL) += static_call.o
 obj-$(CONFIG_HAVE_STATIC_CALL_INLINE) += static_call_inline.o
 obj-$(CONFIG_CFI_CLANG) += cfi.o
+obj-$(CONFIG_PGO_CLANG) += pgo/
 
 obj-$(CONFIG_PERF_EVENTS) += events/
 
