License header:

Makefile:
# SPDX-License-Identifier: GPL-2.0

--- a/Makefile.orig	2021-11-06 22:01:15.345623181 -0700
+++ b/Makefile	2021-11-06 22:02:14.375619346 -0700
@@ -674,6 +674,9 @@ endif # KBUILD_EXTMOD
 # Defaults to vmlinux, but the arch makefile usually adds further targets
 all: vmlinux
 
+CFLAGS_PGO_CLANG := -fprofile-generate
+export CFLAGS_PGO_CLANG
+
 CFLAGS_GCOV	:= -fprofile-arcs -ftest-coverage
 ifdef CONFIG_CC_IS_GCC
 CFLAGS_GCOV	+= -fno-tree-loop-im
