License identified for Makefile:
# SPDX-License-Identifier: GPL-2.0
--- a/Documentation/dev-tools/index.rst.orig	2022-03-20 13:14:17.000000000 -0700
+++ b/Documentation/dev-tools/index.rst	2022-03-24 13:28:20.164416861 -0700
@@ -33,6 +33,7 @@ Documentation/dev-tools/testing-overview
    kselftest
    kunit/index
    ktap
+   pgo
 
 
 .. only::  subproject and html
--- a/Makefile.orig	2022-03-24 13:19:48.507060520 -0700
+++ b/Makefile	2022-03-24 13:29:04.580183747 -0700
@@ -677,6 +677,9 @@ endif # KBUILD_EXTMOD
 # Defaults to vmlinux, but the arch makefile usually adds further targets
 all: vmlinux
 
+CFLAGS_PGO_CLANG := -fprofile-generate
+export CFLAGS_PGO_CLANG
+
 CFLAGS_GCOV	:= -fprofile-arcs -ftest-coverage
 ifdef CONFIG_CC_IS_GCC
 CFLAGS_GCOV	+= -fno-tree-loop-im
--- a/MAINTAINERS.orig	2022-03-24 13:27:25.836255613 -0700
+++ b/MAINTAINERS	2022-03-24 13:29:42.449690194 -0700
@@ -4710,6 +4710,19 @@ T:	git git://git.kernel.org/pub/scm/linu
 F:	include/linux/cfi.h
 F:	kernel/cfi.c
 
+CLANG PROFILE GUIDED OPTIMIZATION SUPPORT
+M:	Sami Tolvanen <samitolvanen@google.com>
+M:	Bill Wendling <wcw@google.com>
+M:	Kees Cook <keescook@chromium.org>
+R:	Nathan Chancellor <nathan@kernel.org>
+R:	Nick Desaulniers <ndesaulniers@google.com>
+L:	clang-built-linux@googlegroups.com
+S:	Supported
+B:	https://github.com/ClangBuiltLinux/linux/issues
+T:	git git://git.kernel.org/pub/scm/linux/kernel/git/kees/linux.git for-next/clang/features
+F:	Documentation/dev-tools/pgo.rst
+F:	kernel/pgo/
+
 CLK API
 M:	Russell King <linux@armlinux.org.uk>
 L:	linux-clk@vger.kernel.org
