License header for kernel/Makefile

# SPDX-License-Identifier: GPL-2.0

--- a/kernel/Makefile.orig	2023-02-11 05:36:30.125263391 -0800
+++ b/kernel/Makefile	2023-02-11 05:36:28.477263337 -0800
@@ -60,6 +60,7 @@ obj-$(CONFIG_PROFILING) += profile.o
 obj-$(CONFIG_STACKTRACE) += stacktrace.o
 obj-y += time/
 obj-$(CONFIG_FUTEX) += futex/
+obj-$(CONFIG_FUTEX2) += futex/
 obj-$(CONFIG_GENERIC_ISA_DMA) += dma.o
 obj-$(CONFIG_SMP) += smp.o
 ifneq ($(CONFIG_SMP),y)
