--- a/tools/testing/selftests/futex/functional/.gitignore.orig	2021-08-29 15:04:50.000000000 -0700
+++ b/tools/testing/selftests/futex/functional/.gitignore	2021-09-22 02:04:26.481617883 -0700
@@ -8,3 +8,4 @@ futex_wait_uninitialized_heap
 futex_wait_wouldblock
 futex_wait
 futex_requeue
+futex_wait_multiple
--- a/tools/testing/selftests/futex/functional/Makefile.orig	2021-08-29 15:04:50.000000000 -0700
+++ b/tools/testing/selftests/futex/functional/Makefile	2021-09-22 02:05:12.718619407 -0700
@@ -17,7 +17,8 @@ TEST_GEN_FILES := \
 	futex_wait_uninitialized_heap \
 	futex_wait_private_mapped_file \
 	futex_wait \
-	futex_requeue
+	futex_requeue \
+	futex_wait_multiple
 
 TEST_PROGS := run.sh
 
--- a/tools/testing/selftests/futex/functional/run.sh.orig	2021-08-29 15:04:50.000000000 -0700
+++ b/tools/testing/selftests/futex/functional/run.sh	2021-09-22 02:13:56.070636652 -0700
@@ -79,3 +79,6 @@ echo
 
 echo
 ./futex_requeue $COLOR
+
+echo
+./futex_wait_multiple $COLOR
