--- a/tools/testing/selftests/futex/functional/.gitignore.orig	2021-09-22 23:39:47.138178967 -0700
+++ b/tools/testing/selftests/futex/functional/.gitignore	2021-09-22 23:40:38.772180669 -0700
@@ -9,3 +9,4 @@ futex_wait_wouldblock
 futex_wait
 futex_requeue
 futex_wait_multiple
+futex2_wait
--- a/tools/testing/selftests/futex/functional/Makefile.orig	2021-09-22 23:40:03.368179502 -0700
+++ b/tools/testing/selftests/futex/functional/Makefile	2021-09-22 23:41:05.026181534 -0700
@@ -19,7 +19,8 @@ TEST_GEN_FILES := \
 	futex_wait_private_mapped_file \
 	futex_wait \
 	futex_requeue \
-	futex_wait_multiple
+	futex_wait_multiple \
+	futex2_wait
 
 TEST_PROGS := run.sh
 
--- a/tools/testing/selftests/futex/functional/run.sh.orig	2021-09-22 23:39:47.140178967 -0700
+++ b/tools/testing/selftests/futex/functional/run.sh	2021-09-22 23:41:20.562182046 -0700
@@ -82,3 +82,6 @@ echo
 
 echo
 ./futex_wait_multiple $COLOR
+
+echo
+./futex2_wait $COLOR
