--- a/include/asm-generic/vmlinux.lds.h.orig	2022-10-13 21:09:50.325123543 -0700
+++ b/include/asm-generic/vmlinux.lds.h	2022-10-13 21:10:30.770762207 -0700
@@ -587,20 +587,6 @@
 
 
 /*
- * .text..L.cfi.jumptable.* contain Control-Flow Integrity (CFI)
- * jump table entries.
- */
-#ifdef CONFIG_CFI_CLANG
-#define TEXT_CFI_JT							\
-		ALIGN_FUNCTION();					\
-		__cfi_jt_start = .;					\
-		*(.text..L.cfi.jumptable .text..L.cfi.jumptable.*)	\
-		__cfi_jt_end = .;
-#else
-#define TEXT_CFI_JT
-#endif
-
-/*
  * Non-instrumentable text section
  */
 #define NOINSTR_TEXT							\
