--- a/block/Kconfig.iosched.orig	2019-02-12 03:32:51.540724829 -0800
+++ b/block/Kconfig.iosched	2019-02-12 03:34:54.415453217 -0800
@@ -71,9 +71,18 @@ choice
 	config DEFAULT_DEADLINE
 		bool "Deadline" if IOSCHED_DEADLINE=y
 
+	config DEFAULT_MQ_DEADLINE
+		bool "MQ Deadline" if MQ_IOSCHED_DEADLINE=y
+
 	config DEFAULT_CFQ
 		bool "CFQ" if IOSCHED_CFQ=y
 
+	config DEFAULT_BFQ
+		bool "BFQ" if IOSCHED_BFQ=y
+
+	config DEFAULT_BFQ_MQ
+		bool "BFQ-MQ" if MQ_IOSCHED_BFQ=y
+
 	config DEFAULT_BFQ_SQ
 		bool "BFQ-SQ" if IOSCHED_BFQ_SQ=y
 		help
@@ -84,6 +93,9 @@ choice
 		  any workload. It also tries to guarantee low latency to
 		  interactive and soft real-time applications.
 
+	config DEFAULT_KYBER
+		bool "KYBER" if MQ_IOSCHED_KYBER=y
+
 	config DEFAULT_NOOP
 		bool "No-op"
 
@@ -92,8 +104,12 @@ endchoice
 config DEFAULT_IOSCHED
 	string
 	default "deadline" if DEFAULT_DEADLINE
+	default "mq-deadline" if DEFAULT_MQ_DEADLINE
 	default "cfq" if DEFAULT_CFQ
+	default "bfq" if DEFAULT_BFQ
+	default "bfq-mq" if DEFAULT_BFQ_MQ
 	default "bfq-sq" if DEFAULT_BFQ_SQ
+	default "kyber" if DEFAULT_KYBER
 	default "noop" if DEFAULT_NOOP
 
 config MQ_IOSCHED_BFQ
