--- a/block/Kconfig.iosched.orig	2019-02-12 03:27:26.588218815 -0800
+++ b/block/Kconfig.iosched	2019-02-12 03:27:07.997502068 -0800
@@ -50,9 +50,18 @@ choice
 	config DEFAULT_DEADLINE
 		bool "Deadline" if IOSCHED_DEADLINE=y
 
+	config DEFAULT_MQ_DEADLINE
+		bool "MQ Deadline" if MQ_IOSCHED_DEADLINE=y
+
 	config DEFAULT_CFQ
 		bool "CFQ" if IOSCHED_CFQ=y
 
+	config DEFAULT_BFQ
+		bool "BFQ" if IOSCHED_BFQ=y
+
+	config DEFAULT_KYBER
+		bool "KYBER" if MQ_IOSCHED_KYBER=y
+
 	config DEFAULT_NOOP
 		bool "No-op"
 
@@ -62,6 +71,9 @@ config DEFAULT_IOSCHED
 	string
 	default "deadline" if DEFAULT_DEADLINE
 	default "cfq" if DEFAULT_CFQ
+	default "bfq" if DEFAULT_BFQ
+	default "kyber" if DEFAULT_KYBER
+	default "mq-deadline" if DEFAULT_MQ_DEADLINE
 	default "noop" if DEFAULT_NOOP
 
 config MQ_IOSCHED_DEADLINE
