--- a/drivers/cpufreq/intel_pstate.c.orig	2025-05-18 01:34:25.610128125 -0700
+++ b/drivers/cpufreq/intel_pstate.c	2025-05-18 01:39:48.592511852 -0700
@@ -3136,6 +3136,12 @@ static int __init intel_pstate_setup(cha
 	if (!strcmp(str, "no_hwp"))
 		no_hwp = 1;
 
+	if (!strcmp(str, "no_turbo") && !global.no_turbo) {
+		global.no_turbo = 1;
+		intel_pstate_update_limits_for_all();
+		arch_set_max_freq_ratio(global.no_turbo);
+	}
+
 	if (!strcmp(str, "enable"))
 		no_load = 0;
 	if (!strcmp(str, "force"))
