--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c.orig	2019-10-30 19:36:42.345731618 -0700
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c	2019-10-30 20:41:45.318876404 -0700
@@ -1111,6 +1111,9 @@ static int initialize_cpsch(struct devic
 
 	INIT_WORK(&dqm->hw_exception_work, kfd_process_hw_exception);
 
+	if (dqm->dev->kfd2kgd->get_iq_wait_times)
+		dqm->dev->kfd2kgd->get_iq_wait_times(dqm->dev->kgd,
+					&dqm->wait_times);
 	return 0;
 }
 
