--- a/drivers/gpu/drm/amd/amdgpu/Makefile.orig	2019-11-02 15:28:25.201031774 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/Makefile	2019-11-02 16:46:58.702437325 -0700
@@ -162,6 +162,7 @@ amdgpu-y += \
 	 amdgpu_amdkfd_gpuvm.o \
 	 amdgpu_amdkfd_gfx_v8.o \
 	 amdgpu_amdkfd_gfx_v9.o \
+	 amdgpu_amdkfd_arcturus.o \
 	 amdgpu_amdkfd_gfx_v10.o
 
 ifneq ($(CONFIG_DRM_AMDGPU_CIK),)
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c.orig	2019-11-02 15:28:26.700092023 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c	2019-11-02 16:48:55.373950675 -0700
@@ -92,6 +92,9 @@ void amdgpu_amdkfd_device_probe(struct a
 	case CHIP_RAVEN:
 		kfd2kgd = amdgpu_amdkfd_gfx_9_0_get_functions();
 		break;
+	case CHIP_ARCTURUS:
+		kfd2kgd = amdgpu_amdkfd_arcturus_get_functions();
+		break;
 	case CHIP_NAVI10:
 		kfd2kgd = amdgpu_amdkfd_gfx_10_0_get_functions();
 		break;
