--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c.orig	2019-05-25 09:00:15.634417677 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c	2019-05-25 09:02:50.610443155 -0700
@@ -28,15 +28,20 @@
 #include <linux/module.h>
 #include <linux/dma-buf.h>
 
+#undef pr_fmt
+#define pr_fmt(fmt) "kfd2kgd: " fmt
+
 const struct kgd2kfd_calls *kgd2kfd;
 
-static const unsigned int compute_vmid_bitmap = 0xFF00;
+static unsigned int compute_vmid_bitmap = 0xFF00;
 
 /* Total memory size in system memory and all GPU VRAM. Used to
  * estimate worst case amount of memory to reserve for page tables
  */
 uint64_t amdgpu_amdkfd_total_mem_size;
 
+extern bool pcie_p2p;
+
 int amdgpu_amdkfd_init(void)
 {
 	struct sysinfo si;
@@ -643,6 +648,7 @@ struct kfd_dev *kgd2kfd_probe(struct kgd
 }
 
 bool kgd2kfd_device_init(struct kfd_dev *kfd,
+                         struct drm_device *ddev,
 			 const struct kgd2kfd_shared_resources *gpu_resources)
 {
 	return false;
