--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-25 22:50:07.155193662 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-25 22:54:03.134491865 -0700
@@ -310,13 +310,7 @@ static void dm_pflip_high_irq(void *inte
 		return;
 	}
 
-	/* Update to correct count(s) if racing with vblank irq */
-#if DRM_VERSION_CODE >= DRM_VERSION(4, 14, 0) || \
-			defined(OS_NAME_SUSE_15)
 	amdgpu_crtc->last_flip_vblank = drm_crtc_accurate_vblank_count(&amdgpu_crtc->base);
-#elif DRM_VERSION_CODE >= DRM_VERSION(4, 8, 0)
-	amdgpu_crtc->last_flip_vblank = drm_accurate_vblank_count(&amdgpu_crtc->base)
-#endif
 
 	/* wake up userspace */
 	if (amdgpu_crtc->event) {
@@ -5234,12 +5228,9 @@ static void amdgpu_dm_commit_planes(stru
 			to_dm_crtc_state(kcl_drm_atomic_get_old_crtc_state_after_commit(state, pcrtc));
 	int planes_count = 0, vpos, hpos;
 	unsigned long flags;
-	u64 last_flip_vblank;
-	bool vrr_active = acrtc_state->freesync_config.state == VRR_STATE_ACTIVE_VARIABLE;
 	struct amdgpu_bo *abo;
 	uint64_t tiling_flags, dcc_address;
 	uint32_t target, target_vblank;
-
 	bool pflip_present = false;
 	uint64_t last_flip_vblank;
 	bool vrr_active = acrtc_state->freesync_config.state == VRR_STATE_ACTIVE_VARIABLE;
