--- a/drivers/gpu/drm/amd/amdkfd/kfd_doorbell.c.orig	2019-05-27 13:15:02.834925757 -0700
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_doorbell.c	2019-05-27 13:45:29.295794575 -0700
@@ -130,7 +130,11 @@ void kfd_doorbell_fini(struct kfd_dev *k
 		iounmap(kfd->doorbell_kernel_ptr);
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0)
+static unsigned int kfd_doorbell_vm_fault(struct vm_fault *vmf)
+{
+	struct kfd_process *process = vmf->vma->vm_private_data;
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
 static int kfd_doorbell_vm_fault(struct vm_fault *vmf)
 {
 	struct kfd_process *process = vmf->vma->vm_private_data;
