--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-24 17:48:36.416919969 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-24 17:49:28.068930204 -0700
@@ -5274,8 +5274,7 @@ static void amdgpu_dm_commit_planes(stru
 
 		dc_plane = dm_new_plane_state->dc_state;
 
-		framebuffer_changed = old_plane_state->fb &&
-			old_plane_state->fb != new_plane_state->fb;
+		plane_needs_flip = old_plane_state->fb && new_plane_state->fb;
 
 		/* For variable refresh rate mode only:
 		 * Get vblank of last completed flip to avoid > 1 vrr flips per
@@ -5288,9 +5287,9 @@ static void amdgpu_dm_commit_planes(stru
 
 		spin_unlock_irqrestore(&crtc->dev->event_lock, flags);
 
-		pflip_present = pflip_present || framebuffer_changed;
+		pflip_present = pflip_present || plane_needs_flip;
 
-		if (framebuffer_changed) {
+		if (plane_needs_flip) {
 			/* Use old throttling in non-vrr fixed refresh rate mode
 			 * to keep flip scheduling based on target vblank counts
 			 * working in a backwards compatible way, e.g., clients
