--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-12 15:37:57.613755488 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-12 15:41:02.210885544 -0700
@@ -304,8 +304,13 @@ static void dm_pflip_high_irq(void *inte
 	}
 
 	/* Update to correct count(s) if racing with vblank irq */
+#if DRM_VERSION_CODE >= DRM_VERSION(4, 14, 0) || \
+		defined(OS_NAME_SUSE_15)
 	amdgpu_crtc->last_flip_vblank = drm_crtc_accurate_vblank_count(&amdgpu_crtc->base);
-
+#elif DRM_VERSION_CODE >= DRM_VERSION(4, 8, 0)
+		/* Update to correct count/ts if racing with vblank irq */
+	amdgpu_crtc->last_flip_vblank = drm_accurate_vblank_count(&amdgpu_crtc->base);
+#endif
 	/* wake up userspace */
 	if (amdgpu_crtc->event) {
 		drm_crtc_send_vblank_event(&amdgpu_crtc->base, amdgpu_crtc->event);
