--- a/drivers/gpu/drm/amd/amdkfd/kfd_priv.h.orig	2019-10-24 01:38:05.855305956 -0700
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_priv.h	2019-10-24 01:40:42.127344376 -0700
@@ -1183,11 +1183,15 @@ int kfd_ipc_init(void);
 /* Check with device cgroup if @kfd device is accessible */
 static inline int kfd_devcgroup_check_permission(struct kfd_dev *kfd)
 {
+#if defined(CONFIG_CGROUP_DEVICE)
 	struct drm_device *ddev = kfd->ddev;
 
 	return devcgroup_check_permission(DEVCG_DEV_CHAR, DRM_MAJOR,
 					  ddev->render->index,
 					  DEVCG_ACC_WRITE | DEVCG_ACC_READ);
+#else
+	return 0;
+#endif
 }
 
 /* Debugfs */
