--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c.orig	2019-05-12 15:44:27.138803615 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c	2019-05-12 15:46:56.864590513 -0700
@@ -2725,10 +2725,12 @@ void amdgpu_device_fini(struct amdgpu_de
 	/* disable all interrupts */
 	amdgpu_irq_disable_all(adev);
 	if (adev->mode_info.mode_config_initialized){
-		if (!amdgpu_device_has_dc_support(adev))
-			drm_helper_force_disable_all(adev->ddev);
-		else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
+		if (amdgpu_device_has_dc_support(adev))
 			drm_atomic_helper_shutdown(adev->ddev);
+		else
+#endif
+			drm_crtc_force_disable_all(adev->ddev);
 	}
 	amdgpu_ib_pool_fini(adev);
 	amdgpu_fence_driver_fini(adev);
