--- a/drivers/gpu/drm/amd/amdgpu/Makefile.orig	2019-09-15 14:19:32.000000000 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/Makefile	2019-10-23 14:17:42.236124311 -0700
@@ -53,8 +53,8 @@ amdgpu-y += amdgpu_device.o amdgpu_kms.o
 	amdgpu_ucode.o amdgpu_bo_list.o amdgpu_ctx.o amdgpu_sync.o \
 	amdgpu_gtt_mgr.o amdgpu_vram_mgr.o amdgpu_virt.o amdgpu_atomfirmware.o \
 	amdgpu_vf_error.o amdgpu_sched.o amdgpu_debugfs.o amdgpu_ids.o \
-	amdgpu_gmc.o amdgpu_xgmi.o amdgpu_csa.o amdgpu_ras.o amdgpu_vm_cpu.o \
-	amdgpu_vm_sdma.o amdgpu_discovery.o
+	amdgpu_gmc.o amdgpu_xgmi.o amdgpu_csa.o amdgpu_sem.o amdgpu_ras.o \
+	amdgpu_vm_cpu.o amdgpu_vm_sdma.o amdgpu_discovery.o
 
 amdgpu-$(CONFIG_PERF_EVENTS) += amdgpu_pmu.o
 
