--- a/drivers/gpu/drm/amd/amdgpu/dce_v10_0.c.orig	2019-05-12 13:07:27.662155772 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/dce_v10_0.c	2019-05-12 13:46:42.064657960 -0700
@@ -1682,7 +1682,12 @@ static void dce_v10_0_afmt_setmode(struc
 	dce_v10_0_audio_write_sad_regs(encoder);
 	dce_v10_0_audio_write_latency_fields(encoder, mode);
 
+#if DRM_VERSION_CODE < DRM_VERSION(4, 14, 0) && \
+	!defined(OS_NAME_SUSE_15)
+	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode);
+#else
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, connector, mode);
+#endif
 	if (err < 0) {
 		DRM_ERROR("failed to setup AVI infoframe: %zd\n", err);
 		return;
--- a/drivers/gpu/drm/amd/amdgpu/dce_v11_0.c.orig	2019-05-12 13:07:27.662155772 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/dce_v11_0.c	2019-05-12 13:47:11.325800016 -0700
@@ -1724,7 +1724,12 @@ static void dce_v11_0_afmt_setmode(struc
 	dce_v11_0_audio_write_sad_regs(encoder);
 	dce_v11_0_audio_write_latency_fields(encoder, mode);
 
+#if DRM_VERSION_CODE < DRM_VERSION(4, 14, 0) && \
+	!defined(OS_NAME_SUSE_15)
+	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode);
+#else
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, connector, mode);
+#endif
 	if (err < 0) {
 		DRM_ERROR("failed to setup AVI infoframe: %zd\n", err);
 		return;
--- a/drivers/gpu/drm/amd/amdgpu/dce_v6_0.c.orig	2019-05-12 13:07:27.662155772 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/dce_v6_0.c	2019-05-12 13:47:38.936877652 -0700
@@ -1431,7 +1431,12 @@ static void dce_v6_0_audio_set_avi_infof
 	ssize_t err;
 	u32 tmp;
 
+#if DRM_VERSION_CODE < DRM_VERSION(4, 14, 0) && \
+	!defined(OS_NAME_SUSE_15)
+	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode);
+#else
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, connector, mode);
+#endif
 	if (err < 0) {
 		DRM_ERROR("failed to setup AVI infoframe: %zd\n", err);
 		return;
--- a/drivers/gpu/drm/amd/amdgpu/dce_v8_0.c.orig	2019-05-12 13:07:27.662155772 -0700
+++ b/drivers/gpu/drm/amd/amdgpu/dce_v8_0.c	2019-05-12 13:48:06.947970881 -0700
@@ -1616,7 +1616,12 @@ static void dce_v8_0_afmt_setmode(struct
 	dce_v8_0_audio_write_sad_regs(encoder);
 	dce_v8_0_audio_write_latency_fields(encoder, mode);
 
+#if DRM_VERSION_CODE < DRM_VERSION(4, 14, 0) && \
+	!defined(OS_NAME_SUSE_15)
+	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode);
+#else
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, connector, mode);
+#endif
 	if (err < 0) {
 		DRM_ERROR("failed to setup AVI infoframe: %zd\n", err);
 		return;
