--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-25 23:35:31.181842040 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-25 23:41:01.366677143 -0700
@@ -311,12 +311,7 @@ static void dm_pflip_high_irq(void *inte
 	}
 
 	/* Update to correct count(s) if racing with vblank irq */
-#if DRM_VERSION_CODE >= DRM_VERSION(4, 14, 0) || \
-			defined(OS_NAME_SUSE_15)
 	amdgpu_crtc->last_flip_vblank = drm_crtc_accurate_vblank_count(&amdgpu_crtc->base);
-#elif DRM_VERSION_CODE >= DRM_VERSION(4, 8, 0)
-	amdgpu_crtc->last_flip_vblank = drm_accurate_vblank_count(&amdgpu_crtc->base);
-#endif
 
 	/* wake up userspace */
 	if (amdgpu_crtc->event) {
@@ -5322,8 +5317,6 @@ static void amdgpu_dm_commit_planes(stru
 	struct amdgpu_bo *abo;
 	uint64_t tiling_flags, dcc_address;
 	uint32_t target, target_vblank;
-	uint64_t last_flip_vblank;
-	bool vrr_active = acrtc_state->freesync_config.state == VRR_STATE_ACTIVE_VARIABLE;
 	bool pflip_present = false;
 	uint64_t last_flip_vblank;
 	bool vrr_active = acrtc_state->freesync_config.state == VRR_STATE_ACTIVE_VARIABLE;
