--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c.orig	2019-05-22 08:55:27.737663231 -0700
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c	2019-05-22 08:56:46.730682038 -0700
@@ -1156,16 +1156,9 @@ static int create_queue_cpsch(struct dev
 	}
 
 	if (q->properties.type == KFD_QUEUE_TYPE_SDMA) {
-		retval = allocate_sdma_queue(dqm, &q->sdma_id);
+		retval = allocate_sdma_queue(dqm, q);
 		if (retval)
 			goto out;
-		q->properties.sdma_queue_id =
-			q->sdma_id / get_num_sdma_engines(dqm);
-		q->properties.sdma_engine_id =
-			q->sdma_id % get_num_sdma_engines(dqm);
-		pr_debug("SDMA id is:    %d\n", q->sdma_id);
-		pr_debug("SDMA queue id: %d\n", q->properties.sdma_queue_id);
-		pr_debug("SDMA engine id: %d\n", q->properties.sdma_engine_id);
 	}
 
 	retval = allocate_doorbell(qpd, q);
