--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-05-12 02:05:38.792972804 -0700
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-05-12 02:06:24.184740861 -0700
@@ -5034,8 +5034,10 @@ static void amdgpu_dm_commit_planes(stru
 
 	flip = kzalloc(sizeof(*flip), GFP_KERNEL);
 
-	if (!flip)
+	if (!flip || !full) {
 		dm_error("Failed to allocate update bundles\n");
+		goto cleanup;
+	}
 	u64 last_flip_vblank;
 	bool vrr_active = acrtc_state->freesync_config.state == VRR_STATE_ACTIVE_VARIABLE;
 
