--- a/arch/x86/crypto/tresor_glue.c.orig	2024-02-10 01:47:14.915220656 -0800
+++ b/arch/x86/crypto/tresor_glue.c	2024-02-10 01:49:22.893719082 -0800
@@ -32,6 +32,7 @@
  * Copyright © 2012-2013 Jussi Kivilinna <jussi.kivilinna@iki.fi>
  */
 
+#include <asm/processor.h>
 #include <crypto/algapi.h>
 #include <crypto/internal/simd.h>
 #include <crypto/internal/skcipher.h>
@@ -155,6 +156,12 @@ void __tresor_decrypt_blk(int key_length
  */
 void tresor_encrypt(struct crypto_tfm *tfm, u8 *dst, const u8 *src)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return;
+
 	struct tresor_ctx *ctx = crypto_tfm_ctx(tfm);
 	__tresor_encrypt_blk(ctx->key_length, dst, src);
 }
@@ -164,6 +171,12 @@ void tresor_encrypt(struct crypto_tfm *t
  */
 void tresor_decrypt(struct crypto_tfm *tfm, u8 *dst, const u8 *src)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return;
+
 	struct tresor_ctx *ctx = crypto_tfm_ctx(tfm);
 	__tresor_decrypt_blk(ctx->key_length, dst, src);
 }
@@ -203,6 +216,12 @@ void __tresor_decrypt(const void *_ctx,
 
 static int ecb_encrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	ECB_WALK_START(req, AES_BLOCK_SIZE, TRESOR_PARALLEL_BLOCKS);
 	ECB_BLOCK(1, __tresor_encrypt);
 	ECB_WALK_END();
@@ -210,6 +229,12 @@ static int ecb_encrypt(struct skcipher_r
 
 static int ecb_decrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	ECB_WALK_START(req, AES_BLOCK_SIZE, TRESOR_PARALLEL_BLOCKS);
 	ECB_BLOCK(1, __tresor_decrypt);
 	ECB_WALK_END();
@@ -217,6 +242,12 @@ static int ecb_decrypt(struct skcipher_r
 
 static int cbc_encrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	CBC_WALK_START(req, AES_BLOCK_SIZE, -1);
 	CBC_ENC_BLOCK(__tresor_encrypt);
 	CBC_WALK_END();
@@ -224,6 +255,12 @@ static int cbc_encrypt(struct skcipher_r
 
 static int cbc_decrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	CBC_WALK_START(req, AES_BLOCK_SIZE, TRESOR_PARALLEL_BLOCKS);
 	CBC_DEC_BLOCK(1, __tresor_decrypt);
 	CBC_WALK_END();
--- a/crypto/ctr.c.orig	2024-02-10 01:44:11.010017114 -0800
+++ b/crypto/ctr.c	2024-02-10 01:50:19.837524423 -0800
@@ -105,6 +105,17 @@ static int crypto_ctr_crypt(struct skcip
 	unsigned int nbytes;
 	int err;
 
+#ifdef CONFIG_CRYPTO_TRESOR
+	const char *algo = crypto_tfm_alg_driver_name(crypto_skcipher_tfm(tfm));
+	if (strstr(algo, "tresor")) {
+		unsigned int ecx = cpuid_ecx(0x00000001);
+		if (ecx & (1 << 25))
+			; // aes-ni: true
+		else
+			return -ENOTSUPP;
+	}
+#endif
+
 	err = skcipher_walk_virt(&walk, req, false);
 
 	while (walk.nbytes >= bsize) {
