--- a/arch/x86/crypto/tresor_asm.S.orig	2024-02-12 19:49:47.878426518 -0800
+++ b/arch/x86/crypto/tresor_asm.S	2024-02-12 19:50:52.433644215 -0800
@@ -219,8 +219,7 @@ rc_tab:	.long	0x00000001, 0x00000002, 0x
 	pxor		%mm6,%mm6
 	pxor		%mm7,%mm7
 	/* leave */
-#ifdef CONFIG_X86_64
-#else
+#ifdef CONFIG_X86_32
 	popa
 	pop		ebp
 #endif
@@ -275,9 +274,9 @@ rc_tab:	.long	0x00000001, 0x00000002, 0x
 	pxor		rhelp,\sse
 .endm
 
+#ifdef CONFIG_X86_64
 /* copy four 32-bit values from debug registers into one 128-bit sse register */
 .macro	xts_dbg_to_sse dbg0 dbg1 dbg2 dbg3 sse
-#ifdef CONFIG_X86_64
 	movq		\dbg0,rax
 	rorq		$32,rax
 	movd		eax,\sse
@@ -293,17 +292,12 @@ rc_tab:	.long	0x00000001, 0x00000002, 0x
 	pxor		rhelp,\sse
 	movq		\dbg3,rax
 	rorq		$32,rax
-#else
-#endif
 	movd		eax,rhelp
 	pslldq		$12,rhelp
 	pxor		rhelp,\sse
-#ifdef CONFIG_X86_64
 	xorq		rax,rax
-#else
-	xorl		eax,eax
-#endif
 .endm
+#endif
 
 /* copy four 32-bit general purpose registers into one 128-bit sse register */
 .macro	gpr_to_sse gpr0 gpr1 gpr2 gpr3 sse
@@ -545,8 +539,10 @@ rc_tab:	.long	0x00000001, 0x00000002, 0x
 	.global		tresor_set_key
 	.global		tresor_encblk_128
 	.global		tresor_decblk_128
+#ifdef CONFIG_X86_64
 	.global		tresor_set_key_xts_tweak_128
 	.global		tresor_encblk_128_xts_tweak
+#endif
 	.extern		crypto_ft_tab
 	.extern		crypto_fl_tab
 	.extern		crypto_it_tab
@@ -589,6 +585,7 @@ tresor_decblk_128:
 	i_lround	rk1
 	epilog
 
+#ifdef CONFIG_X86_64
 /* void tresor_encblk_128_xts_tweak(u8 *out, const u8 *in) */
 tresor_encblk_128_xts_tweak:
 	prolog 
@@ -606,6 +603,7 @@ tresor_encblk_128_xts_tweak:
 	f_nround_	rk9a,rk9b,rk5
 	f_lround	rk10
 	epilog
+#endif
 
 /* void tresor_set_key(const u8 *in_key) */
 tresor_set_key:
@@ -634,9 +632,9 @@ tresor_set_key:
 #endif
 	ret
 
+#ifdef CONFIG_X86_64
 /* void tresor_set_key_xts_tweak_128(const u8 *in_key) */
 tresor_set_key_xts_tweak_128:
-#ifdef CONFIG_X86_64
 	pushq		rcx
 	pushq		rbx
 	pushq		rax
@@ -677,6 +675,5 @@ tresor_set_key_xts_tweak_128:
 	popq		rax
 	popq		rbx
 	popq		rcx
-#else
-#endif
 	ret
+#endif
--- a/arch/x86/crypto/tresor_glue.c.orig	2024-02-12 19:49:48.162423073 -0800
+++ b/arch/x86/crypto/tresor_glue.c	2024-02-12 19:50:29.185925756 -0800
@@ -121,6 +121,7 @@ void tresor_decrypt(struct crypto_tfm *t
 }
 
 
+#ifdef CONFIG_X86_64
 /*
  * Set AES key (the real function this time, not dummy as above)
  */
@@ -135,6 +136,7 @@ void tresor_setkey(const u8 *in_key)
 {
 	on_each_cpu(tresor_setkey_current_cpu, (void *)in_key, 1);
 }
+#endif
 
 static int tresor_skcipher_setkey(struct crypto_skcipher *tfm, const u8 *key, unsigned int keylen)
 {
--- a/init/main.c.orig	2024-02-12 21:06:54.608857768 -0800
+++ b/init/main.c	2024-02-12 21:08:17.651988423 -0800
@@ -1637,7 +1637,9 @@ static noinline void __init kernel_init_
 	alg_test("ecb(tresor)", "ecb(tresor)", 0, 0);
 	alg_test("cbc(tresor)", "cbc(tresor)", 0, 0);
 	alg_test("ctr(tresor)", "ctr(tresor)", 0, 0);
+#ifdef CONFIG_X86_64
 	alg_test("xts(tresor)", "xts(tresor)", 0, 0);
+#endif
 	tresor_lock_tests();
 #endif
 
