--- linux-5.4.90-ot.orig/arch/x86/crypto/tresor_key.c	2021-01-18 20:46:23.418601185 -0800
+++ linux-5.4.90-ot/arch/x86/crypto/tresor_key.c	2021-01-18 20:50:44.532529096 -0800
@@ -152,7 +152,7 @@ static void sha256(const char *message,
 	a = 0; b = 0; c = 0; d = 0;
 	e = 0; f = 0; g = 0; h = 0;
 	memset(chunk, 0, 64);
-	memset(w, 0, 64);
+	memset(w, 0, sizeof(w));
 	wbinvd();
 }
 
@@ -160,12 +160,13 @@ static void sha256(const char *message,
 /* Print to term_fd */
 static int printf_(const char *fmt, ...)
 {
+#define MAX_COL 80
 	ssize_t ret;
-	va_list args; int col = 80; char line[col];
+	va_list args; char line[MAX_COL];
 
 	va_start(args, fmt);
-	vsnprintf(line, col, fmt, args);
-	line[col-1] = 0;
+	vsnprintf(line, MAX_COL, fmt, args);
+	line[MAX_COL-1] = 0;
 	va_end(args);
 
 	tresor_prompt_begin();
