--- a/arch/x86/crypto/tresor_key.c	2020-07-03 19:13:36.804279446 -0700
+++ b/arch/x86/crypto/tresor_key.c	2020-07-03 19:19:31.833224210 -0700
@@ -152,7 +152,7 @@ static void sha256(const char *message,
 	a = 0; b = 0; c = 0; d = 0;
 	e = 0; f = 0; g = 0; h = 0;
 	memset(chunk, 0, 64);
-	memset(w, 0, 64);
+	memset(w, 0, sizeof(w));
 	wbinvd();
 }
 
@@ -160,11 +160,12 @@ static void sha256(const char *message,
 /* Print to term_fd */
 static int printf_(const char *fmt, ...)
 {
-	va_list args; int col = 80; char line[col];
+#define MAX_COL 80
+	va_list args; char line[MAX_COL];
 
 	va_start(args, fmt);
-	vsnprintf(line, col, fmt, args);
-	line[col-1] = 0;
+	vsnprintf(line, MAX_COL, fmt, args);
+	line[MAX_COL-1] = 0;
 	va_end(args);
 
 	return ksys_write(term_fd, line, strlen(line));
