--- a/arch/x86/crypto/tresor_asm.S.orig	2024-02-12 19:08:30.874826396 -0800
+++ b/arch/x86/crypto/tresor_asm.S	2024-02-12 19:14:25.814695797 -0800
@@ -245,8 +245,7 @@ gfbd9e:	.long	0x00000000, 0x0b0d090e, 0x
 	pxor		%mm7,%mm7
 #endif
 	/* leave */
-#ifdef CONFIG_X86_64
-#else
+#ifdef CONFIG_X86_32
 	popa
 	pop		ebp
 #endif
@@ -1087,8 +1086,7 @@ out:
 .endm
 #endif
 
-#ifdef CONFIG_X86_64
-#else
+#ifdef CONFIG_X86_32
 .macro key_schedule32_10
 	/* Generate round keys from rk1 to rk10 for AES-128 */
 	generate_rk_10	rk1,rk1,$0x1
@@ -1306,60 +1304,45 @@ out:
 	.global		tresor_decblk_192
 	.global		tresor_encblk_256
 	.global		tresor_decblk_256
-#endif
 	.global		tresor_set_key_xts_tweak_128
 	.global		tresor_encblk_128_xts_tweak
+#endif
 	.extern		crypto_ft_tab
 	.extern		crypto_fl_tab
 	.extern		crypto_it_tab
 	.extern		crypto_il_tab
 
+#ifdef CONFIG_X86_64
 /* void tresor_encblk_128(u8 *out, const u8 *in) */
 tresor_encblk_128:
-#ifdef CONFIG_X86_64
 	tresor_encblk64_10
-#else
-	tresor_encblk32_10
-#endif
 /* void tresor_decblk_128(u8 *out, const u8 *in) */
 tresor_decblk_128:
-#ifdef CONFIG_X86_64
 	tresor_decblk64_10
-#else
-	tresor_decblk32_10
-#endif
 /* void tresor_encblk_192(u8 *out, const u8 *in) */
 tresor_encblk_192:
-#ifdef CONFIG_X86_64
 	tresor_encblk64_12
-#else
-	ret
-#endif
 /* void tresor_decblk_192(u8 *out, const u8 *in) */
 tresor_decblk_192:
-#ifdef CONFIG_X86_64
 	tresor_decblk64_12
-#else
-	ret
-#endif
 /* void tresor_encblk_256(u8 *out, const u8 *in) */
 tresor_encblk_256:
-#ifdef CONFIG_X86_64
 	tresor_encblk64_14
-#else
-	ret
-#endif
 /* void tresor_decblk_256(u8 *out, const u8 *in) */
 tresor_decblk_256:
-#ifdef CONFIG_X86_64
 	tresor_decblk64_14
 #else
-	ret
+/* void tresor_encblk_128(u8 *out, const u8 *in) */
+tresor_encblk_128:
+	tresor_encblk32_10
+/* void tresor_decblk_128(u8 *out, const u8 *in) */
+tresor_decblk_128:
+	tresor_decblk32_10
 #endif
 
+#ifdef CONFIG_X86_64
 /* void tresor_encblk_128_xts_tweak(u8 *out, const u8 *in) */
 tresor_encblk_128_xts_tweak:
-#ifdef CONFIG_X86_64
 	prolog
 	xts_dbg_to_sse	rk0
 	pxor		rk0,rstate
@@ -1375,8 +1358,6 @@ tresor_encblk_128_xts_tweak:
 	f_nround	rk9
 	f_lround	rk10
 	epilog
-#else
-	ret
 #endif
 
 /* void tresor_set_key(const u8 *in_key) */
@@ -1457,9 +1438,9 @@ tresor_set_key:
 #endif
 	ret
 
+#ifdef CONFIG_X86_64
 /* void tresor_set_key_xts_tweak_128(const u8 *in_key) */
 tresor_set_key_xts_tweak_128:
-#ifdef CONFIG_X86_64
 /*
 	rdi[0] = tweak_key[0]:tk0 = in
 	rdi[1] = tweak_key[1]:tk1 = in
@@ -1489,6 +1470,5 @@ tresor_set_key_xts_tweak_128:
 	db2    tk1 out  tk0 out
 	db3    tk3 out  tk2 out
 */
-#else
-#endif
 	ret
+#endif
--- a/arch/x86/crypto/tresor_glue.c.orig	2024-02-12 19:08:31.046824379 -0800
+++ b/arch/x86/crypto/tresor_glue.c	2024-02-12 19:09:12.786335405 -0800
@@ -187,6 +187,7 @@ void __tresor_decrypt(void *_ctx, u8 *ds
 }
 
 
+#ifdef CONFIG_X86_64
 /*
  * Set AES key (the real function this time, not dummy as above)
  */
@@ -201,6 +202,7 @@ void tresor_setkey(const u8 *in_key)
 {
 	on_each_cpu(tresor_setkey_current_cpu, (void *)in_key, 1);
 }
+#endif
 
 static int tresor_skcipher_setkey(struct crypto_skcipher *tfm, const u8 *key, unsigned int keylen)
 {
--- a/init/main.c.orig	2024-02-12 21:57:33.478023067 -0800
+++ b/init/main.c	2024-02-12 22:04:25.870129449 -0800
@@ -1637,7 +1637,9 @@ static noinline void __init kernel_init_
 	alg_test("ecb(tresor)", "ecb(tresor)", 0, 0);
 	alg_test("cbc(tresor)", "cbc(tresor)", 0, 0);
 	alg_test("ctr(tresor)", "ctr(tresor)", 0, 0);
+#ifdef CONFIG_X86_64
 	alg_test("xts(tresor)", "xts(tresor)", 0, 0);
+#endif
 	tresor_lock_tests();
 #endif
 
