--- a/crypto/testmgr.c.orig	2019-06-19 20:31:57.844074650 -0700
+++ b/crypto/testmgr.c	2019-06-19 20:35:38.822696237 -0700
@@ -1549,12 +1549,24 @@ static int test_skcipher_vec_cfg(const c
 	struct kvec input;
 	int err;
 
+#ifdef CONFIG_CRYPTO_TRESOR
+	if (strstr(driver, "tresor")) {
+		if (tresor_tests_locked) {
+			return 0;
+		}
+	}
+#endif
+
 	/* Set the key */
 	if (vec->wk)
 		crypto_skcipher_set_flags(tfm, CRYPTO_TFM_REQ_FORBID_WEAK_KEYS);
 	else
 		crypto_skcipher_clear_flags(tfm,
 					    CRYPTO_TFM_REQ_FORBID_WEAK_KEYS);
+#ifdef CONFIG_CRYPTO_TRESOR
+	if (strstr(driver, "tresor"))
+		tresor_setkey(vec->key);
+#endif
 	err = crypto_skcipher_setkey(tfm, vec->key, vec->klen);
 	if (err) {
 		if (vec->fail) /* expectedly failed to set key? */
@@ -1565,13 +1577,6 @@ static int test_skcipher_vec_cfg(const c
 	}
 	if (vec->fail) {
 		pr_err("alg: skcipher: %s setkey unexpectedly succeeded on test vector %u\n",
-#ifdef CONFIG_CRYPTO_TRESOR
-	if (tresor_tests_locked) {
-		ret = 0;
-		goto out;
-	}
-#endif
-
 		       driver, vec_num);
 		return -EINVAL;
 	}
@@ -1594,11 +1599,6 @@ static int test_skcipher_vec_cfg(const c
 		}
 		iv = NULL;
 	}
-#ifdef CONFIG_CRYPTO_TRESOR
-			if (strstr(algo, "tresor"))
-				tresor_setkey(template[i].key);
-#endif
-
 
 	/* Build the src/dst scatterlists */
 	input.iov_base = enc ? (void *)vec->ptext : (void *)vec->ctext;
@@ -1663,11 +1663,6 @@ static int test_skcipher_vec_cfg(const c
 		pr_err("alg: skcipher: %s %s corrupted dst sgl on test vector %u, cfg=\"%s\"\n",
 		       driver, op, vec_num, cfg->name);
 		return -EINVAL;
-#ifdef CONFIG_CRYPTO_TRESOR
-			if (strstr(algo, "tresor"))
-				tresor_setkey(template[i].key);
-#endif
-
 	}
 
 	/* Check for the correct output (ciphertext or plaintext) */
