--- a/arch/x86/crypto/tresor_glue.c.orig	2024-02-09 22:16:26.308083862 -0800
+++ b/arch/x86/crypto/tresor_glue.c	2024-02-09 22:30:17.688463137 -0800
@@ -32,6 +32,7 @@
  * Copyright © 2012-2013 Jussi Kivilinna <jussi.kivilinna@iki.fi>
  */
 
+#include <asm/processor.h>
 #include <crypto/algapi.h>
 #include <crypto/b128ops.h>
 #include <crypto/internal/simd.h>
@@ -191,6 +192,12 @@ static int tresor_skcipher_setkey(struct
 
 static int ecb_encrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	ECB_WALK_START(req, AES_BLOCK_SIZE, -1);
 	ECB_BLOCK(1, tresor_encrypt);
 	ECB_WALK_END();
@@ -198,6 +205,12 @@ static int ecb_encrypt(struct skcipher_r
 
 static int ecb_decrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	ECB_WALK_START(req, AES_BLOCK_SIZE, -1);
 	ECB_BLOCK(1, tresor_decrypt);
 	ECB_WALK_END();
@@ -206,6 +219,12 @@ static int ecb_decrypt(struct skcipher_r
 
 static int cbc_encrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	CBC_WALK_START(req, AES_BLOCK_SIZE, -1);
 	CBC_ENC_BLOCK(tresor_encrypt);
 	CBC_WALK_END();
@@ -213,6 +232,12 @@ static int cbc_encrypt(struct skcipher_r
 
 static int cbc_decrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	CBC_WALK_START(req, AES_BLOCK_SIZE, -1);
 	CBC_DEC_BLOCK(1, tresor_decrypt);
 	CBC_WALK_END();
--- a/crypto/ctr.c.orig	2021-10-31 13:53:10.000000000 -0700
+++ b/crypto/ctr.c	2024-02-09 22:43:34.716062559 -0800
@@ -5,6 +5,7 @@
  * (C) Copyright IBM Corp. 2007 - Joy Latten <latten@us.ibm.com>
  */
 
+#include <asm/processor.h>
 #include <crypto/algapi.h>
 #include <crypto/ctr.h>
 #include <crypto/internal/cipher.h>
@@ -105,6 +106,17 @@ static int crypto_ctr_crypt(struct skcip
 	unsigned int nbytes;
 	int err;
 
+#ifdef CONFIG_CRYPTO_TRESOR
+	const char *algo = crypto_tfm_alg_driver_name(crypto_skcipher_tfm(tfm));
+	if (strstr(algo, "tresor")) {
+		unsigned int ecx = cpuid_ecx(0x00000001);
+		if (ecx & (1 << 25))
+			; // aes-ni: true
+		else
+			return -ENOTSUPP;
+	}
+#endif
+
 	err = skcipher_walk_virt(&walk, req, false);
 
 	while (walk.nbytes >= bsize) {
