--- a/arch/x86/crypto/tresor_glue.c.orig	2024-02-09 12:55:35.883677504 -0800
+++ b/arch/x86/crypto/tresor_glue.c	2024-02-09 12:58:03.217997603 -0800
@@ -32,6 +32,7 @@
  * Copyright © 2012-2013 Jussi Kivilinna <jussi.kivilinna@iki.fi>
  */
 
+#include <asm/processor.h>
 #include <crypto/algapi.h>
 #include <crypto/internal/simd.h>
 #include <crypto/internal/skcipher.h>
@@ -203,6 +204,12 @@ void __tresor_decrypt(const void *_ctx,
 
 static int ecb_encrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	ECB_WALK_START(req, AES_BLOCK_SIZE, TRESOR_PARALLEL_BLOCKS);
 	ECB_BLOCK(1, __tresor_encrypt);
 	ECB_WALK_END();
@@ -210,6 +217,12 @@ static int ecb_encrypt(struct skcipher_r
 
 static int ecb_decrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	ECB_WALK_START(req, AES_BLOCK_SIZE, TRESOR_PARALLEL_BLOCKS);
 	ECB_BLOCK(1, __tresor_decrypt);
 	ECB_WALK_END();
@@ -217,6 +230,12 @@ static int ecb_decrypt(struct skcipher_r
 
 static int cbc_encrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	CBC_WALK_START(req, AES_BLOCK_SIZE, -1);
 	CBC_ENC_BLOCK(__tresor_encrypt);
 	CBC_WALK_END();
@@ -224,6 +243,12 @@ static int cbc_encrypt(struct skcipher_r
 
 static int cbc_decrypt(struct skcipher_request *req)
 {
+	unsigned int ecx = cpuid_ecx(0x00000001);
+	if (ecx & (1 << 25))
+		; // aes-ni: true
+	else
+		return -ENOTSUPP;
+
 	CBC_WALK_START(req, AES_BLOCK_SIZE, TRESOR_PARALLEL_BLOCKS);
 	CBC_DEC_BLOCK(1, __tresor_decrypt);
 	CBC_WALK_END();
