--- a/mm/uksm.c.orig	2019-07-14 22:16:59.515130930 -0700
+++ b/mm/uksm.c	2019-07-16 12:27:29.195610690 -0700
@@ -1503,7 +1503,7 @@ static int write_protect_page(struct vm_
 
 	BUG_ON(PageTransCompound(page));
 
-       mmu_notifier_range_init(&range, mm, pvmw.address, pvmw.address + PAGE_SIZE);
+       mmu_notifier_range_init(&range, MMU_NOTIFY_UNMAP, 0, vma, mm, pvmw.address, pvmw.address + PAGE_SIZE);
 	mmu_notifier_invalidate_range_start(&range);
 
 	if (!page_vma_mapped_walk(&pvmw))
@@ -1608,7 +1608,7 @@ static int replace_page(struct vm_area_s
 	if (!pmd_present(*pmd))
 		goto out;
 
-       mmu_notifier_range_init(&range, mm, addr, addr + PAGE_SIZE);
+       mmu_notifier_range_init(&range, MMU_NOTIFY_UNMAP, 0, vma, mm, addr, addr + PAGE_SIZE);
 	mmu_notifier_invalidate_range_start(&range);
 
 	ptep = pte_offset_map_lock(mm, pmd, addr, &ptl);
