diff '--color=auto' -urp ollama-0.5.7.orig/Makefile ollama-0.5.7/Makefile
--- ollama-0.5.7.orig/Makefile	2025-01-20 15:49:02.957711115 -0800
+++ ollama-0.5.7/Makefile	2025-01-20 15:59:43.996772960 -0800
@@ -19,8 +19,7 @@ else
 	CUDA_12_COMPILER =
 endif
 
-ifeq ($(CUSTOM_CPU_FLAGS),)
-ifeq ($(ARCH),amd64)
+ifeq ($(OLLAMA_SKIP_CPU_RUNNER),)
 	RUNNER_TARGETS=cpu
 endif
 # Without CUSTOM_CPU_FLAGS we default to build both v11 and v12 if present
@@ -32,13 +31,6 @@ ifneq ($(CUDA_12_COMPILER),)
 	RUNNER_TARGETS += cuda_v12
 endif
 endif
-else # CUSTOM_CPU_FLAGS is set, we'll build only the latest cuda version detected
-ifneq ($(CUDA_12_COMPILER),)
-	RUNNER_TARGETS += cuda_v12
-else ifneq ($(CUDA_11_COMPILER),)
-	RUNNER_TARGETS += cuda_v11
-endif
-endif
 
 ifeq ($(OLLAMA_SKIP_ROCM_GENERATE),)
 ifneq ($(HIP_COMPILER),)
diff '--color=auto' -urp ollama-0.5.7.orig/make/Makefile.cpu ollama-0.5.7/make/Makefile.cpu
--- ollama-0.5.7.orig/make/Makefile.cpu	2025-01-16 01:14:04.000000000 -0800
+++ ollama-0.5.7/make/Makefile.cpu	2025-01-20 15:59:33.180855671 -0800
@@ -3,10 +3,15 @@
 include make/common-defs.make
 
 CPU_GOFLAGS="-ldflags=-w -s \"-X=github.com/ollama/ollama/version.Version=$(VERSION)\"  $(TARGET_LDFLAGS)"
-ifeq ($(ARCH),amd64)
-ifeq ($(origin CUSTOM_CPU_FLAGS),undefined)
-	RUNNERS = cpu_avx cpu_avx2
+RUNNERS =
+ifeq ($(OLLAMA_SKIP_CPU_RUNNER_GENERIC),)
+	RUNNERS += cpu
 endif
+ifeq ($(OLLAMA_SKIP_CPU_RUNNER_AVX),)
+	RUNNERS += cpu_avx
+endif
+ifeq ($(OLLAMA_SKIP_CPU_RUNNER_AVX2),)
+	RUNNERS += cpu_avx2
 endif
 
 DIST_RUNNERS = $(addprefix $(RUNNERS_DIST_DIR)/,$(addsuffix /ollama_llama_server$(EXE_EXT),$(RUNNERS)))
@@ -16,6 +21,11 @@ cpu: $(BUILD_RUNNERS)
 
 dist: $(DIST_RUNNERS)
 
+$(RUNNERS_BUILD_DIR)/cpu/ollama_llama_server$(EXE_EXT): TARGET_CPU_FLAGS=$(CUSTOM_CPU_FLAGS)
+$(RUNNERS_BUILD_DIR)/cpu/ollama_llama_server$(EXE_EXT): ./llama/*.go ./llama/runner/*.go $(COMMON_SRCS) $(COMMON_HDRS)
+	@-mkdir -p $(dir $@)
+	GOARCH=$(ARCH) go build -buildmode=pie $(CPU_GOFLAGS) -trimpath -tags $(subst $(space),$(comma),$(TARGET_CPU_FLAGS)) -o $@ ./cmd/runner
+
 $(RUNNERS_BUILD_DIR)/cpu_avx/ollama_llama_server$(EXE_EXT): TARGET_CPU_FLAGS="avx"
 $(RUNNERS_BUILD_DIR)/cpu_avx/ollama_llama_server$(EXE_EXT): ./llama/*.go ./llama/runner/*.go $(COMMON_SRCS) $(COMMON_HDRS)
 	@-mkdir -p $(dir $@)
