--- a/ml/backend/ggml/ggml/src/CMakeLists.txt.orig	2025-02-13 22:38:23.000000000 -0800
+++ b/ml/backend/ggml/ggml/src/CMakeLists.txt	2025-02-18 23:47:13.369095057 -0800
@@ -283,20 +283,13 @@ endfunction()
 
 ggml_add_backend(CPU)
 
+set(GGML_CPU_ALL_VARIANTS @GGML_CPU_ALL_VARIANTS@)
 if (GGML_CPU_ALL_VARIANTS)
     if (NOT GGML_BACKEND_DL)
         message(FATAL_ERROR "GGML_CPU_ALL_VARIANTS requires GGML_BACKEND_DL")
     endif()
     add_custom_target(ggml-cpu)
-    ggml_add_cpu_backend_variant(sandybridge    AVX)
-    ggml_add_cpu_backend_variant(haswell        AVX F16C AVX2 FMA)
-    ggml_add_cpu_backend_variant(skylakex       AVX F16C AVX2 FMA AVX512)
-    ggml_add_cpu_backend_variant(icelake        AVX F16C AVX2 FMA AVX512 AVX512_VBMI AVX512_VNNI)
-    ggml_add_cpu_backend_variant(alderlake      AVX F16C AVX2 FMA AVX_VNNI)
-    if (NOT MSVC)
-        # MSVC doesn't support AMX
-        ggml_add_cpu_backend_variant(sapphirerapids AVX F16C AVX2 FMA AVX512 AVX512_VBMI AVX512_VNNI AVX512_BF16 AMX_TILE AMX_INT8)
-    endif()
+    ggml_add_cpu_backend_variant(custom         @CPU_FEATURES@)
 else ()
     ggml_add_cpu_backend_variant_impl("")
 endif()
