--- a/make/cuda.make.orig	2024-12-23 18:56:47.055874035 -0800
+++ b/make/cuda.make	2024-12-23 19:23:09.148695161 -0800
@@ -25,8 +25,8 @@ else ifeq ($(OS),linux)
 	GPU_VECTOR_FLAGS=$(if $(filter avx512,$(GPU_RUNNER_CPU_FLAGS)),avx512f avx512dq avx512bw) $(filter-out avx512,$(GPU_RUNNER_CPU_FLAGS))
 	GPU_COMPILER_EXTRA_FLAGS = -fPIC -Wno-unused-function -std=c++17
 	GPU_LIBS = $(sort $(wildcard $(addsuffix *.$(SHARED_EXT).*,$(addprefix $(GPU_LIB_DIR)/$(SHARED_PREFIX),$(GPU_RUNNER_LIBS_SHORT)))))
-	GPU_COMPILER_CFLAGS = $(CFLAGS) -Xcompiler -fPIC -D_GNU_SOURCE
-	GPU_COMPILER_CXXFLAGS = $(CXXFLAGS) -Xcompiler -fPIC -D_GNU_SOURCE
+	GPU_COMPILER_CFLAGS = $(CFLAGS) -Xcompiler -fPIC -D_GNU_SOURCE $(NVCC_FLAGS)
+	GPU_COMPILER_CXXFLAGS = $(CXXFLAGS) -Xcompiler -fPIC -D_GNU_SOURCE $(NVCC_FLAGS)
 endif
 GPU_DIST_LIB_DEPS= $(sort $(addprefix $(DIST_GPU_RUNNER_DEPS_DIR)/,$(notdir $(GPU_LIBS))))
 
@@ -35,6 +35,7 @@ GPU_RUNNER_ARCH_FLAGS := $(foreach arch,
 GPU_COMPILER_CUFLAGS = \
 	$(GPU_COMPILER_EXTRA_FLAGS) \
 	-Xcompiler "$(addprefix $(CPU_FLAG_PREFIX),$(GPU_VECTOR_FLAGS))" \
+	$(NVCC_FLAGS) \
 	-t2 \
 	-DGGML_CUDA_DMMV_X=32 \
 	-DGGML_CUDA_MMV_Y=1 \
