--- a/Makefile.orig	2024-12-17 09:49:37.000000000 -0800
+++ b/Makefile	2024-12-23 17:29:06.959581233 -0800
@@ -7,6 +7,18 @@ include make/cuda-v11-defs.make
 include make/cuda-v12-defs.make
 include make/rocm-defs.make
 
+ifeq ($(USE_CUDA),1)
+ifeq ($(CUDA_SLOT),11)
+	CUDA_12_COMPILER =
+endif
+ifeq ($(CUDA_SLOT),12)
+	CUDA_11_COMPILER =
+endif
+else
+	CUDA_11_COMPILER =
+	CUDA_12_COMPILER =
+endif
+
 ifeq ($(CUSTOM_CPU_FLAGS),)
 ifeq ($(ARCH),amd64)
 	RUNNER_TARGETS=cpu
