--- opencl.eselect.orig	2015-06-27 02:45:20.016237601 -0700
+++ opencl.eselect	2015-07-03 22:25:02.303004901 -0700
@@ -133,6 +133,21 @@
 	popd &> /dev/null
 }
 
+setup_icd_symlinks() {
+	local target=$1
+	local arches=(32 64)
+	local vendors=(mesa amd amdapp)
+	
+	for arch in "${arches[@]}"; do
+		for vendor in "${vendors[@]}"; do
+			if [[ "${target}" == "${vendor}" ]]; then
+				rm "/etc/OpenCL/vendors/ocl${arch}.icd"
+				ln -s "/etc/OpenCL/profiles/${vendor}/${vendor}ocl${arch}.icd" "/etc/OpenCL/vendors/ocl${arch}.icd"
+			fi
+		done
+	done
+}
+
 set_new_implementation() {
 	local cl_implem=$1
 	local avail_implems=$(get_implementations)
@@ -207,6 +222,9 @@
 			"${PREFIX}/${libdir}/OpenCL/vendors/${cl_implem}/include/CL/" \
 			"${PREFIX}/${libdir}/OpenCL/global/include/CL/"
 
+		setup_icd_symlinks \
+			${cl_implem}
+
 		# Setup the $LDPATH
 		if [[ "x${REMOVE_ONLY}" == "xfalse" ]]; then
 			# we need this relative to ROOT
