See https://github.com/nodejs/node-gyp/issues/2673#issuecomment-1239619438
--- texturelab-0.3.3/node_modules/node-gyp/gyp/pylib/gyp/input.py.orig	2023-01-08 22:15:48.309000000 -0800
+++ texturelab-0.3.3/node_modules/node-gyp/gyp/pylib/gyp/input.py	2023-01-08 22:27:07.980218740 -0800
@@ -1183,7 +1183,7 @@ def EvalSingleCondition(cond_expr, true_
         else:
             ast_code = compile(cond_expr_expanded, "<string>", "eval")
             cached_conditions_asts[cond_expr_expanded] = ast_code
-        env = {"__builtins__": {}, "v": StrictVersion}
+        env = {"__builtins__": {"openssl_fips": ""}, "v": StrictVersion}
         if eval(ast_code, env, variables):
             return true_dict
         return false_dict
