diff -urp MIVisionX-rocm-5.6.0.orig/rocAL/rocAL_pybind/CMakeLists.txt MIVisionX-rocm-5.6.0/rocAL/rocAL_pybind/CMakeLists.txt
--- MIVisionX-rocm-5.6.0.orig/rocAL/rocAL_pybind/CMakeLists.txt	2023-08-14 18:34:29.270204910 -0700
+++ MIVisionX-rocm-5.6.0/rocAL/rocAL_pybind/CMakeLists.txt	2023-08-14 18:35:56.978202092 -0700
@@ -101,8 +101,10 @@ if(ROCAL_PYTHON)
             set(CMAKE_INSTALL_PREFIX_PYTHON ${Python3_STDARCH} CACHE PATH "rocAL default installation path" FORCE)
         endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
         set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-        add_subdirectory(third_party_lib/pybind11)
-        include_directories(third_party_lib/pybind11/include/)
+        #add_subdirectory(third_party_lib/pybind11)
+        cmake_policy(SET CMP0057 NEW)
+        find_package(pybind11)
+        include_directories(${PYBIND11_INCLUDES})
 
 
         # AMD RPP
diff -urp MIVisionX-rocm-5.6.0.orig/rocAL/rocAL_pybind/third_party_lib/CMakeLists.txt MIVisionX-rocm-5.6.0/rocAL/rocAL_pybind/third_party_lib/CMakeLists.txt
--- MIVisionX-rocm-5.6.0.orig/rocAL/rocAL_pybind/third_party_lib/CMakeLists.txt	2023-08-14 18:34:29.270204910 -0700
+++ MIVisionX-rocm-5.6.0/rocAL/rocAL_pybind/third_party_lib/CMakeLists.txt	2023-08-14 18:36:18.818201391 -0700
@@ -21,4 +21,4 @@
 cmake_minimum_required(VERSION 3.0)
 project(third_party_lib)
 
-add_subdirectory(pybind11)
+#add_subdirectory(pybind11)
