diff -urp libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-canvas.c libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-canvas.c
--- libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-canvas.c	2021-12-07 01:03:53.000000000 -0800
+++ libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-canvas.c	2023-06-19 20:22:21.989562733 -0700
@@ -18,6 +18,8 @@
 #include "caca-font.h"
 #include "common.h"
 
+extern void *_caca_alloc2d(size_t width, size_t height, size_t elem_size);
+
 VALUE cCanvas;
 
 #define simple_func(x)                                  \
@@ -594,7 +596,7 @@ static VALUE render_canvas(VALUE self, V
     f = DATA_PTR(font);
     caca_render_canvas(_SELF, f, buf, NUM2UINT(width), NUM2UINT(height), NUM2UINT(pitch));
 
-    b = rb_str_new(buf, width*height*4);
+    b = rb_str_new((const char*)(void*)buf, width*height*4);
     free(buf);
     return b;
 }
@@ -742,7 +744,7 @@ void Init_caca_canvas(VALUE mCaca)
     rb_define_method(cCanvas, "put_str", put_str, 3);
     rb_define_method(cCanvas, "printf", cprintf, -1);
 
-    rb_define_method(cCanvas, "get_attr", get_attr, 3);
+    rb_define_method(cCanvas, "get_attr", get_attr, 2);
     rb_define_method(cCanvas, "attr=", set_attr, 1);
     rb_define_method(cCanvas, "set_attr", set_attr2, 1);
     rb_define_method(cCanvas, "put_attr", put_attr, 3);
@@ -773,7 +775,7 @@ void Init_caca_canvas(VALUE mCaca)
     rb_define_method(cCanvas, "draw_triangle", draw_triangle, 7);
     rb_define_method(cCanvas, "draw_thin_triangle", draw_thin_triangle, 6);
     rb_define_method(cCanvas, "fill_triangle", fill_triangle, 7);
-    rb_define_method(cCanvas, "fill_triangle_textured", fill_triangle_textured, 4);
+    rb_define_method(cCanvas, "fill_triangle_textured", fill_triangle_textured, 3);
     rb_define_method(cCanvas, "dither_bitmap", dither_bitmap, 6);
 
     rb_define_method(cCanvas, "frame_count", get_frame_count, 0);
diff -urp libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-display.c libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-display.c
--- libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-display.c	2021-12-07 01:03:53.000000000 -0800
+++ libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-display.c	2023-06-19 20:22:21.989562733 -0700
@@ -226,7 +226,7 @@ static VALUE get_event(VALUE self, VALUE
     return e;
 }
 
-static VALUE driver_list(void)
+static VALUE driver_list(VALUE self)
 {
     VALUE ary;
     char const* const* list;
diff -urp libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-font.c libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-font.c
--- libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-font.c	2021-12-07 01:03:53.000000000 -0800
+++ libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-font.c	2023-06-19 20:22:21.989562733 -0700
@@ -41,7 +41,7 @@ static VALUE font_initialize(VALUE self,
     return self;
 }
 
-static VALUE font_list(void)
+static VALUE font_list(VALUE self)
 {
     VALUE ary;
     char const* const* list;
diff -urp libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/common.h libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/common.h
--- libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/common.h	2021-12-07 01:03:53.000000000 -0800
+++ libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/common.h	2023-06-19 20:22:21.989562733 -0700
@@ -16,7 +16,7 @@
 #define _SELF  (DATA_PTR(self))
 
 #define get_singleton_double_list(x)                      \
-static VALUE x##_list(void)                     \
+static VALUE x##_list(VALUE self)                     \
 {                                               \
     VALUE ary, ary2;                            \
     char const* const* list;                    \
diff -urp libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/t/tc_display.rb libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/t/tc_display.rb
--- libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/t/tc_display.rb	2021-12-07 01:03:53.000000000 -0800
+++ libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/t/tc_display.rb	2023-06-19 20:22:44.933563302 -0700
@@ -28,7 +28,8 @@ class TC_Canvas < MiniTest::Test
         d.title = "Test !"
     end
     def test_set_cursor
-        d = Caca::Display.new()
+        c = Caca::Canvas.new(3, 3)
+        d = Caca::Display.new(c)
         d.cursor = 1
     end
 end
