diff -urp libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/python/examples/img2txt.py libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/python/examples/img2txt.py
--- libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/python/examples/img2txt.py	2021-12-07 01:03:53.000000000 -0800
+++ libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/python/examples/img2txt.py	2023-06-21 21:25:49.989944514 -0700
@@ -17,6 +17,9 @@
 # http://www.wtfpl.net/ for more details.
 #
 
+from __future__ import division
+from builtins import str
+from past.utils import old_div
 import getopt
 import os
 import sys
@@ -127,7 +130,7 @@ def main():
     #parse command line
     try:
         opts, args = getopt.getopt(sys.argv[1:], shortopts, longopts)
-    except getopt.GetoptError, err:
+    except getopt.GetoptError as err:
         sys.stderr.write("%s\n" % str(err))
         sys.exit(127)
 
@@ -166,22 +169,22 @@ def main():
         try:
             img = Image.open(args[0])
             img.load()
-        except IOError, err:
+        except IOError as err:
             sys.stderr.write("%s\n" % err)
             sys.exit(127)
 
     if not width and not height:
         width = 60
-        height = width * img.size[1] * font_width / img.size[0] / font_height
+        height = old_div(old_div(width * img.size[1] * font_width, img.size[0]), font_height)
     elif width and not height:
-        height = width * img.size[1] * font_width / img.size[0] / font_height
+        height = old_div(old_div(width * img.size[1] * font_width, img.size[0]), font_height)
     elif not width and height:
-        width = height * img.size[0] * font_height / img.size[1] / font_width
+        width = old_div(old_div(height * img.size[0] * font_height, img.size[1]), font_width)
 
     #init canvas
     try:
         cv = Canvas(width, height)
-    except CanvasError, err:
+    except CanvasError as err:
         sys.stderr.write("%s\n" % err)
         sys.exit(127)
 
@@ -199,7 +202,7 @@ def main():
 
         dit = Dither(BPP, img.size[0], img.size[1], DEPTH * img.size[0],
                         RMASK, GMASK, BMASK, AMASK)
-    except DitherError, err:
+    except DitherError as err:
         sys.stderr.write("%s\n" % err)
         sys.exit(127)
 
@@ -233,7 +236,7 @@ if __name__ == "__main__":
     #Import PIL package
     try:
         from PIL import Image
-    except ImportError, err:
+    except ImportError as err:
         sys.stderr.write("You need to install PIL module !\n")
         sys.exit(2)
 
