diff -urp libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-canvas.c libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-canvas.c
--- libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752.orig/ruby/caca-canvas.c	2023-06-19 23:17:13.613823038 -0700
+++ libcaca-f42aa68fc798db63b7b2a789ae8cf5b90b57b752/ruby/caca-canvas.c	2023-06-19 23:18:40.173825186 -0700
@@ -714,11 +714,8 @@ simple_func(clear_dirty_rect_list)
 
 /****/
 
-void Init_caca_canvas(VALUE mCaca)
+void trash_Init_caca_canvas(VALUE mCaca)
 {
-    cCanvas = rb_define_class_under(mCaca, "Canvas", rb_cObject);
-    rb_define_alloc_func(cCanvas, canvas_alloc);
-
     rb_define_method(cCanvas, "initialize", canvas_initialize, 2);
     rb_define_method(cCanvas, "width", get_canvas_width, 0);
     rb_define_method(cCanvas, "width=", set_canvas_width, 1);
@@ -808,3 +805,11 @@ void Init_caca_canvas(VALUE mCaca)
 
 }
 
+void Init_caca_canvas(VALUE mCaca)
+{
+    cCanvas = rb_define_class_under(mCaca, "Canvas", rb_cObject);
+    rb_define_alloc_func(cCanvas, canvas_alloc);
+
+
+}
+
