--- a/Tools/TestWebKitAPI/Tests/WTF/StringImpl.cpp.orig	2025-01-24 17:54:30.095823522 -0800
+++ b/Tools/TestWebKitAPI/Tests/WTF/StringImpl.cpp	2025-01-24 17:53:57.952042114 -0800
@@ -751,6 +751,7 @@ TEST(WTF, StaticPrivateSymbolImpl)
     ASSERT_TRUE(symbol.isPrivate());
 }
 
+#if 0
 TEST(WTF, ExternalStringImplCreate8bit)
 {
     constexpr LChar buffer[] = "hello";
@@ -866,5 +867,6 @@ TEST(WTF, ExternalStringToSymbol)
 
     ASSERT_TRUE(freeFunctionCalled);
 }
+#endif
 
 } // namespace TestWebKitAPI
