--- a/dotnet-cli-2.1.505/test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/FileInfoAssertions.cs.orig	2019-04-02 17:04:15.478943089 -0700
+++ b/dotnet-cli-2.1.505/test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/FileInfoAssertions.cs	2019-04-02 19:41:54.477607763 -0700
@@ -46,7 +46,7 @@ namespace Microsoft.DotNet.Tools.Test.Ut
             var lastWriteTimeUtc = _fileInfo.LastWriteTimeUtc;
 
             Execute.Assertion
-                .ForCondition(lastWriteTimeUtc != null)
+                .ForCondition(lastWriteTimeUtc != DateTime.MinValue)
                 .BecauseOf(because, reasonArgs) 
                 .FailWith($"Expected File {_fileInfo.FullName} to have a LastWriteTimeUTC, but it is null.");
             return new AndWhichConstraint<FileInfoAssertions, DateTimeOffset>(this, lastWriteTimeUtc);
