diff -urp SkyFire.548-5.4.7.orig/src/server/game/DataStores/DB2Structure.h SkyFire.548-5.4.7/src/server/game/DataStores/DB2Structure.h
--- SkyFire.548-5.4.7.orig/src/server/game/DataStores/DB2Structure.h	2014-05-15 18:49:10.000000000 -0700
+++ SkyFire.548-5.4.7/src/server/game/DataStores/DB2Structure.h	2017-02-23 16:16:29.257253787 -0800
@@ -22,6 +22,7 @@
 
 #include "Common.h"
 #include "ItemPrototype.h"
+#include "DBCStructure.h"
 
 // GCC has alternative #pragma pack(N) syntax and old gcc version does not support pack(push, N), also any gcc version does not support it at some platform
 #if defined(__GNUC__)
@@ -204,4 +205,4 @@ struct SpellReagentsEntry
 #pragma pack(pop)
 #endif
 
-#endif
\ No newline at end of file
+#endif
