diff -urp SkyFire.548-5.4.7.orig/src/server/game/Server/WorldSession.h SkyFire.548-5.4.7/src/server/game/Server/WorldSession.h
--- SkyFire.548-5.4.7.orig/src/server/game/Server/WorldSession.h	2014-05-15 18:49:10.000000000 -0700
+++ SkyFire.548-5.4.7/src/server/game/Server/WorldSession.h	2017-02-23 16:03:27.476777512 -0800
@@ -33,6 +33,7 @@
 #include "Cryptography/BigNumber.h"
 #include "Opcodes.h"
 #include "AccountMgr.h"
+#include "Object.h"
 
 class Creature;
 class GameObject;
