--- a/ext/google/protobuf_c/ruby-upb.h.orig	2022-10-26 10:50:48.000000000 -0700
+++ b/ext/google/protobuf_c/ruby-upb.h	2022-12-22 23:16:17.522862205 -0800
@@ -769,7 +769,7 @@ upb_DecodeStatus upb_Decode(const char*
 
 #include <setjmp.h>
 
-#include "third_party/utf8_range/utf8_range.h"
+#include <utf8_range.h>
 
 /** upb/msg_internal.h ************************************************************/
 /*
