--- a/ext/google/protobuf_c/extconf.rb.orig	2022-10-26 10:50:48.000000000 -0700
+++ b/ext/google/protobuf_c/extconf.rb	2022-12-22 23:26:21.847081424 -0800
@@ -17,12 +17,12 @@ if RUBY_PLATFORM =~ /linux/
   # Instruct the linker to point memcpy calls at our __wrap_memcpy wrapper.
   $LDFLAGS += " -Wl,-wrap,memcpy"
 end
+$LDFLAGS += " -lutf8_range"
 
 $VPATH << "$(srcdir)/third_party/utf8_range"
 $INCFLAGS << "$(srcdir)/third_party/utf8_range"
 
 $srcs = ["protobuf.c", "convert.c", "defs.c", "message.c",
-         "repeated_field.c", "map.c", "ruby-upb.c", "wrap_memcpy.c",
-         "naive.c", "range2-neon.c", "range2-sse.c"]
+         "repeated_field.c", "map.c", "ruby-upb.c", "wrap_memcpy.c"]
 
 create_makefile(ext_name)
