diff -urp RadeonProRenderBlenderAddon-3.5.0.orig/src/rprblender/__init__.py RadeonProRenderBlenderAddon-3.5.0/src/rprblender/__init__.py
--- RadeonProRenderBlenderAddon-3.5.0.orig/src/rprblender/__init__.py	2022-11-04 10:03:40.000000000 -0700
+++ RadeonProRenderBlenderAddon-3.5.0/src/rprblender/__init__.py	2022-12-24 19:02:58.806867674 -0800
@@ -20,7 +20,7 @@ import bpy
 bl_info = {
     "name": "Radeon ProRender",
     "author": "AMD",
-    "version": (3, 5, 1),
+    "version": (3, 5, 0), # oteodoro: bumped version so the generated zip file name matches
     "blender": (2, 80, 0),
     "location": "Info header, render engine menu",
     "description": "Radeon ProRender rendering plugin for Blender 2.8x",
