diff -urp RadeonProRenderBlenderAddon-3.5.2.orig/src/rprblender/__init__.py RadeonProRenderBlenderAddon-3.5.2/src/rprblender/__init__.py
--- RadeonProRenderBlenderAddon-3.5.2.orig/src/rprblender/__init__.py	2022-11-07 10:03:41.000000000 -0800
+++ RadeonProRenderBlenderAddon-3.5.2/src/rprblender/__init__.py	2022-12-24 18:59:11.770232447 -0800
@@ -20,7 +20,7 @@ import bpy
 bl_info = {
     "name": "Radeon ProRender",
     "author": "AMD",
-    "version": (3, 5, 1),
+    "version": (3, 5, 2), # oteodoro: bumped version so the generated zip file name matches
     "blender": (2, 80, 0),
     "location": "Info header, render engine menu",
     "description": "Radeon ProRender rendering plugin for Blender 2.8x",
