diff -ur boinc-client_release-7.4-7.4.42.orig/html/inc/db.inc boinc-client_release-7.4-7.4.42/html/inc/db.inc
--- boinc-client_release-7.4-7.4.42.orig/html/inc/db.inc	2015-12-23 10:41:09.088782101 -0800
+++ boinc-client_release-7.4-7.4.42/html/inc/db.inc	2015-12-23 10:43:42.659519420 -0800
@@ -49,6 +49,7 @@
     if ($host == null) {
         $host = "localhost";
     }
+    error_reporting(E_ALL ^ E_DEPRECATED);
     $link = mysql_pconnect($host, $user, $pass);
     if (!$link) {
         return 1;
diff -ur boinc-client_release-7.4-7.4.42.orig/html/inc/db_conn.inc boinc-client_release-7.4-7.4.42/html/inc/db_conn.inc
--- boinc-client_release-7.4-7.4.42.orig/html/inc/db_conn.inc	2015-12-23 10:41:09.088782101 -0800
+++ boinc-client_release-7.4-7.4.42/html/inc/db_conn.inc	2015-12-23 10:43:44.082572585 -0800
@@ -41,6 +41,7 @@
                 $this->db_conn = new mysqli("p:".$host, $user, $passwd);
             }
         } else {
+            error_reporting(E_ALL ^ E_DEPRECATED);
             $this->db_conn = mysql_pconnect($host, $user, $passwd);
         }
         if (!$this->db_conn) {
diff -ur boinc-client_release-7.4-7.4.42.orig/html/inc/util_ops.inc boinc-client_release-7.4-7.4.42/html/inc/util_ops.inc
--- boinc-client_release-7.4-7.4.42.orig/html/inc/util_ops.inc	2015-12-23 10:41:09.090782176 -0800
+++ boinc-client_release-7.4-7.4.42/html/inc/util_ops.inc	2015-12-23 10:43:40.259429756 -0800
@@ -94,6 +94,7 @@
     print "Database password (if any): ";
     $pass = rtrim(fgets($in, 80));
 
+    error_reporting(E_ALL ^ E_DEPRECATED);
     $retval = mysql_pconnect($host, $user, $pass);
     if (!$retval) {
         die("Can't connect to DB\n");
