diff -ur boinc-client_release-7.4-7.4.42.orig/html/ops/size_census.php boinc-client_release-7.4-7.4.42/html/ops/size_census.php
--- boinc-client_release-7.4-7.4.42.orig/html/ops/size_census.php	2016-01-10 15:20:02.988902226 -0800
+++ boinc-client_release-7.4-7.4.42/html/ops/size_census.php	2016-01-10 15:38:05.039994930 -0800
@@ -40,7 +40,7 @@
     $query = "select et_avg, host.on_frac, host.active_frac, host.gpu_active_frac, app_version.plan_class " .
         " from DBNAME.host_app_version, DBNAME.host, DBNAME.app_version " .
         " where host_app_version.app_version_id = app_version.id " .
-        " and app_version.appid = $app->id " .
+        " and app_version.appid = " . $app->id  .
         " and et_n > 0 " .
         " and host.id = host_app_version.host_id";
     $result = $db->do_query($query);
diff -ur boinc-client_release-7.4-7.4.42.orig/sched/sched_score.cpp boinc-client_release-7.4-7.4.42/sched/sched_score.cpp
--- boinc-client_release-7.4-7.4.42.orig/sched/sched_score.cpp	2016-01-10 15:20:03.033903939 -0800
+++ boinc-client_release-7.4-7.4.42/sched/sched_score.cpp	2016-01-10 15:31:17.590548080 -0800
@@ -43,7 +43,7 @@
 // given the host's estimated speed, determine its size class
 //
 static int get_size_class(APP& app, double es) {
-    for (int i=0; i<app.n_size_classes-1; i++) {
+    for (int i=0; i<app.n_size_classes; i++) {
         if (es < app.size_class_quantiles[i]) return i;
     }
     return app.n_size_classes - 1;
diff -ur boinc-client_release-7.4-7.4.42.orig/sched/sched_shmem.cpp boinc-client_release-7.4-7.4.42/sched/sched_shmem.cpp
--- boinc-client_release-7.4-7.4.42.orig/sched/sched_shmem.cpp	2016-01-10 15:20:03.033903939 -0800
+++ boinc-client_release-7.4-7.4.42/sched/sched_shmem.cpp	2016-01-10 15:25:21.324019322 -0800
@@ -139,7 +139,7 @@
                 );
                 return ERR_FOPEN;
             }
-            for (int i=0; i<app.n_size_classes-1; i++) {
+            for (int i=0; i<app.n_size_classes; i++) {
                 char buf[256];
                 char* p = fgets(buf, 256, f);
                 if (!p) {
