--- boinc-client_release-7.4-7.4.42.orig/stripchart/stripchart.cnf	2015-12-23 18:43:55.137916343 -0800
+++ boinc-client_release-7.4-7.4.42/stripchart/stripchart.cnf	2015-12-23 20:57:48.792219858 -0800
@@ -4,7 +4,7 @@
 $majorversion = 2; $minorversion = 1;
 
 # Directory where gnuplot is
-$gnuplot = "/usr/local/bin"
+$gnuplot = "/usr/bin"
  
 # Temporary files
 $suffix = rand(10000);
@@ -48,11 +48,11 @@
 
 # Where is the list of datafiles for stripchart.cgi?
 # Note: there is a sample copy in the samples directory: 
-$datafilelist = "datafiles";
+$datafilelist = "../lib/datafiles";
  
 # Where is the list of user-definied cgi queries?
 # Note: this file gets created by the cgi - must put it somewhere that the cgi user can write to 
-$queryfilelist = "querylist";
+$queryfilelist = "../lib/querylist";
 
 # What time is it right now?
 $rightnow = time;
