--- boinc-client_release-7.4-7.4.42.orig/sched/Makefile.am	2015-12-27 03:00:06.218699313 -0800
+++ boinc-client_release-7.4-7.4.42/sched/Makefile.am	2015-12-27 08:53:01.607155194 -0800
@@ -49,7 +49,7 @@
 libsched_fcgi_la_CFLAGS = -D_USING_FCGI_ $(AM_CPPFLAGS)
 libsched_fcgi_la_CXXFLAGS = -D_USING_FCGI_ $(AM_CPPFLAGS)
 libsched_fcgi_la_LDFLAGS= -L$(libdir) -rpath $(libdir) -version-number $(LIBBOINC_VERSION)
-libsched_fcgi_la_LIBADD=
+libsched_fcgi_la_LIBADD=$(MYSQL_LIBS)
 
 endif
 # end of "if ENABLE_FCGI"
