--- boinc-client_release-7.4-7.4.42.orig/samples/wrapper/Makefile	2016-01-12 23:13:33.809855250 -0800
+++ boinc-client_release-7.4-7.4.42/samples/wrapper/Makefile	2016-01-12 23:58:33.633992679 -0800
@@ -16,7 +16,8 @@
     -L$(BOINC_API_DIR) \
     -L$(BOINC_LIB_DIR) \
     -L$(BOINC_ZIP_DIR) \
-    -L.
+    -L. \
+    -L/usr/lib
 
 PROGS = wrapper
 
@@ -37,5 +38,5 @@
 	regexp_memory.o \
 	regexp_report.o
 
-wrapper: wrapper.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a $(REGEXP_OBJS)
+wrapper: wrapper.o libstdc++.a $(REGEXP_OBJS)
 	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -o wrapper wrapper.o $(REGEXP_OBJS) libstdc++.a $(PTHREAD) -lboinc_api -lboinc -lboinc_zip
