diff -urp samtal-0.1.0_p9999.orig/src/app/qml/BasePage.qml samtal-0.1.0_p9999/src/app/qml/BasePage.qml
--- samtal-0.1.0_p9999.orig/src/app/qml/BasePage.qml	2022-12-04 12:02:41.827759727 -0800
+++ samtal-0.1.0_p9999/src/app/qml/BasePage.qml	2022-12-04 12:03:31.801730422 -0800
@@ -8,7 +8,7 @@ import Fluid.Controls 1.0 as FluidContro
 FluidControls.Page {
     actions: [
         FluidControls.Action {
-            iconName: "content/add"
+            icon.source: FluidControls.Utils.iconUrl("content/add")
             toolTip: qsTr("Join a new channel")
             enabled: connection.connected
             onTriggered: joinDialog.open()
diff -urp samtal-0.1.0_p9999.orig/src/app/qml/BufferSideBar.qml samtal-0.1.0_p9999/src/app/qml/BufferSideBar.qml
--- samtal-0.1.0_p9999.orig/src/app/qml/BufferSideBar.qml	2022-12-04 12:02:41.827759727 -0800
+++ samtal-0.1.0_p9999/src/app/qml/BufferSideBar.qml	2022-12-04 12:03:13.180996118 -0800
@@ -38,7 +38,7 @@ Pane {
         }
         */
         delegate: FluidControls.ListItem {
-            iconName: "communication/chat"
+            icon.source: FluidControls.Utils.iconUrl("communication/chat")
             text: model.title || model.name || model.display || qsTr("Unknown")
             highlighted: model.buffer === currentBuffer
 
diff -urp samtal-0.1.0_p9999.orig/src/app/qml/PlaceholderPage.qml samtal-0.1.0_p9999/src/app/qml/PlaceholderPage.qml
--- samtal-0.1.0_p9999.orig/src/app/qml/PlaceholderPage.qml	2022-12-04 12:02:41.827759727 -0800
+++ samtal-0.1.0_p9999/src/app/qml/PlaceholderPage.qml	2022-12-04 12:03:54.874640306 -0800
@@ -10,7 +10,7 @@ import Fluid.Controls 1.0 as FluidContro
 FluidControls.Page {
     actions: [
         FluidControls.Action {
-            iconName: "content/add"
+            icon.source: FluidControls.Utils.iconUrl("content/add")
             toolTip: qsTr("Add network")
             onTriggered: addNetworkDialog.open()
         }
@@ -99,7 +99,7 @@ FluidControls.Page {
     FluidControls.Placeholder {
         anchors.fill: parent
 
-        iconName: "communication/chat_bubble_outline"
+        icon.source: FluidControls.Utils.iconUrl("communication/chat_bubble_outline")
         text: qsTr("Liri IRC")
         subText: "Join a room using the + button."
     }
diff -urp samtal-0.1.0_p9999.orig/src/app/qml/UsersSideBar.qml samtal-0.1.0_p9999/src/app/qml/UsersSideBar.qml
--- samtal-0.1.0_p9999.orig/src/app/qml/UsersSideBar.qml	2022-12-04 12:02:41.827759727 -0800
+++ samtal-0.1.0_p9999/src/app/qml/UsersSideBar.qml	2022-12-04 12:03:13.180996118 -0800
@@ -36,17 +36,17 @@ Pane {
             }
         }
         delegate: FluidControls.ListItem {
-            iconName: {
+            icon.source: {
                 switch (model.mode) {
                 case "o":
-                    return "action/supervisor_account";
+                    return FluidControls.Utils.iconUrl("action/supervisor_account");
                 case "v":
-                    return "action/verified_user";
+                    return FluidControls.Utils.iconUrl("action/verified_user");
                 default:
                     break;
                 }
 
-                return "action/account_box";
+                return FluidControls.Utils.iconUrl("action/account_box");
             }
             text: model.title
             opacity: model.away ? 0.5 : 1.0
