diff -ur tolua++-1.0.93.orig/src/bin/lua/feature.lua tolua++-1.0.93/src/bin/lua/feature.lua
--- tolua++-1.0.93.orig/src/bin/lua/feature.lua	2016-02-06 18:16:33.368161556 -0800
+++ tolua++-1.0.93/src/bin/lua/feature.lua	2016-02-06 18:18:26.114458125 -0800
@@ -132,7 +132,7 @@
  if not fname or fname == '' then
  	fname = self.name
  end
-  n = string.gsub(n..'_'.. (fname), "[<>:, \.%*&]", "_")
+  n = string.gsub(n..'_'.. (fname), "[<>:, %.%*&]", "_")
 
   return n
 end
