diff -ur tolua++-1.0.93.orig/src/bin/lua/clean.lua tolua++-1.0.93/src/bin/lua/clean.lua
--- tolua++-1.0.93.orig/src/bin/lua/clean.lua	2016-02-06 17:55:04.060910854 -0800
+++ tolua++-1.0.93/src/bin/lua/clean.lua	2016-02-06 17:56:15.720650688 -0800
@@ -9,18 +9,18 @@
 ESC2 = "\007"
 
 MASK = { -- the substitution order is important
- {ESC1, "\\'"},
- {ESC2, '\\"'},
- {STR1, "'"},
- {STR2, '"'},
- {STR3, "%[%["},
- {STR4, "%]%]"},
- {REM , "%-%-"},
+ {ESC1, "\\'", "\\'"},
+ {ESC2, '\\"', '\\"'},
+ {STR1, "'", "'"},
+ {STR2, '"', '"'},
+ {STR3, "%[%[", "[["},
+ {STR4, "%]%]", "]]"},
+ {REM , "%-%-", "--"},
 }
 
 function mask (s)
  for i = 1,getn(MASK)  do
-  s = gsub(s,MASK[i][2],MASK[i][1])
+  s = gsub(s,MASK[i][2],MASK[i][3])
  end
  return s
 end
