diff -ur tolua++-1.0.93.orig/src/bin/lua/declaration.lua tolua++-1.0.93/src/bin/lua/declaration.lua
--- tolua++-1.0.93.orig/src/bin/lua/declaration.lua	2016-02-06 18:12:10.877160204 -0800
+++ tolua++-1.0.93/src/bin/lua/declaration.lua	2016-02-06 18:13:34.711354159 -0800
@@ -520,7 +520,7 @@
  end
 
  -- check the form: mod type* name
- local s1 = gsub(s,"(%b\[\])",function (n) return gsub(n,'%*','\1') end)
+ local s1 = gsub(s,"(%b%[%])",function (n) return gsub(n,'%*','\1') end)
  t = split_c_tokens(s1,'%*')
  if t.n == 2 then
   t[2] = gsub(t[2],'\1','%*') -- restore * in dimension expression
