diff -urp nano-ycmd-9999.orig/src/Makefile.am nano-ycmd-9999/src/Makefile.am
--- nano-ycmd-9999.orig/src/Makefile.am	2023-07-04 15:59:46.988439733 -0700
+++ nano-ycmd-9999/src/Makefile.am	2023-07-04 16:05:28.320446691 -0700
@@ -151,8 +151,8 @@ revision.h: FORCE
 			echo "#define $(SOMETHING)" > $@
 FORCE:
 
-bin_PROGRAMS = 	nano
-nano_SOURCES = \
+bin_PROGRAMS = 	ynano
+ynano_SOURCES = \
 		definitions.h \
 		prototypes.h \
 		browser.c \
@@ -173,29 +173,29 @@ nano_SOURCES = \
 		winio.c
 
 if USE_YCMD
-nano_SOURCES += \
+ynano_SOURCES += \
 		ycmd.c
 endif
 
-nano_LDADD = @LIBINTL@ $(top_builddir)/lib/libgnu.a \
+ynano_LDADD = @LIBINTL@ $(top_builddir)/lib/libgnu.a \
 		$(LIB_CLOCK_GETTIME) $(LIBTHREAD)
 
 if USE_YCMD
-nano_LDADD += @NEON_LIBS@
+ynano_LDADD += @NEON_LIBS@
 endif
 
 if HAVE_NETTLE
-nano_LDADD += @NETTLE_LIBS@
+ynano_LDADD += @NETTLE_LIBS@
 endif
 if HAVE_OPENSSL
-nano_LDADD += @OPENSSL_LIBS@
+ynano_LDADD += @OPENSSL_LIBS@
 endif
 if HAVE_LIBGCRYPT
-nano_LDADD += @GLIB_LIBS@
-nano_LDFLAGS = $(LIBGCRYPT_LIBS)
+ynano_LDADD += @GLIB_LIBS@
+ynano_LDFLAGS = $(LIBGCRYPT_LIBS)
 endif
 if HAVE_YCMD
-nano_LDADD += @GLIB_LIBS@
+ynano_LDADD += @GLIB_LIBS@
 endif
 
 install-exec-hook:
