diff -urp nano-ycmd-052b4866f3b24caeed877ae6f017f422d1443ed9.orig/src/Makefile.am nano-ycmd-052b4866f3b24caeed877ae6f017f422d1443ed9/src/Makefile.am
--- nano-ycmd-052b4866f3b24caeed877ae6f017f422d1443ed9.orig/src/Makefile.am	2017-02-05 20:47:06.000000000 -0800
+++ nano-ycmd-052b4866f3b24caeed877ae6f017f422d1443ed9/src/Makefile.am	2017-02-05 20:52:53.131060090 -0800
@@ -59,8 +59,8 @@ revision.h: update_revision
 
 .PHONY: update_revision
 
-bin_PROGRAMS = 	nano
-nano_SOURCES =	browser.c \
+bin_PROGRAMS = 	ynano
+ynano_SOURCES =	browser.c \
 		chars.c \
 		color.c \
 		cut.c \
@@ -79,23 +79,23 @@ nano_SOURCES =	browser.c \
 		ycmd.c \
 		winio.c
 
-nano_LDADD = @GLIB_LIBS@ @LIBINTL@ @NEON_LIBS@
+ynano_LDADD = @GLIB_LIBS@ @LIBINTL@ @NEON_LIBS@
 if HAVE_NETTLE
-nano_LDADD += @NETTLE_LIBS@
+ynano_LDADD += @NETTLE_LIBS@
 endif
 if HAVE_OPENSSL
-nano_LDADD += @OPENSSL_LIBS@
+ynano_LDADD += @OPENSSL_LIBS@
 endif
 if HAVE_LIBGCRYPT
-nano_LDADD += @GLIB_LIBS@
-nano_LDFLAGS = $(LIBGCRYPT_LIBS)
+ynano_LDADD += @GLIB_LIBS@
+ynano_LDFLAGS = $(LIBGCRYPT_LIBS)
 endif
 if HAVE_YCMD
-nano_LDADD += @GLIB_LIBS@
+ynano_LDADD += @GLIB_LIBS@
 endif
 
 
 install-exec-hook:
-	cd $(DESTDIR)$(bindir) && rm -f rnano && $(LN_S) nano rnano
+	cd $(DESTDIR)$(bindir) && rm -f rnano && $(LN_S) ynano rnano
 uninstall-hook:
 	cd $(DESTDIR)$(bindir) && rm -f rnano
Only in nano-ycmd-052b4866f3b24caeed877ae6f017f422d1443ed9/src: Makefile.am.orig
Only in nano-ycmd-052b4866f3b24caeed877ae6f017f422d1443ed9/src: Makefile.am.rej
