diff -urp nano-ycmd-9999.orig/autogen.sh nano-ycmd-9999/autogen.sh
--- nano-ycmd-9999.orig/autogen.sh	2023-07-02 08:45:17.874077620 -0700
+++ nano-ycmd-9999/autogen.sh	2023-07-02 08:51:13.830086452 -0700
@@ -29,25 +29,31 @@ modules="
 	wcwidth
 "
 
-# Make sure the local gnulib git repo is up-to-date.
-if [ ! -d "gnulib" ]; then
-	git clone --depth=1111 ${gnulib_url}
+if [[ "${GNULIB_USE_TARBALL:-0}" == "0" ]] ; then
+	# Make sure the local gnulib git repo is up-to-date.
+	if [ ! -d "gnulib" ]; then
+		git clone --depth=1111 ${gnulib_url}
+	fi
+	cd gnulib >/dev/null || exit 1
+	curr_hash=$(git log -1 --format=%H)
+	if [ "${gnulib_hash}" != "${curr_hash}" ]; then
+		echo "Pulling..."
+		git pull
+		git checkout --force ${gnulib_hash}
+	fi
+	cd .. >/dev/null || exit 1
 fi
-cd gnulib >/dev/null || exit 1
-curr_hash=$(git log -1 --format=%H)
-if [ "${gnulib_hash}" != "${curr_hash}" ]; then
-	echo "Pulling..."
-	git pull
-	git checkout --force ${gnulib_hash}
-fi
-cd .. >/dev/null || exit 1
 
 echo "Autopoint..."
 autopoint --force
 
 rm -rf lib
 echo "Gnulib-tool..."
-./gnulib/gnulib-tool --import ${modules}
+if [[ "${GNULIB_USE_SYSTEM:-1}" == "1" ]] ; then
+	/usr/bin/gnulib-tool --import ${modules}
+else
+	./gnulib/gnulib-tool --import ${modules}
+fi
 echo
 
 echo "Aclocal..."
