diff -urp nano-ycmd-f7199c52643b43dad27ffbd83219760323536bbe.orig/src/Makefile.am nano-ycmd-f7199c52643b43dad27ffbd83219760323536bbe/src/Makefile.am
--- nano-ycmd-f7199c52643b43dad27ffbd83219760323536bbe.orig/src/Makefile.am	2021-04-17 04:24:13.000000000 -0700
+++ nano-ycmd-f7199c52643b43dad27ffbd83219760323536bbe/src/Makefile.am	2021-04-17 12:48:11.113607074 -0700
@@ -133,8 +133,8 @@ revision.h: FORCE
 			echo "#define $(SOMETHING)" > $@
 FORCE:
 
-bin_PROGRAMS = 	nano
-nano_SOURCES = \
+bin_PROGRAMS = 	ynano
+ynano_SOURCES = \
 		definitions.h \
 		prototypes.h \
 		browser.c \
@@ -155,26 +155,26 @@ nano_SOURCES = \
 		ycmd.c \
 		winio.c
 
-nano_LDADD = @LIBINTL@ $(top_builddir)/lib/libgnu.a \
+ynano_LDADD = @LIBINTL@ $(top_builddir)/lib/libgnu.a \
 		$(LIB_CLOCK_GETTIME) $(LIBTHREAD)
 
-nano_LDADD += @NEON_LIBS@
+ynano_LDADD += @NEON_LIBS@
 
 if HAVE_NETTLE
-nano_LDADD += @NETTLE_LIBS@
+ynano_LDADD += @NETTLE_LIBS@
 endif
 if HAVE_OPENSSL
-nano_LDADD += @OPENSSL_LIBS@
+ynano_LDADD += @OPENSSL_LIBS@
 endif
 if HAVE_LIBGCRYPT
-nano_LDADD += @GLIB_LIBS@
-nano_LDFLAGS = $(LIBGCRYPT_LIBS)
+ynano_LDADD += @GLIB_LIBS@
+ynano_LDFLAGS = $(LIBGCRYPT_LIBS)
 endif
 if HAVE_YCMD
-nano_LDADD += @GLIB_LIBS@
+ynano_LDADD += @GLIB_LIBS@
 endif
 
 install-exec-hook:
-	cd $(DESTDIR)$(bindir) && rm -f rnano && $(LN_S) nano rnano
+	cd $(DESTDIR)$(bindir) && rm -f rnano && $(LN_S) ynano rnano
 uninstall-hook:
 	cd $(DESTDIR)$(bindir) && rm -f rnano
Only in nano-ycmd-f7199c52643b43dad27ffbd83219760323536bbe/src: Makefile.am.orig
Only in nano-ycmd-f7199c52643b43dad27ffbd83219760323536bbe/src: Makefile.am.rej
