diff -urp nano-ycmd-c44e2e2a0d97a0c5d0e93dcab3a769ba1a4d294c.orig/src/Makefile.am nano-ycmd-c44e2e2a0d97a0c5d0e93dcab3a769ba1a4d294c/src/Makefile.am
--- nano-ycmd-c44e2e2a0d97a0c5d0e93dcab3a769ba1a4d294c.orig/src/Makefile.am	2017-02-04 23:02:01.000000000 -0800
+++ nano-ycmd-c44e2e2a0d97a0c5d0e93dcab3a769ba1a4d294c/src/Makefile.am	2017-02-04 23:11:18.824436398 -0800
@@ -59,8 +59,8 @@ revision.h: update_revision
 
 .PHONY: update_revision
 
-bin_PROGRAMS = 	nano
-nano_SOURCES =	browser.c \
+bin_PROGRAMS = 	ynano
+ynano_SOURCES =	browser.c \
 		chars.c \
 		color.c \
 		cut.c \
@@ -80,22 +80,22 @@ nano_SOURCES =	browser.c \
 		ycmd.c \
 		winio.c
 
-nano_LDADD = @GLIB_LIBS@ @LIBINTL@ @NEON_LIBS@
+ynano_LDADD = @GLIB_LIBS@ @LIBINTL@ @NEON_LIBS@
 if HAVE_NETTLE
-nano_LDADD += @NETTLE_LIBS@
+ynano_LDADD += @NETTLE_LIBS@
 endif
 if HAVE_OPENSSL
-nano_LDADD += @OPENSSL_LIBS@
+ynano_LDADD += @OPENSSL_LIBS@
 endif
 if HAVE_LIBGCRYPT
-nano_LDADD += @GLIB_LIBS@
+ynano_LDADD += @GLIB_LIBS@
 endif
 if HAVE_YCMD
-nano_LDADD += @GLIB_LIBS@
+ynano_LDADD += @GLIB_LIBS@
 endif
 
 
 install-exec-hook:
-	cd $(DESTDIR)$(bindir) && rm -f rnano && $(LN_S) nano rnano
+	cd $(DESTDIR)$(bindir) && rm -f rnano && $(LN_S) ynano rnano
 uninstall-hook:
 	cd $(DESTDIR)$(bindir) && rm -f rnano
