diff -urp finit-4.6.orig/contrib/Makefile.am finit-4.6/contrib/Makefile.am
--- finit-4.6.orig/contrib/Makefile.am	2023-12-29 14:24:08.607680126 -0800
+++ finit-4.6/contrib/Makefile.am	2023-12-29 14:27:16.469971290 -0800
@@ -1,5 +1,5 @@
 SUBDIRS    = alpine debian gentoo void
-bin_SCRIPT = service
+bin_SCRIPTS = service
 docsdir   := @docdir@/contrib
 docs_DATA  = README.md finit.conf procmon.sh
 EXTRA_DIST = $(docs_DATA)
