--- AppImageKit-08800854de05f4f6f7c1f3901dc165b8518822e1/src/runtime.c.orig	2020-08-11 11:40:12.475681204 -0700
+++ AppImageKit-08800854de05f4f6f7c1f3901dc165b8518822e1/src/runtime.c	2020-08-11 11:40:30.956399927 -0700
@@ -66,6 +66,7 @@
 
 //#include "notify.c"
 extern int notify(char *title, char *body, int timeout);
+extern ssize_t appimage_get_elf_size(const char* fname);
 struct stat st;
 
 static ssize_t fs_offset; // The offset at which a filesystem image is expected = end of this ELF
