--- a/setup.py.orig	2024-03-18 06:01:38.000000000 -0700
+++ b/setup.py	2024-06-09 18:08:08.537726297 -0700
@@ -288,7 +288,7 @@ def get_ext_modules():
                 libraries=['tbb'],  # TODO: if --debug or -g, use 'tbb_debug'
                 library_dirs=[
                     # for Linux
-                    os.path.join(tbb_root, 'lib', 'intel64', 'gcc4.4'),
+                    os.path.join(tbb_root, 'lib', 'lib64', 'intel64', 'gcc4.4'),
                     # for MacOS
                     os.path.join(tbb_root, 'lib'),
                     # for Windows
