--- a/cmake/Modules/FindGRPC.cmake.orig	2024-06-15 10:21:32.000000000 -0700
+++ b/cmake/Modules/FindGRPC.cmake	2025-10-28 16:25:21.800373370 -0700
@@ -3,7 +3,7 @@ option(ENABLE_GRPC_REFLECTION "Link to g
 # FIXME(kirillbobyrev): Check if gRPC and Protobuf headers can be included at
 # configure time.
 find_package(Threads REQUIRED)
-if (GRPC_INSTALL_PATH)
+if (GRPC_INSTALL_PATH AND PROTOBUF_INSTALL_PATH)
   # This setup requires gRPC to be built from sources using CMake and installed
   # to ${GRPC_INSTALL_PATH} via -DCMAKE_INSTALL_PREFIX=${GRPC_INSTALL_PATH}.
   # Libraries will be linked according to gRPC build policy which generates
@@ -11,7 +11,7 @@ if (GRPC_INSTALL_PATH)
   # it's On (NOTE: This is a variable passed to gRPC CMake build invocation,
   # LLVM's BUILD_SHARED_LIBS has no effect).
   set(protobuf_MODULE_COMPATIBLE TRUE)
-  find_package(Protobuf CONFIG REQUIRED HINTS ${GRPC_INSTALL_PATH})
+  find_package(Protobuf CONFIG REQUIRED HINTS ${PROTOBUF_INSTALL_PATH})
   message(STATUS "Using protobuf ${Protobuf_VERSION}")
   find_package(gRPC CONFIG REQUIRED HINTS ${GRPC_INSTALL_PATH})
   message(STATUS "Using gRPC ${gRPC_VERSION}")
