--- setiathome-cpu-8.00.orig/AKv8/client/analyzeFuncs.cpp	2016-01-02 10:59:33.196551865 -0800
+++ setiathome-cpu-8.00/AKv8/client/analyzeFuncs.cpp	2016-01-02 11:25:16.754296493 -0800
@@ -790,7 +790,7 @@
             SETIERROR(WRITE_FAILED,"from outfile.printf (best triplet) in seti_analyze()");
         }
     }
-    retval=outfile.printf("<setiathome_version>"PACKAGE_VERSION"</setiathome_version>\n");
+    //retval=outfile.printf("<setiathome_version>"PACKAGE_VERSION"</setiathome_version>\n");
 
 #ifdef BOINC_APP_GRAPHICS
     if (sah_graphics) strcpy(sah_graphics->status, "Work unit done");
