diff -urp setiathome-cpu-8.00.orig/AKv8/client/analyzeFuncs.cpp setiathome-cpu-8.00/AKv8/client/analyzeFuncs.cpp
--- setiathome-cpu-8.00.orig/AKv8/client/analyzeFuncs.cpp	2017-09-26 10:38:13.874643207 -0700
+++ setiathome-cpu-8.00/AKv8/client/analyzeFuncs.cpp	2017-09-26 10:42:53.441248017 -0700
@@ -790,7 +790,7 @@ int seti_analyze (ANALYSIS_STATE& state)
             SETIERROR(WRITE_FAILED,"from outfile.printf (best triplet) in seti_analyze()");
         }
     }
-    retval=outfile.printf("<setiathome_version>" PACKAGE_VERSION "</setiathome_version>\n");
+    retval=outfile.printf("<setiathome_version> __PACKAGE_VERSION__ </setiathome_version>\n");
 
 #ifdef BOINC_APP_GRAPHICS
     if (sah_graphics) strcpy(sah_graphics->status, "Work unit done");
