--- setiathome-gpu-8.00.orig/AKv8/client/analyzePoT.cpp	2016-01-02 01:58:49.216959827 -0800
+++ setiathome-gpu-8.00/AKv8/client/analyzePoT.cpp	2016-01-02 02:23:57.605654900 -0800
@@ -567,6 +567,9 @@
 	size_t globalThreadsGaussian=0;
   if(!SkipGauss && (analysis_state.PoT_activity == POT_DOING_GAUSS ||
                     analysis_state.PoT_activity == POT_INACTIVE)) {
+#ifdef BOINC_APP_GRAPHICS
+    if (sah_graphics) strcpy(sah_graphics->status, "Searching for Gaussians");
+#endif
 	   // fprintf(stderr,"Gaussian: icfft=%d\n",analysis_state.icfft);
 #if USE_CUDA
 		need_gaussian_cpu_processing=PC_FindGaussians_cu(
@@ -1152,9 +1155,7 @@
 	clFlush(cq_gaussian_neg);
 #endif
 #else
-#ifdef BOINC_APP_GRAPHICS
-    if (sah_graphics) strcpy(sah_graphics->status, "Searching for Gaussians");
-#endif
+
 
     // If we are back from being interrupted in the middle of gaussian PoT
     // analysis, load state and continue.  Otherwise start anew, skipping
