--- setiathome-gpu-8.22.orig/AKv8/client/analyzePoT.cpp	2017-01-19 09:43:27.446716463 -0800
+++ setiathome-gpu-8.22/AKv8/client/analyzePoT.cpp	2017-01-19 09:50:03.102930640 -0800
@@ -641,6 +641,9 @@ int analyze_pot(
 	size_t globalThreadsGaussian=0;
   if(!SkipGauss && (analysis_state.PoT_activity == POT_DOING_GAUSS ||
                     analysis_state.PoT_activity == POT_INACTIVE)) {
+#ifdef BOINC_APP_GRAPHICS
+    if (sah_graphics) strcpy(sah_graphics->status, "Searching for Gaussians");
+#endif
 	   // fprintf(stderr,"Gaussian: icfft=%d\n",analysis_state.icfft);
 #if USE_CUDA
 		need_gaussian_cpu_processing=PC_FindGaussians_cu(
@@ -1143,9 +1146,6 @@ if(1||analysis_state.icfft==10)
 #endif
 	clFlush(cq_gaussian);//R: orders to start computations on GPU with Gaussian block of kernels
 #else
-#ifdef BOINC_APP_GRAPHICS
-    if (!nographics()) strcpy(sah_graphics->status, "Searching for Gaussians");
-#endif
 
     // If we are back from being interrupted in the middle of gaussian PoT
     // analysis, load state and continue.  Otherwise start anew, skipping
