--- AKv8/client/seti.h.orig	2015-12-04 16:42:14.047670387 -0800
+++ AKv8/client/seti.h	2015-12-04 16:44:09.788120891 -0800
@@ -179,14 +179,6 @@
 extern ANALYSIS_STATE analysis_state;
 extern SETI_WU_INFO swi;
 
-#ifndef GUISO
-class GDATA;
-extern GDATA *sah_graphics;
-#else
-class SAH_GRAPHICS;
-extern SAH_GRAPHICS *sah_graphics;
-#endif
-
 #ifdef DYNAMIC_GRAPHICS
 typedef int (*int_func_t)();
 extern int_func_t f_glut_is_initialized;
