--- AKv8/client/main.cpp.orig	2015-12-04 09:16:38.223659561 -0800
+++ AKv8/client/main.cpp	2015-12-04 09:18:22.125676294 -0800
@@ -65,26 +65,26 @@
 
 #ifdef BOINC_APP_GRAPHICS
 #include "sah_gfx_main.h"
-#include "graphics2.h"
+//#include "graphics2.h"
 #endif
 
-#ifdef BOINC_APP_GRAPHICS
-#include "reduce.h"
-#include "graphics_data.h"
-#include "graphics_api.h"
-#ifdef DYNAMIC_GRAPHICS
-#include "graphics_lib.h"
-#endif
-#endif
+//#ifdef BOINC_APP_GRAPHICS
+//#include "reduce.h"
+//#include "graphics_data.h"
+//#include "graphics_api.h"
+//#ifdef DYNAMIC_GRAPHICS
+//#include "graphics_lib.h"
+//#endif
+//#endif
 
-#ifdef BOINC_APP_GRAPHICS
-#include "graphics_api.h"
-#ifdef DYNAMIC_GRAPHICS
-#include "graphics_lib.h"
-#endif
-#include "graphics_data.h"
+//#ifdef BOINC_APP_GRAPHICS
+//#include "graphics_api.h"
+//#ifdef DYNAMIC_GRAPHICS
+//#include "graphics_lib.h"
+//#endif
+//#include "graphics_data.h"
 #include "sah_gfx.h"
-#endif
+//#endif
 
 #include "util.h"
 #include "s_util.h"
