--- setiathome-gpu-8.00.orig/AKv8/client/gaussfit.cpp	2016-01-02 01:58:49.183958568 -0800
+++ setiathome-gpu-8.00/AKv8/client/gaussfit.cpp	2016-01-02 02:29:10.687663745 -0800
@@ -690,6 +690,7 @@
            -lcgf(0.5*null_dof,std::max(null_chisqr*0.5*gauss_bins,0.5*null_dof+1))
            +lcgf_null;
 }
+#if USE_OPENCL
 #if SIGNALS_ON_GPU
 void LogGaussian(cl_GPUGaussian& gpu_gaussian,bool best){
 	if(gpu_gaussian.icfft==-1)return;//R: empty slot
@@ -722,6 +723,9 @@
 		  Counter<Gaussian_new_best>::update(1);
 #endif
 		  *best_gauss = gi;
+#ifdef BOINC_APP_GRAPHICS
+                  if (!nographics()) sah_graphics->gi.copy(&gi);
+#endif      
 		  if(verbose>=2 && verbose <6){
 			  fprintf(stderr,"LogGaussian best gaussian updated: score=%.7g, fft_len=%d, PoT=%d, Offset=%d,\n\tPeak=%.7g, TrueMean=%.7g,ChiSq=%.7g,null_hyp=%.7g,PoTMaxPower=%.7g,icfft=%d\n",
 			gi.score,ChirpFftPairs[gpu_gaussian.icfft].FftLen,gi.bin,gi.fft_ind,
@@ -737,6 +741,9 @@
 		  Counter<Gaussian_new_best>::update(1);
 #endif
 		  *best_gauss = gi;
+#ifdef BOINC_APP_GRAPHICS
+                  if (!nographics()) sah_graphics->gi.copy(&gi);
+#endif      
 		  if(verbose>=2 && verbose <6){
 			  fprintf(stderr,"LogGaussian best gaussian updated: score=%.7g, fft_len=%d, PoT=%d, Offset=%d,\n\tPeak=%.7g, TrueMean=%.7g,ChiSq=%.7g,null_hyp=%.7g,PoTMaxPower=%.7g,icfft=%d\n",
 			gi.score,ChirpFftPairs[gpu_gaussian.icfft].FftLen,gi.bin,gi.fft_ind,
@@ -772,6 +779,7 @@
 
 }
 #endif
+#endif
 int ChooseGaussEvent(
   int ifft,
   float PeakPower,
@@ -882,10 +890,7 @@
       gi.g.pot.set_size(swi.analysis_cfg.gauss_pot_length);
       float_to_uchar(fp_PoT, &(gi.g.pot[0]), swi.analysis_cfg.gauss_pot_length, scale_factor);
 
-#ifdef BOINC_APP_GRAPHICS
-      if (graphics)
-        sah_graphics->gi.copy(&gi);
-#endif
+
 #if 0
 	  if(analysis_state.icfft==9 && ChirpFftPairs[analysis_state.icfft].FftLen==4096 && bin==713 && ifft==18){
 		  fprintf(stderr,"Forced gaussian dump:\n");
