--- AKv8/client/Makefile.am.orig	2015-12-04 10:21:30.755176574 -0800
+++ AKv8/client/Makefile.am	2015-12-04 10:20:28.931807490 -0800
@@ -30,6 +30,8 @@
 CLIENT_LD_ADD = $(PTHREAD_LIBS) $(APP_LIBS) $(BOINC_LIBS)
 CLIENT_LD_ADD += $(AM_LIBS)
 
+CXXFLAGS += $(BOINC_CFLAGS) -I$(top_srcdir)/db -I$(top_srcdir)/../src
+
 if AVX2
 #uje: Have to refuse lower SIMD-usage not to confuse compiler later
 CLIENT_CXX_FLAGS+=-mavx2 -mfpmath=sse -DUSE_AVX2
