diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AKv8/client/Makefile.am setiathome-7.08/AKv8/client/Makefile.am
--- setiathome-7.08.orig/AKv8/client/Makefile.am	2015-12-07 14:56:19.449194244 -0800
+++ setiathome-7.08/AKv8/client/Makefile.am	2015-12-07 15:05:28.761951758 -0800
@@ -110,7 +110,7 @@
     CLIENT_BIN = $(CLIENT_PROG) $(CLIENT_NATIVE) $(DEBUG_NATIVE) $(DEBUG_PROG)
     GUI_BIN = seti_graphics
     GUI_C_FLAGS = $(CLIENT_C_FLAGS) $(GRAPHICS_CFLAGS)
-    GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
+    GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
     CLIENT_LD_FLAGS += $(LDSTATIC)
     CLIENT_LD_ADD += -L$(BOINCDIR)/api -L$(BOINCDIR)/lib $(BOINC_LIBS) $(AM_LIBS)
     SAH_GRX_SOURCES = sah_gfx_main.cpp
