--- setiathome-gpu-8.00.orig/AKv8/client/spike.cpp	2016-01-02 01:58:49.082954715 -0800
+++ setiathome-gpu-8.00/AKv8/client/spike.cpp	2016-01-02 02:39:23.989215474 -0800
@@ -367,13 +367,14 @@
     // if best_spike.s.fft_len == 0, there is not yet a first spike
     if (si.score > best_spike->score || best_spike->s.fft_len == 0) {
       *best_spike 			= si;
+#ifdef BOINC_APP_GRAPHICS
+      if (!nographics()) sah_graphics->si.copy(&si);
+#endif
 	  if(verbose>=2 && verbose<6){//Best Spike dump
 	  fprintf(stderr,"New best spike:score:%.5g, power: %.5g, index=%d, fft_len=%d, ifft=%d,icfft=%d\n",
 		  si.score,si.s.peak_power,si.bin,si.s.fft_len,si.fft_ind,analysis_state.icfft);
 	  }
-#ifdef BOINC_APP_GRAPHICS
-      if (!nographics()) sah_graphics->si.copy(&si);
-#endif
+
     }
 
     // Report a signal if it excceeds threshold.
@@ -437,6 +438,9 @@
 #endif
     if (new_best) {
       *best_spike 			= si;
+#ifdef BOINC_APP_GRAPHICS
+      if (!nographics()) sah_graphics->si.copy(&si);
+#endif      
 	  if(verbose>=2 && verbose<6){//Best Spike dump
 		fprintf(stderr,"R New best spike:score:%.5g, power: %.5g, index=%d, fft_len=%d, ifft=%d,icfft=%d\n",
 		  si.score,si.s.peak_power,si.bin,si.s.fft_len,si.fft_ind,analysis_state.icfft);
@@ -475,6 +479,9 @@
     time_to_ra_dec(si.s.time, &si.s.ra, &si.s.decl);
 	if(best && best_spike->s.peak_power<si.s.peak_power){//R: update only if really new one
 		*best_spike = si;
+#ifdef BOINC_APP_GRAPHICS
+                if (!nographics()) sah_graphics->si.copy(&si);
+#endif    
 		if(verbose>=2 && verbose<6){ //Best spike dump
 	     fprintf(stderr,"LogSpike New best spike: score:%.5g, power: %.5g, index=%d, fft_len=%d, ifft=%d, icfft=%d\n",
 		  si.score,si.s.peak_power,si.bin,si.s.fft_len,si.fft_ind,gpu_spike.icfft);
@@ -540,6 +547,9 @@
 
 	if (new_best) {
       *best_spike = si;
+#ifdef BOINC_APP_GRAPHICS
+      if (!nographics()) sah_graphics->si.copy(&si);
+#endif      
 	  if(verbose>=2 && verbose<6){ //Best spike dump
 	  fprintf(stderr,"R1 New best spike: score:%.5g, power: %.5g, index=%d, fft_len=%d, ifft=%d, icfft=%d\n",
 		  si.score,si.s.peak_power,si.bin,si.s.fft_len,si.fft_ind,analysis_state.icfft);
@@ -602,6 +612,9 @@
 
 	if (si.score > best_spike->score || best_spike->s.fft_len == 0) {
       *best_spike 			= si;
+#ifdef BOINC_APP_GRAPHICS
+      if (!nographics()) sah_graphics->si.copy(&si);
+#endif
 	  if(verbose>=2 && verbose<6){ //Best spike dump
 	  fprintf(stderr,"R2 New best spike: score:%.5g, power: %.5g, index=%d, fft_len=%d, ifft=%d, icfft=%d\n",
 		  si.score,si.s.peak_power,si.bin,si.s.fft_len,si.fft_ind,icfft);
