diff -urp setiathome-gpu-8.22.orig/AKv8/client/analyzePoT.cpp setiathome-gpu-8.22/AKv8/client/analyzePoT.cpp
--- setiathome-gpu-8.22.orig/AKv8/client/analyzePoT.cpp	2017-01-19 20:13:05.640567376 -0800
+++ setiathome-gpu-8.22/AKv8/client/analyzePoT.cpp	2017-01-19 20:18:17.194557051 -0800
@@ -484,7 +484,7 @@ int analyze_pot(
        PulsePoTLen,      // length of PoT segment passed to pulse finders
        Overlap,          // PoT segment overlap in bins
        TOffset,          // index into ThisPoT of current pulse segment
-       //PulsePoTNum  = 0, // the oridinal position of a pulse PoT w/in a full PoT
+       PulsePoTNum  = 0, // the oridinal position of a pulse PoT w/in a full PoT
        NumPulsePoTs = 0, // the number of pulse PoTs w/in a full PoT.  This is
 			 //   constant regardless of FFT or PoT length and is
 			 //   determined by slew rate.
@@ -1238,7 +1238,7 @@ if(1||analysis_state.icfft==10)
       ThisPoT = 1; // skip the DC bin on start of new cfft pair
     }
 
-    //PulsePoTNum = 0;
+    PulsePoTNum = 0;
 
 #ifdef BOINC_APP_GRAPHICS
     // Inital display of Local Progress
@@ -3145,9 +3145,9 @@ CPU_pulse_processing:
 
       // loop through time for each frequency.  PulsePoTNum is
       // used only for progress calculation.
-      for(TOffset = 0, /*PulsePoTNum = 1,*/ TOffsetOK = true;
+      for(TOffset = 0, PulsePoTNum = 1, TOffsetOK = true;
           TOffsetOK;
-          /*PulsePoTNum++,*/ TOffset += AdvanceBy) {
+          PulsePoTNum++, TOffset += AdvanceBy) {
 
         // Create PowerOfTime array for pulse detection.  If there
         // are not enough points left in this PoT, adjust TOffset
@@ -3279,9 +3279,9 @@ CPU_pulse_processing:
 
       // loop through time for each frequency.  PulsePoTNum is
       // used only for progress calculation.
-      for(TOffset = 0, /*PulsePoTNum = 1,*/ TOffsetOK = true;
+      for(TOffset = 0, PulsePoTNum = 1, TOffsetOK = true;
           TOffsetOK;
-          /*PulsePoTNum++,*/ TOffset += AdvanceBy) {
+          PulsePoTNum++, TOffset += AdvanceBy) {
 
         // Create PowerOfTime array for pulse detection.  If there
         // are not enough points left in this PoT, adjust TOffset
