--- AKv8/client/analyzeFuncs.cpp.orig	2015-12-04 15:44:15.353052817 -0800
+++ AKv8/client/analyzeFuncs.cpp	2015-12-04 15:52:46.066587532 -0800
@@ -62,13 +62,9 @@
 #include "counters.h"
 #endif
 #ifdef BOINC_APP_GRAPHICS
+#include "graphics2.h"
 #include "reduce.h"
-#include "graphics_data.h"
-#include "graphics_api.h"
 #include "sah_gfx_main.h"
-#ifdef DYNAMIC_GRAPHICS
-#include "graphics_lib.h"
-#endif
 #endif
 #include "diagnostics.h"
 
