--- setiathome-gpu-8.00.orig/AKv8/client/main.cpp	2016-01-02 01:58:49.216959827 -0800
+++ setiathome-gpu-8.00/AKv8/client/main.cpp	2016-01-02 02:34:24.553712383 -0800
@@ -104,12 +104,14 @@
   printf(
     "options:\n"
     " -help  show options\n"
-//#ifdef BOINC_APP_GRAPHICS
+#ifdef BOINC_APP_GRAPHICS
     " -nographics  run without graphics\n"
-//#endif
+    " -standalone (implies -nographics)\n"
+#else
+    " -standalone \n"
+#endif
     " -version  show version info\n"
     " -verbose  print running status\n"
-    " -standalone \n"
 #if (USE_OPENCL || USE_CUDA)
 	" -period_iterations_num <N>  splits single longest PulseFind kernes call on N calls\n"
 	" -sbs <N>  single memory buffer size in MB (can't be less than 64M or more than device capability)\n"
@@ -203,16 +205,7 @@
   setbuf(stdout, 0);
 	//fprintf(debug,"after setbuf()\n");fflush(debug);
   bool standalone = false;
-#ifdef BOINC_APP_GRAPHICS
-  nographics_flag=false;
-
-//  if (atexit(atexit_handler)) {
-//    exit(ATEXIT_FAILURE);
-//  }
 
-#else
-  nographics_flag=true;
-#endif
 
   g_argv[0]=argv[0];
 //  for(int k=0;k<argc;k++)
@@ -231,10 +224,13 @@
       verbose = 1;
     } else if (!strncmp(p, "st", 2)) {
         standalone = true;
-//#ifdef BOINC_APP_GRAPHICS
-    } else if (!strncmp(p, "no", 2)) {
+#ifdef BOINC_APP_GRAPHICS
+        nographics_flag = true;
+#endif
+    } else if (!strncmp(p, "nog", 2)) {
+#ifdef BOINC_APP_GRAPHICS
         nographics_flag = true;
-//#endif
+#endif      
     }
 /*#if __linux__ || __APPLE__
 	else
