--- a/sah_v7_opt/src/cl_cutted.hpp.orig	2019-03-17 23:52:11.707584614 -0700
+++ b/sah_v7_opt/src/cl_cutted.hpp	2019-03-18 00:07:20.719000362 -0700
@@ -188,6 +188,8 @@
 
 #include <cstring>
 
+#include "cl_boinc.h"
+
 /*! \namespace cl
  *
  * \brief The OpenCL C++ bindings are defined within this namespace.
--- a/src/cl_cutted.hpp.orig	2019-03-17 23:51:52.188848029 -0700
+++ b/src/cl_cutted.hpp	2019-03-18 00:08:23.405381170 -0700
@@ -188,6 +188,8 @@
 
 #include <cstring>
 
+#include "cl_boinc.h"
+
 /*! \namespace cl
  *
  * \brief The OpenCL C++ bindings are defined within this namespace.
