diff -urp setiathome-gpu-8.22.orig/sah_v7_opt/src/GPU_lock.cpp setiathome-gpu-8.22/sah_v7_opt/src/GPU_lock.cpp
--- setiathome-gpu-8.22.orig/sah_v7_opt/src/GPU_lock.cpp	2017-01-19 20:20:33.088785951 -0800
+++ setiathome-gpu-8.22/sah_v7_opt/src/GPU_lock.cpp	2017-01-19 20:24:26.706773991 -0800
@@ -663,7 +663,7 @@ convertToString(const char *filename)
 		{
 			f.close();
 			fprintf(stderr,"ERROR: can't allocate mem for CL file, exiting...\n");
-			boinc_temporary_exit(300,"Can't read CL file",true);
+			boinc_temporary_exit(300,"Can't read CL file");
 		}
 #undef read
 		f.read(str, fileSize);
@@ -675,7 +675,7 @@ convertToString(const char *filename)
 		return s;
 	}
 	fprintf(stderr,"ERROR: can't open CL file, exiting...\n");
-	boinc_temporary_exit(300,"Can't read CL file",true);
+	boinc_temporary_exit(300,"Can't read CL file");
 
 }
 
@@ -941,7 +941,7 @@ FROMSRC:
                 fprintf(stderr, "CL file build log on device %s\n", devicename);
                 fprintf(stderr, "%s\n", build_log);
                 free(build_log);
-				boinc_temporary_exit(180,"CL file build failure",true);
+				boinc_temporary_exit(180,"CL file build failure");
 	}
 
 	//R: extract binary and save it to file
diff -urp setiathome-gpu-8.22.orig/src/GPU_lock.cpp setiathome-gpu-8.22/src/GPU_lock.cpp
--- setiathome-gpu-8.22.orig/src/GPU_lock.cpp	2017-01-19 20:20:32.380758709 -0800
+++ setiathome-gpu-8.22/src/GPU_lock.cpp	2017-01-19 20:24:28.995862054 -0800
@@ -663,7 +663,7 @@ convertToString(const char *filename)
 		{
 			f.close();
 			fprintf(stderr,"ERROR: can't allocate mem for CL file, exiting...\n");
-			boinc_temporary_exit(300,"Can't read CL file",true);
+			boinc_temporary_exit(300,"Can't read CL file");
 		}
 #undef read
 		f.read(str, fileSize);
@@ -675,7 +675,7 @@ convertToString(const char *filename)
 		return s;
 	}
 	fprintf(stderr,"ERROR: can't open CL file, exiting...\n");
-	boinc_temporary_exit(300,"Can't read CL file",true);
+	boinc_temporary_exit(300,"Can't read CL file");
 
 }
 
@@ -941,7 +941,7 @@ FROMSRC:
                 fprintf(stderr, "CL file build log on device %s\n", devicename);
                 fprintf(stderr, "%s\n", build_log);
                 free(build_log);
-				boinc_temporary_exit(180,"CL file build failure",true);
+				boinc_temporary_exit(180,"CL file build failure");
 	}
 
 	//R: extract binary and save it to file
