--- AP/client/ap_client_main.cpp.orig	2015-12-03 19:00:55.172188889 -0800
+++ AP/client/ap_client_main.cpp	2015-12-03 19:26:14.677864874 -0800
@@ -561,7 +561,7 @@
 #endif
 }
 #if _GNU_SOURCE || __APPLE__ || __MINGW32__ || __FreeBSD__
-  ap_shmem_init(app_init_data); //uje: Renamed ap_graphics_init(app_init_data) to avoid confusions with stock code
+  ap_graphics_init(app_init_data); //uje: Renamed ap_graphics_init(app_init_data) to avoid confusions with stock code
 #else // !_GNU_SOURCE
   // We've moved the state variable into the graphics shmem segment,
   // so we always need to initialize graphics.
