--- AP/client/Makefile.am.orig	2015-12-03 11:17:43.592442430 -0800
+++ AP/client/Makefile.am	2015-12-03 11:20:16.229209551 -0800
@@ -60,20 +60,20 @@
 CLIENT_LD_ADD += $(top_srcdir)./../lib/linux32/libfftw3f.a
 endif
 
-##if ENABLE_GUI
-##     CLIENT_BIN = $(CLIENT_PROG) $(CLIENT_NATIVE) $(DEBUG_NATIVE) $(DEBUG_PROG)
-##     GUI_BIN = ap_graphics
-##     GUI_C_FLAGS = $(CLIENT_C_FLAGS) $(GRAPHICS_CFLAGS)
-##     GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
-##     CLIENT_LD_FLAGS += $(LDSTATIC)
-##     CLIENT_LD_ADD += -L$(BOINCDIR)/api -L$(BOINCDIR)/lib $(BOINC_LIBS) $(AM_LIBS)
-##     AP_GRX_SOURCES = ap_gfx_main.cpp
-##else
+if ENABLE_GUI
+     CLIENT_BIN = $(CLIENT_PROG) $(CLIENT_NATIVE) $(DEBUG_NATIVE) $(DEBUG_PROG)
+     GUI_BIN = ap_graphics
+     GUI_C_FLAGS = $(CLIENT_C_FLAGS) $(GRAPHICS_CFLAGS)
+     GUI_LD_ADD = $(GRAPHICS_LIBS_RAW)
+     CLIENT_LD_FLAGS += $(LDSTATIC)
+     CLIENT_LD_ADD += -L$(BOINCDIR)/api -L$(BOINCDIR)/lib $(BOINC_LIBS) $(AM_LIBS)
+     AP_GRX_SOURCES = ap_gfx_main.cpp
+else
 #no graphics
-##   AP_GRX_SOURCES =
+     AP_GRX_SOURCES =
      CLIENT_LD_ADD += $(AM_LIBS)
      CLIENT_BIN = $(CLIENT_PROG) $(DEBUG_PROG)
-##endif
+endif
 
 all: client-bin
 
@@ -123,7 +123,8 @@
 	ap_graphics.h					\
 	sbtf.h						\
         ap_version.h                                    \
-	ap_shmem.h
+	ap_shmem.h                                      \
+	$(AP_GRX_SOURCES)
 
 if OPENCL
 ap_client_opencl_SOURCES = \
@@ -139,7 +140,6 @@
 	../../src/OpenCL_FFT/fft_setup.cpp		\
 	../../src/OpenCL_FFT/procs.h
 endif
-##	$(AP_GRX_SOURCES)
 
 ap_client_SOURCES = $(ap_client_cpu_SOURCES) $(ap_client_opencl_SOURCES)
 
@@ -149,29 +149,29 @@
 ap_client_LDADD = $(CLIENT_LD_ADD)
 ap_client_LINK = $(CXX) $(ap_client_CXXFLAGS) $(ap_client_LDFLAGS) -o $@
 
-##if ENABLE_GUI
-##noinst_PROGRAMS += ap_graphics
-
-##ap_graphics_SOURCES = \
-##    graphics_main.cpp \
-##    ap_gfx_base.cpp \
-##    ap_graphics.cpp
-
-##ap_graphics_CFLAGS = $(GUI_C_FLAGS)
-##ap_graphics_CXXFLAGS = $(GUI_C_FLAGS)
-##ap_graphics_LDFLAGS = $(GUI_LD_FLAGS)
-##ap_graphics_LDADD = $(GUI_LD_ADD) $(BOINC_LIBS)
+if ENABLE_GUI
+noinst_PROGRAMS += ap_graphics
 
-##$(DEBUG_NATIVE): $(GUI_BUILD)
-##	@RM@ -f $(DEBUG_NATIVE)
-##	@LN@ ap_graphics $(DEBUG_NATIVE)
+ap_graphics_SOURCES = \
+    graphics_main.cpp \
+    ap_gfx_base.cpp \
+    ap_graphics.cpp
+
+ap_graphics_CFLAGS = $(GUI_C_FLAGS)
+ap_graphics_CXXFLAGS = $(GUI_C_FLAGS)
+ap_graphics_LDFLAGS = $(GUI_LD_FLAGS)
+ap_graphics_LDADD = $(GUI_LD_ADD) $(BOINC_LIBS)
+
+$(DEBUG_NATIVE): $(GUI_BUILD)
+	@RM@ -f $(DEBUG_NATIVE)
+	@LN@ ap_graphics $(DEBUG_NATIVE)
+
+$(CLIENT_NATIVE): $(DEBUG_NATIVE)
+	@RM@ -f $(CLIENT_NATIVE)
+	@CP@ $(DEBUG_NATIVE) $(CLIENT_NATIVE)
 
-##$(CLIENT_NATIVE): $(DEBUG_NATIVE)
-##	@RM@ -f $(CLIENT_NATIVE)
-##	@CP@ $(DEBUG_NATIVE) $(CLIENT_NATIVE)
 
-
-##endif
+endif
 
 
 ##win_config.h: $(top_srcdir)/config.h
