--- AP/client/ap_gfx_main.h.orig	2015-12-02 11:42:30.421184364 -0800
+++ AP/client/ap_gfx_main.h	2015-12-02 11:42:43.717687713 -0800
@@ -54,13 +54,13 @@
 #include "boinc_api.h"
 
 extern REDUCED_ARRAY_GEN rarray;
-extern AP_GDATA* ap_graphics;
+extern AP_GDATA* ap_gdata;
 extern bool nographics_flag;
 
 inline bool nographics() {
     if (nographics_flag) return true;
-    if (!ap_graphics) return true;
-    if (!ap_graphics->countdown) return true;
+    if (!ap_gdata) return true;
+    if (!ap_gdata->countdown) return true;
     return false;
 }
 extern void ap_graphics_init(APP_INIT_DATA&);
