diff -urp taoframework-2.1.0.orig/build-debug.sh taoframework-2.1.0/build-debug.sh
--- taoframework-2.1.0.orig/build-debug.sh	2017-01-09 14:13:32.539704344 -0800
+++ taoframework-2.1.0/build-debug.sh	2017-01-09 14:19:01.309132316 -0800
@@ -1,2 +1,2 @@
 #!/bin/sh
-nant mono-2.0 package-debug
+nant mono-4.0 package-debug
diff -urp taoframework-2.1.0.orig/build.sh taoframework-2.1.0/build.sh
--- taoframework-2.1.0.orig/build.sh	2017-01-09 14:13:32.539704344 -0800
+++ taoframework-2.1.0/build.sh	2017-01-09 14:19:03.741224270 -0800
@@ -1,2 +1,2 @@
 #!/bin/sh
-nant mono-2.0 package
+nant mono-4.0 package
diff -urp taoframework-2.1.0.orig/clean.sh taoframework-2.1.0/clean.sh
--- taoframework-2.1.0.orig/clean.sh	2017-01-09 14:13:32.540704382 -0800
+++ taoframework-2.1.0/clean.sh	2017-01-09 14:19:29.580201222 -0800
@@ -4,4 +4,4 @@
 rm -rf dist
 
 # Build Solutions Using NAnt 
-nant -t:mono-2.0 -buildfile:tao.build clean
+nant -t:mono-4.0 -buildfile:tao.build clean
diff -urp taoframework-2.1.0.orig/create-installer.sh taoframework-2.1.0/create-installer.sh
--- taoframework-2.1.0.orig/create-installer.sh	2017-01-09 14:13:32.541704420 -0800
+++ taoframework-2.1.0/create-installer.sh	2017-01-09 14:19:19.276811655 -0800
@@ -1,2 +1,2 @@
 #!/bin/sh
-nant mono-2.0 create-installer
+nant mono-4.0 create-installer
diff -urp taoframework-2.1.0.orig/src/Tao.DevIl/Tao.DevIl.build taoframework-2.1.0/src/Tao.DevIl/Tao.DevIl.build
--- taoframework-2.1.0.orig/src/Tao.DevIl/Tao.DevIl.build	2017-01-09 14:13:32.890717611 -0800
+++ taoframework-2.1.0/src/Tao.DevIl/Tao.DevIl.build	2017-01-09 14:17:33.229802326 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.FFmpeg/Tao.FFmpeg.build taoframework-2.1.0/src/Tao.FFmpeg/Tao.FFmpeg.build
--- taoframework-2.1.0.orig/src/Tao.FFmpeg/Tao.FFmpeg.build	2017-01-09 14:13:32.896717837 -0800
+++ taoframework-2.1.0/src/Tao.FFmpeg/Tao.FFmpeg.build	2017-01-09 14:17:37.053946897 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.FreeType/Tao.FreeType.build taoframework-2.1.0/src/Tao.FreeType/Tao.FreeType.build
--- taoframework-2.1.0.orig/src/Tao.FreeType/Tao.FreeType.build	2017-01-09 14:13:32.906718215 -0800
+++ taoframework-2.1.0/src/Tao.FreeType/Tao.FreeType.build	2017-01-09 14:17:28.366618474 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.GlBindGen/Tao.GlBindGen.build taoframework-2.1.0/src/Tao.GlBindGen/Tao.GlBindGen.build
--- taoframework-2.1.0.orig/src/Tao.GlBindGen/Tao.GlBindGen.build	2017-01-09 14:13:32.910718366 -0800
+++ taoframework-2.1.0/src/Tao.GlBindGen/Tao.GlBindGen.build	2017-01-09 14:17:36.093910603 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.Lua/Tao.Lua.build taoframework-2.1.0/src/Tao.Lua/Tao.Lua.build
--- taoframework-2.1.0.orig/src/Tao.Lua/Tao.Lua.build	2017-01-09 14:13:32.930719122 -0800
+++ taoframework-2.1.0/src/Tao.Lua/Tao.Lua.build	2017-01-09 14:17:29.806672915 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.Ode/Tao.Ode.build taoframework-2.1.0/src/Tao.Ode/Tao.Ode.build
--- taoframework-2.1.0.orig/src/Tao.Ode/Tao.Ode.build	2017-01-09 14:13:32.935719311 -0800
+++ taoframework-2.1.0/src/Tao.Ode/Tao.Ode.build	2017-01-09 14:17:32.157761798 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.OpenGl/Tao.OpenGl.build taoframework-2.1.0/src/Tao.OpenGl/Tao.OpenGl.build
--- taoframework-2.1.0.orig/src/Tao.OpenGl/Tao.OpenGl.build	2017-01-09 14:13:32.972720710 -0800
+++ taoframework-2.1.0/src/Tao.OpenGl/Tao.OpenGl.build	2017-01-09 14:17:30.877713405 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.PhysFs/Tao.PhysFs.build taoframework-2.1.0/src/Tao.PhysFs/Tao.PhysFs.build
--- taoframework-2.1.0.orig/src/Tao.PhysFs/Tao.PhysFs.build	2017-01-09 14:13:32.976720861 -0800
+++ taoframework-2.1.0/src/Tao.PhysFs/Tao.PhysFs.build	2017-01-09 14:17:34.237840434 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/src/Tao.Sdl/Tao.Sdl.build taoframework-2.1.0/src/Tao.Sdl/Tao.Sdl.build
--- taoframework-2.1.0.orig/src/Tao.Sdl/Tao.Sdl.build	2017-01-09 14:13:32.997721655 -0800
+++ taoframework-2.1.0/src/Tao.Sdl/Tao.Sdl.build	2017-01-09 14:17:35.181876124 -0800
@@ -27,6 +27,18 @@
         <property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
         <property name="nant.settings.currentframework" value="mono-2.0" />
     </target>
diff -urp taoframework-2.1.0.orig/tao.build taoframework-2.1.0/tao.build
--- taoframework-2.1.0.orig/tao.build	2017-01-09 14:13:32.542704458 -0800
+++ taoframework-2.1.0/tao.build	2017-01-09 14:17:25.343504187 -0800
@@ -29,6 +29,18 @@
 	<property name="nant.settings.currentframework" value="net-2.0" />
     </target>
 
+    <target name="mono-4.5" description="Sets framework to mono 4.5">
+        <property name="nant.settings.currentframework" value="mono-4.5" />
+    </target>
+
+    <target name="mono-4.0" description="Sets framework to mono 4.0">
+        <property name="nant.settings.currentframework" value="mono-4.0" />
+    </target>
+
+    <target name="mono-3.0" description="Sets framework to mono 3.0">
+        <property name="nant.settings.currentframework" value="mono-3.0" />
+    </target>
+
     <target name="mono-2.0" description="Sets framework to mono 2.0">
 	<call target="check-osx" />
 	<property name="nant.settings.currentframework" value="mono-2.0" />
diff -urp taoframework-2.1.0.orig/taoframework.spec.suse.example taoframework-2.1.0/taoframework.spec.suse.example
--- taoframework-2.1.0.orig/taoframework.spec.suse.example	2017-01-09 14:13:32.542704458 -0800
+++ taoframework-2.1.0/taoframework.spec.suse.example	2017-01-09 14:20:25.611319811 -0800
@@ -108,10 +108,10 @@ This package contains SDL Bindings
 %setup -c  
   
 %build  
-nant mono-2.0 build-release  
+nant mono-4.0 build-release  
   
 %install  
-nant mono-2.0 install -D:gacdir="$RPM_BUILD_ROOT%{_prefix}" -D:pkgconfig="$RPM_BUILD_ROOT%{_prefix}/lib/pkgconfig"  
+nant mono-4.0 install -D:gacdir="$RPM_BUILD_ROOT%{_prefix}" -D:pkgconfig="$RPM_BUILD_ROOT%{_prefix}/lib/pkgconfig"  
   
 %clean  
 rm -rf "$RPM_BUILD_ROOT"  
@@ -171,4 +171,4 @@ rm -rf "$RPM_BUILD_ROOT"
 %defattr(-,root,root)  
 %{_prefix}/lib/mono/tao-sdl/Tao.Sdl.dll  
 %{_prefix}/lib/mono/gac/Tao.Sdl  
-%{_prefix}/lib/pkgconfig/tao-sdl.pc  
\ No newline at end of file
+%{_prefix}/lib/pkgconfig/tao-sdl.pc  
