diff -ur knet-9999.orig/src/NetworkServer.cpp knet-9999/src/NetworkServer.cpp
--- knet-9999.orig/src/NetworkServer.cpp	2016-03-07 08:52:14.139390423 -0800
+++ knet-9999/src/NetworkServer.cpp	2016-03-07 10:26:48.807745932 -0800
@@ -85,7 +85,10 @@
 		if (listenSockets[i]->TransportLayer() == SocketOverUDP)
 			acceptNewConnections = false; ///\todo At this point, if in UDP mode, we should have destroyed all connections that use this socket!
 		else
-			owner->DeleteSocket(listenSockets[i]); 
+                {
+			owner->DeleteSocket(listenSockets[i]);
+                        listenSockets[i] = 0;
+                }
 	}
 
 	// Now forget all sockets - not getting them back in any way.
@@ -167,6 +170,8 @@
 {
 	CleanupDeadConnections();
 
+	LOG(LogError, "Number of listen sockets: %d", listenSockets.size());
+
 	for(size_t i = 0; i < listenSockets.size(); ++i)
 	{
 		Socket *listen = listenSockets[i];
