diff -urp shell-0.9.0_p9999.orig/src/compositor/CMakeLists.txt shell-0.9.0_p9999/src/compositor/CMakeLists.txt
--- shell-0.9.0_p9999.orig/src/compositor/CMakeLists.txt	2022-12-12 22:31:41.972989537 -0800
+++ shell-0.9.0_p9999/src/compositor/CMakeLists.txt	2022-12-12 22:33:13.632594265 -0800
@@ -3,6 +3,9 @@ check_include_file("sys/prctl.h" HAVE_SY
 
 if(FEATURE_enable_systemd)
     find_package(Libsystemd REQUIRED)
+    set(SYSTEMD_LIBS PkgConfig::Libsystemd)
+else()
+    set(SYSTEMD_LIBS "")
 endif()
 
 if(NOT TARGET Liri::Qt5GSettings)
@@ -78,7 +81,10 @@ liri_extend_target("LiriShell"
 
 liri_extend_target("LiriShell"
     CONDITION FEATURE_enable_systemd
-    DEFINES HAVE_SYSTEMD
-    LIBRARIES PkgConfig::Libsystemd)
+    LIBRARIES ${SYSTEMD_LIBS})
+
+if(FEATURE_enable_systemd)
+    target_compile_definitions("LiriShell" PUBLIC HAVE_SYSTEMD)
+endif()
 
 liri_finalize_executable("LiriShell")
Only in shell-0.9.0_p9999/src/compositor: CMakeLists.txt.orig
diff -urp shell-0.9.0_p9999.orig/src/helper/CMakeLists.txt shell-0.9.0_p9999/src/helper/CMakeLists.txt
--- shell-0.9.0_p9999.orig/src/helper/CMakeLists.txt	2022-12-12 22:31:41.968989380 -0800
+++ shell-0.9.0_p9999/src/helper/CMakeLists.txt	2022-12-12 22:33:13.636594422 -0800
@@ -3,6 +3,9 @@ check_include_file("sys/prctl.h" HAVE_SY
 
 if(FEATURE_enable_systemd)
     find_package(Libsystemd REQUIRED)
+    set(SYSTEMD_LIBS PkgConfig::Libsystemd)
+else()
+    set(SYSTEMD_LIBS "")
 endif()
 
 if(NOT TARGET Liri::WaylandClient)
@@ -49,7 +52,10 @@ liri_extend_target("LiriShellHelper"
 
 liri_extend_target("LiriShellHelper"
     CONDITION FEATURE_enable_systemd
-    DEFINES HAVE_SYSTEMD
-    LIBRARIES PkgConfig::Libsystemd)
+    LIBRARIES ${SYSTEMD_LIBS})
+
+if(FEATURE_enable_systemd)
+    target_compile_definitions("LiriShellHelper" PUBLIC HAVE_SYSTEMD)
+endif()
 
 liri_finalize_executable("LiriShellHelper")
Only in shell-0.9.0_p9999/src/helper: CMakeLists.txt.orig
