diff -urp shell-0.9.0_p9999.orig/src/compositor/qml/windows/XWaylandWindow.qml shell-0.9.0_p9999/src/compositor/qml/windows/XWaylandWindow.qml
--- shell-0.9.0_p9999.orig/src/compositor/qml/windows/XWaylandWindow.qml	2022-12-12 23:45:38.447424459 -0800
+++ shell-0.9.0_p9999/src/compositor/qml/windows/XWaylandWindow.qml	2022-12-12 23:46:04.240436284 -0800
@@ -30,15 +30,19 @@ FluidCore.Object {
     readonly property alias fullscreen: __private.fullscreen
     readonly property alias resizing: __private.resizing
     readonly property bool decorated: {
-        // Set true if it has a titlebar and decoration
-        if (__private.windowType !== Qt.Popup)
+        if (toplevelHandle && toplevelHandle.appId == "")
+            return __private.decorate && !__private.fullscreen;
+        else if (__private.windowType !== Qt.Popup)
+            // Set true if it has a titlebar and decoration
             return __private.decorate && !__private.fullscreen;
         else
             return false;
     }
     readonly property bool bordered: {
-        // Set true if it has resize handles
-        if (decorated)
+        if (toplevelHandle && toplevelHandle.appId == "")
+            return !__private.maximized && !__private.fullscreen;
+        else if (decorated)
+            // Set true if it has resize handles
             return !__private.maximized && !__private.fullscreen;
         else
             return false;
diff -urp shell-0.9.0_p9999.orig/src/compositor/qml/windows/XdgToplevelWindow.qml shell-0.9.0_p9999/src/compositor/qml/windows/XdgToplevelWindow.qml
--- shell-0.9.0_p9999.orig/src/compositor/qml/windows/XdgToplevelWindow.qml	2022-12-12 23:45:38.447424459 -0800
+++ shell-0.9.0_p9999/src/compositor/qml/windows/XdgToplevelWindow.qml	2022-12-12 23:48:17.165650577 -0800
@@ -30,15 +30,19 @@ FluidCore.Object {
     readonly property alias fullscreen: __private.fullscreen
     readonly property alias resizing: __private.resizing
     readonly property bool decorated: {
-        // Set true if it has a titlebar and decorations
-        if (__private.decorationMode === QtWayland.XdgToplevel.ServerSideDecoration)
+        if (toplevel && toplevel.appId == "")
+            return !__private.fullscreen;
+        else if (__private.decorationMode === QtWayland.XdgToplevel.ServerSideDecoration)
+            // Set true if it has a titlebar and decorations
             return !__private.fullscreen;
         else
             return false;
     }
     readonly property bool bordered: {
-        // Set true if it has resize handles
-        if (__private.decorationMode === QtWayland.XdgToplevel.ServerSideDecoration)
+        if (toplevel && toplevel.appId == "")
+            return !__private.maximized && !__private.fullscreen;
+        else if (__private.decorationMode === QtWayland.XdgToplevel.ServerSideDecoration)
+            // Set true if it has resize handles
             return !__private.maximized && !__private.fullscreen;
         else
             return false;
@@ -47,7 +51,9 @@ FluidCore.Object {
     readonly property real titleBarHeight: decorated ? 32 : 0
     readonly property bool focusable: true
     readonly property bool maximizable: {
-        if (__private.minSize.width <= 0 && __private.minSize.height <= 0 && __private.maxSize.width <= 0 && __private.maxSize.height <= 0)
+        if (toplevel && toplevel.appId == "")
+            return true;
+        else if (__private.minSize.width <= 0 && __private.minSize.height <= 0 && __private.maxSize.width <= 0 && __private.maxSize.height <= 0)
             return true;
         return __private.minSize !== __private.maxSize;
     }
