diff -urp shell-0.9.0_p9999.orig/src/compositor/CMakeLists.txt shell-0.9.0_p9999/src/compositor/CMakeLists.txt
--- shell-0.9.0_p9999.orig/src/compositor/CMakeLists.txt	2022-11-30 21:50:23.817058589 -0800
+++ shell-0.9.0_p9999/src/compositor/CMakeLists.txt	2022-11-30 21:55:41.921512361 -0800
@@ -3,6 +3,9 @@ check_include_file("sys/prctl.h" HAVE_SY
 
 if(FEATURE_enable_systemd)
     find_package(Libsystemd REQUIRED)
+    set(SYSTEMD_LIBS PkgConfig::Libsystemd)
+else()
+    set(SYSTEMD_LIBS "")
 endif()
 
 if(NOT TARGET Liri::Qt5GSettings)
@@ -77,7 +80,10 @@ liri_extend_target("LiriShell"
 
 liri_extend_target("LiriShell"
     CONDITION FEATURE_enable_systemd
-    DEFINES HAVE_SYSTEMD
-    LIBRARIES PkgConfig::Libsystemd)
+    LIBRARIES ${SYSTEMD_LIBS})
+
+if(FEATURE_enable_systemd)
+    target_compile_definitions("LiriShell" PUBLIC HAVE_SYSTEMD)
+endif()
 
 liri_finalize_executable("LiriShell")
diff -urp shell-0.9.0_p9999.orig/src/helper/CMakeLists.txt shell-0.9.0_p9999/src/helper/CMakeLists.txt
--- shell-0.9.0_p9999.orig/src/helper/CMakeLists.txt	2022-11-30 21:50:23.841059529 -0800
+++ shell-0.9.0_p9999/src/helper/CMakeLists.txt	2022-11-30 21:55:39.969435928 -0800
@@ -3,6 +3,9 @@ check_include_file("sys/prctl.h" HAVE_SY
 
 if(FEATURE_enable_systemd)
     find_package(Libsystemd REQUIRED)
+    set(SYSTEMD_LIBS PkgConfig::Libsystemd)
+else()
+    set(SYSTEMD_LIBS "")
 endif()
 
 if(NOT TARGET Liri::AuroraClient)
@@ -47,7 +50,10 @@ liri_extend_target("LiriShellHelper"
 
 liri_extend_target("LiriShellHelper"
     CONDITION FEATURE_enable_systemd
-    DEFINES HAVE_SYSTEMD
-    LIBRARIES PkgConfig::Libsystemd)
+    LIBRARIES ${SYSTEMD_LIBS})
+
+if(FEATURE_enable_systemd)
+    target_compile_definitions("LiriShellHelper" PUBLIC HAVE_SYSTEMD)
+endif()
 
 liri_finalize_executable("LiriShellHelper")
diff -urp shell-0.9.0_p9999.orig/src/lockscreen/CMakeLists.txt shell-0.9.0_p9999/src/lockscreen/CMakeLists.txt
--- shell-0.9.0_p9999.orig/src/lockscreen/CMakeLists.txt	2022-11-30 21:50:23.897061721 -0800
+++ shell-0.9.0_p9999/src/lockscreen/CMakeLists.txt	2022-11-30 21:55:40.961474771 -0800
@@ -3,6 +3,9 @@ check_include_file("sys/prctl.h" HAVE_SY
 
 if(FEATURE_enable_systemd)
     find_package(Libsystemd REQUIRED)
+    set(SYSTEMD_LIBS PkgConfig::Libsystemd)
+else()
+    set(SYSTEMD_LIBS "")
 endif()
 
 if(NOT TARGET Liri::AuroraClient)
@@ -48,7 +51,10 @@ liri_extend_target("LiriShellLockscreen"
 
 liri_extend_target("LiriShellLockscreen"
     CONDITION FEATURE_enable_systemd
-    DEFINES HAVE_SYSTEMD
-    LIBRARIES PkgConfig::Libsystemd)
+    LIBRARIES ${SYSTEMD_LIBS})
+
+if(FEATURE_enable_systemd)
+    target_compile_definitions("LiriShellLockscreen" PUBLIC HAVE_SYSTEMD)
+endif()
 
 liri_finalize_executable("LiriShellLockscreen")
