--- a/src/components/messageInput.tsx.orig	2025-08-10 09:07:55.099254256 -0700
+++ b/src/components/messageInput.tsx	2025-08-10 09:08:43.083442214 -0700
@@ -63,7 +63,8 @@ export default function MessageInput({
             // both are 16000
             const audioCtx = new AudioContext();
             const buffer = audioCtx.createBuffer(1, audio.length, 16000);
-            buffer.copyToChannel(audio, 0, 0);
+            const audioArrayBuffer = new Float32Array(audio.buffer);
+            buffer.copyToChannel(audioArrayBuffer, 0, 0);
             transcriber.start(buffer);
             break;
           }
