--- a/next.config.js.orig	2025-02-04 13:05:45.000000000 -0800
+++ b/next.config.js	2025-02-08 08:13:18.416206723 -0800
@@ -16,6 +16,7 @@ const nextConfig = {
     root: process.env.BASE_PATH || "",
   },
   optimizeFonts: false,
+  productionBrowserSourceMaps: false,
   webpack: (config, { webpack, buildId }) => {
     // See https://webpack.js.org/configuration/resolve/#resolvealias
     config.resolve.alias = {
@@ -28,19 +29,11 @@ const nextConfig = {
       new CopyPlugin({
         patterns: [
           {
-            from: "./node_modules/onnxruntime-web/dist/ort-wasm-simd-threaded.wasm",
-            to: "static/chunks/[name][ext]",
-          },
-          {
-            from: "./node_modules/onnxruntime-web/dist/ort-wasm-threaded.wasm",
+            from: "./node_modules/onnxruntime-web/dist/ort-wasm-simd-threaded.jsep.wasm",
             to: "static/chunks/[name][ext]",
           },
           {
-            from: "./node_modules/onnxruntime-web/dist/ort-wasm.wasm",
-            to: "static/chunks/[name][ext]",
-          },
-          {
-            from: "./node_modules/onnxruntime-web/dist/ort-wasm-simd.wasm",
+            from: "./node_modules/onnxruntime-web/dist/ort-wasm-simd-threaded.wasm",
             to: "static/chunks/[name][ext]",
           },
           {
