--- a/third_party/FBGEMM/fbgemm_gpu/CMakeLists.txt.orig	2024-11-20 19:57:18.708316698 -0800
+++ b/third_party/FBGEMM/fbgemm_gpu/CMakeLists.txt	2024-11-20 20:14:55.869383241 -0800
@@ -29,6 +29,8 @@ set(THIRDPARTY ${FBGEMM}/third_party)
 ################################################################################
 
 option(FBGEMM_CPU_ONLY  "Build FBGEMM_GPU without GPU support" OFF)
+option(USE_AVX2         "Use avx2?" ON)
+option(USE_AVX512       "Use avx512?" ON)
 option(USE_ROCM         "Build FBGEMM_GPU for ROCm" OFF)
 
 if(((EXISTS "/opt/rocm-@ROCM_VERSION@/") OR (EXISTS $ENV{ROCM_PATH}))
@@ -444,7 +446,7 @@ if(NOT FBGEMM_CPU_ONLY AND WSL_MODE)
   set(AVX512_FLAGS "-Xcompiler;-mavx2;-Xcompiler;-mf16c;-Xcompiler;-mfma;-Xcompiler;-mavx512f;-Xcompiler;-mavx512bw;-Xcompiler;-mavx512dq;-Xcompiler;-mavx512vl;-fopenmp")
 endif()
 
-if(CXX_AVX2_FOUND)
+if(CXX_AVX2_FOUND AND USE_AVX2)
   set_source_files_properties(${gen_cpu_source_files}
     PROPERTIES COMPILE_OPTIONS
     "${AVX2_FLAGS}")
@@ -513,25 +515,25 @@ set(fbgemm_sources_avx2
 set(fbgemm_sources_avx512
   "${FBGEMM}/src/EmbeddingSpMDMAvx512.cc")
 
-if(CXX_AVX2_FOUND)
+if(CXX_AVX2_FOUND AND USE_AVX2)
   set_source_files_properties(${fbgemm_sources_avx2}
     PROPERTIES COMPILE_OPTIONS
     "${AVX2_FLAGS}")
 endif()
 
-if(CXX_AVX512_FOUND)
+if(CXX_AVX512_FOUND AND USE_AVX512)
   set_source_files_properties(${fbgemm_sources_avx512}
     PROPERTIES COMPILE_OPTIONS
     "${AVX512_FLAGS}")
 endif()
 
 set(fbgemm_sources ${fbgemm_sources_normal})
-if(CXX_AVX2_FOUND)
+if(CXX_AVX2_FOUND AND USE_AVX2)
   set(fbgemm_sources
     ${fbgemm_sources}
     ${fbgemm_sources_avx2})
 endif()
-if(NOT USE_ROCM AND CXX_AVX512_FOUND)
+if(NOT USE_ROCM AND CXX_AVX2_FOUND AND USE_AVX2 AND CXX_AVX512_FOUND AND USE_AVX512)
   set(fbgemm_sources
     ${fbgemm_sources}
     ${fbgemm_sources_avx2}
@@ -627,7 +629,7 @@ if(NOT FBGEMM_CPU_ONLY)
   endif()
 endif()
 
-if(CXX_AVX2_FOUND)
+if(CXX_AVX2_FOUND AND USE_AVX2)
   set_source_files_properties(${fbgemm_gpu_sources_static_cpu}
     PROPERTIES COMPILE_OPTIONS
     "${AVX2_FLAGS}")
