diff -urp gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21.orig/ycmd.cpp gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21/ycmd.cpp
--- gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21.orig/ycmd.cpp	2017-01-16 22:39:07.352874909 -0800
+++ gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21/ycmd.cpp	2017-01-16 22:42:41.971693254 -0800
@@ -105,7 +105,8 @@ bool Ycmd::startServer(){
 	gchar* cwd = g_get_current_dir();
 	gchar py[] = "python3.4";
 	gchar iss[] = "--idle_suicide_seconds=10800";
-	gchar * args[6] = { py, NULL, NULL, NULL, iss, NULL }; /* python; ycmd path; port, config; iss */ // TODO: Add log-level option
+	gchar klf[] = "--keep_logfiles";
+	gchar * args[7] = { py, NULL, NULL, NULL, iss, klf, NULL }; /* python; ycmd path; port, config; iss */ // TODO: Add log-level option
 	// ycmd path
 	char * expanded_path = realpath(ycmdsettings["ycmd_path"].asString().c_str(),NULL);
 	if(!expanded_path){
