diff -urp gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21.orig/ycmd.cpp gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21/ycmd.cpp
--- gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21.orig/ycmd.cpp	2014-12-16 23:08:20.000000000 -0800
+++ gycm-1963a9f6b51a3aff7c44780dcb13d056a8659b21/ycmd.cpp	2017-01-16 02:31:17.493626667 -0800
@@ -224,7 +224,7 @@ int Ycmd::handler(const char * buf, size
 		return 0;
 	
 	if(returned.isMember("exception")){
-		msgwin_status_add("[ycmd] %s: %s", returned["exception"]["TYPE"].asCString(), returned["message"].asCString());
+		//msgwin_status_add("[ycmd] %s: %s", returned["exception"]["TYPE"].asCString(), returned["message"].asCString());
 		#ifndef NDEBUG
 		std::cout << returned.toStyledString();
 		#endif
