--- AKv8/client/spike.cpp.orig	2015-12-04 17:24:20.438043968 -0800
+++ AKv8/client/spike.cpp	2015-12-04 17:24:37.493709338 -0800
@@ -72,6 +72,9 @@
 #include "spike.h"
 //#include <Accelerate/Accelerate.h>
 
+#include "gdata.h"
+extern GDATA *sah_graphics;
+
 #if defined(__VEC__) || defined(__SSE3__)
 //#define ZERO		((vFloat) {0.0f, 0.0f, 0.0f, 0.0f})
 //#define IZERO		((vUInt32) {0, 0, 0, 0})
