--- AKv8/client/Makefile.am.orig	2015-11-30 01:48:24.691325436 -0800
+++ AKv8/client/Makefile.am	2015-11-30 01:48:53.059414684 -0800
@@ -13,7 +13,7 @@
 LIBS += $(BOINCDIR)/api/.libs/libboinc_api.a $(BOINCDIR)/lib/.libs/libboinc.a
 endif
 
-AM_MAKEFLAGS = -s
+#AM_MAKEFLAGS = -s
 
 CLIENT_CXX_FLAGS = $(APP_CXXFLAGS) $(CFLAGS) $(DEFS) \
 		-DTEXT_UI -DNDEBUG -DCLIENT \
@@ -125,7 +125,7 @@
 $(DEBUG_PROG): seti_boinc
 	@RM@ -f $(DEBUG_PROG)
 	@LN@ seti_boinc $(DEBUG_PROG)
-	@ECHO@ ***** FINSHED BUILDING DEBUG CLIENT PROGRAM *****
+	@echo ***** FINSHED BUILDING DEBUG CLIENT PROGRAM *****
 
 $(CLIENT_PROG): seti_boinc
 	@CP@ seti_boinc $(CLIENT_PROG)
@@ -133,7 +133,7 @@
 if OPENCL
 	@CP@ MultiBeam_Kernels.cl MultiBeam_Kernels_r$(SVN_REV_NUM).cl
 endif
-	@ECHO@ ***** FINSHED BUILDING CLIENT PROGRAM *****
+	@echo ***** FINSHED BUILDING CLIENT PROGRAM *****
 
 noinst_PROGRAMS = seti_boinc
 
