--- AKv8/client/main.cpp.orig	2015-12-04 15:42:37.386306253 -0800
+++ AKv8/client/main.cpp	2015-12-04 15:55:10.845126260 -0800
@@ -64,27 +64,10 @@
 #include "filesys.h"
 
 #ifdef BOINC_APP_GRAPHICS
+#include "graphics2.h"
 #include "sah_gfx_main.h"
-//#include "graphics2.h"
-#endif
-
-//#ifdef BOINC_APP_GRAPHICS
-//#include "reduce.h"
-//#include "graphics_data.h"
-//#include "graphics_api.h"
-//#ifdef DYNAMIC_GRAPHICS
-//#include "graphics_lib.h"
-//#endif
-//#endif
-
-//#ifdef BOINC_APP_GRAPHICS
-//#include "graphics_api.h"
-//#ifdef DYNAMIC_GRAPHICS
-//#include "graphics_lib.h"
-//#endif
-//#include "graphics_data.h"
 #include "sah_gfx.h"
-//#endif
+#endif
 
 #include "util.h"
 #include "s_util.h"
