--- AKv8/client/sah_gfx_base.cpp.orig	2015-12-04 15:46:59.445328717 -0800
+++ AKv8/client/sah_gfx_base.cpp	2015-12-04 15:55:17.052363737 -0800
@@ -60,11 +60,12 @@
 #include "db/schema_master.h"
 #include "boinc_gl.h"
 #include "parse.h"
-#include "graphics_api.h"
+#ifdef BOINC_APP_GRAPHICS
+#include "graphics2.h"
+#endif
 #include "sah_gfx_base.h"
 #include "gutil.h"
 #include "reduce.h"
-#include "graphics_data.h"
 
 #define PI2 (2*3.1415926)
 
