diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AKv8/Makefile.incl setiathome-7.08/AKv8/Makefile.incl
--- setiathome-7.08.orig/AKv8/Makefile.incl	2015-12-07 10:45:11.537272908 -0800
+++ setiathome-7.08/AKv8/Makefile.incl	2015-12-07 12:56:04.556214471 -0800
@@ -32,10 +32,10 @@
 APP_LIBS = @ASMLIB_LIBS@ @APP_LIBS@
 APP_LDFLAGS = @ASMLIB_LDFLAGS@ @APP_LDFLAGS@
 APP_CFLAGS = @ASMLIB_CFLAGS@ @APP_CFLAGS@
-## GRAPHICS_CFLAGS = @GRAPHICS_CFLAGS@
+GRAPHICS_CFLAGS = @GRAPHICS_CFLAGS@
 DEFS = @DEFS@
-## GRAPHICS_LIBS = @GRAPHICS_LIBS@
-## SETILIBDIR = @SETILIBDIR@
+GRAPHICS_LIBS = @GRAPHICS_LIBS@
+SETILIBDIR = @SETILIBDIR@
 SSE_LEVEL=@SSE_LEVEL@
 SVN_REV_NUM = @SVN_REV_NUM@
 OCL_Y = @OCL_Y@
diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AKv8/client/sah_gfx.cpp setiathome-7.08/AKv8/client/sah_gfx.cpp
--- setiathome-7.08.orig/AKv8/client/sah_gfx.cpp	2015-12-07 10:45:11.555273616 -0800
+++ setiathome-7.08/AKv8/client/sah_gfx.cpp	2015-12-07 11:18:52.687496346 -0800
@@ -713,7 +713,7 @@
     sah_graphics.resize(w, h);
 }
 
-void boinc_app_mouse_move(int x, int y, bool left, bool middle, bool right) {
+void boinc_app_mouse_move(int x, int y, int left, int middle, int right) {
     if (left) {
         sah_graphics.pitch_angle += (y-mouse_y)*.1;
         sah_graphics.roll_angle += (x-mouse_x)*.1;
@@ -729,7 +729,7 @@
     }
 }
 
-void boinc_app_mouse_button(int x, int y, int which, bool is_down) {
+void boinc_app_mouse_button(int x, int y, int which, int is_down) {
     if (is_down) {
         mouse_down = true;
         mouse_x = x;
Binary files setiathome-7.08.orig/AKv8/client/seti_graphics and setiathome-7.08/AKv8/client/seti_graphics differ
Binary files setiathome-7.08.orig/AKv8/client/seti_graphics-sah_gfx.o and setiathome-7.08/AKv8/client/seti_graphics-sah_gfx.o differ
Only in setiathome-7.08/AKv8/client: stderrgfx.txt
