diff -ur -x '*.Po' -x requests -x Makefile -x Makefile.in -x config.log setiathome-7.08.orig/AKv8/client/sah_gfx_base.cpp setiathome-7.08/AKv8/client/sah_gfx_base.cpp
--- setiathome-7.08.orig/AKv8/client/sah_gfx_base.cpp	2015-12-07 15:36:18.539492495 -0800
+++ setiathome-7.08/AKv8/client/sah_gfx_base.cpp	2015-12-07 15:40:40.230270015 -0800
@@ -67,6 +67,7 @@
 #include "sah_gfx_base.h"
 #include "gutil.h"
 #include "reduce.h"
+#include "filesys.h"
 
 #define PI2 (2*3.1415926)
 
@@ -367,6 +368,7 @@
 	longjmp(jb,1);
 }
 
+#if 0
 void SAH_GRAPHICS_BASE::setup_given_prefs() {
 	char filename[256];
 	FORCE_FRAME_POINTER;
@@ -410,4 +412,32 @@
 	signal(SIGBUS,oldsig[2]);
 #endif
 }
+#endif
+
+void SAH_GRAPHICS_BASE::setup_given_prefs() {
+        char filename[256];
+
+    boinc_max_fps = max_fps;
+    boinc_max_gfx_cpu_frac = max_cpu/100.;
+    if (starfield_size) {
+        starfield.build_stars(starfield_size, starfield_speed);
+    }
+    seti_logo_texture.present = false;
+    user_logo_texture.present = false;
+    background_texture.present = false;
+
+    double fsize=0;
+    boinc_resolve_filename(SETI_LOGO_FILENAME, filename, sizeof(filename));
+    if (!file_size(filename,fsize) && (fsize>4096)) {
+        seti_logo_texture.load_image_file(filename);
+    } else {
+        fprintf(stderr,"Warning: unable to load JPEG file. File size=%d\n",(int)fsize);
+            seti_logo_texture.present=false;
+    }
+    boinc_resolve_filename(USER_LOGO_FILENAME, filename, sizeof(filename));
+    user_logo_texture.load_image_file(filename);
+    boinc_resolve_filename(BACKGROUND_FILENAME, filename, sizeof(filename));
+    background_texture.load_image_file(filename);
+}
+
 
Binary files setiathome-7.08.orig/AKv8/client/seti_graphics and setiathome-7.08/AKv8/client/seti_graphics differ
Binary files setiathome-7.08.orig/AKv8/client/seti_graphics-sah_gfx_base.o and setiathome-7.08/AKv8/client/seti_graphics-sah_gfx_base.o differ
