diff -urp opentk-2014-07-23.orig/Source/OpenTK/Input/Joystick.cs opentk-2014-07-23/Source/OpenTK/Input/Joystick.cs
--- opentk-2014-07-23.orig/Source/OpenTK/Input/Joystick.cs	2014-07-24 01:15:34.000000000 -0700
+++ opentk-2014-07-23/Source/OpenTK/Input/Joystick.cs	2017-01-03 20:02:12.902287580 -0800
@@ -79,7 +79,15 @@ namespace OpenTK.Input
             return implementation.GetState(index);
         }
 
-        internal static Guid GetGuid(int index)
+        /// <summary>
+        /// Retrieves the ID of the device connected
+        /// at the specified index.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="Guid"/> representing the id of the connected device.
+        /// </returns>
+        /// <param name="index">The zero-based index of the device to poll.</param>
+        public static Guid GetGuid(int index)
         {
             return implementation.GetGuid(index);
         }
