diff -urp amdgpu-3.8-30.orig/amd/dkms/Makefile amdgpu-3.8-30/amd/dkms/Makefile
--- amdgpu-3.8-30.orig/amd/dkms/Makefile	2020-08-12 08:24:21.000000000 -0700
+++ amdgpu-3.8-30/amd/dkms/Makefile	2020-09-24 17:41:55.665503058 -0700
@@ -43,6 +43,9 @@ RHEL_MINOR := $(shell $(call get_rhel_ve
 ifneq (,$(RHEL_MAJOR))
 OS_NAME = "rhel"
 OS_VERSION = "$(RHEL_MAJOR).$(RHEL_MINOR)"
+else ifeq (gentoo,$(shell sed -n 's/^ID=\(.*\)/\1/p' /etc/os-release | tr -d '\"'))
+OS_NAME = "gentoo"
+OS_VERSION = $(shell cat /etc/gentoo-release | cut -f5 -d' ')
 else ifneq (,$(wildcard /etc/os-release))
 OS_NAME = "$(shell sed -n 's/^ID=\(.*\)/\1/p' /etc/os-release | tr -d '\"')"
 # On CentOS/RHEL, users could have installed a kernel not distributed from RHEL
