diff -urp amdgpu-3.8-30.orig/amd/dkms/Makefile amdgpu-3.8-30/amd/dkms/Makefile
--- amdgpu-3.8-30.orig/amd/dkms/Makefile	2020-09-24 17:45:52.514828674 -0700
+++ amdgpu-3.8-30/amd/dkms/Makefile	2020-09-24 18:52:45.461128082 -0700
@@ -201,6 +201,7 @@ export CONFIG_DRM_AMDGPU_USERPTR=y
 export CONFIG_DRM_AMD_DC=y
 export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_DRM_AMD_DC_DCN1_01=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DCONFIG_DRM_TTM_DMA_PAGE_POOL
@@ -210,6 +211,7 @@ subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_0
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_01
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 
 # Trying to enable DCN2/3 with core2 optimizations will result in
