--- a/amd/dkms/post-remove.sh.orig	2020-05-28 16:10:26.000000000 -0700
+++ b/amd/dkms/post-remove.sh	2020-06-03 00:37:11.292018606 -0700
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- a/amd/dkms/pre-build.sh.orig	2020-05-28 16:10:26.000000000 -0700
+++ b/amd/dkms/pre-build.sh	2020-06-03 00:37:13.952121772 -0700
@@ -84,6 +84,6 @@ export KERNELVER
 	-include $SRCTREE/include/linux/kconfig.h" \
 	./configure)
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-3.5-32/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-3.5-32/firmware/amdgpu $FW_DIR
