--- a/Makefile.orig	2019-10-03 05:15:38.464000637 -0700
+++ b/Makefile	2019-10-03 11:53:40.548847650 -0700
@@ -192,6 +192,7 @@ export CONFIG_DRM_AMDGPU_USERPTR=y
 export CONFIG_DRM_AMD_DC=y
 export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_DRM_AMD_DC_DCN1_01=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DBUILD_AS_DKMS
@@ -201,5 +202,6 @@ subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_0
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_01
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 obj-m += scheduler/ amd/amdgpu/ ttm/ amd/amdkcl/
--- a/amd/dkms/Makefile.orig	2019-10-03 05:15:38.464000637 -0700
+++ b/amd/dkms/Makefile	2019-10-03 11:53:40.548847650 -0700
@@ -192,6 +192,7 @@ export CONFIG_DRM_AMDGPU_USERPTR=y
 export CONFIG_DRM_AMD_DC=y
 export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_DRM_AMD_DC_DCN1_01=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DBUILD_AS_DKMS
@@ -201,5 +202,6 @@ subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_0
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_01
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 obj-m += scheduler/ amd/amdgpu/ ttm/ amd/amdkcl/
