diff -urp amdgpu-4.2-21.orig/amd/dkms/post-install.sh amdgpu-4.2-21/amd/dkms/post-install.sh
--- amdgpu-4.2-21.orig/amd/dkms/post-install.sh	2021-04-26 23:34:46.000000000 -0700
+++ amdgpu-4.2-21/amd/dkms/post-install.sh	2021-05-20 16:57:11.369286344 -0700
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware/updates"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-4.2-21/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/updates"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-4.2-21/firmware/amdgpu $FW_DIR
diff -urp amdgpu-4.2-21.orig/amd/dkms/post-remove.sh amdgpu-4.2-21/amd/dkms/post-remove.sh
--- amdgpu-4.2-21.orig/amd/dkms/post-remove.sh	2021-04-26 23:34:46.000000000 -0700
+++ amdgpu-4.2-21/amd/dkms/post-remove.sh	2021-05-20 16:57:09.399208707 -0700
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware/updates"
-rm -rf $FW_DIR/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware/updates"
+#rm -rf $FW_DIR/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
