--- a/amd/dkms/post-remove.sh.orig	2020-08-17 15:57:45.000000000 -0700
+++ b/amd/dkms/post-remove.sh	2020-08-21 23:01:49.936483113 -0700
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- a/amd/dkms/pre-build.sh.orig	2020-08-17 15:57:45.000000000 -0700
+++ b/amd/dkms/pre-build.sh	2020-08-21 23:01:52.116567133 -0700
@@ -84,6 +84,6 @@ export KERNELVER
 	-include $SRCTREE/include/linux/kconfig.h" \
 	./configure)
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-3.8-30/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-3.8-30/firmware/amdgpu $FW_DIR
