diff -urp amdgpu-5.4.orig/amd/dkms/Makefile amdgpu-5.4/amd/dkms/Makefile
--- amdgpu-5.4.orig/amd/dkms/Makefile	2023-11-13 16:03:21.495856776 -0800
+++ amdgpu-5.4/amd/dkms/Makefile	2023-11-13 16:05:09.287854187 -0800
@@ -150,6 +150,7 @@ LINUXINCLUDE := \
 	$(USER_INCLUDE)
 
 export CONFIG_HSA_AMD=y
+export CONFIG_HSA_AMD_AMDKCL=y
 export CONFIG_DRM_TTM=m
 export CONFIG_DRM_TTM_DMA_PAGE_POOL=y
 export CONFIG_DRM_AMDGPU=m
@@ -162,6 +163,7 @@ export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 subdir-ccflags-y += -DCONFIG_HSA_AMD
+subdir-ccflags-y += -DCONFIG_HSA_AMD_AMDKCL
 subdir-ccflags-y += -DCONFIG_DRM_TTM_DMA_PAGE_POOL
 subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_CIK
 subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_SI
diff -urp amdgpu-5.4.orig/Makefile amdgpu-5.4/Makefile
--- amdgpu-5.4.orig/Makefile	2023-01-27 18:49:53.000000000 -0800
+++ amdgpu-5.4/Makefile	2023-11-13 16:05:10.807854150 -0800
@@ -147,6 +147,7 @@ LINUXINCLUDE := \
 	$(USER_INCLUDE)
 
 export CONFIG_HSA_AMD=y
+export CONFIG_HSA_AMD_AMDKCL=y
 export CONFIG_DRM_TTM=m
 export CONFIG_DRM_TTM_DMA_PAGE_POOL=y
 export CONFIG_DRM_AMDGPU=m
@@ -157,6 +158,7 @@ export CONFIG_DRM_AMDGPU_USERPTR=y
 export CONFIG_DRM_AMD_DC=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
+subdir-ccflags-y += -DCONFIG_HSA_AMD_AMDKCL
 subdir-ccflags-y += -DCONFIG_DRM_TTM_DMA_PAGE_POOL
 subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_CIK
 subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_SI
