--- a/amd/dkms/pre-build.sh.orig	2020-12-29 15:12:43.707538366 -0800
+++ b/amd/dkms/pre-build.sh	2020-12-29 15:13:51.002126454 -0800
@@ -97,6 +97,6 @@ export KERNELVER
 	-include $SRCTREE/include/linux/kconfig.h" \
 	./configure)
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-3.9-19/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-3.9-19/firmware/amdgpu $FW_DIR
--- a/amd/dkms/post-remove.sh.orig	2020-11-10 22:26:09.000000000 -0800
+++ b/amd/dkms/post-remove.sh	2020-12-29 15:12:43.707538366 -0800
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
