--- a/amd/amdgpu/dce_v11_0.c.orig	2019-10-03 06:20:26.573651147 -0700
+++ b/amd/amdgpu/dce_v11_0.c	2019-10-03 06:22:22.876237019 -0700
@@ -1735,9 +1735,11 @@ static void dce_v11_0_afmt_setmode(struc
 	dce_v11_0_audio_write_sad_regs(encoder);
 	dce_v11_0_audio_write_latency_fields(encoder, mode);
 
-#if defined(HAVE_2ARGS_DRM_HDMI_AVI_INFOFRAME_FROM_DISPLAY_MODE)
+#if DRM_VERSION_CODE >= DRM_VERSION(5, 1, 0)
+	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, connector, mode);
+#elif defined(HAVE_2ARGS_DRM_HDMI_AVI_INFOFRAME_FROM_DISPLAY_MODE) && DRM_VERSION_CODE < DRM_VERSION(4, 14, 0)
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode);
-#else
+#elif DRM_VERSION_CODE < DRM_VERSION(5, 1, 0)
 	err = drm_hdmi_avi_infoframe_from_display_mode(&frame, mode, false);
 #endif
 	if (err < 0) {
