--- a/amd/display/amdgpu_dm/amdgpu_dm.c.orig	2019-10-03 15:29:01.125863032 -0700
+++ b/amd/display/amdgpu_dm/amdgpu_dm.c	2019-10-03 16:12:54.841046590 -0700
@@ -68,6 +68,9 @@
 #include <drm/drm_atomic_helper.h>
 #include <drm/drm_dp_mst_helper.h>
 #include <drm/drm_fb_helper.h>
+#if DRM_VERSION_CODE >= DRM_VERSION(5, 1, 0)
+#include <drm/drm_probe_helper.h>
+#endif
 #include <drm/drm_edid.h>
 
 #if defined(HAVE_DRM_AUDIO_COMPONENT_HEADER)
