--- a/post-remove.sh.orig	2019-10-01 16:27:12.000000000 -0700
+++ b/post-remove.sh	2019-11-23 14:05:07.088914177 -0800
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware/$1"
-rm -rf $FW_DIR/radeon
-rm -rf $FW_DIR/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware/$1"
+#rm -rf $FW_DIR/radeon
+#rm -rf $FW_DIR/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- a/pre-build.sh.orig	2019-10-01 16:27:12.000000000 -0700
+++ b/pre-build.sh	2019-11-23 14:05:11.825100512 -0800
@@ -44,7 +44,7 @@ find scheduler -name '*.c' -exec grep EX
 export KERNELVER
 ./configure
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-3.3-19/firmware/radeon $FW_DIR
-cp -ar /usr/src/amdgpu-3.3-19/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-3.3-19/firmware/radeon $FW_DIR
+#cp -ar /usr/src/amdgpu-3.3-19/firmware/amdgpu $FW_DIR
