diff -urp amdgpu-4.0-26.orig/amd/dkms/post-remove.sh amdgpu-4.0-26/amd/dkms/post-remove.sh
--- amdgpu-4.0-26.orig/amd/dkms/post-remove.sh	2021-01-19 02:15:52.000000000 -0800
+++ amdgpu-4.0-26/amd/dkms/post-remove.sh	2021-04-04 23:51:53.230518450 -0700
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
diff -urp amdgpu-4.0-26.orig/amd/dkms/pre-build.sh amdgpu-4.0-26/amd/dkms/pre-build.sh
--- amdgpu-4.0-26.orig/amd/dkms/pre-build.sh	2021-01-19 02:15:52.000000000 -0800
+++ amdgpu-4.0-26/amd/dkms/pre-build.sh	2021-04-04 23:51:49.486371528 -0700
@@ -97,6 +97,6 @@ export KERNELVER
 	-include $SRCTREE/include/linux/kconfig.h" \
 	./configure)
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-4.0-26/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-4.0-26/firmware/amdgpu $FW_DIR
