--- amdgpu-3.9-17.orig/amd/dkms/post-remove.sh	2020-10-21 13:03:36.000000000 -0700
+++ amdgpu-3.9-17/amd/dkms/post-remove.sh	2020-11-01 20:41:10.620294213 -0800
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-FW_DIR="/lib/firmware"
-rm -rf $FW_DIR/*/amdgpu
-[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
+#FW_DIR="/lib/firmware"
+#rm -rf $FW_DIR/*/amdgpu
+#[[ ! $(ls -A $FW_DIR) ]] && rm -rf $FW_DIR
--- amdgpu-3.9-17.orig/amd/dkms/pre-build.sh	2020-10-21 13:03:36.000000000 -0700
+++ amdgpu-3.9-17/amd/dkms/pre-build.sh	2020-11-01 20:41:32.421142225 -0800
@@ -97,6 +97,6 @@ export KERNELVER
 	-include $SRCTREE/include/linux/kconfig.h" \
 	./configure)
 
-FW_DIR="/lib/firmware/$KERNELVER"
-mkdir -p $FW_DIR
-cp -ar /usr/src/amdgpu-3.9-17/firmware/amdgpu $FW_DIR
+#FW_DIR="/lib/firmware/$KERNELVER"
+#mkdir -p $FW_DIR
+#cp -ar /usr/src/amdgpu-3.9-17/firmware/amdgpu $FW_DIR
