--- a/amd/amdkcl/kcl_drm.c.orig	2019-10-03 04:49:34.368207511 -0700
+++ b/amd/amdkcl/kcl_drm.c	2019-10-03 05:12:02.459506844 -0700
@@ -1,6 +1,7 @@
 #include <kcl/kcl_drm.h>
 #include "kcl_common.h"
 
+#if DRM_VERSION_CODE < DRM_VERSION(5, 1, 0)
 #if !defined(HAVE_DRM_CRTC_FORCE_DISABLE_ALL)
 /**
  * drm_crtc_force_disable_all - Forcibly turn off all enabled CRTCs
@@ -30,6 +31,7 @@ out:
 }
 EXPORT_SYMBOL(drm_crtc_force_disable_all);
 #endif
+#endif
 
 void (*_kcl_drm_fb_helper_cfb_fillrect)(struct fb_info *info,
 				const struct fb_fillrect *rect);
