diff -urp amdgpu-4.3-52.orig/amd/dkms/Makefile amdgpu-4.3-52/amd/dkms/Makefile
--- amdgpu-4.3-52.orig/amd/dkms/Makefile	2021-08-10 23:11:04.941679016 -0700
+++ amdgpu-4.3-52/amd/dkms/Makefile	2021-08-10 23:12:18.597808276 -0700
@@ -153,6 +153,7 @@ export CONFIG_DRM_AMDGPU_USERPTR=y
 export CONFIG_DRM_AMD_DC=y
 export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_HP_HOOK_WORKAROUND=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DCONFIG_DRM_TTM_DMA_PAGE_POOL
@@ -162,6 +163,7 @@ subdir-ccflags-y += -DCONFIG_DRM_AMDGPU_
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_0
 subdir-ccflags-y += -DCONFIG_HP_HOOK_WORKAROUND
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 ifeq ($(call _is_kcl_macro_defined,HAVE_AMDKCL_HMM_MIRROR_ENABLED),y)
 export CONFIG_HSA_AMD_SVM=y
