--- a/Makefile.orig	2020-01-07 01:49:10.387626644 -0800
+++ b/Makefile	2020-01-07 01:50:45.911321951 -0800
@@ -197,6 +197,7 @@ export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_DRM_AMD_DC_DCN1_01=y
 export CONFIG_DRM_AMD_DC_DCN2_0=y
 export CONFIG_DRM_AMD_DC_DCN2_1=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DBUILD_AS_DKMS
@@ -208,5 +209,6 @@ subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_01
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN2_0
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN2_1
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 obj-m += scheduler/ amd/amdgpu/ ttm/ amd/amdkcl/
--- a/amd/dkms/Makefile.orig	2020-01-07 01:49:10.387626644 -0800
+++ b/amd/dkms/Makefile	2020-01-07 01:50:37.901012049 -0800
@@ -197,6 +197,7 @@ export CONFIG_DRM_AMD_DC_DCN1_0=y
 export CONFIG_DRM_AMD_DC_DCN1_01=y
 export CONFIG_DRM_AMD_DC_DCN2_0=y
 export CONFIG_DRM_AMD_DC_DCN2_1=y
+export CONFIG_MMU_NOTIFIER=y
 
 subdir-ccflags-y += -DCONFIG_HSA_AMD
 subdir-ccflags-y += -DBUILD_AS_DKMS
@@ -208,5 +209,6 @@ subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN1_01
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN2_0
 subdir-ccflags-y += -DCONFIG_DRM_AMD_DC_DCN2_1
+subdir-ccflags-y += -DCONFIG_MMU_NOTIFIER
 
 obj-m += scheduler/ amd/amdgpu/ ttm/ amd/amdkcl/
