--- a/m4/drm-dp-atomic-find-vcpi-slots.m4.orig	2020-02-03 09:11:12.000000000 -0800
+++ b/m4/drm-dp-atomic-find-vcpi-slots.m4	2020-03-16 03:25:20.487038015 -0700
@@ -7,7 +7,8 @@ AC_DEFUN([AC_AMDGPU_DRM_DP_ATOMIC_FIND_V
 	AC_KERNEL_TRY_COMPILE_SYMBOL([
 		#include <drm/drm_dp_mst_helper.h>
 	], [
-		drm_dp_atomic_find_vcpi_slots(NULL, NULL, NULL, 0);
+		int retval;
+		retval = drm_dp_atomic_find_vcpi_slots(NULL, NULL, NULL, 0);
 	], [drm_dp_atomic_find_vcpi_slots], [drivers/gpu/drm/drm_dp_mst_topology.c], [
 		AC_MSG_RESULT(yes)
 		AC_DEFINE(HAVE_DRM_DP_ATOMIC_FIND_VCPI_SLOTS, 1, [drm_dp_atomic_find_vcpi_slots() is available])
--- a/amd/dkms/m4/drm-dp-atomic-find-vcpi-slots.m4.orig	2020-02-03 09:11:12.000000000 -0800
+++ b/amd/dkms/m4/drm-dp-atomic-find-vcpi-slots.m4	2020-03-16 03:25:18.296952887 -0700
@@ -7,7 +7,8 @@ AC_DEFUN([AC_AMDGPU_DRM_DP_ATOMIC_FIND_V
 	AC_KERNEL_TRY_COMPILE_SYMBOL([
 		#include <drm/drm_dp_mst_helper.h>
 	], [
-		drm_dp_atomic_find_vcpi_slots(NULL, NULL, NULL, 0);
+		int retval;
+		retval = drm_dp_atomic_find_vcpi_slots(NULL, NULL, NULL, 0);
 	], [drm_dp_atomic_find_vcpi_slots], [drivers/gpu/drm/drm_dp_mst_topology.c], [
 		AC_MSG_RESULT(yes)
 		AC_DEFINE(HAVE_DRM_DP_ATOMIC_FIND_VCPI_SLOTS, 1, [drm_dp_atomic_find_vcpi_slots() is available])
