--- a/test/warning-settings.cpp.orig	2019-10-07 17:15:21.000000000 -0700
+++ b/test/warning-settings.cpp	2023-06-02 11:56:30.816367630 -0700
@@ -29,6 +29,13 @@ int main() {
   if (GCC == 7) // FIXME: I _think_ this is a false positive but need
                 // to double check
     printf("-Walloc-size-larger-than=-1 ");
+  if (sizeof(void *) == 4) {
+    disable_gcc_warning("class-memaccess");
+    disable_gcc_warning("restrict");
+    disable_gcc_error("alloc-size-larger-than=");
+  } else {
+    disable_gcc_error("class-memaccess");
+  }
   disable_clang_warning("return-type-c-linkage");
   disable_clang_warning("tautological-compare");
   printf("\n");
