diff -ur MonoGame-3.4.orig/Build/GenerateProject.CSharp.xslt MonoGame-3.4/Build/GenerateProject.CSharp.xslt
--- MonoGame-3.4.orig/Build/GenerateProject.CSharp.xslt	2016-01-21 02:23:45.441871260 -0800
+++ MonoGame-3.4/Build/GenerateProject.CSharp.xslt	2016-01-21 02:30:31.802257667 -0800
@@ -18,6 +18,10 @@
     {
       return origName.Replace('.','_');
     }
+    public string UnixPath(string from)
+    {
+        return from.Replace('\\', '/');
+    }
     public string GetRelativePath(string from, string to)
     {
       try
@@ -1281,16 +1285,19 @@
                     </xsl:if>
                   </xsl:for-each>
                   <HintPath>
-                    <xsl:value-of
-                      select="user:GetRelativePath(
-                        concat(
-                          $project/@Path,
-                          '\',
-                          $project/@Name,
-                          '.',
-                          /Input/Generation/Platform,
-                          '.csproj'),
-                        @Path)" />
+                    <xsl:if test="/Input/Generation/HostPlatform = 'Linux'">/<xsl:value-of select="user:UnixPath(@Path)"/></xsl:if>
+                    <xsl:if test="/Input/Generation/HostPlatform != 'Linux'">
+                      <xsl:value-of
+                        select="user:GetRelativePath(
+                          concat(
+                            $project/@Path,
+                            '\',
+                            $project/@Name,
+                            '.',
+                            /Input/Generation/Platform,
+                            '.csproj'),
+                          @Path)" />
+                    </xsl:if>
                   </HintPath>
                 </Reference>
               </xsl:for-each>
@@ -1307,16 +1316,19 @@
                       </xsl:if>
                     </xsl:for-each>
                     <HintPath>
-                      <xsl:value-of
-                        select="user:GetRelativePath(
-                          concat(
-                            $project/@Path,
-                            '\',
-                            $project/@Name,
-                            '.',
-                            /Input/Generation/Platform,
-                            '.csproj'),
-                          @Path)" />
+                      <xsl:if test="/Input/Generation/HostPlatform = 'Linux'">/<xsl:value-of select="user:UnixPath(@Path)"/></xsl:if>
+                      <xsl:if test="/Input/Generation/HostPlatform != 'Linux'">
+                        <xsl:value-of
+                          select="user:GetRelativePath(
+                            concat(
+                              $project/@Path,
+                              '\',
+                              $project/@Name,
+                              '.',
+                              /Input/Generation/Platform,
+                              '.csproj'),
+                            @Path)" />
+                      </xsl:if>
                     </HintPath>
                   </Reference>
                 </xsl:for-each>
