diff -urN MonoGame-3.4.orig/IDE/MonoDevelop/MonoDevelop.MonoGame/AddinInfo.cs MonoGame-3.4/IDE/MonoDevelop/MonoDevelop.MonoGame/AddinInfo.cs
--- MonoGame-3.4.orig/IDE/MonoDevelop/MonoDevelop.MonoGame/AddinInfo.cs	1969-12-31 16:00:00.000000000 -0800
+++ MonoGame-3.4/IDE/MonoDevelop/MonoDevelop.MonoGame/AddinInfo.cs	2016-01-24 11:46:11.658837276 -0800
@@ -0,0 +1,16 @@
+using System;
+using Mono.Addins;
+using Mono.Addins.Description;
+
+[assembly:Addin ("MonoGame",
+        Namespace = "MonoDevelop",
+        Version = MonoDevelop.BuildInfo.Version,
+        Category = "IDE extensions")]
+
+[assembly:AddinName ("MonoGame")]
+[assembly:AddinDescription ("MonoGame")]
+
+[assembly:AddinDependency ("Core", MonoDevelop.BuildInfo.Version)]
+[assembly:AddinDependency ("Ide", MonoDevelop.BuildInfo.Version)]
+[assembly:AddinDependency ("DesignerSupport", MonoDevelop.BuildInfo.Version)]
+
diff -urN MonoGame-3.4.orig/IDE/MonoDevelop/MonoDevelop.MonoGame/MonoDevelop.MonoGame.csproj MonoGame-3.4/IDE/MonoDevelop/MonoDevelop.MonoGame/MonoDevelop.MonoGame.csproj
--- MonoGame-3.4.orig/IDE/MonoDevelop/MonoDevelop.MonoGame/MonoDevelop.MonoGame.csproj	2016-01-24 11:43:29.316601363 -0800
+++ MonoGame-3.4/IDE/MonoDevelop/MonoDevelop.MonoGame/MonoDevelop.MonoGame.csproj	2016-01-24 11:44:16.191402000 -0800
@@ -65,6 +65,7 @@
     <Compile Include="MonoGameBuildExtension.cs" />
     <Compile Include="MonoGameConditions.cs" />
     <Compile Include="ContentItemTemplate.cs" />
+    <Compile Include="AddinInfo.cs" />
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
   <ItemGroup>
