diff -urp Urho3D-1.8-ALPHA.orig/Source/Urho3D/CMakeLists.txt Urho3D-1.8-ALPHA/Source/Urho3D/CMakeLists.txt
--- Urho3D-1.8-ALPHA.orig/Source/Urho3D/CMakeLists.txt	2019-10-05 21:46:51.000000000 -0700
+++ Urho3D-1.8-ALPHA/Source/Urho3D/CMakeLists.txt	2020-10-05 12:38:53.389726707 -0700
@@ -372,6 +372,7 @@ elseif (TVOS)
     # TODO: Temporary workaround as CMake does not recognize AppleTV platform yet
     add_dependencies (${TARGET_NAME} APPLETV_POST_CMAKE_FIX)
 elseif (EMSCRIPTEN)
+    set_target_properties(${TARGET_NAME} PROPERTIES OUTPUT_NAME "Urho3D_web")
     install (TARGETS ${TARGET_NAME} LIBRARY DESTINATION ${DEST_LIBRARY_DIR} PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ ARCHIVE DESTINATION ${DEST_LIBRARY_DIR})
 else ()
     install (TARGETS ${TARGET_NAME} RUNTIME DESTINATION ${DEST_RUNTIME_DIR} LIBRARY DESTINATION ${DEST_LIBRARY_DIR} ARCHIVE DESTINATION ${DEST_LIBRARY_DIR})
@@ -556,6 +557,9 @@ if (ANDROID)
     # After migration to Gradle build system we do not support Android cross-compiling build via pkgconfig anymore, however, this file is still being provided as reference only
     set (RENAME RENAME Urho3D-${ANDROID_ABI}.pc)
 endif ()
+if (WEB)
+    set (RENAME RENAME Urho3D_web.pc)
+endif ()
 install (FILES ${CMAKE_CURRENT_BINARY_DIR}/Urho3D.pc DESTINATION ${DEST_PKGCONFIG_DIR} ${RENAME})
 
 # Define post build steps
