diff -urp Urho3D-1.6.orig/Source/Urho3D/CMakeLists.txt Urho3D-1.6/Source/Urho3D/CMakeLists.txt
--- Urho3D-1.6.orig/Source/Urho3D/CMakeLists.txt	2016-11-20 23:21:49.262823301 -0800
+++ Urho3D-1.6/Source/Urho3D/CMakeLists.txt	2016-11-20 23:48:16.614431632 -0800
@@ -176,9 +176,9 @@ if (URHO3D_LUA)
         if (IOS)
             set (IOS_FIX CMAKE_COMMAND /usr/bin/env -i PATH=$ENV{PATH} ${CMAKE_COMMAND} BUILD_COMMAND bash -c "sed -i '' 's/EFFECTIVE_PLATFORM_NAME//g' CMakeScripts/install_postBuildPhase.make*")
         endif ()
-        ExternalProject_Add (tolua++
-            SOURCE_DIR /usr/bin
-            CMAKE_ARGS -DDEST_RUNTIME_DIR=${CMAKE_BINARY_DIR}/bin/tool -DBAKED_CMAKE_SOURCE_DIR=${CMAKE_SOURCE_DIR} -DURHO3D_UPDATE_SOURCE_TREE=${URHO3D_UPDATE_SOURCE_TREE} ${IOS_FIX})
+        #ExternalProject_Add (tolua++
+        #    SOURCE_DIR /usr/bin
+        #    CMAKE_ARGS -DDEST_RUNTIME_DIR=${CMAKE_BINARY_DIR}/bin/tool -DBAKED_CMAKE_SOURCE_DIR=${CMAKE_SOURCE_DIR} -DURHO3D_UPDATE_SOURCE_TREE=${URHO3D_UPDATE_SOURCE_TREE} ${IOS_FIX})
     else ()
         # Otherwise, build it internally as per normal
         
