diff -ur Urho3D-1.5.orig/Source/Urho3D/Resource/JSONFile.cpp Urho3D-1.5/Source/Urho3D/Resource/JSONFile.cpp
--- Urho3D-1.5.orig/Source/Urho3D/Resource/JSONFile.cpp	2016-02-04 19:17:37.315648382 -0800
+++ Urho3D-1.5/Source/Urho3D/Resource/JSONFile.cpp	2016-02-04 19:26:56.479109825 -0800
@@ -200,8 +200,7 @@
             {
                 const char* _name = i->first_.CString();
                 rapidjson::Value value;
-                rapidjson::Value name;
-		name = _name;
+                rapidjson::Value name(_name,allocator);
                 rapidjsonValue.AddMember(name, value, allocator);
                 ToRapidjsonValue(rapidjsonValue[name], i->second_, allocator);
             }
