--- cmake/cef_variables.cmake.orig	2021-09-01 21:55:56.000000000 -0700
+++ cmake/cef_variables.cmake	2021-09-04 23:39:22.309532284 -0700
@@ -85,7 +85,6 @@ if(OS_LINUX)
     -fPIC                           # Generate position-independent code for shared libraries
     -fstack-protector               # Protect some vulnerable functions from stack-smashing (security feature)
     -funwind-tables                 # Support stack unwinding for backtrace()
-    -fvisibility=hidden             # Give hidden visibility to declarations that are not explicitly marked as visible
     --param=ssp-buffer-size=4       # Set the minimum buffer size protected by SSP (security feature, related to stack-protector)
     -pipe                           # Use pipes rather than temporary files for communication between build stages
     -pthread                        # Use the pthread library
@@ -104,7 +103,6 @@ if(OS_LINUX)
     -fno-exceptions                 # Disable exceptions
     -fno-rtti                       # Disable real-time type information
     -fno-threadsafe-statics         # Don't generate thread-safe statics
-    -fvisibility-inlines-hidden     # Give hidden visibility to inlined class member functions
     -std=c++14                      # Use the C++14 language standard
     -Wsign-compare                  # Warn about mixed signed/unsigned type comparisons
     )
