--- a/packaging/CMakeLists.txt.orig	2023-09-17 22:11:16.323098131 -0700
+++ b/packaging/CMakeLists.txt	2023-09-17 22:12:15.327097343 -0700
@@ -107,24 +107,6 @@ install(FILES ${CMAKE_BINARY_DIR}/hipamd
 install(FILES ${CMAKE_BINARY_DIR}/hipamd/hip-config-nvidia.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/hip COMPONENT dev)
 #End dev files install
 
-#Begin doc files install
-find_program(DOXYGEN_EXE doxygen)
-if(DOXYGEN_EXE)
-    if(EXISTS "${HIP_COMMON_DIR}/docs/doxygen-input/doxy.cfg")
-        add_custom_target(build_doxygen ALL
-            COMMAND HIP_PATH=${HIP_COMMON_DIR} doxygen ${HIP_COMMON_DIR}/docs/doxygen-input/doxy.cfg)
-    elseif(EXISTS "${HIP_COMMON_DIR}/docs/.doxygen/Doxyfile")
-        add_custom_target(build_doxygen ALL
-            COMMAND HIP_PATH=${HIP_COMMON_DIR} doxygen ${HIP_COMMON_DIR}/docs/.doxygen/Doxyfile)
-    else()
-        message(FATAL_ERROR "Unable to find doxygen config file")
-    endif()
-    install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/RuntimeAPI/html
-            DESTINATION ${CMAKE_INSTALL_DOCDIR}/RuntimeAPI COMPONENT doc)
-endif()
-#End doc files install
-
-
 ##################################
 # Packaging steps COMMON Variables
 ##################################
