diff -urp hipamd-rocm-5.1.3.orig/packaging/CMakeLists.txt hipamd-rocm-5.1.3/packaging/CMakeLists.txt
--- hipamd-rocm-5.1.3.orig/packaging/CMakeLists.txt	2023-08-01 01:33:47.340724374 -0700
+++ hipamd-rocm-5.1.3/packaging/CMakeLists.txt	2023-08-01 01:34:35.888725638 -0700
@@ -87,8 +87,10 @@ install(DIRECTORY ${hip_SOURCE_DIR}/incl
         DESTINATION include/hip COMPONENT dev)
 install(DIRECTORY ${hip_SOURCE_DIR}/include/hip/nvidia_detail
         DESTINATION include/hip COMPONENT dev)
+if(HIP_PLATFORM STREQUAL "amd")
 install(FILES ${CMAKE_BINARY_DIR}/include/hip/amd_detail/hip_prof_str.h
         DESTINATION include/hip/amd_detail COMPONENT dev)
+endif()
 install(FILES ${CMAKE_BINARY_DIR}/include/hip/hip_version.h
         DESTINATION include/hip COMPONENT dev)
 install(FILES ${CMAKE_BINARY_DIR}/.hipVersion DESTINATION bin COMPONENT dev)
