diff -urp hipamd-rocm-5.6.0.orig/CMakeLists.txt hipamd-rocm-5.6.0/CMakeLists.txt
--- hipamd-rocm-5.6.0.orig/CMakeLists.txt	2023-03-15 02:13:57.000000000 -0700
+++ hipamd-rocm-5.6.0/CMakeLists.txt	2023-08-01 12:24:26.813741179 -0700
@@ -51,6 +51,7 @@ option(BUILD_HIPIFY_CLANG "Enable buildi
 option(__HIP_ENABLE_PCH "Enable/Disable pre-compiled hip headers" ON)
 option(HIP_OFFICIAL_BUILD "Enable/Disable for mainline/staging builds" ON)
 option(FILE_REORG_BACKWARD_COMPATIBILITY "Enable File Reorg with backward compatibility" ON)
+option(SKIP_TESTS "Skip building tests" ON)
 set(HIPCC_BIN_DIR "" CACHE STRING "HIPCC and HIPCONFIG binary directories")
 
 if(__HIP_ENABLE_PCH)
@@ -446,7 +447,9 @@ execute_process(COMMAND "${CMAKE_COMMAND
 if(${RUN_HIT} EQUAL 0)
     execute_process(COMMAND "${CMAKE_COMMAND}" -E copy_directory "${HIP_COMMON_BIN_DIR}" "${HIP_ROOT_DIR}/bin" RESULT_VARIABLE RUN_HIT ERROR_QUIET)
 endif()
-if(HIP_CATCH_TEST EQUAL "1")
+if(SKIP_TESTS)
+	message(STATUS "Skipping building tests")
+elseif(HIP_CATCH_TEST EQUAL "1")
 	message(STATUS "Building of catch tests through hipamd is no longer supported. Testing targets will not be available. catch tests have been moved to an independent github project hip-tests. Please refer to hip-tests Readme for build instructions! ")
 else()
     if(${RUN_HIT} EQUAL 0)
