diff -urp hipamd-rocm-5.2.3.orig/packaging/CMakeLists.txt hipamd-rocm-5.2.3/packaging/CMakeLists.txt
--- hipamd-rocm-5.2.3.orig/packaging/CMakeLists.txt	2023-09-24 14:35:41.428043669 -0700
+++ hipamd-rocm-5.2.3/packaging/CMakeLists.txt	2023-09-24 14:37:30.776042411 -0700
@@ -94,16 +94,6 @@ install(FILES ${CMAKE_BINARY_DIR}/.hipVe
 install(DIRECTORY ${HIP_COMMON_DIR}/cmake/ DESTINATION lib/cmake/hip COMPONENT dev)
 #End dev files install
 
-#Begin doc files install
-find_program(DOXYGEN_EXE doxygen)
-if(DOXYGEN_EXE)
-    add_custom_target(build_doxygen ALL
-                  COMMAND HIP_PATH=${HIP_COMMON_DIR} doxygen ${HIP_COMMON_DIR}/docs/doxygen-input/doxy.cfg)
-    install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/RuntimeAPI/html
-            DESTINATION share/doc/hip/RuntimeAPI COMPONENT doc)
-endif()
-#End doc files install
-
 #Begin samples files install
 install(DIRECTORY ${HIP_COMMON_DIR}/samples DESTINATION share/hip COMPONENT samples)
 #End samples files install
