--- a/packaging/CMakeLists.txt.orig	2023-08-01 01:21:37.428705363 -0700
+++ b/packaging/CMakeLists.txt	2023-08-01 01:25:11.156710930 -0700
@@ -82,8 +82,10 @@ install(DIRECTORY ${hip_SOURCE_DIR}/incl
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hip COMPONENT dev)
 install(DIRECTORY ${hip_SOURCE_DIR}/include/hip/nvidia_detail
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hip COMPONENT dev)
-install(FILES ${CMAKE_BINARY_DIR}/include/hip/amd_detail/hip_prof_str.h
-        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hip/amd_detail COMPONENT dev)
+if(HIP_PLATFORM STREQUAL "amd")
+  install(FILES ${CMAKE_BINARY_DIR}/include/hip/amd_detail/hip_prof_str.h
+          DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hip/amd_detail COMPONENT dev)
+endif()
 install(FILES ${CMAKE_BINARY_DIR}/include/hip/hip_version.h
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hip COMPONENT dev)
 install(DIRECTORY ${HIP_COMMON_DIR}/cmake/ DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/hip COMPONENT dev)
