diff -urp hipamd-rocm-5.1.3.orig/src/CMakeLists.txt hipamd-rocm-5.1.3/src/CMakeLists.txt
--- hipamd-rocm-5.1.3.orig/src/CMakeLists.txt	2024-04-01 14:27:44.655022469 -0700
+++ hipamd-rocm-5.1.3/src/CMakeLists.txt	2024-04-01 14:28:24.922656681 -0700
@@ -147,6 +147,9 @@ target_include_directories(amdhip64
 target_compile_definitions(amdhip64 PRIVATE __HIP_PLATFORM_AMD__)
 target_link_libraries(amdhip64 PRIVATE ${OPENGL_LIBRARIES})
 target_link_libraries(amdhip64 PRIVATE ${CMAKE_DL_LIBS})
+target_link_options(amdhip64 PUBLIC -Wl,-rpath=@ROCM_PATH@/@LIBDIR@)
+target_link_directories(amdhip64 PUBLIC @ROCM_PATH@/@LIBDIR@)
+target_link_libraries(amdhip64 PUBLIC hsa-runtime64 amd_comgr)
 # Additional dependencies for hipRTC
 if(WIN32)
   target_link_libraries(amdhip64 PRIVATE Dbghelp.lib)
Only in hipamd-rocm-5.1.3/src: CMakeLists.txt.orig
Only in hipamd-rocm-5.1.3/src: CMakeLists.txt.rej
