diff -urp hipamd.orig/src/CMakeLists.txt hipamd/src/CMakeLists.txt
--- hipamd.orig/src/CMakeLists.txt	2024-04-01 13:03:47.872537957 -0700
+++ hipamd/src/CMakeLists.txt	2024-04-01 13:07:36.826511614 -0700
@@ -145,6 +145,9 @@ target_include_directories(amdhip64
 target_compile_definitions(amdhip64 PRIVATE __HIP_PLATFORM_AMD__)
 target_link_libraries(amdhip64 PRIVATE ${OPENGL_LIBRARIES})
 target_link_libraries(amdhip64 PRIVATE ${CMAKE_DL_LIBS})
+target_link_options(amdhip64 PUBLIC -Wl,-rpath=@ROCM_PATH@/@ROCM_LIBDIR@)
+target_link_directories(amdhip64 PUBLIC @ROCM_PATH@/@ROCM_LIBDIR@)
+target_link_libraries(amdhip64 PUBLIC hsa-runtime64 amd_comgr)
 
 target_link_directories(amdhip64 PRIVATE ${CLR_PATH}/rocclr_build)
 
