diff -urp hipamd-rocm-5.2.3.orig/src/hiprtc/CMakeLists.txt hipamd-rocm-5.2.3/src/hiprtc/CMakeLists.txt
--- hipamd-rocm-5.2.3.orig/src/hiprtc/CMakeLists.txt	2024-04-01 16:26:19.143782791 -0700
+++ hipamd-rocm-5.2.3/src/hiprtc/CMakeLists.txt	2024-04-01 16:29:06.874297705 -0700
@@ -85,6 +85,8 @@ if(WIN32)
   target_link_libraries(hiprtc PRIVATE Dbghelp.lib)
 endif()
 
+target_link_options(hiprtc PUBLIC -Wl,-rpath=@ROCM_PATH@/@LIBDIR@)
+target_link_directories(hiprtc PUBLIC @ROCM_PATH@/@LIBDIR@)
 target_link_libraries(hiprtcobject PRIVATE ${CMAKE_DL_LIBS})
 
 if(BUILD_SHARED_LIBS)
