diff -urp ROCclr-rocm-5.1.3.orig/cmake/ROCclrHSA.cmake ROCclr-rocm-5.1.3/cmake/ROCclrHSA.cmake
--- ROCclr-rocm-5.1.3.orig/cmake/ROCclrHSA.cmake	2022-05-01 22:28:44.000000000 -0700
+++ ROCclr-rocm-5.1.3/cmake/ROCclrHSA.cmake	2023-08-30 13:07:59.430655367 -0700
@@ -20,11 +20,10 @@
 
 find_package(hsa-runtime64 1.3 REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    /usr/
   PATH_SUFFIXES
     cmake/hsa-runtime64
-    lib/cmake/hsa-runtime64
-    lib64/cmake/hsa-runtime64)
+    @LIBDIR@/cmake/hsa-runtime64)
 target_link_libraries(rocclr PUBLIC hsa-runtime64::hsa-runtime64)
 
 find_package(NUMA)
diff -urp ROCclr-rocm-5.1.3.orig/cmake/ROCclrLC.cmake ROCclr-rocm-5.1.3/cmake/ROCclrLC.cmake
--- ROCclr-rocm-5.1.3.orig/cmake/ROCclrLC.cmake	2022-05-01 22:28:44.000000000 -0700
+++ ROCclr-rocm-5.1.3/cmake/ROCclrLC.cmake	2023-08-30 13:07:59.430655367 -0700
@@ -20,10 +20,10 @@
 
 find_package(amd_comgr REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    /usr/
   PATH_SUFFIXES
     cmake/amd_comgr
-    lib/cmake/amd_comgr)
+    @LIBDIR@/cmake/amd_comgr)
 
 target_compile_definitions(rocclr PUBLIC WITH_LIGHTNING_COMPILER USE_COMGR_LIBRARY)
 if(BUILD_SHARED_LIBS)
diff -urp ROCclr-rocm-5.1.3.orig/elf/test/CMakeLists.txt ROCclr-rocm-5.1.3/elf/test/CMakeLists.txt
--- ROCclr-rocm-5.1.3.orig/elf/test/CMakeLists.txt	2022-05-01 22:28:44.000000000 -0700
+++ ROCclr-rocm-5.1.3/elf/test/CMakeLists.txt	2023-08-30 13:07:59.430655367 -0700
@@ -26,24 +26,24 @@ cmake_minimum_required(VERSION 3.5.1)
 
 find_package(amd_comgr REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    /usr/
   PATH_SUFFIXES
     cmake/amd_comgr
-    lib/cmake/amd_comgr)
+    @LIBDIR@/cmake/amd_comgr)
 
 find_package(hsa-runtime64 REQUIRED CONFIG
   PATHS
-    /opt/rocm/
+    /usr/
   PATH_SUFFIXES
-    cmake/hsa-runtime64)
+    @LIBDIR@/cmake/hsa-runtime64)
 
 find_package(Threads REQUIRED)
 
 # Look for ROCclr which contains elfio
 find_package(ROCclr REQUIRED CONFIG
   PATHS
-    /opt/rocm
-    /opt/rocm/rocclr)
+    /usr/
+    /usr/rocclr)
 
 add_executable(elf_test main.cpp)
 set_target_properties(
@@ -60,4 +60,4 @@ add_definitions(-DUSE_COMGR_LIBRARY -DCO
 
 target_link_libraries(elf_test PRIVATE amdrocclr_static)
 
-#-------------------------------------elf_test--------------------------------------#
\ No newline at end of file
+#-------------------------------------elf_test--------------------------------------#
diff -urp ROCclr-rocm-5.1.3.orig/elf/test/main.cpp ROCclr-rocm-5.1.3/elf/test/main.cpp
--- ROCclr-rocm-5.1.3.orig/elf/test/main.cpp	2022-05-01 22:28:44.000000000 -0700
+++ ROCclr-rocm-5.1.3/elf/test/main.cpp	2023-08-30 13:07:59.430655367 -0700
@@ -45,8 +45,8 @@ static constexpr size_t rodataSymbolInfo
 
 // Elf::COMMENT,        ".comment",        1, SHT_PROGBITS, 0,
 static const amd::Elf::SymbolInfo commentSymbolInfos_[] = {
-    { ".comment", nullptr, 0, "compile", "-g -I/opt/include",  strlen("-g -I/opt/include") + 1 },
-    { ".comment", nullptr, 0, "link",    "-g -l/opt/rocm/lib", strlen("-g -l/opt/rocm/lib") + 1 },
+    { ".comment", nullptr, 0, "compile", "-g -I/usr/include",  strlen("-g -I/usr/include") + 1 },
+    { ".comment", nullptr, 0, "link",    "-g -l/usr/@LIBDIR@", strlen("-g -l/usr/@LIBDIR@") + 1 },
   };
 static constexpr size_t commentSymbolInfosSize_ = sizeof(commentSymbolInfos_)
     / sizeof(commentSymbolInfos_[0]);
diff -urp ROCclr-rocm-5.1.3.orig/os/os_posix.cpp ROCclr-rocm-5.1.3/os/os_posix.cpp
--- ROCclr-rocm-5.1.3.orig/os/os_posix.cpp	2022-05-01 22:28:44.000000000 -0700
+++ ROCclr-rocm-5.1.3/os/os_posix.cpp	2023-08-30 13:09:23.490657060 -0700
@@ -447,7 +447,7 @@ uint64_t Os::timerResolutionNanos() {
 
 const char* Os::libraryExtension() { return MACOS_SWITCH(".dylib", ".so"); }
 
-const char* Os::libraryPrefix() { return "lib"; }
+const char* Os::libraryPrefix() { return "@LIBDIR@"; }
 
 const char* Os::objectExtension() { return ".o"; }
 
