--- a/build/config/compiler/BUILD.gn.orig	2024-09-05 09:54:07.627681077 -0700
+++ b/build/config/compiler/BUILD.gn	2024-09-05 10:26:37.579904114 -0700
@@ -202,6 +202,7 @@ declare_args() {
   simple_template_names = is_clang && !is_nacl && !is_win && !is_apple
 
   #use_mold = false
+  #custom_optimization_level = ""
 }
 
 declare_args() {
@@ -2589,9 +2590,17 @@ config("optimize_max") {
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O2" ] + common_optimize_on_cflags
+      if (custom_optimization_level != "") {
+        cflags = [ "-O$custom_optimization_level" ] + common_optimize_on_cflags
+      } else {
+        cflags = [ "-O2" ] + common_optimize_on_cflags
+      }
+    }
+    if (custom_optimization_level != "") {
+      rustflags = [ "-Copt-level=$custom_optimization_level" ]
+    } else {
+      rustflags = [ "-Copt-level=3" ]
     }
-    rustflags = [ "-Copt-level=3" ]
   }
 }
 
@@ -2625,9 +2634,17 @@ config("optimize_speed") {
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O3" ] + common_optimize_on_cflags
+      if (custom_optimization_level != "") {
+        cflags = [ "-O$custom_optimization_level" ] + common_optimize_on_cflags
+      } else {
+        cflags = [ "-O3" ] + common_optimize_on_cflags
+      }
+    }
+    if (custom_optimization_level != "") {
+      rustflags = [ "-Copt-level=$custom_optimization_level" ]
+    } else {
+      rustflags = [ "-Copt-level=3" ]
     }
-    rustflags = [ "-Copt-level=3" ]
   }
 }
 
