--- a/build/config/compiler/BUILD.gn.orig	2026-03-11 08:27:02.586102775 -0700
+++ b/build/config/compiler/BUILD.gn	2026-03-11 08:28:39.663105655 -0700
@@ -418,6 +419,11 @@ config("compiler") {
     configs += [ "//build/config/zos:compiler" ]
   }
 
+  if (use_system_clang) {
+    cflags += [ "-stdlib=libc++" ]
+    ldflags += [ "-stdlib=libc++" ]
+  }
+
   if (use_march != "") {
     cflags += [ "-march=${use_march}" ]
   }
