Subject:  Make SIMD instructions optional for zlib
Patch Author:  Orson Teodoro <orsonteodoro@hotmail.com>
Date: Wed Oct 20 03:40:04 PM PDT 2021 (Unix time: 1634769604)
--- a/third_party/zlib/BUILD.gn.orig	2021-10-18 12:24:23.000000000 -0700
+++ b/third_party/zlib/BUILD.gn	2021-10-20 16:50:51.850339504 -0700
@@ -4,6 +4,11 @@
 
 import("//build/config/compiler/compiler.gni")
 
+declare_args() {
+  use_sse4_2 = true
+  use_ssse3 = true
+}
+
 if (build_with_chromium) {
   import("//testing/test.gni")
 }
@@ -57,7 +62,7 @@ use_x86_x64_optimizations =
     (current_cpu == "x86" || current_cpu == "x64") && !is_ios
 
 config("zlib_adler32_simd_config") {
-  if (use_x86_x64_optimizations) {
+  if (use_x86_x64_optimizations && use_ssse3) {
     defines = [ "ADLER32_SIMD_SSSE3" ]
     if (is_win) {
       defines += [ "X86_WINDOWS" ]
@@ -80,7 +85,7 @@ source_set("zlib_adler32_simd") {
       "adler32_simd.h",
     ]
 
-    if (!is_win || is_clang) {
+    if ((!is_win || is_clang) && use_ssse3) {
       cflags = [ "-mssse3" ]
     }
   }
@@ -193,7 +198,7 @@ source_set("zlib_inflate_chunk_simd") {
 }
 
 config("zlib_crc32_simd_config") {
-  if (use_x86_x64_optimizations) {
+  if (use_x86_x64_optimizations && use_sse4_2) {
     defines = [ "CRC32_SIMD_SSE42_PCLMUL" ]
   }
 }
@@ -207,7 +212,7 @@ source_set("zlib_crc32_simd") {
       "crc32_simd.h",
     ]
 
-    if (!is_win || is_clang) {
+    if ((!is_win || is_clang) && use_sse4_2) {
       cflags = [
         "-msse4.2",
         "-mpclmul",
@@ -222,7 +227,7 @@ source_set("zlib_crc32_simd") {
 }
 
 config("zlib_x86_simd_config") {
-  if (use_x86_x64_optimizations) {
+  if (use_x86_x64_optimizations && use_sse4_2) {
     defines = [
       "CRC32_SIMD_SSE42_PCLMUL",
       "DEFLATE_FILL_WINDOW_SSE2",
@@ -239,7 +244,7 @@ source_set("zlib_x86_simd") {
       "fill_window_sse.c",
     ]
 
-    if (!is_win || is_clang) {
+    if ( ( !is_win || is_clang ) && use_sse4_2) {
       cflags = [
         "-msse4.2",
         "-mpclmul",
