--- a/chrome/browser/pdf/BUILD.gn.orig	2026-02-26 00:11:33.007008322 -0800
+++ b/chrome/browser/pdf/BUILD.gn	2026-02-26 00:11:52.430401138 -0800
@@ -2,6 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//chrome/common/features.gni")
 import("//pdf/features.gni")
 import("//printing/buildflags/buildflags.gni")
 
@@ -32,7 +33,13 @@ source_set("pdf") {
     "//build:chromeos_buildflags",
     "//chrome/browser:browser_process",
     "//chrome/browser:resources",
-    "//chrome/browser/glic",
+  ]
+  if (enable_glic || enable_glic_android) {
+    deps += [
+      "//chrome/browser/glic",
+    ]
+  }
+  deps += [
     "//chrome/browser/pdf:pdf_pref_names",
     "//chrome/browser/profiles:profile",
     "//chrome/browser/resources/pdf:resources",
