--- a/v8/BUILD.gn.orig	2026-05-16 22:31:47.449746105 -0700
+++ b/v8/BUILD.gn	2026-05-16 22:58:50.089494773 -0700
@@ -809,8 +809,8 @@ assert(!v8_enable_pointer_compression_8g
 assert(!v8_enable_sandbox || v8_enable_external_code_space,
        "The sandbox requires the external code space")
 
-assert(!v8_enable_sandbox || use_safe_libcxx,
-       "The sandbox requires libc++ hardening")
+#assert(!v8_enable_sandbox || use_safe_libcxx,
+#       "The sandbox requires libc++ hardening")
 
 assert(!v8_enable_memory_corruption_api || v8_enable_sandbox,
        "The Memory Corruption API requires the sandbox")
