--- a/third_party/pdfium/BUILD.gn.orig	2026-01-26 10:29:08.268779237 -0800
+++ b/third_party/pdfium/BUILD.gn	2026-01-26 11:24:45.150559873 -0800
@@ -4,6 +4,7 @@
 
 import("//build/config/clang/clang.gni")
 import("//build/config/gclient_args.gni")
+import("//build/config/linux/pkg_config.gni")
 import("//testing/test.gni")
 import("pdfium.gni")
 
@@ -24,6 +25,27 @@ group("freetype_common") {
   }
 }
 
+if (use_system_freetype) {
+  pkg_config("freetype2_from_pkgconfig") {
+    defines = [ "USE_SYSTEM_FREETYPE" ]
+    packages = [ "freetype2" ]
+  }
+}
+
+if (use_system_lcms2) {
+  pkg_config("lcms2_from_pkgconfig") {
+    defines = [ "USE_SYSTEM_LCMS2" ]
+    packages = [ "lcms2" ]
+  }
+}
+
+if (use_system_libopenjpeg2) {
+  pkg_config("libopenjpeg2_from_pkgconfig") {
+    defines = [ "USE_SYSTEM_LIBOPENJPEG2" ]
+    packages = [ "libopenjp2" ]
+  }
+}
+
 config("pdfium_common_config") {
   cflags = []
   cflags_cc = []
@@ -31,7 +53,18 @@ config("pdfium_common_config") {
   include_dirs = [ "." ]
   defines = []
 
-  if (!use_system_libopenjpeg2) {
+  configs = []
+  if (use_system_freetype) {
+    configs += [ ":freetype2_from_pkgconfig" ]
+  }
+
+  if (use_system_lcms2) {
+    configs += [ ":lcms2_from_pkgconfig" ]
+  }
+
+  if (use_system_libopenjpeg2) {
+    configs += [ ":libopenjpeg2_from_pkgconfig" ]
+  } else {
     defines += [ "OPJ_STATIC" ]
   }
 
--- a/third_party/pdfium/core/fxge/freetype/fx_freetype.cpp.orig	2026-01-06 16:50:30.000000000 -0800
+++ b/third_party/pdfium/core/fxge/freetype/fx_freetype.cpp	2026-01-26 11:21:36.545140029 -0800
@@ -13,7 +13,11 @@
 #include "core/fxge/cfx_gemodule.h"
 
 #define DEFINE_PS_TABLES
-#include "third_party/freetype/include/pstables.h"
+#ifdef USE_SYSTEM_FREETYPE
+#  include <freetype2/pstables.h>
+#else
+#  include "third_party/freetype/include/pstables.h"
+#endif
 
 namespace {
 
