--- a/v8/BUILD.gn.orig	2025-05-13 12:27:04.361980303 -0700
+++ b/v8/BUILD.gn	2025-05-13 12:36:23.307012599 -0700
@@ -515,6 +515,10 @@ declare_args() {
 
   # Deinterleaving load support.
   v8_enable_wasm_deinterleaved_mem_ops = false
+
+  # Unbreak mksnapshot
+  v8_mksnapshot_use_8gb_heap = false
+  v8_mksnapshot_disable_drumbrake = false
 }
 
 # Derived defaults.
@@ -2475,6 +2479,18 @@ template("run_mksnapshot") {
       data += [ "$root_out_dir/mksnapshot_output${suffix}.log" ]
     }
 
+    if (v8_mksnapshot_use_8gb_heap) {
+      args += [
+        "--max-old-space-size=8192"
+      ]
+    }
+
+    if (v8_mksnapshot_disable_drumbrake) {
+      args += [
+        "--no-drumbrake"
+      ]
+    }
+
     args += [
       "./" + rebase_path(get_label_info(":mksnapshot($v8_snapshot_toolchain)",
                                         "root_out_dir") + "/mksnapshot",
