--- a/build/config/compiler/BUILD.gn.orig	2026-01-06 16:50:30.000000000 -0800
+++ b/build/config/compiler/BUILD.gn	2026-01-19 11:55:38.706742729 -0800
@@ -49,6 +49,7 @@ if (!is_clang) {
 declare_args() {
   # This allows overriding the location of lld.
   lld_path = default_lld_path
+  use_sse3 = true
 }
 
 declare_args() {
@@ -1298,7 +1299,9 @@ config("compiler_cpu_abi") {
         cflags += [ "-m64" ]
         ldflags += [ "-m64" ]
       }
-      cflags += [ "-msse3" ]
+      if (use_sse3) {
+        cflags += [ "-msse3" ]
+      }
     } else if (current_cpu == "x86") {
       if (is_clang && !is_android && !is_chromeos_device) {
         cflags += [ "--target=i386-unknown-linux-gnu" ]
@@ -1307,10 +1310,12 @@ config("compiler_cpu_abi") {
         cflags += [ "-m32" ]
         ldflags += [ "-m32" ]
       }
-      cflags += [
-        "-mfpmath=sse",
-        "-msse3",
-      ]
+      if (use_sse3) {
+        cflags += [
+          "-mfpmath=sse",
+          "-msse3",
+        ]
+      }
     } else if (current_cpu == "arm") {
       if (is_clang && !is_android && !is_chromeos_device) {
         cflags += [ "--target=arm-linux-gnueabihf" ]
--- a/build/config/win/BUILD.gn.orig	2026-01-06 16:50:30.000000000 -0800
+++ b/build/config/win/BUILD.gn	2026-01-19 12:00:45.456439515 -0800
@@ -42,6 +42,8 @@ declare_args() {
   #  and with this switch, clang emits it like this:
   #    foo/bar.cc:12:34: error: something went wrong
   use_clang_diagnostics_format = false
+
+  use_sse3 = true
 }
 
 # This is included by reference in the //build/config/compiler config that
@@ -120,7 +122,9 @@ config("compiler") {
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      if (use_sse3) {
+        cflags += [ "-msse3" ]
+      }
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
