--- a/third_party/libvpx/BUILD.gn.orig	2025-05-06 18:17:34.491392965 -0700
+++ b/third_party/libvpx/BUILD.gn	2025-05-07 09:51:46.451365859 -0700
@@ -5,6 +5,7 @@
 import("//build/config/android/config.gni")
 import("//build/config/arm.gni")
 import("//build/config/chromeos/ui_mode.gni")
+import("//build/config/simd.gni")
 import("//testing/test.gni")
 import("//third_party/libvpx/libvpx_srcs.gni")
 import("//third_party/libvpx/libvpx_test_srcs.gni")
@@ -614,39 +615,83 @@ static_library("libvpx") {
   configs += [ ":libvpx_config" ]
   deps = []
   if (current_cpu == "x86" || (current_cpu == "x64" && !is_msan)) {
-    deps += [
-      ":libvpx_asm",
-      ":libvpx_intrinsics_avx",
-      ":libvpx_intrinsics_avx2",
-      ":libvpx_intrinsics_avx512",
-      ":libvpx_intrinsics_mmx",
-      ":libvpx_intrinsics_sse2",
-      ":libvpx_intrinsics_sse4_1",
-      ":libvpx_intrinsics_ssse3",
-    ]
+    if (use_mmx && use_sse2 && use_sse3 && use_ssse3) {
+      deps += [
+        ":libvpx_asm",
+      ]
+    }
+    if (use_avx) {
+       deps += [
+       ":libvpx_intrinsics_avx",
+      ]
+    }
+    if (use_avx2) {
+      deps += [
+        ":libvpx_intrinsics_avx2",
+      ]
+    }
+    if (use_avx512) {
+      deps += [
+        ":libvpx_intrinsics_avx512",
+      ]
+    }
+    if (use_mmx) {
+      deps += [
+        ":libvpx_intrinsics_mmx",
+      ]
+    }
+    if (use_sse2) {
+      deps += [
+        ":libvpx_intrinsics_sse2",
+      ]
+    }
+    if (use_sse4_1) {
+      deps += [
+        ":libvpx_intrinsics_sse4_1",
+      ]
+    }
+    if (use_ssse3) {
+      deps += [
+        ":libvpx_intrinsics_ssse3",
+      ]
+    )
   }
   if (cpu_arch_full == "arm-neon-highbd" || cpu_arch_full == "arm-neon" ||
       cpu_arch_full == "arm-neon-cpu-detect" || current_cpu == "arm64") {
     deps += [ ":libvpx_intrinsics_neon" ]
   }
   if (current_cpu == "arm64") {
-    deps += [ ":libvpx_intrinsics_neon_dotprod" ]
-    deps += [ ":libvpx_intrinsics_neon_i8mm" ]
+    if (use_neon && use_dotprod) {
+      deps += [ ":libvpx_intrinsics_neon_dotprod" ]
+    }
+    if (use_neon && use_i8mm) {
+      deps += [ ":libvpx_intrinsics_neon_i8mm" ]
+    }
     if (!is_win) {
-      deps += [
-        ":libvpx_intrinsics_sve",
-        ":libvpx_intrinsics_sve2",
-      ]
+      if (use_sve) {
+        deps += [
+          ":libvpx_intrinsics_sve",
+        ]
+      }
+      if (use_sve2) {
+        deps += [
+          ":libvpx_intrinsics_sve2",
+        ]
+      }
     }
   }
   if (is_android) {
     deps += [ "//third_party/cpu_features:ndk_compat" ]
   }
   if (current_cpu == "arm" && arm_assembly_sources != []) {
-    deps += [ ":libvpx_assembly_arm" ]
+    if (use_neon) {
+      deps += [ ":libvpx_assembly_arm" ]
+    }
   }
   if (current_cpu == "loong64") {
-    deps += [ ":libvpx_loongarch_lsx" ]
+    if (use_lsx) {
+      deps += [ ":libvpx_loongarch_lsx" ]
+    }
   }
 
   public_configs = [ ":libvpx_public_config" ]
