--- a/build/config/compiler/compiler.gni.orig	2024-08-21 15:46:04.998060500 -0700
+++ b/build/config/compiler/compiler.gni	2024-08-27 13:44:13.260481947 -0700
@@ -36,6 +36,9 @@ declare_args() {
   # The default linker everywhere else.
   use_lld = is_clang && current_os != "zos"
 
+  # enable together with use_lld
+  use_mold = false
+
   # If true, optimize for size.
   # Default to favoring speed over size for platforms not listed below.
   optimize_for_size =
--- a/build/config/compiler/pgo/BUILD.gn.orig	2024-08-21 15:46:04.998060500 -0700
+++ b/build/config/compiler/pgo/BUILD.gn	2024-08-27 13:53:44.162811153 -0700
@@ -148,16 +148,21 @@ config("pgo_optimization_flags") {
     # Enable basic block layout based on the extended TSP problem. This aims to
     # improve icache utilization and reduce the binary size.
     if (use_thin_lto) {
+      assert(!use_mold, "use_mold=true is not allowed with use_thin_lto=true")
       if (is_win) {
         ldflags = [ "-mllvm:-enable-ext-tsp-block-placement=1" ]
       } else {
         ldflags = [ "-Wl,-mllvm,-enable-ext-tsp-block-placement=1" ]
       }
     } else {
-      cflags += [
-        "-mllvm",
-        "-enable-ext-tsp-block-placement=1",
-      ]
+      if (use_mold) {
+        # Pass
+      } else {
+        cflags += [
+          "-mllvm",
+          "-enable-ext-tsp-block-placement=1",
+        ]
+      }
     }
   }
 }
--- a/build/config/compiler/BUILD.gn.orig	2024-08-21 15:46:04.998060500 -0700
+++ b/build/config/compiler/BUILD.gn	2024-08-27 13:52:13.156350770 -0700
@@ -571,7 +571,10 @@ config("compiler") {
       "-mllvm",
       "-instcombine-lower-dbg-declare=0",
     ]
-    if (!is_debug && use_thin_lto && is_a_target_toolchain) {
+    if (use_mold) {
+      assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+      # Pass
+    } else if (!is_debug && use_thin_lto && is_a_target_toolchain) {
       if (is_win) {
         ldflags += [ "-mllvm:-instcombine-lower-dbg-declare=0" ]
       } else {
@@ -588,7 +591,10 @@ config("compiler") {
         "-mllvm",
         "-split-threshold-for-reg-with-hint=0",
       ]
-      if (use_thin_lto && is_a_target_toolchain) {
+      if (use_mold) {
+        assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+        # Pass
+      } else if (use_thin_lto && is_a_target_toolchain) {
         if (is_win) {
           ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
         } else {
@@ -711,7 +717,10 @@ config("compiler") {
   #
   # [0]: https://pinpoint-dot-chromeperf.appspot.com/job/15efb0313e0000
   # [1]: https://pinpoint-dot-chromeperf.appspot.com/job/157f0b42be0000
-  if (!is_debug && use_thin_lto && is_a_target_toolchain) {
+  if (use_mold) {
+    assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+    # Pass
+  } else if (!is_debug && use_thin_lto && is_a_target_toolchain) {
     assert(use_lld, "LTO is only supported with lld")
 
     cflags += [
@@ -928,7 +937,10 @@ config("compiler") {
     assert(
         is_android && host_os == "linux",
         "MLGO is currently only supported for targeting Android on a linux host")
-    if (use_thin_lto) {
+    if (use_mold) {
+      assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+      # Pass
+    } else if (use_thin_lto) {
       ldflags += [ "-Wl,-mllvm,-enable-ml-inliner=release" ]
     }
   }
@@ -1502,7 +1514,12 @@ config("compiler_codegen") {
     # This can be removed once https://bugs.llvm.org/show_bug.cgi?id=40348
     # has been resolved, and -mno-outline is obeyed by the linker during
     # ThinLTO.
-    ldflags += [ "-Wl,-mllvm,-enable-machine-outliner=never" ]
+    if (use_mold) {
+      assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+      # Pass
+    } else {
+      ldflags += [ "-Wl,-mllvm,-enable-machine-outliner=never" ]
+    }
   }
 
   asmflags = cflags
--- a/third_party/llvm/llvm/utils/gn/build/BUILD.gn.orig	2024-08-21 15:56:56.796674000 -0700
+++ b/third_party/llvm/llvm/utils/gn/build/BUILD.gn	2024-08-28 23:31:12.573530134 -0700
@@ -38,6 +38,8 @@ declare_args() {
 
   # The version of host gcc. Ignored if is_clang is true.
   gcc_version = 9
+
+  use_mold = false
 }
 
 assert(!llvm_build_instrumented_coverage || is_clang,
@@ -269,7 +271,9 @@ config("compiler_defaults") {
   }
 
   # On Windows, the linker is not invoked through the compiler driver.
-  if (use_lld && current_os != "win") {
+  if (use_mold && current_os != "win") {
+    ldflags += [ "-fuse-ld=mold" ]
+  } else if (use_lld && current_os != "win") {
     ldflags += [ "-fuse-ld=lld" ]
   }
 
--- a/third_party/rust-src/src/llvm-project/llvm/utils/gn/build/BUILD.gn.orig	2024-08-21 16:00:28.649322300 -0700
+++ b/third_party/rust-src/src/llvm-project/llvm/utils/gn/build/BUILD.gn	2024-08-28 23:31:15.385485283 -0700
@@ -38,6 +38,8 @@ declare_args() {
 
   # The version of host gcc. Ignored if is_clang is true.
   gcc_version = 9
+
+  use_mold = false
 }
 
 assert(!llvm_build_instrumented_coverage || is_clang,
@@ -274,7 +276,9 @@ config("compiler_defaults") {
   }
 
   # On Windows, the linker is not invoked through the compiler driver.
-  if (use_lld && current_os != "win") {
+  if (use_mold && current_os != "win") {
+    ldflags += [ "-fuse-ld=mold" ]
+  } else if (use_lld && current_os != "win") {
     ldflags += [ "-fuse-ld=lld" ]
   }
 
--- a/third_party/instrumented_libs/noble/BUILD.gn.orig	2024-08-21 15:47:04.259141700 -0700
+++ b/third_party/instrumented_libs/noble/BUILD.gn	2024-08-28 23:30:49.949890977 -0700
@@ -5,6 +5,10 @@
 import("//build/config/python.gni")
 import("//build/config/sanitizers/sanitizers.gni")
 
+declare_args() {
+  use_mold = false
+}
+
 static_library("stat_wrapper") {
   sources = [ "stat_wrapper.c" ]
 }
@@ -121,7 +125,17 @@ template("instrumented_library") {
     ]
     package_ldflags = [
       "-fsanitize=memory",
-      "-fuse-ld=lld",
+    ]
+    if (use_mold) {
+      package_ldflags += [
+        "-fuse-ld=mold",
+      ]
+    } else {
+      package_ldflags += [
+        "-fuse-ld=lld",
+      ]
+    }
+    package_ldflags += [
       "-Wl,--wrap=stat",
       "-Wl,--wrap=fstat",
       "-Wl,--wrap=lstat",
--- a/build/config/compiler/BUILD.gn.orig	2024-08-27 22:30:13.510510204 -0700
+++ b/build/config/compiler/BUILD.gn	2024-08-28 23:31:09.989571349 -0700
@@ -200,6 +200,8 @@ declare_args() {
   # debugging information which is required by some specialized
   # debugging tools.
   simple_template_names = is_clang && !is_nacl && !is_win && !is_apple
+
+  use_mold = 1
 }
 
 declare_args() {
@@ -371,7 +373,9 @@ config("compiler") {
       }
     }
 
-    if (use_lld) {
+    if (use_mold) {
+      ldflags += [ "-fuse-ld=mold" ]
+    } if (use_lld) {
       ldflags += [ "-fuse-ld=lld" ]
       if (lld_path != "") {
         ldflags += [ "-B$lld_path" ]
