--- a/third_party/fuzztest/BUILD.gn.orig	2025-04-27 16:35:54.000000000 -0700
+++ b/third_party/fuzztest/BUILD.gn	2025-05-07 15:58:06.706462449 -0700
@@ -13,6 +13,7 @@
 
 import("//build/config/android/config.gni")
 import("//build/config/sanitizers/sanitizers.gni")
+import("//build/config/simd.gni")
 import("//build_overrides/build.gni")
 
 # Retain full optimization level for the engine parts independent of the
@@ -41,7 +42,9 @@ config("fuzztest_internal_config") {
 
   # int_utils.h depends on an SSE 4.2 intrinsic.
   if (current_cpu == "x64") {
-    cflags += [ "-msse4.2" ]
+    if (use_sse4_2) {
+      cflags += [ "-msse4.2" ]
+    }
   }
 }
 
