--- a/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc.orig	2024-09-10 11:42:04.774961000 -0700
+++ b/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc	2024-09-18 13:03:47.782161002 -0700
@@ -5,6 +5,7 @@
 #include "chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.h"
 
 #include "base/functional/bind.h"
+#include "base/metrics/histogram_functions.h"
 #include "base/metrics/histogram_macros.h"
 #include "chrome/browser/accessibility/accessibility_state_utils.h"
 #include "chrome/browser/net/system_network_context_manager.h"
