diff -u b/chrome/browser/pdf/BUILD.gn b/chrome/browser/pdf/BUILD.gn
--- b/chrome/browser/pdf/BUILD.gn	2026-02-25 23:00:16.084961151 -0800
+++ b/chrome/browser/pdf/BUILD.gn	2026-02-25 22:48:20.891478619 -0800
@@ -2,10 +2,11 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//chrome/common/features.gni")
 import("//pdf/features.gni")
 
 assert(enable_pdf)
 
 source_set("pdf") {
   sources = [
     "chrome_pdf_stream_delegate.cc",
@@ -31,7 +37,13 @@
     "//build:chromeos_buildflags",
     "//chrome/browser:browser_process",
     "//chrome/browser:resources",
-    "//chrome/browser/glic",
+  ]
+  if (enable_glic || enable_glic_android) {
+    deps += [
+      "//chrome/browser/glic",
+    ]
+  }
+  deps += [
     "//chrome/browser/pdf:pdf_pref_names",
     "//chrome/browser/profiles:profile",
     "//chrome/browser/resources/pdf:resources",
