--- a/v8/src/api/api-arguments.h.orig	2026-02-12 11:44:24.000000000 -0800
+++ b/v8/src/api/api-arguments.h	2026-02-18 20:02:30.346449977 -0800
@@ -193,8 +193,15 @@ class PropertyCallbackArguments final :
   template <typename T>
   static DirectHandle<AccessorInfo> GetAccessorInfo(
       const PropertyCallbackInfo<T>& info) {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
     return Cast<AccessorInfo>(
         DirectHandle<Object>::FromSlot(&info.args_[kCallbackInfoIndex]));
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
   }
 
   // Returns whether given v8::PropertyCallbackInfo<T> object is named/indexed.
@@ -324,7 +331,14 @@ class FunctionCallbackArguments : public
   // v8::FunctionCallbackInfo<T>.
   template <typename T>
   static Tagged<Object> GetTarget(const FunctionCallbackInfo<T>& info) {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
     return Tagged<Object>(info.values_[kTargetIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
   }
 
  private:
--- a/v8/include/v8-function-callback.h.orig	2026-02-18 20:12:00.397218520 -0800
+++ b/v8/include/v8-function-callback.h	2026-02-18 20:04:07.096282764 -0800
@@ -673,7 +673,14 @@ Local<Value> FunctionCallbackInfo<T>::op
 
 template <typename T>
 Local<Object> FunctionCallbackInfo<T>::This() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return Local<Object>::FromSlot(&values_[kReceiverIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
@@ -689,7 +696,14 @@ Local<Value> FunctionCallbackInfo<T>::Ne
 
 template <typename T>
 Local<Value> FunctionCallbackInfo<T>::Data() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   auto target = Local<v8::Data>::FromSlot(&values_[kTargetIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
   return api_internal::GetFunctionTemplateData(GetIsolate(), target);
 }
 
@@ -700,12 +714,26 @@ Isolate* FunctionCallbackInfo<T>::GetIso
 
 template <typename T>
 ReturnValue<T> FunctionCallbackInfo<T>::GetReturnValue() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return ReturnValue<T>(&values_[kReturnValueIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
 bool FunctionCallbackInfo<T>::IsConstructCall() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return I::SmiValue(values_[kFrameTypeIndex]) == I::kFrameTypeApiConstructExit;
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
@@ -715,8 +743,15 @@ int FunctionCallbackInfo<T>::Length() co
 
 template <typename T>
 bool PropertyCallbackInfo<T>::IsNamed() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return I::SmiValue(args_[kFrameTypeIndex]) ==
          I::kFrameTypeApiNamedAccessorExit;
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
@@ -726,7 +761,14 @@ Isolate* PropertyCallbackInfo<T>::GetIso
 
 template <typename T>
 Local<Value> PropertyCallbackInfo<T>::Data() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   internal::Address callback_info = args_[kCallbackInfoIndex];
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
   internal::Address data =
       I::ReadTaggedPointerField(callback_info, I::kCallbackInfoDataOffset);
   return Local<Value>::New(GetIsolate(), data);
@@ -734,26 +776,54 @@ Local<Value> PropertyCallbackInfo<T>::Da
 
 template <typename T>
 Local<Object> PropertyCallbackInfo<T>::This() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return Local<Object>::FromSlot(&args_[kThisIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
 Local<Object> PropertyCallbackInfo<T>::HolderV2() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return Local<Object>::FromSlot(&args_[kHolderIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
 ReturnValue<T> PropertyCallbackInfo<T>::GetReturnValue() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return ReturnValue<T>(&args_[kReturnValueIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
 bool PropertyCallbackInfo<T>::ShouldThrowOnError() const {
   if constexpr (!HasShouldThrowOnError()) return false;
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   if (args_[kShouldThrowOnErrorIndex] !=
       I::IntegralToSmi(I::kInferShouldThrowMode)) {
     return args_[kShouldThrowOnErrorIndex] != I::IntegralToSmi(I::kDontThrow);
   }
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
   return v8::internal::ShouldThrowOnError(
       reinterpret_cast<v8::internal::Isolate*>(GetIsolate()));
 }
--- a/third_party/swiftshader/third_party/llvm-subzero/lib/Support/regcomp.c.orig	2026-02-12 11:44:24.000000000 -0800
+++ b/third_party/swiftshader/third_party/llvm-subzero/lib/Support/regcomp.c	2026-02-16 14:30:34.348287424 -0800
@@ -222,7 +222,14 @@ llvm_regcomp(llvm_regex_t *preg, const c
 	g->mlen = 0;
 	g->nsub = 0;
 	g->ncategories = 1;	/* category 0 is "everything else" */
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
 	g->categories = &g->catspace[-(CHAR_MIN)];
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 	(void) memset((char *)g->catspace, 0, NC*sizeof(cat_t));
 	g->backrefs = 0;
 
