--- a/third_party/libaom/BUILD.gn.orig	2025-05-30 12:50:32.000000000 -0700
+++ b/third_party/libaom/BUILD.gn	2025-06-03 15:37:01.078617647 -0700
@@ -5,6 +5,7 @@
 import("//build/buildflag_header.gni")
 import("//build/config/android/config.gni")
 import("//build/config/arm.gni")
+import("//build/config/simd.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//testing/test.gni")
 import("//third_party/libaom/libaom_srcs.gni")
@@ -338,35 +339,83 @@ static_library("libaom") {
   }
   deps = []
   if (current_cpu == "x86" || (current_cpu == "x64" && !is_msan)) {
-    deps += [
-      ":libaom_intrinsics_avx",
-      ":libaom_intrinsics_avx2",
-      ":libaom_intrinsics_sse2",
-      ":libaom_intrinsics_sse3",
-      ":libaom_intrinsics_sse4_1",
-      ":libaom_intrinsics_sse4_2",
-      ":libaom_intrinsics_ssse3",
-      ":libaom_nasm",
-    ]
+    if (use_avx) {
+      deps += [
+        ":libaom_intrinsics_avx",
+      ]
+    }
+    if (use_avx2) {
+      deps += [
+        ":libaom_intrinsics_avx2",
+      ]
+    }
+    if (use_sse2) {
+      deps += [
+        ":libaom_intrinsics_sse2",
+      ]
+    }
+    if (use_sse3) {
+      deps += [
+        ":libaom_intrinsics_sse3",
+      ]
+    }
+    if (use_sse4_1) {
+      deps += [
+        ":libaom_intrinsics_sse4_1",
+      ]
+    }
+    if (use_sse4_2) {
+      deps += [
+        ":libaom_intrinsics_sse4_2",
+      ]
+    }
+    if (use_ssse3) {
+      deps += [
+        ":libaom_intrinsics_ssse3",
+      ]
+    }
+    if (use_sse2 && use_ssse3) {
+      deps += [
+        ":libaom_nasm",
+      ]
+    }
   }
   if (current_cpu == "arm64" || current_cpu == "arm64e" ||
       cpu_arch_full == "arm-neon" || cpu_arch_full == "arm-neon-cpu-detect") {
-    deps += [ ":libaom_intrinsics_neon" ]
+    if (use_neon) {
+      deps += [ ":libaom_intrinsics_neon" ]
+    }
   }
   if (current_cpu == "arm64" || current_cpu == "arm64e") {
     # This is needed by all arm boards due to aom_arm_cpu_caps()
     sources += [ "source/libaom/aom_ports/aarch64_cpudetect.c" ]
-    deps += [
-      ":libaom_intrinsics_arm_crc32",
-      ":libaom_intrinsics_neon_dotprod",
-      ":libaom_intrinsics_neon_i8mm",
-    ]
-    if (!is_win) {
+    if (use_crc32) {
       deps += [
-        ":libaom_intrinsics_sve",
-        ":libaom_intrinsics_sve2",
+        ":libaom_intrinsics_arm_crc32",
       ]
     }
+    if (use_neon && use_dotprod) {
+      deps += [
+        ":libaom_intrinsics_neon_dotprod",
+      ]
+    }
+    if (use_neon && use_dotprod && use_i8mm) {
+      deps += [
+        ":libaom_intrinsics_neon_i8mm",
+      ]
+    }
+    if (!is_win) {
+      if (use_dotprod && use_i8mm && use_sve) {
+        deps += [
+          ":libaom_intrinsics_sve",
+        ]
+      }
+      if (use_i8mm && use_sve2) {
+        deps += [
+          ":libaom_intrinsics_sve2",
+        ]
+      }
+    }
   }
   if (current_cpu == "arm") {
     # This is needed by all arm boards due to aom_arm_cpu_caps()
