--- a/third_party/pdfium/BUILD.gn.orig	2026-01-26 07:39:54.652291968 -0800
+++ b/third_party/pdfium/BUILD.gn	2026-01-26 07:42:34.857385448 -0800
@@ -4,6 +4,7 @@
 
 import("//build/config/clang/clang.gni")
 import("//build/config/gclient_args.gni")
+import("//build/config/linux/pkg_config.gni")
 import("//testing/test.gni")
 import("pdfium.gni")
 
@@ -24,6 +25,20 @@ group("freetype_common") {
   }
 }
 
+if (use_system_lcms2) {
+  pkg_config("lcms2_from_pkgconfig") {
+    defines = [ "USE_SYSTEM_LCMS2" ]
+    packages = [ "lcms2" ]
+  }
+}
+
+if (use_system_libopenjpeg2) {
+  pkg_config("libopenjpeg2_from_pkgconfig") {
+    defines = [ "USE_SYSTEM_LIBOPENJPEG2" ]
+    packages = [ "libopenjp2" ]
+  }
+}
+
 config("pdfium_common_config") {
   cflags = []
   cflags_cc = []
@@ -31,7 +46,13 @@ config("pdfium_common_config") {
   include_dirs = [ "." ]
   defines = []
 
-  if (!use_system_libopenjpeg2) {
+  if (use_system_lcms2) {
+    public_configs = [ ":lcms2_from_pkgconfig" ]
+  }
+
+  if (use_system_libopenjpeg2) {
+    configs = [ ":libopenjpeg2_from_pkgconfig" ]
+  } else {
     defines += [ "OPJ_STATIC" ]
   }
 
