--- a/third_party/pdfium/skia/BUILD.gn.orig	2025-04-27 16:35:54.000000000 -0700
+++ b/third_party/pdfium/skia/BUILD.gn	2025-05-07 00:20:15.899758513 -0700
@@ -4,6 +4,7 @@
 
 import("//build/config/features.gni")
 import("//build/config/freetype/freetype.gni")
+import("//build/config/simd.gni")
 import("//build/config/ui.gni")
 import("//testing/test.gni")
 import("//third_party/skia/gn/codec.gni")
@@ -413,7 +414,9 @@ skia_source_set("skia_opts") {
   defines = []
 
   if (current_cpu == "x86" || current_cpu == "x64") {
-    deps = [ ":skia_opts_hsw" ]
+    if (use_avx2 && use_bmi && use_bmi2 && use_f16c && use_fma) {
+      deps = [ ":skia_opts_hsw" ]
+    }
   } else if (current_cpu == "arm") {
     # The assembly uses the frame pointer register (r7 in Thumb/r11 in
     # ARM), the compiler doesn't like that.
@@ -508,9 +511,17 @@ source_set("skcms_TransformBaseline_and_
 source_set("skcms") {
   deps = [
     ":skcms_TransformBaseline_and_public",
-    ":skcms_TransformHsw",
-    ":skcms_TransformSkx",
   ]
+  if (use_sse2 && use_sse3 && use_ssse3 && use_sse4_1 && use_sse4_2 && use_pclmul && use_avx && use_avx2 && use_bmi && use_bmi2 && use_fma) {
+    deps += [
+      ":skcms_TransformHsw",
+    ]
+  }
+  if (use_sse2 && use_sse3 && use_ssse3 && use_sse4_1 && use_sse4_2 && use_pclmul && use_avx && use_avx2 && use_bmi && use_bmi2 && use_fma && use_aes && use_avx512) {
+    deps += [
+      ":skcms_TransformSkx",
+    ]
+  }
   public =
       rebase_path(skcms_public_headers, ".", "//third_party/skia/modules/skcms")
   include_dirs = [ "//third_party/skia/modules/skcms" ]
