--- a/build/config/compiler/BUILD.gn.orig	2026-03-10 19:07:32.179017738 -0700
+++ b/build/config/compiler/BUILD.gn	2026-03-10 19:23:48.324802785 -0700
@@ -1036,7 +1036,9 @@ config("compiler") {
   # C11/C++11 compiler flags setup.
   # ---------------------------
   if (is_linux || is_chromeos || is_android || current_os == "aix") {
-    if (is_clang) {
+    if (is_clang && use_system_clang) {
+      standard_prefix = "gnu"
+    } else if (is_clang) {
       standard_prefix = "c"
 
       # Since we build with -std=c* and not -std=gnu*, _GNU_SOURCE will not be
