--- a/build/config/compiler/compiler.gni.orig	2026-03-09 12:52:02.000000000 -0700
+++ b/build/config/compiler/compiler.gni	2026-03-12 08:22:08.095823559 -0700
@@ -28,11 +28,15 @@ if (is_apple) {
 }
 
 declare_args() {
+  use_mold = false
+}
+
+declare_args() {
   # Set to true to use lld, the LLVM linker.
   # In late bring-up on macOS (see docs/mac_lld.md).
   # Tentatively used on iOS.
   # The default linker everywhere else.
-  use_lld = is_clang && current_os != "zos"
+  use_lld = is_clang && current_os != "zos" && !use_mold
 
   # If false, build using C++20; otherwise, build using C++23.
   use_cxx23 = true
--- a/build/config/compiler/pgo/BUILD.gn.orig	2026-03-09 12:52:01.000000000 -0700
+++ b/build/config/compiler/pgo/BUILD.gn	2026-03-12 08:22:08.095823559 -0700
@@ -9,6 +9,10 @@ import("//build/config/compiler/pgo/pgo.
 import("//build/config/features.gni")
 import("//build/toolchain/toolchain.gni")
 
+declare_args() {
+  #use_mold = false
+}
+
 # Configuration that enables PGO instrumentation.
 config("pgo_instrumentation_flags") {
   visibility = [ ":default_pgo_flags" ]
@@ -177,16 +181,21 @@ config("pgo_optimization_flags") {
     # Enable basic block layout based on the extended TSP problem. This aims to
     # improve icache utilization and reduce the binary size.
     if (use_thin_lto) {
+      assert(!use_mold, "use_mold=true is not allowed with use_thin_lto=true")
       if (is_win) {
         ldflags = [ "-mllvm:-enable-ext-tsp-block-placement=1" ]
       } else {
         ldflags = [ "-Wl,-mllvm,-enable-ext-tsp-block-placement=1" ]
       }
     } else {
-      cflags += [
-        "-mllvm",
-        "-enable-ext-tsp-block-placement=1",
-      ]
+      if (use_mold) {
+        # Pass
+      } else {
+        cflags += [
+          "-mllvm",
+          "-enable-ext-tsp-block-placement=1",
+        ]
+      }
     }
   }
 }
--- a/build/config/compiler/BUILD.gn.orig	2026-03-12 08:20:27.907083115 -0700
+++ b/build/config/compiler/BUILD.gn	2026-03-12 08:22:38.218844032 -0700
@@ -210,6 +210,8 @@ declare_args() {
   # away from the opt-in wexit_time_destructors config, and into the default
   # compiler config.
   is_wexit_time_destructors_default = true
+
+  #use_mold = false
 }
 
 declare_args() {
@@ -471,7 +473,9 @@ config("compiler") {
       }
     }
 
-    if (use_lld) {
+    if (use_mold) {
+      ldflags += [ "-fuse-ld=mold" ]
+    } else if (use_lld) {
       ldflags += [ "-fuse-ld=lld" ]
       if (lld_path != "") {
         ldflags += [ "-B" + rebase_path(lld_path, root_build_dir) ]
@@ -704,7 +708,10 @@ config("compiler") {
       "-mllvm",
       "-instcombine-lower-dbg-declare=0",
     ]
-    if (!is_debug && use_thin_lto && is_a_target_toolchain) {
+    if (use_mold) {
+      assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+      # Pass
+    } else if (!is_debug && use_thin_lto && is_a_target_toolchain) {
       if (is_win) {
         ldflags += [ "-mllvm:-instcombine-lower-dbg-declare=0" ]
       } else {
@@ -719,7 +726,10 @@ config("compiler") {
         "-mllvm",
         "-split-threshold-for-reg-with-hint=0",
       ]
-      if (use_thin_lto && is_a_target_toolchain) {
+      if (use_mold) {
+        assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+        # Pass
+      } else if (use_thin_lto && is_a_target_toolchain) {
         if (is_win) {
           ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
         } else {
@@ -838,7 +848,10 @@ config("compiler") {
   #
   # [0]: https://pinpoint-dot-chromeperf.8pp2p8t.qjz9zk/job/15efb0313e0000
   # [1]: https://pinpoint-dot-chromeperf.8pp2p8t.qjz9zk/job/157f0b42be0000
-  if (!is_debug && use_thin_lto && is_a_target_toolchain) {
+  if (use_mold) {
+    assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+    # Pass
+  } else if (!is_debug && use_thin_lto && is_a_target_toolchain) {
     assert(use_lld, "LTO is only supported with lld")
 
     cflags += [
@@ -1055,7 +1068,10 @@ config("compiler") {
     assert(
         is_android && host_os == "linux",
         "MLGO is currently only supported for targeting Android on a linux host")
-    if (use_thin_lto) {
+    if (use_mold) {
+      assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+      # Pass
+    } else if (use_thin_lto) {
       ldflags += [ "-Wl,-mllvm,-enable-ml-inliner=release" ]
       if (is_high_end_android) {
         # Besides using the arm64 - trained model, instruct the inline advisor
@@ -1644,7 +1660,12 @@ config("compiler_codegen") {
     # has been resolved, and -mno-outline is obeyed by the linker during
     # ThinLTO.
     if (use_thin_lto && use_lld) {
-      ldflags += [ "-Wl,-mllvm,-enable-machine-outliner=never" ]
+      if (use_mold) {
+        assert(!use_thin_lto, "use_thin_lto=true is not allowed with use_mold=true")
+        # Pass
+      } else {
+        ldflags += [ "-Wl,-mllvm,-enable-machine-outliner=never" ]
+      }
     }
   }
 
--- a/third_party/instrumented_libs/noble/BUILD.gn.orig	2026-03-09 12:52:01.000000000 -0700
+++ b/third_party/instrumented_libs/noble/BUILD.gn	2026-03-12 08:22:08.095823559 -0700
@@ -9,6 +9,10 @@ static_library("stat_wrapper") {
   sources = [ "stat_wrapper.c" ]
 }
 
+declare_args() {
+  use_mold = false
+}
+
 group("locally_built") {
   visibility = [ "//third_party/instrumented_libs:deps" ]
 
@@ -121,7 +125,17 @@ template("instrumented_library") {
     ]
     package_ldflags = [
       "-fsanitize=memory",
-      "-fuse-ld=lld",
+    ]
+    if (use_mold) {
+      package_ldflags += [
+        "-fuse-ld=mold",
+      ]
+    } else {
+      package_ldflags += [
+        "-fuse-ld=lld",
+      ]
+    }
+    package_ldflags += [
       "-Wl,--wrap=stat",
       "-Wl,--wrap=fstat",
       "-Wl,--wrap=lstat",
