--- a/build/config/compiler/BUILD.gn.orig	2026-03-09 22:17:36.714826618 -0700
+++ b/build/config/compiler/BUILD.gn	2026-03-09 22:32:41.362863567 -0700
@@ -53,6 +53,7 @@ declare_args() {
 
   enable_frame_pointers = true
   use_system_clang = false
+  use_glibc = true
 }
 
 declare_args() {
@@ -732,6 +733,29 @@ config("compiler") {
       cflags += [ "-ftrivial-auto-var-init=zero", "-Wno-unused-command-line-argument" ]
     }
 
+    if (is_clang && use_system_clang) {
+      if (use_glibc) {
+        cflags += [ "-DMB_LEN_MAX=16" ]
+        cflags += [ "-D_POSIX_SEM_VALUE_MAX=32767" ]
+        if (current_cpu == "arm64"
+            || current_cpu == "loong64"
+            || current_cpu == "mips64"
+            || current_cpu == "mips64el"
+            || current_cpu == "ppc64"
+            || current_cpu == "riscv64"
+            || current_cpu == "s390x"
+            || current_cpu == "x64") {
+          cflags += [ "-DSSIZE_MAX=9223372036854775807" ]
+        } else {
+          cflags += [ "-DSSIZE_MAX=2147483647" ]
+        }
+      }
+      if (is_linux) {
+        cflags += [ "-DPATH_MAX=4096" ]
+        cflags += [ "-DNAME_MAX=255" ]
+      }
+    }
+
     # Linker warnings.
     if (fatal_linker_warnings && !is_apple && current_os != "aix" &&
         current_os != "zos") {
