--- a/build/config/BUILD.gn.orig	2022-08-01 12:59:58.000000000 -0700
+++ b/build/config/BUILD.gn	2022-08-21 21:54:39.803579619 -0700
@@ -23,6 +23,9 @@ if (is_android) {
   import("//build/config/android/abi.gni")
 }
 
+declare_args() {
+  gcc_pgi = false
+}
 # ==============================================
 #   PLEASE DO NOT ADD MORE THINGS TO THIS LIST
 # ==============================================
@@ -224,6 +227,9 @@ config("default_libs") {
       "pthread",
       "rt",
     ]
+    if (gcc_pgi) {
+      libs += [ "gcov" ]
+    }
   }
 }
 
