--- a/components/spellcheck/spellcheck_build_features.gni.orig	2024-09-15 16:12:49.050637383 -0700
+++ b/components/spellcheck/spellcheck_build_features.gni	2024-09-15 16:13:04.666379154 -0700
@@ -5,7 +5,7 @@
 import("//build/config/features.gni")
 
 # Enable the spell checker.
-enable_spellcheck = use_blink
+enable_spellcheck = false
 
 # Use the operating system's spellchecker rather than hunspell. This does
 # not affect the "red underline" spellchecker which can consult Google's
