--- a/net/cookies/cookie_util.cc.orig	2026-03-15 10:02:24.928185341 -0700
+++ b/net/cookies/cookie_util.cc	2026-03-15 10:05:35.790660757 -0700
@@ -840,8 +840,9 @@ bool IsCookiePartitionedValid(
 
 void ParseRequestCookieLine(std::string_view header_value,
                             ParsedRequestCookies* parsed_cookies) {
-  std::string::const_iterator i = header_value.begin();
-  while (i != header_value.end()) {
+  auto i = header_value.begin();
+  auto end_it = header_value.end();
+  while (i != end_it) {
     // Here we are at the beginning of a cookie.
 
     // Eat whitespace.
@@ -849,9 +850,9 @@ void ParseRequestCookieLine(std::string_
     if (i == header_value.end()) return;
 
     // Find cookie name.
-    std::string::const_iterator cookie_name_beginning = i;
+    auto cookie_name_beginning = i;
     while (i != header_value.end() && *i != '=') ++i;
-    auto cookie_name = std::string_view(cookie_name_beginning, i);
+    auto cookie_name = std::string_view(cookie_name_beginning, i - cookie_name_beginning);
 
     // Find cookie value.
     std::string_view cookie_value;
