--- v8/gni/v8.gni.orig	2024-09-13 10:38:56.572267440 -0700
+++ v8/gni/v8.gni	2024-09-13 11:13:08.929293690 -0700
@@ -189,7 +189,7 @@ if (v8_enable_backtrace == "") {
 # Chromium is configured to use the perfetto client library, v8 should also
 # use perfetto for tracing.
 if (build_with_chromium) {
-  v8_use_perfetto = true
+  v8_use_perfetto = false
 }
 
 # Includes profiles to optimize builtins if
