--- a/components/os_crypt/sync/BUILD.gn.orig	2026-01-05 14:53:38.000000000 -0800
+++ b/components/os_crypt/sync/BUILD.gn	2026-01-08 09:22:34.823538137 -0800
@@ -7,6 +7,10 @@ import("//build/config/features.gni")
 import("//build/config/ui.gni")
 import("//components/os_crypt/sync/features.gni")
 
+declare_args() {
+  use_system_libsecret = false
+}
+
 component("sync") {
   # New code should use os_crypt async.
   visibility = [
@@ -87,6 +91,11 @@ component("sync") {
       configs += [ "//build/config/linux:glib" ]
       deps += [ "//third_party/libsecret" ]
       defines += [ "USE_LIBSECRET" ]
+      if (use_system_libsecret) {
+        include_dirs = [
+          "/usr/include/libsecret-1"
+        ]
+      }
     }
     if (use_dbus) {
       sources += [
