--- a/build/toolchain/gcc_toolchain.gni.orig	2021-07-30 20:59:53.500101503 -0700
+++ b/build/toolchain/gcc_toolchain.gni	2021-07-30 21:01:45.431517867 -0700
@@ -410,7 +410,7 @@ template("gcc_toolchain") {
       rspfile = sofile + ".rsp"
       pool = "//build/toolchain:link_pool($default_toolchain)"
 
-      if (defined(invoker.strip)) {
+      if (defined(invoker.strip) && invoker.strip != "llvm-strip") {
         unstripped_sofile = "{{root_out_dir}}/lib.unstripped/$soname"
       } else {
         unstripped_sofile = sofile
