--- a/third_party/dav1d/BUILD.gn.orig	2025-05-17 02:52:34.716897118 -0700
+++ b/third_party/dav1d/BUILD.gn	2025-05-17 02:54:44.925110084 -0700
@@ -52,6 +52,16 @@ if (enable_nasm) {
   }
 }
 
+config("dav1d_asm_config") {
+  if (current_cpu == "x64") {
+    cflags = [ "-fPIC" ]
+    asmflags = [
+      "-f", "elf64",
+      "-DPIC",
+    ]
+  }
+}
+
 config("public_dav1d_config") {
   include_dirs = [ "version" ]
 
@@ -179,6 +189,10 @@ if (enable_nasm) {
       rebase_path("$platform_config_root/config.asm", root_build_dir),
     ]
 
+    if (current_cpu == "x64") {
+      configs += [ ":dav1d_asm_config" ]
+    }
+
     defines = []
     if (needs_stack_alignment) {
       defines += [ "STACK_ALIGNMENT=$stack_alignment" ]
