--- a/build/config/compiler/BUILD.gn.orig	2026-03-09 21:25:00.012295222 -0700
+++ b/build/config/compiler/BUILD.gn	2026-03-09 21:24:00.942120250 -0700
@@ -52,6 +52,7 @@ declare_args() {
   lld_path = default_lld_path
 
   enable_frame_pointers = true
+  use_system_clang = false
 }
 
 declare_args() {
@@ -343,7 +344,7 @@ assert(is_bot ||
 # Since this is referenced both when passing the file and when deciding which
 # warnings to enable, store it in a separate variable here.
 use_clang_warning_suppression_file =
-    is_clang && clang_warning_suppression_file != ""
+    is_clang && !use_system_clang && clang_warning_suppression_file != ""
 
 # default_include_dirs ---------------------------------------------------------
 #
