--- a/ui/webui/resources/tools/bundle_js.gni.orig	2025-06-04 19:05:34.138035568 -0700
+++ b/ui/webui/resources/tools/bundle_js.gni	2025-06-04 19:09:11.264898305 -0700
@@ -115,8 +115,18 @@ template("bundle_js") {
     }
     external_paths += [ "//resources/polymer/v3_0/|$polymer_path" ]
 
-    lit_path = rebase_path("${root_gen_dir}/third_party/lit/v3_0/minified/",
-                           root_build_dir)
+    minify = optimize
+    bundle = optimize && defined(invoker.optimize_webui_in_files)
+
+    if (bundle) {
+      lit_path = rebase_path("${root_gen_dir}/third_party/lit/v3_0/bundled/",
+                             root_build_dir)
+    }
+    if (minify) {
+      lit_path = rebase_path("${root_gen_dir}/third_party/lit/v3_0/minified/",
+                             root_build_dir)
+    }
+
     if (!is_chrome_untrusted) {
       external_paths += [ "chrome://resources/lit/v3_0/|$lit_path" ]
     }
