Subject:  Make SIMD instructions optional for zlib
Patch Author:  Orson Teodoro <orsonteodoro@hotmail.com>
--- a/third_party/skia/third_party/zlib/BUILD.gn.orig	2023-03-06 23:23:16.342635000 -0800
+++ b/third_party/skia/third_party/zlib/BUILD.gn	2023-03-11 18:41:47.396575686 -0800
@@ -6,6 +6,12 @@
 import("../third_party.gni")
 import("zlib.gni")
 
+declare_args() {
+  use_sse4_2 = true
+  use_ssse3 = true
+  use_sse2 = true
+}
+
 if (skia_use_system_zlib) {
   system("zlib") {
     libs = [ "z" ]
@@ -15,11 +21,11 @@ if (skia_use_system_zlib) {
   use_arm_neon_optimizations =
       (current_cpu == "arm" || current_cpu == "arm64") && !(is_win && !is_clang)
   use_x86_x64_optimizations =
-      (current_cpu == "x86" || current_cpu == "x64") && !is_ios
+      (current_cpu == "x86" || current_cpu == "x64") && !is_ios && use_sse2
 
   config("zlib_simd_config") {
     defines = []
-    if (use_x86_x64_optimizations) {
+    if (use_x86_x64_optimizations && use_ssse3) {
       if (is_win) {
         defines += [ "X86_WINDOWS" ]
       } else {
@@ -57,7 +63,7 @@ if (skia_use_system_zlib) {
   source_set("zlib_adler32_simd") {
     visibility = [ ":*" ]
     configs += [ ":zlib_simd_config" ]
-    if (use_x86_x64_optimizations) {
+    if (use_x86_x64_optimizations && use_ssse3) {
       defines = [ "ADLER32_SIMD_SSSE3" ]
       if (!is_win || is_clang) {
         cflags = [ "-mssse3" ]
@@ -87,7 +93,7 @@ if (skia_use_system_zlib) {
       }
     }
 
-    if (use_x86_x64_optimizations) {
+    if (use_x86_x64_optimizations && use_sse4_2) {
       defines = [ "CRC32_SIMD_SSE42_PCLMUL" ]
       if (!is_win || is_clang) {
         cflags = [
@@ -130,7 +136,7 @@ if (skia_use_system_zlib) {
   source_set("zlib_x86_x64_simd") {
     visibility = [ ":*" ]
     configs += [ ":zlib_simd_config" ]
-    if (use_x86_x64_optimizations) {
+    if (use_x86_x64_optimizations && use_sse4_2) {
       defines = [
         "CRC32_SIMD_SSE42_PCLMUL",
         "DEFLATE_FILL_WINDOW_SSE2",
--- a/third_party/zlib/BUILD.gn.orig	2023-03-06 23:21:31.309786300 -0800
+++ b/third_party/zlib/BUILD.gn	2023-03-11 18:41:05.688575147 -0800
@@ -4,6 +4,12 @@
 
 import("//build/config/compiler/compiler.gni")
 
+declare_args() {
+  use_sse4_2 = true
+  use_ssse3 = true
+  use_sse2 = true
+}
+
 if (build_with_chromium) {
   import("//testing/test.gni")
 }
@@ -67,10 +73,10 @@ if ((current_cpu == "arm" || current_cpu
 }
 
 use_x86_x64_optimizations =
-    (current_cpu == "x86" || current_cpu == "x64") && !is_ios
+    (current_cpu == "x86" || current_cpu == "x64") && !is_ios && use_sse2
 
 config("zlib_adler32_simd_config") {
-  if (use_x86_x64_optimizations) {
+  if (use_x86_x64_optimizations && use_ssse3) {
     defines = [ "ADLER32_SIMD_SSSE3" ]
     if (is_win) {
       defines += [ "X86_WINDOWS" ]
@@ -87,13 +93,13 @@ config("zlib_adler32_simd_config") {
 source_set("zlib_adler32_simd") {
   visibility = [ ":*" ]
 
-  if (use_x86_x64_optimizations) {
+  if (use_x86_x64_optimizations && use_ssse3) {
     sources = [
       "adler32_simd.c",
       "adler32_simd.h",
     ]
 
-    if (!is_win || is_clang) {
+    if ((!is_win || is_clang) && use_ssse3) {
       cflags = [ "-mssse3" ]
     }
   }
@@ -207,7 +213,7 @@ source_set("zlib_inflate_chunk_simd") {
 }
 
 config("zlib_crc32_simd_config") {
-  if (use_x86_x64_optimizations) {
+  if (use_x86_x64_optimizations && use_sse4_2) {
     defines = [ "CRC32_SIMD_SSE42_PCLMUL" ]
   }
 }
@@ -215,7 +221,7 @@ config("zlib_crc32_simd_config") {
 source_set("zlib_crc32_simd") {
   visibility = [ ":*" ]
 
-  if (use_x86_x64_optimizations) {
+  if (use_x86_x64_optimizations && use_sse4_2) {
     sources = [
       "crc32_simd.c",
       "crc32_simd.h",
