diff -u b/v8/include/v8-function-callback.h b/v8/include/v8-function-callback.h
--- b/v8/include/v8-function-callback.h	2026-02-18 18:05:27.457673134 -0800
+++ b/v8/include/v8-function-callback.h	2026-02-18 08:07:40.122268408 -0800
@@ -673,7 +673,14 @@
 
 template <typename T>
 Local<Object> FunctionCallbackInfo<T>::This() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return Local<Object>::FromSlot(&values_[kReceiverIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
@@ -700,7 +707,14 @@
 
 template <typename T>
 ReturnValue<T> FunctionCallbackInfo<T>::GetReturnValue() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return ReturnValue<T>(&values_[kReturnValueIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
@@ -719,7 +733,14 @@
 
 template <typename T>
 bool FunctionCallbackInfo<T>::IsConstructCall() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return I::SmiValue(values_[kFrameTypeIndex]) == I::kFrameTypeApiConstructExit;
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
@@ -732,12 +753,26 @@
 
 template <typename T>
 Local<Object> PropertyCallbackInfo<T>::HolderV2() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return Local<Object>::FromSlot(&args_[kHolderIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
 ReturnValue<T> PropertyCallbackInfo<T>::GetReturnValue() const {
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
   return ReturnValue<T>(&args_[kReturnValueIndex]);
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 }
 
 template <typename T>
--- a/third_party/swiftshader/third_party/llvm-subzero/lib/Support/regcomp.c.orig	2026-02-12 11:44:24.000000000 -0800
+++ b/third_party/swiftshader/third_party/llvm-subzero/lib/Support/regcomp.c	2026-02-16 14:30:34.348287424 -0800
@@ -222,7 +222,14 @@ llvm_regcomp(llvm_regex_t *preg, const c
 	g->mlen = 0;
 	g->nsub = 0;
 	g->ncategories = 1;	/* category 0 is "everything else" */
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warray-bounds"
+#endif
 	g->categories = &g->catspace[-(CHAR_MIN)];
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 	(void) memset((char *)g->catspace, 0, NC*sizeof(cat_t));
 	g->backrefs = 0;
 
