diff '--color=auto' -urp lua-5.1.5.orig/Makefile lua-5.1.5/Makefile
--- lua-5.1.5.orig/Makefile	2012-02-10 01:50:23.000000000 -0800
+++ lua-5.1.5/Makefile	2026-05-19 18:29:36.060967503 -0700
@@ -9,16 +9,16 @@ PLAT= none
 
 # Where to install. The installation starts in the src and doc directories,
 # so take care if INSTALL_TOP is not an absolute path.
-INSTALL_TOP= /usr/local
+INSTALL_TOP= @STAGING_PREFIX@
 INSTALL_BIN= $(INSTALL_TOP)/bin
-INSTALL_INC= $(INSTALL_TOP)/include
-INSTALL_LIB= $(INSTALL_TOP)/lib
-INSTALL_MAN= $(INSTALL_TOP)/man/man1
+INSTALL_INC= $(INSTALL_TOP)/include/lua@LUA_SLOT@
+INSTALL_LIB= $(INSTALL_TOP)/@LIBDIR@
+INSTALL_MAN= $(INSTALL_TOP)/share/man/man1
 #
 # You probably want to make INSTALL_LMOD and INSTALL_CMOD consistent with
 # LUA_ROOT, LUA_LDIR, and LUA_CDIR in luaconf.h (and also with etc/lua.pc).
-INSTALL_LMOD= $(INSTALL_TOP)/share/lua/$V
-INSTALL_CMOD= $(INSTALL_TOP)/lib/lua/$V
+INSTALL_LMOD= $(INSTALL_TOP)/share/lua/@LUA_SLOT@
+INSTALL_CMOD= $(INSTALL_TOP)/@LIBDIR@/lua/@LUA_SLOT@
 
 # How to install. If your install program does not support "-p", then you
 # may have to run ranlib on the installed liblua.a (do "make ranlib").
@@ -40,10 +40,18 @@ RANLIB= ranlib
 # Convenience platforms targets.
 PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
+SO_CURRENT = 0
+SO_AGE = 0
+SO_REVISION = 0
+SO_MAJOR = $(shell expr $(SO_CURRENT) - $(SO_AGE))
+SO_MINOR = $(SO_REVISION)
+SO_PATCH = $(SO_AGE)
+
 # What to install.
-TO_BIN= lua luac
+TO_BIN= lua@LUA_SLOT@ luac@LUA_SLOT@
 TO_INC= lua.h luaconf.h lualib.h lauxlib.h ../etc/lua.hpp
-TO_LIB= liblua.a
+TO_LIB_A= liblua@LUA_SLOT@.a
+TO_LIB_SO= liblua@LUA_SLOT@.so.$(SO_MAJOR).$(SO_MINOR).$(SO_PATCH)
 TO_MAN= lua.1 luac.1
 
 # Lua version and release.
@@ -62,11 +70,14 @@ install: dummy
 	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN) $(INSTALL_LMOD) $(INSTALL_CMOD)
 	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN)
 	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC)
-	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB)
+	cd src && $(INSTALL_DATA) $(TO_LIB_A) $(INSTALL_LIB)
+	cd src && $(INSTALL_DATA) $(TO_LIB_SO) $(INSTALL_LIB)
+	ln -s $(TO_LIB_SO) $(INSTALL_LIB)/liblua@LUA_SLOT@.so.$(SO_MAJOR)
+	ln -s liblua@LUA_SLOT@.so.$(SO_MAJOR) $(INSTALL_LIB)/liblua@LUA_SLOT@.so
 	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN)
 
 ranlib:
-	cd src && cd $(INSTALL_LIB) && $(RANLIB) $(TO_LIB)
+	cd src && cd $(INSTALL_LIB) && $(RANLIB) $(TO_LIB_A)
 
 local:
 	$(MAKE) install INSTALL_TOP=..
@@ -110,7 +121,8 @@ pecho:
 	@echo "R = $(R)"
 	@echo "TO_BIN = $(TO_BIN)"
 	@echo "TO_INC = $(TO_INC)"
-	@echo "TO_LIB = $(TO_LIB)"
+	@echo "TO_LIB_A = $(TO_LIB_A)"
+	@echo "TO_LIB_SO = $(TO_LIB_SO)"
 	@echo "TO_MAN = $(TO_MAN)"
 
 # echo config parameters as Lua code
diff '--color=auto' -urp lua-5.1.5.orig/src/Makefile lua-5.1.5/src/Makefile
--- lua-5.1.5.orig/src/Makefile	2012-02-13 12:41:22.000000000 -0800
+++ lua-5.1.5/src/Makefile	2026-05-19 18:36:27.196455804 -0700
@@ -8,11 +8,12 @@
 PLAT= none
 
 CC= gcc
-CFLAGS= -O2 -Wall $(MYCFLAGS)
+CFLAGS= -O2 -Wall -fPIC $(MYCFLAGS) $(USERCFLAGS)
+LDFLAGS= $(MYLDFLAGS) $(USERLDFLAGS)
 AR= ar rcu
 RANLIB= ranlib
 RM= rm -f
-LIBS= -lm $(MYLIBS)
+LIBS= -lm $(MYLIBS) $(USERLIBS)
 
 MYCFLAGS=
 MYLDFLAGS=
@@ -22,22 +23,32 @@ MYLIBS=
 
 PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
-LUA_A=	liblua.a
+LUA_A=	liblua@LUA_SLOT@.a
+
+SO_CURRENT = 0
+SO_AGE = 0
+SO_REVISION = 0
+SO_MAJOR = $(shell expr $(SO_CURRENT) - $(SO_AGE))
+SO_MINOR = $(SO_REVISION)
+SO_PATCH = $(SO_AGE)
+LUA_SO=	liblua@LUA_SLOT@.so.$(SO_MAJOR).$(SO_MINOR).$(SO_PATCH)
+
 CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
 	lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o  \
 	lundump.o lvm.o lzio.o
 LIB_O=	lauxlib.o lbaselib.o ldblib.o liolib.o lmathlib.o loslib.o ltablib.o \
 	lstrlib.o loadlib.o linit.o
 
-LUA_T=	lua
+LUA_T=	lua@LUA_SLOT@
 LUA_O=	lua.o
 
-LUAC_T=	luac
+LUAC_T=	luac@LUA_SLOT@
 LUAC_O=	luac.o print.o
 
 ALL_O= $(CORE_O) $(LIB_O) $(LUA_O) $(LUAC_O)
-ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T)
+ALL_T= $(LUA_A) $(LUA_SO) $(LUA_T) $(LUAC_T)
 ALL_A= $(LUA_A)
+ALL_SO= $(LUA_SO)
 
 default: $(PLAT)
 
@@ -47,10 +58,17 @@ o:	$(ALL_O)
 
 a:	$(ALL_A)
 
+so:	$(ALL_SO)
+
 $(LUA_A): $(CORE_O) $(LIB_O)
 	$(AR) $@ $(CORE_O) $(LIB_O)	# DLL needs all object files
 	$(RANLIB) $@
 
+$(LUA_SO): $(BASE_O)
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,liblua@LUA_SLOT@.so.$(SO_MAJOR) -o $@ $^
+	ln -sf $(LUA_SO) liblua@LUA_SLOT@.so.$(SO_MAJOR)
+	ln -sf liblua@LUA_SLOT@.so.$(SO_MAJOR) liblua@LUA_SLOT@.so
+
 $(LUA_T): $(LUA_O) $(LUA_A)
 	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
 
