diff -urp cecil-master.orig/Mono.Cecil.nuspec cecil-master/Mono.Cecil.nuspec
--- cecil-master.orig/Mono.Cecil.nuspec	2017-09-08 10:58:17.000000000 -0700
+++ cecil-master/Mono.Cecil.nuspec	2017-09-17 13:15:35.752842614 -0700
@@ -1,14 +1,15 @@
 <?xml version="1.0"?>
 <package xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
-    <id>Mono.Cecil</id>
-    <version>0.10.0.0-beta6</version>
+    <id>$id$</id>
+    <version>$version$</version>
     <title>Mono.Cecil</title>
     <authors>Jb Evain</authors>
     <owners>Jb Evain</owners>
     <licenseUrl>http://opensource.org/licenses/mit-license.php</licenseUrl>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <projectUrl>http://github.com/jbevain/cecil/</projectUrl>
+    <iconUrl>file:///var/lib/layman/dotnet/dev-dotnet/cecil/files/niget_icon_64x64.png</iconUrl>
     <summary>Cecil is a library written by Jb Evain to generate and inspect programs and libraries in the ECMA CIL format.</summary>
     <description>Cecil is a library written by Jb Evain to generate and inspect programs and libraries in the ECMA CIL format. It has full support for generics, and support some debugging symbol format. In simple English, with Cecil, you can load existing managed assemblies, browse all the contained types, modify them on the fly and save back to the disk the modified assembly.</description>
     <language>en-US</language>
@@ -29,8 +30,8 @@
     </dependencies>
   </metadata>
   <files>
-    <file src="bin\net_3_5_Release\*.dll" target="lib/net35" />
-    <file src="bin\net_4_0_Release\*.dll" target="lib/net40" />
-    <file src="bin\netstandard_Release\netstandard1.3\*.dll" target="lib/netstandard1.3" />
+    <file src="bin/net_3_5_$configuration$/*.dll" target="lib/net35" />
+    <file src="bin/net_4_0_$configuration$/*.dll" target="lib/net40" />
+    <file src="bin/net_4_5_$configuration$/*.dll" target="lib/net45" />
   </files>
 </package>
Only in cecil-master: Mono.Cecil.nuspec.orig
Only in cecil-master: Mono.Cecil.nuspec.rej
