diff -urp cecil-master.orig/Mono.Cecil.props cecil-master/Mono.Cecil.props
--- cecil-master.orig/Mono.Cecil.props	2016-12-14 10:03:42.000000000 -0800
+++ cecil-master/Mono.Cecil.props	2017-01-03 04:51:22.651946850 -0800
@@ -37,6 +37,10 @@
     <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
     <DefineConstants>$(DefineConstants);NET_4_0;</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition=" $(Configuration.StartsWith('net_4_5')) ">
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <DefineConstants>$(DefineConstants);NET_4_5;</DefineConstants>
+  </PropertyGroup>
   <PropertyGroup Condition=" $(Configuration.StartsWith('pcl')) ">
     <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <TargetFrameworkProfile>Profile344</TargetFrameworkProfile>
@@ -54,12 +58,16 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)' == 'net_4_0_Debug' ">
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'net_4_5_Debug' ">
+  </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)' == 'pcl_Debug' ">
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)' == 'net_3_5_Release' ">
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)' == 'net_4_0_Release' ">
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'net_4_5_Release' ">
+  </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)' == 'pcl_Release' ">
   </PropertyGroup>
   <!-- This optional import allows products that distribute Cecil to tweak settings that will affect its 
