diff -urp hipFFT-rocm-5.3.3.orig/clients/cmake/FindFFTW.cmake hipFFT-rocm-5.3.3/clients/cmake/FindFFTW.cmake
--- hipFFT-rocm-5.3.3.orig/clients/cmake/FindFFTW.cmake	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/clients/cmake/FindFFTW.cmake	2023-09-04 17:46:47.855417369 -0700
@@ -45,11 +45,11 @@ if( FFTW_FIND_REQUIRED_FLOAT OR FFTW_FIN
   find_library( FFTW_LIBRARIES_SINGLE
       NAMES fftw3f fftw3f-3 fftw3 fftw3-3
       HINTS
-          ${FFTW_ROOT}/lib
-          $ENV{FFTW_ROOT}/lib
+          ${FFTW_ROOT}/@LIBDIR@
+          $ENV{FFTW_ROOT}/@LIBDIR@
       PATHS
-          /usr/lib
-          /usr/local/lib
+          /usr/@LIBDIR@
+          /usr/local/@LIBDIR@
       PATH_SUFFIXES
           x86_64-linux-gnu
       DOC "FFTW dynamic library single"
@@ -74,11 +74,11 @@ if( FFTW_FIND_REQUIRED_DOUBLE )
   find_library( FFTW_LIBRARIES_DOUBLE
       NAMES fftw3
       HINTS
-          ${FFTW_ROOT}/lib
-          $ENV{FFTW_ROOT}/lib
+          ${FFTW_ROOT}/@LIBDIR@
+          $ENV{FFTW_ROOT}/@LIBDIR@
       PATHS
-          /usr/lib
-          /usr/local/lib
+          /usr/@LIBDIR@
+          /usr/local/@LIBDIR@
       PATH_SUFFIXES
           x86_64-linux-gnu
       DOC "FFTW dynamic library double"
diff -urp hipFFT-rocm-5.3.3.orig/clients/CMakeLists.txt hipFFT-rocm-5.3.3/clients/CMakeLists.txt
--- hipFFT-rocm-5.3.3.orig/clients/CMakeLists.txt	2023-09-04 17:46:34.071417554 -0700
+++ hipFFT-rocm-5.3.3/clients/CMakeLists.txt	2023-09-04 17:46:47.859417369 -0700
@@ -33,7 +33,7 @@ if( WIN32 )
     "Install path prefix, prepended onto install directories" )
 else( )
   set( CMAKE_INSTALL_PREFIX
-    "/opt/rocm"
+    "/usr"
     CACHE
     PATH
     "Install path prefix, prepended onto install directories" )
@@ -82,8 +82,8 @@ if( BUILD_CLIENTS_TESTS )
       ${CMAKE_CURRENT_BINARY_DIR}/src/gtest/googletest/include
       )
     set( GTEST_LIBRARIES
-      ${CMAKE_CURRENT_BINARY_DIR}/src/gtest-build/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gtest${CMAKE_STATIC_LIBRARY_SUFFIX}
-      ${CMAKE_CURRENT_BINARY_DIR}/src/gtest-build/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gtest_main${CMAKE_STATIC_LIBRARY_SUFFIX}
+      ${CMAKE_CURRENT_BINARY_DIR}/src/gtest-build/@LIBDIR@/${CMAKE_STATIC_LIBRARY_PREFIX}gtest${CMAKE_STATIC_LIBRARY_SUFFIX}
+      ${CMAKE_CURRENT_BINARY_DIR}/src/gtest-build/@LIBDIR@/${CMAKE_STATIC_LIBRARY_PREFIX}gtest_main${CMAKE_STATIC_LIBRARY_SUFFIX}
       )
 
     ExternalProject_Add( gtest
diff -urp hipFFT-rocm-5.3.3.orig/cmake/dependencies.cmake hipFFT-rocm-5.3.3/cmake/dependencies.cmake
--- hipFFT-rocm-5.3.3.orig/cmake/dependencies.cmake	2023-09-04 17:46:34.071417554 -0700
+++ hipFFT-rocm-5.3.3/cmake/dependencies.cmake	2023-09-04 17:46:47.859417369 -0700
@@ -51,7 +51,7 @@ else()
 endif()
 
 # ROCm
-find_package( ROCM 0.7.3 CONFIG PATHS /opt/rocm )
+find_package( ROCM 0.7.3 CONFIG PATHS /usr )
 if(NOT ROCM_FOUND)
   set( rocm_cmake_tag "master" CACHE STRING "rocm-cmake tag to download" )
   set( PROJECT_EXTERN_DIR "${CMAKE_CURRENT_BINARY_DIR}/extern" )
diff -urp hipFFT-rocm-5.3.3.orig/CMakeLists.txt hipFFT-rocm-5.3.3/CMakeLists.txt
--- hipFFT-rocm-5.3.3.orig/CMakeLists.txt	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/CMakeLists.txt	2023-09-04 17:46:47.859417369 -0700
@@ -32,20 +32,20 @@ if( WIN32 )
   set( CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/package" CACHE PATH
     "Install path prefix, prepended onto install directories" )
 else()
-  set( CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH
+  set( CMAKE_INSTALL_PREFIX "/usr" CACHE PATH
     "Install path prefix, prepended onto install directories" )
 endif()
 
 
 # Workarounds..
-list( APPEND CMAKE_PREFIX_PATH /opt/rocm/llvm /opt/rocm )
-list( APPEND CMAKE_MODULE_PATH ${ROCM_PATH}/lib/cmake/hip /opt/rocm/lib/cmake/hip /opt/rocm/hip/cmake )
+list( APPEND CMAKE_PREFIX_PATH /usr/lib/llvm/@LLVM_SLOT@ /usr )
+list( APPEND CMAKE_MODULE_PATH ${ROCM_PATH}/@LIBDIR@/cmake/hip /usr/@LIBDIR@/cmake/hip )
 
 # This has to be initialized before the project() command appears
 # Set the default of CMAKE_BUILD_TYPE to be release, unless user specifies with -D.
 # MSVC_IDE does not use CMAKE_BUILD_TYPE
 if( NOT DEFINED CMAKE_CONFIGURATION_TYPES AND NOT DEFINED CMAKE_BUILD_TYPE )
-  set( CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." )
+#_cmake_modify_IGNORE   set( CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." )
 endif()
 
 project( hipfft LANGUAGES CXX )
@@ -193,7 +193,7 @@ endif()
 if(WIN32)
   set(CPACK_SOURCE_GENERATOR "ZIP")
   set(CPACK_GENERATOR "ZIP")
-  set(CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE)
+#_cmake_modify_IGNORE   set(CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE)
   set(INSTALL_PREFIX "C:/hipSDK")
   set(CPACK_SET_DESTDIR OFF)
   set(CPACK_PACKAGE_INSTALL_DIRECTORY "C:/hipSDK")
@@ -220,7 +220,6 @@ if( ROCM_FOUND )
     set( package_name hipfft-alt )
   endif()
 
-  set( HIPFFT_CONFIG_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" CACHE PATH "Path placed into ldconfig file" )
 
   rocm_create_package(
     NAME ${package_name}
diff -urp hipFFT-rocm-5.3.3.orig/deps/CMakeLists.txt hipFFT-rocm-5.3.3/deps/CMakeLists.txt
--- hipFFT-rocm-5.3.3.orig/deps/CMakeLists.txt	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/deps/CMakeLists.txt	2023-09-04 17:46:47.859417369 -0700
@@ -31,13 +31,13 @@ list( APPEND CMAKE_MODULE_PATH ${CMAKE_C
 # Consider removing this in the future
 # It can be annoying for visual studio developers to build a project that tries to install into 'program files'
 if( WIN32 AND CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT )
-  set( CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/package" CACHE PATH "Install path prefix, prepended onto install directories" FORCE )
+#_cmake_modify_IGNORE   set( CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/package" CACHE PATH "Install path prefix, prepended onto install directories" FORCE )
 endif( )
 
 # This has to be initialized before the project() command appears
 # Set the default of CMAKE_BUILD_TYPE to be release, unless user specifies with -D.  MSVC_IDE does not use CMAKE_BUILD_TYPE
 if( NOT DEFINED CMAKE_CONFIGURATION_TYPES AND NOT DEFINED CMAKE_BUILD_TYPE )
-  set( CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." )
+#_cmake_modify_IGNORE   set( CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." )
 endif()
 
 # The superbuild does not build anything itself; all compiling is done in external projects
diff -urp hipFFT-rocm-5.3.3.orig/deps/external-gtest.cmake hipFFT-rocm-5.3.3/deps/external-gtest.cmake
--- hipFFT-rocm-5.3.3.orig/deps/external-gtest.cmake	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/deps/external-gtest.cmake	2023-09-04 17:46:47.859417369 -0700
@@ -90,11 +90,7 @@ ExternalProject_Get_Property( googletest
 set( package_dir "${PREFIX_GTEST}" )
 if( CMAKE_CONFIGURATION_TYPES )
   # Create a package by bundling libraries and header files
-  if( BUILD_64 )
-    set( LIB_DIR lib64 )
-  else( )
-    set( LIB_DIR lib )
-  endif( )
+  set( LIB_DIR @LIBDIR@ )
 
   set( gtest_lib_dir "<BINARY_DIR>/${LIB_DIR}" )
   ExternalProject_Add_Step( googletest createPackage
diff -urp hipFFT-rocm-5.3.3.orig/.githooks/pre-commit hipFFT-rocm-5.3.3/.githooks/pre-commit
--- hipFFT-rocm-5.3.3.orig/.githooks/pre-commit	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/.githooks/pre-commit	2023-09-04 17:46:47.859417369 -0700
@@ -4,7 +4,7 @@
 # are installed, and if so, uses the installed version to format
 # the staged changes.
 
-base=/opt/rocm/hcc/bin/clang-format
+base=/usr/hcc/bin/clang-format
 format=""
 
 # Redirect output to stderr.
diff -urp hipFFT-rocm-5.3.3.orig/README.md hipFFT-rocm-5.3.3/README.md
--- hipFFT-rocm-5.3.3.orig/README.md	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/README.md	2023-09-04 17:46:47.859417369 -0700
@@ -25,8 +25,8 @@ If you are transitioning from the hipFFT
 this standalone hipFFT version; please modify your build following
 this example:
 
-* previously: `hipcc hipfft_1d_z2z.cpp -L/opt/rocm/lib -lrocfft`
-* during transition: `hipcc -I/opt/rocm/hipfft/include hipfft_1d_z2z.cpp -L/opt/rocm/lib -lhipfft -lrocfft`
+* previously: `hipcc hipfft_1d_z2z.cpp -L/usr/@LIBDIR@ -lrocfft`
+* during transition: `hipcc -I/usr/hipfft/include hipfft_1d_z2z.cpp -L/usr/@LIBDIR@ -lhipfft -lrocfft`
 
 ## Building from source
 
diff -urp hipFFT-rocm-5.3.3.orig/rmake.py hipFFT-rocm-5.3.3/rmake.py
--- hipFFT-rocm-5.3.3.orig/rmake.py	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/rmake.py	2023-09-04 17:46:47.859417369 -0700
@@ -108,7 +108,7 @@ def config_cmd():
         cmake_options.extend( generator )
         toolchain = os.path.join( src_path, "toolchain-windows.cmake" )
     else:
-        rocm_path = os.getenv( 'ROCM_PATH', "/opt/rocm")
+        rocm_path = os.getenv( 'ROCM_PATH', "/usr")
         cmake_platform_opts.append( f"-DROCM_DIR:PATH={rocm_path}")
         cmake_platform_opts.append( f"-DCPACK_PACKAGING_INSTALL_PREFIX={rocm_path}" )
         cmake_platform_opts.append( f"-DCMAKE_INSTALL_PREFIX=rocfft-install" )
diff -urp hipFFT-rocm-5.3.3.orig/toolchain-linux.cmake hipFFT-rocm-5.3.3/toolchain-linux.cmake
--- hipFFT-rocm-5.3.3.orig/toolchain-linux.cmake	2022-08-29 14:12:18.000000000 -0700
+++ hipFFT-rocm-5.3.3/toolchain-linux.cmake	2023-09-04 17:47:09.615417079 -0700
@@ -21,9 +21,9 @@
 # #############################################################################
 
 if (DEFINED ENV{ROCM_PATH})
-  set(rocm_bin "$ENV{ROCM_PATH}/hip/bin")
+  set(rocm_bin "$ENV{ROCM_PATH}/bin")
 else()
-  set(rocm_bin "/opt/rocm/hip/bin")
+  set(rocm_bin "/usr/bin")
 endif()
 
 set(CMAKE_CXX_COMPILER "${rocm_bin}/hipcc")
Only in hipFFT-rocm-5.3.3: toolchain-linux.cmake.orig
Only in hipFFT-rocm-5.3.3: toolchain-linux.cmake.rej
