diff -urp gst-plugins-rs-fdc3ea68e8276ca9eaac077997d5d2e975af58ad.orig/meson.build gst-plugins-rs-fdc3ea68e8276ca9eaac077997d5d2e975af58ad/meson.build
--- gst-plugins-rs-fdc3ea68e8276ca9eaac077997d5d2e975af58ad.orig/meson.build	2021-06-15 19:33:46.664834349 -0700
+++ gst-plugins-rs-fdc3ea68e8276ca9eaac077997d5d2e975af58ad/meson.build	2021-06-15 19:34:15.105941089 -0700
@@ -153,14 +153,14 @@ cc = meson.get_compiler('c')
 csound_option = get_option('csound')
 csound_dep = dependency('', required: false) # not-found dependency
 if not csound_option.disabled()
-  csound_dep = cc.find_library('csound64', required: false)
+  csound_dep = cc.find_library('csound', required: false)
   if not csound_dep.found()
     python3 = import('python').find_installation('python3')
     res = run_command(python3, '-c', 'import os; print(os.environ["CSOUND_LIB_DIR"])')
     if res.returncode() == 0
-      csound_dep = cc.find_library('csound64', dirs: res.stdout(), required: csound_option)
+      csound_dep = cc.find_library('csound', dirs: res.stdout(), required: csound_option)
     elif csound_option.enabled()
-      error('csound option is enabled, but csound64 library could not be found and CSOUND_LIB_DIR was not set')
+      error('csound option is enabled, but csound library could not be found and CSOUND_LIB_DIR was not set')
     endif
   endif
 endif
