diff -urp gst-plugins-rs-9999.orig/meson.build gst-plugins-rs-9999/meson.build
--- gst-plugins-rs-9999.orig/meson.build	2022-10-25 19:29:44.997299546 -0700
+++ gst-plugins-rs-9999/meson.build	2022-10-25 19:35:27.206647714 -0700
@@ -39,57 +39,147 @@ endif
 # workspace name -> lib name
 # kept in the same order as the `members` list in Cargo.toml
 plugins = {
-  'gst-plugin-audiofx': 'libgstrsaudiofx',
-  'gst-plugin-claxon': 'libgstclaxon',
   # csound has an external dependency, see below
-  'gst-plugin-lewton': 'libgstlewton',
-  'gst-plugin-spotify': 'libgstspotify',
-
-  'gst-plugin-file': 'libgstrsfile',
   # sodium has an external dependency, see below
-  'gst-plugin-threadshare': 'libgstthreadshare',
-
-  'gst-plugin-fmp4': 'libgstfmp4',
-
-  'gst-plugin-aws': 'libgstaws',
-  'gst-plugin-hlssink3': 'libgsthlssink3',
-  'gst-plugin-ndi': 'libgstndi',
-  'gst-plugin-onvif': 'libgstrsonvif',
-  'gst-plugin-raptorq': 'libgstraptorq',
-  'gst-plugin-reqwest': 'libgstreqwest',
-  'gst-plugin-rtp': 'libgstrsrtp',
-  'gst-plugin-webrtchttp': 'libgstwebrtchttp',
-  'gst-plugin-webrtc': 'libgstrswebrtc',
-
-  'gst-plugin-textahead': 'libgsttextahead',
-  'gst-plugin-json': 'libgstjson',
-  'gst-plugin-regex': 'libgstregex',
-  'gst-plugin-textwrap': 'libgsttextwrap',
-
-  'gst-plugin-fallbackswitch': 'libgstfallbackswitch',
-  'gst-plugin-togglerecord': 'libgsttogglerecord',
-  'gst-plugin-tracers': 'libgstrstracers',
-  'gst-plugin-uriplaylistbin': 'libgsturiplaylistbin',
-
-  'gst-plugin-cdg': 'libgstcdg',
   # closedcaption has an external dependency, see below
   # dav1d has an external dependency, see below
-  'gst-plugin-ffv1': 'libgstffv1',
-  'gst-plugin-flavors': 'libgstrsflv',
-  'gst-plugin-gif': 'libgstgif',
   # gtk4 has an external dependency, see below
-  'gst-plugin-hsv': 'libgsthsv',
-  'gst-plugin-png': 'libgstrspng',
-  'gst-plugin-rav1e': 'libgstrav1e',
   # videofx has an external dependency, see below
 
   # FIXME: libwebp-sys2 will build its bundled version on msvc and apple platforms
   # https://github.com/qnighy/libwebp-sys2-rs/issues/4
-  'gst-plugin-webp': 'libgstrswebp',
 }
 
 extra_env = {}
 
+if get_option('audiofx').enabled()
+  plugins += {'gst-plugin-audiofx' : 'libgstrsaudiofx'}
+endif
+
+if get_option('aws').enabled()
+  plugins += {'gst-plugin-aws' : 'libgstaws'}
+endif
+
+if get_option('cdg').enabled()
+  plugins += {'gst-plugin-cdg' : 'libgstcdg'}
+endif
+
+if get_option('claxon').enabled()
+  plugins += {'gst-plugin-claxon' : 'libgstclaxon'}
+endif
+
+if get_option('fallbackswitch').enabled()
+  plugins += {'gst-plugin-fallbackswitch' : 'libgstfallbackswitch'}
+endif
+
+if get_option('ffv1').enabled()
+  plugins += {'gst-plugin-ffv1' : 'libgstffv1'}
+endif
+
+if get_option('file').enabled()
+  plugins += {'gst-plugin-file' : 'libgstrsfile'}
+endif
+
+if get_option('flavors').enabled()
+  plugins += {'gst-plugin-flavors' : 'libgstrsflv'}
+endif
+
+if get_option('fmp4').enabled()
+  plugins += {'gst-plugin-fmp4' : 'libgstfmp4'}
+endif
+
+if get_option('gif').enabled()
+  plugins += {'gst-plugin-gif' : 'libgstgif'}
+endif
+
+if get_option('gtk4').enabled()
+  plugins += {'gst-plugin-gtk4' : 'libgstgtk4'}
+endif
+
+if get_option('hlssink3').enabled()
+  plugins += {'gst-plugin-hlssink3' : 'libgsthlssink3'}
+endif
+
+if get_option('hsv').enabled()
+  plugins += {'gst-plugin-hsv' : 'libgsthsv'}
+endif
+
+if get_option('json').enabled()
+  plugins += {'gst-plugin-json' : 'libgstjson'}
+endif
+
+if get_option('lewton').enabled()
+  plugins += {'gst-plugin-lewton' : 'libgstlewton'}
+endif
+
+if get_option('ndi').enabled()
+  plugins += {'gst-plugin-ndi' : 'libgstndi'}
+endif
+
+if get_option('onvif').enabled()
+  plugins += {'gst-plugin-onvif' : 'libgstrsonvif'}
+endif
+
+if get_option('raptorq').enabled()
+  plugins += {'gst-plugin-raptorq' : 'libgstraptorq'}
+endif
+
+if get_option('rav1e').enabled()
+  plugins += {'gst-plugin-rav1e' : 'libgstrav1e'}
+endif
+
+if get_option('regex').enabled()
+  plugins += {'gst-plugin-regex' : 'libgstregex'}
+endif
+
+if get_option('reqwest').enabled()
+  plugins += {'gst-plugin-reqwest' : 'libgstreqwest'}
+endif
+
+if get_option('png').enabled()
+  plugins += {'gst-plugin-png' : 'libgstrspng'}
+endif
+
+if get_option('rtp').enabled()
+  plugins += {'gst-plugin-rtp' : 'libgstrtp'}
+endif
+
+if get_option('spotify').enabled()
+  plugins += {'gst-plugin-spotify' : 'libgstspotify'}
+endif
+
+if get_option('textahead').enabled()
+  plugins += {'gst-plugin-textahead' : 'libgsttextahead'}
+endif
+
+if get_option('textwrap').enabled()
+  plugins += {'gst-plugin-textwrap' : 'libgsttextwrap'}
+endif
+
+if get_option('threadshare').enabled()
+  plugins += {'gst-plugin-threadshare' : 'libgstthreadshare'}
+endif
+
+if get_option('togglerecord').enabled()
+  plugins += {'gst-plugin-togglerecord' : 'libgsttogglerecord'}
+endif
+
+if get_option('tracers').enabled()
+  plugins += {'gst-plugin-tracers' : 'libgstrstracers'}
+endif
+
+if get_option('uriplaylistbin').enabled()
+  plugins += {'gst-plugin-uriplaylistbin' : 'libgsturiplaylistbin'}
+endif
+
+if get_option('webp').enabled()
+  plugins += {'gst-plugin-webp' : 'libgstrswebp'}
+endif
+
+if get_option('webrtchttp').enabled()
+  plugins += {'gst-plugin-webrtchttp' : 'libgstwebrtchttp'}
+endif
+
 if dependency('cairo-gobject', required : get_option('videofx')).found()
   plugins += {'gst-plugin-videofx': 'libgstrsvideofx',}
 endif
@@ -102,13 +192,15 @@ if dependency('dav1d', version : '>= 1.0
   plugins += {'gst-plugin-dav1d' : 'libgstdav1d'}
 endif
 
-sodium = get_option ('sodium')
-if sodium == 'system'
-  dependency('libsodium')
-  plugins += {'gst-plugin-sodium': 'libgstsodium'}
-  extra_env += {'SODIUM_USE_PKG_CONFIG': '1'}
-elif sodium == 'built-in'
-  plugins += {'gst-plugin-sodium': 'libgstsodium'}
+sodium = get_option ('sodium').enabled()
+if get_option ('sodium').enabled()
+  if get_option ('sodium-built-in').enabled()
+    plugins += {'gst-plugin-sodium': 'libgstsodium'}
+  else
+    dependency('libsodium')
+    plugins += {'gst-plugin-sodium': 'libgstsodium'}
+    extra_env += {'SODIUM_USE_PKG_CONFIG': '1'}
+  endif
 endif
 
 cc = meson.get_compiler('c')
Only in gst-plugins-rs-9999: meson.build.orig
Only in gst-plugins-rs-9999: meson.build.rej
diff -urp gst-plugins-rs-9999.orig/meson_options.txt gst-plugins-rs-9999/meson_options.txt
--- gst-plugins-rs-9999.orig/meson_options.txt	2022-10-25 19:29:44.997299546 -0700
+++ gst-plugins-rs-9999/meson_options.txt	2022-10-25 19:30:10.766304503 -0700
@@ -1,10 +1,43 @@
 option('videofx', type : 'feature', value : 'auto', description : 'Build videofx plugin')
 option('closedcaption', type : 'feature', value : 'auto', description : 'Build closedcaption plugin')
 option('dav1d', type : 'feature', value : 'auto', description : 'Build dav1d plugin')
-option('sodium', type : 'combo',
-    choices : ['system', 'built-in', 'disabled'], value : 'built-in',
-    description : 'Weither to use libsodium from the system or the built-in version from the sodiumoxide crate')
+option('sodium', type : 'feature', value : 'auto', description : 'Build sodium plugin')
+option('sodium-built-in', type : 'feature', value : 'disabled', description : 'Weither to use libsodium from the system or the built-in version from the sodiumoxide crate')
 option('csound', type : 'feature', value : 'auto', description : 'Build csound plugin')
+
+option('audiofx', type: 'feature', value : 'auto', description : 'Build the audiofx plugin')
+option('aws', type: 'feature', value : 'auto', description : 'Build the aws plugin')
+option('cdg', type: 'feature', value : 'auto', description : 'Build the cdg plugin')
+option('claxon', type: 'feature', value : 'auto', description : 'Build the claxon plugin')
+option('fallbackswitch', type: 'feature', value : 'auto', description : 'Build the fallbackswitch plugin')
+option('ffv1', type: 'feature', value : 'auto', description : 'Build the ffv1 plugin')
+option('file', type: 'feature', value : 'auto', description : 'Build the file plugin')
+option('flavors', type: 'feature', value : 'auto', description : 'Build the flavors plugin')
+option('fmp4', type: 'feature', value : 'auto', description : 'Build the fmp4 plugin')
+option('gif', type: 'feature', value : 'auto', description : 'Build the gif plugin')
+option('hlssink3', type: 'feature', value : 'auto', description : 'Build the hlssink3 plugin')
+option('lewton', type: 'feature', value : 'auto', description : 'Build the lewton plugin')
+option('ndi', type: 'feature', value : 'auto', description : 'Build the ndi plugin')
+option('onvif', type: 'feature', value : 'auto', description : 'Build the onvif plugin')
+option('raptorq', type: 'feature', value : 'auto', description : 'Build the raptorq plugin')
+option('rav1e', type: 'feature', value : 'auto', description : 'Build the rav1e plugin')
+option('reqwest', type: 'feature', value : 'auto', description : 'Build the reqwest plugin')
+option('png', type: 'feature', value : 'auto', description : 'Build the rspng plugin')
+option('rtp', type: 'feature', value : 'auto', description : 'Build the rtpav1 plugin')
+option('textahead', type: 'feature', value : 'auto', description : 'Build the ahead plugin')
+option('textwrap', type: 'feature', value : 'auto', description : 'Build the textwrap plugin')
+option('threadshare', type: 'feature', value : 'auto', description : 'Build the threadshare plugin')
+option('togglerecord', type: 'feature', value : 'auto', description : 'Build the togglerecord plugin')
+option('tracers', type: 'feature', value : 'auto', description : 'Build the tracers plugin')
+option('spotify', type: 'feature', value : 'auto', description : 'Build the spotify plugin')
+option('uriplaylistbin', type: 'feature', value : 'auto', description : 'Build the uriplaylistbin plugin')
+option('hsv', type: 'feature', value : 'auto', description : 'Build the hsv plugin')
+option('json', type: 'feature', value : 'auto', description : 'Build the json plugin')
+option('regex', type: 'feature', value : 'auto', description : 'Build the regex plugin')
+option('webp', type: 'feature', value : 'auto', description : 'Build the webp plugin')
+option('webrtc', type: 'feature', value : 'auto', description : 'Build the webrtc plugin')
+option('webrtchttp', type: 'feature', value : 'auto', description : 'Build the webrtchttp plugin')
+
 option('gtk4', type : 'feature', value : 'auto', description : 'Build GTK4 plugin')
 
 # Common options
