--- a/meson.build.orig	2021-06-15 12:54:09.621129432 -0700
+++ b/meson.build	2021-06-15 13:00:19.935135555 -0700
@@ -186,7 +186,7 @@ if not csound_dep.found() and not csound
   res = run_command(python3, '-c', 'import os; print(os.environ["CSOUND_LIB_DIR"])')
   if res.returncode() == 0
     csound_libdir = res.stdout().strip()
-    csound_dep = cc.find_library('csound64', dirs: csound_libdir, required: false)
+    csound_dep = cc.find_library('csound', dirs: csound_libdir, required: false)
     if csound_dep.found()
       extra_env += {'CSOUND_LIB_DIR': csound_libdir}
     endif
