--- a/rife_ncnn_vulkan_python/rife_ncnn_vulkan.py.orig	2024-10-18 17:09:35.000000000 -0700
+++ b/rife_ncnn_vulkan_python/rife_ncnn_vulkan.py	2024-12-16 13:51:14.040183756 -0800
@@ -62,7 +62,10 @@ class Rife:
         # determine if rife-v2 is used
         rife_v2 = ("rife-v2" in model) or ("rife-v3" in model)
         rife_v4 = "rife-v4" in model or "rife4" in model or "rife-4" in model
-        padding = 64 if  ("rife-v4.25" in model) or ("rife-v4.26" in model) else padding = 32
+        if ("rife-v4.25" in model) or ("rife-v4.26" in model):
+            padding = 64
+        else:
+            padding = 32
 
         # create raw RIFE wrapper object
         self._rife_object = wrapped.RifeWrapped(
