--- grpc-1.51.1/Makefile.orig	2022-11-22 15:31:21.000000000 -0800
+++ grpc-1.51.1/Makefile	2022-12-23 11:32:56.691398836 -0800
@@ -537,9 +537,14 @@ GRPC_ABSEIL_MERGE_LIBS = $(LIBDIR)/$(CON
 
 # Setup re2 dependency
 
+EMBED_RE2 ?= false                                  # oteodoro:  optional embed
+ifeq ($(EMBED_RE2),true)                            # oteodoro:  optional embed
 RE2_DEP = $(LIBDIR)/$(CONFIG)/libre2.a
 RE2_MERGE_OBJS = $(LIBRE2_OBJS)
 RE2_MERGE_LIBS = $(LIBDIR)/$(CONFIG)/libre2.a
+else                                                # oteodoro:  optional embed
+LIBS += re2                                         # oteodoro:  optional embed
+endif                                               # oteodoro:  optional embed
 
 # Setup upb dependency
 
