--- a/src/ruby/ext/grpc/extconf.rb	2022-12-23 11:19:44.064418749 -0800
+++ b/src/ruby/ext/grpc/extconf.rb	2022-12-23 11:23:42.133722404 -0800
@@ -69,11 +69,12 @@ if apple_toolchain && !cross_compiling
 end
 
 # Don't embed on TruffleRuby (constant-time crypto is unsafe with Sulong, slow build times)
-ENV['EMBED_OPENSSL'] = (RUBY_ENGINE != 'truffleruby').to_s
+ENV['EMBED_OPENSSL'] = 'true'                                   # oteodoro:  force no embed
 # Don't embed on TruffleRuby (the system zlib is already linked for the zlib C extension, slow build times)
-ENV['EMBED_ZLIB'] = (RUBY_ENGINE != 'truffleruby').to_s
+ENV['EMBED_ZLIB'] = 'false'                                     # oteodoro:  force no embed
 
-ENV['EMBED_CARES'] = 'true'
+ENV['EMBED_CARES'] = 'false'                                    # oteodoro:  force no embed
+ENV['EMBED_ABSEIL'] = 'false'                                   # oteodoro:  force no embed
 
 ENV['ARCH_FLAGS'] = RbConfig::CONFIG['ARCH_FLAG']
 if apple_toolchain && !cross_compiling
