--- a/tensorflow/tools/pip_package/setup.py.orig	2024-03-13 12:57:52.155821432 -0700
+++ b/tensorflow/tools/pip_package/setup.py	2024-03-13 12:59:20.627069166 -0700
@@ -89,8 +89,8 @@ REQUIRED_PACKAGES = [
     'gast >=0.2.1,!=0.5.0,!=0.5.1,!=0.5.2',
     'google_pasta >= 0.1.1',
     'h5py >= 3.10.0',
-    'libclang >= 13.0.0',
-    'ml_dtypes ~= 0.3.1',
+    #'libclang >= 13.0.0',
+    'ml_dtypes >= 0.3.1',
     # TODO(b/304751256): Adjust the numpy pin to a single version, when ready
     'numpy >= 1.23.5, < 2.0.0 ; python_version <= "3.11"',
     'numpy >= 1.26.0, < 2.0.0 ; python_version >= "3.12"',
@@ -107,7 +107,7 @@ REQUIRED_PACKAGES = [
     'typing_extensions >= 3.6.6',
     'wrapt >= 1.11.0',
     # TODO(b/305196096): Remove the <3.12 condition once the pkg is updated
-    'tensorflow-io-gcs-filesystem >= 0.23.1 ; python_version < "3.12"',
+    #'tensorflow-io-gcs-filesystem >= 0.23.1 ; python_version < "3.12"',
     # grpcio does not build correctly on big-endian machines due to lack of
     # BoringSSL support.
     # See https://github.com/tensorflow/tensorflow/issues/17882.
@@ -139,8 +139,8 @@ FAKE_REQUIRED_PACKAGES = [
     _VERSION + ';platform_system=="Windows"',
 ]
 
-if platform.system() == 'Linux' and platform.machine() == 'x86_64':
-  REQUIRED_PACKAGES.append(FAKE_REQUIRED_PACKAGES)
+# if platform.system() == 'Linux' and platform.machine() == 'x86_64':
+#   REQUIRED_PACKAGES.append(FAKE_REQUIRED_PACKAGES)
 
 if collaborator_build:
   # If this is a collaborator build, then build an "installer" wheel and
