diff -urp nvorbis-9999.20160922.orig/UnitTests/UnitTests.csproj nvorbis-9999.20160922/UnitTests/UnitTests.csproj
--- nvorbis-9999.20160922.orig/UnitTests/UnitTests.csproj	2017-01-03 17:05:28.445451010 -0800
+++ nvorbis-9999.20160922/UnitTests/UnitTests.csproj	2017-01-03 17:09:07.866876179 -0800
@@ -64,7 +64,7 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Import Project="$(SolutionDir)\.nuget\nuget.targets" />
+  <Import Project="$(SolutionDir)\.nuget\NuGet.targets" />
   <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
        Other similar extension points exist, see Microsoft.Common.targets.
   <Target Name="BeforeBuild">
@@ -72,4 +72,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
