--- a/cmake/developer_package/packaging/archive.cmake.orig	2024-01-08 04:42:05.000000000 -0800
+++ b/cmake/developer_package/packaging/archive.cmake	2024-06-07 04:54:27.900176984 -0700
@@ -21,17 +21,31 @@ endif()
 macro(ov_archive_cpack_set_dirs)
     # common "archive" package locations
     # TODO: move current variables to OpenVINO specific locations
-    set(OV_CPACK_INCLUDEDIR runtime/include)
-    set(OV_CPACK_IE_CMAKEDIR runtime/cmake)
-    set(OV_CPACK_NGRAPH_CMAKEDIR runtime/cmake)
-    set(OV_CPACK_OPENVINO_CMAKEDIR runtime/cmake)
-    set(OV_CPACK_DOCDIR docs)
-    set(OV_CPACK_LICENSESDIR licenses)
-    set(OV_CPACK_SAMPLESDIR samples)
-    set(OV_CPACK_WHEELSDIR tools)
-    set(OV_CPACK_TOOLSDIR tools)
-    set(OV_CPACK_DEVREQDIR tools)
-    set(OV_CPACK_PYTHONDIR python)
+    if(ENABLE_PYTHON)
+        set(OV_CPACK_INCLUDEDIR openvino/thirdparty/include)
+        set(OV_CPACK_IE_CMAKEDIR openvino/thirdparty/$ENV{LIBDIR}/cmake)
+        set(OV_CPACK_NGRAPH_CMAKEDIR openvino/thirdparty/$ENV{LIBDIR}/cmake)
+        set(OV_CPACK_OPENVINO_CMAKEDIR openvino/thirdparty/$ENV{LIBDIR}/cmake)
+        set(OV_CPACK_DOCDIR openvino/thirdparty/share/openvino/docs)
+        set(OV_CPACK_LICENSESDIR openvino/thirdparty/share/openvino/licenses)
+        set(OV_CPACK_SAMPLESDIR openvino/thirdparty/share/openvino/samples)
+        set(OV_CPACK_WHEELSDIR openvino/thirdparty/$ENV{LIBDIR}/openvino/tools)
+        set(OV_CPACK_TOOLSDIR openvino/thirdparty/$ENV{LIBDIR}/openvino/tools)
+        set(OV_CPACK_DEVREQDIR openvino/thirdparty/$ENV{LIBDIR}/openvino/tools)
+        set(OV_CPACK_PYTHONDIR .)
+    else()
+        set(OV_CPACK_INCLUDEDIR include)
+        set(OV_CPACK_IE_CMAKEDIR $ENV{LIBDIR}/cmake)
+        set(OV_CPACK_NGRAPH_CMAKEDIR $ENV{LIBDIR}/cmake)
+        set(OV_CPACK_OPENVINO_CMAKEDIR $ENV{LIBDIR}/cmake)
+        set(OV_CPACK_DOCDIR share/openvino/docs)
+        set(OV_CPACK_LICENSESDIR share/openvino/licenses)
+        set(OV_CPACK_SAMPLESDIR share/openvino/samples)
+        set(OV_CPACK_WHEELSDIR $ENV{LIBDIR}/openvino/tools)
+        set(OV_CPACK_TOOLSDIR $ENV{LIBDIR}/openvino/tools)
+        set(OV_CPACK_DEVREQDIR $ENV{LIBDIR}/openvino/tools)
+        set(OV_CPACK_PYTHONDIR .)
+    endif()
 
     if(USE_BUILD_TYPE_SUBFOLDER)
         set(build_type ${CMAKE_BUILD_TYPE})
@@ -48,9 +62,15 @@ macro(ov_archive_cpack_set_dirs)
         set(OV_CPACK_RUNTIMEDIR runtime/lib/${ARCH_FOLDER}/${build_type})
         set(OV_CPACK_ARCHIVEDIR runtime/lib/${ARCH_FOLDER}/${build_type})
     else()
-        set(OV_CPACK_LIBRARYDIR runtime/lib/${ARCH_FOLDER})
-        set(OV_CPACK_RUNTIMEDIR runtime/lib/${ARCH_FOLDER})
-        set(OV_CPACK_ARCHIVEDIR runtime/lib/${ARCH_FOLDER})
+        if(ENABLE_PYTHON)
+            set(OV_CPACK_LIBRARYDIR openvino/thirdparty/$ENV{LIBDIR}/${ARCH_FOLDER})
+            set(OV_CPACK_RUNTIMEDIR openvino/thirdparty/$ENV{LIBDIR}/${ARCH_FOLDER})
+            set(OV_CPACK_ARCHIVEDIR openvino/thirdparty/$ENV{LIBDIR}/${ARCH_FOLDER})
+        else()
+            set(OV_CPACK_LIBRARYDIR $ENV{LIBDIR}/${ARCH_FOLDER})
+            set(OV_CPACK_RUNTIMEDIR $ENV{LIBDIR}/${ARCH_FOLDER})
+            set(OV_CPACK_ARCHIVEDIR $ENV{LIBDIR}/${ARCH_FOLDER})
+        endif()
     endif()
     set(OV_CPACK_PLUGINSDIR ${OV_CPACK_RUNTIMEDIR})
 endmacro()
--- a/scripts/CMakeLists.txt.orig	2024-01-08 04:42:05.000000000 -0800
+++ b/scripts/CMakeLists.txt	2024-06-07 04:54:24.412206763 -0700
@@ -43,18 +43,32 @@ elseif(WIN32)
     endif()
 endif()
 
-install(PROGRAMS "${_setupvars_file_full_path}"
-        DESTINATION .
+if(ENABLE_PYTHON)
+    install(PROGRAMS "${_setupvars_file_full_path}"
+        DESTINATION openvino/thirdparty/$ENV{LIBDIR}/openvino/bin/
         COMPONENT ${OV_CPACK_COMP_SETUPVARS}
         ${OV_CPACK_COMP_SETUPVARS_EXCLUDE_ALL})
+else()
+    install(PROGRAMS "${_setupvars_file_full_path}"
+        DESTINATION $ENV{LIBDIR}/openvino/bin/
+        COMPONENT ${OV_CPACK_COMP_SETUPVARS}
+        ${OV_CPACK_COMP_SETUPVARS_EXCLUDE_ALL})
+endif()
 
 # install install_dependencies
 
 if(LINUX)
     ov_cpack_add_component(${OV_CPACK_COMP_INSTALL_DEPENDENCIES} HIDDEN)
 
-    install(PROGRAMS "${CMAKE_CURRENT_SOURCE_DIR}/install_dependencies/install_openvino_dependencies.sh"
-            DESTINATION install_dependencies/
+    if(ENABLE_PYTHON)
+        install(PROGRAMS "${CMAKE_CURRENT_SOURCE_DIR}/install_dependencies/install_openvino_dependencies.sh"
+            DESTINATION openvino/thirdparty/$ENV{LIBDIR}/openvino/bin/
             COMPONENT ${OV_CPACK_COMP_INSTALL_DEPENDENCIES}
             ${OV_CPACK_COMP_INSTALL_DEPENDENCIES_EXCLUDE_ALL})
+    else()
+        install(PROGRAMS "${CMAKE_CURRENT_SOURCE_DIR}/install_dependencies/install_openvino_dependencies.sh"
+            DESTINATION $ENV{LIBDIR}/openvino/bin/
+            COMPONENT ${OV_CPACK_COMP_INSTALL_DEPENDENCIES}
+            ${OV_CPACK_COMP_INSTALL_DEPENDENCIES_EXCLUDE_ALL})
+    endif()
 endif()
--- a/src/bindings/python/src/pyopenvino/CMakeLists.txt.orig	2024-01-08 04:42:05.000000000 -0800
+++ b/src/bindings/python/src/pyopenvino/CMakeLists.txt	2024-06-07 05:02:48.231899818 -0700
@@ -124,7 +124,7 @@ ov_set_install_rpath(${PROJECT_NAME} ${O
 ov_cpack_add_component(${OV_CPACK_COMP_OPENVINO_REQ_FILES} HIDDEN)
 
 install(FILES ${OpenVINOPython_SOURCE_DIR}/requirements.txt
-        DESTINATION ${OV_CPACK_PYTHONDIR}
+        DESTINATION ${OV_CPACK_PYTHONDIR}/openvino
         COMPONENT ${OV_CPACK_COMP_OPENVINO_REQ_FILES}
         ${OV_CPACK_COMP_OPENVINO_REQ_FILES_EXCLUDE_ALL})
 
