--- webkitgtk-2.9.2/Source/cmake/OptionsGTK.cmake.orig	2015-06-14 10:14:46.602693952 -0700
+++ webkitgtk-2.9.2/Source/cmake/OptionsGTK.cmake	2015-06-14 10:16:47.465340129 -0700
@@ -256,6 +256,12 @@
     find_package(LLVM)
     if (NOT LLVM_FOUND)
         message(FATAL_ERROR "LLVM is needed for ENABLE_FTL_JIT")
+    endif ()
+    if (ENABLE_FTL_NATIVE_CALL_INLINING)
+        find_package(Clang)
+        if (NOT CLANG_FOUND)
+            message(FATAL_ERROR "Clang is needed for ENABLE_FTL_NATIVE_CALL_INLINING")
+        endif ()
     endif ()
     SET_AND_EXPOSE_TO_BUILD(HAVE_LLVM TRUE)
 endif ()
