--- webkitgtk-2.4.9/Source/JavaScriptCore/GNUmakefile.list.am.orig	2015-11-27 10:49:48.717696993 -0800
+++ webkitgtk-2.4.9/Source/JavaScriptCore/GNUmakefile.list.am	2015-11-27 12:16:55.238108391 -0800
@@ -436,6 +436,25 @@
 	Source/JavaScriptCore/disassembler/LLVMDisassembler.cpp \
 	Source/JavaScriptCore/disassembler/LLVMDisassembler.h \
 	Source/JavaScriptCore/disassembler/X86Disassembler.cpp \
+	Source/JavaScriptCore/llvm/InitializeLLVM.cpp \
+	Source/JavaScriptCore/llvm/InitializeLLVM.h \
+	Source/JavaScriptCore/llvm/LLVMAPI.cpp \
+	Source/JavaScriptCore/llvm/LLVMAPI.h \
+	Source/JavaScriptCore/llvm/InitializeLLVMLinux.cpp \
+	Source/JavaScriptCore/llvm/InitializeLLVMPOSIX.cpp \
+	Source/JavaScriptCore/llvm/InitializeLLVMPOSIX.h \
+	Source/JavaScriptCore/ftl/FTLSlowPathCallKey.cpp \
+	Source/JavaScriptCore/ftl/FTLSlowPathCallKey.h \
+	Source/JavaScriptCore/ftl/FTLInlineCacheSize.cpp \
+	Source/JavaScriptCore/ftl/FTLInlineCacheSize.h \
+	Source/JavaScriptCore/ftl/FTLSaveRestore.cpp \
+	Source/JavaScriptCore/ftl/FTLSaveRestore.h \
+	Source/JavaScriptCore/ftl/FTLStackMaps.cpp \
+	Source/JavaScriptCore/ftl/FTLStackMaps.h \
+	Source/JavaScriptCore/ftl/FTLSlowPathCall.cpp \
+	Source/JavaScriptCore/ftl/FTLSlowPathCall.h \
+	Source/JavaScriptCore/ftl/FTLLocation.cpp \
+	Source/JavaScriptCore/ftl/FTLLocation.h \
 	Source/JavaScriptCore/ftl/FTLAbbreviatedTypes.h \
 	Source/JavaScriptCore/ftl/FTLAbbreviations.h \
 	Source/JavaScriptCore/ftl/FTLAbstractHeap.cpp \
