diff -urp rustc-1.19.0-src.orig/src/bootstrap/lib.rs rustc-1.19.0-src/src/bootstrap/lib.rs
--- rustc-1.19.0-src.orig/src/bootstrap/lib.rs	2017-07-17 15:10:50.000000000 -0700
+++ rustc-1.19.0-src/src/bootstrap/lib.rs	2017-12-29 14:37:15.937792932 -0800
@@ -105,7 +105,7 @@ mod doc;
 mod flags;
 mod install;
 mod native;
-mod sanity;
+//mod sanity;
 mod step;
 pub mod util;
 
@@ -293,7 +293,7 @@ impl Build {
         self.verbose("finding compilers");
         cc::find(self);
         self.verbose("running sanity check");
-        sanity::check(self);
+        //sanity::check(self);
         // If local-rust is the same major.minor as the current version, then force a local-rebuild
         let local_version_verbose = output(
             Command::new(&self.rustc).arg("--version").arg("--verbose"));
