diff -urp firefox-52.2.0esr.orig/media/libopus/include/opus_defines.h firefox-52.2.0esr/media/libopus/include/opus_defines.h
--- firefox-52.2.0esr.orig/media/libopus/include/opus_defines.h	2017-04-10 19:13:12.000000000 -0700
+++ firefox-52.2.0esr/media/libopus/include/opus_defines.h	2017-08-08 14:18:18.179312667 -0700
@@ -60,6 +60,8 @@ extern "C" {
 #define OPUS_ALLOC_FAIL       -7
 /**@}*/
 
+#define FLOAT_APPROX 1
+
 /** @cond OPUS_INTERNAL_DOC */
 /**Export control for opus functions */
 
