--- firefox-57.0.orig/toolkit/moz.configure	2017-12-30 07:53:13.264576263 -0800
+++ firefox-57.0/toolkit/moz.configure	2017-12-30 08:22:59.204547674 -0800
@@ -826,7 +826,7 @@ with only_when(building_stylo_bindgen):
             clang_path=clang_resolved,
         )
 
-    set_config('MOZ_LIBCLANG_PATH', bindgen_config_paths.libclang_path)
+    #set_config('MOZ_LIBCLANG_PATH', bindgen_config_paths.libclang_path)
     set_config('MOZ_CLANG_PATH', bindgen_config_paths.clang_path)
     set_config('MOZ_STYLO_BINDGEN', depends_if('--enable-stylo-build-bindgen')(lambda _: True))
 
