--- a/third_party/rust/cubeb-coreaudio/install_rustfmt_clippy.sh.orig	2020-07-20 13:56:19.000000000 -0700
+++ b/third_party/rust/cubeb-coreaudio/install_rustfmt_clippy.sh	2020-08-20 20:49:10.588839724 -0700
@@ -1,7 +1,7 @@
 # https://github.com/rust-lang/rustup-components-history/blob/master/README.md
 # https://github.com/rust-lang/rust-clippy/blob/27acea0a1baac6cf3ac6debfdbce04f91e15d772/.travis.yml#L40-L46
 if ! rustup component add rustfmt; then
-    TARGET=$(rustc -Vv | awk '/host/{print $2}')
+    TARGET=${CROSSCOMPILE}
     NIGHTLY=$(curl -s "https://rust-lang.github.io/rustup-components-history/${TARGET}/rustfmt")
     curl -sSL "https://static.rust-lang.org/dist/${NIGHTLY}/rustfmt-nightly-${TARGET}.tar.xz" | \
     tar -xJf - --strip-components=3 -C ~/.cargo/bin
@@ -9,7 +9,7 @@ if ! rustup component add rustfmt; then
 fi
 
 if ! rustup component add clippy; then
-    TARGET=$(rustc -Vv | awk '/host/{print $2}')
+    TARGET=${CROSSCOMPILE}
     NIGHTLY=$(curl -s "https://rust-lang.github.io/rustup-components-history/${TARGET}/clippy")
     rustup default nightly-${NIGHTLY}
     rustup component add rustfmt
