--- a/inference-engine/ie_bridges/python/wheel/CMakeLists.txt.orig	2021-11-16 09:55:28.000000000 -0800
+++ b/inference-engine/ie_bridges/python/wheel/CMakeLists.txt	2024-06-08 11:27:21.648795833 -0700
@@ -54,19 +54,37 @@ if(LINUX)
     endif()
 endif()
 
-add_custom_command(TARGET ie_wheel
-    PRE_BUILD
-        COMMAND ${CMAKE_COMMAND} -E rm -rf "${CMAKE_CURRENT_BINARY_DIR}/site-packages"
-        COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} clean bdist_wheel
-            --dist-dir ${CMAKE_BINARY_DIR}/wheels
-            --build=${WHEEL_BUILD}
-            --plat-name=${WHEEL_PLATFORM}
-    POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E rm "${CMAKE_CURRENT_SOURCE_DIR}/.env"
-    WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
-    COMMENT "Building Python wheel ${WHEEL_PACKAGE_NAME}"
-    VERBATIM
-)
+option(PIP22 "Using pip 22 or later?" OFF)
+if(PIP22)
+    add_custom_command(TARGET ie_wheel
+        PRE_BUILD
+            COMMAND ${CMAKE_COMMAND} -E rm -rf "${CMAKE_CURRENT_BINARY_DIR}/site-packages"
+            COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/site-packages/python/${PYTHON_VERSION}"
+            COMMAND ${PYTHON_EXECUTABLE} -m pip wheel
+                --no-deps
+                --wheel-dir ${CMAKE_BINARY_DIR}/wheels
+                --build-option --build-number=${WHEEL_BUILD}
+                --build-option --plat-name=${WHEEL_PLATFORM}
+                "${CMAKE_CURRENT_SOURCE_DIR}"
+        WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+        COMMENT "Building Python wheel ${WHEEL_PACKAGE_NAME}"
+        VERBATIM
+    )
+else()
+    add_custom_command(TARGET ie_wheel
+        PRE_BUILD
+            COMMAND ${CMAKE_COMMAND} -E rm -rf "${CMAKE_CURRENT_BINARY_DIR}/site-packages"
+            COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/site-packages/python/${PYTHON_VERSION}"
+            COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} bdist_wheel
+                --dist-dir ${CMAKE_BINARY_DIR}/wheels
+                --build=${WHEEL_BUILD}
+                --plat-name=${WHEEL_PLATFORM}
+            COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} clean
+        WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+        COMMENT "Building Python wheel ${WHEEL_PACKAGE_NAME}"
+        VERBATIM
+    )
+endif()
 
 set_property(TARGET ie_wheel
     APPEND
