diff -urp rocm_smi_lib-rocm-5.6.0.orig/CMakeLists.txt rocm_smi_lib-rocm-5.6.0/CMakeLists.txt
--- rocm_smi_lib-rocm-5.6.0.orig/CMakeLists.txt	2023-07-17 00:18:26.632983886 -0700
+++ rocm_smi_lib-rocm-5.6.0/CMakeLists.txt	2023-07-17 00:20:17.976981006 -0700
@@ -244,8 +244,12 @@ install(EXPORT rocm_smiTargets
 set(CPACK_RPM_PACKAGE_LICENSE "NCSA")
 # install license file in share/doc/rocm_smi-asan folder
 install(FILES ${CPACK_RESOURCE_FILE_LICENSE}
+        DESTINATION
+        share/doc/rocm_smi-asan
         COMPONENT asan)
 install( FILES ${CPACK_RESOURCE_FILE_LICENSE}
+        DESTINATION
+        share/doc/rocm_smi-asan
         COMPONENT dev)
 
 ###########################
