diff -urp rocm_smi_lib-rocm-5.1.3.orig/CMakeLists.txt rocm_smi_lib-rocm-5.1.3/CMakeLists.txt
--- rocm_smi_lib-rocm-5.1.3.orig/CMakeLists.txt	2023-08-26 10:19:58.023759306 -0700
+++ rocm_smi_lib-rocm-5.1.3/CMakeLists.txt	2023-08-26 10:20:15.191759732 -0700
@@ -46,7 +46,7 @@ set(PKG_VERSION_NUM_COMMIT 0)
 
 ## Define default variable and variables for the optional build target
 ##  rocm_smi_lib-dev
-set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE STRING "Default installation directory.")
+#_cmake_modify_IGNORE set(CMAKE_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE STRING "Default installation directory.")
 set(COMMON_SRC_ROOT ${CMAKE_CURRENT_SOURCE_DIR} CACHE STRING "Location source code common root.")
 set(ROCM_SMI_PACKAGE rocm-smi-lib)
 
@@ -55,7 +55,7 @@ project(${AMD_SMI_LIBS_TARGET})
 set(COMMON_PROJ_ROOT ${PROJECT_SOURCE_DIR})
 
 ## Verbose output.
-set(CMAKE_VERBOSE_MAKEFILE on)
+#_cmake_modify_IGNORE set(CMAKE_VERBOSE_MAKEFILE on)
 
 if (CMAKE_COMPILER_IS_GNUCC AND
                          CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.4.0)
@@ -162,7 +162,7 @@ configure_file(rocm_smi-config.cmake.in
     "${PROJECT_BINARY_DIR}/rocm_smi-config.cmake" @ONLY)
 # ... for the install tree
 set(CONF_INCLUDE_DIRS "${CMAKE_INSTALL_PREFIX}/rocm_smi/include")
-set(CONF_LIB_DIR  "${CMAKE_INSTALL_PREFIX}/rocm_smi/lib")
+set(CONF_LIB_DIR  "${CMAKE_INSTALL_PREFIX}/rocm_smi/@LIBDIR@")
 set(CONF_LIBS "${ROCM_SMI_LIB_NAME}.so")
 configure_file(rocm_smi-config.cmake.in
   "${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/rocm_smi-config.cmake" @ONLY)
@@ -173,10 +173,10 @@ configure_file(rocm_smi-config-version.c
 install(FILES
   "${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/rocm_smi-config.cmake"
   "${PROJECT_BINARY_DIR}/rocm_smi-config-version.cmake"
-  DESTINATION "${ROCM_SMI}/lib/cmake" COMPONENT dev)
+  DESTINATION "@LIBDIR@/cmake" COMPONENT dev)
 # Install the export set for use with the install-tree
 install(EXPORT rocm_smiTargets DESTINATION
-  "${ROCM_SMI}/lib/cmake" COMPONENT dev)
+  "@LIBDIR@/cmake" COMPONENT dev)
 
 #License file
 set(CPACK_RPM_PACKAGE_LICENSE "NCSA")
@@ -185,7 +185,7 @@ set(CPACK_RPM_PACKAGE_LICENSE "NCSA")
 # Packaging directives
 ###########################
 set(CPACK_GENERATOR "DEB;RPM"  CACHE STRING "Default packaging generators.")
-set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE STRING "Default packaging prefix.")
+set(CPACK_PACKAGING_INSTALL_PREFIX "@EPREFIX_ROCM_PATH@" CACHE STRING "Default packaging prefix.")
 set ( ENABLE_LDCONFIG ON CACHE BOOL "Set library links and caches using ldconfig.")
 set(CPACK_PACKAGE_NAME ${ROCM_SMI_PACKAGE})
 set(CPACK_PACKAGE_VENDOR "Advanced Micro Devices, Inc.")
diff -urp rocm_smi_lib-rocm-5.1.3.orig/oam/CMakeLists.txt rocm_smi_lib-rocm-5.1.3/oam/CMakeLists.txt
--- rocm_smi_lib-rocm-5.1.3.orig/oam/CMakeLists.txt	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/oam/CMakeLists.txt	2023-08-26 10:20:15.191759732 -0700
@@ -7,7 +7,7 @@ message("                    CMake OAM (
 message("&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&")
 
 ## Verbose output.
-set(CMAKE_VERBOSE_MAKEFILE on)
+#_cmake_modify_IGNORE set(CMAKE_VERBOSE_MAKEFILE on)
 
 # Required Defines first:
 
@@ -18,7 +18,7 @@ message("Build Configuration:")
 # message("-------------Version: " ${CMAKE_CXX_COMPILER_VERSION})
 message("--------Proj Src Dir: " ${PROJECT_SOURCE_DIR})
 # message("--------Proj Bld Dir: " ${PROJECT_BINARY_DIR})
-# message("--------Proj Lib Dir: " ${PROJECT_BINARY_DIR}/lib)
+# message("--------Proj Lib Dir: " ${PROJECT_BINARY_DIR}/@LIBDIR@)
 # message("--------Proj Exe Dir: " ${PROJECT_BINARY_DIR}/bin)
 # message("--------RSMI Lib Dir: " ${RSMI_LIB_DIR})
 # message("--------RSMI Inc Dir: " ${OAM_INC_DIR})
@@ -101,12 +101,12 @@ target_include_directories(${OAM_TARGET}
 ## Add the install directives for the runtime library.
 install(TARGETS ${OAM_TARGET}
           EXPORT rocm_smiTargets
-          LIBRARY DESTINATION ${OAM_NAME}/lib
-          ARCHIVE DESTINATION ${OAM_NAME}/lib
+          LIBRARY DESTINATION @LIBDIR@
+          ARCHIVE DESTINATION @LIBDIR@
           COMPONENT ${OAM_COMPONENT})
 install(FILES ${COMMON_SRC_ROOT}/oam/include/oam/oam_mapi.h
               ${COMMON_SRC_ROOT}/oam/include/oam/amd_oam.h
-                                        DESTINATION oam/include/oam)
+                                        DESTINATION include/oam)
 
 # Generate Doxygen documentation
 if (DOXYGEN_FOUND)
diff -urp rocm_smi_lib-rocm-5.1.3.orig/python_smi_tools/README.md rocm_smi_lib-rocm-5.1.3/python_smi_tools/README.md
--- rocm_smi_lib-rocm-5.1.3.orig/python_smi_tools/README.md	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/python_smi_tools/README.md	2023-08-26 10:20:15.191759732 -0700
@@ -30,7 +30,7 @@ $(uname -a) | cut -d ' ' -f 3)
 
 For detailed and up to date usage information, we recommend consulting the help:
 ```shell
-/opt/rocm/bin/rocm-smi -h
+@EPREFIX_ROCM_PATH@/bin/rocm-smi -h
 ```
 
 For convenience purposes, following is the output from the -h flag:
diff -urp rocm_smi_lib-rocm-5.1.3.orig/python_smi_tools/rocm_smi.py rocm_smi_lib-rocm-5.1.3/python_smi_tools/rocm_smi.py
--- rocm_smi_lib-rocm-5.1.3.orig/python_smi_tools/rocm_smi.py	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/python_smi_tools/rocm_smi.py	2023-08-26 10:20:15.191759732 -0700
@@ -66,7 +66,7 @@ def driverInitialized():
     """
     driverInitialized = ''
     try:
-        driverInitialized = str(subprocess.check_output("cat /proc/modules|grep amdgpu", shell=True))
+        driverInitialized = str(subprocess.check_output("test -d /sys/module/amdgpu", shell=True))
     except subprocess.CalledProcessError:
         pass
     if len(driverInitialized) > 0:
Only in rocm_smi_lib-rocm-5.1.3/python_smi_tools: rocm-smi.rej
diff -urp rocm_smi_lib-rocm-5.1.3.orig/python_smi_tools/rsmiBindings.py rocm_smi_lib-rocm-5.1.3/python_smi_tools/rsmiBindings.py
--- rocm_smi_lib-rocm-5.1.3.orig/python_smi_tools/rsmiBindings.py	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/python_smi_tools/rsmiBindings.py	2023-08-26 10:20:15.191759732 -0700
@@ -10,16 +10,7 @@ from enum import Enum
 import os
 
 # Use ROCm installation path if running from standard installation
-path_librocm = os.path.dirname(os.path.realpath(__file__)) + '/../lib/librocm_smi64.so'
-if not os.path.isfile(path_librocm):
-    print('Unable to find %s . Trying /opt/rocm*' % path_librocm)
-    for root, dirs, files in os.walk('/opt', followlinks=True):
-        if 'librocm_smi64.so' in files:
-            path_librocm = os.path.join(os.path.realpath(root), 'librocm_smi64.so')
-    if os.path.isfile(path_librocm):
-        print('Using lib from %s' % path_librocm)
-    else:
-        print('Unable to find librocm_smi64.so')
+path_librocm = '@EPREFIX_ROCM_PATH@/@LIBDIR@/librocm_smi64.so'
 
 # ----------> TODO: Support static libs as well as SO
 
diff -urp rocm_smi_lib-rocm-5.1.3.orig/README.md rocm_smi_lib-rocm-5.1.3/README.md
--- rocm_smi_lib-rocm-5.1.3.orig/README.md	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/README.md	2023-08-26 10:20:15.191759732 -0700
@@ -29,7 +29,7 @@ After the ROCm SMI library git repositor
 ##### ```$ cd build```
 ##### ```$ cmake <location of root of ROCm SMI library CMakeLists.txt>```
 ##### ```$ make```
-##### ```# Install library file and header; default location is /opt/rocm```
+##### ```# Install library file and header; default location is @EPREFIX_ROCM_PATH@```
 ##### ```$ make install```
 The built library will appear in the `build` folder.
 
@@ -43,7 +43,7 @@ The reference manual, `refman.pdf` will
 In order to verify the build and capability of ROCm SMI on your system and to see an example of how ROCm SMI can be used, you may build and run the tests that are available in the repo. To build the tests, follow these steps:
 
 ##### ```# Set environment variables used in CMakeLists.txt file```
-##### ```$ ROCM_DIR=<parent dir. to lib/ and inc/, containing RSMI library and header>```
+##### ```$ ROCM_DIR=<parent dir. to @LIBDIR@/ and inc/, containing RSMI library and header>```
 ##### ```$ mkdir <location for test build>```
 ##### ```$ cd <location for test build>```
 ##### ```$ cmake -DROCM_DIR=$ROCM_DIR <ROCm SMI source root>/tests/rocm_smi_test```
diff -urp rocm_smi_lib-rocm-5.1.3.orig/rocm_smi/CMakeLists.txt rocm_smi_lib-rocm-5.1.3/rocm_smi/CMakeLists.txt
--- rocm_smi_lib-rocm-5.1.3.orig/rocm_smi/CMakeLists.txt	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/rocm_smi/CMakeLists.txt	2023-08-26 10:20:15.191759732 -0700
@@ -7,7 +7,7 @@ message("                    CMake ROCm
 message("&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&")
 
 ## Verbose output.
-set(CMAKE_VERBOSE_MAKEFILE on)
+#_cmake_modify_IGNORE set(CMAKE_VERBOSE_MAKEFILE on)
 
 # Required Defines first:
 
@@ -18,7 +18,7 @@ message("Build Configuration:")
 # message("-------------Version: " ${CMAKE_CXX_COMPILER_VERSION})
 message("--------Proj Src Dir: " ${PROJECT_SOURCE_DIR})
 # message("--------Proj Bld Dir: " ${PROJECT_BINARY_DIR})
-# message("--------Proj Lib Dir: " ${PROJECT_BINARY_DIR}/lib)
+# message("--------Proj Lib Dir: " ${PROJECT_BINARY_DIR}/@LIBDIR@)
 # message("--------Proj Exe Dir: " ${PROJECT_BINARY_DIR}/bin)
 # message("--------RSMI Lib Dir: " ${RSMI_LIB_DIR})
 # message("--------RSMI Inc Dir: " ${RSMI_INC_DIR})
@@ -103,35 +103,30 @@ if ("${CMAKE_BUILD_TYPE}" STREQUAL Relea
   endif()
 endif ()
 
-## Add symlinks from top level ROCm lib dir to rocm-smi lib so files
+## Add symlinks from top level ROCm @LIBDIR@ dir to rocm-smi @LIBDIR@ so files
 if(${BUILD_SHARED_LIBS})
   add_custom_target ( so-link ALL WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
                     COMMAND ${CMAKE_COMMAND} -E create_symlink
-                          ../${ROCM_SMI}/lib/${ROCM_SMI_LIB_NAME}.so so-link )
+                          ../${ROCM_SMI}/@LIBDIR@/${ROCM_SMI_LIB_NAME}.so so-link ) # TODO: recheck
   add_custom_target ( so-major-link ALL WORKING_DIRECTORY
                   ${CMAKE_CURRENT_BINARY_DIR} COMMAND ${CMAKE_COMMAND}
                   -E create_symlink
-                  ../${ROCM_SMI}/lib/${ROCM_SMI_LIB_NAME}.so.${VERSION_MAJOR}
-                                                               so-major-link )
-
-  install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/so-link DESTINATION lib RENAME
-                                                     ${ROCM_SMI_LIB_NAME}.so )
-  install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/so-major-link DESTINATION lib
-                             RENAME ${ROCM_SMI_LIB_NAME}.so.${VERSION_MAJOR} )
+                  ../${ROCM_SMI}/@LIBDIR@/${ROCM_SMI_LIB_NAME}.so.${VERSION_MAJOR} # TODO: recheck
+                                                                so-major-link )
 endif()
 ## Add the install directives for the runtime library.
 install(TARGETS ${ROCM_SMI_TARGET}
           EXPORT rocm_smiTargets
-          LIBRARY DESTINATION ${ROCM_SMI}/lib
-          ARCHIVE DESTINATION ${ROCM_SMI}/lib
+          LIBRARY DESTINATION @LIBDIR@
+          ARCHIVE DESTINATION @LIBDIR@
           COMPONENT ${ROCM_SMI_COMPONENT})
 
 install(FILES ${COMMON_SRC_ROOT}/include/rocm_smi/rocm_smi.h
-                                        DESTINATION rocm_smi/include/rocm_smi)
+                                        DESTINATION include/rocm_smi)
 install(FILES ${COMMON_SRC_ROOT}/include/rocm_smi/${ROCM_SMI_TARGET}Config.h
-                                        DESTINATION rocm_smi/include/rocm_smi)
+                                        DESTINATION include/rocm_smi)
 install(FILES ${COMMON_SRC_ROOT}/include/rocm_smi/kfd_ioctl.h
-                                        DESTINATION rocm_smi/include/rocm_smi)
+                                        DESTINATION include/rocm_smi)
 install(FILES ${COMMON_SRC_ROOT}/python_smi_tools/rsmiBindings.py
                                         DESTINATION rocm_smi/bindings)
 install(FILES ${COMMON_SRC_ROOT}/python_smi_tools/rocm_smi.py
@@ -140,14 +135,6 @@ install(FILES ${COMMON_SRC_ROOT}/python_
                                         DESTINATION bin)
 install(FILES ${COMMON_SRC_ROOT}/python_smi_tools/rocm-smi
                                         DESTINATION bin)
-add_custom_target(bindings_link ALL
-                    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-                    COMMAND ${CMAKE_COMMAND} -E create_symlink
-                       ../rocm_smi/bindings/rsmiBindings.py bindings_link)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/bindings_link
-                     DESTINATION bin RENAME rsmiBindings.py)
-
-
 # Generate Doxygen documentation
 find_package(Doxygen)
 find_package(LATEX COMPONENTS PDFLATEX)
diff -urp rocm_smi_lib-rocm-5.1.3.orig/RPM/post.in rocm_smi_lib-rocm-5.1.3/RPM/post.in
--- rocm_smi_lib-rocm-5.1.3.orig/RPM/post.in	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/RPM/post.in	2023-08-26 10:20:15.191759732 -0700
@@ -1,5 +1,5 @@
 # left-hand term originates from ENABLE_LDCONFIG = ON/OFF at package build
 if [ "@ENABLE_LDCONFIG@" == "ON" ]; then
-  echo -e "@CPACK_PACKAGING_INSTALL_PREFIX@/rocm_smi/lib\n@CPACK_PACKAGING_INSTALL_PREFIX@/rocm_smi/lib64" > /etc/ld.so.conf.d/x86_64-librocm_smi_lib.conf
+  echo -e "@CPACK_PACKAGING_INSTALL_PREFIX@/rocm_smi/@LIBDIR@\n@CPACK_PACKAGING_INSTALL_PREFIX@/rocm_smi/@LIBDIR@" > /etc/ld.so.conf.d/x86_64-librocm_smi_lib.conf
   ldconfig
 fi
diff -urp rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/CMakeLists.txt rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/CMakeLists.txt
--- rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/CMakeLists.txt	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/CMakeLists.txt	2023-08-26 10:20:15.191759732 -0700
@@ -48,7 +48,7 @@ endif()
 # Required Defines first:
 
 set(RSMI_INC_DIR ${ROCM_DIR}/include)
-set(RSMI_LIB_DIR ${ROCM_DIR}/lib)
+set(RSMI_LIB_DIR ${ROCM_DIR}/@LIBDIR@)
 #
 # Determine RSMI Header files are present
 #  (no external source dependencies)
@@ -101,7 +101,7 @@ message("------------Compiler: " ${CMAKE
 message("-------------Version: " ${CMAKE_CXX_COMPILER_VERSION})
 message("--------Proj Src Dir: " ${PROJECT_SOURCE_DIR})
 message("--------Proj Bld Dir: " ${PROJECT_BINARY_DIR})
-message("--------Proj Lib Dir: " ${PROJECT_BINARY_DIR}/lib)
+message("--------Proj Lib Dir: " ${PROJECT_BINARY_DIR}/@LIBDIR@)
 message("--------Proj Exe Dir: " ${PROJECT_BINARY_DIR}/bin)
 message("--------RSMI Lib Dir: " ${RSMI_LIB_DIR})
 message("")
@@ -109,11 +109,11 @@ message("")
 #
 # Set the build type based on user input
 #
-set(CMAKE_BUILD_TYPE ${BUILD_TYPE})
+#_cmake_modify_IGNORE set(CMAKE_BUILD_TYPE ${BUILD_TYPE})
 #
 # Flag to enable / disable verbose output.
 #
-set(CMAKE_VERBOSE_MAKEFILE on)
+#_cmake_modify_IGNORE set(CMAKE_VERBOSE_MAKEFILE on)
 #
 # Compiler pre-processor definitions.
 #
@@ -220,7 +220,7 @@ add_executable(${RSMITST} ${rsmitstSourc
 target_link_libraries(${RSMITST} ${RSMITST_LIBS} c stdc++ pthread)
 
 install(TARGETS ${RSMITST}
-        ARCHIVE DESTINATION ${PROJECT_BINARY_DIR}/lib
-        LIBRARY DESTINATION ${PROJECT_BINARY_DIR}/lib
+        ARCHIVE DESTINATION ${PROJECT_BINARY_DIR}/@LIBDIR@
+        LIBRARY DESTINATION ${PROJECT_BINARY_DIR}/@LIBDIR@
         RUNTIME DESTINATION ${PROJECT_BINARY_DIR}/bin)
 
diff -urp rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/docs/Pkgconfig.md rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/docs/Pkgconfig.md
--- rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/docs/Pkgconfig.md	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/docs/Pkgconfig.md	2023-08-26 10:20:15.191759732 -0700
@@ -136,11 +136,11 @@ CMake Error at /usr/share/cmake/Modules/
 These failures are common if you installed GoogleTest yourself and have not
 sourced it from a distro or other package manager. If so, you need to tell
 pkg-config where it can find the `.pc` files containing the information.
-Say you installed GoogleTest to `/usr/local`, then it might be that the
-`.pc` files are installed under `/usr/local/lib64/pkgconfig`. If you set
+Say you installed GoogleTest to `@EPREFIX@/usr`, then it might be that the
+`.pc` files are installed under `@EPREFIX@/usr/@LIBDIR@/pkgconfig`. If you set
 
 ```
-export PKG_CONFIG_PATH=/usr/local/lib64/pkgconfig
+export PKG_CONFIG_PATH=@EPREFIX@/usr/@LIBDIR@/pkgconfig
 ```
 
 pkg-config will also try to look in `PKG_CONFIG_PATH` to find `gtest_main.pc`.
diff -urp rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googlemock/configure.ac rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googlemock/configure.ac
--- rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googlemock/configure.ac	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googlemock/configure.ac	2023-08-26 10:20:15.191759732 -0700
@@ -136,7 +136,7 @@ AS_IF([test "x${HAVE_BUILT_GTEST}" = "xy
       GTEST_CPPFLAGS='-I$(top_srcdir)/../googletest/include'
       GTEST_CXXFLAGS='-g'
       GTEST_LDFLAGS=''
-      GTEST_LIBS='$(top_builddir)/../googletest/lib/libgtest.la'
+      GTEST_LIBS='$(top_builddir)/../googletest/@LIBDIR@/libgtest.la'
       GTEST_VERSION="${GTEST_MIN_VERSION}"])
 
 # TODO(chandlerc@google.com) Check the types, structures, and other compiler
diff -urp rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googlemock/Makefile.am rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googlemock/Makefile.am
--- rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googlemock/Makefile.am	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googlemock/Makefile.am	2023-08-26 10:20:15.191759732 -0700
@@ -21,7 +21,7 @@ if HAVE_PTHREADS
 endif
 
 # Build rules for libraries.
-lib_LTLIBRARIES = lib/libgmock.la lib/libgmock_main.la
+lib_LTLIBRARIES = @LIBDIR@/libgmock.la @LIBDIR@/libgmock_main.la
 
 lib_libgmock_la_SOURCES = src/gmock-all.cc
 
@@ -48,7 +48,7 @@ pkginclude_internal_HEADERS = \
   include/gmock/internal/custom/gmock-port.h
 
 lib_libgmock_main_la_SOURCES = src/gmock_main.cc
-lib_libgmock_main_la_LIBADD = lib/libgmock.la
+lib_libgmock_main_la_LIBADD = @LIBDIR@/libgmock.la
 
 # Build rules for tests. Automake's naming for some of these variables isn't
 # terribly obvious, so this is a brief reference:
@@ -65,7 +65,7 @@ AM_LDFLAGS = $(GTEST_LDFLAGS)
 TESTS += test/gmock-spec-builders_test
 check_PROGRAMS += test/gmock-spec-builders_test
 test_gmock_spec_builders_test_SOURCES = test/gmock-spec-builders_test.cc
-test_gmock_spec_builders_test_LDADD = $(GTEST_LIBS) lib/libgmock.la
+test_gmock_spec_builders_test_LDADD = $(GTEST_LIBS) @LIBDIR@/libgmock.la
 
 # This tests using Google Mock in multiple translation units.  It also
 # verifies that libgmock_main and libgmock work.
@@ -75,7 +75,7 @@ test_gmock_link_test_SOURCES = \
   test/gmock_link2_test.cc \
   test/gmock_link_test.cc \
   test/gmock_link_test.h
-test_gmock_link_test_LDADD = $(GTEST_LIBS) lib/libgmock_main.la  lib/libgmock.la
+test_gmock_link_test_LDADD = $(GTEST_LIBS) @LIBDIR@/libgmock_main.la  @LIBDIR@/libgmock.la
 
 if HAVE_PYTHON
   # Tests that fused gmock files compile and work.
diff -urp rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googlemock/scripts/gmock-config.in rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googlemock/scripts/gmock-config.in
--- rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googlemock/scripts/gmock-config.in	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googlemock/scripts/gmock-config.in	2023-08-26 10:20:15.195759732 -0700
@@ -38,7 +38,7 @@ Compiler flag queries output the union o
 
   # When using an installed Google Mock, but with installation overrides:
   export GMOCK_PREFIX="/opt"
-  g++ $(gmock-config --libdir="/opt/lib64" ...) ...
+  g++ $(gmock-config --libdir="@EPREFIX@/usr/@LIBDIR@" ...) ...
 
  Help:
   --usage                    brief usage information
@@ -216,7 +216,7 @@ if test "${this_bindir}" = "${this_bindi
   # TODO(chandlerc@google.com): This is a dangerous dependency on libtool, we
   # should work to remove it, and/or remove libtool altogether, replacing it
   # with direct references to the library and a link path.
-  gmock_libs="${build_dir}/lib/libgmock.la"
+  gmock_libs="${build_dir}/@LIBDIR@/libgmock.la"
   gmock_ldflags=""
 
   # We provide hooks to include from either the source or build dir, where the
diff -urp rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googletest/Makefile.am rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googletest/Makefile.am
--- rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googletest/Makefile.am	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googletest/Makefile.am	2023-08-26 10:20:15.195759732 -0700
@@ -181,7 +181,7 @@ else
 endif
 
 # Build rules for libraries.
-lib_LTLIBRARIES = lib/libgtest.la lib/libgtest_main.la
+lib_LTLIBRARIES = @LIBDIR@/libgtest.la @LIBDIR@/libgtest_main.la
 
 lib_libgtest_la_SOURCES = src/gtest-all.cc
 
@@ -215,7 +215,7 @@ pkginclude_internal_HEADERS = \
   include/gtest/internal/custom/gtest-printers.h
 
 lib_libgtest_main_la_SOURCES = src/gtest_main.cc
-lib_libgtest_main_la_LIBADD = lib/libgtest.la
+lib_libgtest_main_la_LIBADD = @LIBDIR@/libgtest.la
 
 # Build rules for samples and tests. Automake's naming for some of
 # these variables isn't terribly obvious, so this is a brief
@@ -252,43 +252,43 @@ check_PROGRAMS += samples/sample1_unitte
     samples/sample10_unittest
 
 samples_sample1_unittest_SOURCES = samples/sample1_unittest.cc samples/sample1.cc
-samples_sample1_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample1_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 samples_sample2_unittest_SOURCES = samples/sample2_unittest.cc samples/sample2.cc
-samples_sample2_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample2_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 samples_sample3_unittest_SOURCES = samples/sample3_unittest.cc
-samples_sample3_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample3_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 samples_sample4_unittest_SOURCES = samples/sample4_unittest.cc samples/sample4.cc
-samples_sample4_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample4_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 samples_sample5_unittest_SOURCES = samples/sample5_unittest.cc samples/sample1.cc
-samples_sample5_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample5_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 samples_sample6_unittest_SOURCES = samples/sample6_unittest.cc
-samples_sample6_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample6_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 samples_sample7_unittest_SOURCES = samples/sample7_unittest.cc
-samples_sample7_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample7_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 samples_sample8_unittest_SOURCES = samples/sample8_unittest.cc
-samples_sample8_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la
+samples_sample8_unittest_LDADD = @LIBDIR@/libgtest_main.la \
+                                 @LIBDIR@/libgtest.la
 
 # Also verify that libgtest works by itself.
 samples_sample9_unittest_SOURCES = samples/sample9_unittest.cc
-samples_sample9_unittest_LDADD = lib/libgtest.la
+samples_sample9_unittest_LDADD = @LIBDIR@/libgtest.la
 samples_sample10_unittest_SOURCES = samples/sample10_unittest.cc
-samples_sample10_unittest_LDADD = lib/libgtest.la
+samples_sample10_unittest_LDADD = @LIBDIR@/libgtest.la
 
 # This tests most constructs of gtest and verifies that libgtest_main
 # and libgtest work.
 TESTS += test/gtest_all_test
 check_PROGRAMS += test/gtest_all_test
 test_gtest_all_test_SOURCES = test/gtest_all_test.cc
-test_gtest_all_test_LDADD = lib/libgtest_main.la \
-                            lib/libgtest.la
+test_gtest_all_test_LDADD = @LIBDIR@/libgtest_main.la \
+                            @LIBDIR@/libgtest.la
 
 # Tests that fused gtest files compile and work.
 FUSED_GTEST_SRC = \
diff -urp rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googletest/scripts/gtest-config.in rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googletest/scripts/gtest-config.in
--- rocm_smi_lib-rocm-5.1.3.orig/tests/rocm_smi_test/gtest/googletest/scripts/gtest-config.in	2022-03-21 07:31:15.000000000 -0700
+++ rocm_smi_lib-rocm-5.1.3/tests/rocm_smi_test/gtest/googletest/scripts/gtest-config.in	2023-08-26 10:20:15.195759732 -0700
@@ -38,7 +38,7 @@ Compiler flag queries output the union o
 
   # When using an installed Google Test, but with installation overrides:
   export GTEST_PREFIX="/opt"
-  g++ $(gtest-config --libdir="/opt/lib64" ...) ...
+  g++ $(gtest-config --libdir="@EPREFIX@/usr/@LIBDIR@" ...) ...
 
  Help:
   --usage                    brief usage information
@@ -214,7 +214,7 @@ if test "${this_bindir}" = "${this_bindi
   # TODO(chandlerc@google.com): This is a dangerous dependency on libtool, we
   # should work to remove it, and/or remove libtool altogether, replacing it
   # with direct references to the library and a link path.
-  gtest_libs="${build_dir}/lib/libgtest.la @PTHREAD_CFLAGS@ @PTHREAD_LIBS@"
+  gtest_libs="${build_dir}/@LIBDIR@/libgtest.la @PTHREAD_CFLAGS@ @PTHREAD_LIBS@"
   gtest_ldflags=""
 
   # We provide hooks to include from either the source or build dir, where the
