--- a/src/zig_llvm.h.orig	2025-12-30 16:48:24.182118263 -0800
+++ b/src/zig_llvm.h	2025-12-30 16:48:49.764065508 -0800
@@ -307,7 +307,12 @@ enum ZigLLVM_EnvironmentType {
     ZigLLVM_Amplification,
     ZigLLVM_OpenHOS,
 
-    ZigLLVM_LastEnvironmentType = ZigLLVM_OpenHOS
+    // Distro backported values
+    ZigLLVM_GNUT64,
+    ZigLLVM_GNUEABIT64,
+    ZigLLVM_GNUEABIHFT64,
+
+    ZigLLVM_LastEnvironmentType = ZigLLVM_GNUEABIHFT64
 };
 
 enum ZigLLVM_ObjectFormatType {
