diff -urp geany-1.38.orig/src/editor.c geany-1.38/src/editor.c
--- geany-1.38.orig/src/editor.c	2021-10-09 08:02:14.000000000 -0700
+++ geany-1.38/src/editor.c	2023-06-30 22:17:23.218999232 -0700
@@ -2228,6 +2228,7 @@ gboolean editor_start_auto_complete(Gean
 	const gchar *wordchars;
 	GeanyFiletype *ft;
 
+	return FALSE;
 	g_return_val_if_fail(editor != NULL, FALSE);
 
 	if (! editor_prefs.auto_complete_symbols && ! force)
