diff -urp Nancy-1.4.3.orig/src/Nancy.Testing/Nancy.Testing.csproj Nancy-1.4.3/src/Nancy.Testing/Nancy.Testing.csproj
--- Nancy-1.4.3.orig/src/Nancy.Testing/Nancy.Testing.csproj	2015-12-21 04:00:06.000000000 -0800
+++ Nancy-1.4.3/src/Nancy.Testing/Nancy.Testing.csproj	2017-01-30 05:25:11.327810610 -0800
@@ -85,10 +85,7 @@
     <DocumentationFile>bin\MonoRelease\Nancy.Testing.XML</DocumentationFile>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="CsQuery, Version=1.3.3.5, Culture=neutral, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\CsQuery.1.3.3\lib\net40\CsQuery.dll</HintPath>
-    </Reference>
+    <Reference Include="CsQuery" />
     <Reference Include="System">
     </Reference>
     <Reference Include="System.Core">
@@ -188,4 +185,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
diff -urp Nancy-1.4.3.orig/src/Nancy.Testing.Tests/Nancy.Testing.Tests.csproj Nancy-1.4.3/src/Nancy.Testing.Tests/Nancy.Testing.Tests.csproj
--- Nancy-1.4.3.orig/src/Nancy.Testing.Tests/Nancy.Testing.Tests.csproj	2015-12-21 04:00:06.000000000 -0800
+++ Nancy-1.4.3/src/Nancy.Testing.Tests/Nancy.Testing.Tests.csproj	2017-01-30 05:25:12.463853926 -0800
@@ -80,10 +80,7 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="CsQuery, Version=1.3.3.5, Culture=neutral, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\CsQuery.1.3.3\lib\net40\CsQuery.dll</HintPath>
-    </Reference>
+    <Reference Include="CsQuery" />
     <Reference Include="FakeItEasy">
       <HintPath>..\packages\FakeItEasy.1.19.0\lib\net40\FakeItEasy.dll</HintPath>
     </Reference>
@@ -192,4 +189,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
