--- /var/tmp/portage/net-libs/webkit-gtk-2.8.5/work/webkitgtk-2.8.5/Source/WebCore/platform/network/gtk/CredentialBackingStore.cpp.orig	2015-10-23 22:33:38.334953752 -0700
+++ webkitgtk-2.8.5/Source/WebCore/platform/network/gtk/CredentialBackingStore.cpp	2015-10-23 23:57:05.282997899 -0700
@@ -91,7 +91,7 @@
         return;
     }
 
-    size_t length;
+    gsize length;
     GRefPtr<SecretValue> secretValue = adoptGRef(secret_item_get_secret(secretItem.get()));
     const char* passwordData = secret_value_get(secretValue.get(), &length);
     String password = String::fromUTF8(passwordData, length);
