diff -urp static-tools-runtime-fuse-9999.orig/src/runtime/Makefile static-tools-runtime-fuse-9999/src/runtime/Makefile
--- static-tools-runtime-fuse-9999.orig/src/runtime/Makefile	2023-12-22 11:55:38.656637248 -0800
+++ static-tools-runtime-fuse-9999/src/runtime/Makefile	2023-12-22 11:56:08.812638052 -0800
@@ -1,6 +1,6 @@
 CC            = gcc
 CFLAGS        = -std=gnu99 -s -Os -D_FILE_OFFSET_BITS=64 -DGIT_COMMIT=\"${GIT_COMMIT}\" -T data_sections.ld -ffunction-sections -fdata-sections -Wl,--gc-sections -static
-LIBS          = -lsquashfuse -lsquashfuse_ll -lzstd -lz
+LIBS          = -lsquashfuse -lsquashfuse_ll -lzstd -lz -llzma
 
 all: runtime-fuse2 runtime-fuse3
 
