diff -urp cppsharp-9999.20160115.orig/src/CppParser/Parser.cpp cppsharp-9999.20160115/src/CppParser/Parser.cpp
--- cppsharp-9999.20160115.orig/src/CppParser/Parser.cpp	2017-01-05 13:32:04.106570024 -0800
+++ cppsharp-9999.20160115/src/CppParser/Parser.cpp	2017-01-05 13:34:45.477721942 -0800
@@ -47,7 +47,7 @@
 #include <CodeGen/TargetInfo.h>
 #include <CodeGen/CGCall.h>
 #include <CodeGen/CGCXXABI.h>
-#include <Driver/ToolChains.h>
+#include <clang/lib/Driver/ToolChains.h>
 
 #if defined(__APPLE__) || defined(__linux__)
 #ifndef _GNU_SOURCE
@@ -3533,4 +3533,4 @@ Declaration* Parser::GetDeclarationFromF
         Decl->LineNumberEnd = MinLineNumberEnd;
     }
     return Decl;
-}
\ No newline at end of file
+}
