diff -urp cppsharp-9999.20160115.orig/src/CppParser/AST.cpp cppsharp-9999.20160115/src/CppParser/AST.cpp
--- cppsharp-9999.20160115.orig/src/CppParser/AST.cpp	2017-01-05 13:28:09.952636440 -0800
+++ cppsharp-9999.20160115/src/CppParser/AST.cpp	2017-01-05 13:30:07.287114114 -0800
@@ -10,6 +10,7 @@
 #include <string>
 #include <vector>
 #include <llvm/Support/Path.h>
+#include <llvm/ADT/SmallString.h>
 
 // copy from widenPath ('llvm/lib/Support/Windows/Path.inc')
 static std::string normalizePath(const std::string & File) {
@@ -944,4 +945,4 @@ VerbatimBlockLineComment::VerbatimBlockL
 
 DEF_STRING(VerbatimBlockLineComment, Text)
 
-} } }
\ No newline at end of file
+} } }
